--- setup.py.old	2008-11-25 18:03:16.000000000 +0100
+++ setup.py	2008-11-25 18:04:14.000000000 +0100
@@ -341,12 +341,12 @@
               include_dirs=["Bio"]
               ),
 #Commented out due to the build dependency on flex, see Bug 2619
-#   Extension('Bio.PDB.mmCIF.MMCIFlex',
-#              ['Bio/PDB/mmCIF/lex.yy.c',
-#               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
-#              include_dirs=["Bio"],
-#              libraries=["fl"]
-#              ),
+    Extension('Bio.PDB.mmCIF.MMCIFlex',
+              ['Bio/PDB/mmCIF/lex.yy.c',
+               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
+              include_dirs=["Bio"],
+              libraries=["fl"]
+              ),
     Extension('Bio.Nexus.cnexus',
               ['Bio/Nexus/cnexus.c']
               ),
