diff --git a/Bio/SeqIO/SffIO.py b/Bio/SeqIO/SffIO.py
index 2bb0dac..735d55b 100644
--- a/Bio/SeqIO/SffIO.py
+++ b/Bio/SeqIO/SffIO.py
@@ -941,7 +941,8 @@ def _check_eof(handle, index_offset, index_length):
                       BiopythonParserWarning)
 
     offset = handle.tell()
-    assert offset % 8 == 0
+    assert offset % 8 == 0, \
+        "Wanted offset %i %% 8 = %i to be zero" % (offset, offset % 8)
     # Should now be at the end of the file...
     extra = handle.read(4)
     if extra == _sff:
