diff -Nu -Nu -uarN arp_warp_7.0.1.orig/share/arpwarp_setup_base.bash arp_warp_7.0.1/share/arpwarp_setup_base.bash
--- arp_warp_7.0.1.orig/share/arpwarp_setup_base.bash	2007-10-21 21:46:13.000000000 +0200
+++ arp_warp_7.0.1/share/arpwarp_setup_base.bash	2008-04-06 13:33:05.000000000 +0200
@@ -13,10 +13,10 @@
     identifier=${nameprocessor}'-'${namesystem}
 fi
 #
-arpwarphome="$1X"
+arpwarphome="/opt/arp-warp-bin"
 #
 export warpbin=${arpwarphome}/bin/bin-${identifier}
-export warpdoc=${arpwarphome}/manual
+export warpdoc=/usr/share/doc/arp-warp-bin-7.0.1/html/
 export PATH=${warpbin}:${PATH}
 #
 testccp='0'
@@ -49,7 +49,7 @@
     fi
 
 ##    2. Complement by other potential places for python :
-    search_list="${search_list} /usr/local/bin/python /usr/bin/python"
+    search_list="${search_list} /usr/bin/python"
     flex_wARP_python="/dev/null"
 
 ##    3. Pick up the first one of the list which seems reasonable :
@@ -73,30 +73,6 @@
             export flex_wARP_bin=${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
             break
         fi
-#        3.c Otherwise, we should be able to create the byte-code using the python.
-        /bin/mkdir -p ${arpwarphome}/byte-code/python-${pythonvrs1}.${pythonvrs2} 1> /dev/null 2>&1
-        if [ 0 != $? ] ; then
-            /bin/mkdir -p ${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2} 1> /dev/null 2>&1
-            export flex_wARP_bin=${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
-        else
-            export flex_wARP_bin=${arpwarphome}/byte-code/python-${pythonvrs1}.${pythonvrs2}
-        fi
-        pushd ${flex_wARP_bin} 1> /dev/null 2>&1
-        if [ 0 != $? ] ; then 
-            export flex_wARP_bin=${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
-            /bin/mkdir -p ${flex_wARP_bin} 1> /dev/null 2>&1
-            pushd ${flex_wARP_bin} 1> /dev/null 2>&1
-        fi
-        /bin/cp -p ${arpwarphome}/flex-wARP-src/*.py .
-        ${python_to_test} ./compile.py
-        /bin/rm -f *.py
-        popd 1> /dev/null 2>&1
-        if [ -e ${flex_wARP_bin}/CPyWARP.pyc ] ; then
-            export flex_wARP_python=${python_to_test}
-            echo "Created a new set of flex-wARP byte code in the directory :"
-            echo "  '${flex_wARP_bin}'"
-            break
-        fi
     done
 ##    4. If nothing is available, print out a message.
     if [ "/dev/null" != ${flex_wARP_python} ] ; then
@@ -107,6 +83,7 @@
         echo "WARNING while setting up ARP/wARP"
         echo "No valid 'python' command was found"
         echo "flex-warp ('ARP/wARP Expert System' in the GUI) will not be available."
+	echo "Did you upgrade your python? Please run 'emerge sci-chemistry/arp-warp-bin!'"
         echo
         export flex_wARP_warpbin="Unavailable"
         export flex_wARP_cbin="Unavailable"
@@ -115,17 +92,3 @@
     fi
 fi
 
-#
-# Setting up library path for SGI
-if [[ "${namesystem}" = 'IRIX64' || "${namesystem}" = 'IRIX' ]]; then
-    if test "$LD_LIBRARY_PATH"; then
-        export LD_LIBRARY_PATH=${warpbin}:${LD_LIBRARY_PATH}
-    else
-        export LD_LIBRARY_PATH=${warpbin}
-    fi
-fi
-#   echo
-#   echo 'ARP/wARP Version 7.0.1 has been setup'
-#
-
-
diff -Nu -Nu -uarN arp_warp_7.0.1.orig/share/arpwarp_setup_base.csh arp_warp_7.0.1/share/arpwarp_setup_base.csh
--- arp_warp_7.0.1.orig/share/arpwarp_setup_base.csh	2007-10-21 21:46:13.000000000 +0200
+++ arp_warp_7.0.1/share/arpwarp_setup_base.csh	2008-04-06 13:32:48.000000000 +0200
@@ -13,10 +13,10 @@
     set identifier = ${nameprocessor}'-'${namesystem}
 endif
 #
-set arpwarphome = '$1X'
+set arpwarphome = '/opt/arp-warp-bin'
 #
 setenv warpbin ${arpwarphome}/bin/bin-${identifier}
-setenv warpdoc ${arpwarphome}/manual
+setenv warpdoc /usr/share/doc/arp-warp-bin-7.0.1/html/
 set path = ( $warpbin $path )
 #
 # Settings for python
@@ -47,7 +47,7 @@
 endif
 
 ##    2. Complement by other potential places for python :
-set search_list = "${search_list} /usr/local/bin/python /usr/bin/python"
+set search_list = "${search_list} /usr/bin/python"
 setenv flex_wARP_python "/dev/null"
 
 ##    3. Pick up the first one of the list which seems reasonable :
@@ -71,30 +71,6 @@
         setenv flex_wARP_bin ${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
         break
     endif
-#        3.c Otherwise, we should be able to create the byte-code using the python.
-    /bin/mkdir -p ${arpwarphome}/byte-code/python-${pythonvrs1}.${pythonvrs2} >& /dev/null
-    if ( 0 != $status ) then
-        /bin/mkdir -p ${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2} >& /dev/null
-        setenv flex_wARP_bin ${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
-    else
-        setenv flex_wARP_bin ${arpwarphome}/byte-code/python-${pythonvrs1}.${pythonvrs2}
-    endif
-    pushd ${flex_wARP_bin} >& /dev/null
-    if ( 0 != $status ) then
-        setenv flex_wARP_bin ${HOME}/.flex-wARP/python-${pythonvrs1}.${pythonvrs2}
-        /bin/mkdir -p ${flex_wARP_bin} >& /dev/null
-        pushd ${flex_wARP_bin} >& /dev/null
-    endif
-    /bin/cp -p ${arpwarphome}/flex-wARP-src/*.py .
-    ${python_to_test} ./compile.py
-    /bin/rm -f *.py
-    popd >& /dev/null
-    if ( -e ${flex_wARP_bin}/CPyWARP.pyc ) then
-        setenv flex_wARP_python ${python_to_test}
-        echo "Created a new set of flex-wARP byte code in the directory :"
-        echo "  '${flex_wARP_bin}'"
-        break
-    endif
 end
 ##    4. If nothing is available, print out a message.
 if ( "/dev/null" != ${flex_wARP_python} ) then
@@ -105,22 +81,13 @@
     echo "WARNING while setting up ARP/wARP"
     echo "No valid 'python' command was found"
     echo "flex-warp ('ARP/wARP Expert System' in the GUI) will not be available."
+    echo "Did you upgrade your python? Please run 'emerge sci-chemistry/arp-warp-bin!'"
     echo
     setenv flex_wARP_warpbin "Unavailable"
     setenv flex_wARP_cbin "Unavailable"
     setenv flex_wARP_python "/dev/null"
     setenv flex_wARP_bin "Unavailable"
 endif
-
-#
-# Setting up library path for SGI
-if ( ${namesystem} == 'IRIX64' || ${namesystem} == 'IRIX' ) then
-    if ( ${?LD_LIBRARY_PATH} ) then
-        setenv LD_LIBRARY_PATH ${warpbin}:${LD_LIBRARY_PATH}
-    else
-        setenv LD_LIBRARY_PATH ${warpbin}
-    endif
-endif
 #
 goto endsetup
 #
@@ -132,7 +99,4 @@
 #
 endsetup:
 #
-#echo
-#echo 'ARP/wARP Version 7.0.1 has been setup'
-#
 
