--- src/makefiles/default_rules.mk.orig	2015-11-06 18:43:31.000000000 -0400
+++ src/makefiles/default_rules.mk	2015-11-09 18:42:33.609774157 -0400
@@ -14,7 +14,7 @@
         LIBFILE = lib$(LIBNAME).so
         #LDLIBS  += -l$(LIBNAME)
       endif
-      LDFLAGS += -Wl,-rpath=$(shell readlink -f $(KALDILIBDIR)) -L.
+      LDFLAGS += -L.
       LDFLAGS += $(foreach dep,$(ADDLIBS), -L$(dir $(dep)) )
       XDEPENDS = $(foreach dep,$(ADDLIBS), $(dir $(dep))/lib$(notdir $(basename $(dep))).so )
     else  # Platform not supported
@@ -75,7 +75,7 @@
 	@{ result=0;			\
 	for x in $(TESTFILES); do	\
 	  printf "Running $$x ...";	\
-	  ./$$x >$$x.testlog 2>&1;	\
+	  LD_LIBRARY_PATH=$(KALDILIBDIR):${LD_LIBRARY_PATH} ./$$x >$$x.testlog 2>&1;	\
 	  if [ $$? -ne 0 ]; then	\
 	    echo "... FAIL $$x";	\
 	    result=1;			\
@@ -99,3 +99,9 @@
 # removing automatic making of "depend" as it's quite slow.
 #.depend.mk: depend
 -include .depend.mk
+
+print-binfiles:
+	@echo $(addprefix $(shell pwd)/, $(BINFILES))
+
+print-libfiles:
+	@echo $(addsuffix .so, $(addprefix $(shell pwd)/lib, $(LIBNAME)))
