*** Makefile.orig	2015-02-28 16:31:57.377401743 -0400
--- Makefile	2015-02-28 16:48:58.153175376 -0400
***************
*** 30,36 ****
  
  # Reset the default goal, so that the all target will become default
  .DEFAULT_GOAL :=
! all: checkversion test_dependencies kaldi.mk mklibdir $(SUBDIRS)
  	-echo Done
  
  mklibdir:
--- 30,36 ----
  
  # Reset the default goal, so that the all target will become default
  .DEFAULT_GOAL :=
! all: checkversion kaldi.mk mklibdir $(SUBDIRS)
  	-echo Done
  
  mklibdir:
***************
*** 88,103 ****
  ext: test_dependencies ext_depend $(SUBDIRS) $(EXT_SUBDIRS)
  	-echo Done
  
- ifndef OPENFST_VER
- $(error Please rerun configure: OPENFST_VER is not defined, likely kaldi.mk was produced by older configure script.)
- endif
- # Note: OPENFST_VER is determined by configure and added to kaldi.mk
- OPENFST_VER_NUM := $(shell echo $(OPENFST_VER) | sed 's/\./ /g' | xargs printf "%d%02d%02d")
- test_dependencies:
- ifeq ("$(shell expr $(OPENFST_VER_NUM) \< 10302)","1")
- 	$(error OpenFst $(OPENFST_VER) is not supported. You now need OpenFst >= 1.3.2.)
- endif
- 
  check_portaudio:
  	@[ -d ../tools/portaudio ] || ( cd ../tools;  ./install_portaudio.sh )
  
--- 88,93 ----
***************
*** 131,136 ****
--- 121,136 ----
  ext_depend: check_portaudio
  	-for x in $(EXT_SUBDIRS); do $(MAKE) -C $$x depend; done
  
+ print-binfiles: $(addsuffix /print-binfiles, $(SUBDIRS))
+ 
+ %/print-binfiles:
+ 	@$(MAKE) --no-print-directory -C $(dir $@) print-binfiles
+ 
+ print-libfiles: $(addsuffix /print-libfiles, $(SUBDIRS))
+ 
+ %/print-libfiles:
+ 	@$(MAKE) --no-print-directory -C $(dir $@) print-libfiles
+ 
  
  .PHONY: $(SUBDIRS)
  $(SUBDIRS) : mklibdir
