--- src/Makefile.orig	2017-03-03 01:33:22.000000000 +0100
+++ src/Makefile	2017-03-03 18:06:59.912864196 +0100
@@ -32,7 +32,6 @@
 # Reset the default goal, so that the all target will become default
 .DEFAULT_GOAL :=
 all: 
-	$(MAKE) checkversion 
 	$(MAKE) kaldi.mk 
 	$(MAKE) mklibdir
 	$(MAKE) subdirs
@@ -52,18 +51,6 @@
 	@true
 endif
 
-.PHONY: checkversion
-checkversion:
-ifeq ($(shell ./configure --version),$(CONFIGURE_VERSION))
-	@echo "The version of configure script matches kaldi.mk version. Good."
-else
-	@echo ""
-	@echo "The kaldi.mk file was generated using a different version of configure script. Please rerun the configure again"
-	@test -f ./kaldi.mk && echo  "Hint: Previous configure command line: " && head -n 2 ./kaldi.mk | grep configure | sed 's/^# *//g'
-	@echo ""
-	@false
-endif
-
 biglib: $(SUBDIRS_LIB)
 ifeq ($(KALDI_FLAVOR), dynamic)
 ifeq ($(shell uname), Darwin)
@@ -136,6 +123,15 @@
 ext_depend: check_portaudio
 	-for x in $(EXT_SUBDIRS); do $(MAKE) -C $$x depend; done
 
+print-binfiles: $(addsuffix /print-binfiles, $(SUBDIRS))
+
+%/print-binfiles:
+	@$(MAKE) --no-print-directory -C $(dir $@) print-binfiles
+
+print-libfiles: $(addsuffix /print-libfiles, $(SUBDIRS))
+
+%/print-libfiles:
+	@$(MAKE) --no-print-directory -C $(dir $@) print-libfiles
 
 .PHONY: $(SUBDIRS)
 $(SUBDIRS) : mklibdir
