*** src/mira/readpool.C.ori	2010-11-21 11:42:29.000000000 -0600
--- src/mira/readpool.C	2011-01-13 10:20:38.000000000 -0600
***************
*** 1751,1757 ****
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_QUALITY_LEFT  : {
! 	  REP_thepool[idoffound].setLQClipoff(atoi(ECD->c_str()));
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_QUALITY_RIGHT  : {
--- 1751,1757 ----
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_QUALITY_LEFT  : {
! 	  REP_thepool[idoffound].setLQClipoff(atoi(ECD->c_str())-1);
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_QUALITY_RIGHT  : {
***************
*** 1759,1765 ****
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_VECTOR_LEFT  : {
! 	  REP_thepool[idoffound].setLSClipoff(atoi(ECD->c_str()));
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_VECTOR_RIGHT  : {
--- 1759,1765 ----
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_VECTOR_LEFT  : {
! 	  REP_thepool[idoffound].setLSClipoff(atoi(ECD->c_str())-1);
  	  break;
  	}
  	case NCBIInfoXML::NCBIXML_CLIP_VECTOR_RIGHT  : {
