--- ibutils-1.2.orig/ibdm/ibdm/Makefile.in	2009-05-27 17:26:57.000000000 +0200
+++ ibutils-1.2/ibdm/ibdm/Makefile.in	2009-10-14 17:18:24.950978547 +0200
@@ -293,6 +293,7 @@
 libibdm_la_LDFLAGS = -Wl,-rpath -Wl,$(TCL_PREFIX)/lib -version-info $(LIB_VER_TRIPLET) -no-undefined
 ibdmsh_SOURCES = ibdmsh_wrap.cpp
 ibdmsh_LDADD = -libdmcom $(TCL_LIBS)
+ibdmsh_DEPENDENCIES = libibdmcom.la
 ibdmsh_LDFLAGS = -static -Wl,-rpath -Wl,$(TCL_PREFIX)/lib
 
 # track latest GIT version for this tree:
--- ibutils-1.2.orig/ibdm/ibdm/Makefile.am	2009-05-27 17:26:45.000000000 +0200
+++ ibutils-1.2/ibdm/ibdm/Makefile.am	2009-10-14 17:18:44.763825873 +0200
@@ -87,6 +87,7 @@
 bin_PROGRAMS  = ibdmsh
 ibdmsh_SOURCES = ibdmsh_wrap.cpp
 ibdmsh_LDADD =  -libdmcom $(TCL_LIBS)
+ibdmsh_DEPENDENCIES = libibdmcom.la
 ibdmsh_LDFLAGS = -static -Wl,-rpath -Wl,$(TCL_PREFIX)/lib
 
 $(srcdir)/Fabric.cpp: $(srcdir)/git_version.h
