--- ibutils-1.2.orig/ibdm/ibdm/TopoMatch.cpp	2009-05-27 17:26:45.000000000 +0200
+++ ibutils-1.2/ibdm/ibdm/TopoMatch.cpp	2009-10-14 16:02:49.535651042 +0200
@@ -676,7 +676,7 @@
   IBSystem *p_system = p_node->p_system;
 
   // we always mark the board of the node by examining all but the "UXXX"
-  char *p_lastSlash = rindex(p_node->name.c_str(), '/');
+  const char *p_lastSlash = rindex(p_node->name.c_str(), '/');
   char nodeBoardName[512];
   int  boardNameLength;
   if (!p_lastSlash) {
--- ibutils-1.2.orig/ibdm/ibdm/SysDef.cpp	2009-05-27 17:26:45.000000000 +0200
+++ ibutils-1.2/ibdm/ibdm/SysDef.cpp	2009-10-14 16:03:12.178724234 +0200
@@ -79,7 +79,7 @@
 
       // the device number should be embedded in the master name of
       // the node: MT23108 ...
-      char *p_digit;
+      const char *p_digit;
       if ((p_digit = strpbrk(p_inst->master.c_str(), "0123456789")) != NULL)
         sscanf(p_digit,"%u", &p_node->devId);
 
