--- clipper-config.in	2009-02-10 18:42:47.000000000 +0100
+++ clipper-config.in.new	2009-05-20 14:30:52.000000000 +0200
@@ -79,13 +79,13 @@
 
 if test "$echo_cflags" = "yes" ; then
  # and include mccp4 cflags? yes
- cflags="-I${includedir} @CCP4_CXXFLAGS@ @MCCP4_CXXFLAGS@ @MMDB_CXXFLAGS@ @FFTW_CXXFLAGS@"
+ cflags="-I${includedir} @CCP4_CXXFLAGS@ @MMDB_CXXFLAGS@ @FFTW_CXXFLAGS@"
  echo $cflags
 fi
 
 if test "$echo_libs" = "yes" ; then
  # note BOOST_LIBS should be part of CCTBX_LIBS
- libs="-lclipper-minimol -lclipper-mmdb -lclipper-cns -lclipper-cif -lclipper-contrib -lclipper-ccp4 -lclipper-phs -lclipper-core @CCP4_LIBS@ @MMDB_LIBS@ @FFTW_LIBS@ @CCTBX_LIBS@"
+ libs="-lclipper-minimol -lclipper-mmdb -lclipper-cns -lclipper-cif -lclipper-contrib -lclipper-ccp4 -lclipper-phs -lclipper-core @CCP4_LIBS@ @MMDB_LIBS@ @FFTW_LIBS@"
  if test "${libdir}" != "/usr/lib" ; then
   echo -L${libdir} $libs
  else
