--- abyss-2.2.5/Common/ContigProperties.h.ori	2020-09-18 20:42:12.000000000 +0200
+++ abyss-2.2.5/Common/ContigProperties.h	2021-04-14 12:31:23.682158000 +0200
@@ -171,7 +171,7 @@
 template <typename T>
 ContigProperties& operator+=(ContigProperties& a, const T& b)
 {
-	assert((int)a.length + (int)b.distance > 0);
+	assert((int)a.length + (int)b.distance >= 0);
 	a.length += b.distance;
 	return a;
 }
