--- abyss-2.2.5/MergePaths/PathConsensus.cpp	2020-09-18 20:42:12.000000000 +0200
+++ abyss-2.2.5/MergePaths/PathConsensus.cpp	2021-04-12 22:26:28.780547000 +0200
@@ -379,7 +379,7 @@
 		unsigned overlap, Sequence& seq,
 		const Sequence& s, const ContigNode& node, const Path& path)
 {
-	assert(s.length() > overlap);
+	assert(s.length() >= overlap);
 	Sequence ao;
 	Sequence bo(s, 0, overlap);
 	Sequence o;
