diff -r -U2 sympy-0.6.2.orig/doc/Makefile sympy-0.6.2/doc/Makefile
--- sympy-0.6.2.orig/doc/Makefile	2008-08-17 22:00:24.000000000 +0700
+++ sympy-0.6.2/doc/Makefile	2008-08-29 00:09:11.000000000 +0700
@@ -1,11 +1,8 @@
 # Makefile for Sphinx documentation
 #
-RST2HTML     = rst2html
+RST2HTML     = rst2html.py
 
 # You can set these variables from the command line.
 SPHINXOPTS   =
-SPHINXVER    = 0.4.1
-SPHINXBUILDpy  = sphinx/Sphinx-$(SPHINXVER)/sphinx-build.py
-SPHINXBUILD  = PYTHONPATH=.. python $(SPHINXBUILDpy)
 PAPER        =
 
@@ -31,11 +28,5 @@
 	-rm -rf sphinx
 
-$(SPHINXBUILDpy):
-	rm -rf sphinx
-	mkdir sphinx
-	cd sphinx; wget http://pypi.python.org/packages/source/S/Sphinx/Sphinx-$(SPHINXVER).tar.gz;
-	cd sphinx; tar xzf Sphinx-$(SPHINXVER).tar.gz
-
-html: $(SPHINXBUILDpy) spt-printable
+html: spt-printable
 	mkdir -p src/.static
 	mkdir -p _build/html _build/doctrees
@@ -52,5 +43,5 @@
 	$(RST2HTML) src/sympy-patches-tutorial.txt > _build/html/spt-printable.html
 
-htmlapi: $(SPHINXBUILDpy)
+htmlapi:
 	mkdir -p api/.static
 	mkdir -p api/modules
@@ -77,5 +68,5 @@
 	      ".hhp project file in _build/htmlhelp."
 
-latex: $(SPHINXBUILDpy)
+latex:
 	mkdir -p _build/latex _build/doctrees
 	$(SPHINXBUILD) -b latex $(ALLSPHINXOPTS) _build/latex
