diff -ru likwid-3.1.1/config.mk likwid-3.1.1-nju/config.mk
--- likwid-3.1.1/config.mk	2014-02-12 17:14:16.000000000 +0100
+++ likwid-3.1.1-nju/config.mk	2014-02-19 00:22:02.530917183 +0100
@@ -10,12 +10,12 @@
 
 # Path were to install likwid
 PREFIX = /usr/local#NO SPACE
-MANPREFIX = $(PREFIX)/man#NO SPACE
+MANPREFIX = $(PREFIX)/share/man#NO SPACE
 
 # For the daemon based secure msr/pci access configure
 # the absolute path to the msr daemon executable.
 # $(PREFIX)/bin/likwid-accessD
-ACCESSDAEMON = $(PREFIX)/bin/likwid-accessD#NO SPACE
+ACCESSDAEMON = /usr/bin/likwid-accessD#NO SPACE
 
 # Build the accessDaemon. Have a look in the WIKI for details.
 BUILDDAEMON = true#NO SPACE
@@ -26,7 +26,7 @@
 ACCESSMODE = accessdaemon#NO SPACE
 
 # Change to true to a build shared library instead of a static one
-SHARED_LIBRARY = false#NO SPACE
+SHARED_LIBRARY = true#NO SPACE
 
 # Build Fortran90 module interface for marker API. Adopt Fortran compiler
 # in ./make/include_<COMPILER>.mk if necessary. Default: ifort .
