diff -ur likwid-3.0/config.mk likwid-3.0-nju/config.mk
--- likwid-3.0/config.mk	2012-11-29 18:16:13.000000000 +0100
+++ likwid-3.0-nju/config.mk	2013-01-30 15:32:47.896140004 +0100
@@ -10,12 +10,12 @@
 
 # Path were to install likwid
 PREFIX = /usr/local#NO SPACE
-MANPREFIX = $(PREFIX)/man#NO SPACE
+MANPREFIX = $(PREFIX)/share/man#NO SPACE
 
 # For the daemon based secure msr/pci access configure
 # the absolute path to the msr daemon executable.
 # $(PREFIX)/bin/likwid-accessD
-ACCESSDAEMON = $(PREFIX)/bin/likwid-accessD#NO SPACE
+ACCESSDAEMON = /usr/bin/likwid-accessD#NO SPACE
 # Path to the msrd-socket in system daemon mode.
 SYSDAEMONSOCKETPATH = /var/run/likwid-msrd.sock#NO SPACE
 
@@ -31,7 +31,7 @@
 ENABLE_SNB_UNCORE = false#NO SPACE
 
 # Change to YES to a build shared library instead of a static one
-SHARED_LIBRARY = false#NO SPACE
+SHARED_LIBRARY = true#NO SPACE
 
 # Instrument likwid-bench for use with likwid-perfctr
 INSTRUMENT_BENCH = false#NO SPACE
