diff -ur likwid-3.0.old/config.mk likwid-3.0/config.mk
--- likwid-3.0.old/config.mk	2013-02-19 16:06:21.867704619 +0100
+++ likwid-3.0/config.mk	2013-02-19 16:06:08.277704352 +0100
@@ -28,7 +28,7 @@
 ACCESSMODE = direct#NO SPACE
 
 # Set to true to enable SandyBridge Uncore support
-ENABLE_SNB_UNCORE = false#NO SPACE
+ENABLE_SNB_UNCORE = true#NO SPACE
 
 # Change to YES to a build shared library instead of a static one
 SHARED_LIBRARY = true#NO SPACE
