diff -ur likwid-3.0/config.mk likwid-3.0-new/config.mk
--- likwid-3.0/config.mk	2012-11-29 18:16:13.000000000 +0100
+++ likwid-3.0-new/config.mk	2013-01-29 09:05:08.406319092 +0100
@@ -20,12 +20,12 @@
 SYSDAEMONSOCKETPATH = /var/run/likwid-msrd.sock#NO SPACE
 
 # Build the accessDaemon. Have a look in the WIKI for details.
-BUILDDAEMON = false#NO SPACE
+BUILDDAEMON = true#NO SPACE
 
 # Set the default mode for MSR access.
 # This can usually be overriden on the commandline.
 # Valid values are: direct, accessdaemon, sysdaemon
-ACCESSMODE = direct#NO SPACE
+ACCESSMODE = accessdaemon#NO SPACE
 
 # Set to true to enable SandyBridge Uncore support
 ENABLE_SNB_UNCORE = false#NO SPACE
Nur in likwid-3.0-new: liblikwid.a.
Nur in likwid-3.0-new: liblikwidpin.so.
Nur in likwid-3.0-new: likwid-accessD.
Nur in likwid-3.0-new: likwid-features.
Nur in likwid-3.0-new: likwid-memsweeper.
Nur in likwid-3.0-new: likwid-perfctr.
Nur in likwid-3.0-new: likwid-pin.
Nur in likwid-3.0-new: likwid-powermeter.
Nur in likwid-3.0-new: likwid-topology.
