--- a/src/Makefile	2022-07-20 11:47:44.078639381 +0200
+++ b/src/Makefile	2022-07-22 19:24:00.389938450 +0200
@@ -9,10 +9,11 @@
 # options for various compilers
 #
 
-# Intels ifort, default in optimized mode
-F90 = ifort
-comp = -c -O -module ${Mdir} -I${InDir} 
-LFlagsB = -O  
+
+# gentoo toolchain
+F90 = ${F90}
+comp = -c -O -J${Mdir} -I${InDir}  ${FFLAGS} ${FCFLAGS} ${CFLAGS}
+LFlagsB = -O ${LDFLAGS}
 
 # Intels ifort, debug modus
 ifeq (${F90},ifortg)
