--- bmp/Makefile.in
+++ bmp/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- gif/Makefile.in
+++ gif/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- ico/Makefile.in
+++ ico/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- jpeg/Makefile.in
+++ jpeg/Makefile.in
@@ -322,7 +322,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- pcx/Makefile.in
+++ pcx/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- pixmap/Makefile.in
+++ pixmap/Makefile.in
@@ -324,7 +324,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- png/Makefile.in
+++ png/Makefile.in
@@ -322,7 +322,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- ppm/Makefile.in
+++ ppm/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- ps/Makefile.in
+++ ps/Makefile.in
@@ -320,7 +320,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- sgi/Makefile.in
+++ sgi/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- sun/Makefile.in
+++ sun/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- tga/Makefile.in
+++ tga/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- tiff/Makefile.in
+++ tiff/Makefile.in
@@ -322,7 +322,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- window/Makefile.in
+++ window/Makefile.in
@@ -323,7 +323,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- xbm/Makefile.in
+++ xbm/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
--- xpm/Makefile.in
+++ xpm/Makefile.in
@@ -318,7 +318,7 @@
 
 VPATH = $(srcdir)
 
-.c.$(OBJEXT):
+.c.$(OBJEXT): init.c
 	$(COMPILE) -c `@CYGPATH@ $<` -o $@
 
 init.c:	$(srcdir)/../init.c
