--- bmp/bmp.c
+++ bmp/bmp.c
@@ -446,9 +446,7 @@
 	    }
 	    if ((rleBuf[0]==0) && (rleBuf[1]==2)) {
 	        /* Deltarecord */
-		/* printf ("Deltarecord\n"); fflush (stdout); */
-		x += rleBuf[2];
-		y += rleBuf[3];
+		Tcl_AppendResult(interp, "Delta records not supported.", (char *)NULL);
 	    }
 	}
     }
--- ps/ps.c
+++ ps/ps.c
@@ -171,7 +171,7 @@
 		(tkimg_Read(handle, (char *) buf, 40) == 40))) {
 	    int w, h, zoomx, zoomy;
 	    char *p = buf;
-	    buf[41] = 0;
+	    buf[40] = 0;
 	    w = - (int) strtoul(p, &p, 0);
 	    h = - (int) strtoul(p, &p, 0);
 	    w += strtoul(p, &p, 0);
--- tiff/tiff.c
+++ tiff/tiff.c
@@ -561,7 +561,7 @@
     int srcX, srcY;
 {
     myblock bl;
-    unsigned char *pixelPtr = block.pixelPtr;
+    unsigned char *pixelPtr;
     uint32 w, h;
     size_t npixels;
     uint32 *raster;
