--- a/misc/CMake/BRLCAD_CMakeFiles.cmake	2015-04-27 12:18:13.159285916 -0400
+++ b/misc/CMake/BRLCAD_CMakeFiles.cmake	2015-04-27 12:19:40.722856052 -0400
@@ -82,6 +82,10 @@
 	endif("${ITEM}" MATCHES "${TARGET_FLAG}")
       endforeach(TARGET_FLAG ${FUZZY_TARGET_FLAGS})
 
+      if("${ITEM}" MATCHES "MODULE")
+            set(CMAKEFILES_DO_TEST 0)
+      endif("")
+
       # Handled target flags, proceeding with tests.
       if(CMAKEFILES_DO_TEST)
 	get_filename_component(ITEM_PATH "${ITEM}" PATH)
