--- brlcad-9999/CMakeLists.txt	2012-02-11 11:37:28.287828624 -0500
+++ brlcad-9999/CMakeLists.txt	2012-02-11 11:37:54.018158044 -0500
@@ -1586,7 +1586,7 @@
   endforeach()
   foreach(path_label ${PATH_LABELS})
     set(CURRENT_LABEL ${${path_label}_LABEL})
-    string(LENGTH ${CURRENT_LABEL} CURRENT_LENGTH)
+    string(LENGTH "${CURRENT_LABEL}" CURRENT_LENGTH)
     while(${PATH_LABEL_LENGTH} GREATER ${CURRENT_LENGTH})
       set(CURRENT_LABEL " ${CURRENT_LABEL}")
       string(LENGTH "${CURRENT_LABEL}" CURRENT_LENGTH)
