--- lib-src-reg/LPinterface.hh	2009-04-30 13:23:20.000000000 +0200
+++ lib-src-reg/LPinterface.hh.new	2009-06-17 15:22:07.000000000 +0200
@@ -77,7 +77,12 @@
   dd_FreeLPSolution(_solptr);
   dd_FreeLPData    (_lpptr);
   dd_FreeMatrix    (_matrixptr);
-  dd_clear_global_constants();
+//  dd_clear_global_constants();
+  dd_clear(dd_zero);                                                                       
+  dd_clear(dd_minuszero);                                                                  
+  dd_clear(dd_one);                                                                        
+  dd_clear(dd_minusone);                                                                   
+  dd_clear(dd_purezero);                                                                   
 }
 
 #endif
