diff -Naur tmp-old/src/56_xc/m_libxc_functionals.F90 tmp-new/src/56_xc/m_libxc_functionals.F90
--- src/56_xc/m_libxc_functionals.F90	2010-04-02 18:35:00.000000000 +0000
+++ src/56_xc/m_libxc_functionals.F90	2010-04-07 15:06:14.000000000 +0000
@@ -363,14 +363,14 @@
         if (funcs(i)%id == 0) cycle
 
         !Get the potential (and possibly the energy)
-        if (iand(xc_f90_info_provides(funcs(i)%info), XC_PROVIDES_EXC) .ne. 0) then
+        if (iand(xc_f90_info_flags(funcs(i)%info), XC_FLAGS_HAVE_EXC) .ne. 0) then
           select case (funcs(i)%family)
           case (XC_FAMILY_LDA)
             call xc_f90_lda_exc_vxc(funcs(i)%conf,1,rhotmp(1),exctmp,vxctmp(1))
           case (XC_FAMILY_GGA)
             call xc_f90_gga_exc_vxc(funcs(i)%conf,1,rhotmp(1),sigma(1),exctmp,vxctmp(1),vsigma(1))
           case (XC_FAMILY_MGGA)
-            call xc_f90_mgga_exc_vxc(funcs(i)%conf,rhotmp(1),sigma(1),lrhotmp(1),&
+            call xc_f90_mgga_exc_vxc(funcs(i)%conf,1,rhotmp(1),sigma(1),lrhotmp(1),&
                         tautmp(1),exctmp,vxctmp(1),vsigma(1),vxclrhotmp(1),vxctautmp(1))
 !            write(message, '(7a)' )ch10,&
 !                 &    ' libxc_functionals_init : ERROR -',ch10,&
@@ -388,7 +388,7 @@
           case (XC_FAMILY_GGA)
             call xc_f90_gga_vxc(funcs(i)%conf,1,rhotmp(1),sigma(1),vxctmp(1),vsigma(1))
           case (XC_FAMILY_MGGA)
-            call xc_f90_mgga_vxc(funcs(i)%conf,rhotmp(1),sigma(1),lrhotmp(1),&
+            call xc_f90_mgga_vxc(funcs(i)%conf,1,rhotmp(1),sigma(1),lrhotmp(1),&
                         tautmp(1),vxctmp(1),vsigma(1),vxclrhotmp(1),vxctautmp(1))
           end select
         end if
