diff -Naur tmp-old/src/53_ffts/ccfft.F90 tmp-new/src/53_ffts/ccfft.F90
--- src/53_ffts/ccfft.F90	2009-03-11 05:12:18.000000000 +0000
+++ src/53_ffts/ccfft.F90	2009-04-14 12:13:31.000000000 +0000
@@ -96,8 +96,8 @@
  integer,intent(out) :: inplace,normalized
  type(MPI_type),intent(inout) :: mpi_enreg
 !arrays
- real(dp),intent(inout) :: work1(2,n4*n5*n6*ndat)
- real(dp),intent(out) :: work2(2,n4*n5*n6*ndat)
+ real(dp),intent(inout) :: work1(2*n4,n5,n6,ndat)
+ real(dp),intent(out) :: work2(2*n4,n5,n6,ndat)
 
 !Local variables ------------------------------
 !scalars
