diff -Naur abinit-6.12.1_orig/Makefile.am abinit-6.12.1/Makefile.am
--- Makefile.am	2012-01-30 19:42:26.000000000 +0000
+++ Makefile.am	2012-03-22 10:32:50.000000000 +0000
@@ -1279,12 +1279,16 @@
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/12_hide_mpi/lib12_hide_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/13_memory_mpi/lib13_memory_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/14_hidewrite/lib14_hidewrite.a
+if DO_BUILD_15_GPU_TOOLBOX
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/15_gpu_toolbox/lib15_gpu_toolbox.a
+endif
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/16_hideleave/lib16_hideleave.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/18_timing/lib18_timing.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/27_toolbox_oop/lib27_toolbox_oop.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/28_numeric_noabirule/lib28_numeric_noabirule.a
+if DO_BUILD_32_CONTRACT
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/32_contract/lib32_contract.a
+endif
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/32_util/lib32_util.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/42_geometry/lib42_geometry.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/42_nlstrain/lib42_nlstrain.a
@@ -1298,7 +1302,9 @@
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/49_gw_toolbox_oop/lib49_gw_toolbox_oop.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/51_manage_mpi/lib51_manage_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/52_fft_mpi_noabirule/lib52_fft_mpi_noabirule.a
+if DO_BUILD_52_MANAGE_CUDA
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/52_manage_cuda/lib52_manage_cuda.a
+endif
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_abiutil/lib53_abiutil.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_ffts/lib53_ffts.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_spacepar/lib53_spacepar.a
diff -Naur abinit-6.12.1_orig/Makefile.in abinit-6.12.1/Makefile.in
--- Makefile.in	2012-03-22 09:23:51.000000000 +0000
+++ Makefile.in	2012-03-22 10:33:22.000000000 +0000
@@ -2638,12 +2638,12 @@
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/12_hide_mpi/lib12_hide_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/13_memory_mpi/lib13_memory_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/14_hidewrite/lib14_hidewrite.a
-	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/15_gpu_toolbox/lib15_gpu_toolbox.a
+@DO_BUILD_15_GPU_TOOLBOX_TRUE@	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/15_gpu_toolbox/lib15_gpu_toolbox.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/16_hideleave/lib16_hideleave.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/18_timing/lib18_timing.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/27_toolbox_oop/lib27_toolbox_oop.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/28_numeric_noabirule/lib28_numeric_noabirule.a
-	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/32_contract/lib32_contract.a
+@DO_BUILD_32_CONTRACT_TRUE@	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/32_contract/lib32_contract.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/32_util/lib32_util.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/42_geometry/lib42_geometry.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/42_nlstrain/lib42_nlstrain.a
@@ -2657,7 +2657,7 @@
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/49_gw_toolbox_oop/lib49_gw_toolbox_oop.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/51_manage_mpi/lib51_manage_mpi.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/52_fft_mpi_noabirule/lib52_fft_mpi_noabirule.a
-	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/52_manage_cuda/lib52_manage_cuda.a
+@DO_BUILD_52_MANAGE_CUDA_TRUE@	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/52_manage_cuda/lib52_manage_cuda.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_abiutil/lib53_abiutil.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_ffts/lib53_ffts.a
 	cd $(libabinit_tmpdir) && $(AR) xv $(abs_top_builddir)/src/53_spacepar/lib53_spacepar.a
