diff -Naur abinit-6.2.2_orig/src/66_paw/indgrid.F90 abinit-6.2.2/src/66_paw/indgrid.F90
--- src/66_paw/indgrid.F90	2010-08-07 04:31:29.000000000 +0000
+++ src/66_paw/indgrid.F90	2010-08-11 17:16:01.000000000 +0000
@@ -62,6 +62,7 @@
 !arrays
  integer :: id(3)
  integer,allocatable :: gc(:,:),gf(:,:)
+ character(len=500) :: msg
 
 ! *************************************************************************
 !
@@ -117,7 +118,10 @@
 !Check coatofin to make sure there are no zeros!
  do ii=1,ubound(coatofin,1)
    if (coatofin(ii)==0) then
-     MSG_ERROR('A zero was found in coatofin. Check that the fine FFT mesh is finer in each dimension than the coarse FFT mesh.')
+     msg='A zero was found in coatofin.&
+& Check that the fine FFT mesh is finer&
+& in each dimension than the coarse FFT mesh.'
+     MSG_ERROR(msg)
      stop
    end if
  end do
