diff -Naur abinit-6.2.2_orig/config/m4/conn-trio.m4 abinit-6.2.2/config/m4/conn-trio.m4
--- config/m4/conn-trio.m4	2010-08-07 04:31:29.000000000 +0000
+++ config/m4/conn-trio.m4	2010-08-11 15:03:20.000000000 +0000
@@ -183,6 +183,10 @@
     AC_MSG_CHECKING([whether the specified HDF library works])
     AC_LINK_IFELSE([AC_LANG_PROGRAM([],
       [[
-        call h5fcreate_f
+        use hdf5
+        character (len = 42) :: filename
+        integer :: mode, hferr
+        integer(hid_t) :: hfid
+        call h5fcreate_f(filename, mode, hfid, hferr)
       ]])], [abi_trio_hdf_has_libs="yes"], [abi_trio_hdf_has_libs="no"])
     AC_MSG_RESULT([${abi_trio_hdf_has_libs}])
@@ -250,7 +251,9 @@
     AC_LINK_IFELSE([AC_LANG_PROGRAM([],
       [[
         use netcdf
-        call nf90_open
+        character (len = 42) :: filename
+        integer :: status, mode, ncid
+        status=nf90_open(filename, mode, ncid)
       ]])], [abi_trio_netcdf_has_libs="yes"], [abi_trio_netcdf_has_libs="no"])
     AC_MSG_RESULT([${abi_trio_netcdf_has_libs}])
   fi
