diff -Naur abinit-6.2.2_orig/config/m4/conn-fft.m4 abinit-6.2.2/config/m4/conn-fft.m4
--- config/m4/conn-fft.m4	2010-08-07 04:31:29.000000000 +0000
+++ config/m4/conn-fft.m4	2010-08-12 08:53:17.000000000 +0000
@@ -393,6 +393,12 @@
     AC_DEFINE([HAVE_FFT_MPI],1,[Define to 1 if you have an optimized MPI-parallel FFT library.])
   fi
 
+  dnl Transmit information
+  lib_fft_fcflags="${abi_fft_fcflags}"
+  lib_fft_ldflags="${abi_fft_ldflags}"
+  lib_fft_incs="${abi_fft_incs}"
+  lib_fft_libs="${abi_fft_libs}"
+
   dnl Restore build environment
   AC_LANG_POP
   LIBS="${abi_saved_LIBS}"
diff -Naur abinit-6.2.2_orig/src/98_main/Makefile.am abinit-6.2.2/src/98_main/Makefile.am
--- src/98_main/Makefile.am	2010-08-07 04:32:39.000000000 +0000
+++ src/98_main/Makefile.am	2010-08-12 08:43:42.000000000 +0000
@@ -101,8 +101,10 @@
 LIB_ETSF_IO = @lib_etsf_io_libs@
 LIB_FOX = @lib_fox_libs@
 LIB_LIBXC = @lib_libxc_libs@
+LIB_DFT = @lib_dft_libs@
 LIB_NETCDF = @lib_netcdf_libs@
 LIB_WANNIER90 = @lib_wannier90_libs@
+LIB_TRIO = @lib_trio_libs@
 
 # Binary list
 bin_PROGRAMS = \
@@ -131,10 +133,12 @@
 abinetcdf_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
 	$(CPPFLAGS) \
-	@lib_netcdf_incs@
+	@lib_netcdf_incs@ \
+	@lib_trio_incs@
 abinetcdf_LDADD = \
 	$(LIB_10_DEFS) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	$(FC_LIBS)
 
 # abinit.F90 ---> abinit
@@ -144,9 +148,11 @@
 	$(CPPFLAGS) \
 	@lib_bigdft_incs@ \
 	@lib_libxc_incs@ \
+	@lib_dft_incs@ \
 	@lib_fox_incs@ \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_fft_incs@ \
 	@lib_math_incs@ \
 	@lib_linalg_incs@ \
@@ -231,9 +237,11 @@
 	$(LIB_WANNIER90) \
 	$(LIB_BIGDFT) \
 	$(LIB_LIBXC) \
+	$(LIB_DFT) \
 	$(LIB_FOX) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_fft_libs@ \
 	@lib_math_libs@ \
 	@lib_linalg_libs@ \
@@ -250,6 +258,7 @@
 	$(CPPFLAGS) \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 aim_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -280,6 +289,7 @@
 	$(LIB_01_INTERFACES_EXT) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -291,6 +301,7 @@
 	@lib_bigdft_incs@ \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 anaddb_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -339,6 +350,7 @@
 	$(LIB_BIGDFT) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -357,6 +369,7 @@
 	@lib_etsf_io_incs@ \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 conducti_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -404,6 +417,7 @@
 	$(LIB_ETSF_IO) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -415,6 +429,7 @@
 	@lib_etsf_io_incs@ \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_fft_incs@ \
 	@lib_linalg_incs@
 cut3d_FCFLAGS = \
@@ -465,6 +480,7 @@
 	$(LIB_ETSF_IO) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_fft_libs@ \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
@@ -516,6 +532,7 @@
 	$(CPPFLAGS) \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 lwf_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -539,6 +556,7 @@
 	$(LIB_01_INTERFACES_EXT) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -550,6 +568,7 @@
 	$(CPPFLAGS) \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
  macroave_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -576,6 +595,7 @@
 	$(LIB_01_INTERFACES_EXT) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 endif
@@ -588,6 +608,7 @@
 	@lib_bigdft_incs@ \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 mrgddb_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -635,6 +656,7 @@
 	$(LIB_BIGDFT) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -646,6 +668,7 @@
 	@lib_bigdft_incs@ \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 mrggkk_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -673,6 +696,7 @@
 	$(LIB_BIGDFT) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -684,6 +708,7 @@
 	@lib_bigdft_incs@ \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 mrgscr_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -733,6 +758,7 @@
 	$(LIB_BIGDFT) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -744,6 +770,7 @@
 	@lib_etsf_io_incs@ \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 newsp_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -791,6 +818,7 @@
 	$(LIB_ETSF_IO) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -802,6 +830,7 @@
 	@lib_etsf_io_incs@ \
 	@lib_bigdft_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 optic_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -848,6 +877,7 @@
 	$(LIB_ETSF_IO) \
 	$(LIB_BIGDFT) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
@@ -858,6 +888,7 @@
 	$(CPPFLAGS) \
 	@lib_etsf_io_incs@ \
 	@lib_netcdf_incs@ \
+	@lib_trio_incs@ \
 	@lib_linalg_incs@
 ujdet_FCFLAGS = \
 	$(AM_FCFLAGS) \
@@ -912,6 +943,7 @@
 	$(LIB_01_GSL_EXT) \
 	$(LIB_ETSF_IO) \
 	$(LIB_NETCDF) \
+	$(LIB_TRIO) \
 	@lib_linalg_libs@ \
 	$(FC_LIBS)
 
