--- makefile_refmac.inc1.orig	2008-08-01 22:17:14.000000000 -0700
+++ makefile_refmac.inc1	2008-08-01 22:18:46.000000000 -0700
@@ -69,9 +69,7 @@
 bridge.o                    \
 i0.o                        \
 i1.o                        \
-chbevl.o                    \
-$(LLIBCIF)                  \
-$(LLIBLAPACK)
+chbevl.o                    
 #$(LLIBCCP)                  \
 #crt0.o                      \
 
@@ -95,8 +93,7 @@
 emin.o                      \
 libcheck_min.o              \
 mol2_file.o                 \
-sdf_files.o                 \
-$(LLIBCCP)
+sdf_files.o                 
 
 SUBM =                      \
 main_makecif.o              \
@@ -116,9 +113,7 @@
 make_PDB.o                  \
 makecif.o                   \
 make_unix_ccp4.o            \
-make_vdw.o                  \
-$(LLIBCCP)                  \
-$(LLIBCIF)
+make_vdw.o                  
 
 SUBT =                      \
 tlsgroups.o                 \
@@ -142,9 +137,7 @@
 inout_routines.o            \
 errwrt_1.o                  \
 linalgebra.o                \
-subnumrec.o                 \
-$(LLIBCCP)                  \
-$(LLIBCIF)
+subnumrec.o                 
 #--------------------------------------------------------------------------------
 
 ed = echo 'g/^CMDW-5.0/s//        /@w@q@' | tr "@" "\012" | ed -s
@@ -154,16 +147,16 @@
 all:	refmac libcheck makecif
 
 refmac:	$(SUBR)
-	$(FC) $(FFLAGS) -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP) $(LLIBCIF) $(LLIBLAPACK)
 
 libcheck:	$(SUBL)
-	$(FC) $(FFLAGS) -o libcheck${VERSION} $(SUBL)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o libcheck${VERSION} $(SUBL) $(LLIBCCP)
 
 makecif:	$(SUBM)
-	$(FC) $(FFLAGS) -o makecif${VERSION} $(SUBM)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o makecif${VERSION} $(SUBM) $(LLIBCCP) $(LLIBCIF)
 
 tlsgroups:	$(SUBT)
-	$(FC) $(FFLAGS) -o tlsgroups${VERSION} $(SUBT)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o tlsgroups${VERSION} $(SUBT) $(LLIBCCP) $(LLIBCIF)
 
 clean:
 	/bin/rm *.o
