diff --git a/makefile_refmac.inc1 b/makefile_refmac.inc1
index 79d408e..e66c887 100644
--- a/makefile_refmac.inc1
+++ b/makefile_refmac.inc1
@@ -73,9 +73,7 @@ messy_pdb.o                 \
 bridge.o                    \
 i0.o                        \
 i1.o                        \
-chbevl.o                    \
-$(LLIBCIF)                  \
-$(LLIBLAPACK)
+chbevl.o                    
 #$(LLIBCCP)                  \
 #crt0.o                      \
 #heavy.o                     \
@@ -107,9 +105,7 @@ cell_symm_subrs.o           \
 iheap_sort.o                \
 linalgebra.o                \
 math_refmac.o               \
-subnumrec.o                 \
-$(LLIBLAPACK)               \
-$(LLIBCCP)
+subnumrec.o                 
 
 #--------------------------------------------------------------------------------
 
@@ -138,10 +134,10 @@ hkon_secder_tch: hkon_secder_tch.f ncs_simil.f90
 	$(FC) $(FFLAGS) -c ncs_simil.f90 hkon_secder_tch.f
 
 refmac:	$(SUBR)
-	$(FC) $(FFLAGS) -pg -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o refmac${VERSION}   ${SUBR} $(LLIBOTHERS) $(LLIBCCP) $(LLIBCIF) $(LLIBLAPACK)
 
 libcheck:	$(SUBL)
-	$(FC) $(FFLAGS) -o libcheck${VERSION} $(SUBL)
+	$(FC) $(FFLAGS) $(LDFLAGS) -o libcheck${VERSION} $(SUBL) $(LLIBLAPACK) $(LLIBCCP)
 
 clean:
 	/bin/rm *.o
