# fixes wrong handling of long path and filenames
diff -Nur xnec2c.orig/src/main.c xnec2c/src/main.c
--- xnec2c.orig/src/main.c	2010-07-10 13:58:40.000000000 +0200
+++ xnec2c/src/main.c	2010-07-10 13:59:07.000000000 +0200
@@ -138,7 +138,10 @@
 	{
 	  case 'i' : /* specify input file name */
 		if( strlen(optarg) > 80 )
-		  stop( "Input file name too long", 1 );
+		{
+		  fprintf ( stderr, "Input file name too long\n" );
+		  exit(1);
+		}
 		strcpy( infile, optarg );
 		break;
 
diff -Nur xnec2c.orig/src/misc.c xnec2c/src/misc.c
--- xnec2c.orig/src/misc.c	2010-07-10 13:58:40.000000000 +0200
+++ xnec2c/src/misc.c	2010-07-10 13:59:11.000000000 +0200
@@ -234,8 +234,8 @@
   Close_File( fp );
   if( (*fp = fopen(fname, mode)) == NULL )
   {
-	char mesg[88] = "xnec2c: ";
-	strcat( mesg, fname );
+	char mesg[110] = "xnec2c: ";
+	strncat( mesg, fname, 80 );
 	perror( mesg );
 	strcat( mesg, ": Failed to open file" );
 	stop( mesg, 1 );
