--- Makefile	2009-07-07 05:40:18.000000000 +0200
+++ Makefile.docfix	2009-11-25 00:47:33.000000000 +0100
@@ -241,15 +241,15 @@
 docPs:
 	rm -rf bin/doc
 	mkdir bin/doc
-	for i in 1 2 3; do latex doc/manual.tex -output-directory=bin/doc/; done
+	for i in 1 2 3; do latex -output-directory=bin/doc/ doc/manual.tex; done
 	cd bin; dvips doc/manual.dvi
 docPdf:
 	rm -rf bin/doc
 	mkdir bin/doc
-	for i in 1 2 3; do pdflatex doc/manual.tex -output-directory=bin/doc/; done
+	for i in 1 2 3; do pdflatex -output-directory=bin/doc/ doc/manual.tex; done
 	mv bin/doc/manual.pdf bin
 docDviOnce: # Useful to view changes when writing the manual
-	latex doc/manual.tex -output-directory=bin/doc
+	latex -output-directory=bin/doc doc/manual.tex 
 
 # It may seem wasteful to run doxygen three times to generate three
 # kinds of output. However, the latex output for creating a pdf file
