diff -Naur pdb2pqr-1.0.2/propka/configure pdb2pqr-1.0.2-new/propka/configure
--- pdb2pqr-1.0.2/propka/configure	2005-12-16 09:50:58.000000000 -0600
+++ pdb2pqr-1.0.2-new/propka/configure	2006-03-21 09:18:26.000000000 -0600
@@ -2861,7 +2861,7 @@
 fi
 if test -z "$F77"; then
   ac_ct_F77=$F77
-  for ac_prog in g77 f77 xlf frt pgf77 fl32 af77 fort77 f90 xlf90 pgf90 epcf90 f95 fort xlf95 lf95 g95
+  for ac_prog in g77 f77 gfortran xlf frt pgf77 fl32 af77 fort77 f90 xlf90 pgf90 epcf90 f95 fort xlf95 lf95 g95
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
