--- src/Algorithm/LinearSolvers/IpMumpsSolverInterface.cpp.orig	2013-07-03 13:29:25.140918850 -0700
+++ src/Algorithm/LinearSolvers/IpMumpsSolverInterface.cpp	2013-07-03 13:28:22.481558184 -0700
@@ -17,9 +17,9 @@
   // The following line is a fix for otherwise twice-defined global variable
   // (This would have to be taken out for a parallel MUMPS version!)
 # define MPI_COMM_WORLD IPOPT_MPI_COMM_WORLD
-# include "mpi.h"
-}
 
+}
+# include "mpi.h"
 #include "IpMumpsSolverInterface.hpp"
 
 #include "dmumps_c.h"
