--- customize.py.orig	2007-02-13 22:56:17.000000000 +0000
+++ customize.py	2007-02-13 22:59:20.000000000 +0000
@@ -13,9 +13,9 @@
 # will be used, except on Mac OS X, where the vecLib framework will be used
 # if found.
 
-use_system_lapack = 0
+use_system_lapack = 1
 lapack_library_dirs = []
-lapack_libraries = []
+lapack_libraries = ['lapack']
 lapack_extra_link_args = []
 
 # Example: using ATLAS
@@ -35,11 +35,11 @@
 
 # Set use_dotblas to 1 to use BLAS for the matrix multiplication routines.
 # Put the directory that 'cblas.h' is in into dotblas_include_dirs
-use_dotblas = 0
-dotblas_include_dirs = []
+use_dotblas = 1
+dotblas_include_dirs = ['/usr/include/atlas']
 dotblas_cblas_header = '<cblas.h>'
 dotblas_library_dirs = lapack_library_dirs
-dotblas_libraries = lapack_libraries
+dotblas_libraries = ['cblas', 'blas', 'atlas', 'g2c']
 dotblas_extra_link_args = []
 
 # Example: using ATLAS
