--- a/pyproject.toml	2022-06-28 11:08:09.624127900 -0400
+++ b/pyproject.toml	2022-07-11 16:33:14.810411482 -0400
@@ -20,8 +20,11 @@
     "Topic :: Utilities",
     "Typing :: Typed",
 ]
-
-include = ["CHANGELOG.md", "tests"]
+include = [
+  { path = "LICENSE", format = "sdist" },
+  { path = "README.md", format = "sdist" },
+  { path = "CHANGELOG.md", format = "sdist" },
+]

 [tool.poetry.urls]
 Discord = "https://nekit.dev/discord"
