--- nextPhredPipeline.cpp.ori	2011-01-08 20:44:45.000000000 +0100
+++ nextPhredPipeline.cpp	2011-01-08 20:47:29.000000000 +0100
@@ -476,7 +476,7 @@
 void nextPhredPipeline :: openAndLockControlFile() {
 
 
-   nFDControlFile_ = open( filControlFile_.data(), O_CREAT |O_RDWR );
+   nFDControlFile_ = open( filControlFile_.data(), O_CREAT |O_RDWR, 0600 );
    if ( nFDControlFile_ == -1 ) {
       THROW_FILE_ERROR( filControlFile_ );
    }
