diff --git a/pkgs/luafilesystem/Makefile b/pkgs/luafilesystem/Makefile
index 9beaf12..7926e29 100644
--- a/pkgs/luafilesystem/Makefile
+++ b/pkgs/luafilesystem/Makefile
@@ -28,7 +28,7 @@ $(SONAME):
 	ln -s $(SONAMEV) $@
 
 $(LIBRARY): $(OBJ)
-	$(CC) $(CFLAGS) $(LIB_OPTION) -o $(LIBRARY) $(OBJ) -lc
+	$(CC) $(CFLAGS) $(LIB_OPTION) -o $(LIBRARY) $(OBJ) $(LDFLAGS) -lc
 
 install: $(LIB) all
 	cp -a *.so* $(LIB)
diff --git a/pkgs/tcl2lua/Makefile b/pkgs/tcl2lua/Makefile
index dbce85e..be42a85 100644
--- a/pkgs/tcl2lua/Makefile
+++ b/pkgs/tcl2lua/Makefile
@@ -25,7 +25,7 @@ $(SONAME):
 	ln -s $(SONAMEV) $@
 
 $(LIBRARY): $(OBJ)
-	$(CC) $(CFLAGS) $(LIB_OPTION) -o $(LIBRARY) $(OBJ) -lc $(LIBS)
+	$(CC) $(CFLAGS) $(LIB_OPTION) -o $(LIBRARY) $(OBJ) $(LDFLAGS) -lc $(LIBS)
 
 install: all
 	cp -a *.so* $(LIB)
