--- config/scripts/move.sh.SH.orig	2008-04-08 16:08:16.817366051 +0000
+++ config/scripts/move.sh.SH	2008-04-08 19:05:44.332687325 +0000
@@ -445,33 +445,8 @@
 cd $g4install
 #echo `pwd`
 
-
-# Remove 'verbosity' in 'tar' command
-# If not INCLUDE then install full tree in 'src' directory!
-#
-if [ X$g4includes_flag = Xn ] ; then
-if [ X$g4install != X$g4final_install ] ; then
-echo ""
-echo "Installing sources and headers..."
-echo ""
-mkdir -p $g4final_install/src/geant4/
-tar cf - Configure config/ source/ environments/ examples/ include/ |(cd $g4final_install/src/geant4/; tar xf -)
-fi
-fi
-
-########################################
-# There is no env!
-if [ X$g4includes_flag = Xy ] ; then
-
-if [ X$g4install != X$g4final_install ] ; then
-echo ""
-echo "Installing sources..."
-echo ""
-
-mkdir -p $g4final_install/src/geant4/
-tar cf - Configure config/ source/ environments/ examples/ |(cd $g4final_install/src/geant4/; tar xf -)
-fi
-
+# Remove all the stuff to not install sources
+# 
 echo ""
 echo "Installing headers..."
 echo ""
