diff -U2 --recursive mathgl-1.6.2.orig/examples/full_test.cpp mathgl-1.6.2/examples/full_test.cpp
--- mathgl-1.6.2.orig/examples/full_test.cpp	2008-04-04 22:21:20.000000000 +0700
+++ mathgl-1.6.2/examples/full_test.cpp	2008-05-03 10:57:37.000000000 +0700
@@ -253,6 +253,6 @@
 int sample_a4(mglGraph *gr, const void *)	// font features
 {
-	setlocale(LC_CTYPE, "ru_RU.cp1251");
-	gr->Puts(mglPoint(0,1),"Text can be in english и на русском");
+	//setlocale(LC_CTYPE, "ru_RU.cp1251");
+	gr->Puts(mglPoint(0,1),"Text can be in english or russian");
 	gr->Puts(mglPoint(0,0.6),"It can be \\wire{wire} and \\big{big}");
 	gr->Puts(mglPoint(0,0.2),"One can change style in string: "
