--- setup.bak.py	2006-01-17 12:54:55.000000000 -0500
+++ setup.py	2006-01-17 12:59:47.000000000 -0500
@@ -203,47 +203,20 @@
 build_enthought(ext_modules, packages)
 
 if BUILD_GTK:
-    try:
-        import gtk
-    except ImportError:
-        print 'GTK requires pygtk'
-        BUILD_GTK=0
-    except RuntimeError:
-        print 'pygtk present but import failed'
-
-if BUILD_GTK:
-        build_gdk(ext_modules, packages, NUMERIX)
-        rc['backend'] = 'GTK'
+    build_gdk(ext_modules, packages, NUMERIX)
+    rc['backend'] = 'GTK'
 
 if BUILD_TKAGG:
-    try: import Tkinter
-    except ImportError: print 'TKAgg requires TkInter'
-    else:
-        BUILD_AGG = 1
-        build_tkagg(ext_modules, packages, NUMERIX)
-        rc['backend'] = 'TkAgg'
+    BUILD_AGG = 1
+    build_tkagg(ext_modules, packages, NUMERIX)
+    rc['backend'] = 'TkAgg'
         
 if BUILD_WXAGG:
-    try: import wxPython
-    except ImportError:
-        if BUILD_WXAGG != 'auto':
-            print 'WXAgg\'s accelerator requires wxPython'
-        BUILD_WXAGG = 0
-    else:
-        BUILD_AGG = 1
-        build_wxagg(ext_modules, packages, NUMERIX,
-            not (isinstance(BUILD_WXAGG, str) # don't about if BUILD_WXAGG
-                 and BUILD_WXAGG.lower() == 'auto')) # is "auto"
-        rc['backend'] = 'WXAgg'
-
-if BUILD_GTKAGG:
-    try:
-        import gtk
-    except ImportError:
-        print 'GTKAgg requires pygtk'
-        BUILD_GTKAGG=0
-    except RuntimeError:
-        print 'pygtk present but import failed'
+    BUILD_AGG = 1
+    build_wxagg(ext_modules, packages, NUMERIX,
+        not (isinstance(BUILD_WXAGG, str) # don't about if BUILD_WXAGG
+             and BUILD_WXAGG.lower() == 'auto')) # is "auto"
+    rc['backend'] = 'WXAgg'
 
 if BUILD_GTKAGG:
     BUILD_AGG = 1
