diff -Nur matplotlib-0.87.6.orig/lib/matplotlib/__init__.py matplotlib-0.87.6/lib/matplotlib/__init__.py
--- matplotlib-0.87.6.orig/lib/matplotlib/__init__.py	2006-09-27 23:33:28.000000000 +0100
+++ matplotlib-0.87.6/lib/matplotlib/__init__.py	2006-10-10 19:08:36.000000000 +0100
@@ -339,9 +339,10 @@
         path = os.environ['MATPLOTLIBDATA']
         if os.path.isdir(path): return path
 
-    path = os.sep.join([os.path.dirname(__file__), 'mpl-data'])
+    path = os.sep.join(['/usr/share/matplotlib', 'mpl-data'])
     if os.path.isdir(path): return path
-
+    raise RuntimeError('Could not find the matplotlib data files')
+	
     # setuptools' namespace_packages may highjack this init file
     # so need to try something known to be in matplotlib, not basemap
     import matplotlib.afm
@@ -925,7 +926,7 @@
     if os.path.exists(fname): return fname
 
 
-    path =  get_data_path() # guaranteed to exist or raise
+    path =  '/etc' # guaranteed to exist or raise
     fname = os.path.join(path, 'matplotlibrc')
     if not os.path.exists(fname):
         warnings.warn('Could not find matplotlibrc; using defaults')
diff -Nur matplotlib-0.87.6.orig/setup.py matplotlib-0.87.6/setup.py
--- matplotlib-0.87.6.orig/setup.py	2006-09-27 23:29:29.000000000 +0100
+++ matplotlib-0.87.6/setup.py	2006-10-10 19:00:19.000000000 +0100
@@ -84,7 +84,7 @@
 data.extend(glob.glob('images/*.svg'))
 data.extend(glob.glob('images/*.png'))
 data.extend(glob.glob('images/*.ppm'))
-data.append('matplotlibrc')
+#data.append('matplotlibrc')
 
 data_files=[('matplotlib/mpl-data', data),
             ('matplotlib/mpl-data/Matplotlib.nib',
