--- 3d/mplot3d.py~	2006-02-15 16:37:51.000000000 +0000
+++ 3d/mplot3d.py	2006-03-05 20:02:48.000000000 +0000
@@ -966,7 +966,8 @@
 
     def contour3D(self, X, Y, Z, *args, **kwargs):
         had_data = self.has_data()
-        levels, colls = self.contour(X, Y, Z, *args, **kwargs)
+        C = self.contour(X, Y, Z, *args, **kwargs)
+        levels, colls = (C.levels, C.collections)
         for z,linec in zip(levels,colls):
             zl = []
             for s in linec._segments:
@@ -978,7 +979,8 @@
     def contourf3D(self, X, Y, Z, *args, **kwargs):
         had_data = self.has_data()
         
-        levels, colls = self.contourf(X, Y, Z, 20)
+        C = self.contourf(X, Y, Z, *args, **kwargs)
+        levels, colls = (C.levels, C.collections)        
         print len(levels),len(colls)
         for z1,z2,linec in zip(levels,levels[1:],colls):
             zs = [z1] * (len(linec._verts[0])/2)
