diff -arNu aria2.2/aria2.py aria2.2.new/aria2.py
--- aria2.2/aria2.py	2008-02-14 11:02:31.000000000 +0100
+++ aria2.2.new/aria2.py	2009-07-14 12:01:02.000000000 +0200
@@ -427,7 +427,7 @@
         
 def check_numeric_slice():
 
-    msg = "\nThe version of Numeric (%s) is known to be incompatible with ARIA.\nConsider reverting to a more stable version (like 23.8).\n"
+    msg = "\nThe version of Numeric (%s) is known to be incompatible with ARIA.\nSo we are using numpy instead.\n"
     
     try:
         from Numeric import ones, __version__ as NUMERIC_VERSION
diff -arNu aria2.2/check.py aria2.2.new/check.py
--- aria2.2/check.py	2009-07-14 11:50:54.000000000 +0200
+++ aria2.2.new/check.py	2009-07-14 12:02:56.000000000 +0200
@@ -55,18 +55,6 @@
     failed = 0
     
     try:
-        import Numeric
-        msg += 'ok (numeric).'
-        
-    except:
-        #msg =  'could not import Numeric module.'
-        failed = 1
-
-    if not failed:
-        print msg
-        return 'numeric'#, None
-
-    try:
        import numpy.oldnumeric
        version = LooseVersion(numpy.oldnumeric.__version__)
 
@@ -79,14 +67,14 @@
            msg += 'version > %s required for Numpy' % str(NUMPY_MIN_VERSION)
            
     except:        
-        msg += 'could not import Numeric/Numpy module.'
+        msg += 'could not import Numpy module.'
 
     print msg
     return None
 
 def check_numeric_slice():
 
-    msg = "\nWARNING: This version of Numeric (%s) is known to be incompatible with ARIA.\nConsider reverting to a more stable version (like 23.8).\n"
+    msg = "\nWARNING: This version of Numeric (%s) is known to be incompatible with ARIA.\nSo we use numpy instead.\n"
     
     from Numeric import ones, __version__ as NUMERIC_VERSION
     if not ones(10)[2:]:
diff -arNu aria2.2/src/py/Merger.py aria2.2.new/src/py/Merger.py
--- aria2.2/src/py/Merger.py	2007-03-05 16:50:12.000000000 +0100
+++ aria2.2.new/src/py/Merger.py	2009-07-14 12:06:55.000000000 +0200
@@ -213,12 +213,9 @@
                 new_long = []
 
                 try:
-                    from RandomArray import permutation
+                    from numpy.oldnumeric.random_array import permutation
                 except:
-                    try:
-                        from numpy.oldnumeric.random_array import permutation
-                    except:
-                        return n, []
+                    return n, []
 
                 ids = permutation(len(long_range))
            
diff -arNu aria2.2/src/py/N.py aria2.2.new/src/py/N.py
--- aria2.2/src/py/N.py	2007-12-04 17:01:46.000000000 +0100
+++ aria2.2.new/src/py/N.py	2009-07-14 12:07:16.000000000 +0200
@@ -1,7 +1,4 @@
 try:
-    from Numeric import *
+    from numpy.oldnumeric import *
 except:
-    try:
-        from numpy.oldnumeric import *
-    except:
-        pass
+    pass
