Adjust installation path of examples.

Patch by Marius Brehler.

--- b/examples/CMakeLists.txt
+++ a/examples/CMakeLists.txt
@@ -13,7 +13,7 @@ MACRO(CREATE_EXAMPLES BACKEND)
         TARGET_LINK_LIBRARIES(${EXAMPLE_NAME} af${BACKEND} ${THREAD_LIB_FLAG})
         SET_TARGET_PROPERTIES(${EXAMPLE_NAME} PROPERTIES COMPILE_FLAGS -DAF_${DEF_NAME} OUTPUT_NAME ${FNAME}_${BACKEND})
 
-#        INSTALL(TARGETS ${EXAMPLE_NAME} DESTINATION arrayfire/examples)
+        INSTALL(TARGETS ${EXAMPLE_NAME} DESTINATION share/arrayfire/examples)
      ENDFOREACH()
 ENDMACRO(CREATE_EXAMPLES)
 
