Debundle wcstools

--- a/setup.py
+++ b/setup.py
@@ -13,36 +13,8 @@
 topDir=os.getcwd()
 sourceDir="PyWCSTools"+os.path.sep+"wcssubs-3.8.4"+os.path.sep
 
-#oFiles=glob.glob(sourceDir+"*.o")
-#print oFiles
-oFiles=['PyWCSTools/wcssubs-3.8.4/cel.o', 'PyWCSTools/wcssubs-3.8.4/wcs.o', 'PyWCSTools/wcssubs-3.8.4/proj.o', 'PyWCSTools/wcssubs-3.8.4/distort.o', 'PyWCSTools/wcssubs-3.8.4/wcsinit.o', 'PyWCSTools/wcssubs-3.8.4/wcslib.o', 'PyWCSTools/wcssubs-3.8.4/poly.o', 'PyWCSTools/wcssubs-3.8.4/platepos.o', 'PyWCSTools/wcssubs-3.8.4/zpxpos.o', 'PyWCSTools/wcssubs-3.8.4/iget.o', 'PyWCSTools/wcssubs-3.8.4/imio.o', 'PyWCSTools/wcssubs-3.8.4/dsspos.o', 'PyWCSTools/wcssubs-3.8.4/tnxpos.o', 'PyWCSTools/wcssubs-3.8.4/wcscon.o', 'PyWCSTools/wcssubs-3.8.4/fitsfile.o', 'PyWCSTools/wcssubs-3.8.4/dateutil.o', 'PyWCSTools/wcssubs-3.8.4/imhfile.o', 'PyWCSTools/wcssubs-3.8.4/lin.o', 'PyWCSTools/wcssubs-3.8.4/fileutil.o', 'PyWCSTools/wcssubs-3.8.4/wcstrig.o', 'PyWCSTools/wcssubs-3.8.4/slasubs.o', 'PyWCSTools/wcssubs-3.8.4/sph.o', 'PyWCSTools/wcssubs-3.8.4/worldpos.o', 'PyWCSTools/wcssubs-3.8.4/hget.o', 'PyWCSTools/wcssubs-3.8.4/hput.o']
-
 exampleScripts=glob.glob("scripts"+os.path.sep+"*.py")
 
-class build_PyWCSTools_ext(build_ext):
-	
-    def build_extensions(self):
-        
-        os.chdir(sourceDir)
-        
-        cc=distutils.ccompiler.new_compiler(distutils.ccompiler.get_default_compiler())
-        distutils.command.config.customize_compiler(cc)
-        
-        # Suppress warnings from compiling WCSTools wcssubs-3.8.4
-        if "-Wstrict-prototypes" in cc.compiler_so:
-            cc.compiler_so.pop(cc.compiler_so.index("-Wstrict-prototypes"))
-        if "-Wall" in cc.compiler_so:
-            cc.compiler_so.pop(cc.compiler_so.index("-Wall"))
-
-        WCSToolsCFiles=glob.glob("*.c")
-        WCSToolsCFiles.pop(WCSToolsCFiles.index("wcs_wrap.c"))
-        WCSToolsCFiles.pop(WCSToolsCFiles.index("wcscon_wrap.c"))
-        cc.compile(WCSToolsCFiles)
-        
-        os.chdir(topDir)
-        
-        build_ext.build_extensions(self)		
-
 setup(name='astLib',
     version='0.6.0',
     url='http://astlib.sourceforge.net',
@@ -65,12 +37,11 @@
     PyWCSTools is distributed (and developed) as part of astLib.""",
     packages=['astLib', 'PyWCSTools'],
     package_data={'astLib': ['data/*']},
-    cmdclass={"build_ext": build_PyWCSTools_ext},
     scripts=exampleScripts,
     ext_modules=[
         Extension('PyWCSTools._wcscon', [sourceDir+"wcscon_wrap.c"], 
-        extra_objects=oFiles),
+        libraries=['wcstools']),
         Extension('PyWCSTools._wcs', [sourceDir+"wcs_wrap.c"], 
-        extra_objects=oFiles)
+        libraries=['wcstools'])
     ]
 	)
