diff -U2 --recursive udav-0.3.2.orig/configure.ac udav-0.3.2/configure.ac
--- udav-0.3.2.orig/configure.ac	2008-04-03 01:24:14.000000000 +0700
+++ udav-0.3.2/configure.ac	2008-04-30 13:43:19.000000000 +0700
@@ -18,6 +18,6 @@
 	)
 
-AC_CHECK_HEADERS([FL/Fl_Export.H],,[echo "Please install FLTK library"; exit])
-AC_CHECK_LIB([fltk], [main], [FLTK_LIBS="-lfltk -lfltk_images" AC_SUBST(FLTK_LIBS)], 
+AC_CHECK_HEADERS([fltk-1.1/FL/Fl_Export.H],,[echo "Please install FLTK library"; exit])
+AC_CHECK_LIB([fltk], [main], [FLTK_LIBS="-L/usr/lib/fltk-1.1 -lfltk -lfltk_images -lXft" AC_SUBST(FLTK_LIBS)], 
 	[echo "Please install FLTK library (libfltk)"; exit],\
 	)
diff -U2 --recursive udav-0.3.2.orig/src/Makefile.am udav-0.3.2/src/Makefile.am
--- udav-0.3.2.orig/src/Makefile.am	2008-04-03 01:24:14.000000000 +0700
+++ udav-0.3.2/src/Makefile.am	2008-04-30 13:53:00.000000000 +0700
@@ -8,3 +8,3 @@
 
 udav_LDADD = $(MGL_LIBS) $(FLTK_LIBS)
-udav_CPPFLAGS = -O2 -DDOC_DIR=\"$(docdir)/\"
+udav_CPPFLAGS = -I/usr/include/fltk-1.1 -DDOC_DIR=\"$(docdir)/\"
