--- configure.ac.orig	2008-04-03 07:24:14.000000000 +1300
+++ configure.ac	2008-05-11 16:34:17.000000000 +1200
@@ -17,10 +17,9 @@
 	[echo "Please install MathGL library (libmgl)"; exit],\
 	)
 
-AC_CHECK_HEADERS([FL/Fl_Export.H],,[echo "Please install FLTK library"; exit])
-AC_CHECK_LIB([fltk], [main], [FLTK_LIBS="-lfltk -lfltk_images" AC_SUBST(FLTK_LIBS)], 
-	[echo "Please install FLTK library (libfltk)"; exit],\
-	)
+AC_PATH_PROG(FLTKCONFIG,fltk-config)
+CXXFLAGS="`$FLTKCONFIG --cflags` $CXXFLAGS"
+LDFLAGS="`$FLTKCONFIG --use-images --ldflags` $LDFLAGS"
 
 
 AC_OUTPUT(Makefile src/Makefile)
--- src/Makefile.am.orig	2008-05-11 16:39:57.000000000 +1200
+++ src/Makefile.am	2008-05-11 16:40:28.000000000 +1200
@@ -7,4 +7,4 @@
 data.cpp     Fl_Table.cxx  main.cpp  option.cpp  table.cpp
 
 udav_LDADD = $(MGL_LIBS) $(FLTK_LIBS)
-udav_CPPFLAGS = -O2 -DDOC_DIR=\"$(docdir)/\"
+udav_CPPFLAGS = $(CPPFLASGS) -DDOC_DIR=\"$(docdir)/\"
