diff -Naur parallel-2.0.0/src/pserver.cc parallel-2.0.0.new/src/pserver.cc
--- parallel-2.0.0/src/pserver.cc	2009-05-08 09:17:57.000000000 -0400
+++ parallel-2.0.0.new/src/pserver.cc	2009-12-10 21:48:02.000000000 -0500
@@ -61,8 +61,6 @@
 // SSIZE_MAX might be for 64-bit. Limit to 2^31-1
 #define BUFF_SIZE 2147483647
 
-static bool quitting_gracefully = false;
-
 // Handle server SIGTERM SIGQUIT
 
 static RETSIGTYPE
@@ -99,7 +97,8 @@
 do_octave_atexit_server (void)
 {
   static bool deja_vu = false;
-
+  extern bool quitting_gracefully;
+   
   while (! octave_atexit_functions.empty ())
     {
       std::string fcn = octave_atexit_functions.top ();
