diff -urN spl-0.6.0-rc3.orig/include/Makefile.am spl-0.6.0-rc3/include/Makefile.am
--- spl-0.6.0-rc3.orig/include/Makefile.am	2011-04-07 22:30:50.000000000 +0400
+++ spl-0.6.0-rc3/include/Makefile.am	2011-04-12 03:12:40.000000000 +0400
@@ -15,8 +15,7 @@
 noinst_HEADERS += $(top_srcdir)/include/vm/*.h
 
 install-data-local:
-	destname=spl-$(SPL_META_VERSION)/$(LINUX_VERSION); \
-	instdest=$(DESTDIR)/usr/src/$$destname; \
+	instdest=$(DESTDIR)/${includedir}/spl; \
 	instfiles=`find . -name '*.h'`; \
         for instfile in $$instfiles; do \
 		$(INSTALL) -D $$instfile $$instdest/$$instfile; \
diff -urN spl-0.6.0-rc3.orig/Makefile.am spl-0.6.0-rc3/Makefile.am
--- spl-0.6.0-rc3.orig/Makefile.am	2011-04-07 22:30:50.000000000 +0400
+++ spl-0.6.0-rc3/Makefile.am	2011-04-12 03:12:22.000000000 +0400
@@ -29,8 +29,7 @@
 
 if CONFIG_KERNEL
 install-data-local:
-	destname=spl-$(SPL_META_VERSION)/$(LINUX_VERSION); \
-	instdest=$(DESTDIR)/usr/src/$$destname; \
+	instdest=$(DESTDIR)/${includedir}/spl; \
 	echo "$(SPL_META_VERSION)" >$$instdest/spl.release; \
 	for instfile in $(noinst_HEADERS) module/$(LINUX_SYMBOLS); do \
 		$(INSTALL) -D $$instfile $$instdest/$$instfile; \
