--- ocfs2-tools-1.4.4-buggy/configure	2010-10-04 18:10:31.327238001 +0200
+++ ocfs2-tools-1.4.4/configure	2010-10-05 10:27:52.607238001 +0200
@@ -5534,7 +5534,12 @@
 echo $ECHO_N "checking for cpg_initialize in -lcpg... $ECHO_C" >&6
     while test "x$TRY_PATH" != "x"; do
         TRY="`echo $TRY_PATH | cut -f1 -d:`"
-        TRY_PATH="`echo $TRY_PATH | cut -f2- -d:`"
+        NEW_TRY_PATH="`echo $TRY_PATH | cut -f2- -d:`"
+        if test "x$NEW_TRY_PATH" != "x$TRY_PATH"; then
+            TRY_PATH="$NEW_TRY_PATH"
+        else
+            TRY_PATH=""
+        fi
 
         if test "x$TRY" != "x"; then
             TRY="-L${TRY}/${cpg_package}"
@@ -5680,7 +5685,13 @@
 echo $ECHO_N "checking for saCkptInitialize in -lSaCkpt... $ECHO_C" >&6
     while test "x$TRY_PATH" != "x"; do
         TRY="`echo $TRY_PATH | cut -f1 -d:`"
-        TRY_PATH="`echo $TRY_PATH | cut -f2- -d:`"
+        NEW_TRY_PATH="`echo $TRY_PATH | cut -f2- -d:`"
+        if test "x$NEW_TRY_PATH" != "x$TRY_PATH"; then
+            TRY_PATH="$NEW_TRY_PATH"
+        else
+            TRY_PATH=""
+        fi
+	
 
         if test "x$TRY" != "x"; then
             TRY="-L${TRY}/openais"
