diff -urN torque-2.0.0p2.org/src/iff/Makefile.in torque-2.0.0p2/src/iff/Makefile.in
--- torque-2.0.0p2.org/src/iff/Makefile.in	2005-12-11 14:03:23.000000000 +0100
+++ torque-2.0.0p2/src/iff/Makefile.in	2005-12-11 14:12:44.000000000 +0100
@@ -114,7 +114,7 @@
 	$(INSTALL) -m 4755 $(TARGET) $(DESTDIR)$(sbindir)
 
 pbs_iff: $(OBJS) $(PBS_LIBS)
-	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $(OBJS) $(PBS_LIBS) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -Wl,-z,now -o $@ $(OBJS) $(PBS_LIBS) $(LIBS)
 
 @mk_cleanup@
 @mk_tail@
diff -urN torque-2.0.0p2.org/src/mom_rcp/Makefile.in torque-2.0.0p2/src/mom_rcp/Makefile.in
--- torque-2.0.0p2.org/src/mom_rcp/Makefile.in	2005-12-11 14:03:23.000000000 +0100
+++ torque-2.0.0p2/src/mom_rcp/Makefile.in	2005-12-11 14:13:21.000000000 +0100
@@ -115,7 +115,7 @@
 	$(INSTALL) -m 4755 $(TARGET) $(DESTDIR)$(sbindir)
 
 $(TARGET): $(OBJS)
-	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $(OBJS) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -Wl,-z,now -o $@ $(OBJS) $(LIBS)
 
 
# needed for make depend

