diff -Naur torque-2.0.0p8/src/lib/Libpbs/Makefile.in torque-2.0.0p8.new/src/lib/Libpbs/Makefile.in
--- torque-2.0.0p8/src/lib/Libpbs/Makefile.in	2006-01-27 18:42:09.000000000 +0000
+++ torque-2.0.0p8.new/src/lib/Libpbs/Makefile.in	2006-03-21 21:59:09.525967999 +0000
@@ -93,7 +93,7 @@
 INCS = -I@top_builddir@/src/include -I$(top_srcdir)/src/include \
        -I$(srcdir)/../Libdis -I$(srcdir)/../Libifl
 
-CFLAGS = @CFLAGS@ $(INCS) @DEFS@ -DIFF_PATH=\"$(IFF_PATH)\"
+CFLAGS = @CFLAGS@ $(INCS) @DEFS@ -DIFF_PATH=\"$(IFF_PATH)\" -fPIC
 
 TARGET = libpbs.a
 
