diff -Naur torque-2.1.0p0/src/iff/Makefile.in torque-2.1.0p0.new/src/iff/Makefile.in
--- torque-2.1.0p0/src/iff/Makefile.in	2006-04-06 22:56:32.000000000 +0100
+++ torque-2.1.0p0.new/src/iff/Makefile.in	2006-05-19 23:51:52.000000000 +0100
@@ -146,7 +146,7 @@
 am_pbs_iff_OBJECTS = iff2.$(OBJEXT)
 pbs_iff_OBJECTS = $(am_pbs_iff_OBJECTS)
 pbs_iff_DEPENDENCIES = ../lib/Libpbs/libtorque.la
-pbs_iff_LDFLAGS =
+pbs_iff_LDFLAGS = -Wl,-z,now
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)/src/include
diff -Naur torque-2.1.0p0/src/mom_rcp/Makefile.in torque-2.1.0p0.new/src/mom_rcp/Makefile.in
--- torque-2.1.0p0/src/mom_rcp/Makefile.in	2006-04-06 22:56:34.000000000 +0100
+++ torque-2.1.0p0.new/src/mom_rcp/Makefile.in	2006-05-19 23:52:42.000000000 +0100
@@ -145,7 +145,7 @@
 pbs_rcp_OBJECTS = $(am_pbs_rcp_OBJECTS)
 pbs_rcp_LDADD = $(LDADD)
 pbs_rcp_DEPENDENCIES =
-pbs_rcp_LDFLAGS =
+pbs_rcp_LDFLAGS = -Wl,-z,now
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)/src/include
