--- torque-2.0.0p5/buildutils/makedepend-sh.old	2005-12-15 21:14:28.000000000 -0500
+++ torque-2.0.0p5/buildutils/makedepend-sh	2006-01-23 12:39:36.000000000 -0500
@@ -572,11 +572,11 @@
                 #
                 test -n "$mksrcs" && echo $d/$s >> $SRCS
                 test "$verbose" -lt 2 && errout='2> /dev/null'
-    
+   
                 eval $CPP $arg_cc $d/$s $errout | \
                   sed -n -e "s;^\# [0-9][0-9 ]*\"\(.*\)\";$f: \1;p" | \
-                  grep -v "//\$" | \
-                  grep -v "$s\$" | grep -v command | grep -v built-in | \
+                  grep -v "$s\$" | \
+                  grep -v ">$" | \
                   sed -e 's;\([^ :]*: [^ ]*\).*;\1;' \
                   >> $TMP
                 ;;
