--- src/configure.orig	2010-06-01 19:49:30.000000000 +0100
+++ src/configure	2010-06-01 19:52:05.000000000 +0100
@@ -2421,13 +2421,13 @@
 }
 EOF
 				echocheck "AMD ACML support"
-				if cc_check -lacml -lcblas -lgfortran
+				if cc_check $(pkg-config --libs cblas lapack)
 				then
 					echores "yes"
 					HAVE_ACML='#define HAVE_ACML 1'
 					HAVE_LAPACK='#define HAVE_LAPACK 1'
 					DEFINES="$DEFINES -DHAVE_ACML -DHAVE_LAPACK"
-					LINKFLAGS="$LINKFLAGS -lacml -lcblas -lgfortran"
+					LINKFLAGS="$LINKFLAGS $(pkg-config --libs cblas lapack)"
 				else
 					echores "no"
 
