diff --git a/Build/CMake/FindOpenCascade.cmake b/Build/CMake/FindOpenCascade.cmake
index aefa011..346678d 100644
--- a/Build/CMake/FindOpenCascade.cmake
+++ b/Build/CMake/FindOpenCascade.cmake
@@ -4,13 +4,6 @@
 # OPENCASCADE_FOUND If false, do not try to use OPENCASCADE
 
 find_path ( OPENCASCADE_INCLUDE_DIR Standard.hxx PATHS
-                /usr/include/opencascade
-                /usr/include/oce
-                /usr/local/include/opencascade
-                /usr/local/include/oce
-                /usr/local/inc
-                /opt/OpenCASCADE/inc
-                ${CMAKE_INCLUDE_PATH}/OpenCascade
                 $ENV{CASROOT}/inc
 )
 
@@ -18,50 +11,56 @@ if ( WIN32 )
   add_definitions ( -DWNT )
 endif ( WIN32 )
 
-set ( OC_REDHAT_RPM /opt/OpenCASCADE/lib64 )
+find_path ( OPENCASCADE_LIBRARY_DIR libTKernel.so PATHS
+               $ENV{CASROOT}/lib64
+)
+
 find_library ( OPENCASCADE_LIB_TKERNEL
                  NAMES TKernel
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKBO
                  NAMES TKBO
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKPRIM
                  NAMES TKPrim
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKMESH
                  NAMES TKMesh
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKBREP
                  NAMES TKBRep
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKTOPALGO
                  NAMES TKTopAlgo
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKMATH
                  NAMES TKMath
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 find_library ( OPENCASCADE_LIB_TKG2D
                  NAMES TKG2d
-                 PATHS ${OC_REDHAT_RPM}         
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 
 find_library ( OPENCASCADE_LIB_TKG3D
                  NAMES TKG3d
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 
 find_library ( OPENCASCADE_LIB_TKGEOMBASE
                  NAMES TKGeomBase
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 
 find_library ( OPENCASCADE_LIB_TKGEOMAlgo
                  NAMES TKGeomAlgo
+                 PATHS ${OPENCASCADE_LIBRARY_DIR}
 )
 
 set ( OPENCASCADE_LIBRARIES
