diff --git a/Framework/Geometry/CMakeLists.txt b/Framework/Geometry/CMakeLists.txt
index bb1d146..466b072 100644
--- a/Framework/Geometry/CMakeLists.txt
+++ b/Framework/Geometry/CMakeLists.txt
@@ -351,10 +351,8 @@ else ()
   add_definitions ( -DNO_OPENCASCADE )
 endif ()
 
-# A few defines needed for OpenCascade on the Mac
-if ( APPLE )
-  add_definitions ( -DHAVE_IOSTREAM -DHAVE_LIMITS -DHAVE_IOMANIP )
-endif ()
+# A few defines needed for OpenCascade
+add_definitions ( -DHAVE_IOSTREAM -DHAVE_LIMITS -DHAVE_IOMANIP )
 
 include_directories ( ${OPENGL_INCLUDE_DIR} )
 include_directories ( ${GSL_INCLUDE_DIR} )
