--- src/DoConfig	2009-04-03 08:57:43.000000000 +0200
+++ src/DoConfig.new	2009-05-01 14:28:00.000000000 +0200
@@ -25,11 +25,13 @@
 'LDFLAGS_CXX' => '$(LDFLAGS)',
 'LDLIBS'      => '-lm',
 'LDLIBS_CXX'  => '$(LDLIBS)',
+'PICFLAG'     => '-fPIC',
 'CPPFLAGS'    => '',
-'PREFIX'      => '/usr/local',
+'PREFIX'      => '/usr/',
 'LIBDIR'      => '$(PREFIX)/lib',
 'INCLUDEDIR'  => '$(PREFIX)/include',
 'DOCDIR'      => '$(PREFIX)/share/doc',
+'SHMAKE'      => 'non-gld',
 
 'GMP_PREFIX'  => undef,
 'GMP_INCDIR'  => undef,
@@ -84,11 +86,6 @@
 
 foreach $arg (@ARGV) {
 
-   if ($arg =~ '-h|help|-help|--help') {
-      system("more ../doc/config.txt");
-      exit;
-   }
-
    if (($name, $val) = ($arg =~ /(.*?)=(.*)/)) {
       
       if (exists($MakeFlag{$name}) && ($val =~ 'on|off')) {
--- src/mfile	2009-05-01 16:44:21.000000000 +0200
+++ src/mfile.new	2009-05-01 16:56:02.000000000 +0200
@@ -75,6 +75,11 @@
 # where to install NTL
 
 ###############################################################
+# choose the appropriate shared object makefile
+###############################################################
+SHMAKE=@{SHMAKE}
+
+###############################################################
 #
 #  Second, if you want to use GMP (the GNU Multi-Precision library),
 #  define the variables GMP_INCDIR, GMP_LIBDIR, GMP_LIB below.
@@ -137,6 +142,16 @@
 WIZARD=@{WIZARD}
 # Set to off if you want to bypass the wizard; otherwise, set to on.
 
+###############################################################
+#
+# New addition for shared library building. With gcc you need to
+# choose the Position Indepent Code flag. You have a choice of 
+# -fpic better code but in rare case not available (ppc)
+# -fPIC slightly slower code but guaranted to work anywhere.
+#
+###############################################################
+
+PICFLAG=@{PICFLAG}
 
 #################################################################
 #
@@ -169,6 +184,26 @@
 
 OBJ=$(O19)
 
+SHOBJ=FFT.lo FacVec.lo GF2.lo GF2E.lo GF2EX.lo GF2EXFactoring.lo GF2X.lo GF2X1.lo \
+	GF2XFactoring.lo GF2XVec.lo GetTime.lo HNF.lo ctools.lo LLL.lo LLL_FP.lo \
+	LLL_QP.lo LLL_RR.lo LLL_XD.lo RR.lo WordVector.lo ZZ.lo ZZVec.lo \
+	ZZX.lo ZZX1.lo ZZXCharPoly.lo ZZXFactoring.lo ZZ_p.lo ZZ_pE.lo ZZ_pEX.lo \
+	ZZ_pEXFactoring.lo ZZ_pX.lo ZZ_pX1.lo ZZ_pXCharPoly.lo ZZ_pXFactoring.lo \
+	fileio.lo lip.lo lzz_p.lo lzz_pE.lo lzz_pEX.lo lzz_pEXFactoring.lo \
+	lzz_pX.lo lzz_pX1.lo lzz_pXCharPoly.lo lzz_pXFactoring.lo \
+	mat_GF2.lo mat_GF2E.lo mat_RR.lo mat_ZZ.lo mat_ZZ_p.lo \
+	mat_ZZ_pE.lo mat_lzz_p.lo mat_lzz_pE.lo mat_poly_ZZ.lo \
+	mat_poly_ZZ_p.lo mat_poly_lzz_p.lo pair_GF2EX_long.lo \
+	pair_GF2X_long.lo pair_ZZX_long.lo pair_ZZ_pEX_long.lo \
+	pair_ZZ_pX_long.lo pair_lzz_pEX_long.lo pair_lzz_pX_long.lo \
+	quad_float.lo tools.lo vec_GF2.lo vec_GF2E.lo vec_GF2XVec.lo \
+	vec_RR.lo vec_ZZ.lo vec_ZZVec.lo vec_ZZ_p.lo vec_ZZ_pE.lo \
+	vec_double.lo vec_long.lo vec_lzz_p.lo vec_lzz_pE.lo vec_quad_float.lo \
+	vec_vec_GF2.lo vec_vec_GF2E.lo vec_vec_RR.lo vec_vec_ZZ.lo \
+	vec_vec_ZZ_p.lo vec_vec_ZZ_pE.lo vec_vec_long.lo vec_vec_lzz_p.lo \
+	vec_vec_lzz_pE.lo vec_xdouble.lo xdouble.lo \
+	G_LLL_FP.lo G_LLL_QP.lo G_LLL_XD.lo G_LLL_RR.lo vec_ulong.lo vec_vec_ulong.lo
+
 # library source files
 
 
@@ -316,7 +351,7 @@
 LINK = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 LINK_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS_CXX)
 
-
+.SUFFIXES: .lo
 
 # 'make all' does a complete make, including all setup.
 # It also creates the file 'all', which means you should
@@ -374,18 +409,32 @@
 lip.o:	lip.c g_lip_impl.h c_lip_impl.h lip_gmp_aux_impl.h
 	$(LCOMP) $(COMPILE) $(GMP_INCDIR) lip.c
 
+lip.lo: lip.c g_lip_impl.h c_lip_impl.h lip_gmp_aux_impl.h
+	$(LCOMP) $(COMPILE) $(PICFLAG) $(GMP_INCDIR) lip.c -o lip.lo
+
+
 ctools.o:	ctools.c
 	$(LCOMP) $(COMPILE) ctools.c
 
 
+ctools.lo: ctools.c
+	$(LCOMP) $(COMPILE) $(PICFLAG) ctools.c -o ctools.lo
+
+
 GetTime.o: GetTime.c 
 	$(LCOMP) $(COMPILE) GetTime.c
 
 
+GetTime.lo: GetTime.c 
+	$(LCOMP) $(COMPILE) $(PICFLAG) GetTime.c -o GetTime.lo
 
 .c.o: 
 	$(LCOMP) $(COMPILE_CXX) $(GF2X_INCDIR) $<
 
+.c.lo: 
+	$(LCOMP) $(COMPILE_CXX) $(PICFLAG) -o $@ $<
+
+
 .c: 
 @{LSTAT}	$(LINK_CXX) -o $@ $< ntl.a $(GMP_LIBDIR) $(GMP_LIB) $(GF2X_LIBDIR) $(GF2X_LIB) $(LDLIBS_CXX) #LSTAT
 @{LSHAR}	$(LIBTOOL) --mode=link $(LINK_CXX) -o $@ $< libntl.la #LSHAR
@@ -455,19 +504,18 @@
 #
 #################################################################
 
-clobber:	
+clobber:	clean
 	rm -f ntl.a mach_desc.h ../include/NTL/mach_desc.h  GetTime.c 
 	rm -f lip_gmp_aux_impl.h ../include/NTL/gmp_aux.h
-	sh RemoveProg $(PROGS) MakeDesc TestGetTime gen_lip_gmp_aux gen_gmp_aux
-	rm -f *.o
-	rm -rf small
+	sh RemoveProg $(PROGS)
+	rm -f libntl*.so*
 	rm -f cfileout mfileout
 	rm -rf .libs *.lo libntl.la
 	rm -f all
 
 clean:
 	sh RemoveProg MakeDesc TestGetTime gen_lip_gmp_aux gen_gmp_aux
-	rm -f *.o 
+	rm -f *.o *.lo
 	rm -rf small
 @{LSHAR}	- $(LIBTOOL) --mode=clean rm -f libntl.la *.lo #LSHAR
 
--- src/mfile	2009-05-01 17:07:58.000000000 +0200
+++ src/mfile.new	2009-05-01 17:19:09.000000000 +0200
@@ -592,3 +592,4 @@
 
 
 
+sinclude $(SHMAKE).mk
