diff --git a/Makefile b/Makefile
index ab410bd..cb0a1c1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,18 +1,37 @@
-build: ripser
+TARGET	=	ripser
 
+DESTDIR	?=
 
-all: ripser ripser-coeff ripser-debug
+prefix	?= /usr/local
+exec_prefix	?= $(prefix)
+bindir	?= $(exec_prefix)/bin
 
+CFLAGS	+= -std=c++11 -Wall -Wextra -Wpedantic -Wstrict-overflow
+
+ifeq ($(INDICATE_PROGRESS),1)
+	CFLAGS += -D INDICATE_PROGRESS
+endif
+
+ifeq ($(USE_GOOGLE_HASHMAP),1)
+	CFLAGS += -D USE_GOOGLE_HASHMAP
+endif
+
+ifeq ($(NDEBUG),1)
+	CFLAGS += -D NDEBUG
+endif
+
+build: all
+
+all: ripser ripser-coeff
 
 ripser: ripser.cpp
-	c++ -std=c++11 -Wall ripser.cpp -o ripser -Ofast -D NDEBUG
+	$(CXX) $(CFLAGS) $(CXXFLAGS) ripser.cpp -o ripser
 
 ripser-coeff: ripser.cpp
-	c++ -std=c++11 -Wall ripser.cpp -o ripser-coeff -Ofast -D NDEBUG -D USE_COEFFICIENTS
-
-ripser-debug: ripser.cpp
-	c++ -std=c++11 -Wall ripser.cpp -o ripser-debug -g
+	$(CXX) $(CFLAGS) $(CXXFLAGS) -D USE_COEFFICIENTS ripser.cpp -o ripser-coeff
 
+install: all
+	install -m 755 ripser ripser-coeff $(bindir)
 
 clean:
-	rm -f ripser ripser-coeff ripser-debug
+	rm -f ripser ripser-coeff
