From 4fea6631cd69c90816777e75955a80e81bb04901 Mon Sep 17 00:00:00 2001
From: Michael Schubert <mschu.dev@gmail.com>
Date: Sat, 26 Jul 2025 15:55:58 +0200
Subject: [PATCH] fix: if file is present and no sha256 provided assume pass

---
 src/snakemake/assets/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/snakemake/assets/__init__.py b/src/snakemake/assets/__init__.py
index f0be54693b..19b0912c1f 100644
--- a/src/snakemake/assets/__init__.py
+++ b/src/snakemake/assets/__init__.py
@@ -541,7 +541,7 @@ def deploy(cls) -> None:
             if target_path.exists():
                 with open(target_path, "rb") as fin:
                     # file is already present, check if it is up to date
-                    if (asset.sha256 is not None) and (
+                    if (asset.sha256 is None) or (
                         asset.sha256 == hashlib.sha256(fin.read()).hexdigest()
                     ):
                         continue
