--- root/core/utils/Module.mk.orig	2012-07-22 11:18:33.000000000 +0400
+++ root/core/utils/Module.mk	2012-07-22 11:20:02.633436297 +0400
@@ -95,6 +95,10 @@
 	$(MAKEDIR)
 	cp $< $@
 
+$(UTILSDIRS)/%_tmp.cxx: $(UTILSDIRS)/%.cxx
+	$(MAKEDIR)
+	cp $< $@
+
 $(call stripsrc,$(UTILSDIRS)/rootcint_tmp.o): $(call stripsrc,$(UTILSDIRS)/rootcint_tmp.cxx)
 
 $(call stripsrc,$(UTILSDIRS)/RStl_tmp.o): $(call stripsrc,$(UTILSDIRS)/RStl_tmp.cxx)
