From 08b08412bafc24fa635b0fdb832097a3aa2fa1d2 Mon Sep 17 00:00:00 2001
From: Pere Mato <pere.mato@cern.ch>
Date: Sun, 21 Sep 2014 19:27:30 +0200
Subject: [PATCH] Fix for ROOT-6728 - Compilation error in TTimeStamp

---
 core/base/src/TTimeStamp.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/base/src/TTimeStamp.cxx b/core/base/src/TTimeStamp.cxx
index 8709220..f0ff12d 100644
--- a/core/base/src/TTimeStamp.cxx
+++ b/core/base/src/TTimeStamp.cxx
@@ -813,7 +813,7 @@ void TTimeStamp::DumpTMStruct(const tm_t &tmstruct)
           tmstruct.tm_isdst);
 #if (defined(linux) && !defined(R__WINGCC)) || defined(R__MACOSX)
    printf(",\n      tm_gmtoff %6ld, tm_zone \"%s\"",
-#if defined(__USE_BSD) || defined(R__MACOSX)
+#if defined(__USE_BSD) || defined(R__MACOSX) || defined(__USE_MISC)
    tmstruct.tm_gmtoff, tmstruct.tm_zone);
 #else
    tmstruct.__tm_gmtoff, tmstruct.__tm_zone);
