diff -ur root-5.34.00.orig/build/rmkdepend/main.c root-5.34.00/build/rmkdepend/main.c
--- root-5.34.00.orig/build/rmkdepend/main.c	2012-06-05 15:31:46.000000000 +0200
+++ root-5.34.00/build/rmkdepend/main.c	2012-07-05 20:42:58.938953353 +0200
@@ -60,7 +60,7 @@
 #  endif
 # elif defined(__CYGWIN__)
 extern int fchmod(int, mode_t);
-#else
+# elif !defined(__GLIBC__) || __GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 16)
 extern int fchmod(int, int);
 # endif
 #endif
diff -ur root-5.34.00.orig/cint/tool/rmkdepend/main.c root-5.34.00/cint/tool/rmkdepend/main.c
--- root-5.34.00.orig/cint/tool/rmkdepend/main.c	2012-06-05 15:31:30.000000000 +0200
+++ root-5.34.00/cint/tool/rmkdepend/main.c	2012-07-05 20:42:58.492949104 +0200
@@ -58,7 +58,7 @@
 #if !defined(MAC_OS_X_VERSION_10_4)
 extern int fchmod();
 #endif
-#else
+#elif !defined(__GLIBC__) || __GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 16)
 extern int fchmod();
 #endif
 #endif
diff -ur root-5.34.00.orig/core/unix/src/TUnixSystem.cxx root-5.34.00/core/unix/src/TUnixSystem.cxx
--- root-5.34.00.orig/core/unix/src/TUnixSystem.cxx	2012-06-05 15:31:17.000000000 +0200
+++ root-5.34.00/core/unix/src/TUnixSystem.cxx	2012-07-05 20:29:29.969242837 +0200
@@ -98,6 +98,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <errno.h>
+#include <sys/resource.h>
 #include <sys/wait.h>
 #include <time.h>
 #include <sys/time.h>
