--- root/build/rmkdepend/main.c.orig	2012-05-09 15:00:06.000000000 +0400
+++ root/build/rmkdepend/main.c	2013-12-22 01:58:44.939245160 +0400
@@ -52,18 +52,6 @@
 #else
 #include <io.h>
 #endif
-#if !defined(__hpux)
-# if defined(__APPLE__)
-#  include <AvailabilityMacros.h>
-#  if !defined(MAC_OS_X_VERSION_10_4)
-extern int fchmod(int, int);
-#  endif
-# elif defined(__CYGWIN__)
-extern int fchmod(int, mode_t);
-#else
-extern int fchmod(int, int);
-# endif
-#endif
 
 #ifdef MINIX
 #define USE_CHMOD 1
