diff --git a/montecarlo/pythia8/Module.mk b/montecarlo/pythia8/Module.mk
index b3d0ee4..86fb3d5 100644
--- a/montecarlo/pythia8/Module.mk
+++ b/montecarlo/pythia8/Module.mk
@@ -50,7 +50,7 @@ $(PYTHIA8LIB):  $(PYTHIA8O) $(PYTHIA8DO) $(ORDER_) $(MAINLIBS) $(PYTHIA8LIBDEP)
 $(PYTHIA8DS):   $(PYTHIA8H) $(PYTHIA8L) $(ROOTCINTTMPDEP)
 		$(MAKEDIR)
 		@echo "Generating dictionary $@..."
-		$(ROOTCINTTMP) -f $@ -c -I$(FPYTHIA8INCDIR) $(PYTHIA8H) $(PYTHIA8L)
+		$(ROOTCINTTMP) -f $@ -c -p -I$(FPYTHIA8INCDIR) $(PYTHIA8H) $(PYTHIA8L)
 
 $(PYTHIA8MAP):  $(RLIBMAP) $(MAKEFILEDEP) $(PYTHIA8L)
 		$(RLIBMAP) -o $@ -l $(PYTHIA8LIB) \
