diff --git a/config/RConfigure.in b/config/RConfigure.in
index 5950401..e4fbfbf 100644
--- a/config/RConfigure.in
+++ b/config/RConfigure.in
@@ -11,6 +11,7 @@
 #define ROOTDATADIR   "@datadir@"
 #define ROOTDOCDIR    "@docdir@"
 #define ROOTMACRODIR  "@macrodir@"
+#define ROOTTUTDIR    "@tutdir@"
 #define ROOTSRCDIR    "@srcdir@"
 #define ROOTICONPATH  "@iconpath@"
 #define EXTRAICONPATH "@extraiconpath@"
diff --git a/configure b/configure
index 131df92..2c8ea95 100755
--- a/configure
+++ b/configure
@@ -7537,6 +7537,7 @@ sed \
     -e "s|@prefix@|$prefix|"               \
     -e "s|@srcdir@|$srcdir|"               \
     -e "s|@ttffontdir@|$fontdir|"          \
+    -e "s|@tutdir@|$tutdir|"               \
     -e "s|@setresuid@|$setresuid|"         \
     -e "s|@hasmathmore@|$hasmathmore|"     \
     -e "s|@haspthread@|$haspthread|"       \
diff --git a/core/base/src/TApplication.cxx b/core/base/src/TApplication.cxx
index 95fb1e5..3830955 100644
--- a/core/base/src/TApplication.cxx
+++ b/core/base/src/TApplication.cxx
@@ -794,6 +794,22 @@ Long_t TApplication::ProcessRemote(const char *line, Int_t *)
    return 1;
 }
 
+namespace {
+   static int PrintFile(const char* filename) {
+      TString sFileName(filename);
+      gSystem->ExpandPathName(sFileName);
+      if (gSystem->AccessPathName(sFileName)) {
+         Error("ProcessLine()", "Cannot find file %s", filename);
+         return 1;
+      }
+      std::ifstream instr(sFileName);
+      TString content;
+      content.ReadFile(instr);
+      Printf("%s", content.Data());
+      return 0;
+   }
+}
+
 //______________________________________________________________________________
 Long_t TApplication::ProcessLine(const char *line, Bool_t sync, Int_t *err)
 {
@@ -827,11 +843,41 @@ Long_t TApplication::ProcessLine(const char *line, Bool_t sync, Int_t *err)
       return 0;
    }
 
-   if (!strncmp(line, "?", 1)) {
+   if (!strncmp(line, "?", 1) || !strncmp(line, ".help", 5)) {
       Help(line);
       return 1;
    }
 
+   if (!strncmp(line, ".demo", 5)) {
+      if (gROOT->IsBatch()) {
+         Error("ProcessLine", "Cannot show demos in batch mode!");
+         return 1;
+      }
+#ifdef ROOTTUTDIR
+      ProcessLine(".x " ROOTTUTDIR "/demos.C");
+#else
+      ProcessLine(".x $(ROOTSYS)/tutorials/demos.C");
+#endif
+      return 0;
+   }
+
+   if (!strncmp(line, ".license", 8)) {
+#ifdef ROOTDOCDIR
+      return PrintFile(ROOTDOCDIR "/LICENSE");
+#else
+      return PrintFile("$(ROOTSYS)/LICENSE");
+#endif
+   }
+
+   if (!strncmp(line, ".credits", 8)) {
+#ifdef ROOTDOCDIR
+      return PrintFile(ROOTDOCDIR "/CREDITS");
+#else
+      return PrintFile("$(ROOTSYS)/README/CREDITS");
+#endif
+     
+   }
+
    if (!strncmp(line, ".pwd", 4)) {
       if (gDirectory)
          Printf("Current directory: %s", gDirectory->GetPath());
