--- root.orig/configure	2010-12-17 22:34:50.987000643 +0300
+++ root/configure	2010-12-17 22:35:55.046516404 +0300
@@ -2703,8 +2703,8 @@
 if test ! "x$enable_pgsql" = "xno"; then
     # Check for PgSQL include and library
     check_header "libpq-fe.h" "$pgsqlincdir" $PGSQL $PGSQL/include \
-        /usr/include /usr/local/include /usr/local/pgsql/include \
-        /usr/include/pgsql /usr/include/postgresql \
+        /usr/include/postgresql /usr/include /usr/local/include \
+        /usr/local/pgsql/include /usr/include/pgsql \
         /usr/include/postgresql/8.0 /usr/pgsql/include \
         /usr/local/include/pgsql /usr/local/include/postgresql \
         /usr/include/postgresql/8.0 /usr/pgsql /usr/local/pgsql \
@@ -2786,7 +2786,7 @@
     else
         check_header "sqlext.h" "$odbcincdir" \
             $ODBC $ODBC/include $ODBCINCDIR /usr/local/include \
-            /usr/include $finkdir/include
+            /usr/include $finkdir/include /usr/include/iodbc
         odbcinc=$found_hdr
         odbcincdir=$found_dir
 

