--- etc/dictpch/makepch-gentoo.sh.orig	2015-04-08 14:27:07.781026830 +0300
+++ etc/dictpch/makepch-gentoo.sh	2015-04-12 13:40:48.226060739 +0300
@@ -13,8 +13,8 @@
 # Copyright (c) 2014 Rene Brun and Fons Rademakers
 # Author: Axel Naumann <axel@cern.ch>, 2014-10-16
 
-rootdir=.
-cfgdir=etc/dictpch
+rootdir=ROOTDIR_TEMPLATE
+cfgdir=etc/root/dictpch
 allheaders=$cfgdir/allHeaders.h
 alllinkdefs=$cfgdir/allLinkDefs.h
 cppflags=$cfgdir/allCppflags.txt
@@ -36,7 +36,7 @@
     rootbuild=-rootbuild
 fi
 
-cxxflags="-D__CLING__ -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DROOT_PCH -I$rootdir/include -I$rootdir/etc -I$rootdir/$cfgdir -I$rootdir/etc/cling `cat $rootdir/$cppflags`"
+cxxflags="-D__CLING__ -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DROOT_PCH -I$rootdir/usr/include/root -I$rootdir/etc/root -I$rootdir/$cfgdir -I$rootdir/etc/root/cling `cat $rootdir/$cppflags`"
 
 if ! [ "x$1" = "x" ]; then
     cxxflags="$cxxflags $1"
@@ -44,7 +44,7 @@
 
 # generate pch
 touch allDict.cxx.h
-$rootdir/bin/rootcling $rootbuild -1 -f allDict.cxx -noDictSelection -c $cxxflags $allheaders $@ $alllinkdefs
+$rootdir/usr/bin/rootcling -1 -f allDict.cxx -noDictSelection -c $cxxflags $allheaders $@ $alllinkdefs
 res=$?
 if [ $res -eq 0 ] ; then
   mv allDict_rdict.pch $pch
