--- root/io/io/Module.mk.orig	2011-11-24 20:12:56.000000000 +0400
+++ root/io/io/Module.mk	2012-03-09 03:45:42.850549826 +0400
@@ -67,9 +67,4 @@
 distclean::     distclean-$(MODNAME)
 
 ##### extra rules ######
-#ifeq ($(GCC_VERS_FULL),gcc-4.4.0)
-ifeq ($(GCC_VERS),gcc-4.4)
-ifneq ($(filter -O%,$(OPT)),)
    $(call stripsrc,$(IODIRS)/TStreamerInfoReadBuffer.o): CXXFLAGS += -DR__EXPLICIT_FUNCTION_INSTANTIATION
-endif
-endif
