--- configure.orig	2008-07-07 09:43:12.732330045 +0100
+++ configure	2008-07-07 15:45:27.122015928 +0100
@@ -1648,7 +1648,7 @@
       if pkg-config --exists ftgl ; then
 	  result "ok"
 	  ftglincdir=`pkg-config --cflags-only-I ftgl | sed 's/-I//g'`
-	  ftgllibs=`pkg-config --libs-only-l ftgl | sed s'/-lftgl/-lftgl_pic/'`
+	  ftgllibs=`pkg-config --libs-only-l ftgl`
 	  ftgllibdir=`pkg-config --libs-only-L ftgl | sed 's/-L//g'`
 	  enable_builtin_ftgl=no
       else
@@ -2753,12 +2753,12 @@
        found_dirz=""
        # libz must be before libpng, if libz is not found don't
        # add libpng which needs libz
-       # note that failure to find system libungif is not fatal since
-       # libAfterImage contains embedded libungif source if needed
+       # note that failure to find system libgif is not fatal since
+       # libAfterImage contains embedded libgif source if needed
        if test ! "x$enable_astiff" = "xno" ; then
-           aslibs="libjpeg libtiff libungif libz libpng"
+           aslibs="libjpeg libtiff libgif libz libpng"
        else
-           aslibs="libjpeg libungif libz libpng"
+           aslibs="libjpeg libgif libz libpng"
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
@@ -2771,7 +2771,7 @@
            if test "x$k" = "xlibpng" && test "x$found_libz" = "x" ; then
                break;
            fi
-           if test "x$k" = "xlibungif" && test "x$found_lib" = "x" ; then
+           if test "x$k" = "xlibgif" && test "x$found_lib" = "x" ; then
                asgifincdir=""
            fi
            if test ! "x$found_lib" = "x" ; then
