--- root/config/Makefile.in.orig	2013-01-21 22:04:43.011452938 +0400
+++ root/config/Makefile.in	2013-01-21 22:04:18.027352209 +0400
@@ -25,9 +25,9 @@
 USECONFIG      := @useconfig@
 MKLIBOPTIONS   := @mkliboption@
 
-EXTRA_CFLAGS   := -Iinclude @cflags@
-EXTRA_CXXFLAGS := -Iinclude @cflags@
-EXTRA_LDFLAGS  := @ldflags@
+EXTRA_CFLAGS   := -Iinclude @cflags@ $(CFLAGS)
+EXTRA_CXXFLAGS := -Iinclude @cflags@ $(CXXFLAGS)
+EXTRA_LDFLAGS  := @ldflags@ $(LDFLAGS)
 WINRTDEBUG     := @winrtdebug@
 
 GLBPATCHFLAGS  := @glbpatchcflags@
