--- ccp4i/crank/plugins/shelxd/crank_shelxd.tcl 	2009-09-02 21:28:51.000000000 +0200
+++ ccp4i/crank/plugins/shelxd/crank_shelxd.tcl		2009-08-17 21:17:55.000000000 +0200
@@ -114,7 +114,7 @@
         puts $smalllog "Trial       CCweak       CC   \$\$ \$\$"
         for {set i 1} {$i <= $ntrial} {incr i} {
 	    if { [info exists accweak($i)] && [info exists acc($i)] } {
-		if { $accweak($i) > maxaccweak } {
+		if { $accweak($i) > $maxaccweak } {
 		    set maxaccweak accweak($i)
 		    set trialmaxaccweak $i
 		    set maxacc acc($i)
