--- ccp4i/tasks/refmac5.tcl	2009-03-13 08:53:52.000000000 +0100
+++ ccp4i/tasks/refmac5.tcl.new	2009-03-27 08:40:49.093519853 +0100
@@ -6,7 +6,7 @@
 #     A copy of the CCP4 licence can be obtained by writing to the
 #     CCP4 Secretary, Daresbury Laboratory, Warrington WA4 4AD, UK.
 #
-#CCP4i_cvs_Id $Id: refmac5.tcl,v 1.59.2.7 2009/02/02 14:25:03 nds65 Exp $
+#CCP4i_cvs_Id $Id: refmac5.tcl,v 1.59.2.11 2009/03/26 17:02:41 rmk65 Exp $
 # ======================================================================
 # refmac.tcl --
 #
@@ -48,6 +48,22 @@
       WarningMessage "Input Warning: DelFwt map name not specified. Will use default file name."
     }
   }
+  
+  # Check to see if need FREER column but it has not been set
+  if { $array(EXCLUDE_FREER) } {
+    if { $array(FREE) == "Unassigned" || $array(FREE) == "" } {
+      WarningMessage "Free R label has not been set."
+      return 0
+    }
+  }
+  
+  if { $array(MLSC_REF_SET) == "free" } {
+    if { $array(FREE) == "Unassigned" || $array(FREE) == "" } {
+      WarningMessage "You are using free reflections to fit SigmaA but Free R label not set."
+      return 0
+    }
+  }
+    
 
   if { $array(MAKE_LIBRARY) != "" } { 
     append array(INPUT_FILES) " MAKE_LIBRARY" }
@@ -612,6 +628,24 @@
 # If XYZIN set e.g. re-running then initialise chain list
   refmac_update_chain_menu $arrayname
 
+# catch for ATOM compatibility between 6.1.0 and 6.1.1
+  catch {
+  if { $array(ATOM) != "" } {
+    set array(NATOMS) 1
+    set array(ATOM,1) $array(ATOM)
+    set array(ATOM_FP,1) $array(ATOM_FP)
+    set array(ATOM_FPP,1) $array(ATOM_FPP)
+#   set array(ATOM,0) $array(ATOM)
+#   set array(ATOM_FP,0) $array(ATOM_FP)
+#   set array(ATOM_FPP,0) $array(ATOM_FPP)
+    }
+    unset array(ATOM)
+    unset array(ATOM_FP)
+    unset array(ATOM_FPP)
+# have to update PARAM_LIST element too, this may need watching
+    regsub -all -- {ATOM ATOM_FP ATOM_FPP} $array(PARAM_LIST) { } array(PARAM_LIST)
+  }
+
   set array(IFMAPNAME) $preferences(IFMAPNAME)
 
   # See if Coot package installed and available
