--- a/ccp4i/scripts/import.script
+++ b/ccp4i/scripts/import.script
@@ -188,6 +188,10 @@ END\n"
         
       set status [Execute $cmd $ctruncate_script program_status report  ]
 
+      # Copy the output mtz from Ctruncate to the tmpmtz variable so that it will become 
+      # HKLOUT. Note that ctruncate carries over the I/SIGI columns by default.
+
+      set tmpmtz $truncatemtz
   }
 
 #=====================================================================
