 gle-1.5.6/gle/gle.i | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gle-1.5.6/gle/gle.i b/gle-1.5.6/gle/gle.i
index bc48694..76170f1 100644
--- a/gle-1.5.6/gle/gle.i
+++ b/gle-1.5.6/gle/gle.i
@@ -149,7 +149,7 @@ glePolyCone_c4f (int npoints,	 /* numpoints in poly-line */
                    gleColor4f color_array[1],	/* colors at polyline verts */
                    gleDouble radius_array[1]); /* cone radii at polyline verts */
 
-#typemaps for Numeric arrays
+/* typemaps for Numeric arrays */
 DOUBLE_ARRAY2D( contour, [1][2], ncp )
 %apply double ARRAY2D[ANY][ANY] { double cont_normal[1][2] };
 %apply double VECTOR_NULL[ANY] { double up[3] };
@@ -172,7 +172,7 @@ gleExtrusion_c4f (int ncp,         /* number of contour points */
                 gleDouble point_array[1][3], /* polyline vertices */
                 gleColor4f color_array[1]); /* colors at polyline verts */
 
-#typemaps for Numeric arrays
+/* typemaps for Numeric arrays */
 %apply double VECTOR[ANY] { double twist_array[1] };
 
 /* extrude 2D contour, specifying local rotations (twists) */
@@ -195,7 +195,7 @@ gleTwistExtrusion_c4f (int ncp,         /* number of contour points */
                 gleDouble twist_array[1]);   /* countour twists (in degrees) */
 
 
-#typemaps for Numeric arrays
+/* typemaps for Numeric arrays */
 %apply double ARRAY3D_NULL[ANY][ANY][ANY] { double xform_array[1][2][3] };
 
 /* extrude 2D contour, specifying local affine tranformations */
@@ -218,7 +218,7 @@ gleSuperExtrusion_c4f (int ncp,  /* number of contour points */
                 gleColor4f color_array[1],        /* color at polyline verts */
                 gleDouble xform_array[1][2][3]);   /* 2D contour xforms */
 
-#typemaps for Numeric arrays
+/* typemaps for Numeric arrays */
 %apply double ARRAY2D_NULL[ANY][ANY] { double startXform[2][3] };
 %apply double ARRAY2D_NULL[ANY][ANY] { double dXformdTheta[2][3] };
 
