--- CDFPlayer_9.0.1_LINUX.sh
+++ CDFPlayer_9.0.1_LINUX.sh
@@ -261,11 +261,11 @@
                     fi
                 done
                 chmod a+x $0 || echo Please add execution rights on $0
-                if test `echo "$0" | cut -c1` = "/"; then # Spawn a terminal!
-                    exec $XTERM -title "$label" -e "$0" --xwin "$@"
-                else
-                    exec $XTERM -title "$label" -e "./$0" --xwin "$@"
-                fi
+                #if test `echo "$0" | cut -c1` = "/"; then # Spawn a terminal!
+                #    exec $XTERM -title "$label" -e "$0" --xwin "$@"
+                #else
+                #    exec $XTERM -title "$label" -e "./$0" --xwin "$@"
+                #fi
             fi
         fi
     fi
