 setup.py |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 0191ba2..bda3e41 100644
--- a/setup.py
+++ b/setup.py
@@ -268,9 +268,6 @@ class build_EXT(build_ext):
                     if path.isfile(statlib1) and path.isfile(statlib2):
                         self.spawn(["ranlib", "-s", statlib1])
                         self.spawn(["ranlib", "-s", statlib2])
-            togl_build(self,self.include_dirs,
-                       self.library_dirs,
-                       togl_libs)
         #if platform in ["linux2", "win32"]:
         if os.name in ['nt']:
             tkdnd_build(self)
@@ -716,8 +713,8 @@ glxlib_ext = MyExtension("extent._glxlib",
                            ], #sources
                           define_macros = macros,
                           include_dirs = glx_incldirs,
-                          library_dirs = glx_libdirs,
-                          libraries = ['X11'] + gl_libs,
+                          library_dirs = glx_libdirs + ['@GENTOO_PORTAGE_EPREFIX@/usr/lib64/Togl2.0/'],
+                          libraries = ['X11', 'GLEW', 'GLEWmx'] + gl_libs,
                           headers = {
                              'name': path.join("GL","glx.h"),
                              'dir': h_dir,
