--- trinityrnaseq_r20140413p1/trinity-plugins/coreutils/build_parallel_sort.sh.ori	2014-06-25 10:54:43.000000000 +0200
+++ trinityrnaseq_r20140413p1/trinity-plugins/coreutils/build_parallel_sort.sh	2014-06-25 10:56:52.000000000 +0200
@@ -29,7 +29,11 @@
   cd ${DIR}/bin
   SORTPATH=`which sort`
   ln -s ${SORTPATH}
+  if [ ! -x ./sort-softlink]; then
+    if [ ! -x ${DIR}/bin/sort ]; then
   chmod +x ${DIR}/bin/sort
+    fi
+  fi
   exit 0
 fi 
 
