--- src/Makefile.in.orig	2010-08-19 23:38:00.081085063 +1200
+++ src/Makefile.in	2010-08-19 23:38:56.018083508 +1200
@@ -15,8 +15,8 @@
 LTRELEASE = `echo ${RELEASE} | sed -e "s/\./:/g"`
 
 CPHASE = ${LIBTOOL} --tag=CXX --mode=compile @CXX@ -c @CFLAGS@
-LPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@ -version-info ${LTRELEASE} -rpath @prefix@/@libdir@ @LIBS@
-MPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@
+LPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@ -version-info ${LTRELEASE} @LDFLAGS@ -rpath @prefix@/@libdir@ @LIBS@
+MPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@ @LDFLAGS@
 IPHASE = ${LIBTOOL} --mode=install @INSTALL@
 FPHASE = ${LIBTOOL} --mode=finish @prefix@/@libdir@
 
