--- src/Makefile.in.orig	2009-04-03 22:17:57.000000000 +1300
+++ src/Makefile.in	2009-04-03 22:18:19.000000000 +1300
@@ -103,7 +103,6 @@
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/bin
 	@INSTALL@ -m 755 -d ${DEVDESTDIR}@prefix@/include/modglue
 	${IPHASE} libmodglue.la ${DESTDIR}@prefix@/lib
-	mv ${DESTDIR}@prefix@/lib/libmodglue*.so ${DEVDESTDIR}@prefix@/lib
 #	${IPHASE} libmid.la ${DESTDIR}@prefix@/lib
 	${FPHASE}
 	${IPHASE} ptywrap ${DESTDIR}@prefix@/bin
