--- Makefile.in.orig	2010-08-21 22:04:51.877629329 +1200
+++ Makefile.in	2010-08-21 22:06:12.887628815 +1200
@@ -7,7 +7,7 @@
 #examples
 
 library:
-	( cd src && $(MAKE) );
+	( cd src && $(MAKE) library tools );
 
 static:
 	( cd src && $(MAKE) static );
--- src/Makefile.in.orig	2010-08-21 22:05:26.411630332 +1200
+++ src/Makefile.in	2010-08-21 22:08:20.961629492 +1200
@@ -49,21 +49,21 @@
 
 #------- Tests ----------------------------------------
 
-test_child_process: test_child_process.o
+test_child_process: library test_child_process.o
 	${MPHASE} -o test_child_process $< libmodglue.la
 
-test_ext_process: test_ext_process.o
+test_ext_process: library test_ext_process.o
 	${MPHASE} -o test_ext_process $< libmodglue.la
 
-test_pipe: test_pipe.o
+test_pipe: library test_pipe.o
 	${MPHASE} -o test_pipe $< libmodglue.la
 
-test_writes: test_writes.o
+test_writes: library test_writes.o
 	${MPHASE} -o test_writes $< libmodglue.la
 
 #------- Tools ----------------------------------------
 
-prompt: prompt.o
+prompt: library prompt.o
 	${MPHASE} -o prompt prompt.o `pkg-config sigc++-2.0 --libs` libmodglue.la 
 
 prompt_static: prompt.cc library
@@ -77,10 +77,10 @@
 endif
 
 
-ptywrap: ptywrap.o
+ptywrap: library ptywrap.o
 	${MPHASE} -o ptywrap ptywrap.o @LINK_UTIL@
 
-ptywrap_static: ptywrap.o
+ptywrap_static: library ptywrap.o
 	${CPHASE} ${TIMESTAMP} -o ptywrap.o \
           -I@top_srcdir@/include -DSTATICBUILD ptywrap.cc
 ifeq ($(strip $(MACTEST)),)
