--- Makefile.in.orig	2013-09-04 14:12:33.436620240 +1200
+++ Makefile.in	2013-09-04 14:13:31.051711085 +1200
@@ -8,7 +8,7 @@
 
 library:
 	echo @libdir@
-	( cd src && $(MAKE) );
+	( cd src && $(MAKE) library tools );
 
 static:
 	( cd src && $(MAKE) static );
--- src/Makefile.in.orig	2013-09-04 14:13:46.820735926 +1200
+++ src/Makefile.in	2013-09-04 14:15:46.020923434 +1200
@@ -56,21 +56,21 @@
 
 #------- Tests ----------------------------------------
 
-test_child_process: test_child_process.o
+test_child_process: library test_child_process.o
 	${MPHASE} -o test_child_process $< libmodglue.la
 
-test_ext_process: test_ext_process.o
+test_ext_process: library test_ext_process.o
 	${MPHASE} -o test_ext_process $< libmodglue.la
 
-test_pipe: test_pipe.o
+test_pipe: library test_pipe.o
 	${MPHASE} -o test_pipe $< libmodglue.la
 
-test_writes: test_writes.o
+test_writes: library test_writes.o
 	${MPHASE} -o test_writes $< libmodglue.la
 
 #------- Tools ----------------------------------------
 
-prompt: prompt.o
+prompt: library prompt.o
 	${MPHASE} -o prompt prompt.o `pkg-config sigc++-2.0 --libs` libmodglue.la 
 #	chrpath -d .libs/prompt
 
@@ -85,10 +85,10 @@
 endif
 
 
-ptywrap: ptywrap.o
+ptywrap: library ptywrap.o
 	${MPHASE} -o ptywrap ptywrap.o @LINK_UTIL@
 
-ptywrap_static: ptywrap.o
+ptywrap_static: library ptywrap.o
 	${CPHASE} ${TIMESTAMP} -o ptywrap.o \
           -I@top_srcdir@/include -DSTATICBUILD ptywrap.cc
 ifeq ($(strip $(MACTEST)),)
