--- Makefile.in.orig	2010-03-04 10:34:54.000000000 +1300
+++ Makefile.in	2010-08-23 23:21:34.931886424 +1200
@@ -3,6 +3,9 @@
 
 .PHONY: all install library static examples uninstall clean distclean
 
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+
 all: library 
 #examples
 
@@ -17,8 +20,8 @@
 
 install:
 	( cd src && $(MAKE) install );
-	@INSTALL@ -d -m 755 ${DESTDIR}@prefix@/@libdir@/pkgconfig
-	@INSTALL@ -m 644 modglue.pc ${DESTDIR}@prefix@/@libdir@/pkgconfig/
+	@INSTALL@ -d -m 755 ${DESTDIR}@libdir@/pkgconfig
+	@INSTALL@ -m 644 modglue.pc ${DESTDIR}@libdir@/pkgconfig/
 	@INSTALL@ -d -m 755 ${DESTDIR}@prefix@/share/man/man1
 	@INSTALL@ man/man1/prompt.1 ${DESTDIR}@prefix@/share/man/man1
 	@INSTALL@ man/man1/ptywrap.1 ${DESTDIR}@prefix@/share/man/man1
--- src/Makefile.in.orig	2010-03-04 10:34:54.000000000 +1300
+++ src/Makefile.in	2010-08-23 23:21:41.165883457 +1200
@@ -1,6 +1,8 @@
 
 
 MACTEST= @MAC_OS_X@
+prefix = @prefix@
+exec_prefix = @exec_prefix@
 
 ifeq ($(strip $(MACTEST)),)
 LIBTOOL = libtool
@@ -18,7 +20,7 @@
 LPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@ -version-info ${LTRELEASE} -rpath @prefix@/@libdir@ @LIBS@
 MPHASE = ${LIBTOOL} --tag=CXX --mode=link @CXX@
 IPHASE = ${LIBTOOL} --mode=install @INSTALL@
-FPHASE = ${LIBTOOL} --mode=finish @prefix@/@libdir@
+FPHASE = ${LIBTOOL} --mode=finish @libdir@
 
 TIMESTAMP = -D"DATETIME=\"`date | sed -e 's/  / /'`\"" -DHOSTNAME=\"`hostname`\"
 
@@ -98,13 +100,13 @@
 
 install: library tools
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@
-	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/@libdir@
-	@INSTALL@ -m 755 -d ${DEVDESTDIR}@prefix@/@libdir@
+	@INSTALL@ -m 755 -d ${DESTDIR}@libdir@
+	@INSTALL@ -m 755 -d ${DEVDESTDIR}@libdir@
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/bin
 	@INSTALL@ -m 755 -d ${DEVDESTDIR}@prefix@/include/modglue
-	${IPHASE} libmodglue.la ${DESTDIR}@prefix@/@libdir@
+	${IPHASE} libmodglue.la ${DESTDIR}@libdir@
 ifneq ($(DESTDIR),$(DEVDESTDIR))
-	mv ${DESTDIR}@prefix@/@libdir@/libmodglue*.so ${DEVDESTDIR}@prefix@/@libdir@
+	mv ${DESTDIR}@libdir@/libmodglue*.so ${DEVDESTDIR}@libdir@
 endif
 #	${IPHASE} libmid.la ${DESTDIR}@prefix@/lib
 	${FPHASE}
