--- zphota.f.orig	2012-06-14 22:11:52.000000000 +0100
+++ zphota.f	2012-06-14 22:12:18.000000000 +0100
@@ -1725,7 +1725,7 @@
          if (cattyp(1:4).eq.'LONG') then
             j=2*imagm+2
             if (gbcont.eq.-1) then 
-               read(paravc(j),'(E24.0)') conti
+               read(paravc(j),'(I24)') conti
                cont = DBLE(conti)
             else
                cont=gbcont
