diff -Nur SPR-3.3.1.orig/configure.ac SPR-3.3.1/configure.ac
--- SPR-3.3.1.orig/configure.ac	2008-08-11 22:11:07.000000000 +0100
+++ SPR-3.3.1/configure.ac	2008-12-12 17:28:54.562084262 +0000
@@ -114,7 +114,7 @@
     AC_SUBST([ROOTLIBS])
 fi
 
-AM_CONDITIONAL(WITH_ROOT, [test $with_root != xno])
+AM_CONDITIONAL(WITH_ROOT, [test x$with_root != xno])
 
 ################################
 
@@ -125,10 +125,6 @@
 # HippoDraw (not used)
 AM_CONDITIONAL([WITH_HIPPODRAW], false)
 
-AM_CXXFLAGS="-ansi -pedantic -O4"
-AC_SUBST(AM_CPPFLAGS)
-AC_SUBST(AM_CXXFLAGS)
-
 AC_CONFIG_FILES(Makefile)
 AC_CONFIG_FILES(math/Makefile)
 AC_CONFIG_FILES(src/Makefile)
diff -Nur SPR-3.3.1.orig/src/Makefile.am SPR-3.3.1/src/Makefile.am
--- SPR-3.3.1.orig/src/Makefile.am	2008-08-11 18:26:00.000000000 +0100
+++ SPR-3.3.1/src/Makefile.am	2008-12-12 17:47:25.244039219 +0000
@@ -53,14 +53,14 @@
 libSPR_la_CPPFLAGS += $(ROOTCPPFLAGS)
 libSPR_la_CXXFLAGS = $(CXXFLAGS) $(AM_CXXFLAGS) $(ROOTCXXFLAGS)
 libSPR_la_LDFLAGS = $(ROOTLDFLAGS) -lstdc++
-libSPR_la_LIBADD = $(top_srcdir)/math/libSPRmath.la
+libSPR_la_LIBADD = $(top_builddir)/math/libSPRmath.la  $(ROOTLIBS)
 
 #####################################################################
 
 APPCPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir) $(ROOTCPPFLAGS)
 APPCXXFLAGS = $(ROOTCXXFLAGS)
-APPLDFLAGS  = -L/usr/lib -L$(top_srcdir)/src $(ROOTLDFLAGS)
-APPLDADD    = -lSPR $(ROOTLIBS)
+APPLDFLAGS  = $(ROOTLDFLAGS)
+APPLDADD    = libSPR.la
 
 #if WITH_ROOT
 #    APPCPPFLAGS += -DSPRROOTTUPLE
@@ -77,81 +77,53 @@
 AM_CPPFLAGS = $(APPCPPFLAGS)
 AM_CXXFLAGS = $(APPCXXFLAGS)
 AM_LDFLAGS  = $(APPLDFLAGS)
-AM_LIBS     = $(APPLDADD)
+LDADD 	    = $(APPLDADD)
 
-noinst_PROGRAMS = exampleBootstrapVarianceEst exampleCorrTest exampleUserCuts \
+check_PROGRAMS = exampleBootstrapVarianceEst exampleCorrTest exampleUserCuts \
                   exampleMatrixTest
+
 exampleBootstrapVarianceEst_SOURCES = exampleBootstrapVarianceEst.cc
-exampleBootstrapVarianceEst_LDADD   = $(APPLDADD)
 exampleCorrTest_SOURCES             = exampleCorrTest.cc
-exampleCorrTest_LDADD               = $(APPLDADD)
+exampleUserCuts_SOURCES 	    = exampleUserCuts.cc
 exampleMatrixTest_SOURCES           = exampleMatrixTest.cc
-exampleMatrixTest_LDADD             = $(APPLDADD)
 
 bin_PROGRAMS = SprFisherLogitApp             SprDecisionTreeApp         SprExploratoryAnalysisApp \
                SprMultiClassApp              SprGoFDecisionTreeApp      SprAdaBoostBinarySplitApp \
                SprRBFNetApp                  SprAdaBoostDecisionTreeApp SprStdBackpropApp \
                SprCombinerApp                SprBaggerDecisionTreeApp   SprBumpHunterApp  \
                SprInteractiveAnalysisApp     SprOutputAnalyzerApp       SprOutputWriterApp \
-               SprBoosterApp                 SprBaggerApp               exampleUserCuts \
+               SprBoosterApp                 SprBaggerApp               \
                SprVariableImportanceApp      SprIOTestApp               SprAddBaggersApp \
                SprTransformationApp          SprSplitterApp             SprAddNRemoveRApp \
                SprGEPApp                     SprCrossValidatorApp       SprAddColumnsForMCLApp \
 SprIndicatorMatrixApp
 
 SprFisherLogitApp_SOURCES = SprFisherLogitApp.cc
-SprFisherLogitApp_LDADD = $(APPLDADD)
 SprDecisionTreeApp_SOURCES = SprDecisionTreeApp.cc
-SprDecisionTreeApp_LDADD = $(APPLDADD)
 SprExploratoryAnalysisApp_SOURCES = SprExploratoryAnalysisApp.cc
-SprExploratoryAnalysisApp_LDADD = $(APPLDADD)
 SprMultiClassApp_SOURCES = SprMultiClassApp.cc
-SprMultiClassApp_LDADD = $(APPLDADD)
 SprGoFDecisionTreeApp_SOURCES = SprGoFDecisionTreeApp.cc
-SprGoFDecisionTreeApp_LDADD = $(APPLDADD)
 SprAdaBoostBinarySplitApp_SOURCES = SprAdaBoostBinarySplitApp.cc
-SprAdaBoostBinarySplitApp_LDADD = $(APPLDADD)
 SprRBFNetApp_SOURCES = SprRBFNetApp.cc
-SprRBFNetApp_LDADD = $(APPLDADD)
 SprAdaBoostDecisionTreeApp_SOURCES = SprAdaBoostDecisionTreeApp.cc
-SprAdaBoostDecisionTreeApp_LDADD = $(APPLDADD)
 SprStdBackpropApp_SOURCES = SprStdBackpropApp.cc
-SprStdBackpropApp_LDADD = $(APPLDADD)
 SprCombinerApp_SOURCES = SprCombinerApp.cc
-SprCombinerApp_LDADD = $(APPLDADD)
 SprBaggerDecisionTreeApp_SOURCES = SprBaggerDecisionTreeApp.cc
-SprBaggerDecisionTreeApp_LDADD = $(APPLDADD)
 SprBumpHunterApp_SOURCES = SprBumpHunterApp.cc
-SprBumpHunterApp_LDADD = $(APPLDADD)
 SprInteractiveAnalysisApp_SOURCES = SprInteractiveAnalysisApp.cc
-SprInteractiveAnalysisApp_LDADD = $(APPLDADD)
 SprOutputAnalyzerApp_SOURCES = SprOutputAnalyzerApp.cc
-SprOutputAnalyzerApp_LDADD = $(APPLDADD)
 SprOutputWriterApp_SOURCES = SprOutputWriterApp.cc
-SprOutputWriterApp_LDADD = $(APPLDADD)
 SprBoosterApp_SOURCES = SprBoosterApp.cc
-SprBoosterApp_LDADD = $(APPLDADD)
 SprBaggerApp_SOURCES = SprBaggerApp.cc
-SprBaggerApp_LDADD = $(APPLDADD)
-exampleUserCuts_SOURCES = exampleUserCuts.cc
-exampleUserCuts_LDADD = $(APPLDADD)
 SprVariableImportanceApp_SOURCES = SprVariableImportanceApp.cc
-SprVariableImportanceApp_LDADD = $(APPLDADD)
 SprIOTestApp_SOURCES = SprIOTestApp.cc
-SprIOTestApp_LDADD = $(APPLDADD)
 SprAddBaggersApp_SOURCES = SprAddBaggersApp.cc
-SprAddBaggersApp_LDADD = $(APPLDADD)
 SprTransformationApp_SOURCES = SprTransformationApp.cc
-SprTransformationApp_LDADD = $(APPLDADD)
 SprSplitterApp_SOURCES = SprSplitterApp.cc
-SprSplitterApp_LDADD = $(APPLDADD)
 SprAddNRemoveRApp_SOURCES = SprAddNRemoveRApp.cc
-SprAddNRemoveRApp_LDADD = $(APPLDADD)
 SprGEPApp_SOURCES = SprGEPApp.cc
-SprGEPApp_LDADD = $(APPLDADD)
 SprCrossValidatorApp_SOURCES = SprCrossValidatorApp.cc
-SprCrossValidatorApp_LDADD = $(APPLDADD)
 SprAddColumnsForMCLApp_SOURCES = SprAddColumnsForMCLApp.cc
-SprAddColumnsForMCLApp_LDADD = $(APPLDADD)
 SprIndicatorMatrixApp_SOURCES = SprIndicatorMatrixApp.cc
-SprIndicatorMatrixApp_LDADD = $(APPLDADD)
+
+TESTS= $(check_PROGRAMS)
