Index: configure.in
===================================================================
RCS file: /cvsroot/gnuplot/gnuplot/configure.in,v
retrieving revision 1.262
diff -u -r1.262 configure.in
--- configure.in	7 Feb 2010 23:18:14 -0000	1.262
+++ configure.in	14 Feb 2010 14:06:49 -0000
@@ -16,6 +16,7 @@
 
 VERSION_MAJOR="`cat $srcdir/VERSION`"
 PATCHLEVEL="`cat $srcdir/PATCHLEVEL`"
+AC_DEFINE_UNQUOTED([PATCHLEVEL],["$PATCHLEVEL"],[Gnuplot patchlevel])
 
 dnl configure.in body
 
@@ -1145,6 +1146,8 @@
 AC_CHECK_PROGS(LRELEASE, lrelease-qt4 lrelease, no)
 AM_CONDITIONAL(HAVE_LRELEASE, test "${LRELEASE}" != no)
 
+AC_ARG_VAR(DIST_CONTACT,[Contact address for modified and binary distributed gnuplot versions])
+AM_CONDITIONAL(IS_DIST_RELEASE, test -n "${DIST_CONTACT}")
 
 dnl build lisp files
 if test "$with_lisp_files" != no; then
Index: src/plot.c
===================================================================
RCS file: /cvsroot/gnuplot/gnuplot/src/plot.c,v
retrieving revision 1.107
diff -u -r1.107 plot.c
--- src/plot.c	6 Jan 2010 17:29:03 -0000	1.107
+++ src/plot.c	14 Feb 2010 14:06:49 -0000
@@ -375,7 +375,12 @@
 		    "  -p  --persist\n"
 		    "  -e  \"command1; command2; ...\"\n"
 		    "gnuplot %s patchlevel %s\n"
+#ifdef DIST_CONTACT
+		    "Report bugs to "DIST_CONTACT"\n"
+		    "            or %s\n",
+#else
 		    "Report bugs to %s\n",
+#endif
 		    gnuplot_version, gnuplot_patchlevel, bug_email);
 	    return 0;
 
Index: src/Makefile.am
===================================================================
RCS file: /cvsroot/gnuplot/gnuplot/src/Makefile.am,v
retrieving revision 1.77
diff -u -r1.77 Makefile.am
--- src/Makefile.am	7 Feb 2010 23:18:14 -0000	1.77
+++ src/Makefile.am	14 Feb 2010 14:06:49 -0000
@@ -36,6 +36,10 @@
 
 AM_CPPFLAGS = -I../term -I$(top_srcdir)/term -DBINDIR=\"$(bindir)\" -DX11_DRIVER_DIR=\"$(X11_DRIVER_DIR)\" -DGNUPLOT_PS_DIR=\"$(GNUPLOT_PS_DIR)\" -DGNUPLOT_JS_DIR=\"$(GNUPLOT_JS_DIR)\" -DGNUPLOT_LUA_DIR=\"$(GNUPLOT_LUA_DIR)\"  -DCONTACT=\"$(EMAIL)\" -DHELPFILE=\"$(HELPFILE)\" -DGNUPLOT_X11=\"$(GNUPLOT_X11)\" -DXAPPLRESDIR=\"$(XAPPLRESDIR)\"
 
+if IS_DIST_RELEASE
+AM_CPPFLAGS += -DDIST_CONTACT="\"$(DIST_CONTACT)\""
+endif
+
 gnuplot_SOURCES = alloc.c alloc.h ansichek.h axis.c axis.h binary.c \
 breaders.c breaders.h bitmap.c bitmap.h color.c color.h command.c \
 command.h contour.c contour.h datafile.c datafile.h dynarray.c dynarray.h \
Index: src/version.c
===================================================================
RCS file: /cvsroot/gnuplot/gnuplot/src/version.c,v
retrieving revision 1.96
diff -u -r1.96 version.c
--- src/version.c	3 Feb 2010 10:00:46 -0000	1.96
+++ src/version.c	14 Feb 2010 14:06:49 -0000
@@ -40,7 +40,7 @@
 
 
 const char gnuplot_version[] = "4.5";
-const char gnuplot_patchlevel[] = "0";
+const char gnuplot_patchlevel[] = PATCHLEVEL;
 const char gnuplot_date[] = "February 2010";
 const char gnuplot_copyright[] = "Copyright (C) 1986-1993, 1998, 2004, 2007-2010";
 
