diff -Naur gnuplot-4.4.0-rc1.orig/share/LaTeX/Makefile.am gnuplot-4.4.0-rc1/share/LaTeX/Makefile.am
--- gnuplot-4.4.0-rc1.orig/share/LaTeX/Makefile.am	2009-12-13 14:16:41.000000000 +0100
+++ gnuplot-4.4.0-rc1/share/LaTeX/Makefile.am	2010-01-08 20:30:20.000000000 +0100
@@ -11,12 +11,10 @@
 	  installdir=`kpsexpand ${prefix}/usr/'$$TEXMFLOCAL'/tex/latex/gnuplot |sed "s%^.*share/%$(prefix)/share/%"` ; \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$$installdir ; \
 	  if [ ! -f $(DESTDIR)$$installdir/gnuplot.cfg ]; then \
-	    $(INSTALL_DATA) $(srcdir)/gnuplot.cfg $(DESTDIR)$$installdir/gnuplot.cfg && \
-	    texhash; \
+	    $(INSTALL_DATA) $(srcdir)/gnuplot.cfg $(DESTDIR)$$installdir/gnuplot.cfg; \
 	  fi ; \
 	  if [ ! -f $(DESTDIR)$$installdir/gnuplot-lua-tikz.sty ]; then \
-	    $(INSTALL_DATA) $(srcdir)/gnuplot-lua-tikz.sty $(DESTDIR)$$installdir/gnuplot-lua-tikz.sty && \
-	    texhash; \
+	    $(INSTALL_DATA) $(srcdir)/gnuplot-lua-tikz.sty $(DESTDIR)$$installdir/gnuplot-lua-tikz.sty; \
 	  fi ; \
 	  fi
 
