diff -durr mrfast-2.1.0.3.orig/MrFAST.c mrfast-2.1.0.3/MrFAST.c
--- mrfast-2.1.0.3.orig/MrFAST.c	2011-12-18 17:48:16.431973604 -0500
+++ mrfast-2.1.0.3/MrFAST.c	2011-12-18 17:55:11.400723582 -0500
@@ -7358,6 +7358,7 @@
 	char fname1[FILE_NAME_LENGTH];
 	char fname2[FILE_NAME_LENGTH];
 	char l;
+	int l_int;
 	int loc1, loc2;
 	int err1, err2;
 	char dir1, dir2;
@@ -7399,28 +7400,28 @@
 
 		//tmp = fwrite (&(mi2[k].cigarSize), sizeof(int), 1, out);
 		
-		tmp = fread(&l, sizeof(int), 1, in);
-		tmp = fread(cigar1, sizeof(char), l, in);
-		cigar1[(int)l]='\0';
+		tmp = fread(&l_int, sizeof(int), 1, in);
+		tmp = fread(cigar1, sizeof(char), l_int, in);
+		cigar1[l_int]='\0';
 		//tmp = fwrite ((mi2[k].cigar), sizeof(char), mi2[k].cigarSize, out);
 
 		//tmp = fwrite (&(mi2[k].mdSize), sizeof(int), 1, out);
-		tmp = fread(&l, sizeof(int), 1, in);
-		tmp = fread(editString1, sizeof(char), l, in);
-		editString1[(int)l]='\0';
+		tmp = fread(&l_int, sizeof(int), 1, in);
+		tmp = fread(editString1, sizeof(char), l_int, in);
+		editString1[l_int]='\0';
 		//tmp = fwrite ((mi2[k].md), sizeof(char), mi2[k].mdSize, out);
 
 		tmp = fread(&loc2, sizeof(int), 1, in);
 		tmp = fread(&err2, sizeof(int), 1, in);
 		tmp = fread(&sc2, sizeof(float), 1, in);
 
-		tmp = fread(&l, sizeof(int), 1, in);
-		tmp = fread(cigar2, sizeof(char), l, in);
-		cigar2[(int)l]='\0';
+		tmp = fread(&l_int, sizeof(int), 1, in);
+		tmp = fread(cigar2, sizeof(char), l_int, in);
+		cigar2[l_int]='\0';
 		/* FARHAD:  l is of type char; and you fread into it with sizeof(char) above; but as sizeof(int) here; are you sure? 1 byte vs. 4 bytes */
-		tmp = fread(&l, sizeof(int), 1, in);
-		tmp = fread(editString2, sizeof(char), l, in);
-		editString2[(int)l]='\0';
+		tmp = fread(&l_int, sizeof(int), 1, in);
+		tmp = fread(editString2, sizeof(char), l_int, in);
+		editString2[l_int]='\0';
 	
 		convertMDCigarToMatrix(cigar1, strlen(cigar1), editString1, strlen(editString1), tmp_matrix1);
 		convertMDCigarToMatrix(cigar2, strlen(cigar2), editString2, strlen(editString2), tmp_matrix2);
