--- /var/tmp/portage/octave-epstk-2.1/work/epstk21/m/einit.m	2005-04-08 03:47:39.000000000 -0400
+++ /var/tmp/portage/octave-epstk-2.1/work/epstk21/m/einit.m	2006-03-20 13:04:55.000000000 -0500
@@ -6,19 +6,19 @@
 %%
 
 % Path of epstk-mfiles
-ePath='./';  % local directory
-%ePath='/usr/share/octave/site/m/octave-epstk/';%or e.g. on a linux system
+%ePath='./';  % local directory
+ePath='/usr/share/octave/site/m/octave-epstk/';%or e.g. on a linux system
 
 % Read user-parameter-file (octave on linux system)
-%eInitUserFile=[system('echo -n $HOME') '/.epstkInit.m'];
-%if exist(eInitUserFile)~=2
-%  [userParaText n]=etxtread([ePath 'einit.m']); 
-%  userParaStartPos=findstr(userParaText(1:1000),'% Parameter for epstk');
-%  etxtwrit(userParaText(userParaStartPos(2):n),eInitUserFile);
-%  clear userParaText
-%end
-%source(eInitUserFile); 
-%return
+eInitUserFile=[system('echo -n $HOME') '/.epstkInit.m'];
+if exist(eInitUserFile)~=2
+  [userParaText n]=etxtread([ePath 'einit.m']); 
+  userParaStartPos=findstr(userParaText(1:1000),'% Parameter for epstk');
+  etxtwrit(userParaText(userParaStartPos(2):n),eInitUserFile);
+  clear userParaText
+end
+source(eInitUserFile); 
+return
 
 % Parameter for epstk
 % written by stefan.mueller@fgan.de (C) 2005
