diff --git a/include/ccp4.setup-bash b/include/ccp4.setup-bash
index 9848101..d76fe3b 100755
--- a/include/ccp4.setup-bash
+++ b/include/ccp4.setup-bash
@@ -37,8 +37,8 @@ fi
 # This is usually the directory in which you ran the tar command to unpack the
 # code, and is assumed to be shared between machines at a multi-machine site.
 
-setenv CCP4_MASTER	/xtal                    
-setenv CCP4		$CCP4_MASTER/ccp4-6.1.2
+setenv CCP4_MASTER /usr
+setenv CCP4	$CCP4_MASTER
 
 # Check for existence of CCP4_MASTER
 if (! test -d $CCP4_MASTER) ; then
@@ -54,7 +54,7 @@ fi
 # files. A dedicated scratch filesystem is probably better than (/usr)/tmp
 # BINSORT_SCR: a scratch directory for binsort's use; normally same as CCP4_SCR
 
-setenv CCP4_SCR      /tmp/$USER	
+setenv CCP4_SCR     /tmp/$USER	
 # check to see if this exists and if not try to make it
 test -d $CCP4_SCR || mkdir $CCP4_SCR
 test -d $CCP4_SCR || \
@@ -70,7 +70,7 @@ setenv CCP4I_TOP ${CCP4}/ccp4i
 #               For 'standard' installations this is /usr/local/bin
 #               but note the SGI distributed version of Tcl/Tk is not 
 #               appropriate version
-setenv CCP4I_TCLTK /usr/local/bin        
+setenv CCP4I_TCLTK /usr/bin        
 # CCP4I_HELP - directory contain ccp4i help - default is $CCP4I_TOP/help
 setenv CCP4I_HELP ${CCP4I_TOP}/help
 
@@ -128,7 +128,7 @@ setenv GFORTRAN_UNBUFFERED_ALL Y
 #                                       # for multiple systems
 #   foo)
     setenv CBIN      $CCP4/bin		
-    setenv CLIB      $CCP4/lib	
+    setenv CLIB      $CCP4/GENTOOLIBDIR
 
     setenv CCP4_BROWSER  firefox             
 
@@ -188,9 +188,9 @@ fi
 #Python path for the subprocess fixed module
 
 if [ $ccp4_first_in_path -eq 1 ]; then
-    setenv PYTHONPATH ${CCP4}/share/python:${PYTHONPATH}
+    setenv PYTHONPATH ${CCP4}/share/ccp4/smartie:${CCP4I_TOP}/python:${PYTHONPATH}
 else
-    setenv PYTHONPATH ${PYTHONPATH}:${CCP4}/share/python
+    setenv PYTHONPATH ${PYTHONPATH}:${CCP4}/share/ccp4/smartie:${CCP4I_TOP}/python
 fi
 
 ### PLOT_COMMAND PRINT_COMMAND for the XCCPJIFFY programs to compile ###
@@ -292,14 +292,12 @@ test -r $CCP4/include/ccp4.msg && cat $CCP4/include/ccp4.msg
 
 ### dbCCP4i database handler setup ###
 
-setenv DBCCP4I_TOP ${CCP4}/share/dbccp4i
+setenv DBCCP4I_TOP ${CCP4}/share/ccp4/dbccp4i
 
 ### IMOSFLM environment is now defined here
 
 setenv MOSFLM_EXEC ${CBIN}/ipmosflm
 setenv MOSFLM_WISH ${CCP4I_TCLTK}/wish
-setenv IMOSFLM_VERSION "1.0.0"
-
 
 ### MMCIFDIC is now defined here. cif_mmdic.lib is a binary file ===
 # so on multiple installs it need to be unique and not in $CLIBD
@@ -309,21 +307,13 @@ setenv MMCIFDIC $CLIB/cif_mmdic.lib
 ### MOLREPLIB Point to the location of the files bs-zeros.dat  ps.resource  symlib.blc
 # They are distributed with CCP4 in $CLIBD. This env must have the trailing \ or /
 
-setenv MOLREPLIB $CCP4/lib/data/monomers/
+setenv MOLREPLIB $CCP4/share/ccp4/data/monomers/
 
 ### CRANK points to location of Crank automation suite within ccp4i
 
-setenv CRANK $CCP4/ccp4i/crank
+setenv CRANK $CCP4I_TOP/bin/crank
 setenv CCP4_CRANK 1
 
-### XIA2 setup
-# XIAROOT directory (default is under the share folder)
-
-if test -e $CCP4/share/XIAROOT/setup.sh
-then
-. $CCP4/share/XIAROOT/setup.sh
-fi
-
 # Note: The PATH is updated with the xia2 Applications directory in the 
 # construct below.
 
@@ -339,7 +329,7 @@ fi
 # They are distributed with CCP4 in $CLIBD/monomers. This env must have 
 # the trailing \ or /
 
-setenv CLIBD_MON $CCP4/lib/data/monomers/
+setenv CLIBD_MON $CCP4/share/ccp4/data/monomers/
 
 ### PUBLIC_FONT84, CCP4_HELPDIR, HELPDIR, MOSHELPFILE ###
 # (essential for the relevant programs)
@@ -347,10 +337,10 @@ setenv PUBLIC_FONT84   $CLIB/font84.dat  # plot84 fonts [used by plot84lib]
 
 # location of VMS-style help files required for mosflm [used by hlplib, chelp,
 #   chlp, (ip)mosflm]:
-setenv CCP4_HELPDIR    $CCP4/help/            # NB trailing /
+setenv CCP4_HELPDIR    $CCP4I_TOP/help/            # NB trailing /
 
 # Rasmol path. Needed for prog. to find rasmol.hlp
-setenv RASMOLPATH $CCP4/x-windows/RasMol/src
+setenv RASMOLPATH $CCP4/GENTOOLIBDIR/rasmol
 
 # CCP4 executables and scripts live in $CBIN and $CETC respectively; put them
 # on the path in an appropriate order for scripts to be used as wrappers for
@@ -373,32 +363,14 @@ done
 
 # Optional (useful) additional environment variables
   setenv CCP4             $CCP4
-  setenv CDOC             $CCP4/doc
-  setenv CHTML            $CCP4/html
-  setenv CETC             $CCP4/etc
-  setenv CEXAM            $CCP4/examples
+  setenv CDOC             $DBCCP4I_TOP/doc
+  setenv CHTML            $CCP4I_TOP/html
+  setenv CETC             $CCP4I_TOP/etc
+  setenv CEXAM            $CLIB/ccp4/examples
   setenv CINCL            $CCP4/include
-  setenv CLIBD            $CCP4/lib/data
-  setenv CLIBS            $CCP4/lib/src
-  setenv CPROG            $CCP4/src
+  setenv CLIBD            $CCP4/share/ccp4/data
 
 if [ "`basename $SHELL`" != "ksh" ]; then
-# useful aliases, at least for developers
-  alias ccp4   'pushd $CCP4>/dev/null'
-  alias xtal   'pushd $CCP4_MASTER>/dev/null'
-  alias cbin   'pushd $CBIN>/dev/null'
-  alias cetc   'pushd $CETC>/dev/null'
-  alias cprog  'pushd $CPROG>/dev/null'
-  alias cincl  'pushd $CINCL>/dev/null'
-  alias clib   'pushd $CLIB>/dev/null'
-  alias clibd  'pushd $CLIBD>/dev/null'
-  alias clibs  'pushd $CLIBS>/dev/null'
-  alias cbin   'pushd $CBIN>/dev/null'
-  alias cexam  'pushd $CEXAM>/dev/null'
-  alias cdoc   'pushd $CDOC>/dev/null'
-  alias chtml  'pushd $CHTML>/dev/null'
-  alias ccp4help '$CCP4_BROWSER $CHTML/INDEX.html \&'
-
 ### laue, lget, linkimages ###
 
 ### only if we have the Laue software suite...
@@ -430,23 +402,6 @@ if [ "`basename $SHELL`" != "ksh" ]; then
     alias lbin   'pushd $CBIN>/dev/null'
 
   fi
-
+else
   unset alias  # clean up
 fi             # ksh test
-
-# Set-up cctbx environment
-test -r $CCP4/lib/cctbx/cctbx_build/setpaths.sh && . $CCP4/lib/cctbx/cctbx_build/setpaths.sh
-
-# Set-up phaser environment
-if ( test -d $CCP4/src/phaser) ; then
-  phaser_mtype=`$CCP4/src/phaser/bin/machine_type`
-  phaser_version=`grep PHASER_VERSION $CCP4/src/phaser/conf/version.sh | awk '{print $3}'`
-  phaser_setup_file="${CCP4}/src/phaser/phaser-${phaser_version}/build/${phaser_mtype}/setpaths.sh"
-  test -r $phaser_setup_file && . $phaser_setup_file
-fi
-
-#BALBES root configuration
-if ( test -d $CCP4/share/balbes/BALBES_0.0.1)
-then
-  setenv BALBES_ROOT $CCP4/share/balbes/BALBES_0.0.1
-fi
diff --git a/include/ccp4.setup-csh b/include/ccp4.setup-csh
index 42f0820..ed3c671 100755
--- a/include/ccp4.setup-csh
+++ b/include/ccp4.setup-csh
@@ -24,8 +24,8 @@
 # This is usually the directory in which you ran the tar command to unpack the
 # code, and is assumed to be shared between machines at a multi-machine site.
 
-setenv CCP4_MASTER	/xtal                  
-setenv CCP4		$CCP4_MASTER/ccp4-6.1.2
+setenv CCP4_MASTER	/usr
+setenv CCP4		$CCP4_MASTER
 
 # Check for existence of CCP4_MASTER
 if (! -e $CCP4_MASTER) then
@@ -57,7 +57,7 @@ setenv CCP4I_TOP ${CCP4}/ccp4i
 #               For 'standard' installations this is /usr/local/bin
 #               but note the SGI distributed version of Tcl/Tk is not 
 #               appropriate version
-setenv CCP4I_TCLTK /usr/local/bin        
+setenv CCP4I_TCLTK /usr/bin        
 # CCP4I_HELP - directory contain ccp4i help - default is $CCP4I_TOP/help
 setenv CCP4I_HELP ${CCP4I_TOP}/help
 
@@ -115,7 +115,7 @@ setenv GFORTRAN_UNBUFFERED_ALL Y
 #                                       # for multiple systems
 #   case 'foo':
     setenv CBIN      $CCP4/bin		
-    setenv CLIB      $CCP4/lib	
+    setenv CLIB      $CCP4/GENTOOLIBDIR	
 
     setenv CCP4_BROWSER  firefox          
 
@@ -180,9 +180,9 @@ endif
 #Python path for the subprocess fixed module
 if (${?PYTHONPATH}) then
   if ($ccp4_first_in_path) then
-      setenv PYTHONPATH ${CCP4}/share/python:${PYTHONPATH}
+      setenv PYTHONPATH ${CCP4}/share/ccp4/smartie:${CCP4I_TOP}/python:${PYTHONPATH}
   else
-      setenv PYTHONPATH ${PYTHONPATH}:${CCP4}/share/python
+      setenv PYTHONPATH ${PYTHONPATH}:${CCP4}/share/ccp4/smartie:${CCP4I_TOP}/python
   endif
 else
     setenv PYTHONPATH ${CCP4}/share/python
@@ -284,13 +284,12 @@ if (-e $CCP4/include/ccp4.msg) cat $CCP4/include/ccp4.msg
 
 ### dbCCP4i database handler setup ###
 
-setenv DBCCP4I_TOP ${CCP4}/share/dbccp4i
+setenv DBCCP4I_TOP ${CCP4}/share/ccp4/dbccp4i
 
 ### IMOSFLM environment is now defined here
 
 setenv MOSFLM_EXEC ${CBIN}/ipmosflm
 setenv MOSFLM_WISH ${CCP4I_TCLTK}/wish
-setenv IMOSFLM_VERSION "1.0.0"
 
 ### MMCIFDIC is now defined here. cif_mmdic.lib is a binary file ===
 # so on multiple installs it need to be unique and not in $CLIBD
@@ -300,20 +299,13 @@ setenv MMCIFDIC $CLIB/cif_mmdic.lib
 ### MOLREPLIB Point to the location of the files bs-zeros.dat  ps.resource  symlib.blc
 # They are distributed with CCP4 in $CLIBD. This env must have the trailing \ or /
 
-setenv MOLREPLIB $CCP4/lib/data/monomers/
+setenv MOLREPLIB $CCP4/share/ccp4/data/monomers/
 
 ### CRANK points to location of Crank automation suite within ccp4i
 
-setenv CRANK $CCP4/ccp4i/crank
+setenv CRANK $CCP4I_TOP/bin/crank
 setenv CCP4_CRANK 1
 
-### XIA2 setup
-# XIAROOT directory (default is under the share folder)
-
-if (-e $CCP4/share/XIAROOT/setup.csh) then
-   source $CCP4/share/XIAROOT/setup.csh
-endif
-
 # Note: The PATH is updated with the xia2 Applications directory in the 
 # construct below.
 
@@ -328,7 +320,7 @@ endif
 # They are distributed with CCP4 in $CLIBD/monomers. This env must have 
 # the trailing \ or /
 
-setenv CLIBD_MON $CCP4/lib/data/monomers/
+setenv CLIBD_MON $CCP4/share/ccp4/data/monomers/
 
 ### PUBLIC_FONT84, CCP4_HELPDIR, HELPDIR, MOSHELPFILE ###
 # (essential for the relevant programs)
@@ -339,7 +331,7 @@ setenv PUBLIC_FONT84   $CLIB/font84.dat  # plot84 fonts [used by plot84lib]
 setenv CCP4_HELPDIR    $CCP4/help/            # NB trailing /
 
 # Rasmol path. Needed for prog. to find rasmol.hlp
-setenv RASMOLPATH $CCP4/x-windows/RasMol/src
+setenv RASMOLPATH $CCP4/GENTOOLIBDIR/rasmol
 
 # CCP4 executables and scripts live in $CBIN and $CETC respectively; put them
 # on the path in an appropriate order for scripts to be used as wrappers for
@@ -366,30 +358,12 @@ end
 
 # Optional (useful) additional environment variables
   setenv CCP4             $CCP4
-  setenv CDOC             $CCP4/doc
-  setenv CHTML            $CCP4/html
-  setenv CETC             $CCP4/etc
-  setenv CEXAM            $CCP4/examples
+  setenv CDOC             $DBCCP4I_TOP/doc
+  setenv CHTML            $CCP4I_TOP/html
+  setenv CETC             $CCP4I_TOP/etc
+  setenv CEXAM            $CLIB/ccp4/examples
   setenv CINCL            $CCP4/include
-  setenv CLIBD            $CCP4/lib/data
-  setenv CLIBS            $CCP4/lib/src
-  setenv CPROG            $CCP4/src
-
-# useful aliases, at least for developers
-  alias ccp4   'pushd $CCP4>/dev/null'
-  alias xtal   'pushd $CCP4_MASTER>/dev/null'
-  alias cbin   'pushd $CBIN>/dev/null'
-  alias cetc   'pushd $CETC>/dev/null'
-  alias cprog  'pushd $CPROG>/dev/null'
-  alias cincl  'pushd $CINCL>/dev/null'
-  alias clib   'pushd $CLIB>/dev/null'
-  alias clibd  'pushd $CLIBD>/dev/null'
-  alias clibs  'pushd $CLIBS>/dev/null'
-  alias cbin   'pushd $CBIN>/dev/null'
-  alias cexam  'pushd $CEXAM>/dev/null'
-  alias cdoc   'pushd $CDOC>/dev/null'
-  alias chtml  'pushd $CHTML>/dev/null'
-  alias ccp4help '$CCP4_BROWSER $CHTML/INDEX.html &'
+  setenv CLIBD            $CCP4/share/ccp4/data
 
 ### laue, lget, linkimages ###
 
@@ -421,19 +395,3 @@ end
 #
 
   endif 
-
-# Set-up cctbx environment
-if (-e $CCP4/lib/cctbx/cctbx_build/setpaths.csh) source $CCP4/lib/cctbx/cctbx_build/setpaths.csh
-
-# Set-up phaser environment
-if (-d $CCP4/src/phaser) then
-  set phaser_mtype = `$CCP4/src/phaser/bin/machine_type`
-  set phaser_version = `grep PHASER_VERSION $CCP4/src/phaser/conf/version.csh | awk '{print $3}'`
-  set phaser_setup_file = "${CCP4}/src/phaser/phaser-${phaser_version}/build/${phaser_mtype}/setpaths.csh"
-  if (-e $phaser_setup_file) source $phaser_setup_file
-endif
-
-#BALBES root configuration
-if ( -d $CCP4/share/balbes/BALBES_0.0.1) then
-  setenv BALBES_ROOT $CCP4/share/balbes/BALBES_0.0.1
-endif
