--- src/rapper/LOOP/Makefile.in	2008-05-29 16:39:40.000000000 +0200
+++ src/rapper/LOOP/Makefile.in.new	2009-03-12 20:40:03.157602172 +0100
@@ -111,9 +111,9 @@
 pyexecdir = @pyexecdir@
 pythondir = @pythondir@
 rapper_LDFLAGS = @rapper_LDFLAGS@
-bin_PROGRAMS = rapper
+bin_PROGRAMS = rappermc
 
-rapper_SOURCES = \
+rappermc_SOURCES = \
 kip.h aaconv.cpp atomconv.cpp kip_snake.cpp \
 kip_pdb.cpp measure.cpp lib_frags.cpp ran2.cpp \
 heap.cpp queue.cpp 3Dgeom.cpp space_grid.cpp list.cpp search_residue.cpp bitfield.cpp \
@@ -129,7 +129,7 @@
 residue_range.cpp pop_search_frags.cpp saxs.cpp lib_choral.cpp pdf.cpp WeightedDistribMean.cpp
 
 
-rapper_LDADD = ../gc7.0/libgc.la ../libxml2/libxml2.la
+rappermc_LDADD = ../gc7.0/libgc.la ../libxml2/libxml2.la
 dist_pkgdata_DATA = params.xml
 
 INCLUDES = -I. -I../gc7.0/include -I../libxml2/include
@@ -139,7 +139,7 @@
 bin_PROGRAMS = rapper$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS)
 
-am_rapper_OBJECTS = aaconv.$(OBJEXT) atomconv.$(OBJEXT) \
+am_rappermc_OBJECTS = aaconv.$(OBJEXT) atomconv.$(OBJEXT) \
 	kip_snake.$(OBJEXT) kip_pdb.$(OBJEXT) measure.$(OBJEXT) \
 	lib_frags.$(OBJEXT) ran2.$(OBJEXT) heap.$(OBJEXT) \
 	queue.$(OBJEXT) 3Dgeom.$(OBJEXT) space_grid.$(OBJEXT) \
@@ -164,8 +164,8 @@
 	iterator.$(OBJEXT) residue_range.$(OBJEXT) \
 	pop_search_frags.$(OBJEXT) saxs.$(OBJEXT) lib_choral.$(OBJEXT) \
 	pdf.$(OBJEXT) WeightedDistribMean.$(OBJEXT)
-rapper_OBJECTS = $(am_rapper_OBJECTS)
-rapper_DEPENDENCIES = ../gc7.0/libgc.la ../libxml2/libxml2.la
+rappermc_OBJECTS = $(am_rapper_OBJECTS)
+rappermc_DEPENDENCIES = ../gc7.0/libgc.la ../libxml2/libxml2.la
 
 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir)
@@ -233,11 +233,11 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-DIST_SOURCES = $(rapper_SOURCES)
+DIST_SOURCES = $(rappermc_SOURCES)
 DATA = $(dist_pkgdata_DATA)
 
 DIST_COMMON = $(dist_pkgdata_DATA) Makefile.am Makefile.in
-SOURCES = $(rapper_SOURCES)
+SOURCES = $(rappermc_SOURCES)
 
 all: all-am
 
--- src/rapper/LOOP/Makefile.am 2008-01-24 15:08:17.000000000 +0100
+++ src/rapper/LOOP/Makefile.am.new     2009-03-12 23:01:56.697533874 +0100
@@ -1,6 +1,6 @@
-bin_PROGRAMS = rapper
+bin_PROGRAMS = rappermc
 
-rapper_SOURCES = \
+rappermc_SOURCES = \
 kip.h aaconv.cpp atomconv.cpp kip_snake.cpp \
 kip_pdb.cpp measure.cpp lib_frags.cpp ran2.cpp \
 heap.cpp queue.cpp 3Dgeom.cpp space_grid.cpp list.cpp search_residue.cpp bitfield.cpp \
@@ -15,7 +15,7 @@
 hash_table.cpp surface.cpp parse.cpp band.cpp edensity.cpp read_map.cpp iterator.cpp \
 residue_range.cpp pop_search_frags.cpp saxs.cpp lib_choral.cpp pdf.cpp WeightedDistribMean.cpp
 
-rapper_LDADD = ../gc7.0/libgc.la ../libxml2/libxml2.la
+rappermc_LDADD = ../gc7.0/libgc.la ../libxml2/libxml2.la
 
 pkgdatadir = $(datadir)/rapper
 dist_pkgdata_DATA = params.xml
