coreutils now installs a binary called truncate.

--- src/Makefile.in.orig	2009-03-02 23:25:45.000000000 -0800
+++ src/Makefile.in	2009-03-02 23:25:53.000000000 -0800
@@ -41,7 +41,7 @@
 professs r500 rantan rebatch reindex revise rfcorr rotamer rotmat \
 rstats rwcontents sapi scaleit scalepack2mtz seqwt sfall sigmaa sortmtz sortwater \
 stgrid stnet surface tffc tracer \
-truncate unique vecref vectors volume watertidy watncs watpeak \
+ftruncate unique vecref vectors volume watertidy watncs watpeak \
 wilson $(CIFTARGETS)
 
 BALBESTARGETS = search_DB get_structure_DB f2cif check_file_DB check_cell_sg \
@@ -982,7 +982,7 @@
 surface:	$(srcdir)/surface.f
 tffc:		$(srcdir)/tffc.f
 tracer:		$(srcdir)/tracer.f
-truncate:	$(srcdir)/truncate.f
+ftruncate:	$(srcdir)/truncate.f
 unique:		$(srcdir)/unique.f
 vecref:		$(srcdir)/vecref.f
 vectors:	$(srcdir)/vectors.f
--- examples/unix/runnable/truncate.exam.orig	2009-03-02 23:35:04.000000000 -0800
+++ examples/unix/runnable/truncate.exam	2009-03-02 23:35:11.000000000 -0800
@@ -15,7 +15,7 @@
   exit 1
 fi
 #
-truncate hklin $CCP4_SCR/aucn_mrg.mtz \
+ftruncate hklin $CCP4_SCR/aucn_mrg.mtz \
    hklout $CCP4_SCR/aucn_trn.mtz <<EOF-trunc
 title DMSO red aucn2 Data - mosflm
 truncate yes
@@ -28,7 +28,7 @@
 # do this if you already have Fs but want to look at the
 # output graphs of TRUNCATE
 
-truncate hklin $CEXAM/rnase/rnase18.mtz <<EOF
+ftruncate hklin $CEXAM/rnase/rnase18.mtz <<EOF
 title Analysis of rnase data
 labin F=FNAT SIGF=SIGFNAT
 nresidue 192
