--- EMBOSS-6.3.1/ajax/graphics/ajgraph.c.ori	2014-07-04 18:25:18.000000000 +0200
+++ EMBOSS-6.3.1/ajax/graphics/ajgraph.c	2014-07-04 18:25:32.000000000 +0200
@@ -1329,8 +1329,8 @@
 
 void ajGraphicsSetPenwidth(float penwidth)
 {
-    ajDebug("=g= plwid(%.2f) [width]\n", penwidth);
-    plwid((PLINT)penwidth);
+    ajDebug("=g= plwidth(%.2f) [width]\n", penwidth);
+    plwidth((PLINT)penwidth);
 
     return;
 }
