--- EMBOSS-6.3.1/configure.in	2013-09-11 03:55:15.510117066 +0200
+++ configure.in	2013-09-11 03:48:10.410115955 +0200
@@ -449,7 +447,8 @@
     if test "`uname -a | grep Darwin`"; then
     OSXX=`sw_vers -productVersion | sed 's/\(10\.[[0-9]]*\).*/\1/'`
     if test ${OSXX} '>' '10.4'; then
-    CFLAGS="$CFLAGS -I/usr/X11/include -L/usr/X11/lib"
+    CPPFLAGS="$CPPFLAGS -I/usr/X11/include"
+    LDFLAGS="$LDFLAGS -L/usr/X11/lib"
     fi
     fi
     fi
@@ -461,7 +460,7 @@
 AM_CONDITIONAL(ESYSTEMLIBS, test "$have_systemlibs" = "yes")
 AC_SUBST(ESYSTEMLIBS)
 AC_SUBST(DEFINE_USE_SYSTEM_PCRE)
-AC_SUBST(PLPLOT_CFLAGS)
+AC_SUBST(PLPLOT_CPPFLAGS)
 AC_SUBST(PLPLOT_LIBS)
 
 
--- EMBOSS-6.3.1/ajax/graphics/Makefile.am	2013-09-11 03:27:30.110112715 +0200
+++ ajax/graphics/Makefile.am	2013-09-11 03:58:50.860117628 +0200
@@ -19,7 +19,7 @@
 
 
 INCLUDES = \
-$(EXTRA_INCLUDES) -I$(top_srcdir)/ajax/core
+$(EXTRA_INCLUDES) -I$(top_srcdir)/ajax/core -I$(top_srcdir)/plplot
 
 AJAXGSRC = ajgraph.c ajhist.c
 
--- EMBOSS-6.3.1/ajax/acd/Makefile.am	2013-09-11 03:27:30.110112715 +0200
+++ ajax/acd/Makefile.am	2013-09-11 03:59:10.310117679 +0200
@@ -24,7 +24,8 @@
 -I$(top_srcdir)/ajax/core \
 -I$(top_srcdir)/ajax/graphics \
 -I$(top_srcdir)/ajax/ensembl \
--I$(top_srcdir)/ajax/ajaxdb
+-I$(top_srcdir)/ajax/ajaxdb \
+-I$(top_srcdir)/plplot
 
 ACDSRC = ajacd.c
 
--- EMBOSS-6.3.1/nucleus/Makefile.am	2013-09-11 03:27:30.110112715 +0200
+++ nucleus/Makefile.am	2013-09-11 03:59:28.630117727 +0200
@@ -32,7 +32,7 @@
 	$(EXTRA_INCLUDES) \
 	-I$(top_srcdir)/ajax/core -I$(top_srcdir)/ajax/graphics \
 	-I$(top_srcdir)/ajax/ensembl -I$(top_srcdir)/ajax/ajaxdb \
-	-I$(top_srcdir)/ajax/acd
+	-I$(top_srcdir)/ajax/acd -I$(top_srcdir)/plplot
 
 NUCLSRC =  embaln.c embcom.c embcons.c embdata.c embdbi.c embdmx.c \
 	embdomain.c embest.c embexit.c embgroup.c embiep.c embindex.c \
--- EMBOSS-6.3.1/emboss/Makefile.am	2013-09-11 03:27:30.110112715 +0200
+++ emboss/Makefile.am	2013-09-11 03:59:46.340117773 +0200
@@ -89,7 +89,7 @@
 	$(EXTRA_INCLUDES) \
 	-I$(top_srcdir)/ajax/core -I$(top_srcdir)/ajax/graphics \
 	-I$(top_srcdir)/ajax/ensembl -I$(top_srcdir)/ajax/ajaxdb \
-	-I$(top_srcdir)/ajax/acd
+	-I$(top_srcdir)/ajax/acd -I$(top_srcdir)/plplot
 
 
 aaindexextract_SOURCES = aaindexextract.c
--- EMBOSS-6.3.1/Makefile.am	2013-09-11 04:19:01.770120791 +0200
+++ Makefile.am	2013-09-11 04:19:18.460120835 +0200
@@ -13,8 +13,8 @@
 JEMBOSS_DIR = jemboss
 endif
 
-SUBDIRS = $(EXTRA_DIRS) ajax nucleus emboss test doc $(JEMBOSS_DIR)
-DIST_SUBDIRS = $(EXTRA_DIRS) ajax nucleus emboss test doc $(JEMBOSS_DIR)
+SUBDIRS = $(EXTRA_DIRS) plplot ajax nucleus emboss test doc $(JEMBOSS_DIR)
+DIST_SUBDIRS = $(EXTRA_DIRS) plplot ajax nucleus emboss test doc $(JEMBOSS_DIR)
 
 # AJAX_FIXED_ROOT = \"`pwd`/emboss/acd\"
 
--- EMBOSS-6.3.1/plplot/Makefile.am	2013-09-11 09:34:44.620170277 +0200
+++ plplot/Makefile.am	2013-09-11 09:35:35.820170411 +0200
@@ -10,26 +10,27 @@
 if AMPNG
 
 if AMPDF
-TESTPDF_CFLAGS = -Wl,-G -lhpdf
+TESTPDF_LIBS = -Wl,-G -lhpdf
 else
-TESTPDF_CFLAGS = -Wl,-G
+TESTPDF_LIBS = -Wl,-G
 endif
 
-TESTPNG_CFLAGS = $(TESTPDF_CFLAGS) -lgd -lpng -lz -lX11 -lm
+TESTPNG_LIBS = $(TESTPDF_LIBS) -lgd -lpng -lz -lX11 -lm
 else
-TESTPNG_CFLAGS = $(TESTPDF_CFLAGS) -lX11
+TESTPNG_LIBS = $(TESTPDF_LIBS) -lX11
 endif
 endif
 endif
 
-AM_CFLAGS = $(TESTPNG_CFLAGS)
+AM_LIBS = $(TESTPNG_LIBS)
 
 if ISCYGWIN
 CYGWIN_LDFLAGS = -no-undefined -Wl,--add-stdcall-alias -Wl,--export-all-symbols -Wl,--enable-auto-import
 CYGWIN_LIBS = -lX11 -lgd -lpng -lz
 endif
 
-INCLUDES = $(X_CFLAGS) -I$(srcdir)/ -I/usr/include/gd -DPREFIX=\"$(prefix)\" -DBUILD_DIR=\"$(srcdir)\" -DDRV_DIR=\"$(srcdir)\" -DEMBOSS_TOP=\"$(EMBOSS_TOP)\"
+INCLUDES = -I$(srcdir)/ -I/usr/include/gd -DPREFIX=\"$(prefix)\" -DBUILD_DIR=\"$(srcdir)\" -DDRV_DIR=\"$(srcdir)\" -DEMBOSS_TOP=\"$(EMBOSS_TOP)\"
+LIBS = $(AM_LIBS)
 
 PLPLOTSRC = pdfutils.c plargs.c plbox.c plcont.c plcore.c \
 plctrl.c plcvt.c pldtik.c plfill.c plhist.c plline.c plmap.c \
--- EMBOSS-6.3.1/m4/pngdriver.m4	2013-09-11 10:39:07.680180369 +0200
+++ m4/pngdriver.m4	2013-09-11 10:44:16.000181174 +0200
@@ -63,7 +63,8 @@
 	  if test "`uname`" = "SunOS"; then
 	        AC_CHECK_LIB(iconv, libiconv_close, ICCHECK=1, ICCHECK=0, -L${ALT_HOME}/lib -liconv)
 	if test $ICCHECK = "1" ; then
-	        LDFLAGS="${LDFLAGS} -L${ALT_HOME}/lib -liconv"
+	        LDFLAGS="${LDFLAGS} -L${ALT_HOME}/lib"
+		LIBS="${LIBS} -liconv"
 	fi
 	    LDFLAGS="$LDFLAGS -R$ALT_HOME/lib"
           fi
--- EMBOSS-6.3.1/m4/java.m4	2013-09-11 10:39:03.490180358 +0200
+++ m4/java.m4	2013-09-11 10:46:08.240181467 +0200
@@ -180,7 +180,7 @@
 
 	  if test "$withval" = "yes" ; then
 	        CFLAGS="$CFLAGS -D_REENTRANT" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 
@@ -192,38 +192,38 @@
 
 	  if test "$withval" = "linux" ; then
 	        CFLAGS="$CFLAGS -D_REENTRANT" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 
 	  if test "$withval" = "solaris" ; then
 	        CFLAGS="$CFLAGS -D_POSIX_C_SOURCE=199506L" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 	  if test "$withval" = "macos" ; then
 	        CFLAGS="$CFLAGS"
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 	  if test "$withval" = "hpux" ; then
 	        CFLAGS="$CFLAGS -Ae +z -DNATIVE -D_POSIX_C_SOURCE=199506L" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 	  if test "$withval" = "irix" ; then
 	        CFLAGS="$CFLAGS" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 	  if test "$withval" = "aix" ; then
 	        CFLAGS="$CFLAGS -D_REENTRANT" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 	  if test "$withval" = "osf" ; then
 	        CFLAGS="$CFLAGS -D_REENTRANT -D_OSF_SOURCE" 
-		LDFLAGS="$LDFLAGS -lpthread"
+		LDFLAGS="$LDFLAGS"
 		LIBS="$LIBS -lpthread"
 	  fi
 else
--- EMBOSS-6.3.1/m4/mysql.m4	2013-09-11 11:03:07.160184129 +0200
+++ m4/mysql.m4	2013-09-11 11:04:39.440184370 +0200
@@ -85,7 +85,6 @@
             AC_MSG_CHECKING([for MySQL libraries])
 
 dnl         MYSQL_CFLAGS="`$MYSQL_CONFIG --cflags`"
-            MYSQL_CFLAGS="`$MYSQL_CONFIG --include`"
             MYSQL_LIBS="`$MYSQL_CONFIG --libs`"
 
             MYSQL_VERSION=`$MYSQL_CONFIG --version`
--- EMBOSS-6.3.1/m4/mysql.m4	2013-09-11 11:38:09.330189621 +0200
+++ m4/mysql.m4	2013-09-11 11:41:02.170190072 +0200
@@ -170,6 +170,4 @@
     fi
 
     AC_SUBST([MYSQL_VERSION])
-    AC_SUBST([MYSQL_CFLAGS])
-    AC_SUBST([MYSQL_LIBS])
 ])
--- EMBOSS-6.3.1/configure.in	2013-09-11 12:23:17.330196695 +0200
+++ configure.in	2013-09-11 12:23:21.480196706 +0200
@@ -8,6 +8,8 @@
 AC_INIT(ajax/core/ajmath.c)
 AM_INIT_AUTOMAKE([EMBOSS],[6.3.1])
 
+AC_CONFIG_MACRO_DIR([m4])
+
 # Make sure CFLAGS is defined to stop LT_INIT adding -g
 CFLAGS="$CFLAGS "
 
--- EMBOSS-6.3.1/nucleus/Makefile.am	2013-09-11 12:33:14.260198254 +0200
+++ nucleus/Makefile.am	2013-09-11 12:33:16.650198260 +0200
@@ -58,4 +58,4 @@
 endif
 
 libnucleus_la_LDFLAGS =  $(LINKFLAGS)
-libnucleus_la_LIBADD = ../ajax/core/libajax.la ../ajax/graphics/libajaxg.la ../ajax/acd/libacd.la
+libnucleus_la_LIBADD = -L../ajax/core -L../ajax/graphics -L../ajax/acd -lajax -lajaxg -lacd
--- EMBOSS-6.3.1/ajax/ensembl/Makefile.am	2013-09-11 12:41:06.340199487 +0200
+++ ajax/ensembl/Makefile.am	2013-09-11 12:41:40.460199577 +0200
@@ -56,5 +56,5 @@
 LINKFLAGS = $(VERS_INF)
 endif
 
-libensembl_la_LDFLAGS = $(LINKFLAGS)
-libensembl_la_LIBADD = ../core/libajax.la
+libensembl_la_LDFLAGS = $(LINKFLAGS) -L../core
+libensembl_la_LIBADD = -lajax
--- EMBOSS-6.3.1/ajax/graphics/Makefile.am	2013-09-11 12:42:17.730199674 +0200
+++ ajax/graphics/Makefile.am	2013-09-11 12:42:47.000199750 +0200
@@ -35,5 +35,5 @@
 LINKFLAGS = $(VERS_INF)
 endif
 
-libajaxg_la_LDFLAGS = $(LINKFLAGS)
-libajaxg_la_LIBADD = ../core/libajax.la $(EXTRA_LIBS)
+libajaxg_la_LDFLAGS = $(LINKFLAGS) -L../core/
+libajaxg_la_LIBADD = -lajax $(EXTRA_LIBS)
--- EMBOSS-6.3.1/ajax/acd/Makefile.am	2013-09-11 12:43:11.660199815 +0200
+++ ajax/acd/Makefile.am	2013-09-11 12:44:01.520199945 +0200
@@ -41,5 +41,5 @@
 LINKFLAGS = $(VERS_INF)
 endif
 
-libacd_la_LDFLAGS = $(LINKFLAGS)
-libacd_la_LIBADD = ../core/libajax.la ../graphics/libajaxg.la ../ajaxdb/libajaxdb.la
+libacd_la_LDFLAGS = $(LINKFLAGS) -L../core -L../graphics -L../ajaxdb
+libacd_la_LIBADD = -lajax -lajaxg -lajaxdb
--- EMBOSS-6.3.1/ajax/ajaxdb/Makefile.am	2013-09-11 12:44:10.980199970 +0200
+++ ajax/ajaxdb/Makefile.am	2013-09-11 12:44:40.040200046 +0200
@@ -37,5 +37,5 @@
 LINKFLAGS = $(VERS_INF)
 endif
 
-libajaxdb_la_LDFLAGS = $(LINKFLAGS)
-libajaxdb_la_LIBADD = ../core/libajax.la ../ensembl/libensembl.la
+libajaxdb_la_LDFLAGS = $(LINKFLAGS) -L../core -L../ensembl
+libajaxdb_la_LIBADD = -lajax -lensembl
--- EMBOSS-6.3.1/configure.in	2013-09-11 13:24:39.720206315 +0200
+++ configure.in	2013-09-11 13:24:58.610206364 +0200
@@ -319,10 +319,8 @@
 AX_LIB_MYSQL
 AX_LIB_POSTGRESQL
 
-CFLAGS="$CFLAGS $MYSQL_CFLAGS"
 LIBS="${MYSQL_LIBS} ${LIBS}"
 
-CFLAGS="$POSTGRESQL_CFLAGS $CFLAGS"
 LIBS="$POSTGRESQL_LIBS $LIBS"
 
 
--- EMBOSS-6.3.1/ajax/core/Makefile.am	2013-09-11 13:47:36.790209912 +0200
+++ ajax/core/Makefile.am	2013-09-11 13:47:40.550209922 +0200
@@ -65,5 +65,5 @@
 LINKFLAGS = $(VERS_INF)
 endif
 
-libajax_la_LDFLAGS = $(LINKFLAGS)
+libajax_la_LDFLAGS =
 libajax_la_LIBADD = $(EXTRA_LIBS) $(POSTGRESQL_LIBS) $(MYSQL_LIBS)
--- EMBOSS-6.3.1/nucleus/Makefile.am	2013-09-11 14:03:21.960212381 +0200
+++ nucleus/Makefile.am	2013-09-11 14:03:58.910212478 +0200
@@ -22,11 +22,14 @@
 EXTRA_INCLUDES = -I$(top_srcdir)/ajax/expat -I$(top_srcdir)/ajax/zlib \
 -I$(top_srcdir)/plplot -I$(top_srcdir)/ajax/pcre
 else
-EXTRA_INCLUDES = $(PLPLOT_CFLAGS)
+EXTRA_INCLUDES =
 endif
 endif
 
 
+# zap the crap
+PLPLOT_CFLAGS =
+
 INCLUDES = $(X_CFLAGS) -I$(srcdir)/ \
 	-I$(top_srcdir)/ajax \
 	$(EXTRA_INCLUDES) \
--- EMBOSS-6.3.1/emboss/Makefile.am	2013-09-11 14:23:50.570215591 +0200
+++ emboss/Makefile.am	2013-09-11 14:24:24.950215680 +0200
@@ -328,9 +328,7 @@
 
 LDADD = ../nucleus/libnucleus.la ../ajax/acd/libacd.la \
 	../ajax/ajaxdb/libajaxdb.la ../ajax/ensembl/libensembl.la \
-	../ajax/graphics/libajaxg.la ../ajax/core/libajax.la \
-	$(EXTRA_LDS) \
-	$(XLIB)
+	../ajax/graphics/libajaxg.la ../ajax/core/libajax.la
 
 pkgdata_DATA = emboss.default.template
 pkgdatadir=$(prefix)/share/$(PACKAGE)
