--- EMBOSS-6.3.1/nucleus/Makefile	2013-09-11 14:50:25.140219756 +0200
+++ nucleus/Makefile	2013-09-11 14:52:00.920220006 +0200
@@ -165,7 +165,7 @@
 ESYSTEMLIBS = 
 EXEEXT = 
 FGREP = /bin/grep -F
-GD_LIBS = -ljpeg -lfontconfig -lfreetype -lz -lm -lgd
+GD_LIBS = 
 GREP = /bin/grep
 HAVE_MEMMOVE = 
 HAVE_STRERROR = 
@@ -180,7 +180,7 @@
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBOBJS = 
-LIBS =   -L/usr/lib64/mysql -lmysqlclient -lpthread -lz -lm -lrt -lssl -lcrypto -ldl -lm  -lhpdf -lpng16  -L/usr/lib64 -lz  -ljpeg -lfontconfig -lfreetype -lz -lm -lgd
+LIBS =   
 LIBTOOL = $(SHELL) $(top_builddir)/libtool
 LIPO = 
 LN_S = ln -s
@@ -219,7 +219,7 @@
 # zap the crap
 PLPLOT_CFLAGS = 
 PLPLOT_CPPFLAGS = 
-PLPLOT_LIBS = -lplplotd -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lcairo -lSM -lICE -lX11 -lXext -lgd -lpng -lz -ljpeg -lfreetype -lSM -lICE -lX11 -lXext -lpthread -lLASi -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfreetype -lfontconfig -lhpdf -lm -lcsirocsa -lnistcd -lqsastime -lfreetype 
+PLPLOT_LIBS = 
 PNG_CFLAGS = -I/usr/include/libpng16 
 PNG_LIBS = -lpng16 
 POSIX_MALLOC_THRESHOLD = -DPOSIX_MALLOC_THRESHOLD=10
@@ -336,7 +336,7 @@
 VERS_INF = -version-info 6:3:0
 LINKFLAGS = $(VERS_INF)
 #LINKFLAGS = $(VERS_INF) $(CYGWIN_LDFLAGS) $(CYGWIN_LIBS)
-libnucleus_la_LDFLAGS = $(LINKFLAGS)
+libnucleus_la_LDFLAGS = $(LDFLAGS)
 libnucleus_la_LIBADD = -L../ajax/core -L../ajax/graphics -L../ajax/acd -lajax -lajaxg -lacd
 all: all-am
 
@@ -407,7 +407,7 @@
 	  rm -f $${locs}; \
 	}
 libnucleus.la: $(libnucleus_la_OBJECTS) $(libnucleus_la_DEPENDENCIES) $(EXTRA_libnucleus_la_DEPENDENCIES) 
-	$(libnucleus_la_LINK) -rpath $(libdir) $(libnucleus_la_OBJECTS) $(libnucleus_la_LIBADD) $(LIBS)
+	$(libnucleus_la_LINK) -rpath $(libdir) $(libnucleus_la_OBJECTS) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
