--- imaxima.el~	2006-05-27 15:36:47.000000000 +0100
+++ imaxima.el	2006-10-12 14:25:26.000000000 +0100
@@ -997,7 +997,7 @@
       (add-hook 'kill-buffer-hook 'imaxima-clean-up t t)
       (imaxima-setup-preoutput-filter)
       (maxima-single-string
-       (format "block(load(\"%s\"), linenum:0)$\n" imaxima-lisp-file))
+       (format "block(load(\"%s\"), linenum:0);\n" imaxima-lisp-file))
       ;; maxima mode tries to run inferior-maxima-mode-hook twice
       ;; due to changes made in 5.9.2 release. To prevent this,
       ;; the following hook must be removed earlier than before.
