--- Makefile	2009-01-25 18:27:52.000000000 +0100
+++ Makefile.new	2009-03-27 23:08:54.000000000 +0100
@@ -2,14 +2,9 @@
 .POSIX:
 
 CC=gcc
-CFLAGS=$$($(SHELL) lfs.sh CFLAGS) -D _XOPEN_SOURCE=500 -pipe -ansi -pedantic \
-    -O3
+CFLAGS += -D_XOPEN_SOURCE=500
 
-
-
-
-LDFLAGS=-s $$($(SHELL) lfs.sh LDFLAGS)
-LIBS=-l pthread -l bz2 $$($(SHELL) lfs.sh LIBS)
+LIBS=-lpthread -lbz2
 
 lbzip2: main.o lbzip2.o lbunzip2.o lbunzip2_single.o lacos_rbtree.o
 	$(CC) -o lbzip2 $(LDFLAGS) main.o lbzip2.o lbunzip2.o \
