--- celestia.cfg.orig	2006-09-19 11:05:24.000000000 +0100
+++ celestia.cfg	2006-09-19 11:05:50.000000000 +0100
@@ -150,7 +150,7 @@
 # or
 #   ExtrasDirectories  [ "D:\\celestia-extras" ]
 #------------------------------------------------------------------------
-  ExtrasDirectories  [ "extras" ]
+  ExtrasDirectories  [ "extras" "~/.celestia" ]
 
 
 #------------------------------------------------------------------------
