--- configure.in.orig	2010-06-22 14:29:13.162433883 +0200
+++ configure.in	2010-06-22 14:30:08.453433847 +0200
@@ -27,6 +27,17 @@
         fi
 ])
 
+dnl Additional arguments (and substitutions)
+dnl Allow the pkg-config directory to be set
+AC_ARG_WITH(pkgconfigdir,
+        AC_HELP_STRING([--with-pkgconfigdir],
+        [Use the specified pkgconfig dir (default is libdir/pkgconfig)]),
+        [pkgconfigdir=${withval}],
+        [pkgconfigdir='${libdir}/pkgconfig'])
+
+AC_SUBST([pkgconfigdir])
+AC_MSG_NOTICE([pkgconfig directory is ${pkgconfigdir}])
+
 dnl Checks for programs
 AC_PROG_CC
 
@@ -62,5 +73,5 @@
 
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$ac_cv_version_script" = "yes")
 
-AC_CONFIG_FILES([Makefile librdmacm.spec])
+AC_CONFIG_FILES([Makefile librdmacm.spec rdmacm.pc:rdmacm.pc.in])
 AC_OUTPUT
--- Makefile.am.orig	2010-06-22 14:29:17.849433892 +0200
+++ Makefile.am	2010-06-22 14:30:30.680832953 +0200
@@ -1,5 +1,8 @@
 INCLUDES = -I$(srcdir)/include
 
+pkgconfigdir = @pkgconfigdir@
+pkgconfig_DATA = rdmacm.pc
+
 lib_LTLIBRARIES = src/librdmacm.la
 
 AM_CFLAGS = -g -Wall -D_GNU_SOURCE
--- /dev/null	2010-05-29 13:22:18.914064236 +0200
+++ rdmacm.pc.in	2010-06-22 14:30:56.082443278 +0200
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PACKAGE_NAME@
+Version: @PACKAGE_VERSION@
+Description: @PACKAGE_STRING@
+Requires:
+Libs: -L${libdir} -lrdmacm
+Cflags: -I${includedir}
