--- examples/Makefile.cmake.orig	2013-04-29 09:13:58.164963641 -0700
+++ examples/Makefile.cmake	2013-04-29 09:18:42.371460644 -0700
@@ -1,7 +1,7 @@
 # The generated "Makefile" from "Makefile.cmake" is only usable after
 # the Armadillo library has been configured and installed by CMake.
 
-CXX=g++
+CXX:=g++
 #CXX=g++-4.2
 ## Under MacOS you may have an old compiler as default (e.g. GCC 4.0).
 ## However, GCC 4.2 or later is available and preferable due to better
@@ -26,7 +26,7 @@
 
 
 
-LIB_FLAGS = -larmadillo $(EXTRA_LIB_FLAGS)
+LIB_FLAGS = $(EXTRA_LIB_FLAGS) -larmadillo
 ## NOTE: on Ubuntu and Debian based systems you may need to add 
 ## -lgfortran to LIB_FLAGS
 
@@ -73,10 +73,10 @@
 all: example1 example2
 
 example1: example1.cpp
-	$(CXX) $(CXXFLAGS)  -o $@  $<  $(LIB_FLAGS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $< $(LIB_FLAGS) -o $@
 
 example2: example2.cpp
-	$(CXX) $(CXXFLAGS)  -o $@  $<  $(LIB_FLAGS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $< $(LIB_FLAGS) -o $@
 
 
 .PHONY: clean
