--- CMakeLists.txt.orig	2013-05-24 21:14:49.626395820 -0700
+++ CMakeLists.txt	2013-05-24 21:15:10.724511364 -0700
@@ -222,11 +222,11 @@
 #   find_package(HDF5)
 #   message(STATUS "HDF5_FOUND     = ${HDF5_FOUND}")
 #   
-#   if(HDF5_FOUND)
-#     set(ARMA_USE_HDF5 true)
-#     set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
-#     set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
-#   endif()
+if(HDF5_FOUND)
+    set(ARMA_USE_HDF5 true)
+    set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${HDF5_INCLUDE_DIRS})
+    set(ARMA_LIBS ${ARMA_LIBS} ${HDF5_LIBRARIES})
+endif()
 # endif()
 
 
