--- a/books/Makefile-generic	2011-04-12 14:42:17.790165650 -0400
+++ b/books/Makefile-generic	2011-04-12 14:55:54.436829607 -0400
@@ -55,7 +55,7 @@
 # Determine the location of this Makefile-generic.  NOTE: We have
 # tried $(realpath $(dir ...)) and similarly for abspath, but they
 # have returned the empty string, presumably an error.
-BOOKS_DIR := $(dir $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST)))
+BOOKS_DIR := /usr/share/acl2/books/
 
 # and use that to determine the default location of ACL2.  To override,
 # set the ACL2 variable on the command line or in the environment.
