--- install_util/GBrowseInstall.pm.ori	2011-07-30 20:42:02.000000000 +0200
+++ install_util/GBrowseInstall.pm	2011-07-30 20:42:19.000000000 +0200
@@ -486,11 +486,11 @@
     my $base = basename($self->install_path->{htdocs});
 
     # Configure the databases, if needed.
-    print STDERR "Updating user account database...\n";
-    my $metadb_script = File::Spec->catfile("bin", "gbrowse_metadb_config.pl");
-    my $perl          = $self->perl;
-    my @inc           = map{"-I$_"} split ':',$self->added_to_INC;
-    system $perl,@inc,$metadb_script;
+#    print STDERR "Updating user account database...\n";
+#    my $metadb_script = File::Spec->catfile("bin", "gbrowse_metadb_config.pl");
+#    my $perl          = $self->perl;
+#    my @inc           = map{"-I$_"} split ':',$self->added_to_INC;
+#    system $perl,@inc,$metadb_script;
     system 'sudo','chown','-R',"$uid.$gid",$destdir.$sessions,$destdir.$userdata;
 
 #    if (Module::Build->y_n(
