--- lib/Bio/Graphics/Browser2/AuthorizedFeatureFile.pm.ori  2011-07-30 17:14:28.000000000 +0200
+++ lib/Bio/Graphics/Browser2/AuthorizedFeatureFile.pm      2011-07-30 17:13:13.000000000 +0200
@@ -30,7 +30,7 @@
 sub setting {
   my $self = shift;
   my ($label,$option,@rest) = @_ >= 2 ? @_ : ('general',@_);
-  $label = 'general' if lc $label eq 'general';  # buglet
+  $label = 'general' if lc($label) eq 'general';  # buglet
   $self->SUPER::setting($label,$option,@rest);
 }
 
