--- install_util/GBrowseInstall.pm.ori	2010-11-27 12:01:58.000000000 +0100
+++ install_util/GBrowseInstall.pm	2010-11-27 12:02:15.000000000 +0100
@@ -436,8 +436,8 @@
 	$self->ownership_warning($databases,$user);
     }
 
-    chmod 0755,File::Spec->catfile($self->install_path->{'etc'},'init.d','gbrowse-slave');
-    $self->fix_selinux;
+    # chmod 0755,File::Spec->catfile($self->install_path->{'etc'},'init.d','gbrowse-slave');
+    # $self->fix_selinux;
 
     my $base = basename($self->install_path->{htdocs});
 
--- install_util/GBrowseInstall.pm.ori	2011-02-14 13:50:41.000000000 +0100
+++ install_util/GBrowseInstall.pm	2011-02-14 13:51:46.000000000 +0100
@@ -470,11 +470,11 @@
     my @inc           = map{"-I$_"} split ':',$self->added_to_INC;
     system $perl,@inc,$metadb_script;
 
-    if (Module::Build->y_n(
-	    "It is recommended that you restart Apache. Shall I try this for you?",'y'
-	)) {
-	system "sudo /etc/init.d/apache2 restart";
-    }
+#    if (Module::Build->y_n(
+#	    "It is recommended that you restart Apache. Shall I try this for you?",'y'
+#	)) {
+#	system "sudo /etc/init.d/apache2 restart";
+#    }
     
     print STDERR "\n***INSTALLATION COMPLETE***\n";
     print STDERR "Load http://localhost/$base for demo and documentation.\n";
