--- install_util/GBrowseInstall.pm.ori	2011-05-11 04:10:08.000000000 +0200
+++ install_util/GBrowseInstall.pm	2011-05-11 04:10:26.000000000 +0200
@@ -463,12 +463,12 @@
 
     my $base = basename($self->install_path->{htdocs});
 
-    # Configure the databases, if needed.
-    print STDERR "Updating user account database...\n";
-    my $metadb_script = File::Spec->catfile("bin", "gbrowse_metadb_config.pl");
-    my $perl          = $self->perl;
-    my @inc           = map{"-I$_"} split ':',$self->added_to_INC;
-    system $perl,@inc,$metadb_script;
+#    # Configure the databases, if needed.
+#    print STDERR "Updating user account database...\n";
+#    my $metadb_script = File::Spec->catfile("bin", "gbrowse_metadb_config.pl");
+#    my $perl          = $self->perl;
+#    my @inc           = map{"-I$_"} split ':',$self->added_to_INC;
+#    system $perl,@inc,$metadb_script;
 
 #    if (Module::Build->y_n(
 #	    "It is recommended that you restart Apache. Shall I try this for you?",'y'
