--- bin/gbrowse_metadb_config.pl.ori	2011-07-30 19:57:33.000000000 +0200
+++ bin/gbrowse_metadb_config.pl	2011-07-30 20:04:51.000000000 +0200
@@ -703,9 +703,9 @@
     local $database->{AutoCommit} = 0;
     local $database->{RaiseError} = 1;
     eval {
-	$database->do("delete from $table");
-	$database->do("insert into $table (schema_version) values ($version)");
-	$database->commit();
+	$database->do("delete from $table") or warn "delete from $table failed with: $database->errstr";
+	$database->do("insert into $table (schema_version) values ($version)") or warn "insert into $table (schema_version) values ($version) failed with: $database->errstr";
+	$database->commit() or warn "commitfailed with: $database->errstr";
     };
     if ($@) {
 	warn "update failed due to $@. Rolling back";
@@ -768,11 +768,11 @@
 	$insert_user->finish;
 	# rename the current users table
 	$database->do('drop table users')
-	    or die "Couldn't drop old users table";
+	    or die "Couldn't drop old users table: $database->errstr";
 	$database->do('alter table users_new rename to users')
-	    or die "Couldn't rename new users table";
+	    or die "Couldn't rename new users table: $database->errstr";
 	$database->do('create index index_session on session(username)')
-	    or die "Couldn't index sessions table";
+	    or die "Couldn't index sessions table: $database->errstr";
 
 	# now do the uploads table
 	# this upgrades to latest version 0
@@ -808,9 +808,9 @@
 	$insert->finish();
 
 	$database->do('drop table uploads')
-	    or die "Couldn't drop old uploads table";
+	    or die "Couldn't drop old uploads table: $database->errstr";
 	$database->do('alter table uploads_new rename to uploads')
-	    or die "Couldn't rename new uploads table";
+	    or die "Couldn't rename new uploads table: $database->errstr";
 
 	# now do the openid_users table
 	# this creates the new one
@@ -916,10 +916,10 @@
 	$select->finish;
 	$insert->finish;
 	$database->do('drop table users')
-	    or die "Couldn't drop old users table";
+	    or die "Couldn't drop old users table: $database->errstr";
 
 	$database->do('alter table users_new rename to users')
-	    or die "Couldn't rename new users table";
+	    or die "Couldn't rename new users table: $database->errstr";
 	$database->commit();
     };
     
