--- install_util/GBrowseInstall.pm.ori	2011-07-30 19:41:55.000000000 +0200
+++ install_util/GBrowseInstall.pm	2011-07-30 19:42:31.000000000 +0200
@@ -465,7 +465,7 @@
     chown $uid,-1,$destdir.$htdocs;
     {
 	local $> = $uid;
-	symlink($images,$htdocs_i);  # so symlinkifowner match works!
+        symlink($images,$destdir.$htdocs_i) or warn "symlink($images,$destdir.$htdocs_i) failed: $!\n";  # so symlinkifowner match works!
     }
     chown $>,-1,$destdir.$self->install_path->{htdocs};
 
