diff -Naur apbs-0.4.0/examples/born/runme apbs-0.4.0-new/examples/born/runme
--- apbs-0.4.0/examples/born/runme	2002-08-13 14:35:41.000000000 +0000
+++ apbs-0.4.0-new/examples/born/runme	2006-03-09 03:15:09.000000000 +0000
@@ -8,5 +8,5 @@
   echo "Generating ion with $i radius..."
   echo "ATOM      1  I   ION     1       0.000   0.000  0.000  1.00  $i" \
     > ion.pqr
-  ${bindir}/apbs apbs.in > OUTPUT-${i} 2>&1 
+  /usr/bin/apbs apbs.in > OUTPUT-${i} 2>&1 
 done
diff -Naur apbs-0.4.0/examples/ion-pmf/runme apbs-0.4.0-new/examples/ion-pmf/runme
--- apbs-0.4.0/examples/ion-pmf/runme	2002-08-13 14:35:41.000000000 +0000
+++ apbs-0.4.0-new/examples/ion-pmf/runme	2006-03-09 03:16:11.000000000 +0000
@@ -14,5 +14,5 @@
   echo "ATOM      1  I   ION     1      $i    0.000   0.000  1.00 2.00"\
      > mol2.pqr
   cat mol1.pqr mol2.pqr > complex.pqr
-  ${bindir}/apbs apbs.in > OUTPUT_${i} 2>&1
+  /usr/bin/apbs apbs.in > OUTPUT_${i} 2>&1
 done
diff -Naur apbs-0.4.0/examples/point-pmf/runme.sh apbs-0.4.0-new/examples/point-pmf/runme.sh
--- apbs-0.4.0/examples/point-pmf/runme.sh	2004-02-12 17:14:02.000000000 +0000
+++ apbs-0.4.0-new/examples/point-pmf/runme.sh	2006-03-09 03:17:11.000000000 +0000
@@ -20,5 +20,5 @@
   echo "ATOM      1  I   ION     1      $i    0.000   0.000  1.00 0.00"\
      > mol2.pqr
   cat mol1.pqr mol2.pqr > complex.pqr
-  ${bindir}/apbs apbs.in > OUTPUT_${i} 2>&1
+  /usr/bin/apbs apbs.in > OUTPUT_${i} 2>&1
 done
