diff --git a/pyproject.toml b/pyproject.toml
index 065a21ebc..3de10abb1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -126,7 +126,7 @@ markers = [
     "skip_with_pyarrow_strings: Tests that should be skipped when pyarrow string conversion is turned on",
     "xfail_with_pyarrow_strings: Tests that should be xfailed when pyarrow string conversion is turned on",
 ]
-addopts = "-v -rsfE --durations=10 --color=yes --cov-config=pyproject.toml"
+addopts = "-v -rsfE --durations=10 --color=yes"
 filterwarnings = [
     "error:::dask[.*]",
     "error:::pandas[.*]",
