--- tools/build/user.settings		2009-04-07 18:51:37.000000000 +0200
+++ tools/build/user.settings.new	2009-04-07 18:51:37.000000000 +0200
@@ -21,16 +21,22 @@
 settings = {
     "user" : {
         "prepends" : {
+            "program_path"  : os.environ["PATH"].split(":"),
+            "include_path"  : [ "/usr/include" ],
+            "library_path"  : [ "/usr/GENTOO_LIBDIR" ],
         },
         "appends" : {
             # Uncomment the line below and the "import os" line above
             # to let SCons find ICC (assuming it's on your PATH already)
-            "program_path"  : os.environ["PATH"].split(":"),
 
             # Uncomment the line below to begin building the "devel" project too.
             #"projects": { "src" : [ "devel", ], },
         },
         "overrides" : {
+            "flags" : {
+                  "compile" : [ GENTOO_CXXFLAGS ],
+                  "link"    : [ GENTOO_LDFLAGS ],
+            },
         },
         "removes" : {
         },
