--- LibreCAD-LibreCAD-460bd78/librecad.pro	2011-08-26 22:26:35.201935340 -0400
+++ LibreCAD-LibreCAD-460bd78/librecad.pro	2011-08-26 22:26:59.138898865 -0400
@@ -24,7 +24,7 @@
     # Get SVN revision number
     # SVNREVISION = $$system(svn info -R | grep -o \"Revision: [0-9]*\" | sed -e \"s/Revision: //\" | head -n1)
     # Temporary disabled getting SCM version
-    SCMREVISION=$$system(git describe --tags)
+    SCMREVISION=1.0.0_rc2
     DEFINES += QC_SCMREVISION=\"$$SCMREVISION\"
     macx { 
         CONFIG += x86 x86_64
