diff --git a/src/ui/forms/qg_commandwidget.ui b/src/ui/forms/qg_commandwidget.ui
index b6ab92a..30fb453 100644
--- a/src/ui/forms/qg_commandwidget.ui
+++ b/src/ui/forms/qg_commandwidget.ui
@@ -50,7 +50,7 @@
             <bool>true</bool>
           </property>
           <property name="undoRedoEnabled">
-            <bool>false</bool>
+            <bool>true</bool>
           </property>
         </widget>
       </item>
