--- mendeleydesktop-1.13.1/bin/mendeleydesktop.orig 2015-01-15 15:49:15.589870680 +0100
+++ mendeleydesktop-1.13.1/bin/mendeleydesktop 2015-01-15 15:46:49.308954069 +0100
@@ -95,11 +95,11 @@
 		# Both version components are non-numeric, treat
 		# them as equal
 		return False
-	elif (version_part_a and version_part_b is None):
+	elif (type(version_part_a) is int and version_part_b is None):
 		# Left version component is numeric, right component is not.
 		# Treat the numeric part as greater
 		return False
-	elif (version_part_b and version_part_a is None):
+	elif (type(version_part_b) is int and version_part_a is None):
 		# Right version component is numeric, left component is not.
 		# Treat the numeric part as greater
 		return True
