--- lam-7.1.2.orig/config/lam_prog_f77.m4	2006-07-19 22:43:12.000000000 -0700
+++ lam-7.1.2/config/lam_prog_f77.m4	2006-07-19 22:43:18.000000000 -0700
@@ -58,7 +58,7 @@
 
 	# If the user did not specify one, look for all the common names
 
-	AC_CHECK_PROGS(F77, [g77 f77 fort77 f90], no)
+	AC_CHECK_PROGS(F77, [gfortran g77 f77 fort77 f90], no)
 	if test -z "$F77" -o "$F77" = "no"; then
 	    AC_MSG_WARN([*** Could not find FORTRAN compiler.  Either use])
 	    AC_MSG_WARN([*** --with-fc to specify the FORTRAN compiler, or])
