diff -urN lam-7.1.4/configure.in lam-7.1.4-patched/configure.in
--- lam-7.1.4/configure.in	2006-06-24 16:42:47.000000000 -0400
+++ lam-7.1.4-patched/configure.in	2008-03-11 00:03:26.000000000 -0400
@@ -2482,6 +2482,7 @@
 
     share/libmpi/Makefile
     share/liblam/Makefile
+    share/liblamf77mpi/Makefile
 
     share/dynamic-ssi/Makefile
     share/dynamic-ssi/boot/Makefile
diff -urN lam-7.1.4/share/liblamf77mpi/Makefile.am lam-7.1.4-patched/share/liblamf77mpi/Makefile.am
--- lam-7.1.4/share/liblamf77mpi/Makefile.am	1969-12-31 19:00:00.000000000 -0500
+++ lam-7.1.4-patched/share/liblamf77mpi/Makefile.am	2008-03-11 00:37:26.000000000 -0400
@@ -0,0 +1,32 @@
+# -*- makefile -*-
+#
+# Copyright (c) 2001-2003 The Trustees of Indiana University.  
+#                         All rights reserved.
+# Copyright (c) 1998-2001 University of Notre Dame. 
+#                         All rights reserved.
+# Copyright (c) 1994-1998 The Ohio State University.  
+#                         All rights reserved.
+# 
+# This file is part of the LAM/MPI software package.  For license
+# information, see the LICENSE file in the top level directory of the
+# LAM/MPI source distribution.
+#
+# $Id: Makefile.am,v 1.1 2003/11/15 14:09:55 jsquyres Exp $
+#
+
+include $(top_srcdir)/config/Makefile.options
+
+if WANT_FORTRAN
+fortran_lib		= liblamf77mpi.la
+else
+fortran_lib		=
+endif
+
+lib_LTLIBRARIES		= $(fortran_lib)
+
+liblamf77mpi_la_SOURCES	=
+liblamf77mpi_la_LIBADD	= \
+			$(top_builddir)/share/libmpi/libmpi.la \
+			$(top_builddir)/share/liblam/liblam.la \
+			$(top_builddir)/share/mpi/f77/liblamf77mpi.la
+liblamf77mpi_la_LDFLAGS	= $(EXTRA_LDFLAGS)
diff -urN lam-7.1.4/share/libmpi/Makefile.am lam-7.1.4-patched/share/libmpi/Makefile.am
--- lam-7.1.4/share/libmpi/Makefile.am	2006-06-24 16:42:43.000000000 -0400
+++ lam-7.1.4-patched/share/libmpi/Makefile.am	2008-03-11 00:12:32.000000000 -0400
@@ -42,5 +42,6 @@
 			$(top_builddir)/share/mpi/libmpiextra.la \
 			$(top_builddir)/share/ssi/libssi_mpi.la \
 			$(top_builddir)/share/memory/libmemory.la \
+			$(top_builddir)/share/liblam/liblam.la \
 			$(impi_lib) $(pmpi_lib)
 libmpi_la_LDFLAGS	= $(EXTRA_LDFLAGS)
diff -urN lam-7.1.4/share/Makefile.am lam-7.1.4-patched/share/Makefile.am
--- lam-7.1.4/share/Makefile.am	2006-06-24 16:42:43.000000000 -0400
+++ lam-7.1.4-patched/share/Makefile.am	2008-03-10 23:32:04.000000000 -0400
@@ -24,4 +24,4 @@
 SUBDIRS			= include args boot etc freq kreq impi memory mpi \
 			  $(LIBLTDL_SUBDIR) nreq pmpi rreq ssi \
 			  threads $(TOTALVIEW_DLL_DIR) trillium tstdio \
-			  liblam libmpi dynamic-ssi
+			  liblam libmpi liblamf77mpi dynamic-ssi
diff -urN lam-7.1.4/share/mpi/f77/Makefile.am lam-7.1.4-patched/share/mpi/f77/Makefile.am
--- lam-7.1.4/share/mpi/f77/Makefile.am	2006-06-24 16:42:35.000000000 -0400
+++ lam-7.1.4-patched/share/mpi/f77/Makefile.am	2008-03-10 23:41:30.000000000 -0400
@@ -16,6 +16,8 @@
 
 include $(top_srcdir)/config/Makefile.options
 
+noinst_LTLIBRARIES = liblamf77mpi.la
+
 # This kinda sucks.  The "ar" command in IRIX 6.5 has a upper limit
 # on the total number of characters in its arguments.  If we include
 # all these fortran files in making libmpi.a, with all the
@@ -34,12 +36,6 @@
 # no extra subdirectories, so it doesn't run into the same character
 # limit with ar.  Ugh.
 
-if WANT_FORTRAN
-fortran_lib		= liblamf77mpi.la
-else
-fortran_lib		=
-endif
-
 
 # Files with stubs of MPI functions that LAM has not yet implemented.
 # These are here because the default action is to *not* compile them
@@ -61,8 +57,6 @@
 mpi_stub_sources	=
 endif
 
-
-lib_LTLIBRARIES		= $(fortran_lib)
 liblamf77mpi_la_SOURCES	= \
 			  $(mpi_stub_sources) \
 			  abort_f.c \
