--- lam-7.1.2.orig/share/liblam/Makefile.am	2006-09-25 22:52:17.000000000 -0700
+++ lam-7.1.2/share/liblam/Makefile.am	2006-09-25 22:52:04.000000000 -0700
@@ -35,5 +35,6 @@
 			$(top_builddir)/share/ssi/libssi_lam.la \
 			$(top_builddir)/share/threads/liblamthreads.la \
 			$(top_builddir)/share/trillium/liblamtrillium.la \
-			$(top_builddir)/share/tstdio/liblamtstdio.la
+			$(top_builddir)/share/tstdio/liblamtstdio.la \
+			$(LIBLAM_EXTRA_LIBS)
 liblam_la_LDFLAGS	= $(EXTRA_LDFLAGS)
