diff -ruN FreeCAD-0.10.3205.orig/acinclude.m4 FreeCAD-0.10.3205/acinclude.m4
--- FreeCAD-0.10.3205.orig/acinclude.m4	2010-06-05 11:16:14.000000000 +0200
+++ FreeCAD-0.10.3205/acinclude.m4	2010-11-07 22:35:02.000000000 +0100
@@ -374,7 +374,7 @@
 fi
 
 # Make sure not to link against X11 libs so that configure succeeds whithout xserver started
-bnv_try_4="$CXX $fc_qt4_lib_core $LIBS -o myqt myqt.o moc_myqt.o"
+bnv_try_4="$CXX -o myqt myqt.o moc_myqt.o $LIBS $fc_qt4_lib_core"
 AC_TRY_EVAL(bnv_try_4)
 if test x"$ac_status" != x0; then
    AC_MSG_ERROR([Failed to link with Qt, bye...])
