--- a/External/catch/CMakeLists.txt	2020-09-28 05:09:11.307595711 -0400
+++ b/External/catch/CMakeLists.txt 	2020-09-28 05:10:35.188570785 -0400
@@ -13,9 +13,6 @@
   message("-- Catch not found in ${CATCH_DIR}")
   include(RDKitUtils)
   set(RELEASE_NO "2.1.2")
-  downloadAndCheckMD5("https://github.com/catchorg/Catch2/archive/v${RELEASE_NO}.tar.gz"
-        "${CMAKE_CURRENT_SOURCE_DIR}/master.tar.gz"
-        "4c5740d7446d9b4e3735dcff0951aaca")
   execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
     ${CMAKE_CURRENT_SOURCE_DIR}/master.tar.gz
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
--- a/External/CoordGen/CMakeLists.txt	2020-09-28 05:13:38.351584109 -0400
+++ b/External/CoordGen/CMakeLists.txt	2020-09-28 05:15:49.396007762 -0400
@@ -38,9 +38,6 @@

     if(NOT EXISTS "${MAEPARSER_DIR}/MaeParser.hpp")
         set(RELEASE_NO "1.2.3")
-        set(MD5 "4ead02a0bf87110d1caae79ab42a5747")
-        downloadAndCheckMD5("https://github.com/schrodinger/maeparser/archive/v${RELEASE_NO}.tar.gz"
-              "${CMAKE_CURRENT_SOURCE_DIR}/maeparser-v${RELEASE_NO}.tar.gz" ${MD5})
         execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
           ${CMAKE_CURRENT_SOURCE_DIR}/maeparser-v${RELEASE_NO}.tar.gz
           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
--- a/External/CoordGen/CMakeLists.txt	2020-09-28 05:31:21.353305949 -0400
+++ b/External/CoordGen/CMakeLists.txt 	2020-09-28 05:32:04.187783628 -0400
@@ -67,9 +67,6 @@

     if(NOT EXISTS "${COORDGEN_DIR}/sketcherMinimizer.h")
         set(RELEASE_NO "1.4.0")
-        set(MD5 "5f663c8809b494f0548dd504c011c739")
-        downloadAndCheckMD5("https://github.com/schrodinger/coordgenlibs/archive/v${RELEASE_NO}.tar.gz"
-              "${CMAKE_CURRENT_SOURCE_DIR}/coordgenlibs-${RELEASE_NO}.tar.gz" ${MD5})
         execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
           ${CMAKE_CURRENT_SOURCE_DIR}/coordgenlibs-${RELEASE_NO}.tar.gz
           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
--- a/External/RingFamilies/CMakeLists.txt	2020-09-18 11:18:37.000000000 -0400
+++ b/External/RingFamilies/CMakeLists.txt 	2020-09-28 05:35:04.427800305 -0400
@@ -8,9 +8,6 @@

 if(NOT EXISTS "${URFLIB_DIR}/RingDecomposerLib.h")
     set(RELEASE_NO "1.1.3_rdkit")
-    set(MD5 "e9a0bcdda8b921a35e812b9888a9a874")
-    downloadAndCheckMD5("https://github.com/rareylab/RingDecomposerLib/archive/v${RELEASE_NO}.tar.gz"
-            "${CMAKE_CURRENT_SOURCE_DIR}/RingDecomposerLib-v${RELEASE_NO}.tar.gz" ${MD5})
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
         ${CMAKE_CURRENT_SOURCE_DIR}/RingDecomposerLib-v${RELEASE_NO}.tar.gz
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
--- a/Code/GraphMol/MolInterchange/CMakeLists.txt	2020-09-18 11:18:37.000000000 -0400
+++ b/Code/GraphMol/MolInterchange/CMakeLists.txt 	2020-09-28 05:36:47.759960868 -0400
@@ -1,8 +1,5 @@

 if(NOT EXISTS "${CMAKE_SOURCE_DIR}/External/rapidjson-1.1.0")
-    downloadAndCheckMD5("https://github.com/Tencent/rapidjson/archive/v1.1.0.tar.gz"
-          "${CMAKE_SOURCE_DIR}/External/rapidjson-1.1.0.tar.gz"
-          "badd12c511e081fec6c89c43a7027bce")
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar zxf
       ${CMAKE_SOURCE_DIR}/External/rapidjson-1.1.0.tar.gz
       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/External)
