diff -uhr CalculiX/cgx_1.8/src/Makefile CalculiX.mine/cgx_1.8/src/Makefile
--- CalculiX/cgx_1.8/src/Makefile	2008-07-26 18:13:29.000000000 -0400
+++ CalculiX.mine/cgx_1.8/src/Makefile	2008-08-06 08:55:08.000000000 -0400
@@ -1,4 +1,4 @@
-CFLAGS = -O2 -Wall \
+CFLAGS = -O3 -Wall \
   -I./ \
   -I/usr/include \
   -I/usr/include/GL \
@@ -7,7 +7,7 @@
 
 LFLAGS = \
-  -L/usr/lib -lGL -lGLU \
+  -L/usr/lib -lGL -lGLU -lglut \
-  -L/usr/X11/lib -lX11 -lXext -lXi -lXmu -lXt -lSM -lICE \
+  -L/usr/X11/lib $(PTHREAD) -lX11 -lXext -lXi -lXmu -lXt -lSM -lICE \
   -lm 
 
 # -L/usr/X11/lib64 -lpthread -lX11 -lXext -lXi -lXmu -lXt -lSM -lICE \
@@ -122,42 +122,6 @@
  writefbd.c \
  contact.c \
  makeTriFromElems.c \
- ../../glut-3.5/src/layerutil.c \
- ../../glut-3.5/src/glut_dials.c \
- ../../glut-3.5/src/glut_ext.c \
- ../../glut-3.5/src/glut_mesa.c \
- ../../glut-3.5/src/glut_menu.c \
- ../../glut-3.5/src/glut_modifier.c \
- ../../glut-3.5/src/glut_space.c \
- ../../glut-3.5/src/glut_shapes.c \
- ../../glut-3.5/src/glut_tablet.c \
- ../../glut-3.5/src/glut_vidresize.c \
- ../../glut-3.5/src/glut_warp.c \
- ../../glut-3.5/src/glut_8x13.c \
- ../../glut-3.5/src/glut_9x15.c \
- ../../glut-3.5/src/glut_hel10.c \
- ../../glut-3.5/src/glut_hel18.c \
- ../../glut-3.5/src/glut_hel12.c \
- ../../glut-3.5/src/glut_tr10.c \
- ../../glut-3.5/src/glut_tr24.c \
- ../../glut-3.5/src/glut_bitmap.c \
- ../../glut-3.5/src/glut_cursor.c \
- ../../glut-3.5/src/glut_event.c \
- ../../glut-3.5/src/glut_get.c \
- ../../glut-3.5/src/glut_overlay.c \
- ../../glut-3.5/src/glut_input.c \
- ../../glut-3.5/src/glut_stroke.c \
- ../../glut-3.5/src/glut_teapot.c \
- ../../glut-3.5/src/glut_winmisc.c \
- ../../glut-3.5/src/glut_fullscrn.c \
- ../../glut-3.5/src/glut_cindex.c \
- ../../glut-3.5/src/glut_roman.c \
- ../../glut-3.5/src/glut_mroman.c \
- ../../glut-3.5/src/glut_swidth.c \
- ../../glut-3.5/src/glut_bwidth.c \
- ../../glut-3.5/src/glut_win.c \
- ../../glut-3.5/src/glut_init.c \
- ../../glut-3.5/src/glut_util.c \
  uselibSNL.cpp \
  ../../libSNL/src/*.cpp
 
