diff --git a/py2bit.c b/py2bit.c
index dc24535..a9080c3 100644
--- a/py2bit.c
+++ b/py2bit.c
@@ -32,7 +32,7 @@ static PyObject *py2bitOpen(PyObject *self, PyObject *args, PyObject *kwds) {
 }
 
 PyObject *py2bitEnter(pyTwoBit_t *self, PyObject *args) {
-    pyTwoBit_t *pytb = self->tb;
+    pyTwoBit_t *pytb = self;
 
     if(!pytb) {
         PyErr_SetString(PyExc_RuntimeError, "The 2bit file handle is not open!");
