Index: system.mk.in
===================================================================
--- system.mk.in	(revision 2189)
+++ system.mk.in	(working copy)
@@ -113,8 +113,8 @@
 
 # Our libraries are I think exclusively used by us, hence we pick
 # our own subdirectory.
-prefix         = @prefix@
-exec_prefix    = @exec_prefix@
+prefix         = ${DESTDIR}@prefix@
+exec_prefix    = ${DESTDIR}@exec_prefix@
 INSTALLLIB     = @libdir@/staden
 INSTALLBIN     = @bindir@
 #INSTALLETC     = @sysconfdir@/staden
