                                                                     
                                                                     
                                                                     
                                             
Index: system.mk.in
===================================================================
--- system.mk.in	(revision 2261)
+++ system.mk.in	(working copy)
@@ -113,14 +113,14 @@
 
 # Our libraries are I think exclusively used by us, hence we pick
 # our own subdirectory.
-prefix         = ${DESTDIR}@prefix@
-exec_prefix    = ${DESTDIR}@exec_prefix@
-INSTALLLIB     = @libdir@/staden
-INSTALLBIN     = @bindir@
-#INSTALLETC     = @sysconfdir@/staden
-INSTALLSHARE   = @datadir@/staden
-INSTALLDOC     = @datadir@/doc
-INSTALLMAN     = @mandir@
+prefix         = @prefix@
+exec_prefix    = @exec_prefix@
+INSTALLLIB     = ${DESTDIR}@libdir@/staden
+INSTALLBIN     = ${DESTDIR}@bindir@
+#INSTALLETC     = ${DESTDIR}@sysconfdir@/staden
+INSTALLSHARE   = ${DESTDIR}@datadir@/staden
+INSTALLDOC     = ${DESTDIR}@datadir@/doc
+INSTALLMAN     = ${DESTDIR}@mandir@
 INSTALLSCRIPTS = $(INSTALLSHARE)/scripts
 INSTALLTCL     = $(INSTALLSHARE)/tcl
 
