diff -ur src/CGALQt/makefile src/CGALQt/makefile
--- src/CGALQt/makefile	2006-05-23 17:15:58.000000000 +0200
+++ src/CGALQt/makefile	2006-05-24 15:59:33.000000000 +0200
@@ -71,15 +71,21 @@
 
 static_lib: lib
 
-lib: lib_no_install
+lib: lib_no_install lib_shared
 	mv $(CGAL_QT_LIB) $(CGAL_LIB_DESTINATION)
+	mv libCGALQt.so $(CGAL_LIB_DESTINATION)
+	rm $(CGAL_OBJECTS) 
 
 lib_no_install: $(CGAL_OBJECTS) 
 	$(CGAL_LIB_CREATE)$(CGAL_QT_LIB) \
 	`ls *$(OBJ_EXT) | awk '{for (i=1; i<=NF;++i){printf "$(CGAL_OBJ_PREFIX)";print $$i}}'`\
 		$(CGAL_LIB_LDFLAGS)
 	$(RANLIB) $(CGAL_QT_LIB)
-	rm $(CGAL_OBJECTS) 
+
+lib_shared: $(CGAL_OBJECTS)
+	$(CGAL_SHARED_LIB_CREATE)libCGALQt.so \
+	`ls *$(OBJ_EXT) | awk '{for (i=1; i<=NF;++i){printf "$(CGAL_OBJ_PREFIX)";print $$i}}'`\
+		$(CGAL_LIB_LDFLAGS)
 
 clean:
 	rm -f $(CGAL_QT_LIB) $(CGAL_OBJECTS) *.moc
