--- configure.ac.orig	2007-11-08 15:38:10.146227506 +0000
+++ configure.ac	2007-11-08 15:38:35.895694885 +0000
@@ -217,11 +217,7 @@
 
 ################## handle the ATLAS library(linear algebra) ##################
 if test "$use_atlas" = "yes"; then
-  if test "$use_pthreads" = "yes"; then
-    LIBS="-llapack -lptcblas -lcblas -latlas $LIBS"
-  else
-    LIBS="-llapack -lcblas -latlas $LIBS"
-  fi
+    LIBS="ATLAS_LIBDIR -llapack -lcblas -latlas $LIBS"
 else
   AC_MSG_ERROR([CBLAS/LAPack library not found! Exiting.])
 fi
--- src/Makefile.am.orig	2007-11-08 15:38:45.072217825 +0000
+++ src/Makefile.am	2007-11-08 15:39:24.090441347 +0000
@@ -4,6 +4,7 @@
 
 if USE_PLPLOT
 CPLOTSOURCE	= cplot.c
+AM_CFLAGS	= @PLPLOT_CFLAGS@ $(CFLAGS)
 endif
 
 bin_PROGRAMS	= scamp
