--- Makefile.globals.orig	2007-09-30 22:59:31.000000000 +0700
+++ Makefile.globals	2007-09-30 22:57:27.000000000 +0700
@@ -21,5 +21,5 @@
 #
 SHELL=/bin/bash
-ALDOR_PROJECT=d:\projects\aldor\aldorsvn
+ALDOR_PROJECT=$(S)/$(PN)
 
 # All these things under $(ALDOR_PROJECT) can actually be in unrelated places,
@@ -33,5 +33,5 @@
 # MACHINE should be one of: linux, win32gcc, win32msvc, sun4os55g64
 # or the machines listed in:  $(ALDOR_PROJECT)/aldor/tools/unix/docc.
-MACHINE=win32msvc
+MACHINE=linux
 
 # This varible sets the name of the libraries to be built
@@ -43,5 +43,5 @@
 # Note: the order in the list sets the order of building the libraries.
 #LIBS_TO_BUILD=aldor axllib debuglib algebra
-LIBS_TO_BUILD=aldor axllib 
+LIBS_TO_BUILD=aldor axllib algebra
 
 # This is used to form the name of the tarball archive used for binary 
