diff -u -r cmake-2.8.1.medium/CMakeLists.txt cmake-2.8.1/CMakeLists.txt
--- cmake-2.8.1.medium/CMakeLists.txt	2010-03-16 21:29:28.000000000 +0100
+++ cmake-2.8.1/CMakeLists.txt	2010-04-04 20:04:47.000000000 +0200
@@ -314,9 +314,6 @@
   ELSE (UNIX)
     SET(BUILD_CursesDialog 0)
   ENDIF (UNIX)
-  IF(BUILD_CursesDialog)
-    ADD_SUBDIRECTORY(Source/CursesDialog/form)
-  ENDIF(BUILD_CursesDialog)
 ENDMACRO (CMAKE_BUILD_UTILITIES)
 
 
diff -u -r cmake-2.8.1.medium/Source/CursesDialog/CMakeLists.txt cmake-2.8.1/Source/CursesDialog/CMakeLists.txt
--- cmake-2.8.1.medium/Source/CursesDialog/CMakeLists.txt	2010-03-16 21:29:34.000000000 +0100
+++ cmake-2.8.1/Source/CursesDialog/CMakeLists.txt	2010-04-04 20:01:38.000000000 +0200
@@ -25,13 +25,18 @@
               CursesDialog/ccmake
    )
 
-INCLUDE_DIRECTORIES(${CMake_SOURCE_DIR}/Source/CursesDialog/form
-                    ${CMake_BINARY_DIR}/Source/CursesDialog/form)
 INCLUDE_DIRECTORIES(${CURSES_INCLUDE_PATH})
 
 
 ADD_EXECUTABLE(ccmake ${CURSES_SRCS} )
 TARGET_LINK_LIBRARIES(ccmake CMakeLib)
-TARGET_LINK_LIBRARIES(ccmake cmForm)
+TARGET_LINK_LIBRARIES(ccmake form)
+TARGET_LINK_LIBRARIES(ccmake ${CURSES_LIBRARY})
+IF(CURSES_EXTRA_LIBRARY)
+  TARGET_LINK_LIBRARIES(ccmake ${CURSES_EXTRA_LIBRARY})
+ENDIF(CURSES_EXTRA_LIBRARY)
+
 
 INSTALL_TARGETS(/bin ccmake)
+
+
diff -u -r cmake-2.8.1.medium/Source/CursesDialog/cmCursesStandardIncludes.h cmake-2.8.1/Source/CursesDialog/cmCursesStandardIncludes.h
--- cmake-2.8.1.medium/Source/CursesDialog/cmCursesStandardIncludes.h	2010-03-16 21:29:35.000000000 +0100
+++ cmake-2.8.1/Source/CursesDialog/cmCursesStandardIncludes.h	2010-04-04 20:01:38.000000000 +0200
@@ -15,8 +15,6 @@
  #define _MSE_INT_H
 #endif
 
-#include <cmFormConfigure.h>
-
 #if defined(__hpux)
 # define _BOOL_DEFINED
 # include <sys/time.h>
