--- cns_solve_1.2.orig/bin/install	2008-01-09 17:21:43.000000000 -0800
+++ cns_solve_1.2/bin/install	2008-01-09 17:21:59.000000000 -0800
@@ -78,12 +78,6 @@
   exit 1
 endif
 #
-if ( $CNS_ARCH =~ unknown* ) then
-  echo "the architecture of the machine is unknown - CNS_ARCH=$CNS_ARCH"
-  echo 'aborting installation'
-  exit 1
-endif
-#
 set prod=`echo $CNS_ARCH | awk 'BEGIN{FS="-"}{print $1}'`
 set cpu=`echo $CNS_ARCH | awk 'BEGIN{FS="-"}{print $2}' | sed 's/[0-9]*\.*[0-9]*$//'`
 set os=`echo $CNS_ARCH | awk 'BEGIN{FS="-"}{print $3}'`
