--- src/binding/f90/buildiface	2011-12-06 09:57:04.000000000 +0100
+++ src/binding/f90/buildiface	2011-12-06 09:57:54.000000000 +0100
@@ -117,9 +117,13 @@
 		  'Type_hindexed-3' => 'int[]',
 		  'Type_indexed-2' => 'int[]',
 		  'Type_indexed-3' => 'int[]',
+		  'Type_hvector-3' => 'int', 
 		  'Type_struct-2' => 'int[]',
 		  'Type_struct-3' => 'int[]',
 		  'Type_struct-4' => 'MPI_Datatype[]',
+		  'Type_extent-2' => 'int',
+		  'Type_lb-2' => 'int',
+		  'Type_ub-2' => 'int',
 		  'Waitall-2' => 'MPI_Request[]',
 		  'Waitall-3' => 'MPI_Status[]',
 		  'Waitany-2' => 'MPI_Request[]',
--- src/binding/f90/mpi_base.f90.in	2011-12-06 09:58:42.000000000 +0100
+++ src/binding/f90/mpi_base.f90.in	2011-12-06 10:00:30.000000000 +0100
@@ -15,9 +15,7 @@
        END SUBROUTINE MPI_COMM_FREE_KEYVAL
 
        SUBROUTINE MPI_TYPE_EXTENT(v0,v1,ierror)
-       USE MPI_CONSTANTS,ONLY:MPI_ADDRESS_KIND
-       INTEGER v0
-       INTEGER(KIND=MPI_ADDRESS_KIND) v1
+       INTEGER v0, v1
        INTEGER ierror
        END SUBROUTINE MPI_TYPE_EXTENT
 
@@ -114,9 +112,7 @@
        END SUBROUTINE MPI_OP_COMMUTATIVE
 
        SUBROUTINE MPI_TYPE_LB(v0,v1,ierror)
-       USE MPI_CONSTANTS,ONLY:MPI_ADDRESS_KIND
-       INTEGER v0
-       INTEGER(KIND=MPI_ADDRESS_KIND) v1
+       INTEGER v0, v1
        INTEGER ierror
        END SUBROUTINE MPI_TYPE_LB
 
@@ -562,9 +558,7 @@
        END SUBROUTINE MPI_TYPE_CREATE_RESIZED
 
        SUBROUTINE MPI_TYPE_UB(v0,v1,ierror)
-       USE MPI_CONSTANTS,ONLY:MPI_ADDRESS_KIND
-       INTEGER v0
-       INTEGER(KIND=MPI_ADDRESS_KIND) v1
+       INTEGER v0, v1
        INTEGER ierror
        END SUBROUTINE MPI_TYPE_UB
 
@@ -822,10 +816,7 @@
        END SUBROUTINE MPI_GET_VERSION
 
        SUBROUTINE MPI_TYPE_HVECTOR(v0,v1,v2,v3,v4,ierror)
-       USE MPI_CONSTANTS,ONLY:MPI_ADDRESS_KIND
-       INTEGER v0, v1
-       INTEGER(KIND=MPI_ADDRESS_KIND) v2
-       INTEGER v3, v4
+       INTEGER v0, v1, v2, v3, v4
        INTEGER ierror
        END SUBROUTINE MPI_TYPE_HVECTOR
 
