--- mpich2-1.0.8/src/pm/gforker/Makefile.in-orig	2008-11-01 13:18:37.685990975 -0400
+++ mpich2-1.0.8/src/pm/gforker/Makefile.in	2008-11-01 13:19:03.584804978 -0400
@@ -176,6 +176,9 @@
 
 # A special alternate installation target when using multiple process managers
 install-alt: mpiexec
+	if [ ! -d ${DESTDIR}${exec_prefix} ] ; then $(MKDIR_P) ${DESTDIR}${exec_prefix} ; fi
+	if [ ! -d ${DESTDIR}${prefix} ] ; then $(MKDIR_P) ${DESTDIR}${prefix} ; fi
+	if [ ! -d ${DESTDIR}${bindir} ] ; then $(MKDIR_P) ${DESTDIR}${bindir} ; fi
 	$(INSTALL_PROGRAM) $(INSTALL_STRIP_FLAG) mpiexec $(DESTDIR)${bindir}/mpiexec.gforker
 
 
