--- lib/Makefile.old	2010-11-13 22:36:58.064960678 +0100
+++ lib/Makefile	2010-11-13 22:40:01.403417781 +0100
@@ -145,7 +145,8 @@
 
 libHYPRE.so: ${FILES_HYPRE}
 	@echo  "Building libHYPRE ... "
-	${BUILD_CC_SHARED} -o $@ ${FILES_HYPRE}
+	${BUILD_CC_SHARED} -Wl,-soname,$@.2.4.0b -o $@.2.4.0b ${FILES_HYPRE}
+	ln -s $@.2.4.0b $@
 
 libHYPRE_LSI.a: ${FILES_LSI}
 	@echo  "Building libHYPRE_LSI ... "
@@ -170,5 +170,6 @@
 
 libHYPRE_LSI.so: ${FILES_LSI}
 	@echo  "Building libHYPRE_LSI ... "
-	${BUILD_CC_SHARED} -o $@ ${FILES_LSI}
+	${BUILD_CC_SHARED} -Wl,-soname,$@.2.4.0b -o $@.2.4.0b ${FILES_LSI}
+	ln -s $@.2.4.0b $@
 
