--- libdeflate-1.2/Makefile	2019-01-15 06:37:48.000000000 +0100
+++ libdeflate-1.2/Makefile	2019-01-22 14:06:13.427054892 +0100
@@ -21,7 +21,7 @@
 	      1>&2 2>/dev/null; then echo $(1); fi)
 
 override CFLAGS :=							\
-	$(CFLAGS) -O2 -fomit-frame-pointer -std=c99 -I. -Icommon	\
+	$(CFLAGS) -fomit-frame-pointer -std=c99 -I. -Icommon	\
 	-Wall -Wundef							\
 	$(call cc-option,-Wpedantic)					\
 	$(call cc-option,-Wdeclaration-after-statement)			\
@@ -34,6 +34,9 @@
 ##############################################################################
 
 PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INCDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
 
 SOVERSION          := 0
 STATIC_LIB_SUFFIX  := .a
@@ -240,20 +243,20 @@
 all:$(DEFAULT_TARGETS)
 
 install:all
-	install -Dm644 -t $(DESTDIR)$(PREFIX)/lib $(STATIC_LIB)
-	install -Dm755 -t $(DESTDIR)$(PREFIX)/lib $(SHARED_LIB)
-	ln -sf $(SHARED_LIB) $(DESTDIR)$(PREFIX)/lib/libdeflate.so
-	install -Dm644 -t $(DESTDIR)$(PREFIX)/include libdeflate.h
-	install -Dm755 gzip $(DESTDIR)$(PREFIX)/bin/libdeflate-gzip
-	ln -f $(DESTDIR)$(PREFIX)/bin/libdeflate-gzip $(DESTDIR)$(PREFIX)/bin/libdeflate-gunzip
+	install -Dm644 -t $(DESTDIR)$(LIBDIR) $(STATIC_LIB)
+	install -Dm755 -t $(DESTDIR)$(LIBDIR) $(SHARED_LIB)
+	ln -sf $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/libdeflate.so
+	install -Dm644 -t $(DESTDIR)$(INCDIR) libdeflate.h
+	install -Dm755 gzip $(DESTDIR)$(BINDIR)/libdeflate-gzip
+	ln -f $(DESTDIR)$(BINDIR)/libdeflate-gzip $(DESTDIR)$(BINDIR)/libdeflate-gunzip
 
 uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/lib/$(STATIC_LIB) \
-		$(DESTDIR)$(PREFIX)/lib/$(SHARED_LIB) \
-		$(DESTDIR)$(PREFIX)/lib/libdeflate.so \
-		$(DESTDIR)$(PREFIX)/include/libdeflate.h \
-		$(DESTDIR)$(PREFIX)/bin/libdeflate-gzip \
-		$(DESTDIR)$(PREFIX)/bin/libdeflate-gunzip
+	rm -f $(DESTDIR)$(LIBDIR)/$(STATIC_LIB) \
+		$(DESTDIR)$(LIBDIR)/$(SHARED_LIB) \
+		$(DESTDIR)$(LIBDIR)/libdeflate.so \
+		$(DESTDIR)$(INCDIR)/libdeflate.h \
+		$(DESTDIR)$(BINDIR)/libdeflate-gzip \
+		$(DESTDIR)$(BINDIR)/libdeflate-gunzip
 
 test_programs:$(TEST_PROGRAMS)
 
