--- PythonQt2.1_Qt4.8/CMakeLists.txt.auto-diff-temp	2012-11-30 16:38:06.374358937 +0100
+++ PythonQt2.1_Qt4.8/CMakeLists.txt	2012-11-30 16:39:37.421438916 +0100
@@ -215,7 +215,7 @@
 # See http://blog.onesadcookie.com/2008/01/installname-magic.html
 #
 set_target_properties(PythonQt  PROPERTIES
-  INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
+  INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}"
   )
 
 target_link_libraries(PythonQt
@@ -228,8 +228,8 @@
 
 install(TARGETS PythonQt
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib)
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(FILES ${headers} DESTINATION include/PythonQt)
 
 MESSAGE(STATUS "<<< Gentoo configuration >>>
