--- configure.in	2008-06-18 11:53:20.000000000 +0200
+++ configure.in.new	2009-08-23 22:10:20.000000000 +0200
@@ -124,11 +124,6 @@
 echo Constraint disabled value $
 fi
 
-
-#C and CPPFLAFS
-CFLAGS="-O9 -ffast-math -fomit-frame-pointer"
-CXXFLAGS="-O9 -ffast-math"
-
 #alm_sse
 mkasm=false
 case "${host-cpu}" in
@@ -224,195 +219,6 @@
 		version.h
 		Makefile
 		include/almost/algorithm/Makefile
-		include/almost/boost/algorithm/Makefile
-		include/almost/boost/algorithm/string/detail/Makefile
-		include/almost/boost/algorithm/string/Makefile
-		include/almost/boost/algorithm/string/std/Makefile
-		include/almost/boost/archive/detail/Makefile
-		include/almost/boost/archive/impl/Makefile
-		include/almost/boost/archive/iterators/Makefile
-		include/almost/boost/archive/Makefile
-		include/almost/boost/assign/Makefile
-		include/almost/boost/assign/std/Makefile
-		include/almost/boost/bind/Makefile
-		include/almost/boost/compatibility/cpp_c_headers/Makefile
-		include/almost/boost/compatibility/Makefile
-		include/almost/boost/config/abi/Makefile
-		include/almost/boost/config/compiler/Makefile
-		include/almost/boost/config/Makefile
-		include/almost/boost/config/platform/Makefile
-		include/almost/boost/config/stdlib/Makefile
-		include/almost/boost/date_time/gregorian/Makefile
-		include/almost/boost/date_time/local_time/Makefile
-		include/almost/boost/date_time/Makefile
-		include/almost/boost/date_time/posix_time/Makefile
-		include/almost/boost/detail/Makefile
-		include/almost/boost/dynamic_bitset/Makefile
-		include/almost/boost/filesystem/Makefile
-		include/almost/boost/format/detail/Makefile
-		include/almost/boost/format/Makefile
-		include/almost/boost/function/detail/Makefile
-		include/almost/boost/function/Makefile
-		include/almost/boost/functional/detail/Makefile
-		include/almost/boost/functional/hash/Makefile
-		include/almost/boost/functional/Makefile
-		include/almost/boost/graph/detail/Makefile
-		include/almost/boost/graph/Makefile
-		include/almost/boost/integer/Makefile
-		include/almost/boost/io/Makefile
-		include/almost/boost/iostreams/detail/adapter/Makefile
-		include/almost/boost/iostreams/detail/broken_overload_resolution/Makefile
-		include/almost/boost/iostreams/detail/config/Makefile
-		include/almost/boost/iostreams/detail/Makefile
-		include/almost/boost/iostreams/detail/streambuf/Makefile
-		include/almost/boost/iostreams/detail/vc6/Makefile
-		include/almost/boost/iostreams/device/Makefile
-		include/almost/boost/iostreams/filter/Makefile
-		include/almost/boost/iostreams/Makefile
-		include/almost/boost/iterator/detail/Makefile
-		include/almost/boost/iterator/Makefile
-		include/almost/boost/lambda/detail/Makefile
-		include/almost/boost/lambda/Makefile
-		include/almost/boost/logic/Makefile
-		include/almost/boost/Makefile
-		include/almost/boost/math/Makefile
-		include/almost/boost/math/special_functions/Makefile
-		include/almost/boost/mpl/aux_/config/Makefile
-		include/almost/boost/mpl/aux_/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/bcc/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/bcc551/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/dmc/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/gcc/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/msvc60/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/msvc70/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/mwcw/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/no_ctps/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/no_ttp/Makefile
-		include/almost/boost/mpl/aux_/preprocessed/plain/Makefile
-		include/almost/boost/mpl/aux_/preprocessor/Makefile
-		include/almost/boost/mpl/aux_/range_c/Makefile
-		include/almost/boost/mpl/aux_/test/Makefile
-		include/almost/boost/mpl/limits/Makefile
-		include/almost/boost/mpl/list/aux_/Makefile
-		include/almost/boost/mpl/list/aux_/preprocessed/Makefile
-		include/almost/boost/mpl/list/aux_/preprocessed/plain/Makefile
-		include/almost/boost/mpl/list/Makefile
-		include/almost/boost/mpl/Makefile
-		include/almost/boost/mpl/map/aux_/Makefile
-		include/almost/boost/mpl/map/aux_/preprocessed/Makefile
-		include/almost/boost/mpl/map/aux_/preprocessed/no_ctps/Makefile
-		include/almost/boost/mpl/map/aux_/preprocessed/plain/Makefile
-		include/almost/boost/mpl/map/aux_/preprocessed/typeof_based/Makefile
-		include/almost/boost/mpl/map/Makefile
-		include/almost/boost/mpl/math/Makefile
-		include/almost/boost/mpl/multiset/aux_/Makefile
-		include/almost/boost/mpl/multiset/Makefile
-		include/almost/boost/mpl/set/aux_/Makefile
-		include/almost/boost/mpl/set/aux_/preprocessed/Makefile
-		include/almost/boost/mpl/set/aux_/preprocessed/plain/Makefile
-		include/almost/boost/mpl/set/Makefile
-		include/almost/boost/mpl/vector/aux_/Makefile
-		include/almost/boost/mpl/vector/aux_/preprocessed/Makefile
-		include/almost/boost/mpl/vector/aux_/preprocessed/no_ctps/Makefile
-		include/almost/boost/mpl/vector/aux_/preprocessed/plain/Makefile
-		include/almost/boost/mpl/vector/aux_/preprocessed/typeof_based/Makefile
-		include/almost/boost/mpl/vector/Makefile
-		include/almost/boost/multi_array/Makefile
-		include/almost/boost/multi_index/detail/Makefile
-		include/almost/boost/multi_index/Makefile
-		include/almost/boost/numeric/conversion/detail/Makefile
-		include/almost/boost/numeric/conversion/Makefile
-		include/almost/boost/numeric/interval/compare/Makefile
-		include/almost/boost/numeric/interval/detail/Makefile
-		include/almost/boost/numeric/interval/ext/Makefile
-		include/almost/boost/numeric/interval/Makefile
-		include/almost/boost/numeric/Makefile
-		include/almost/boost/numeric/ublas/detail/Makefile
-		include/almost/boost/numeric/ublas/Makefile
-		include/almost/boost/optional/Makefile
-		include/almost/boost/parameter/aux_/Makefile
-		include/almost/boost/parameter/Makefile
-		include/almost/boost/pending/detail/Makefile
-		include/almost/boost/pending/Makefile
-		include/almost/boost/pool/detail/Makefile
-		include/almost/boost/pool/Makefile
-		include/almost/boost/preprocessor/arithmetic/detail/Makefile
-		include/almost/boost/preprocessor/arithmetic/Makefile
-		include/almost/boost/preprocessor/array/Makefile
-		include/almost/boost/preprocessor/comparison/Makefile
-		include/almost/boost/preprocessor/config/Makefile
-		include/almost/boost/preprocessor/control/detail/dmc/Makefile
-		include/almost/boost/preprocessor/control/detail/edg/Makefile
-		include/almost/boost/preprocessor/control/detail/Makefile
-		include/almost/boost/preprocessor/control/detail/msvc/Makefile
-		include/almost/boost/preprocessor/control/Makefile
-		include/almost/boost/preprocessor/debug/Makefile
-		include/almost/boost/preprocessor/detail/dmc/Makefile
-		include/almost/boost/preprocessor/detail/Makefile
-		include/almost/boost/preprocessor/facilities/Makefile
-		include/almost/boost/preprocessor/iteration/detail/bounds/Makefile
-		include/almost/boost/preprocessor/iteration/detail/iter/Makefile
-		include/almost/boost/preprocessor/iteration/detail/Makefile
-		include/almost/boost/preprocessor/iteration/Makefile
-		include/almost/boost/preprocessor/list/detail/dmc/Makefile
-		include/almost/boost/preprocessor/list/detail/edg/Makefile
-		include/almost/boost/preprocessor/list/detail/Makefile
-		include/almost/boost/preprocessor/list/Makefile
-		include/almost/boost/preprocessor/logical/Makefile
-		include/almost/boost/preprocessor/Makefile
-		include/almost/boost/preprocessor/punctuation/Makefile
-		include/almost/boost/preprocessor/repetition/detail/dmc/Makefile
-		include/almost/boost/preprocessor/repetition/detail/edg/Makefile
-		include/almost/boost/preprocessor/repetition/detail/Makefile
-		include/almost/boost/preprocessor/repetition/detail/msvc/Makefile
-		include/almost/boost/preprocessor/repetition/Makefile
-		include/almost/boost/preprocessor/selection/Makefile
-		include/almost/boost/preprocessor/seq/detail/Makefile
-		include/almost/boost/preprocessor/seq/Makefile
-		include/almost/boost/preprocessor/slot/detail/Makefile
-		include/almost/boost/preprocessor/slot/Makefile
-		include/almost/boost/preprocessor/tuple/Makefile
-		include/almost/boost/program_options/detail/Makefile
-		include/almost/boost/program_options/Makefile
-		include/almost/boost/ptr_container/detail/Makefile
-		include/almost/boost/ptr_container/Makefile
-		include/almost/boost/python/converter/Makefile
-		include/almost/boost/python/detail/Makefile
-		include/almost/boost/python/Makefile
-		include/almost/boost/python/object/Makefile
-		include/almost/boost/python/suite/indexing/detail/Makefile
-		include/almost/boost/python/suite/indexing/Makefile
-		include/almost/boost/python/suite/Makefile
-		include/almost/boost/random/detail/Makefile
-		include/almost/boost/random/Makefile
-		include/almost/boost/range/detail/Makefile
-		include/almost/boost/range/detail/mfc/Makefile
-		include/almost/boost/range/detail/vc6/Makefile
-		include/almost/boost/range/Makefile
-		include/almost/boost/regex/config/Makefile
-		include/almost/boost/regex/Makefile
-		include/almost/boost/regex/pending/Makefile
-		include/almost/boost/regex/v4/Makefile
-		include/almost/boost/serialization/detail/Makefile
-		include/almost/boost/serialization/Makefile
-		include/almost/boost/signals/detail/Makefile
-		include/almost/boost/signals/Makefile
-		include/almost/boost/thread/detail/Makefile
-		include/almost/boost/thread/Makefile
-		include/almost/boost/tuple/detail/Makefile
-		include/almost/boost/tuple/Makefile
-		include/almost/boost/type_traits/detail/Makefile
-		include/almost/boost/type_traits/Makefile
-		include/almost/boost/utility/detail/Makefile
-		include/almost/boost/utility/Makefile
-		include/almost/boost/variant/detail/Makefile
-		include/almost/boost/variant/Makefile
-		include/almost/boost/wave/cpplexer/Makefile
-		include/almost/boost/wave/cpplexer/re2clex/Makefile
-		include/almost/boost/wave/grammars/Makefile
-		include/almost/boost/wave/Makefile
-		include/almost/boost/wave/util/Makefile
 		include/almost/boost_1_30/config/compiler/Makefile
 		include/almost/boost_1_30/config/Makefile
 		include/almost/boost_1_30/config/platform/Makefile
@@ -441,7 +247,6 @@
 		include/almost/io/Makefile
 		include/almost/kernel/Makefile
 		include/almost/kernel/scratch/Makefile
-		include/almost/libxml/Makefile
 		include/almost/Makefile
 		include/almost/mbmd/Makefile
 		include/almost/misc/Makefile
@@ -460,7 +265,6 @@
 		include/almost/nmr/Makefile
 		include/almost/tmd/Makefile
 		include/almost/template/Makefile
-		include/almost/tinyxml/Makefile
 		include/almost/types/Makefile
 		include/almost/zeta/Makefile
 		include/Makefile
@@ -470,8 +274,6 @@
 	src/lib/Makefile 
 	src/lib/modules/Makefile 
 	src/Makefile 
-	src/tinyxml/Makefile
-	src/libxml/Makefile
 	src/forcefield/Makefile
 	src/examples/Makefile
 	src/tests/Makefile
