--- configure.ac.orig	2009-03-25 18:21:24.000000000 +0000
+++ configure.ac	2009-03-25 18:27:33.000000000 +0000
@@ -20,42 +20,17 @@
 AC_SUBST(DATE3, "$date3")
 
 # Include macros
-sinclude(acx_prog_cc_optim.m4)
 sinclude(acx_pthread.m4)
 sinclude(acx_urbi_resolve_dir.m4)
 
 # Display pakage and version number
 AC_MSG_RESULT([******** Configuring:  $PACKAGE_NAME $VERSION ($date) ********])
 
-# Initialize the list of compilers to consider
-cclist="cc gcc"
-
-# Backup and reset the input CFLAGS
-mycflags="$CFLAGS"
-CFLAGS=""
-myldflags="$LDFLAGS"
-LDFLAGS=""
-
-# Provide special option for the Linux Intel C compiler
-AC_MSG_CHECKING([for Linux Intel C compiler mode])
-AC_ARG_ENABLE(icc,
-	[AC_HELP_STRING([--enable-icc],
-	[Enable special mode for compilation with the Intel compiler \
-(off by default)])],
-        use_icc="yes"
-        cclist="icc $cclist"
-	AC_MSG_RESULT([yes]),
-        use_icc="no"
-	AC_MSG_RESULT([no]))
-
 # Checks for programs.
-# GCC is chosen last because it is likely to yield less optimized code
-AC_PROG_CC([$cclist])
 # C Compiler: Check that it is ANSI C
 AM_PROG_CC_STDC
 # C Compiler: Check that it is POSIX-compliant
 AC_ISC_POSIX
-ACX_PROG_CC_OPTIM
 AC_PROG_RANLIB
 AC_PROG_INSTALL
 
@@ -154,19 +129,6 @@
   use_static="no"
 fi
 
-# Static linking option
-if test "$use_static" = "yes"; then
-  LDFLAGS="-static -shared-libgcc $LDFLAGS"
-fi
-
-# Override automatic CFLAGS and LDFLAGS with those of user
-#if test -n "$mycflags"; then
-#CFLAGS="$mycflags"
-#fi
-#if test -n "$myldflags"; then
-#LDFLAGS="$myldflags"
-#fi
-
 # Display compiler and linker flags
 AC_MSG_RESULT([***************************************************************])
 AC_MSG_RESULT([Compile cmdline:  $CC $CFLAGS])
