diff --recursive -U2 TeXmacs-1.0.6.10-src.orig/src/Typeset/Boxes/Modifier/change_boxes.cpp TeXmacs-1.0.6.10-src/src/Typeset/Boxes/Modifier/change_boxes.cpp
--- TeXmacs-1.0.6.10-src.orig/src/Typeset/Boxes/Modifier/change_boxes.cpp	2007-05-14 13:54:35.000000000 +0700
+++ TeXmacs-1.0.6.10-src/src/Typeset/Boxes/Modifier/change_boxes.cpp	2007-05-16 05:59:55.000000000 +0700
@@ -325,5 +325,5 @@
     finalize ();
   }
-  inline remember_box_rep::~remember_box_rep () {
+  inline ~remember_box_rep () {
     if (logs_ptr != NULL) {
       rectangles& logs= *logs_ptr;
