diff -U2 --recursive TeXmacs-1.0.6.9-src.orig/plugins/maxima/bin/maxima_detect TeXmacs-1.0.6.9-src/plugins/maxima/bin/maxima_detect
--- TeXmacs-1.0.6.9-src.orig/plugins/maxima/bin/maxima_detect	2007-02-11 21:39:53.000000000 +0600
+++ TeXmacs-1.0.6.9-src/plugins/maxima/bin/maxima_detect	2007-05-05 05:31:03.000000000 +0700
@@ -64,7 +64,8 @@
 version 5.9.3
 version 5.10
-version 5.11" >/dev/null
+version 5.11
+version 5.12" >/dev/null
         then
-          # 5.9.1 or 5.9.2 or 5.9.3 or 5.10 or 5.11
+          # 5.9.1 or 5.9.2 or 5.9.3 or 5.10 or 5.11 or 5.12
           maxima -d | grep -F 'maxima-htmldir=' | sed -e \
           's/maxima-htmldir=/"/' -e 's|$|/maxima_toc.html"|'
diff -U2 --recursive TeXmacs-1.0.6.9-src.orig/plugins/maxima/bin/tm_maxima TeXmacs-1.0.6.9-src/plugins/maxima/bin/tm_maxima
--- TeXmacs-1.0.6.9-src.orig/plugins/maxima/bin/tm_maxima	2007-02-11 21:39:53.000000000 +0600
+++ TeXmacs-1.0.6.9-src/plugins/maxima/bin/tm_maxima	2007-05-05 05:31:20.000000000 +0700
@@ -21,5 +21,5 @@
   5.9.1.1* | 5.9.2* | 5.9.3*) exec maxima -u $1 -l $2 -p "$TEXMACS_MAXIMA_PATH/texmacs-maxima-5.9.2.lisp";;
   5.10.*) exec maxima -u $1 -l $2 -p "$TEXMACS_MAXIMA_PATH/texmacs-maxima-5.10.0.lisp";;
-  5.11.*) exec maxima -u $1 -l $2 -p "$TEXMACS_MAXIMA_PATH/texmacs-maxima-5.11.0.lisp";;
+  5.11.* | 5.12.*) exec maxima -u $1 -l $2 -p "$TEXMACS_MAXIMA_PATH/texmacs-maxima-5.11.0.lisp";;
   cygwin)
     CYGWIN_ROOT="C:\cygwin"
