diff --git a/CMakeLists.txt b/CMakeLists.txt
index da138c1..ecab139 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,15 +28,7 @@ install(FILES ${_dpctl_capi_headers}
 # Define CMAKE_INSTALL_xxx: LIBDIR, INCLUDEDIR
 include(GNUInstallDirs)
 
-# Fetch pybind11
-include(FetchContent)
-
-FetchContent_Declare(
-    pybind11
-    URL https://github.com/pybind/pybind11/archive/refs/tags/v2.10.1.tar.gz
-    URL_HASH SHA256=111014b516b625083bef701df7880f78c2243835abdb263065b6b59b960b6bad
-)
-FetchContent_MakeAvailable(pybind11)
+include(/usr/share/cmake/pybind11/pybind11Config.cmake)
 
 add_subdirectory(dpctl)
 
