diff --git a/libsyclinterface/cmake/modules/GetLevelZeroHeaders.cmake b/libsyclinterface/cmake/modules/GetLevelZeroHeaders.cmake
index 2929c33..731f054 100644
--- a/libsyclinterface/cmake/modules/GetLevelZeroHeaders.cmake
+++ b/libsyclinterface/cmake/modules/GetLevelZeroHeaders.cmake
@@ -26,76 +26,10 @@
 
 function(get_level_zero_headers)
 
-    if(EXISTS level-zero)
-      # Update the checkout
-        execute_process(
-            COMMAND ${GIT_EXECUTABLE} fetch
-            RESULT_VARIABLE result
-            ERROR_VARIABLE error
-            WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/level-zero
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-            ERROR_STRIP_TRAILING_WHITESPACE
-        )
-
-        if(NOT result EQUAL 0)
-            message(FATAL_ERROR
-                "Could not update Level Zero sources."
-            )
-        endif()
-    else()
-        # Clone the Level Zero git repo
-        execute_process(
-            COMMAND ${GIT_EXECUTABLE} clone https://github.com/oneapi-src/level-zero.git
-            RESULT_VARIABLE result
-            ERROR_VARIABLE error
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-            ERROR_STRIP_TRAILING_WHITESPACE
-        )
-
-        if(NOT result EQUAL 0)
-            message(FATAL_ERROR
-                "Could not clone Level Zero sources from github.com/oneapi-src/level-zero."
-            )
-        endif()
-    endif()
-
-    # Use git describe to get latest tag name
-    execute_process(
-        COMMAND ${GIT_EXECUTABLE} describe --tags --abbrev=0
-        RESULT_VARIABLE result
-        OUTPUT_VARIABLE latest_tag
-        ERROR_VARIABLE error
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/level-zero
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-        ERROR_STRIP_TRAILING_WHITESPACE
-    )
-
-    if(NOT result EQUAL 0)
-        message(FATAL_ERROR
-            "Could not get the name for the latest release."
-        )
-    endif()
-
-    # Use git describe to get latest tag name
-    execute_process(
-        COMMAND ${GIT_EXECUTABLE} checkout ${latest_tag}
-        RESULT_VARIABLE result
-        ERROR_VARIABLE error
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/level-zero
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-        ERROR_STRIP_TRAILING_WHITESPACE
-    )
-
-    if(NOT result EQUAL 0)
-        message(FATAL_ERROR
-            "Could not checkout the latest release."
-        )
-    endif()
-
     # Populate the path to the headers
     find_path(LEVEL_ZERO_INCLUDE_DIR
         NAMES zet_api.h
-        PATHS ${CMAKE_BINARY_DIR}/level-zero/include
+        PATHS /usr/include/level_zero
         NO_DEFAULT_PATH
         NO_CMAKE_ENVIRONMENT_PATH
         NO_CMAKE_PATH
