--- tools/build_html_index.sh.orig	2007-07-01 09:47:50.000000000 -0700
+++ tools/build_html_index.sh	2007-07-01 09:52:47.000000000 -0700
@@ -129,10 +129,10 @@
 
 #copy over CSS:
 cp -f grassdocs.css $HTMLDIR/
-cp -f grassdocs.css $HTMLDIR/nviz/
+#cp -f grassdocs.css $HTMLDIR/nviz/
 #copy over GRASS logo:
 cp -f grass.smlogo.gif $HTMLDIR/
-cp -f grass.smlogo.gif $HTMLDIR/nviz/
+#cp -f grass.smlogo.gif $HTMLDIR/nviz/
 #copy over GEM docs:
 mkdir -p $HTMLDIR/gem
 cp -f $GEMDIR/docs/GEM-Manual/*.html $HTMLDIR/gem/
@@ -163,7 +163,7 @@
 echo "<tr><td>&nbsp;&nbsp;r3.* </td><td>raster3D commands</td></tr>" >> $FULLINDEX
 echo "<tr><td>&nbsp;&nbsp;v.*  </td><td>vector commands</td></tr>" >> $FULLINDEX
 echo "<tr><td>&nbsp;&nbsp;<a href=\"gis.m.html\">gis.m</a> </td><td>GUI frontend to GIS menus and display</td></tr>" >> $FULLINDEX
-echo "<tr><td>&nbsp;&nbsp;<a href=\"nviz.html\">nviz</a> </td><td>visualization suite</td></tr>" >> $FULLINDEX
+#echo "<tr><td>&nbsp;&nbsp;<a href=\"nviz.html\">nviz</a> </td><td>visualization suite</td></tr>" >> $FULLINDEX
 echo "<tr><td>&nbsp;&nbsp;<a href=\"xganim.html\">xganim</a> </td><td>raster map slideshow</td></tr>" >> $FULLINDEX
 echo "</table>" >> $FULLINDEX
 echo "<p>" >> $FULLINDEX
@@ -245,7 +245,7 @@
 
 #extra stuff for 'nviz' and 'xganim' and GUIs:
 echo "<li><a href=\"gis.m.html\">gis.m</a> and <a href=\"d.m.html\">d.m</a> GIS managers</li>" >> $FILENAME
-echo "<li><a href=\"nviz.html\">nviz</a> visualization and animation tool</li>" >> $FILENAME
+#echo "<li><a href=\"nviz.html\">nviz</a> visualization and animation tool</li>" >> $FILENAME
 echo "<li><a href=\"xganim.html\">xganim</a> tool  for animating a raster map series</li>" >> $FILENAME
 	      
 echo "</ul>" >> $FILENAME
