--- tools/build_html_index.sh.orig	2008-04-16 10:21:15.000000000 +0200
+++ tools/build_html_index.sh	2008-04-16 10:23:52.000000000 +0200
@@ -141 +141 @@
-cp -f grassdocs.css $HTMLDIR/nviz/
+#cp -f grassdocs.css $HTMLDIR/nviz/
@@ -145 +145 @@
-cp -f grass_logo.png $HTMLDIR/nviz/
+#cp -f grass_logo.png $HTMLDIR/nviz/
@@ -182 +182 @@
-echo "<tr><td>&nbsp;&nbsp;<a href=\"nviz.html\">nviz</a> </td><td>visualization suite</td></tr>" >> $FULLINDEX
+#echo "<tr><td>&nbsp;&nbsp;<a href=\"nviz.html\">nviz</a> </td><td>visualization suite</td></tr>" >> $FULLINDEX
@@ -262 +262 @@
-echo "<li><a href=\"nviz.html\">nviz</a> visualization and animation tool</li>" >> $FILENAME
+#echo "<li><a href=\"nviz.html\">nviz</a> visualization and animation tool</li>" >> $FILENAME
