--- lib/ogsf/gsd_img_ppm.c.orig	2007-08-22 19:07:35.000000000 -0700
+++ lib/ogsf/gsd_img_ppm.c	2007-08-22 19:10:58.000000000 -0700
@@ -119,10 +119,9 @@
 
         avcodec_init();
 
-        register_avcodec(&mpeg1video_encoder); /* just mpeg1 */
-	/* -- use with othe formats ...
+        /* register_avcodec(&mpeg1video_encoder);  just mpeg1 */
+	/* -- use with all formats ... */
 	avcodec_register_all();
-	*/
 
         /* find the mpeg1 video encoder */
         codec = avcodec_find_encoder(CODEC_ID_MPEG1VIDEO);
@@ -144,7 +143,7 @@
 	For Xvid the canvas size must be a multiple of (4?). Best try 16, 
 	 720x720. (File->Set Canval Size) Otherwise the codec won't open.
 	*/
-	register_avcodec(&xvid_encoder);
+	avcodec_register_all();
 	codec = avcodec_find_encoder(CODEC_ID_XVID);
 	bitrate=2000000;
 #endif
