--- gasgano/src/Makefile.am.orig	2009-02-01 20:54:37.000000000 +0000
+++ gasgano/src/Makefile.am	2009-02-01 21:08:40.000000000 +0000
@@ -3,10 +3,10 @@
 
 JAVAROOT = .
 
-GNUREGEXP = $(top_srcdir)/external/gnu-regexp.jar
+GNUREGEXP = /usr/share/gnu-regexp-1/lib/gnu-regexp.jar
 JALSTRING = $(top_srcdir)/external/jalString.jar
 JCONNECT = $(top_srcdir)/external/jConnect40.jar
-JUNIT = $(top_srcdir)/external/junit.jar
+JUNIT = /usr/share/junit/lib/junit.jar
 
 AM_JAVACFLAGS = -source 1.4
 
