--- configure.ac.orig	2009-11-23 09:30:12.000000000 +1300
+++ configure.ac	2009-11-23 11:35:59.000000000 +1300
@@ -3,6 +3,7 @@
 AC_PREREQ(2.59)
 AC_INIT(IML, 1.0.3)
 AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_MACRO_DIR(config)
 AM_CONFIG_HEADER(config.h:config-h.in)
 AC_CONFIG_SRCDIR([src/RNSop.c])
 AM_INIT_AUTOMAKE
@@ -55,11 +56,18 @@
 respectively when running configure.
 )])
 
-IML_CHECK_ATLAS(,,[AC_MSG_ERROR(
-ATLAS not found! 
-ATLAS version 3.0 or greater is required for this library to compile. Please make sure ATLAS is installed and specify the header and libraries location with the options --with-atlas-include=<path> and --with-atlas-lib=<path> respectively when running configure.
-)])
+AC_CHECK_HEADERS([cblas.h],
+	[cblas_headers_found=yes; break;])
+
+AS_IF([test "x$cblas_headers_found" != "xyes"],
+	[AC_MSG_ERROR([Unable to find the cblas headers check your installation])])
+
+AC_ARG_WITH(cblas-lib,
+            AC_HELP_STRING([--with-cblas-lib=<lib>],[Specify the flags for cblas.]),
+            with_cblas_lib=$withval)
+ATLAS_LIBS="$with_cblas_lib"
 
+AC_SUBST(ATLAS_LIBS)
 AC_SUBST(LDFLAGS)
 AC_SUBST(ac_aux_dir)
 	
--- Makefile.am~	2009-11-23 11:38:44.000000000 +1300
+++ Makefile.am	2009-11-23 11:38:44.000000000 +1300
@@ -1,3 +1,5 @@
+ACLOCAL_AMFLAGS = -I config
+
 EXTRA_DIST = bootstrap
 
 SUBDIRS = repl config src doc tests examples
