--- configure.ac.orig	2008-06-15 23:51:30.000000000 +0000
+++ configure.ac	2010-08-07 08:38:44.689081000 +0000
@@ -55,10 +55,7 @@
 respectively when running configure.
 )])
 
-IML_CHECK_ATLAS(,,[AC_MSG_ERROR(
-ATLAS not found! 
-ATLAS version 3.0 or greater is required for this library to compile. Please make sure ATLAS is installed and specify the header and libraries location with the options --with-atlas-include=<path> and --with-atlas-lib=<path> respectively when running configure.
-)])
+PKG_CHECK_MODULES([ATLAS], [cblas])
 
 AC_SUBST(LDFLAGS)
 AC_SUBST(ac_aux_dir)
