--- ngsolve/linalg/order.cpp.orig	2008-05-08 17:44:10.000000000 +0100
+++ ngsolve/linalg/order.cpp	2008-05-09 08:50:56.000000000 +0100
@@ -54,7 +54,7 @@
     return ball.Alloc();
   }
 
-  void CliqueEl :: operator delete (void * p, size_t)
+  void CliqueEl :: operator delete (void * p)
   {
     ball.Free (p);
   }
--- ngsolve/linalg/order.hpp.orig	2008-05-08 17:44:10.000000000 +0100
+++ ngsolve/linalg/order.hpp	2008-05-09 08:50:56.000000000 +0100
@@ -118,7 +118,7 @@
   ///
   void * operator new(size_t);
   ///
-  void operator delete (void *, size_t);
+  void operator delete (void *);
 };
   
 
