--- togl/togl.cpp.org	2004-09-29 19:48:06.000000000 +0200
+++ togl/togl.cpp	2006-10-13 21:39:50.000000000 +0200
@@ -138,8 +138,50 @@
 #  include "tkInt8.4p3.h"
 #elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==4
 #  include "tkInt8.4p4.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==5
+#  include "tkInt8.4p5.h"
 #elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==6
 #  include "tkInt8.4p6.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==7
+#  include "tkInt8.4p7.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==8
+#  include "tkInt8.4p8.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==9
+#  include "tkInt8.4p9.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==10
+#  include "tkInt8.4p10.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==11
+#  include "tkInt8.4p11.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==12
+#  include "tkInt8.4p12.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==13
+#  include "tkInt8.4p13.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==14
+#  include "tkInt8.4p14.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==15
+#  include "tkInt8.4p15.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==16
+#  include "tkInt8.4p16.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==17
+#  include "tkInt8.4p17.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==18
+#  include "tkInt8.4p18.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==19
+#  include "tkInt8.4p19.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==20
+#  include "tkInt8.4p20.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==21
+#  include "tkInt8.4p21.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==22
+#  include "tkInt8.4p22.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==23
+#  include "tkInt8.4p23.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==24
+#  include "tkInt8.4p24.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==25
+#  include "tkInt8.4p25.h"
+#elif TK_MAJOR_VERSION==8 && TK_MINOR_VERSION==4 && TK_RELEASE_SERIAL==26
+#  include "tkInt8.4p26.h"
 #else
    Sorry, you will have to edit togl.c to include the right tkInt.h file
 #endif
