diff --git a/src/condor_examples/condor_config.generic b/src/condor_examples/condor_config.generic
index f3f60a5..94c146f 100644
--- a/src/condor_examples/condor_config.generic
+++ b/src/condor_examples/condor_config.generic
@@ -1,15 +1,35 @@
 ######################################################################
+######################################################################
+##                                                                  ##
+##  N O T I C E:                                                    ##
+##                                                                  ##
+##       Customization of this file should be done via the          ##
+##       LOCAL_CONFIG_FILE.                                         ##
+##                                                                  ##
+######################################################################
+######################################################################
+
+
+######################################################################
 ##
 ##  condor_config
 ##
 ##  This is the global configuration file for condor.  Any settings
-##  made here may potentially be overridden in the local configuration
-##  file.  KEEP THAT IN MIND!  To double-check that a variable is
-##  getting set from the configuration file that you expect, use
-##  condor_config_val -v <variable name>
+##  found here * * s h o u l d   b e   c u s t o m i z e d   i n
+##  t h e   l o c a l   c o n f i g u r a t i o n   f i l e. * *
+##
+##  The local configuration file is specified by LOCAL_CONFIG_FILE
+##  below.
+##
+##  For a basic configuration, you may only want to start by
+##  customizing CONDOR_ADMIN.
+##
+##  Note: To double-check where a configuration variable is set from
+##  you can use condor_config_val -v <variable name>,
+##  e.g. condor_config_val -v CONDOR_ADMIN.
 ##
 ##  The file is divided into four main parts:
-##  Part 1:  Settings you MUST customize 
+##  Part 1:  Settings you likely want to customize 
 ##  Part 2:  Settings you may want to customize
 ##  Part 3:  Settings that control the policy of when condor will
 ##           start and stop jobs on your machines
@@ -47,18 +62,15 @@
 ##  #        #    #  #   #      #              #
 ##  #        #    #  #    #     #            #####
 ##
-##  Part 1:  Settings you must customize:
+##  Part 1:  Settings you likely want to customize:
 ######################################################################
 ######################################################################
 
-##  What machine is your central manager?
-CONDOR_HOST	= central-manager-hostname.your.domain
-
 ##--------------------------------------------------------------------
 ##  Pathnames:
 ##--------------------------------------------------------------------
 ##  Where have you installed the bin, sbin and lib condor directories?   
-RELEASE_DIR		= /usr/local/condor
+RELEASE_DIR		= /usr
 
 ##  Where is the local condor directory for each host?  
 ##  This is where the local config file(s), logs and
@@ -84,7 +96,7 @@ CONDOR_ADMIN		= condor-admin@your.domain
 
 ##  Full path to a mail delivery program that understands that "-s"
 ##  means you want to specify a subject:
-MAIL			= /usr/bin/mail
+MAIL			= /bin/mail
 
 ##--------------------------------------------------------------------
 ##  Network domain parameters:
@@ -93,13 +105,13 @@ MAIL			= /usr/bin/mail
 ##  machines don't share a common UID space, set it to 
 ##  UID_DOMAIN = $(FULL_HOSTNAME)
 ##  to specify that each machine has its own UID space.
-UID_DOMAIN		= your.domain
+UID_DOMAIN		= $(FULL_HOSTNAME)
 
 ##  Internet domain of machines sharing a common file system.
 ##  If your machines don't use a network file system, set it to
 ##  FILESYSTEM_DOMAIN = $(FULL_HOSTNAME)
 ##  to specify that each machine has its own file system. 
-FILESYSTEM_DOMAIN	= your.domain
+FILESYSTEM_DOMAIN	= $(FULL_HOSTNAME)
 
 ##  This macro is used to specify a short description of your pool. 
 ##  It should be about 20 characters long. For example, the name of 
@@ -218,7 +230,7 @@ HOSTALLOW_READ = *
 ##    HOSTALLOW_WRITE = *
 ##  but note that this will allow anyone to submit jobs or add
 ##  machines to your pool and is serious security risk.
-HOSTALLOW_WRITE = YOU_MUST_CHANGE_THIS_INVALID_CONDOR_CONFIGURATION_VALUE
+HOSTALLOW_WRITE = $(FULL_HOSTNAME)
 #HOSTALLOW_WRITE = *.your.domain, your-friend's-machine.other.domain
 #HOSTDENY_WRITE = bad-machine.your.domain
 
@@ -872,9 +883,10 @@ SPOOL		= $(LOCAL_DIR)/spool
 EXECUTE		= $(LOCAL_DIR)/execute
 BIN		= $(RELEASE_DIR)/bin
 LIB		= $(RELEASE_DIR)/lib
-INCLUDE		= $(RELEASE_DIR)/include
+INCLUDE		= $(RELEASE_DIR)/include/condor
 SBIN		= $(RELEASE_DIR)/sbin
-LIBEXEC		= $(RELEASE_DIR)/libexec
+LIBEXEC		= $(RELEASE_DIR)/libexec/condor
+SHARE		= $(RELEASE_DIR)/share/condor
 
 ## If you leave HISTORY undefined (comment it out), no history file
 ## will be created. 
@@ -1655,7 +1667,7 @@ JAVA_MAXHEAP_ARGUMENT = -Xmx
 ## them here.  However, do not remove the existing entries, as Condor
 ## needs them.
 
-JAVA_CLASSPATH_DEFAULT = $(LIB) $(LIB)/scimark2lib.jar .
+JAVA_CLASSPATH_DEFAULT = $(SHARE) $(SHARE)/scimark2lib.jar .
 
 ##  JAVA_CLASSPATH_ARGUMENT describes the command-line parameter
 ##  used to introduce a new classpath:
diff --git a/src/condor_examples/customize b/src/condor_examples/customize
index b1c5710..cb2dea1 100644
--- a/src/condor_examples/customize
+++ b/src/condor_examples/customize
@@ -54,7 +54,7 @@ sub get_platform_defaults {
     $_ = $os;
   SWITCH: {
       if(/^Linux/) { 
-	  $mail_path="/usr/bin/mail";
+	  $mail_path="/bin/mail";
 	  $ps_path="/bin/ps auwx";
 	  $console_devs="mouse, console";
 	  last SWITCH;
