--- Makefile.orig	2007-08-01 17:41:38.172069500 +0100
+++ Makefile	2007-08-01 17:57:17.200552750 +0100
@@ -351,40 +351,41 @@
 tclsh	: FORCE
 	@echo "Installing Tcl shared..."
 	cd $(TCLDIRDIR); CC='$(CC) $(OPTS) $(TCLOPTS)' ./configure $(TCLFLAGS) --enable-shared
-	cd $(TCLDIRDIR); $(MAKE) install
+	cd $(TCLDIRDIR); $(MAKE) install-binaries install-libraries
 
 tcl	: FORCE
 	@echo "Installing Tcl noshared..."
 	cd $(TCLDIRDIR); CC='$(CC) $(OPTS) $(TCLOPTS)' ./configure $(TCLFLAGS) --disable-shared
 	cd $(TCLDIRDIR); $(MAKE) install
+	cd $(TCLDIRDIR); $(MAKE) install-binaries install-libraries
 
 tksh	: FORCE
 	@echo "Installing Tk shared..."
 	cd $(TKDIRDIR); CC='$(CC) $(OPTS) $(TCLOPTS)' ./configure $(TCLFLAGS) --enable-shared
-	cd $(TKDIRDIR); $(MAKE) install
+	cd $(TCLDIRDIR); $(MAKE) install-binaries install-libraries
 	$(RM) -r lib/tk8.4/demos
 	$(RM) -r lib/$(TKVER)/demos
 
 tk	: FORCE
 	@echo "Installing Tk noshared..."
 	cd $(TKDIRDIR); CC='$(CC) $(OPTS) $(TCLOPTS)' ./configure $(TCLFLAGS) --disable-shared
-	cd $(TKDIRDIR); $(MAKE) install
+	cd $(TKDIRDIR); $(MAKE) install-binaries install-libraries
 	$(RM) -r lib/$(TKVER)/demos
 
 tktablesh : FORCE
 	@echo "Installing TkTable..."
 	cd $(TKTABLEDIR); CC='$(CC) $(OPTS)' ./configure $(TKTABLEFLAGS) --enable-shared
-	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install 
+	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install-binaries install-libraries
 
 tktable	: FORCE
 	@echo "Installing TkTable..."
 	cd $(TKTABLEDIR); CC='$(CC) $(OPTS)' ./configure $(TKTABLEFLAGS) --disable-shared
-	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install
+	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install-binaries install-libraries
 
 tcllib	: FORCE
 	@echo "Installing TCLLIB..."
 	cd $(TCLLIBDIR); ./configure $(TCLLIBFLAGS)
-	cd $(TCLLIBDIR); $(MAKE) install
+	cd $(TCLLIBDIR); $(MAKE) install-libraries
 
 tkindex : FORCE
 	@echo "Installing Tk mkIndex..."
@@ -403,7 +404,7 @@
 zlibsh  : FORCE
 	@echo "Installing zlib..."
 	cd $(ZLIBDIR); CC=$(CC) CFLAGS='$(OPTS)' ./configure $(ZLIBFLAGS) --shared
-	cd $(ZLIBDIR); $(MAKE) install
+	cd $(ZLIBDIR); $(MAKE) install-binaries install-libraries
 
 tkimg	: FORCE
 	@echo "Installing TKIMG noshared..."
@@ -413,7 +414,7 @@
 tkimgsh	: FORCE
 	@echo "Installing TKIMG shared..."
 	cd $(TKIMGDIR); CC='$(CC) $(OPTS)' ./configure $(TKIMGFLAGS)
-	cd $(TKIMGDIR); $(MAKE) install
+	cd $(TKIMGDIR); $(MAKE) install-binaries install-libraries
 
 tkmpeg	: FORCE
 	@echo "Installing TKMPEG..."
