--- Makefile.orig	2006-10-18 11:47:43.000000000 +0100
+++ Makefile	2006-10-18 11:48:37.000000000 +0100
@@ -286,40 +286,40 @@
 tclsh	: FORCE
 	@echo "Installing Tcl shared..."
 	cd $(TCLDIRDIR); CC='$(CC) $(OPTS)' ./configure $(TCLFLAGS) --enable-shared
-	cd $(TCLDIRDIR); $(MAKE) install
+	cd $(TCLDIRDIR); $(MAKE) install-binaries install-libraries
 
 tcl	: FORCE
 	@echo "Installing Tcl noshared..."
 	cd $(TCLDIRDIR); CC='$(CC) $(OPTS)' ./configure $(TCLFLAGS) --disable-shared
-	cd $(TCLDIRDIR); $(MAKE) install
+	cd $(TCLDIRDIR); $(MAKE) install-binaries install-libraries
 
 tksh	: FORCE
 	@echo "Installing Tk shared..."
 	cd $(TKDIRDIR); CC='$(CC) $(OPTS)' ./configure $(TCLFLAGS) --enable-shared
-	cd $(TKDIRDIR); $(MAKE) install
+	cd $(TKDIRDIR); $(MAKE) install-binaries install-libraries
 	$(RM) -r lib/tk8.4/demos
 	$(RM) -r lib/$(TKVER)/demos
 
 tk	: FORCE
 	@echo "Installing Tk noshared..."
 	cd $(TKDIRDIR); CC='$(CC) $(OPTS)' ./configure $(TCLFLAGS) --disable-shared
-	cd $(TKDIRDIR); $(MAKE) install
+	cd $(TKDIRDIR); $(MAKE) install-binaries install-libraries
 	$(RM) -r lib/$(TKVER)/demos
 
 tktablesh : FORCE
 	@echo "Installing TkTable..."
 	cd $(TKTABLEDIR); CC='$(CC) $(OPTS)' ./configure $(TKTABLEFLAGS) --enable-shared
-	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install 
+	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install-binaries install-libraries 
 
 tktable	: FORCE
 	@echo "Installing TkTable..."
 	cd $(TKTABLEDIR); CC='$(CC) $(OPTS)' ./configure $(TKTABLEFLAGS) --disable-shared
-	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install
+	cd $(TKTABLEDIR); $(MAKE); $(MAKE) install-binaries install-libraries
 
 tcllib	: FORCE
 	@echo "Installing TCLLIB..."
 	cd $(TCLLIBDIR); ./configure $(TCLLIBFLAGS)
-	cd $(TCLLIBDIR); $(MAKE) install
+	cd $(TCLLIBDIR); $(MAKE) install-libraries
 
 tkindex : FORCE
 	@echo "Installing Tk mkIndex..."
@@ -338,7 +338,7 @@
 zlibsh  : FORCE
 	@echo "Installing zlib..."
 	cd $(ZLIBDIR); CC=$(CC) CFLAGS='$(OPTS)' ./configure $(ZLIBFLAGS) --shared
-	cd $(ZLIBDIR); $(MAKE) install
+	cd $(ZLIBDIR); $(MAKE) install-binaries install-libraries
 
 tkimg	: FORCE
 	@echo "Installing TKIMG noshared..."
@@ -348,7 +348,7 @@
 tkimgsh	: FORCE
 	@echo "Installing TKIMG shared..."
 	cd $(TKIMGDIR); CC='$(CC) $(OPTS)' ./configure $(TKIMGFLAGS)
-	cd $(TKIMGDIR); $(MAKE) install
+	cd $(TKIMGDIR); $(MAKE) install-binaries install-libraries
 
 tkmpeg	: FORCE
 	@echo "Installing TKMPEG..."
@@ -371,7 +371,7 @@
 xpash	: FORCE
 	@echo "Installing XPA shared..."
 	cd $(XPADIR); CC=$(CC) CFLAGS='$(OPTS)' LDFLAGS='$(OPTS)' ./configure $(XPAFLAGS) --enable-shared
-	cd $(XPADIR); $(MAKE); $(MAKE) tclxpa; $(MAKE) install
+	cd $(XPADIR); $(MAKE); $(MAKE) tclxpa; $(MAKE) install-binaries install-libraries
 	cd bin; strip xpa*
 
 xpa	: FORCE
@@ -427,7 +427,7 @@
 	@echo "Installing ZIP..."
 	cd $(ZIPDIR); PREFIX=$(root) $(MAKE) CC=$(CC) CFLAGS='$(OPTS) -I. -DUNIX' LFLAGS1='$(OPTS)' -f unix/Makefile generic
 	cd $(ZIPDIR); PREFIX=$(root) $(MAKE) CC=$(CC) CFLAGS='$(OPTS) -I. -DUNIX' LFLAGS1='$(OPTS)' -f unix/Makefile install
-	ce $(ZIPDIR); $(RM) libzip.a
+	cd $(ZIPDIR); $(RM) libzip.a
 	cd $(ZIPDIR); $(AR) -cr libzip.a $(ZIPOBJS)
 	cp $(ZIPDIR)/libzip.a lib/libzip.a
 
