diff --git a/setup.py b/tmp/setup.py
index 39e75824..f7948f64 100755
--- a/setup.py
+++ b/setup.py
@@ -19,11 +19,11 @@ print('found these packages:', pkgs)
 schema_dir = 'nwb-schema/core'
 
 reqs = [
-    'h5py>=2.10,<4',
-    'hdmf>=3.3.2,<4',
-    'numpy>=1.16,<1.23',
-    'pandas>=1.0.5,<2',
-    'python-dateutil>=2.7,<3',
+    'h5py',
+    'hdmf',
+    'numpy',
+    'pandas',
+    'python-dateutil',
     'setuptools'
 ]
 
