--- swig/include/gdal.i.orig	2009-04-24 21:55:06.315863023 -0700
+++ swig/include/gdal.i	2009-04-24 22:10:28.806611326 -0700
@@ -176,6 +176,17 @@
 %include "gdal_typemaps.i"
 #endif
 
+/* Default memberin typemaps required to support SWIG 1.3.39 and above */
+%typemap(memberin) char *Info %{
+/* char* Info memberin typemap */
+$1;
+%}
+
+%typemap(memberin) char *Id %{
+/* char* Info memberin typemap */
+$1;
+%}
+
 //************************************************************************
 //
 // Define the exposed CPL functions.
