Index: Tensile-rocm-4.0.0/setup.py
===================================================================
--- Tensile-rocm-4.0.0.orig/setup.py
+++ Tensile-rocm-4.0.0/setup.py
@@ -30,8 +30,6 @@ setup(
   install_requires=readRequirementsFromTxt(),
   python_requires='>=3.5',
   packages=["Tensile"],
-  package_data={ "Tensile": ["Tensile/cmake/*"] },
-  data_files=[ ("cmake", ["Tensile/cmake/TensileConfig.cmake", "Tensile/cmake/TensileConfigVersion.cmake"]) ],
   include_package_data=True,
   entry_points={"console_scripts": [
     # user runs a benchmark
