Index: Tensile-rocm-4.0.0/Tensile/cmake/TensileConfig.cmake
===================================================================
--- Tensile-rocm-4.0.0.orig/Tensile/cmake/TensileConfig.cmake
+++ Tensile-rocm-4.0.0/Tensile/cmake/TensileConfig.cmake
@@ -26,7 +26,7 @@ if(NOT DEFINED Tensile_ROOT)
 get_filename_component(Tensile_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
 get_filename_component(Tensile_PREFIX "${Tensile_PREFIX}" PATH)
 
-execute_process(COMMAND "${Tensile_PREFIX}/bin/TensileGetPath" OUTPUT_VARIABLE Tensile_ROOT)
+execute_process(COMMAND "TensileGetPath" OUTPUT_VARIABLE Tensile_ROOT)
 endif()
 list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Source/cmake/")
 list(APPEND CMAKE_MODULE_PATH "${Tensile_ROOT}/Source/")
@@ -93,7 +93,7 @@ function(TensileCreateLibraryFiles
     set(Tensile_MERGE_FILES OFF)
   endif()
 
-  set(Script "${Tensile_ROOT}/bin/TensileCreateLibrary")
+  set(Script "TensileCreateLibrary")
   message(STATUS "Tensile script: ${Script}")
 
   set(Options "--new-client-only" "--no-legacy-components")
Index: Tensile-rocm-4.0.0/Tensile/Source/TensileCreateLibrary.cmake
===================================================================
--- Tensile-rocm-4.0.0.orig/Tensile/Source/TensileCreateLibrary.cmake
+++ Tensile-rocm-4.0.0/Tensile/Source/TensileCreateLibrary.cmake
@@ -52,10 +52,7 @@ function(TensileCreateLibraryCmake
   message(STATUS "Tensile_ARCHITECTURE        from TensileCreateLibraryCmake : ${Tensile_ARCHITECTURE}")
   message(STATUS "Tensile_LIBRARY_FORMAT      from TensileCreateLibraryCmake : ${Tensile_LIBRARY_FORMAT}")
 
-  execute_process(COMMAND chmod 755 ${Tensile_ROOT}/bin/TensileCreateLibrary)
-  execute_process(COMMAND chmod 755 ${Tensile_ROOT}/bin/Tensile)
-
-  set(Tensile_CREATE_COMMAND "${Tensile_ROOT}/bin/TensileCreateLibrary")
+  set(Tensile_CREATE_COMMAND "TensileCreateLibrary")
 
   set(Tensile_SOURCE_PATH "${PROJECT_BINARY_DIR}/Tensile")
   message(STATUS "Tensile_SOURCE_PATH=${Tensile_SOURCE_PATH}")
