diff -Naur OpenMM2.0-Source.orig/src/plugins/freeEnergy/platforms/cuda/tests/TestCudaGBVISoftcoreForce.cpp OpenMM2.0-Source/src/plugins/freeEnergy/platforms/cuda/tests/TestCudaGBVISoftcoreForce.cpp
--- OpenMM2.0-Source.orig/src/plugins/freeEnergy/platforms/cuda/tests/TestCudaGBVISoftcoreForce.cpp	2010-05-14 02:25:51.000000000 +0200
+++ OpenMM2.0-Source/src/plugins/freeEnergy/platforms/cuda/tests/TestCudaGBVISoftcoreForce.cpp	2010-10-02 18:37:10.000000000 +0200
@@ -370,7 +370,9 @@
        
 //           positions[8][2] -=  static_cast<double>(ii+1)*0.1;
 //           positions[8][2] -=  0.001;
+#if PRINT_ON == 1
            (void) fprintf( stderr, "r48=%14.6e r28=%14.6e r24=%14.6e\n", positions[8][2]-positions[4][2], positions[8][2], positions[4][2] );
+#endif
        }
 #if 0
        int carbonIndex    = 1;
