--- a/setup.py	2022-01-31 02:48:24.645837186 -0500
+++ b/setup.py	2022-01-31 02:49:55.066949247 -0500
@@ -12,12 +12,12 @@
 schema_dir = 'common/hdmf-common-schema/common'
 
 reqs = [
-    'h5py>=2.10,<4',
-    'numpy>=1.16,<1.22',
-    'scipy>=1.1,<2',
-    'pandas>=1.0.5,<2',
-    'ruamel.yaml>=0.16,<1',
-    'jsonschema>=2.6.0,<4',
+    'h5py',
+    'numpy',
+    'scipy',
+    'pandas',
+    'ruamel.yaml',
+    'jsonschema',
     'setuptools',
 ]
 
