--- a/setup.py	2022-01-31 02:48:24.645837186 -0500
+++ b/setup.py	2022-01-31 02:49:55.066949247 -0500
@@ -18,12 +18,12 @@
 schema_dir = 'common/hdmf-common-schema/common'

 reqs = [
-    'h5py>=2.10,<4',
-    'jsonschema>=2.6.0,<5',
-    'numpy>=1.16,<1.23',
-    'pandas>=1.0.5,<2',
-    'ruamel.yaml>=0.16,<1',
-    'scipy>=1.1,<2',
+    'h5py',
+    'jsonschema',
+    'numpy',
+    'pandas',
+    'ruamel.yaml',
+    'scipy',
     'setuptools',
 ]
