diff -Naurp freeon-beta-2012-05-03.orig/Makefile.am freeon-beta-2012-05-03/Makefile.am
--- freeon-beta-2012-05-03.orig/Makefile.am	2012-05-04 03:53:04.000000000 +0000
+++ freeon-beta-2012-05-03/Makefile.am	2013-05-29 14:42:07.791473885 +0000
@@ -9,14 +9,9 @@
 INTERNAL_LAPACK_DIRS = lapack
 endif
 
-if INTERNAL_HDF5
-INTERNAL_HDF5_DIRS = hdf5-1.8.3
-HDF5_INCLUDES = -I$(srcdir)/hdf5-1.8.3/src -I$(builddir)/hdf5-1.8.3/src
-endif
-
-SUBDIRS = $(INTERNAL_LAPACK_DIRS) $(INTERNAL_HDF5_DIRS) Modules FreeON SCFeqs OneE QCTC HiCu TwoE ONX PostProcessing BasisSets tests Validate
+SUBDIRS = $(INTERNAL_LAPACK_DIRS) Modules FreeON SCFeqs OneE QCTC HiCu TwoE ONX PostProcessing BasisSets tests Validate
 
-DIST_SUBDIRS = $(SUBDIRS) lapack hdf5-1.8.3
+DIST_SUBDIRS = $(SUBDIRS) lapack
 
 EXTRA_DIST = fix_localversion.sh localversion FreeON.bibtex $(top_builddir)/latex/refman.pdf
 
--- freeon-beta-2012-05-03.orig/configure.ac	2012-05-04 04:38:47.000000000 +0000
+++ freeon-beta-2012-05-03/configure.ac	2013-05-29 15:04:02.886109082 +0000
@@ -855,7 +855,6 @@ AC_MSG_RESULT([${use_internal_hdf5}])
 
 AM_CONDITIONAL(INTERNAL_HDF5, [test "${use_internal_hdf5}" = "yes"])
 
-AC_CONFIG_SUBDIRS([hdf5-1.8.3])
 
 if test "${use_internal_hdf5}" = "yes"; then
 
