diff -Naurp freeon-9999.orig/Modules/Makefile.am freeon-9999/Modules/Makefile.am
--- freeon-9999.orig/Modules/Makefile.am	2013-05-28 08:09:39.000000000 +0000
+++ freeon-9999/Modules/Makefile.am	2013-05-29 14:32:24.879721200 +0000
@@ -4,11 +4,6 @@
 INTERNAL_LAPACK_DIRS = lapack
 endif
 
-if INTERNAL_HDF5
-INTERNAL_HDF5_DIRS = hdf5-1.8.3
-HDF5_INCLUDES = -I$(srcdir)/hdf5-1.8.3/src -I$(builddir)/hdf5-1.8.3/src
-endif
-
 DIST_SUBDIRS = lapack hdf5-1.8.3
 SUBDIRS      = $(INTERNAL_HDF5_DIRS) $(INTERNAL_LAPACK_DIRS)
 
--- freeon-9999.orig/configure.ac	2013-05-29 15:34:16.000000000 +0000
+++ freeon-9999/configure.ac	2013-05-29 15:36:22.000063299 +0000
@@ -870,7 +870,6 @@ AM_CONDITIONAL(INTERNAL_HDF5, [test "${u
 if test "${use_internal_hdf5}" = "yes"; then
 
   AC_DEFINE(HAVE_INTERNAL_HDF5, 1, [Use the inernal hdf5 library.])
-  AC_CONFIG_SUBDIRS([Modules/hdf5-1.8.3])
 
 else
 
