diff --git a/CMakeLists.txt b/CMakeLists.txt
index dcc24af..24f444e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -246,7 +246,8 @@ if (USE_GUI)
 
     add_definitions(-DTCL -DOPENGL -DUSE_TOGL_2)
     include_directories(${TCL_INCLUDE_PATH})
-    include_directories(${TK_INCLUDE_PATH})
+    include_directories(${TK_INCLUDE_PATH}/generic)
+    include_directories(${TK_INCLUDE_PATH}/unix)
     set(LIBTOGL togl)
 
     if(WIN32)
@@ -268,7 +269,7 @@ else()
 endif()
 
 if (USE_PYTHON)
-    add_subdirectory(external_dependencies/pybind11)
+    find_package(pybind11)
     find_path(PYBIND_INCLUDE_DIR pybind11/pybind11.h HINTS ${PYTHON_INCLUDE_DIR})
     if( PYBIND_INCLUDE_DIR )
         message(STATUS "Found Pybind11: ${PYBIND_INCLUDE_DIR}")
diff --git a/cmake/SuperBuild.cmake b/cmake/SuperBuild.cmake
index c24b631..2354a09 100644
--- a/cmake/SuperBuild.cmake
+++ b/cmake/SuperBuild.cmake
@@ -39,7 +39,6 @@ endif(NOT WIN32)
 #######################################################################
 if (USE_PYTHON)
   find_path(PYBIND_INCLUDE_DIR pybind11/pybind11.h PATHS ${CMAKE_CURRENT_SOURCE_DIR}/external_dependencies/pybind11/include NO_DEFAULT_PATH)
-    set(NG_INSTALL_PYBIND ON)
     if( NOT PYBIND_INCLUDE_DIR )
       # if the pybind submodule is missing, try to initialize and update all submodules
       execute_process(COMMAND git submodule update --init --recursive WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
