--- configure.ac 	1970-01-01 01:00:00.000000000 +0100
+++ configure.ac	2007-09-21 16:03:04.073238283 +0100
@@ -0,0 +1,13 @@
+#                                               -*- Autoconf -*-
+AC_PREREQ(2.59)
+AC_INIT(ccolamd, 2.7.0, davis@cise.ufl.edu)
+AM_INIT_AUTOMAKE([foreign])
+AC_PROG_INSTALL
+AC_PROG_LIBTOOL
+AC_CHECK_LIB(m, sqrt)
+AC_CHECK_HEADERS(UFconfig.h)
+AC_CONFIG_FILES([
+    Source/Makefile
+    Demo/Makefile
+    Makefile])
+AC_OUTPUT
