--- config/PETSc/packages/imagemagick.py.orig	2012-01-22 17:25:36.440142627 +0000
+++ config/PETSc/packages/imagemagick.py	2012-01-22 17:26:26.091417306 +0000
@@ -4,8 +4,8 @@
   def __init__(self, framework):
     PETSc.package.NewPackage.__init__(self, framework)
     self.functions     = ['IsMagickInstantiated']
-    self.includes      = ['MagicCore/MagickCore.h']
-    self.liblist       = [['libMagicCore.a']]
+    self.includes      = ['MagickCore/MagickCore.h']
+    self.liblist       = [['libMagickCore.a']]
 
 
 
--- src/sys/draw/impls/x/xinit.c.orig	2012-01-22 17:25:46.230393969 +0000
+++ src/sys/draw/impls/x/xinit.c	2012-01-22 17:26:26.091417306 +0000
@@ -328,7 +328,7 @@
 void deflateEnd(void) {;}
 
 #elif defined(PETSC_HAVE_IMAGEMAGICK)
-#include <MagicCore/MagickCore.h>
+#include <MagickCore/MagickCore.h>
 #undef __FUNCT__  
 #define __FUNCT__ "PetscDrawSave_X" 
 PetscErrorCode PetscDrawSave_X(PetscDraw draw,PetscViewer viewer)
