--- config/BuildSystem/config/setCompilers.py	2010-11-24 18:01:17.110124425 +0100
+++ config/BuildSystem/config/setCompilers.py	2010-11-24 18:02:28.710104226 +0100
@@ -1186,12 +1186,7 @@
     for language in languages:
       flag = '-L'
       self.pushLanguage(language)
-      # test '-R' before '-rpath' as sun compilers [c,fortran] don't give proper errors with wrong options.
-      if not Configure.isDarwin():      
-        testFlags = ['-Wl,-rpath,', '-R','-rpath ' , '-Wl,-R,']
-      else:
-        testFlags = []
-      # test '-R' before '-Wl,-rpath' for SUN compilers [as cc on linux accepts -Wl,-rpath, but  f90 & CC do not.
+      testFlags = []
       if self.isSun(self.framework.getCompiler()):
         testFlags.insert(0,'-R')
       for testFlag in testFlags:
