--- config/BuildSystem/config/setCompilers.py.old	2010-09-07 21:41:32.693103846 +0200
+++ config/BuildSystem/config/setCompilers.py	2010-09-07 21:41:45.322093881 +0200
@@ -875,7 +875,7 @@
       languages.append('FC')
     for language in languages:
       self.pushLanguage(language)
-      for testFlag in ['-PIC', '-fPIC', '-KPIC','-qpic']:
+      for testFlag in ['-fPIC', '-PIC', '-KPIC', '-qpic']:
         try:
           self.framework.logPrint('Trying '+language+' compiler flag '+testFlag)
           if not self.checkLinkerFlag(testFlag):
