--- config/BuildSystem/config/setCompilers.py	2013-10-08 19:45:00.581513389 +0200
+++ config/BuildSystem/config/setCompilers.py	2013-05-13 23:35:20.000000000 +0200
@@ -1303,12 +1303,7 @@
     for language in languages:
       flag = '-L'
       self.pushLanguage(language)
-      # test '-R' before '-rpath' as sun compilers [c,fortran] don't give proper errors with wrong options.
-      if not Configure.isDarwin():
-        testFlags = ['-Wl,-rpath,', '-R','-rpath ' , '-Wl,-R,']
-      else:
-        testFlags = []
-      # test '-R' before '-Wl,-rpath' for SUN compilers [as cc on linux accepts -Wl,-rpath, but  f90 & CC do not.
+      testFlags = []
       if self.isSun(self.framework.getCompiler()):
         testFlags.insert(0,'-R')
       for testFlag in testFlags:
