diff --git a/src/kernel/Makefile.am b/src/kernel/Makefile.am
index 4847fa4..4487233 100644
--- a/src/kernel/Makefile.am
+++ b/src/kernel/Makefile.am
@@ -8,7 +8,7 @@ lib_LTLIBRARIES  = libgmxpreprocess@LIBSUFFIX@.la
 
 LDADD = $(lib_LTLIBRARIES) ../mdlib/libmd@LIBSUFFIX@.la ../gmxlib/libgmx@LIBSUFFIX@.la 
 
-libgmxpreprocess@LIBSUFFIX@_la_LIBADD       =
+libgmxpreprocess@LIBSUFFIX@_la_LIBADD       = ../mdlib/libmd@LIBSUFFIX@.la
 libgmxpreprocess@LIBSUFFIX@_la_DEPENDENCIES =
 
 libgmxpreprocess@LIBSUFFIX@_la_SOURCES = \
