diff -Naur gromacs-4.5.7.orig/cmake/CMakeASM-ATTInformation.cmake gromacs-4.5.7/cmake/CMakeASM-ATTInformation.cmake
--- gromacs-4.5.7.orig/cmake/CMakeASM-ATTInformation.cmake	1969-12-31 17:00:00.000000000 -0700
+++ gromacs-4.5.7/cmake/CMakeASM-ATTInformation.cmake	2013-05-03 10:47:52.779868731 -0600
@@ -0,0 +1,13 @@
+
+# This a modified version of CMakeASMInformation.cmake coming with cmake 2.8.6, which
+# supports .S files
+
+# support for AT&T syntax assemblers, e.g. GNU as
+
+SET(ASM_DIALECT "-ATT")
+SET(CMAKE_ASM${ASM_DIALECT}_SOURCE_FILE_EXTENSIONS S;s;asm)
+
+SET(CMAKE_ASM${ASM_DIALECT}_COMPILE_OBJECT "<CMAKE_ASM${ASM_DIALECT}_COMPILER> <FLAGS> -o <OBJECT> <SOURCE>")
+
+INCLUDE(CMakeASMInformation)
+SET(ASM_DIALECT)
diff -Naur gromacs-4.5.7.orig/src/gmxlib/CMakeLists.txt gromacs-4.5.7/src/gmxlib/CMakeLists.txt
--- gromacs-4.5.7.orig/src/gmxlib/CMakeLists.txt	2013-05-03 10:46:29.029305075 -0600
+++ gromacs-4.5.7/src/gmxlib/CMakeLists.txt	2013-05-03 10:50:34.630582295 -0600
@@ -47,18 +47,18 @@
 if(GMX_IA32_ASM)
   file(GLOB GMX_SSEKERNEL_C_SRC   nonbonded/nb_kernel_ia32_${SSETYPE}/*.c)
   if(GMX_ASM_USEASM_NASM)
-    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_ia32_${SSETYPE}/*intel_syntax*.s)    
+    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_ia32_${SSETYPE}/*intel_syntax*.S)    
   else()
-    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_ia32_${SSETYPE}/*${SSETYPE}.s nonbonded/nb_kernel_ia32_${SSETYPE}/*asm.s)
+    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_ia32_${SSETYPE}/*${SSETYPE}.S nonbonded/nb_kernel_ia32_${SSETYPE}/*asm.S)
   endif()
 endif(GMX_IA32_ASM)
 
 if(GMX_X86_64_ASM)
   file(GLOB GMX_SSEKERNEL_C_SRC   nonbonded/nb_kernel_x86_64_${SSETYPE}/*.c)
   if(GMX_ASM_USEASM_NASM)
-    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_x86_64_${SSETYPE}/*intel_syntax*.s)
+    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_x86_64_${SSETYPE}/*intel_syntax*.S)
   else()
-    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_x86_64_${SSETYPE}/*${SSETYPE}.s nonbonded/nb_kernel_x86_64_${SSETYPE}/*asm.s)
+    file(GLOB GMX_SSEKERNEL_ASM_SRC nonbonded/nb_kernel_x86_64_${SSETYPE}/*${SSETYPE}.S nonbonded/nb_kernel_x86_64_${SSETYPE}/*asm.S)
   endif()
 endif(GMX_X86_64_ASM)
 
