diff -Naur gromacs-4.0.5.orig/share/html/Makefile.am gromacs-4.0.5/share/html/Makefile.am
--- gromacs-4.0.5.orig/share/html/Makefile.am	2006-03-06 10:52:20.000000000 +0100
+++ gromacs-4.0.5/share/html/Makefile.am	2009-11-29 22:27:24.000000000 +0100
@@ -4,7 +4,7 @@
 #
 SUBDIRS = .  images  online
 
-htmldir = ${pkgdatadir}/html
+htmldir = @htmldir@/html
 
 html_DATA = online.html gmxfaq.html
 
diff -Naur gromacs-4.0.5.orig/share/html/Makefile.in gromacs-4.0.5/share/html/Makefile.in
--- gromacs-4.0.5.orig/share/html/Makefile.in	2009-05-09 11:12:24.000000000 +0200
+++ gromacs-4.0.5/share/html/Makefile.in	2009-11-29 22:28:16.000000000 +0100
@@ -172,7 +172,7 @@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
-htmldir = ${pkgdatadir}/html
+htmldir = @htmldir@/html
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
diff -Naur gromacs-4.0.5.orig/share/html/images/Makefile.am gromacs-4.0.5/share/html/images/Makefile.am
--- gromacs-4.0.5.orig/share/html/images/Makefile.am	2006-03-06 10:52:20.000000000 +0100
+++ gromacs-4.0.5/share/html/images/Makefile.am	2009-11-29 23:00:57.000000000 +0100
@@ -2,7 +2,7 @@
 #
 # Don't edit - this file is generated automatically from Makefile.am
 #
-imagedir = ${pkgdatadir}/html/images
+imagedir = @htmldir@/html/images
 
 image_DATA = \
 	features.gif					\
diff -Naur gromacs-4.0.5.orig/share/html/images/Makefile.in gromacs-4.0.5/share/html/images/Makefile.in
--- gromacs-4.0.5.orig/share/html/images/Makefile.in	2009-05-09 11:12:24.000000000 +0200
+++ gromacs-4.0.5/share/html/images/Makefile.in	2009-11-29 23:01:20.000000000 +0100
@@ -186,7 +186,7 @@
 #
 # Don't edit - this file is generated automatically from Makefile.am
 #
-imagedir = ${pkgdatadir}/html/images
+imagedir = @htmldir@/html/images
 image_DATA = \
 	features.gif					\
 	flow_leftrightup.gif	flow_vrule.gif 		\
diff -Naur gromacs-4.0.5.orig/share/html/online/Makefile.am gromacs-4.0.5/share/html/online/Makefile.am
--- gromacs-4.0.5.orig/share/html/online/Makefile.am	2008-08-11 11:46:17.000000000 +0200
+++ gromacs-4.0.5/share/html/online/Makefile.am	2009-11-29 23:01:47.000000000 +0100
@@ -2,7 +2,7 @@
 #
 # Don't edit - this file is generated automatically from Makefile.am
 #
-onlinedir = ${pkgdatadir}/html/online
+onlinedir = @htmldir@/html/online
 
 online_DATA = \
   cpt.html	\
diff -Naur gromacs-4.0.5.orig/share/html/online/Makefile.in gromacs-4.0.5/share/html/online/Makefile.in
--- gromacs-4.0.5.orig/share/html/online/Makefile.in	2009-05-09 11:12:24.000000000 +0200
+++ gromacs-4.0.5/share/html/online/Makefile.in	2009-11-29 23:02:15.000000000 +0100
@@ -186,7 +186,7 @@
 #
 # Don't edit - this file is generated automatically from Makefile.am
 #
-onlinedir = ${pkgdatadir}/html/online
+onlinedir = @htmldir@/html/online
 online_DATA = \
   cpt.html	\
   dat.html           g_disre.html      g_sorient.html        ndx.html        \
