cc is not cache by ccache (bug #302048), work around that

diff --git a/configure.ac b/configure.ac
index 3100101..db9d0b6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -288,11 +288,11 @@ AC_DEFINE_UNQUOTED([BUILD_MACHINE],["`uname -srm`"],[Hardware and OS version for
 # normally we use the default compiler, but on alpha/linux we try the compaq ones first
 case "${host_cpu}-${host_os}" in
   alpha*-linux*)
-    cc_names="ccc cc gcc"
+    cc_names="ccc gcc"
     f77_names="fort f77 g77"
     ;;
   *) 
-    cc_names="cc icc xlc gcc"
+    cc_names="icc xlc gcc"
     f77_names="f77 ifort xlf xlf77 cf77 fl32 g77 fort77 f90 xlf90 pgf77 cf77 fort fort77 pgf90"
     ;;
 esac
