diff --git a/Makefile.am b/Makefile.am
index 52a70cb..f94d4db 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -35,11 +35,13 @@ install-exec-hook:
 mdrun:		
 	(cd $(top_builddir)/src/gmxlib && $(MAKE) ; exit 0)
 	(cd $(top_builddir)/src/mdlib && $(MAKE) ; exit 0)
+	(cd $(top_builddir)/src/kernel && $(MAKE) lib_LTLIBRARIES ; exit 0)
 	(cd $(top_builddir)/src/kernel && $(MAKE) mdrun ; exit 0)
 
 install-mdrun:
 	(cd $(top_builddir)/src/gmxlib && $(MAKE) install ; exit 0)
 	(cd $(top_builddir)/src/mdlib && $(MAKE) install ; exit 0)
+	(cd $(top_builddir)/src/kernel && $(MAKE) install-libLTLIBRARIES ; exit 0)
 	(cd $(top_builddir)/src/kernel && $(MAKE) install-mdrun ; exit 0)
 
 fahcore:
diff --git a/src/Makefile.am b/src/Makefile.am
index b538952..08f4597 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,11 +10,13 @@ EXTRA_DIST = config.win
 mdrun:		
 	(cd $(top_builddir)/src/gmxlib && $(MAKE))
 	(cd $(top_builddir)/src/mdlib && $(MAKE))
+	(cd $(top_builddir)/src/kernel && $(MAKE) lib_LTLIBRARIES)
 	(cd $(top_builddir)/src/kernel && $(MAKE) mdrun)
 
 install-mdrun:
 	(cd $(top_builddir)/src/gmxlib && $(MAKE) install ; exit 0)
 	(cd $(top_builddir)/src/mdlib && $(MAKE) install ; exit 0)
+	(cd $(top_builddir)/src/kernel && $(MAKE) install-libLTLIBRARIES ; exit 0)
 	(cd $(top_builddir)/src/kernel && $(MAKE) install-mdrun)
 
 CLEANFILES = *~ \\\#*
