diff -Naur gromacs-4.0.4-orig/configure.ac gromacs-4.0.4/configure.ac
--- gromacs-4.0.4-orig/configure.ac	2009-04-09 21:12:31.000000000 +0200
+++ gromacs-4.0.4/configure.ac	2009-04-09 21:13:38.000000000 +0200
@@ -320,11 +320,11 @@
 case "${host_cpu}-${host_os}" in
   alpha*-linux*)
     cc_names="ccc cc gcc"
-    f77_names="fort f77 g77"
+    f77_names="fort f77 g77 gfortran"
     ;;
   *) 
     cc_names="cc icc xlc gcc"
-    f77_names="f77 ifort xlf xlf77 cf77 fl32 g77 fort77 f90 xlf90 pgf77 cf77 fort fort77 pgf90"
+    f77_names="f77 ifort xlf xlf77 cf77 fl32 g77 fort77 f90 xlf90 pgf77 cf77 fort fort77 pgf90 gfortran"
     ;;
 esac
 # the (non-mpi) c compiler, which is also used for programs executed during build stage
