--- xdsi	2009-06-25 14:05:48.000000000 +0200
+++ xdsi.new	2009-06-29 13:54:52.000000000 +0200
@@ -21,8 +21,8 @@
 # These are the only two things you may change#
 
 
-set Templates "/mydir/templates"
-set initial "/mydir/SLS-2009"
+set Templates "/usr/share/xdsi/templates"
+set initial "/"
 
 
 
@@ -122,7 +122,7 @@
 }
 
 
-CheckExecutables "kpdf gnuplot xds_par xdsstat xdsviewer convert pointless"
+CheckExecutables "xpdf gnuplot xds_par xdsstat xds-viewer convert pointless"
 
 wm resizable . 0 0
 
--- xdsi	2009-06-25 14:05:48.000000000 +0200
+++ xdsi.new	2009-06-29 14:22:57.000000000 +0200
@@ -3063,7 +3063,7 @@
 event add <<Change>> <B1-ButtonRelease> <FocusOut> <Leave> <Expose>
 #bind .plots.dummy.plot  <<Change>> {puts $Plot}
 if {$path != "/var/tmp" } {
-    button .view.button -text Open  -command {exec xdsviewer $View &}
+    button .view.button -text Open  -command {exec xds-viewer $View &}
 } else {
     button .view.button -text "Sorry, no pics" -command { }
 }
--- xdsi	2009-07-06 21:46:47.000000000 +0200
+++ xdsi.new	2009-07-06 21:47:12.000000000 +0200
@@ -1314,7 +1314,7 @@
     if {$Correct == "1" } {
 	PLOTALL $WorkDir $HOME $XdsManual }
     #exec convert scalefactors.png rfactors.png compl.png isigma.png wilson.png uca.png ucb.png ucc.png  plots_manual.pdf
-    #exec kpdf plots_manual.pdf &
+    #exec xpdf plots_manual.pdf &
 }
 
 
@@ -1371,7 +1371,7 @@
     destroy .compare
     cd $outputdir
     exec convert rfactors.png compl.png isigma.png comp.pdf
-    exec kpdf comp.pdf &
+    exec xpdf comp.pdf &
     
 }
 
@@ -1755,7 +1755,7 @@
 	exec convert $outputdir/compall.png $outputdir/compalliosig.png $outputdir/compl.png $outputdir/compall.pdf
 	}
 	exec rm -f $outputdir/compall.png $outputdir/compalliosig.png $outputdir/compl.png
-	exec kpdf $outputdir/compall.pdf &
+	exec xpdf $outputdir/compall.pdf &
 	#exec display $outputdir/compall.png &
 	#exec display $outputdir/compalliosig.png &
     } else { STATUS2 "Sorry, no usable CORRECT.LPs" }
@@ -1935,7 +1935,7 @@
 	    exec convert  scalefactors.png rfactors.png compl.png isigma.png  uca.png ucb.png ucc.png nstrong.png sigmab.png sigmar.png stat1.png stat2.png stat3.png aliens.png  wilson.png  rfacxdsstat.png iosigxdsstat.png peak.png corr.png unique.png rd.png sigano.png plots.pdf 
 	}
 	if { $XdsManual == "1" } {
-	    exec kpdf $outputdir/plots.pdf & 
+	    exec xpdf $outputdir/plots.pdf & 
 	} 
 	
     }
@@ -2004,7 +2004,7 @@
     exec echo "plot \"$outputdir/.ucc\" w lp title \"C\" " >> $outputdir/.plot
     exec gnuplot < $outputdir/.plot >& /dev/null
     #exec convert uca.png ucb.png ucc.png ucs.pdf
-    #exec kpdf ucs.pdf
+    #exec xpdf ucs.pdf
 }
 
 
@@ -3019,7 +3019,7 @@
 eval [concat $b [lsort $path]]
 event add <<Change>> <B1-ButtonRelease> <FocusOut> <Leave> <Expose>
 #bind .plots.dummy.plot  <<Change>> {puts $Plot}
-button .plots.button -text Open -command {exec kpdf $Plot &}
+button .plots.button -text Open -command {exec xpdf $Plot &}
 pack .plots.button -side bottom
 #tk_optionMenu .plots.dummy.plot  Plot  $path
 pack .plots.plot -anchor w
@@ -3156,7 +3156,7 @@
 set b "tk_optionMenu .help.help Help"
 eval [concat $b $path]
 event add <<Change>> <B1-ButtonRelease> <FocusOut> <Leave> <Expose>
-button .help.button -text Open  -command {exec kpdf $Help &}
+button .help.button -text Open  -command {exec xpdf $Help &}
 pack .help.button -side bottom
 pack .help.help -anchor w
 
