diff -Naur fsl.orig/config/buildSettings.mk fsl/config/buildSettings.mk
--- fsl.orig/config/buildSettings.mk	2020-07-23 15:43:42.277064365 -0400
+++ fsl/config/buildSettings.mk	2020-07-23 15:45:17.705069496 -0400
@@ -141,7 +141,7 @@
 # CUDA development environment
 CUDAVER := $(or $(CUDAVER),9.1)
 #$(info $$CUDAVER is [${CUDAVER}])
-CUDA_INSTALLATION = /opt/cuda-${CUDAVER}
+CUDA_INSTALLATION = /opt/cuda
 GENCODE_FLAGS = $(shell ${FSLDIR}/config/common/supportedGencodes.sh ${CUDA_INSTALLATION})
 LIB_CUDA = ${CUDA_INSTALLATION}/lib64
 INC_CUDA = ${CUDA_INSTALLATION}/include
