diff -Naur fsl.orig/src/flameo/Makefile fsl/src/flameo/Makefile
--- fsl.orig/src/flameo/Makefile	2021-01-06 14:04:35.728274475 -0500
+++ fsl/src/flameo/Makefile	2021-01-06 14:08:18.204269285 -0500
@@ -4,6 +4,7 @@
 
 USRINCFLAGS = -I${INC_NEWMAT} -I${INC_PROB} -I${INC_ZLIB} -DCIFTILIB_USE_XMLPP -I${FSLEXTINC} -I${INC_XML2} -I${INC_XML++} -I${INC_XML++CONF} -I${INC_BOOST} -I${FSLDIR}/include/ciftiio
 USRLDFLAGS = -L${LIB_NEWMAT} -L${LIB_PROB} -L${LIB_ZLIB}
+USRCXXFLAGS = -std=c++11
 
 UNAME := $(shell uname)
 ifeq (${UNAME},Darwin)
