diff -Naur fsl.orig/src/eddy/Makefile fsl/src/eddy/Makefile
--- fsl.orig/src/eddy/Makefile	2020-07-30 14:30:06.414884410 -0400
+++ fsl/src/eddy/Makefile	2020-07-30 14:33:03.400889914 -0400
@@ -29,6 +29,8 @@
 TMPCXXFLAGS_2=
 TMPNAME_1=
 TMPNAME_2=
+cuda=1
+cpu=1
 
 ifndef cuda
    ifndef cpu #Default single cpu, no threading
@@ -74,7 +75,7 @@
 ifdef NVCC11
 	NVCC=${NVCC11}
 endif
-NVCCFLAGS=-c -DCOMPILE_GPU -O3 -m 64 -std=c++11
+NVCCFLAGS=-c -DCOMPILE_GPU -O3 -m 64 -std=c++11 @@GENTOO_NVCC_FLAGS@@
 ifeq ($(CLOBBER_CLANG),1)
    NVCCFLAGS+= -DCLOBBER_CLANG
 endif
