--- cctbx_sources/libtbx/env_config.py.orig	2008-09-29 13:09:25.533038638 +0200
+++ cctbx_sources/libtbx/env_config.py	2008-09-29 13:08:51.989034593 +0200
@@ -893,7 +893,7 @@
     else:
       action = self.write_bin_sh_dispatcher
       ext = ""
-      try: os.chmod(source_file, 0755)
-      except OSError: pass
+      # try: os.chmod(source_file, 0755)
+      # except OSError: pass
     target_file_ext = target_file + ext
     remove_or_rename(target_file_ext)
