diff --git a/libtbx/command_line/scons.py b/libtbx/command_line/scons.py
index dee278e..71837ad 100755
--- a/libtbx/command_line/scons.py
+++ b/libtbx/command_line/scons.py
@@ -23,7 +23,7 @@ def dummy_fetch_win32_parallel_msg():
   pass
 
 def run():
-  engine_path = find_scons_engine_path()
+  engine_path = "/usr/lib/scons-1.2.0"
   if (engine_path is not None):
     sys.path.insert(0, engine_path)
     try: import SCons
