diff -aur opencv-1.0.0-orig/otherlibs/highgui/Makefile.am opencv-1.0.0/otherlibs/highgui/Makefile.am
--- opencv-1.0.0-orig/otherlibs/highgui/Makefile.am	2006-10-31 10:49:51.000000000 -0500
+++ opencv-1.0.0/otherlibs/highgui/Makefile.am	2008-06-07 10:15:18.000000000 -0400
@@ -4,8 +4,8 @@
     cvcap_v4l.cpp cvcap_xine.cpp                                 \
     window_carbon.cpp window_gtk.cpp window_w32.cpp
 
-INCLUDES = -I. -I$(top_srcdir)/cxcore/include \
-    -I$(top_srcdir)/cv/include @GTHREAD_CFLAGS@ @GTK_CFLAGS@ \
+INCLUDES = -I. -I$(top_srcdir)/cxcore/include -I$(top_srcdir) \
+    -I$(top_srcdir)/cv/include -I/usr/include/OpenEXR @GTHREAD_CFLAGS@ @GTK_CFLAGS@ \
     @QUICKTIME_CFLAGS@ @CARBON_CFLAGS@
 
 noinst_HEADERS     = \
diff -aur opencv-1.0.0-orig/otherlibs/highgui/grfmt_exr.cpp opencv-1.0.0/otherlibs/highgui/grfmt_exr.cpp
--- opencv-1.0.0-orig/otherlibs/highgui/grfmt_exr.cpp	2006-10-17 07:58:55.000000000 -0400
+++ opencv-1.0.0/otherlibs/highgui/grfmt_exr.cpp	2008-06-07 10:12:42.000000000 -0400
@@ -40,6 +40,7 @@
 //M*/
 
 #include "_highgui.h"
+#include "cvconfig.h"
 
 #ifdef HAVE_ILMIMF
 
diff -aur opencv-1.0.0-orig/otherlibs/highgui/grfmt_png.cpp opencv-1.0.0/otherlibs/highgui/grfmt_png.cpp
--- opencv-1.0.0-orig/otherlibs/highgui/grfmt_png.cpp	2006-10-11 12:51:43.000000000 -0400
+++ opencv-1.0.0/otherlibs/highgui/grfmt_png.cpp	2008-06-07 09:56:54.000000000 -0400
@@ -40,6 +40,7 @@
 //M*/
 
 #include "_highgui.h"
+#include "cvconfig.h"
 
 #ifdef HAVE_PNG
 
@@ -50,10 +51,10 @@
     and png2bmp sample from libpng distribution (Copyright (C) 1999-2001 MIYASAKA Masaru)
 \****************************************************************************************/
 
-#if defined WIN32 || defined HAVE_PNG_H
+#ifdef HAVE_PNG_H
 #include <png.h>
 #else
-#include <libpng/png.h>
+#include <libpng12/png.h>
 #endif
 #include "grfmt_png.h"
 
