--- ../configure.in	2006-11-26 19:03:36.000000000 +0100
+++ configure.in	2006-11-26 19:13:23.000000000 +0100
@@ -81,13 +81,14 @@
 AC_ARG_ENABLE([debug],
               [AS_HELP_STRING([--enable-debug],[build debug version without optimization])],
               [debug=$enableval], 
-              [debug=no])
+              [debug="no"])
+AM_CONDITIONAL([ENABLE_DEBUG], [test x"$enable_debug" = x"yes"])
 
 # does the user want to build demo applications?
 AC_ARG_ENABLE([apps],
               [AS_HELP_STRING([--enable-apps],[build demo applications])],
               [enable_apps=$enableval],
-              [enable_apps="yes"])
+              [enable_apps="no"])
 AM_CONDITIONAL([BUILD_APPS], [test x"$enable_apps" = x"yes"])
 
 
@@ -101,55 +102,64 @@
 AC_ARG_WITH([swig], 
             [AS_HELP_STRING([--with-swig],[use swig wrapper generator])],
             [with_swig=$withval],
-            [with_swig=no])
+            [with_swig="no"])
+AM_CONDITIONAL([WITH_SWIG], [test x"$with_swig" = x"yes"])
 
 # does the user want to compile python modules as well?
 AC_ARG_WITH([python], 
             [AS_HELP_STRING([--with-python],[use python for wrapping])],
             [with_python=$withval],
-            [with_python=yes])
+            [with_python="no"])
+AM_CONDITIONAL([WITH_PYTHON], [test x"$with_python" = x"yes"])
 
 # does the user want OpenCV to use xine video libraries?
 AC_ARG_WITH([xine],
             [AS_HELP_STRING([--with-xine],[use xine libraries (see LICENSE)])],
             [with_xine=$withval],
-            [with_xine=no])
+            [with_xine=i"no"])
+AM_CONDITIONAL([WITH_XINE], [test x"$with_xine" = x"yes"])
 
 # does the user want OpenCV to use ffmpeg video libraries?
 AC_ARG_WITH([ffmpeg],
             [AS_HELP_STRING([--with-ffmpeg],[use ffmpeg libraries (see LICENSE)])],
             [with_ffmpeg=$withval],
             [with_ffmpeg=auto])
+AM_CONDITIONAL([WITH_XINE], [test x"$with_xine" = x"yes"])
 
 # does the user want OpenCV to use libraw1394/libdc1394 for video input?
 AC_ARG_WITH([1394libs],
             [AS_HELP_STRING([--with-1394libs],[use libraw1394/libdc1394])],
             [with_1394libs=$withval],
-            [with_1394libs=auto])
+            [with_1394libs="no"])
+AM_CONDITIONAL([WITH_1394LIBS], [test x"$with_1394libs" = x"yes"])
 
 # does the user want OpenCV to use the video4linux kernel module?
 AC_ARG_WITH([v4l],
             [AS_HELP_STRING([--with-v4l],[use video4linux])],
             [with_v4l=$withval],
-            [with_v4l=auto])
+            [with_v4l="no"])
+AM_CONDITIONAL([WITH_V4L], [test x"$with_v4l" = x"yes"])
 
 # does the user want OpenCV to use QuickTime video libraries?
 AC_ARG_WITH([quicktime],
             [AS_HELP_STRING([--with-quicktime],[use QuickTime libraries])],
             [with_quicktime=$withval],
-            [with_quicktime=yes])
+            [with_quicktime="no"])
+AM_CONDITIONAL([WITH_QUICKTIME], [test x"$with_quicktime" = x"yes"])
 
 # does the user want OpenCV to use the Carbon GUI of Mac OS X?
 AC_ARG_WITH([carbon],
             [AS_HELP_STRING([--with-carbon],[use Carbon windows])],
             [with_carbon=$withval],
-            [with_carbon=yes])
+            [with_carbon="no"])
+AM_CONDITIONAL([WITH_CARBON], [test x"$with_carbon" = x"yes"])
 
 # does the user want OpenCV to use GTK+ 2.0 for the GUI?
 AC_ARG_WITH([gtk],
             [AS_HELP_STRING([--with-gtk],[use gtk+ 2.0 windows])],
             [with_gtk=$withval],
-            [with_gtk=auto])
+            [with_gtk="no"])
+AM_CONDITIONAL([WITH_GTK], [test x"$with_gtk" = x"yes"])
 
 
 ######################################################################
@@ -187,7 +197,7 @@
     AC_MSG_RESULT([yes])
     if test x"$ac_cv_c_compiler_gnu" = "xyes"; then
         CPPFLAGS="-DDEBUG -D_DEBUG $CPPFLAGS"
-        CXXFLAGS="-ggdb -O0 $CXXFLAGS"
+        CXXFLAGS="-ggdb -O0 -g $CXXFLAGS"
     fi
 else
     AC_MSG_RESULT([no])
@@ -202,7 +212,7 @@
     i686-*-*)
         if test x"$ac_cv_c_compiler_gnu" = "xyes"; then
           # default to i686/pentiumpro -- people can override this
-          CXXFLAGS="-g -march=i686 -ffast-math -fomit-frame-pointer $CXXFLAGS"
+          CXXFLAGS="-march=i686 -ffast-math -fomit-frame-pointer $CXXFLAGS"
         fi
         ;;
     powerpc-*-*)
@@ -340,7 +350,7 @@
 
 # if QuickTime is present, it will override everything else,
 # (unless '--without-quicktime' was given)
-if test x"$with_quicktime" = "xyes"; then
+if test x"$with_quicktime" = x"yes"; then
 
     # check other ./configure flags
     if test x"$with_ffmpeg" = "xyes"; then
@@ -690,7 +700,7 @@
 AM_CONDITIONAL([BUILD_PYTHON_WRAPPERS], [test x"$have_python" = "xyes"])
 
 # check for swig itself
-SWIG=""
+SWIG="no"
 if test x"$with_swig" = "xyes"; then
     AC_MSG_NOTICE([Checking for SWIG])
 
@@ -821,6 +831,7 @@
     Use dc1394 & raw1394:     ${have_dc1394}
     Use v4l:                  ${have_v4l}
     Use v4l2:                 ${have_v4l2}
+    Use gtk:		      ${have_gtk}
     
 Wrappers for other languages =========================================
     SWIG                      ${SWIG}

