diff -aur opencv-1.0.0-orig/samples/c/Makefile.am opencv-1.0.0/samples/c/Makefile.am
--- opencv-1.0.0-orig/samples/c/Makefile.am	2006-10-18 10:27:23.000000000 -0400
+++ opencv-1.0.0/samples/c/Makefile.am	2008-06-09 11:58:36.000000000 -0400
@@ -1,3 +1,5 @@
+if INSTALL_SAMPLES
+
 csamplesdir=$(pkgdatadir)/samples/c
 
 dist_csamples_DATA = airplane.jpg baboon.jpg lena.jpg fruits.jpg stuff.jpg \
@@ -10,4 +12,5 @@
 inpaint.cpp laplace.c letter_recog.cpp lkdemo.c minarea.c morphology.c motempl.c \
 mushroom.cpp pyramid_segmentation.c squares.c watershed.cpp
 
+endif
 
diff -aur opencv-1.0.0-orig/samples/python/Makefile.am opencv-1.0.0/samples/python/Makefile.am
--- opencv-1.0.0-orig/samples/python/Makefile.am	2006-10-04 04:02:12.000000000 -0400
+++ opencv-1.0.0/samples/python/Makefile.am	2008-06-09 12:00:25.000000000 -0400
@@ -1,3 +1,5 @@
+if INSTALL_SAMPLES
+if WITH_PYTHON
 csamplesdir=$(pkgdatadir)/samples/python
 
 dist_csamples_SCRIPTS =     \
@@ -29,3 +31,6 @@
     pyramid_segmentation.py \
     squares.py              \
     watershed.py
+
+endif
+endif
