diff -ur amos-2.0.8.orig/src/Align/align.cc amos-2.0.8/src/Align/align.cc
--- amos-2.0.8.orig/src/Align/align.cc	2008-03-19 16:07:24.000000000 +0200
+++ amos-2.0.8/src/Align/align.cc	2009-07-30 19:40:51.000000000 +0300
@@ -247,7 +247,7 @@
 //  Add  incr_val  to the  after  count in this  Vote_t  for  ch .
 
   {
-   char  * p;
+   const char  * p;
 
    p = strchr (ALPHABET, tolower (ch));
    if  (p == NULL)
@@ -285,7 +285,7 @@
 //  count for a blank.
 
   {
-   char  * p;
+   const char  * p;
 
    p = strchr (ALPHABET, tolower (ch));
    if  (p == NULL)
@@ -357,7 +357,7 @@
 
   {
    int  i;
-   char  * p;
+   const char  * p;
 
    for  (i = 0;  i <= ALPHABET_SIZE;  i ++)
      here [i] = after [i] = 0;
diff -ur amos-2.0.8.orig/src/Align/align_poly.cc amos-2.0.8/src/Align/align_poly.cc
--- amos-2.0.8.orig/src/Align/align_poly.cc	2006-03-30 05:03:12.000000000 +0300
+++ amos-2.0.8/src/Align/align_poly.cc	2009-07-30 19:43:44.000000000 +0300
@@ -247,7 +247,7 @@
 //  Add 1 to the  after  count in this  Vote_t  for  ch .
 
   {
-   char  * p;
+   const char  * p;
 
    p = strchr (ALPHABET, tolower (ch));
    if  (p == NULL)
@@ -285,7 +285,7 @@
 //  count for a blank.
 
   {
-   char  * p;
+   const char  * p;
 
    p = strchr (ALPHABET, tolower (ch));
    if  (p == NULL)
@@ -357,7 +357,7 @@
 
   {
    int  i;
-   char  * p;
+   const char  * p;
 
    for  (i = 0;  i <= ALPHABET_SIZE;  i ++)
      here [i] = after [i] = 0;
diff -ur amos-2.0.8.orig/src/AMOS/IDMap_AMOS.cc amos-2.0.8/src/AMOS/IDMap_AMOS.cc
--- amos-2.0.8.orig/src/AMOS/IDMap_AMOS.cc	2006-11-21 17:45:46.000000000 +0200
+++ amos-2.0.8/src/AMOS/IDMap_AMOS.cc	2009-07-30 19:30:03.000000000 +0300
@@ -10,6 +10,7 @@
 #include "IDMap_AMOS.hh"
 #include <string>
 #include <sstream>
+#include <cstdio>
 #include <cstring>
 using namespace AMOS;
 using namespace std;
diff -ur amos-2.0.8.orig/src/AMOS/Message_AMOS.cc amos-2.0.8/src/AMOS/Message_AMOS.cc
--- amos-2.0.8.orig/src/AMOS/Message_AMOS.cc	2005-01-07 23:53:05.000000000 +0200
+++ amos-2.0.8/src/AMOS/Message_AMOS.cc	2009-07-30 19:32:21.000000000 +0300
@@ -7,6 +7,8 @@
 //!
 ////////////////////////////////////////////////////////////////////////////////
 
+#include <cstdio>
+
 #include "Message_AMOS.hh"
 using namespace AMOS;
 using namespace std;
diff -ur amos-2.0.8.orig/src/Common/amp.cc amos-2.0.8/src/Common/amp.cc
--- amos-2.0.8.orig/src/Common/amp.cc	2006-10-25 16:56:41.000000000 +0300
+++ amos-2.0.8/src/Common/amp.cc	2009-07-30 19:33:59.000000000 +0300
@@ -1,6 +1,7 @@
 #include "amp.hh"
 #include <new>
 #include <iostream>
+#include <cstdio>
 #include <cstring>
 #include <ctime>
 using namespace std;
diff -ur amos-2.0.8.orig/src/Foundation/FileSystem.cc amos-2.0.8/src/Foundation/FileSystem.cc
--- amos-2.0.8.orig/src/Foundation/FileSystem.cc	2005-07-09 00:52:03.000000000 +0300
+++ amos-2.0.8/src/Foundation/FileSystem.cc	2009-07-30 19:35:48.000000000 +0300
@@ -55,7 +55,7 @@
   {
     // Check to see if path was given
 
-    char * end_of_path = strrchr(filename, PATH_DELIMINATOR);
+    const char * end_of_path = strrchr(filename, PATH_DELIMINATOR);
     
     if (end_of_path)
     {
diff -ur amos-2.0.8.orig/src/Foundation/Options.cc amos-2.0.8/src/Foundation/Options.cc
--- amos-2.0.8.orig/src/Foundation/Options.cc	2008-06-22 18:19:07.000000000 +0300
+++ amos-2.0.8/src/Foundation/Options.cc	2009-07-30 19:37:40.000000000 +0300
@@ -26,6 +26,7 @@
  */
 
 #include "Options.hh"
+#include <cstdio>
 #include <cstring>
 
 //! Constructor takes command line options in standard argc, argv format
