diff --git a/setup.cfg b/setup.cfg
index fcb594a..dccad97 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -9,7 +9,7 @@ ignore = E127,E201,E203,E221,E222,E241,E271,E272,E301,E302,E303,E701,W504
 
 [tool:pytest]
 testpaths = tests
-addopts   = -v --cov=fsleyes
+addopts   = -v
 markers   =
   overlayclitest: Overlay scene test invoked via FSLeyes command line
   clitest:        Test invoked via FSLeyes command line
