--- a/setup.py	2017-02-23 13:31:01.608732993 +0100
+++ b/setup.py	2017-02-23 13:33:44.290607540 +0100
@@ -10,7 +10,8 @@
 	https://github.com/pypa/sampleproject
 """
 
-from distutils.core import setup
+"""from distutils.core import setup"""
+from setuptools import setup
 
 setup(name='busfactor',
 	version='0.2',
@@ -26,4 +27,10 @@
 							 'data/*.json', 'js/*.js', 'js/bootstrap/*.js',
 							 'js/vendor/*.js', '*.png', 'index.html', 
 							 'LICENSE']},
+	entry_points = {
+		"console_scripts": [
+			"busfactor_gui = busfactor.main:main",
+		]
+	},
+	
 )
