--- salmon-9999/scripts/fetchRapMap.sh	2018-06-25 11:11:07.019798309 +0200
+++ salmon-9999/scripts/fetchRapMap.sh	2018-06-25 11:12:25.131824486 +0200
@@ -29,7 +29,8 @@
 EXPECTED_SHA256=c4ca27de299bee395b404d117d9d98ad9f0d6ee256ea1deb2890ea402893e688
 
 mkdir -p ${EXTERNAL_DIR}
-curl -k -L https://github.com/COMBINE-lab/RapMap/archive/${SVER}.zip -o ${EXTERNAL_DIR}/rapmap.zip
+# curl -k -L https://github.com/COMBINE-lab/RapMap/archive/${SVER}.zip -o ${EXTERNAL_DIR}/rapmap.zip
+cp ../../distdir/salmon-0.10.2_RapMap.zip ${EXTERNAL_DIR}/rapmap.zip
 
 hashcheck=""
 if exists sha256sum; then
