diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1be02ab..57ef279 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -327,17 +327,6 @@ if( DEFINED CUSTOM_BOOST_PATH )
   set(CMAKE_LIBRARY_PATH ${CUSTOM_BOOST_PATH}/lib ${CMAKE_LIBRARY_PATH})
 endif()
 
-##
-# We want static, multithreaded boost libraries
-##
-if(CONDA_BUILD)
-  set(Boost_USE_STATIC_LIBS OFF)
-elseif(USE_SHARED_LIBS)   # CI failed when using an OR statement above...
-  set(Boost_USE_STATIC_LIBS OFF)
-else()
-  set(Boost_USE_STATIC_LIBS ON)
-endif()
-
 set(Boost_USE_MULTITHREADED ON)
 #set(Boost_USE_STATIC_RUNTIME OFF)
 set(Boost_USE_DEBUG_RUNTIME OFF)
@@ -428,7 +417,7 @@ endif()
 ##
 # Set the latest version and look for what we need
 ##
-set(Boost_ADDITIONAL_VERSIONS "1.59.0" "1.60.0" "1.61.0" "1.62.0" "1.63.0" "1.64.0" "1.65.0" "1.66.0" "1.67.0" "1.68.0" "1.69.0" "1.70.0" "1.71.0")
+set(Boost_ADDITIONAL_VERSIONS "1.59.0" "1.60.0" "1.61.0" "1.62.0" "1.63.0" "1.64.0" "1.65.0" "1.66.0" "1.67.0" "1.68.0" "1.69.0" "1.70.0" "1.71.0" "1.74.0" "1.75.0")
 if (NOT BOOST_RECONFIGURE)
 find_package(Boost 1.59.0 COMPONENTS iostreams filesystem system timer chrono program_options)
 message("BOOST_INCLUDEDIR = ${BOOST_INCLUDEDIR}")
