diff --git a/tests/test_command.py b/tests/test_command.py
index a81b04a..cf40233 100644
--- a/tests/test_command.py
+++ b/tests/test_command.py
@@ -76,7 +76,7 @@ def test_force_fasta_output(tmpdir, cores):
     assert_files_equal(cutpath("small.fasta"), out_path)
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="Maybe this can be made to work")
+@pytest.mark.skipif(reason="Maybe this can be made to work")
 def test_non_utf8_locale():
     subprocess.check_call(
         [sys.executable, "-m", "cutadapt", "-o", os.devnull, datapath("small.fastq")],
