--- cgnslib_2.5.orig/Makefile.in	2008-01-19 04:45:16.000000000 +0100
+++ cgnslib_2.5/Makefile.in	2008-10-16 00:31:07.000000000 +0200
@@ -75,15 +75,17 @@
 	-$(RM) config.log config.status Makefile
 
 install : $(CGNSLIB)
-	$(INSTALL_DATA) cgnslib.h $(INCLUDEDIR)/cgnslib.h
-	$(INSTALL_DATA) cgnslib_f.h $(INCLUDEDIR)/cgnslib_f.h
-	$(INSTALL_DATA) cgnswin_f.h $(INCLUDEDIR)/cgnswin_f.h
-	$(INSTALL_DATA) $(CGNSLIB) $(LIBDIR)/$(INSTLIB)
+	$(INSTALL_DATA) -D cgnslib.h $(DESTDIR)/$(INCLUDEDIR)/cgnslib.h
+	$(INSTALL_DATA) -D cgnslib_f.h $(DESTDIR)/$(INCLUDEDIR)/cgnslib_f.h
+	$(INSTALL_DATA) -D cgnswin_f.h $(DESTDIR)/$(INCLUDEDIR)/cgnswin_f.h
+	$(INSTALL_DATA) -D cgns_header.h $(DESTDIR)/$(INCLUDEDIR)/cgns_header.h
+	$(INSTALL_DATA) -D $(CGNSLIB) $(DESTDIR)/$(LIBDIR)/$(INSTLIB).2.0.0
+	(cd $(DESTDIR)/$(LIBDIR)/; ln -s $(INSTLIB).2.0.0 $(INSTLIB); ln -s $(INSTLIB).2.0.0 $(INSTLIB).2)
 
 uninstall :
-	-$(RM) $(INCLUDEDIR)/cgnslib.h
-	-$(RM) $(INCLUDEDIR)/cgnslib_f.h
-	-$(RM) $(INCLUDEDIR)/cgnswin_f.h
+	-$(RM) $(DESTDIR)/$(INCLUDEDIR)/cgnslib.h
+	-$(RM) $(DESTDIR)/$(INCLUDEDIR)/cgnslib_f.h
+	-$(RM) $(DESTDIR)/$(INCLUDEDIR)/cgnswin_f.h
 	-$(RM) $(LIBDIR)/$(INSTLIB)
 
 #---------- mid-level library
--- cgnslib_2.5.orig/configure.in	2008-01-19 04:45:16.000000000 +0100
+++ cgnslib_2.5/configure.in	2008-10-15 22:58:22.000000000 +0200
@@ -423,7 +423,7 @@
       shared=no
     else
       CFGFLAGS="-fPIC $CFGFLAGS"
-      AR="\$(CC) -shared $SYSCFLAGS -Wl,-rpath,$cgnsdir/\$(SYSTEM)"
+      AR="\$(CC) -shared $SYSCFLAGS -Wl,-soname,libcgns.so.2"
       LIBEXT=so
     fi
   fi
--- cgnslib_2.5.orig/configure	2008-01-19 04:45:16.000000000 +0100
+++ cgnslib_2.5/configure	2008-10-15 23:11:41.000000000 +0200
@@ -2048,7 +2048,7 @@
       shared=no
     else
       CFGFLAGS="-fPIC $CFGFLAGS"
-      AR="\$(CC) -shared $SYSCFLAGS -Wl,-rpath,$cgnsdir/\$(SYSTEM)"
+      AR="\$(CC) -shared $SYSCFLAGS -Wl,-soname,libcgns.so.2"
       LIBEXT=so
     fi
   fi
--- cgnslib_2.5.orig/make.defs.in	2006-10-03 01:06:41.000000000 +0200
+++ cgnslib_2.5/make.defs.in	2008-10-15 23:49:36.000000000 +0200
@@ -34,7 +34,7 @@
 O       = @OBJEXT@
 A       = @LIBEXT@
 EXE     = @EXEEXT@
-LIBCGNS = @SYSTEM@/libcgns.@LIBEXT@
+LIBCGNS = @SYSTEM@/libcgns.@LIBEXT@.2
 
 #------------------------------------------------------------------------
 # CC     - C compiler
@@ -75,7 +75,7 @@
 # strip command for executables - set to true if not used
 #------------------------------------------------------------------------
 
-STRIP  = @STRIP@
+STRIP  = true
 
 #------------------------------------------------------------------------
 # library archiver and ranlib
 