diff -arNu -uarN work/auxil/catcheck/catcheck.cc work.new/auxil/catcheck/catcheck.cc
--- work/auxil/catcheck/catcheck.cc	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/catcheck/catcheck.cc	2009-08-31 23:08:22.000000000 +0200
@@ -1,6 +1,7 @@
 
 #include <unistd.h>
 #include <iostream>
+#include <algorithm>
 #include "catcheck.hh"
 
 int main(int argc, char **argv) {
diff -arNu -uarN work/auxil/catcheck/Makefile.am work.new/auxil/catcheck/Makefile.am
--- work/auxil/catcheck/Makefile.am	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/catcheck/Makefile.am	2009-08-31 23:08:57.000000000 +0200
@@ -4,3 +4,8 @@
 check_PROGRAMS = catcheck
 
 catcheck_SOURCES = catcheck.cc catcheck.hh
+
+INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+ 
+catcheck_LDADD = $(MMDB_LIBS)
+
diff -arNu -uarN work/auxil/rama/Makefile.am work.new/auxil/rama/Makefile.am
--- work/auxil/rama/Makefile.am	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/rama/Makefile.am	2009-08-31 23:09:14.000000000 +0200
@@ -5,3 +5,7 @@
 
 rama_data_SOURCES = rama-data.cc
 rama_doughnut_SOURCES = rama-doughnut.cc
+
+rama_data_LDADD = $(CLIPPER_LIBS)
+rama_doughnut_LDADD = $(CLIPPER_LIBS)
+
diff -arNu -uarN work/auxil/sec-str/Makefile.am work.new/auxil/sec-str/Makefile.am
--- work/auxil/sec-str/Makefile.am	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/sec-str/Makefile.am	2009-08-31 23:09:54.000000000 +0200
@@ -5,5 +5,7 @@
 
 sec_str_dist_check_SOURCES = sec-str-dist-check.cc sec-str-dist-check.hh
 
-
+INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+ 
+sec_str_dist_check_LDADD = $(MMDB_LIBS) $(CLIPPER_LIBS)
 
diff -arNu -uarN work/auxil/strand-analysis/Makefile.am work.new/auxil/strand-analysis/Makefile.am
--- work/auxil/strand-analysis/Makefile.am	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/strand-analysis/Makefile.am	2009-08-31 23:10:07.000000000 +0200
@@ -2,7 +2,7 @@
 EXTRA_DIST = Makefile.simple 
 
 
-INCLUDES = -I$(top_srcdir)/aux/sec-str \
+INCLUDES = -I$(top_srcdir)/auxil/sec-str \
 	$(MMDB_CXXFLAGS) 	       \
 	$(CLIPPER_CXXFLAGS) 		
 
diff -arNu -uarN work/auxil/strand-analysis/strand-analysis.cc work.new/auxil/strand-analysis/strand-analysis.cc
--- work/auxil/strand-analysis/strand-analysis.cc	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/strand-analysis/strand-analysis.cc	2009-08-31 23:08:40.000000000 +0200
@@ -4,6 +4,8 @@
 
 #include <unistd.h>
 #include <iostream>
+#include <cstdlib>
+#include <cstring>
 #include "strand-analysis.hh"
 
 int main(int argc, char **argv) {
diff -arNu -uarN work/auxil/vectors/Makefile.am work.new/auxil/vectors/Makefile.am
--- work/auxil/vectors/Makefile.am	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/vectors/Makefile.am	2009-08-31 23:10:22.000000000 +0200
@@ -5,3 +5,7 @@
 
 diff_vec_SOURCES = diff-vec.cc
 
+INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+
+diff_vec_LDADD = $(MMDB_LIBS) $(CLIPPER_LIBS)
+
