diff -arNu -uarN work/auxil/catcheck/catcheck.cc work.new/auxil/catcheck/catcheck.cc
--- work/auxil/catcheck/catcheck.cc	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/catcheck/catcheck.cc	2009-08-31 23:08:22.000000000 +0200
@@ -1,6 +1,7 @@
 
 #include <unistd.h>
 #include <iostream>
+#include <algorithm>
 #include "catcheck.hh"
 
 int main(int argc, char **argv) {
diff -arNu -uarN work/auxil/strand-analysis/strand-analysis.cc work.new/auxil/strand-analysis/strand-analysis.cc
--- work/auxil/strand-analysis/strand-analysis.cc	2009-08-31 23:07:25.000000000 +0200
+++ work.new/auxil/strand-analysis/strand-analysis.cc	2009-08-31 23:08:40.000000000 +0200
@@ -4,6 +4,8 @@
 
 #include <unistd.h>
 #include <iostream>
+#include <cstdlib>
+#include <cstring>
 #include "strand-analysis.hh"
 
 int main(int argc, char **argv) {
