Index: coot-utils/Makefile.am
===================================================================
--- coot-utils/Makefile.am	(revision 3299)
+++ coot-utils/Makefile.am	(working copy)
@@ -90,7 +90,8 @@
    $(top_builddir)/mini-mol/libcoot-mini-mol.la \
    libcoot-map-utils.la \
    libcoot-coord-utils.la \
-    $(top_builddir)/utils/libcoot-utils.la
+    $(top_builddir)/utils/libcoot-utils.la \
+	$(GSL_LIBS)
 
 
 noinst_PROGRAMS = test-h-bonds testshelxreader testcootutils
