Index: surface/Makefile.am
===================================================================
--- surface/Makefile.am	(revision 3299)
+++ surface/Makefile.am	(working copy)
@@ -1,7 +1,7 @@
 
 INCLUDES = -I$(top_srcdir)/coot-utils $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) $(FFTW_CXXFLAGS)
 
-AM_CPPFLAGS = $(OPENMP_CFLAGS)
+AM_CXXFLAGS = $(OPENMP_CFLAGS)
 
 lib_LTLIBRARIES = libccp4mg-surface.la
 
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 3299)
+++ src/Makefile.am	(working copy)
@@ -417,8 +417,8 @@
 # coot_real_LDFLAGS = -static -lstdc++
 coot_real_LDFLAGS =
 
-coot_real_LDADD =  $(OPENMP_CFLAGS)     \
-                $(COOT_EXTRA_LIBS)      \
+coot_real_LDADD =  \
+		$(COOT_EXTRA_LIBS)      \
 		$(GUILE_GTK_LIBS)       \
 		$(GUILE_LDFLAGS)	\
 		$(CLIPPER_LIBS) 	\
