diff --git a/ChangeLog b/ChangeLog
index 46f6362..fd3fd9a 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1963,7 +1963,7 @@
 2005-01-29  Paul Emsley  <paulemsley@ysbl.york.ac.uk>
 
 	* macros/mmdb-ssm.m4 (AM_WITH_MMDBSSM): use $with_ssmlib_prefix in
-	MMDBSSM_CXXFLAGS [Oops].  Hopefully fixes problems spotted by Joel
+	MMDBSSM_CFLAGS [Oops].  Hopefully fixes problems spotted by Joel
 	Bard and Joseph Toth.
 
 	* src/graphics-info-pick.cc: (symmetry_atom_pick): tried to add
@@ -2256,14 +2256,14 @@
 
 2004-11-20  Paul Emsley  <paulemsley@ysbl.york.ac.uk>
 
-	* coords/Makefile.am (INCLUDES): Added CLIPPER_CXXFLAGS [for
+	* coords/Makefile.am (INCLUDES): Added CLIPPER_CFLAGS [for
 	coot-close.cc usage].
 
 2004-11-19  Paul Emsley  <paulemsley@ysbl.york.ac.uk>
 
-	* sequence-view/Makefile.am (INCLUDES): added CLIPPER_CXXFLAGS
+	* sequence-view/Makefile.am (INCLUDES): added CLIPPER_CFLAGS
 
-	* analysis/Makefile.am (INCLUDES): Added CLIPPER_CXXFLAGS
+	* analysis/Makefile.am (INCLUDES): Added CLIPPER_CFLAGS
 
 2004-11-16  Paul Emsley  <emsley@localhost>
 
@@ -2273,7 +2273,7 @@
  
 2004-11-15  Paul Emsley  <emsley@localhost>
 
-	* coot-utils/Makefile.am: add MMDB_CXXFLAGS to INCLUDE [thanks 
+	* coot-utils/Makefile.am: add MMDB_CFLAGS to INCLUDE [thanks 
           Joseph Toth].
 
 	* surface/CXXSphereNode.h: removed the ; again!
diff --git a/analysis/Makefile.am b/analysis/Makefile.am
index 8a873dc..ef4a9cf 100644
--- a/analysis/Makefile.am
+++ b/analysis/Makefile.am
@@ -8,8 +8,8 @@ pkgincludedir = $(includedir)/coot/analysis
 
 INCLUDES = \
 	-I$(top_srcdir) \
-	$(CLIPPER_CXXFLAGS)		\
-	$(MMDB_CXXFLAGS)
+	$(CLIPPER_CFLAGS)		\
+	$(MMDB_CFLAGS)
 
 libcoot_analysis_la_SOURCES = bfkurt.cc mogul.cc kolmogorov.hh kolmogorov.cc cablam.hh cablam.cc
 
diff --git a/angles/Makefile.am b/angles/Makefile.am
index b89276d..935ce06 100644
--- a/angles/Makefile.am
+++ b/angles/Makefile.am
@@ -7,7 +7,7 @@ EXTRA_DIST = \
 pkginclude_HEADERS = AngleInfo.h ncube.h wrapped-ncube.h
 pkgincludedir = $(includedir)/coot/angles
 
-INCLUDES = $(CLIPPER_CXXFLAGS)
+INCLUDES = $(CLIPPER_CFLAGS)
 
 libcoot_AngleInfo_a_sources = AngleInfo.cc AngleInfo.h  \
 	AngleInfo-angle-torsions-autogen.cc AngleInfo-theta_2-autogen.cc
diff --git a/auxil/catcheck/Makefile.am b/auxil/catcheck/Makefile.am
index 41e1766..69e6091 100644
--- a/auxil/catcheck/Makefile.am
+++ b/auxil/catcheck/Makefile.am
@@ -1,7 +1,7 @@
 
 EXTRA_DIST = Makefile.simple
 
-INCLUDES = $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) 
+INCLUDES = $(MMDB_CFLAGS) $(CLIPPER_CFLAGS) 
 
 check_PROGRAMS = catcheck
 
diff --git a/auxil/catcheck/Makefile.simple b/auxil/catcheck/Makefile.simple
index 5157398..0e25fb2 100644
--- a/auxil/catcheck/Makefile.simple
+++ b/auxil/catcheck/Makefile.simple
@@ -1,17 +1,17 @@
 
-# CLIPPER_CXXFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
+# CLIPPER_CFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
 
 
 BASE = /home/paule/autobuild/Linux-kalypso-pre-release
-CLIPPER_CXXFLAGS = -I$(BASE)/include
-MMDB_CXXFLAGS = -I$(BASE)/include/mmdb
+CLIPPER_CFLAGS = -I$(BASE)/include
+MMDB_CFLAGS = -I$(BASE)/include/mmdb
 
 CLIPPER_LIBS = -L$(BASE)/lib -lclipper-ccp4 \
 	-lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb \
 	-lclipper-core -lccp4c -L$(BASE)/lib \
 	-L$(BASE) -lmmdb -lm -lrfftw -lfftw -lz -lm
 
-INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+INCLUDES = $(CLIPPER_CFLAGS) $(MMDB_CFLAGS)
 LDLIBS = $(CLIPPER_LIBS)
 
 catcheck: catcheck.cc catcheck.hh
diff --git a/auxil/rama/Makefile.am b/auxil/rama/Makefile.am
index cdbb54e..12c6731 100644
--- a/auxil/rama/Makefile.am
+++ b/auxil/rama/Makefile.am
@@ -1,7 +1,7 @@
 
 EXTRA_DIST = Makefile.simple fit-rama.scm run.scm test-arr.scm
 
-INCLUDES =  $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+INCLUDES =  $(CLIPPER_CFLAGS) $(MMDB_CFLAGS)
 
 
 check_PROGRAMS = rama-data rama-doughnut 
diff --git a/auxil/rama/Makefile.simple b/auxil/rama/Makefile.simple
index 10fe22e..715a270 100644
--- a/auxil/rama/Makefile.simple
+++ b/auxil/rama/Makefile.simple
@@ -1,12 +1,12 @@
 
-CLIPPER_CXXFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
+CLIPPER_CFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
 
 CLIPPER_LIBS = -L/Users/paulemsley/ccp4/ysbl-distro-based/lib -lclipper-ccp4 \
 	-lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb \
 	-lclipper-core -lccp4c -L/Users/paulemsley/ccp4/ysbl-distro-based/lib \
 	-L/Users/paulemsley/ccp4/ysbl-distro-based -lmmdb -lm -lrfftw -lfftw -lz -lm
 
-INCLUDES = $(CLIPPER_CXXFLAGS)
+INCLUDES = $(CLIPPER_CFLAGS)
 LDLIBS = $(CLIPPER_LIBS)
 
 rama-data: rama-data.cc
diff --git a/auxil/sec-str/Makefile.am b/auxil/sec-str/Makefile.am
index 81f79e8..bfef73a 100644
--- a/auxil/sec-str/Makefile.am
+++ b/auxil/sec-str/Makefile.am
@@ -1,6 +1,6 @@
 
 EXTRA_DIST = Makefile.simple sec-str-dist-check.cc sec-str-dist-check.hh
-INCLUDES = $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) 
+INCLUDES = $(MMDB_CFLAGS) $(CLIPPER_CFLAGS) 
 
 #check_PROGRAMS = sec-str-dist-check
 
diff --git a/auxil/sec-str/Makefile.simple b/auxil/sec-str/Makefile.simple
index eb6db20..8b22930 100644
--- a/auxil/sec-str/Makefile.simple
+++ b/auxil/sec-str/Makefile.simple
@@ -1,27 +1,27 @@
 
-#CLIPPER_CXXFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
+#CLIPPER_CFLAGS = -I/Users/paulemsley/ccp4/ysbl-distro-based/include
 
 #CLIPPER_LIBS = -L/Users/paulemsley/ccp4/ysbl-distro-based/lib -lclipper-ccp4 \
 #	-lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb \
 #	-lclipper-core -lccp4c -L/Users/paulemsley/ccp4/ysbl-distro-based/lib \
 #	-L/Users/paulemsley/ccp4/ysbl-distro-based -lmmdb -lm -lrfftw -lfftw -lz -lm
 
-#INCLUDES = $(CLIPPER_CXXFLAGS)
+#INCLUDES = $(CLIPPER_CFLAGS)
 #LDLIBS = $(CLIPPER_LIBS)
 
 
 # 
 
 BASE = /home/paule/autobuild/Linux-kalypso-pre-release
-CLIPPER_CXXFLAGS = -I$(BASE)/include
-MMDB_CXXFLAGS = -I$(BASE)/include/mmdb
+CLIPPER_CFLAGS = -I$(BASE)/include
+MMDB_CFLAGS = -I$(BASE)/include/mmdb
 
 CLIPPER_LIBS = -L$(BASE)/lib -lclipper-ccp4 \
 	-lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb \
 	-lclipper-core -lccp4c -L$(BASE)/lib \
 	-L$(BASE) -lmmdb -lm -lrfftw -lfftw -lz -lm
 
-INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+INCLUDES = $(CLIPPER_CFLAGS) $(MMDB_CFLAGS)
 LDLIBS = $(CLIPPER_LIBS)
 
 sec-str-dist-check: sec-str-dist-check.cc sec-str-dist-check.hh
diff --git a/auxil/strand-analysis/Makefile.am b/auxil/strand-analysis/Makefile.am
index 11686ba..9c24b3b 100644
--- a/auxil/strand-analysis/Makefile.am
+++ b/auxil/strand-analysis/Makefile.am
@@ -3,8 +3,8 @@ EXTRA_DIST = Makefile.simple strand-analysis.cc
 
 
 INCLUDES = -I$(top_srcdir)/auxil/sec-str \
-	$(MMDB_CXXFLAGS) 	         \
-	$(CLIPPER_CXXFLAGS) 		
+	$(MMDB_CFLAGS) 	         \
+	$(CLIPPER_CFLAGS) 		
 
 
 check_PROGRAMS = 
diff --git a/auxil/strand-analysis/Makefile.simple b/auxil/strand-analysis/Makefile.simple
index d5c21a4..2f58b8d 100644
--- a/auxil/strand-analysis/Makefile.simple
+++ b/auxil/strand-analysis/Makefile.simple
@@ -2,13 +2,13 @@
 
 CLIPPER = /Users/paulemsley/ccp4/ysbl-distro-based
 
-CLIPPER_CXXFLAGS = -I$(CLIPPER)/include
+CLIPPER_CFLAGS = -I$(CLIPPER)/include
 
 CLIPPER_LIBS = -L$(CLIPPER)/lib -lclipper-ccp4 \
 	-lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb \
 	-lclipper-core -lccp4c -lmmdb -lm -lrfftw -lfftw -lz -lm
 
-INCLUDES = -I../sec-str $(CLIPPER_CXXFLAGS)
+INCLUDES = -I../sec-str $(CLIPPER_CFLAGS)
 LDLIBS = $(CLIPPER_LIBS)
 
 .cc:
diff --git a/auxil/vectors/Makefile.am b/auxil/vectors/Makefile.am
index 8826587..4a8d72e 100644
--- a/auxil/vectors/Makefile.am
+++ b/auxil/vectors/Makefile.am
@@ -1,6 +1,6 @@
 
 EXTRA_DIST = Makefile.simple diff-vec.cc
-INCLUDES = $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS)
+INCLUDES = $(MMDB_CFLAGS) $(CLIPPER_CFLAGS)
 check_PROGRAMS = 
 
 #diff_vec_SOURCES = diff-vec.cc
diff --git a/build/Makefile.am b/build/Makefile.am
index 6afe5a3..748341b 100644
--- a/build/Makefile.am
+++ b/build/Makefile.am
@@ -25,8 +25,8 @@ pkgincludedir = $(includedir)/coot/build
 
 INCLUDES =  \
 	-I$(top_srcdir) \
-	$(CLIPPER_CXXFLAGS) \
-	$(MMDB_CXXFLAGS)
+	$(CLIPPER_CFLAGS) \
+	$(MMDB_CFLAGS)
 
 noinst_PROGRAMS = testcabuild
 
diff --git a/ccp4mg-utils/Makefile.am b/ccp4mg-utils/Makefile.am
index 7734f09..c58fe1d 100644
--- a/ccp4mg-utils/Makefile.am
+++ b/ccp4mg-utils/Makefile.am
@@ -26,7 +26,7 @@ pkgincludedir = $(includedir)/coot/ccp4mg-utils
 EXTRA_DIST = convert.sh get-new-utils.sh
 
 INCLUDES =  \
-	$(MMDB_CXXFLAGS) \
+	$(MMDB_CFLAGS) \
 	$(LIBPNG_CFLAGS)
 
 libccp4mg_utils_la_SOURCES = geomutil.cc mgtree.cc cartesian.cc \
diff --git a/ccp4mg-utils/mmut/Makefile.am b/ccp4mg-utils/mmut/Makefile.am
old mode 100755
new mode 100644
index d0651db..d4e0bf8
--- a/ccp4mg-utils/mmut/Makefile.am
+++ b/ccp4mg-utils/mmut/Makefile.am
@@ -70,9 +70,9 @@ INCLUDES = \
 	$(LIBGLADE_CFLAGS)		\
 	@GTK_CFLAGS@ 			\
 	$(GLUT_CFLAGS)			\
-	$(MMDBSSM_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
-	$(CLIPPER_CXXFLAGS) 		
+	$(MMDBSSM_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		
 
 #	$(WITH_SCRIPT_CFLAGS) 
 
diff --git a/ccp4mg-utils/pygl/Makefile.am b/ccp4mg-utils/pygl/Makefile.am
old mode 100755
new mode 100644
index a7e68b5..e26a6c8
--- a/ccp4mg-utils/pygl/Makefile.am
+++ b/ccp4mg-utils/pygl/Makefile.am
@@ -108,9 +108,9 @@ INCLUDES = \
 	$(LIBGLADE_CFLAGS)		\
 	@GTK_CFLAGS@ 			\
 	$(GLUT_CFLAGS)			\
-	$(MMDBSSM_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
-	$(CLIPPER_CXXFLAGS) 		
+	$(MMDBSSM_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		
 
 #	$(WITH_SCRIPT_CFLAGS) 
 
diff --git a/ccp4mg-utils/util/Makefile.am b/ccp4mg-utils/util/Makefile.am
old mode 100755
new mode 100644
index 399f453..3d2ff55
--- a/ccp4mg-utils/util/Makefile.am
+++ b/ccp4mg-utils/util/Makefile.am
@@ -73,9 +73,9 @@ INCLUDES = \
 	$(LIBGLADE_CFLAGS)		\
 	@GTK_CFLAGS@ 			\
 	$(GLUT_CFLAGS)			\
-	$(MMDBSSM_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
-	$(CLIPPER_CXXFLAGS) 		
+	$(MMDBSSM_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		
 
 #	$(WITH_SCRIPT_CFLAGS) 
 
diff --git a/configure.in b/configure.in
index cc549a5..d56f856 100644
--- a/configure.in
+++ b/configure.in
@@ -108,14 +108,18 @@ esac
 
 
 dnl Check for MMDB.
-AM_PATH_MMDB(, [echo Error locating necessary mmdb; exit 1],,)
+PKG_CHECK_MODULES([MMDB],[mmdb >= 1.2],,[AM_PATH_MMDB(, [echo Error locating necessary mmdb; exit 1],,)])
 
-AM_WITH_MMDBSSM
+PKG_CHECK_MODULES([MMDBSSM],[ssm],,[AM_WITH_MMDBSSM])
 AM_WITH_CCP4SRS
 
 dnl Check for Clipper.  we exit if clipper is not found.
 #
-AM_PATH_CLIPPER(, [echo Error locating Clipper - a CNS-enabled version of Clipper \(2.1-090520-ac or later\) is required now; exit 1],AC_MSG_ERROR([Cannot find Clipper]))
+PKG_CHECK_MODULES([CLIPPER], [clipper libccp4c], [
+  CLIPPER_LIBS="${CLIPPER_LIBS} -lrfftw -lfftw"; AC_SUBST([CLIPPER_LIBS])
+  ], [
+  AM_PATH_CLIPPER(, [echo Error locating Clipper - a CNS-enabled version of Clipper \(2.1-090520-ac or later\) is required now; exit 1],AC_MSG_ERROR([Cannot find Clipper]))
+  ])
 
 # we set default coot_guile to true
 coot_guile=true
@@ -186,11 +190,15 @@ AM_PATH_LIBGLADE
 
 dnl =========================== GooCanvas ================================
 
-AM_PATH_GOOCANVAS
-# give something for the preprocessor to chew on...
-if test "$GOOCANVAS_CFLAGS" != "" ; then
-   GOOCANVAS_CFLAGS="-DHAVE_GOOCANVAS $GOOCANVAS_CFLAGS"
-fi
+PKG_CHECK_MODULES([GOOCANVAS], [goocanvas],
+	[GOOCANVAS_CFLAGS="-DHAVE_GOOCANVAS $GOOCANVAS_CFLAGS"; AC_SUBST(GOOCANVAS_CFLAGS)],
+	[
+		AM_PATH_GOOCANVAS
+		# give something for the preprocessor to chew on...
+		if test "$GOOCANVAS_CFLAGS" != "" ; then
+   		GOOCANVAS_CFLAGS="-DHAVE_GOOCANVAS $GOOCANVAS_CFLAGS"
+		fi
+	])
 
 AC_ARG_WITH(enhanced-ligand-tools, [ --with-enhanced-ligand-tools with optional complicated/experimental ligand-related dependencies ],
  coot_enhanced_ligand_additions="true",
diff --git a/coords/Makefile.am b/coords/Makefile.am
index 05fdf70..1e1c5d2 100644
--- a/coords/Makefile.am
+++ b/coords/Makefile.am
@@ -29,7 +29,7 @@ INCLUDES = \
 	-I$(top_srcdir) \
 	$(ENHANCED_LIGAND_TOOLS_CFLAGS)      \
 	$(RDKIT_CXXFLAGS) 		\
-	$(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+	$(CLIPPER_CFLAGS) $(MMDB_CFLAGS)
 
 AM_CXXFLAGS = \
 	-DPKGDATADIR='"$(pkgdatadir)"' \
diff --git a/coot-surface/Makefile.am b/coot-surface/Makefile.am
index d2243d6..2434219 100644
--- a/coot-surface/Makefile.am
+++ b/coot-surface/Makefile.am
@@ -19,7 +19,7 @@
 # 02110-1301, USA
 
 INCLUDES = -I$(top_srcdir) \
-	$(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) $(GTKGL_CFLAGS) $(GLUT_CFLAGS)
+	$(MMDB_CFLAGS) $(CLIPPER_CFLAGS) $(GTKGL_CFLAGS) $(GLUT_CFLAGS)
 
 lib_LTLIBRARIES = libcoot-surface.la
 
diff --git a/coot-utils/Makefile.am b/coot-utils/Makefile.am
index 87810dc..c5b05b4 100644
--- a/coot-utils/Makefile.am
+++ b/coot-utils/Makefile.am
@@ -34,11 +34,10 @@ pkginclude_HEADERS = peak-search.hh coot-trim.hh \
 	read-sm-cif.hh bonded-pairs.hh tree.hh helix-analysis.hh elastic.hh 
 pkgincludedir = $(includedir)/coot/coot-utils
 
-
 INCLUDES = \
 	-I$(top_srcdir)	\
-	$(CLIPPER_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
 	$(GSL_CFLAGS)			\
 	$(WITH_GUILE_CFLAGS)		\
 	$(WITH_PYTHON_CFLAGS)
diff --git a/cootaneer/Makefile.am b/cootaneer/Makefile.am
index 3a1d720..cfca0fd 100644
--- a/cootaneer/Makefile.am
+++ b/cootaneer/Makefile.am
@@ -20,7 +20,7 @@
 EXTRA_DIST = 		\
 	cootaneer-llk-2.40.dat
 
-INCLUDES = 	$(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) 
+INCLUDES = 	$(CLIPPER_CFLAGS) $(MMDB_CFLAGS) 
 
 cootaneer_demo_SOURCES = cootaneer-demo.cpp
 
diff --git a/cootilus/Makefile.am b/cootilus/Makefile.am
index 7e4169d..4492014 100644
--- a/cootilus/Makefile.am
+++ b/cootilus/Makefile.am
@@ -20,7 +20,7 @@
 EXTRA_DIST = 		\
 	nautilus_lib.pdb
 
-INCLUDES = 	$(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) 
+INCLUDES = 	$(CLIPPER_CFLAGS) $(MMDB_CFLAGS) 
 
 cootilus_demo_SOURCES = cootilus-demo.cpp
 
diff --git a/db-main/Makefile.am b/db-main/Makefile.am
index 3a85b9e..1a11cdf 100644
--- a/db-main/Makefile.am
+++ b/db-main/Makefile.am
@@ -6,7 +6,7 @@ AM_CXXFLAGS = \
 	-DDATADIR='"$(datadir)"'
 
 INCLUDES = -I$(top_srcdir) \
-   $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS)
+   $(MMDB_CFLAGS) $(CLIPPER_CFLAGS)
 
 pkginclude_HEADERS = db-main.hh db-strands.hh
 pkgincludedir = $(includedir)/coot/db-main
diff --git a/db-main/Makefile.simple b/db-main/Makefile.simple
index cec561d..c207a43 100644
--- a/db-main/Makefile.simple
+++ b/db-main/Makefile.simple
@@ -19,7 +19,7 @@ CLIPPER = $(HOME)/crystal
 MCCP4   = $(HOME)/crystal
 
 CXX = g++
-MMDB_CXXFLAGS = -I$(MMDB)/include 
+MMDB_CFLAGS = -I$(MMDB)/include 
 FFTW_LIBS = -L$(FFTW)/lib -lrfftw -lfftw
 
 MMDB_LIBS = -L$(MMDB)/lib -lmmdb 
@@ -28,9 +28,9 @@ CLIPPER_LIBS = -L$(CLIPPER)/lib -lclipper-contrib \
 	-L$(CLIPPER)lib -L/h/paule/mapview/lib $(FFTW_LIBS)
 
 MCCP4_CFLAGS = -I$(MCCP4)/include
-CLIPPER_CXXFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
+CLIPPER_CFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
 
-INCLUDES = $(CLIPPER_CXXFLAGS) $(MCCP4_CFLAGS) $(MMDB_CXXFLAGS) -I../mini-mol
+INCLUDES = $(CLIPPER_CFLAGS) $(MCCP4_CFLAGS) $(MMDB_CFLAGS) -I../mini-mol
 
 CXXFLAGS = -g $(INCLUDES)
 LDFLAGS = -static -L../mini-mol -lcoot-mini-mol $(CLIPPER_LIBS) $(MMDB_LIBS) 
@@ -44,4 +44,4 @@ libcoot-db-main.a: db-main.o
 db-main.o: db-main.cc db-main.hh
 
 testmmdbbits: testmmdbbits.cc
-	$(CXX) $(MMDB_CXXFLAGS) $(MMDB_LIBS) $? -o $@
+	$(CXX) $(MMDB_CFLAGS) $(MMDB_LIBS) $? -o $@
diff --git a/density-contour/Makefile.am b/density-contour/Makefile.am
index 7f705e8..f5dcc00 100644
--- a/density-contour/Makefile.am
+++ b/density-contour/Makefile.am
@@ -20,7 +20,7 @@
 
 COORD_INCLUDES = -I$(top_srcdir)
 
-INCLUDES = $(COORD_INCLUDES) $(CLIPPER_CXXFLAGS) 
+INCLUDES = $(COORD_INCLUDES) $(CLIPPER_CFLAGS) 
 
 lib_LTLIBRARIES = libcoot-isosurface.la
 
diff --git a/geometry/Makefile.am b/geometry/Makefile.am
index 6dd665f..57aaa04 100644
--- a/geometry/Makefile.am
+++ b/geometry/Makefile.am
@@ -26,7 +26,7 @@ pkginclude_HEADERS =  protein-geometry.hh read-sbase.hh lbg-graph.hh srs-interfa
 pkgincludedir = $(includedir)/coot/geometry
 
 INCLUDES = -I$(top_srcdir) \
-	 $(CLIPPER_CXXFLAGS) $(CCP4SRS_CXXFLAGS) $(MMDB_CXXFLAGS)
+	 $(CLIPPER_CFLAGS) $(CCP4SRS_CXXFLAGS) $(MMDB_CFLAGS)
 
 AM_CXXFLAGS = \
 	-DDATADIR='"$(datadir)"'       \
diff --git a/geometry/Makefile.simple b/geometry/Makefile.simple
index 370b2fa..c3b0106 100644
--- a/geometry/Makefile.simple
+++ b/geometry/Makefile.simple
@@ -46,7 +46,7 @@ MCCP4   = $(HOME)/mapview
 
 
 CXX = g++
-MMDB_CXXFLAGS = -I$(MMDB)/include 
+MMDB_CFLAGS = -I$(MMDB)/include 
 FFTW_LIBS = -L$(FFTW)/lib -lrfftw -lfftw
 
 MMDB_LIBS = -L$(MMDB)/lib -lmmdb 
@@ -55,9 +55,9 @@ CLIPPER_LIBS = -L$(CLIPPER)/lib -lclipper-contrib \
 	-L$(CLIPPER)lib -L/h/paule/mapview/lib $(FFTW_LIBS)
 
 MCCP4_CFLAGS = -I$(MCCP4)/include
-CLIPPER_CXXFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
+CLIPPER_CFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
 
-INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) -I$(top_srcdir)/coords 
+INCLUDES = $(CLIPPER_CFLAGS) $(MMDB_CFLAGS) -I$(top_srcdir)/coords 
 
 top_srcdir = ..
 
diff --git a/high-res/Makefile.am b/high-res/Makefile.am
index cf81182..3a2af5c 100644
--- a/high-res/Makefile.am
+++ b/high-res/Makefile.am
@@ -29,8 +29,8 @@ AM_CXXFLAGS = $(COOT_GSL_FLAGS) $(GSL_CFLAGS)
 
 INCLUDES = \
 	-I$(top_srcdir)   \
-	$(MMDB_CXXFLAGS)  \
-	$(CLIPPER_CXXFLAGS)
+	$(MMDB_CFLAGS)  \
+	$(CLIPPER_CFLAGS)
 
 libcoot_high_res_la_SOURCES = high-res.cc coot-atom-graph.cc residue-distortions.cc \
 	sequence-assignment.cc
diff --git a/ideal/Makefile.am b/ideal/Makefile.am
index be4dc3e..f4cc8eb 100644
--- a/ideal/Makefile.am
+++ b/ideal/Makefile.am
@@ -32,11 +32,11 @@ pkgincludedir = $(includedir)/coot/ideal
 
 INCLUDES = \
 	-I$(top_srcdir) \
-	$(CLIPPER_CXXFLAGS)        \
-	$(MMDB_CXXFLAGS)
+	$(CLIPPER_CFLAGS)        \
+	$(MMDB_CFLAGS)
 
 AM_CXXFLAGS = $(COOT_GSL_FLAGS) $(GSL_CFLAGS) $(COOT_WINDOWS_CFLAGS) \
-     $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) $(COOT_GSL_FLAGS) 
+     $(CLIPPER_CFLAGS) $(MMDB_CFLAGS) $(COOT_GSL_FLAGS) 
 AM_LDFLAGS = $(GSL_LIBS) $(MMDB_LIBS) $(CLIPPER_LIBS)
 
 
diff --git a/ideal/Makefile.simple b/ideal/Makefile.simple
index 3c32c3c..0abd514 100644
--- a/ideal/Makefile.simple
+++ b/ideal/Makefile.simple
@@ -44,7 +44,7 @@ MMDB    = $(HOME)/mapview
 
 
 CXX = g++
-MMDB_CXXFLAGS = -I$(MMDB)/include 
+MMDB_CFLAGS = -I$(MMDB)/include 
 FFTW_LIBS = -L$(FFTW)/lib -lrfftw -lfftw
 
 MMDB_LIBS = -L$(MMDB)/lib -lmmdb 
@@ -53,9 +53,9 @@ CLIPPER_LIBS = -L$(CLIPPER)/lib -lclipper-contrib \
 	-L$(CLIPPER)lib -L/h/paule/mapview/lib $(FFTW_LIBS)
 
 MCCP4_CFLAGS = -I$(MCCP4)/include
-CLIPPER_CXXFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
+CLIPPER_CFLAGS = -I$(CLIPPER)/include -I$(MCCP4)/include 
 
-INCLUDES = $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) \
+INCLUDES = $(CLIPPER_CFLAGS) $(MMDB_CFLAGS) \
 	-I$(top_srcdir)/coords  \
 	-I$(top_srcdir)/geometry
 
diff --git a/lbg/Makefile.am b/lbg/Makefile.am
index df74b90..7f54567 100644
--- a/lbg/Makefile.am
+++ b/lbg/Makefile.am
@@ -6,7 +6,7 @@ INCLUDES = \
         $(GSL_CFLAGS) \
         $(GOOCANVAS_CFLAGS) \
         $(GTK_CFLAGS) \
-	$(CLIPPER_CXXFLAGS) $(CCP4SRS_CXXFLAGS) $(MMDB_CXXFLAGS) \
+	$(CLIPPER_CFLAGS) $(CCP4SRS_CXXFLAGS) $(MMDB_CFLAGS) \
 	$(COOT_WINDOWS_CFLAGS) \
 	$(ENHANCED_LIGAND_TOOLS_CFLAGS)      \
 	$(RDKIT_CXXFLAGS) \
diff --git a/lbg/Makefile.simple b/lbg/Makefile.simple
index 48d091b..5b148b4 100644
--- a/lbg/Makefile.simple
+++ b/lbg/Makefile.simple
@@ -24,11 +24,11 @@ GSL_LDFLAGS=$(CRYST)/lib/libgsl.la  $(CRYST)/lib/libgslcblas.la -lm
 MMDB=$(CRYST)
 # MMDB=$(HOME)/test/mmdb
 
-MMDB_CXXFLAGS=-I$(MMDB)/include/mmdb
+MMDB_CFLAGS=-I$(MMDB)/include/mmdb
 MMDB_LDFLAGS=$(MMDB)/lib/libmmdb.la
 
 CLIPPER=$(CRYST)
-CLIPPER_CXXFLAGS=-I$(CLIPPER)/include
+CLIPPER_CFLAGS=-I$(CLIPPER)/include
 CLIPPER_LDFLAGS=$(CLIPPER)/lib/libclipper-core.la \
         $(CLIPPER)/lib/librfftw.la \
         $(CLIPPER)/lib/libfftw.la
@@ -40,7 +40,7 @@ LIDIA_CORE_CXXFLAGS = -I ../lidia-core
 LIDIA_CORE_LDFLAGS = $(COOT_INSTALLED)/lib/libcoot-lidia-core.la
 
 
-CXXFLAGS=-g -O2 $(EXTRA_FLAGS) $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) $(LIDIA_CORE_CXXFLAGS) \
+CXXFLAGS=-g -O2 $(EXTRA_FLAGS) $(MMDB_CFLAGS) $(CLIPPER_CFLAGS) $(LIDIA_CORE_CXXFLAGS) \
     -I$(top_srcdir)/compat
 
 lbg:  lbg.o wmolecule.o lbg-callbacks.o lbg-search.o some-coot-utils.o residue-circles.o
diff --git a/lidia-core/Makefile.am b/lidia-core/Makefile.am
index 72b28ca..2a184b4 100644
--- a/lidia-core/Makefile.am
+++ b/lidia-core/Makefile.am
@@ -24,13 +24,13 @@ pkgincludedir = $(includedir)/coot/lidia-core
 
 #  20110408 I add a CResidue,dictionary_residue_restraints_t constructor to molfile_molecule_t
 # simple
-#INCLUDES = $(CLIPPER_CXXFLAGS)
+#INCLUDES = $(CLIPPER_CFLAGS)
 #
 INCLUDES = \
 	 -I$(top_srcdir) \
 	$(ENHANCED_LIGAND_TOOLS_CFLAGS)      \
 	$(RDKIT_CXXFLAGS) 		\
-        $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS)
+        $(CLIPPER_CFLAGS) $(MMDB_CFLAGS)
 
 libcoot_lidia_core_la_SOURCES = lbg-shared.cc lbg-molfile.cc lig-build.cc rdkit-interface.cc \
 	cod-types.cc
diff --git a/ligand/Makefile.am b/ligand/Makefile.am
index 202979e..5173204 100644
--- a/ligand/Makefile.am
+++ b/ligand/Makefile.am
@@ -27,7 +27,7 @@ INCLUDES = \
 	-I$(top_srcdir) \
 	$(COOT_GSL_FLAGS)		\
 	$(GSL_CFLAGS)			\
-	$(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) 
+	$(CLIPPER_CFLAGS) $(MMDB_CFLAGS) 
 
 pkginclude_HEADERS = ligand.hh residue_by_phi_psi.hh wligand.hh monomer-utils.hh \
 	chi-angles.hh dunbrack.hh helix-placement.hh fast-ss-search.hh        \
diff --git a/macros/clipper.m4 b/macros/clipper.m4
index da11e74..ad2af81 100644
--- a/macros/clipper.m4
+++ b/macros/clipper.m4
@@ -56,7 +56,7 @@ if test x$clipper_prefix != x; then
  #
  # should use clipper-config --cflags
  #
- CLIPPER_CXXFLAGS="-I$clipper_prefix/include"
+ CLIPPER_CFLAGS="-I$clipper_prefix/include"
  # -I$clipper_prefix/cctbx
  
  # yes, libmmtz.a is in -L$clipper_prefix/umtz!
@@ -75,7 +75,7 @@ if test x$clipper_prefix != x; then
  #
  ccp4c=ccp4c
 
- CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-minimol -lclipper-cns -lclipper-mmdb -lclipper-core -l$ccp4c $MMDB_LIBS -l${fftw_pre}rfftw -l${fftw_pre}fftw -lz -lm"
+ CLIPPER_LIBS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-minimol -lclipper-cns -lclipper-mmdb -lclipper-core -l$ccp4c $MMDB_LIBS -l${fftw_pre}rfftw -l${fftw_pre}fftw -lz -lm"
  # -L$clipper_prefix/boost/lib -lclipper-cctbx -L$clipper_prefix/cctbx/lib -lsgtbx -luctbx 
 
 
@@ -102,8 +102,8 @@ else
  # this needs to be 'configured' - typically either s or blank.
  fftw_pre=
 
- CLIPPER_CXXFLAGS="$CCP4_CFLAGS"
- CLIPPER_LDOPTS="-lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-minimol -lclipper-cns -lclipper-core $CCP4_LIBS $MMDB_LIBS -l${fftw_pre}rfftw -l${fftw_pre}fftw -lz -lm"
+ CLIPPER_CFLAGS="$CCP4_CFLAGS"
+ CLIPPER_LIBS="-lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-minimol -lclipper-cns -lclipper-core $CCP4_LIBS $MMDB_LIBS -l${fftw_pre}rfftw -l${fftw_pre}fftw -lz -lm"
 fi
 
 # BL: workaround needed for new MinGW
@@ -113,17 +113,17 @@ ac_cv_build_alias=${ac_cv_build_alias:=$build_alias}
 case $ac_cv_build_alias in
         # BL says:: same as for cygwin in mingw
         MINGW*|Mingw*|*mingw*|Cygwin*|CYGWIN*|*cygwin*)
-                CLIPPER_LDOPTS=$CLIPPER_LDOPTS
+                CLIPPER_LIBS=$CLIPPER_LIBS
 	;; 
 	*)
-		CLIPPER_LDOPTS=$CLIPPER_LDOPTS" -lpthread"
+		CLIPPER_LIBS=$CLIPPER_LIBS" -lpthread"
         ;;
 esac
 
 AC_MSG_CHECKING([for Clipper])
 
-	LIBS="$save_LIBS $CLIPPER_LDOPTS"
-	CFLAGS="$save_CFLAGS $CLIPPER_CXXFLAGS"
+	LIBS="$save_LIBS $CLIPPER_LIBS"
+	CFLAGS="$save_CFLAGS $CLIPPER_CFLAGS"
 	# AC_TRY_LINK uses the c compiler, so we will temporarily 
 	# reassign $CC to the c++ compiler.
  	#
@@ -140,7 +140,7 @@ if test x$have_clipper = xyes; then
 
  LIBS="$saved_LIBS"
  CFLAGS="$saved_CFLAGS"
- CLIPPER_LIBS="$CLIPPER_LDOPTS"
+ CLIPPER_LIBS="$CLIPPER_LIBS"
  ifelse([$1], , :, [$1])
 
 else
@@ -148,12 +148,12 @@ else
  LIBS="$saved_LIBS"
  CFLAGS="$saved_CFLAGS"
  CLIPPER_LIBS=""
- CLIPPER_CXXFLAGS=""
+ CLIPPER_CFLAGS=""
  ifelse([$2], , :, [$2])
 
 fi
 
-AC_SUBST(CLIPPER_CXXFLAGS)
+AC_SUBST(CLIPPER_CFLAGS)
 AC_SUBST(CLIPPER_LIBS)
 
 ])
diff --git a/macros/clipper.m4.old b/macros/clipper.m4.old
index de8ab65..4c1b418 100644
--- a/macros/clipper.m4.old
+++ b/macros/clipper.m4.old
@@ -34,7 +34,7 @@ if test x$clipper_prefix != x; then
  # should ideally be CLIPPER_CFLAGS="-I$clipper_prefix/include", and the like
  # when clipper and dependencies get installed.
  #  
- CLIPPER_CXXFLAGS="-I$clipper_prefix/mccp4 -I$clipper_prefix/fftw/include -I$clipper_prefix/boost -I$clipper_prefix/include"
+ CLIPPER_CFLAGS="-I$clipper_prefix/mccp4 -I$clipper_prefix/fftw/include -I$clipper_prefix/boost -I$clipper_prefix/include"
 # -I$clipper_prefix/cctbx
  
  # yes, libmmtz.a is in -L$clipper_prefix/umtz!
@@ -45,20 +45,20 @@ if test x$clipper_prefix != x; then
  # HACK! FIXME
  # added lz, we should have proper autoconf check for this.
  #
- CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper -L$clipper_prefix/mccp4 -lmccp4 $MMDB_LIBS -L$clipper_prefix/fftw/lib -lrfftw -lfftw -lz -lm"
+ CLIPPER_LIBS="-L$clipper_prefix/lib -lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper -L$clipper_prefix/mccp4 -lmccp4 $MMDB_LIBS -L$clipper_prefix/fftw/lib -lrfftw -lfftw -lz -lm"
 # -L$clipper_prefix/boost/lib -lclipper-cctbx -L$clipper_prefix/cctbx/lib -lsgtbx -luctbx 
 else
  # the compiler looks in the "standard" places for clipper.  In real life,
  # it would be quite unlikely that clipper would be installed in /usr/include, 
  # /usr/lib etc. so this code will not usually find the right dependencies.
- CLIPPER_CXXFLAGS=""
- CLIPPER_LDOPTS=""
+ CLIPPER_CFLAGS=""
+ CLIPPER_LIBS=""
 fi
 
 AC_MSG_CHECKING([for Clipper])
 
-	LIBS="$save_LIBS $CLIPPER_LDOPTS"
-	CFLAGS="$save_CFLAGS $CLIPPER_CXXFLAGS"
+	LIBS="$save_LIBS $CLIPPER_LIBS"
+	CFLAGS="$save_CFLAGS $CLIPPER_CFLAGS"
 	# AC_TRY_LINK uses the c compiler, so we will temporarily 
 	# reassign $CC to the c++ compiler.
  	#
@@ -72,7 +72,7 @@ if test x$have_clipper = xyes; then
 
  LIBS="$saved_LIBS"
  CFLAGS="$saved_CFLAGS"
- CLIPPER_LIBS="$CLIPPER_LDOPTS"
+ CLIPPER_LIBS="$CLIPPER_LIBS"
  ifelse([$1], , :, [$1])
 
 else
@@ -80,12 +80,12 @@ else
  LIBS="$saved_LIBS"
  CFLAGS="$saved_CFLAGS"
  CLIPPER_LIBS=""
- CLIPPER_CXXFLAGS=""
+ CLIPPER_CFLAGS=""
  ifelse([$2], , :, [$2])
 
 fi
 
-AC_SUBST(CLIPPER_CXXFLAGS)
+AC_SUBST(CLIPPER_CFLAGS)
 AC_SUBST(CLIPPER_LIBS)
 
 ])
diff --git a/macros/goo-canvas.m4 b/macros/goo-canvas.m4
index e2494a5..cf96688 100644
--- a/macros/goo-canvas.m4
+++ b/macros/goo-canvas.m4
@@ -35,7 +35,7 @@ if test x$goocanvas_prefix != x; then
   # SGI compiler CC (CXX=CC) needs -lm to link maths library, but 
   # GCC c++ does not.
   #
-  GOOCANVAS_LDOPTS="-L$goocanvas_prefix/lib -lgoocanvas"
+  GOOCANVAS_LIBS="-L$goocanvas_prefix/lib -lgoocanvas"
 
 else
   # the compiler looks in the "standard" places for GOOCANVAS. (or uses pkgconfig)
@@ -54,15 +54,15 @@ else
     break;;
   esac
   if test -z "${PKG_CONFIG}"; then
-     GOOCANVAS_LDOPTS="-lgoocanvas"
+     GOOCANVAS_LIBS="-lgoocanvas"
   else
-     GOOCANVAS_LDOPTS=`$PKG_CONFIG goocanvas --libs`
+     GOOCANVAS_LIBS=`$PKG_CONFIG goocanvas --libs`
   fi
 fi
 
 AC_MSG_CHECKING([for Goocanvas])
 
-LIBS="$saved_LIBS $GOOCANVAS_LDOPTS $GTK_LIBS"
+LIBS="$saved_LIBS $GOOCANVAS_LIBS $GTK_LIBS"
 CXXFLAGS="$saved_CXXFLAGS $GOOCANVAS_CFLAGS $GTK_CFLAGS"
 
 
@@ -96,7 +96,7 @@ AC_MSG_RESULT($have_goocanvas)
 if test x$have_goocanvas = xyes; then
 
  GOOCANVAS_CFLAGS="$GOOCANVAS_CFLAGS"
- GOOCANVAS_LIBS="$GOOCANVAS_LDOPTS"
+ GOOCANVAS_LIBS="$GOOCANVAS_LIBS"
 
 ifelse([$1], , :, [$1])
 
diff --git a/macros/mmdb-ssm.m4 b/macros/mmdb-ssm.m4
index 25dc502..21646a5 100644
--- a/macros/mmdb-ssm.m4
+++ b/macros/mmdb-ssm.m4
@@ -34,11 +34,11 @@ if test x$with_ssmlib_prefix != x; then
 
    if test -r "$with_ssmlib_prefix/include/ssm/ssm_superpose.h"; then
 
-      MMDBSSM_CXXFLAGS="-DHAVE_SSMLIB"
+      MMDBSSM_CFLAGS="-DHAVE_SSMLIB"
       # AFAICS, ssm does not put its libs in $acl_libdirstem
       # MMDBSSM_LIBS="-L$with_ssmlib_prefix/$acl_libdirstem -lssm"
       MMDBSSM_LIBS="-L$with_ssmlib_prefix/lib -lssm"
-      ac_MMDBSSM_CXXFLAGS="-I$with_ssmlib_prefix/include"
+      ac_MMDBSSM_CFLAGS="-I$with_ssmlib_prefix/include"
 
       # saved_LIBS="$LIBS"
       # saved_CPPFLAGS="$CPPFLAGS"
@@ -50,11 +50,11 @@ if test x$with_ssmlib_prefix != x; then
    else 
       AC_MSG_FAILURE([ --with-ssmlib-prefix specified but include files not found])
    fi
-   MMDBSSM_CXXFLAGS="$MMDBSSM_CXXFLAGS $ac_MMDBSSM_CXXFLAGS"
+   MMDBSSM_CFLAGS="$MMDBSSM_CFLAGS $ac_MMDBSSM_CFLAGS"
   
 else 
 
-   MMDBSSM_CXXFLAGS=""
+   MMDBSSM_CFLAGS=""
    MMDBSSM_LIBS=""
    with_ssmlib_prefix=no 
 
@@ -62,6 +62,6 @@ fi
 
 AC_MSG_RESULT([$with_ssmlib_prefix])
 
-AC_SUBST(MMDBSSM_CXXFLAGS)
+AC_SUBST(MMDBSSM_CFLAGS)
 AC_SUBST(MMDBSSM_LIBS)
 ])
diff --git a/macros/mmdb.m4 b/macros/mmdb.m4
index 7c1cbcf..72613f3 100644
--- a/macros/mmdb.m4
+++ b/macros/mmdb.m4
@@ -55,7 +55,7 @@ if test x$mmdb_prefix != x; then
 	# --with-mmdb=/some/thing
 	#
 
- # should ideally be MMDB_CXXFLAGS="-I$MMDB_prefix/include", and the like
+ # should ideally be MMDB_CFLAGS="-I$MMDB_prefix/include", and the like
  # when MMDB and dependencies get installed - we infact, include both
  # directories.
  #  
@@ -69,7 +69,7 @@ lib/src
 lib/src/mmdb'
 for ac_dir in $ac_mmdb_dirs; do
    if test -r "$mmdb_prefix/$ac_dir/mmdb/mmdb_manager.h"; then
-      ac_MMDB_CXXFLAGS="-I$mmdb_prefix/$ac_dir"
+      ac_MMDB_CFLAGS="-I$mmdb_prefix/$ac_dir"
       break
    fi
 done
@@ -82,14 +82,14 @@ else
  # the compiler looks in the "standard" places for MMDB.  In real life,
  # it would be quite unlikely that MMDB would be installed in /usr/include, 
  # /usr/lib etc. so this code will not usually find the right dependencies.
- ac_MMDB_CXXFLAGS=""
+ ac_MMDB_CFLAGS=""
  ac_MMDB_LDOPTS="-lmmdb -lm"
 fi
 
 AC_MSG_CHECKING([for MMDB])
 
 	LIBS="$save_LIBS $ac_MMDB_LDOPTS"
-	CXXFLAGS="$save_CXXFLAGS $ac_MMDB_CXXFLAGS"
+	CXXFLAGS="$save_CXXFLAGS $ac_MMDB_CFLAGS"
 	#
 	# AC_TRY_LINK uses the c compiler (set by AC_LANG), so we will
 	# temporarily reassign $CC to the c++ compiler.
@@ -118,7 +118,7 @@ if test x$have_mmdb = xyes; then
     CLINKR_FLAG=-DMMDB_WITHOUT_LINKR
  fi
  CXXFLAGS="$saved_CXXFLAGS"
- MMDB_CXXFLAGS="$ac_MMDB_CXXFLAGS $HASH_FLAG $CISPEP_FLAG $CLINKR_FLAG"
+ MMDB_CFLAGS="$ac_MMDB_CFLAGS $HASH_FLAG $CISPEP_FLAG $CLINKR_FLAG"
  MMDB_LIBS="$ac_MMDB_LDOPTS"
 ifelse([$1], , :, [$1])
 
@@ -134,12 +134,12 @@ else
  LIBS="$saved_LIBS"
  CXXFLAGS="$saved_CXXFLAGS"
  MMDB_LIBS=""
- MMDB_CXXFLAGS=""
+ MMDB_CFLAGS=""
  ifelse([$2], , :, [$2])
 
 fi
 
-AC_SUBST(MMDB_CXXFLAGS)
+AC_SUBST(MMDB_CFLAGS)
 AC_SUBST(MMDB_LIBS)
 
 ])
diff --git a/mini-mol/Makefile.am b/mini-mol/Makefile.am
index 2f6e3a3..4865b0b 100644
--- a/mini-mol/Makefile.am
+++ b/mini-mol/Makefile.am
@@ -25,8 +25,8 @@ pkgincludedir = $(includedir)/coot/mini-mol
 
 INCLUDES =  \
 	-I$(top_srcdir) \
-	$(CLIPPER_CXXFLAGS)    \
-	$(MMDB_CXXFLAGS)
+	$(CLIPPER_CFLAGS)    \
+	$(MMDB_CFLAGS)
 
 AM_CXXFLAGS = $(COOT_WINDOWS_CFLAGS)
 
diff --git a/protein_db/Makefile.am b/protein_db/Makefile.am
index 268fcf6..cdd5c9e 100644
--- a/protein_db/Makefile.am
+++ b/protein_db/Makefile.am
@@ -19,7 +19,7 @@
 
 lib_LTLIBRARIES=libcoot-protein-db.la
 
-INCLUDES= $(CLIPPER_CXXFLAGS)  $(MMDB_CXXFLAGS)
+INCLUDES= $(CLIPPER_CFLAGS)  $(MMDB_CFLAGS)
 
 pkginclude_HEADERS=protein_db.h  protein_db_utils.h rna_db.h
 pkgincludedir = $(includedir)/coot/protein_db
diff --git a/sequence-view/Makefile.am b/sequence-view/Makefile.am
index 7af6384..742e6d7 100644
--- a/sequence-view/Makefile.am
+++ b/sequence-view/Makefile.am
@@ -55,9 +55,9 @@ INCLUDES = \
 	-I$(top_srcdir)/analysis	\
 	-I$(top_srcdir)/cootaneer	\
 	-I$(top_srcdir)/src     	\
-	$(CLIPPER_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
-	$(MMDBSSM_CXXFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
+	$(MMDBSSM_CFLAGS) 		\
 	$(GLUT_CFLAGS)			\
 	$(CANVAS_CFLAGS) 		\
 	$(GTKGL_CONTEXT_CFLAGS)		\
diff --git a/skeleton/Makefile.am b/skeleton/Makefile.am
index 54872fe..bef7d0e 100644
--- a/skeleton/Makefile.am
+++ b/skeleton/Makefile.am
@@ -23,7 +23,7 @@ lib_LTLIBRARIES=libcoot-skeleton.la
 
 INCLUDES = \
 	  -I$(top_srcdir) \
-	 $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) 
+	 $(MMDB_CFLAGS) $(CLIPPER_CFLAGS) 
 
 AM_CXXFLAGS = $(COOT_WINDOWS_CFLAGS)
 
diff --git a/src/Makefile.am b/src/Makefile.am
index af5276b..d0ab482 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -195,11 +195,11 @@ INCLUDES = \
         $(GOOCANVAS_CFLAGS)             \
 	@GTK_CFLAGS@ 			\
 	$(GLUT_CFLAGS)			\
-	$(MMDBSSM_CXXFLAGS) 		\
-	$(MMDB_CXXFLAGS) 		\
+	$(MMDBSSM_CFLAGS) 		\
+	$(MMDB_CFLAGS) 		\
 	$(CCP4SRS_CXXFLAGS) 		\
 	$(GSL_CFLAGS)			\
-	$(CLIPPER_CXXFLAGS) 		\
+	$(CLIPPER_CFLAGS) 		\
 	$(PYGTK_CFLAGS)                 \
 	$(GLOB_CFLAGS)                 \
 	$(CURL_CFLAGS)                 
diff --git a/surface/Makefile.am b/surface/Makefile.am
index 3de5953..2461f58 100644
--- a/surface/Makefile.am
+++ b/surface/Makefile.am
@@ -1,5 +1,5 @@
 
-INCLUDES = -I$(top_srcdir)/coot-utils $(MMDB_CXXFLAGS) $(CLIPPER_CXXFLAGS) $(FFTW_CXXFLAGS)
+INCLUDES = -I$(top_srcdir)/coot-utils $(MMDB_CFLAGS) $(CLIPPER_CFLAGS) $(FFTW_CXXFLAGS)
 
 AM_CXXFLAGS = $(OPENMP_CFLAGS)
 
