 lidia-core/Makefile.am |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/lidia-core/Makefile.am b/lidia-core/Makefile.am
index 17d468d..78bf054 100644
--- a/lidia-core/Makefile.am
+++ b/lidia-core/Makefile.am
@@ -34,3 +34,7 @@ libcoot_lidia_core_la_SOURCES = lbg-molfile.cc lig-build.cc
 # test_molfile_SOURCES: test-molfile.cc
 
 # test_molfile_LDADD: libcoot-lidia-core.la
+
+libcoot_lidia_core_la_LIBADD = \
+	$(top_builddir)/geometry/libcoot-geometry.la \
+	$(MMDB_LIBS) $(CLIPPER_LIBS)
