Index: python/a_rapper_gui.py
===================================================================
--- python/a_rapper_gui.py	(revision 2425)
+++ python/a_rapper_gui.py	(working copy)
@@ -20,8 +20,8 @@
     home_dir = os.getenv('COOT_HOME')
 else:
     home_dir = os.getenv('HOME')
-rapper_dir = os.path.join(os.path.normpath(home_dir), "rappermc")
-rapper_command = "rapper"
+rapper_dir = "/usr/share/rapper"
+rapper_command = "rappermc"
 
 def rapper_process():
     #needed?
