Index: lightgbm-4.5.0/lightgbm/basic.py
===================================================================
--- lightgbm-4.5.0.orig/lightgbm/basic.py
+++ lightgbm-4.5.0/lightgbm/basic.py
@@ -261,8 +261,7 @@ def _log_callback(msg: bytes) -> None:
 
 def _load_lib() -> ctypes.CDLL:
     """Load LightGBM library."""
-    lib_path = find_lib_path()
-    lib = ctypes.cdll.LoadLibrary(lib_path[0])
+    lib = ctypes.cdll.LoadLibrary("lib_lightgbm.so")
     lib.LGBM_GetLastError.restype = ctypes.c_char_p
     callback = ctypes.CFUNCTYPE(None, ctypes.c_char_p)
     lib.callback = callback(_log_callback)  # type: ignore[attr-defined]
