diff --git a/Makefile b/Makefile
index ea7a550..1b3a1d2 100644
--- a/Makefile
+++ b/Makefile
@@ -689,6 +689,9 @@ else
 endif
 endif
 
+	$(LINKER) $(SOFLAGS) -o $(BASE_LIB_PATH)/$(DEB_LIBBLAS) $? $(LDFLAGS) -Wl,--soname,$(DEB_LIBBLAS)
+	$(LINKER) $(SOFLAGS) -o $(BASE_LIB_PATH)/$(DEB_LIBCBLAS) $? $(LDFLAGS) -Wl,--soname,$(DEB_LIBCBLAS)
+
 # Local symlink for shared library.
 # NOTE: We use a '.loc' suffix to avoid filename collisions in case this
 # rule is executed concurrently with the install-lib-symlinks rule, which
diff --git a/common.mk b/common.mk
index e73a5d1..2aa8a23 100644
--- a/common.mk
+++ b/common.mk
@@ -565,11 +565,14 @@ ifeq ($(MK_ENABLE_SHARED),yes)
 ifeq ($(MK_ENABLE_STATIC),no)
 LIBBLIS_L      := $(LIBBLIS_SO)
 LIBBLIS_LINK   := $(LIBBLIS_SO_PATH)
+SET_RPATH      ?= yes
 ifeq ($(IS_WIN),no)
+ifeq ($(SET_RPATH),yes)
 # For Linux and OS X: set rpath property of shared object.
 LDFLAGS        += -Wl,-rpath,$(BASE_LIB_PATH)
 endif
 endif
+endif
 # On windows, use the shared library even if static is created.
 ifeq ($(IS_WIN),yes)
 LIBBLIS_L      := $(LIBBLIS_SO)
diff --git a/configure b/configure
index 35d4f1d..93c2cd6 100755
--- a/configure
+++ b/configure
@@ -3055,7 +3055,7 @@ main()
 		enable_aocl_zen='yes'
 		enable_aocl_zen_01=1
 	else
-		enable_aocl_zen = 'no';
+		enable_aocl_zen='no';
 		enable_aocl_zen_01=0;
 	fi
 
