 exodus/cbind/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/exodus/cbind/CMakeLists.txt b/exodus/cbind/CMakeLists.txt
index 7038c0f..7c0d229 100644
--- a/exodus/cbind/CMakeLists.txt
+++ b/exodus/cbind/CMakeLists.txt
@@ -295,9 +295,9 @@ ENDIF(EXODUS_LIBRARY_PROPERTIES)
 
 IF(NOT EXODUS_INSTALL_NO_LIBRARIES)
   INSTALL(TARGETS exoIIv2c
-    RUNTIME DESTINATION lib COMPONENT Runtime
-    LIBRARY DESTINATION lib COMPONENT Runtime
-    ARCHIVE DESTINATION lib COMPONENT Development)
+    RUNTIME DESTINATION ${LIB_INSTALL_DIR} COMPONENT Runtime
+    LIBRARY DESTINATION ${LIB_INSTALL_DIR} COMPONENT Runtime
+    ARCHIVE DESTINATION ${LIB_INSTALL_DIR} COMPONENT Development)
   INSTALL(FILES ${cbind_HEADERS} DESTINATION include COMPONENT Development)
 ENDIF(NOT EXODUS_INSTALL_NO_LIBRARIES)
 
