diff -Naur exodusii-4.98/cbind/CMakeLists.txt source/cbind/CMakeLists.txt
--- exodusii-4.98/cbind/CMakeLists.txt	2010-10-04 08:12:11.000000000 -0600
+++ source/cbind/CMakeLists.txt	2010-11-03 18:23:07.910378566 -0600
@@ -1,14 +1,14 @@
-
 find_path( NETCDF_INCLUDE_DIR netcdf.h
    $ENV{ACCESS}/inc
    $ENV{NETCDF_DIR}/inc
    $ENV{NETCDF_DIR}/libsrc
-)	
+)
 
-find_path( NETCDF_LIBRARY libnetcdf.a
-   $ENV{ACCESS}/lib
-   $ENV{NETCDF_DIR}/lib
-   $ENV{NETCDF_DIR}/libsrc/.libs
+find_library( NETCDF_LIBRARY
+              NAMES netcdf
+              $ENV{ACCESS}/lib
+              $ENV{NETCDF_DIR}/lib
+              $ENV{NETCDF_DIR}/libsrc/.libs
 )
 
 find_program (NETCDF_NCDUMP ncdump
@@ -224,7 +224,9 @@
 
 ADD_LIBRARY(exoIIv2c ${cbind_SRCS})
 
-TARGET_LINK_LIBRARIES(exoIIv2c ${NETCDF_LIBRARY}/libnetcdf.a)
+TARGET_LINK_LIBRARIES( exoIIv2c
+                       ${NETCDF_LIBRARY}
+                     )
 
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(exoIIv2c PROPERTIES ${VTK_LIBRARY_PROPERTIES})
