--- qtiplot.pro.orig	2006-05-25 14:51:40.000000000 +0200
+++ qtiplot.pro	2006-05-25 14:54:42.000000000 +0200
@@ -13,2 +13,2 @@
-INCLUDEPATH       += ../3rdparty/qwt/include
-INCLUDEPATH  += -I /usr/include/qwtplot3d
+INCLUDEPATH += /usr/include/qwt
+INCLUDEPATH += /usr/include/qwtplot3d
@@ -17,2 +17 @@
-LIBS         += ../3rdparty/qwt/lib/libqwt.a
-LIBS         += -L /usr/lib -lgsl -lgslcblas -lqwtplot3d -lz
+LIBS += -L_LIBDIR_ -lgsl -lgslcblas -lqwt -lqwtplot3d
