diff --git a/qtiplot/src/scripting/scripting.pri b/qtiplot/src/scripting/scripting.pri
index b8667e3..2457fc8 100755
--- a/qtiplot/src/scripting/scripting.pri
+++ b/qtiplot/src/scripting/scripting.pri
@@ -80,7 +80,6 @@ contains(SCRIPTING_LANGS, Python) {
              $${SIP_DIR}/sipqtiPythonScript.cpp\
              $${SIP_DIR}/sipqtiPythonScripting.cpp\
              $${SIP_DIR}/sipqtiFolder.cpp\
-             $${SIP_DIR}/sipqtiQList.cpp\
              $${SIP_DIR}/sipqtiFit.cpp \
              $${SIP_DIR}/sipqtiExponentialFit.cpp \
              $${SIP_DIR}/sipqtiTwoExpFit.cpp \
@@ -105,4 +104,14 @@ contains(SCRIPTING_LANGS, Python) {
 			 $${SIP_DIR}/sipqtiCorrelation.cpp \
 			 $${SIP_DIR}/sipqtiConvolution.cpp \
 			 $${SIP_DIR}/sipqtiDeconvolution.cpp \
+
+exists(../../$${SIP_DIR}/sipqtiQList.cpp) {
+  # SIP < 4.9
+  SOURCES += $${SIP_DIR}/sipqtiQList.cpp
+} else {
+  SOURCES += \
+         $${SIP_DIR}/sipqtiQList0101Folder.cpp\
+         $${SIP_DIR}/sipqtiQList0101Graph.cpp\
+         $${SIP_DIR}/sipqtiQList0101MdiSubWindow.cpp\
+}
 }
