--- qtiplot.pro.orig	2007-09-30 00:00:02.000000000 +0700
+++ qtiplot.pro	2007-09-30 01:39:57.000000000 +0700
@@ -5,5 +5,5 @@
 # building without muParser doesn't work yet
 SCRIPTING_LANGS += muParser
-SCRIPTING_LANGS += Python
+#SCRIPTING_LANGS += Python
 
 # a console displaying output of scripts; particularly useful on Windows
@@ -28,9 +28,8 @@
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
+INCLUDEPATH       += /usr/include/qwtplot3d
+INCLUDEPATH       += /usr/include/qwt5
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
+INCLUDEPATH       += /usr/include/gsl
 INCLUDEPATH       += ../3rdparty/zlib123/include
 
@@ -41,19 +40,5 @@
 ##################### Linux (Mac OS X) ######################################
 
-# statically link against libraries in 3rdparty
-LIBS         += ../3rdparty/muparser/lib/libmuparser.a
-LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
-LIBS         += ../3rdparty/qwt/lib/libqwt.a
-LIBS         += ../3rdparty/gsl/lib/libgsl.a
-LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-
-# dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
-
-##################### Windows ###############################################
-
+LIBS += -lgsl -lgslcblas -lqwtplot3d -lqwt -lmuparser
 
 #############################################################################
@@ -89,10 +74,5 @@
 ###################### TRANSLATIONS #########################################
 
-TRANSLATIONS    = translations/qtiplot_de.ts \
-                  translations/qtiplot_es.ts \
-                  translations/qtiplot_fr.ts \
-                  translations/qtiplot_ru.ts \
-                  translations/qtiplot_ja.ts \
-                  translations/qtiplot_sv.ts
+TRANSLATIONS    =
 
 #system(lupdate -verbose qtiplot.pro)
@@ -102,6 +82,5 @@
 
 documentation.files += ../manual/html \
-                       ../README.html \
-                       ../gpl_licence.txt \
+                       ../README.html
 
 ###################### HEADERS ##############################################
@@ -376,10 +355,4 @@
   }
 
-  win32 {
-    INCLUDEPATH += $$system(call python-includepath.py)
-    LIBS        += $$system(call python-libs-win.py)
-    system($$system(call python-sipcmd.py) -c $${SIP_DIR} src/qti.sip)
-  }
-
 ##################### SIP generated files #####################
 
