--- fitRational1.pro.orig	2007-09-28 22:50:01.000000000 +0700
+++ fitRational1.pro	2007-09-28 22:50:55.000000000 +0700
@@ -12,10 +12,6 @@
 DESTDIR           = ../
  
-# statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl
-LIBS         += ../3rdparty/gsl/lib/libgsl.a
-LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-#dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+INCLUDEPATH += /usr/include/gsl
+LIBS         += -lgsl -lgslcblas
 
 target.path=/usr/lib$${libsuff}/qtiplot/plugins
