diff -Nur qtiplot-0.9.7.7.orig/qtiplot/src/matrix/Matrix.h qtiplot-0.9.7.7/qtiplot/src/matrix/Matrix.h
--- qtiplot-0.9.7.7.orig/qtiplot/src/matrix/Matrix.h	2009-09-18 18:16:33.000000000 +0100
+++ qtiplot-0.9.7.7/qtiplot/src/matrix/Matrix.h	2009-09-18 18:39:50.000000000 +0100
@@ -137,6 +137,7 @@
 	void setColorMapType(ColorMapType mapType);
 
 	QwtLinearColorMap colorMap(){return d_color_map;};
+	QwtLinearColorMap *colorMapPointer(){return &d_color_map;};
 	void setColorMap(const QwtLinearColorMap& map);
 	//! Used when restoring from project files
 	void setColorMap(const QStringList& lst);
diff -Nur qtiplot-0.9.7.7.orig/qtiplot/src/plot2D/Spectrogram.h qtiplot-0.9.7.7/qtiplot/src/plot2D/Spectrogram.h
--- qtiplot-0.9.7.7.orig/qtiplot/src/plot2D/Spectrogram.h	2009-09-18 18:16:33.000000000 +0100
+++ qtiplot-0.9.7.7/qtiplot/src/plot2D/Spectrogram.h	2009-09-18 18:40:14.000000000 +0100
@@ -62,6 +62,7 @@
 	void setDefaultColorMap();
 
 	QwtLinearColorMap colorMap(){return color_map;};
+	QwtLinearColorMap *colorMapPointer(){return &color_map;};
 	void setCustomColorMap(const QwtLinearColorMap& map);
 	void updateData(Matrix *m);
 
diff -Nur qtiplot-0.9.7.7.orig/qtiplot/src/plot3D/Graph3D.h qtiplot-0.9.7.7/qtiplot/src/plot3D/Graph3D.h
--- qtiplot-0.9.7.7.orig/qtiplot/src/plot3D/Graph3D.h	2009-09-18 18:16:33.000000000 +0100
+++ qtiplot-0.9.7.7/qtiplot/src/plot3D/Graph3D.h	2009-09-18 18:40:37.000000000 +0100
@@ -274,6 +274,8 @@
 
 	QString colorMapFile(){return d_color_map_file;};
 	QwtLinearColorMap colorMap(){return d_color_map;};
+	QwtLinearColorMap *colorMapPointer(){return &d_color_map;};
+
 
 	static bool openColorMapFile(ColorVector& cv, QString fname);
 
diff -Nur qtiplot-0.9.7.7.orig/qtiplot/src/scripting/qti.sip qtiplot-0.9.7.7/qtiplot/src/scripting/qti.sip
--- qtiplot-0.9.7.7.orig/qtiplot/src/scripting/qti.sip	2009-09-18 18:16:33.000000000 +0100
+++ qtiplot-0.9.7.7/qtiplot/src/scripting/qti.sip	2009-09-18 18:19:15.000000000 +0100
@@ -505,7 +505,7 @@
 	void setDefaultColorMap();
 	void setGrayScale();
 	void setRainbowColorMap();
-	QwtLinearColorMap colorMap();
+	QwtLinearColorMap *colorMapPointer()/PyName=colorMap/;
 	void setColorMap(const QwtLinearColorMap&);
 
 	void resetView();
@@ -993,7 +993,7 @@
     void setDataColors(const QColor&, const QColor&);
 	QString colorMapFile();
     void setDataColorMap(const QString& fileName);
-	QwtLinearColorMap colorMap();
+	QwtLinearColorMap *colorMapPointer()/PyName=colorMap/;
 	void setDataColorMap(const QwtLinearColorMap&);
 
     void changeTransparency(double) /PyName=setOpacity/;
