diff -Nur hdf5-1.8.3.orig/c++/src/H5AbstractDs.cpp hdf5-1.8.3/c++/src/H5AbstractDs.cpp
--- hdf5-1.8.3.orig/c++/src/H5AbstractDs.cpp	2009-04-15 19:10:01.061734089 +0100
+++ hdf5-1.8.3/c++/src/H5AbstractDs.cpp	2009-04-16 10:10:56.263885963 +0100
@@ -85,6 +85,8 @@
       else if (fromClass() == "Attribute")
 	 throw DataTypeIException("Attribute::getTypeClass", "H5Tget_class returns H5T_NO_CLASS");
    }
+
+   return (type_class);
 }
 
 //--------------------------------------------------------------------------
diff -Nur hdf5-1.8.3.orig/hl/src/H5LTanalyze.c hdf5-1.8.3/hl/src/H5LTanalyze.c
--- hdf5-1.8.3.orig/hl/src/H5LTanalyze.c	2009-04-15 19:10:01.221747499 +0100
+++ hdf5-1.8.3/hl/src/H5LTanalyze.c	2009-04-16 10:10:56.267887639 +0100
@@ -2329,7 +2329,7 @@
    return ret;
 }
 
-int H5LTyyerror(char *msg)
+void H5LTyyerror(char *msg)
 {
    printf("ERROR: %s before \"%s\".\n", msg, yytext);
 }
diff -Nur hdf5-1.8.3.orig/hl/src/H5LTanalyze.l hdf5-1.8.3/hl/src/H5LTanalyze.l
--- hdf5-1.8.3.orig/hl/src/H5LTanalyze.l	2009-04-15 19:10:01.221747499 +0100
+++ hdf5-1.8.3/hl/src/H5LTanalyze.l	2009-04-16 10:10:56.267887639 +0100
@@ -181,7 +181,7 @@
    return ret;
 }
 
-int H5LTyyerror(char *msg)
+void H5LTyyerror(char *msg)
 {
    printf("ERROR: %s before \"%s\".\n", msg, yytext);
 }
diff -Nur hdf5-1.8.3.orig/hl/src/H5LTparse.c hdf5-1.8.3/hl/src/H5LTparse.c
--- hdf5-1.8.3.orig/hl/src/H5LTparse.c	2009-04-15 19:10:01.221747499 +0100
+++ hdf5-1.8.3/hl/src/H5LTparse.c	2009-04-16 10:10:56.267887639 +0100
@@ -64,7 +64,7 @@
 #include<hdf5.h>
 
 extern int yylex();
-extern int yyerror(char *);
+extern void yyerror(char *);
 
 #define STACK_SIZE      16
 
diff -Nur hdf5-1.8.3.orig/test/dt_arith.c hdf5-1.8.3/test/dt_arith.c
--- hdf5-1.8.3.orig/test/dt_arith.c	2009-04-15 19:10:01.121739118 +0100
+++ hdf5-1.8.3/test/dt_arith.c	2009-04-16 10:10:56.271887918 +0100
@@ -3408,6 +3408,7 @@
     else if(run_test==TEST_DENORM || run_test==TEST_SPECIAL)
         return 1;
 #endif
+    return 1;
 }
 
 
diff -Nur hdf5-1.8.3.orig/tools/h5dump/h5dump.c hdf5-1.8.3/tools/h5dump/h5dump.c
--- hdf5-1.8.3.orig/tools/h5dump/h5dump.c	2009-04-15 19:10:01.141740794 +0100
+++ hdf5-1.8.3/tools/h5dump/h5dump.c	2009-04-16 10:10:56.275888757 +0100
@@ -602,6 +602,7 @@
  *
  *-------------------------------------------------------------------------
  */
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
diff -Nur hdf5-1.8.3.orig/tools/h5ls/h5ls.c hdf5-1.8.3/tools/h5ls/h5ls.c
--- hdf5-1.8.3.orig/tools/h5ls/h5ls.c	2009-04-15 19:10:01.165741353 +0100
+++ hdf5-1.8.3/tools/h5ls/h5ls.c	2009-04-16 10:10:56.279888477 +0100
@@ -2147,6 +2147,7 @@
  *
  *-------------------------------------------------------------------------
  */
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
diff -Nur hdf5-1.8.3.orig/tools/h5stat/h5stat.c hdf5-1.8.3/tools/h5stat/h5stat.c
--- hdf5-1.8.3.orig/tools/h5stat/h5stat.c	2009-04-15 19:10:01.181742750 +0100
+++ hdf5-1.8.3/tools/h5stat/h5stat.c	2009-04-16 10:10:56.279888477 +0100
@@ -181,6 +181,7 @@
     { NULL, 0, '\0' }
 };
 
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
