--- hdf5-1.8.3.orig/config/linux-gnulibc1	2009-04-15 19:10:01.037732972 +0100
+++ hdf5-1.8.3/config/linux-gnulibc1	2009-04-16 09:58:54.132299149 +0100
@@ -39,8 +39,8 @@
 if test "X-" = "X-$FC"; then
     case $CC_BASENAME in
         gcc*)
-            FC=g95
-            FC_BASENAME=g95
+            FC=gfortran
+            FC_BASENAME=gfortran
             ;;
 	pgcc*)
             FC=pgf90
--- hdf5-1.8.3.orig/src/Makefile.am	2009-04-15 19:10:01.113738280 +0100
+++ hdf5-1.8.3/src/Makefile.am	2009-04-16 11:07:46.221186334 +0100
@@ -27,7 +27,7 @@
 # a long time to compile it with any optimization on.  H5detect is used
 # to generate H5Tinit.c once. So, optimization is not critical.
 noinst_PROGRAMS = H5detect
-H5detect_CFLAGS = -g
+H5detect_CFLAGS = -g -O0
 
 # Our main target, the HDF5 library
 lib_LTLIBRARIES=libhdf5.la
--- hdf5-1.8.3.orig/configure.in	2009-04-15 19:10:01.345757835 +0100
+++ hdf5-1.8.3/configure.in	2009-04-16 12:40:53.698868563 +0100
@@ -355,7 +355,7 @@
   dnl --------------------------------------------------------------------
   dnl Check for a Fortran 9X compiler and how to include modules.
   dnl 
-  AC_PROG_FC([f90 pgf90 slf90 f95 g95 xlf95 efc ifort ftn],)
+  AC_PROG_FC([gfortran f90 pgf90 slf90 f95 g95 xlf95 efc ifort ftn],)
   AC_F9X_MODS
 
   dnl It seems that libtool (as of Libtool 1.5.14) is trying to
