diff -Nur gcb-1.5.6.orig/configure.ac gcb-1.5.6/configure.ac
--- gcb-1.5.6.orig/configure.ac	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/configure.ac	2009-10-23 20:27:13.000000000 +0100
@@ -20,9 +20,7 @@
 AC_PROG_AWK
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_INSTALL
-AC_PROG_LN_S
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 if test x$_cv_uname_sysname = "xIRIX" ; then
   CHECK_SOURCE_STANDARDS_DEFINE([_SGI_SOURCE],[1],
diff -Nur gcb-1.5.6.orig/src/client/Makefile.am gcb-1.5.6/src/client/Makefile.am
--- gcb-1.5.6.orig/src/client/Makefile.am	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/src/client/Makefile.am	2009-10-23 21:19:34.000000000 +0100
@@ -22,12 +22,13 @@
 
 AM_CFLAGS = -Wall
 
-lib_LIBRARIES = libGCB.a
-libGCB_a_SOURCES = accept.c client_util.c dup.c get_GCB_env.c \
+lib_LTLIBRARIES = libGCB.la
+libGCB_la_SOURCES = accept.c client_util.c dup.c get_GCB_env.c \
   inherit.c route.c serialize.c active.c close.c getsockname.c \
   init.c select.c socket.c async_event.c connect.c execv.c \
   getsockopt.c listen.c send.c bind.c data.c heartbeat.c recv.c \
   sendto.c generic_socket.c wrap.c non_block.c query.c
 
-libGCB_a_LIBADD = ../common/log.o ../common/util.o ../common/ex_cmd.o ../common/gcbstring.o
+libGCB_la_LIBADD = ../common/log.lo ../common/util.lo ../common/ex_cmd.lo ../common/gcbstring.lo
+libGCB_la_LDFLAGS = -Wl,--wrap,socket -Wl,--wrap,bind -Wl,--wrap,listen -Wl,--wrap,connect -Wl,--wrap,accept -Wl,--wrap,select -Wl,--wrap,recv -Wl,--wrap,read -Wl,--wrap,recvfrom -Wl,--wrap,send -Wl,--wrap,sendto -Wl,--wrap,write -Wl,--wrap,close -Wl,--wrap,dup -Wl,--wrap,dup2 -Wl,--wrap,getsockname -Wl,--wrap,getsockaddr -Wl,--wrap,getsockopt -Wl,--wrap,execv -Wl,--wrap,execve -Wl,--wrap,execl
 noinst_HEADERS = client.h
diff -Nur gcb-1.5.6.orig/src/common/Makefile.am gcb-1.5.6/src/common/Makefile.am
--- gcb-1.5.6.orig/src/common/Makefile.am	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/src/common/Makefile.am	2009-10-23 20:28:10.000000000 +0100
@@ -24,7 +24,7 @@
 
 ## This library is not being installed on user's site. This is just for
 ## internal use
-noinst_LIBRARIES = libGCBcomm.a
+noinst_LTLIBRARIES = libGCBcomm.la
 
 ## Source files in library
-libGCBcomm_a_SOURCES = log.c util.c ex_cmd.c get_exec_path.c gcbstring.c
+libGCBcomm_la_SOURCES = log.c util.c ex_cmd.c get_exec_path.c gcbstring.c
diff -Nur gcb-1.5.6.orig/src/server/Makefile.am gcb-1.5.6/src/server/Makefile.am
--- gcb-1.5.6.orig/src/server/Makefile.am	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/src/server/Makefile.am	2009-10-23 20:29:58.000000000 +0100
@@ -20,7 +20,7 @@
 ## Makefile.am -- Process this file with automake to produce Makefile.in
 
 INCLUDES = -I$(top_srcdir)/src/include -I$(top_builddir)/src/include
-LDADD = ../common/libGCBcomm.a
+LDADD = ../common/libGCBcomm.la
 AM_CFLAGS = -Wall
 AM_CXXFLAGS = -Wall
 
diff -Nur gcb-1.5.6.orig/src/test/Makefile.am gcb-1.5.6/src/test/Makefile.am
--- gcb-1.5.6.orig/src/test/Makefile.am	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/src/test/Makefile.am	2009-10-23 20:52:17.000000000 +0100
@@ -24,7 +24,7 @@
 AM_CXXFLAGS = -Wall
 
 # Programs to build and install
-noinst_PROGRAMS = log_t GCB_t EchoServer TCPecho get_exec_path_t\
+check_PROGRAMS = log_t GCB_t EchoServer TCPecho get_exec_path_t\
                   tcp_server_1 tcp_client_1\
                   udp_server_1 udp_client_1\
                   tcp_server_2 tcp_client_2\
@@ -35,16 +35,16 @@
 
 # log_t source and lib
 log_t_SOURCES = log_t.c
-log_t_LDADD = ../common/libGCBcomm.a
+log_t_LDADD = ../common/libGCBcomm.la
 
 get_exec_path_t_SOURCES = get_exec_path_t.c
-get_exec_path_t_LDADD = ../common/libGCBcomm.a
+get_exec_path_t_LDADD = ../common/libGCBcomm.la
 
 # RUDP_t source and lib
 #RUDP_t_SOURCES = RUDP_t.C
-#RUDP_t_LDADD = ../common/libGCBcomm.a
+#RUDP_t_LDADD = ../common/libGCBcomm.la
 
-GCB_TEST_LIB = ../client/libGCB.a ../common/libGCBcomm.a
+GCB_TEST_LIB = ../client/libGCB.la ../common/libGCBcomm.la
 
 # GCB_t source and lib
 GCB_t_SOURCES = GCB_t.C
diff -Nur gcb-1.5.6.orig/src/tools/Makefile.am gcb-1.5.6/src/tools/Makefile.am
--- gcb-1.5.6.orig/src/tools/Makefile.am	2009-10-23 20:26:41.000000000 +0100
+++ gcb-1.5.6/src/tools/Makefile.am	2009-10-23 20:30:20.000000000 +0100
@@ -20,7 +20,7 @@
 ## Makefile.am -- Process this file with automake to produce Makefile.in
 
 INCLUDES = -I$(top_srcdir)/src/include -I$(top_builddir)/src/include
-LDADD = ../common/libGCBcomm.a
+LDADD = ../common/libGCBcomm.la
 AM_CFLAGS = -Wall
 AM_CXXFLAGS = -Wall
 
