diff --git a/glusterfsd/src/Makefile.am b/glusterfsd/src/Makefile.am
index 0609179..548ed3a 100644
--- a/glusterfsd/src/Makefile.am
+++ b/glusterfsd/src/Makefile.am
@@ -20,5 +20,6 @@ uninstall-local:
 install-data-local:
 	$(INSTALL) -d -m 755 $(DESTDIR)$(localstatedir)/run
 	$(INSTALL) -d -m 755 $(DESTDIR)$(localstatedir)/log/glusterfs
+	$(INSTALL) -d -m 755 $(DESTDIR)$(sbindir)
 	rm -f $(DESTDIR)$(sbindir)/glusterfs
 	ln -s glusterfsd $(DESTDIR)$(sbindir)/glusterfs
