diff --git a/Makefile.am b/Makefile.am
index 12686b3..d547b12 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,8 +1,9 @@
 #
 
+ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = src emboss_acd emboss_doc
 
-EXTRA_DIST = copyright depcomp ltmain.sh install-sh config.sub config.guess
+EXTRA_DIST = depcomp ltmain.sh install-sh config.sub config.guess
 
 dist-hook:
 	tar cBf - emboss_acd | ( cd $(distdir); tar xBf - ; cd emboss_acd; rm -rf CVS)
diff --git a/configure.in b/configure.in
index 3c69e1b..d78c862 100644
--- a/configure.in
+++ b/configure.in
@@ -187,15 +187,6 @@ AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
 
-dnl Intel MacOSX 10.6 puts X11 in a non-standard place
-if test "${with_x}" != "no" ; then
-if test "`uname -a | grep Darwin`"; then
-OSXX=`sw_vers -productVersion | sed 's/\(10\.[[0-9]]*\).*/\1/'`
-if test ${OSXX} '>' '10.4'; then
-CFLAGS="$CFLAGS -I/usr/X11/include -L/usr/X11/lib"
-fi
-fi
-fi
 
 dnl Check for header files
 AC_PATH_XTRA
@@ -248,10 +239,10 @@ AX_LIB_MYSQL
 AX_LIB_POSTGRESQL
 
 CFLAGS="$CFLAGS $MYSQL_CFLAGS"
-LDFLAGS="$LDFLAGS $MYSQL_LDFLAGS"
+LIBS="${MYSQL_LIBS} ${LIBS}"
 
-CFLAGS="$CFLAGS $POSTGRESQL_CFLAGS"
-LDFLAGS="$LDFLAGS $POSTGRESQL_LDFLAGS"
+CFLAGS="$POSTGRESQL_CFLAGS $CFLAGS"
+LIBS="$POSTGRESQL_LIBS $LIBS"
 
 
 dnl "Export" these variables for PCRE
@@ -388,9 +379,30 @@ AC_ARG_ENABLE(systemlibs,
    AS_HELP_STRING([--enable-systemlibs], [Use system versions of zlib/expat]))
 if test "${enable_systemlibs}" = "yes" ; then
     have_systemlibs=yes
+    DEFINE_USE_SYSTEM_PCRE="#define"
+    PKG_CHECK_MODULES([PLPLOT], [plplotd])
+    PKG_CHECK_MODULES([ZLIB], [zlib])
+else
+    dnl X11 is only used by plplot
+    dnl Intel MacOSX 10.6 puts X11 in a non-standard place
+    if test "${with_x}" != "no" ; then
+    if test "`uname -a | grep Darwin`"; then
+    OSXX=`sw_vers -productVersion | sed 's/\(10\.[[0-9]]*\).*/\1/'`
+    if test ${OSXX} '>' '10.4'; then
+    CFLAGS="$CFLAGS -I/usr/X11/include -L/usr/X11/lib"
+    fi
+    fi
+    fi
+    if test "${with_x}" != "no" ; then
+    LF_EMBOSS_PATH_XLIB
+    fi
+     AC_MSG_NOTICE(USING bundled LIBS)
 fi          
 AM_CONDITIONAL(ESYSTEMLIBS, test "$have_systemlibs" = "yes")
 AC_SUBST(ESYSTEMLIBS)
+AC_SUBST(DEFINE_USE_SYSTEM_PCRE)
+AC_SUBST(PLPLOT_CFLAGS)
+AC_SUBST(PLPLOT_LIBS)
 
 
 
diff --git a/emboss_doc/html/Makefile.am b/emboss_doc/html/Makefile.am
index eb337f3..ea58a27 100644
--- a/emboss_doc/html/Makefile.am
+++ b/emboss_doc/html/Makefile.am
@@ -1,4 +1,4 @@
 pkgdata_DATA = index.html \
 	eiprscan.html
 
-pkgdatadir=$(prefix)/share/EMBOSS/doc/html/embassy/iprscan
+pkgdatadir=$(docdir)/html/embassy/iprscan
diff --git a/emboss_doc/text/Makefile.am b/emboss_doc/text/Makefile.am
index 5af1309..8110248 100644
--- a/emboss_doc/text/Makefile.am
+++ b/emboss_doc/text/Makefile.am
@@ -1,3 +1,3 @@
 pkgdata_DATA =  eiprscan.txt
 
-pkgdatadir=$(prefix)/share/EMBOSS/doc/programs/text
+pkgdatadir=$(docdir)/programs/text
diff --git a/m4/hpdf.m4 b/m4/hpdf.m4
index a37940f..80b329c 100644
--- a/m4/hpdf.m4
+++ b/m4/hpdf.m4
@@ -33,8 +33,8 @@ ALT_HOME=/usr
 #
 # Locate hpdf
 #
-if test -d "${ALT_HOME}"
-then
+#if test -d "${ALT_HOME}"
+#then
 
 #
 # Keep a copy if it fails
@@ -45,13 +45,13 @@ then
 #
 # Set 
 #
-        LDFLAGS="${LDFLAGS} -L${ALT_HOME}/lib"
-        CPPFLAGS="$CPPFLAGS -I$ALT_HOME/include"
+#        LDFLAGS="${LDFLAGS} -L${ALT_HOME}/lib"
+#        CPPFLAGS="$CPPFLAGS -I$ALT_HOME/include"
 
 #
 # Check for libharu in ALT_HOME
 #
-        AC_CHECK_LIB(hpdf, HPDF_New, CHECK=1, CHECK=0, -L${ALT_HOME}/lib)
+        AC_CHECK_LIB(hpdf, HPDF_New, CHECK=1, CHECK=0)
 #
 #
 # If everything found okay then proceed to include png driver in config.
@@ -80,10 +80,4 @@ then
 	  echo "No pdf support (libhpdf) found."
 	fi
 
-else
-        if test $withval != "no"; then
-		echo "Directory $ALT_HOME does not exist"
-		exit 0
-        fi
-fi
 ])
diff --git a/m4/mysql.m4 b/m4/mysql.m4
index 12f8fe9..5cd8b90 100644
--- a/m4/mysql.m4
+++ b/m4/mysql.m4
@@ -26,7 +26,7 @@
 #   This macro calls:
 #
 #     AC_SUBST(MYSQL_CFLAGS)
-#     AC_SUBST(MYSQL_LDFLAGS)
+#     AC_SUBST(MYSQL_LIBS)
 #     AC_SUBST(MYSQL_VERSION)
 #
 #   And sets:
@@ -68,7 +68,7 @@ AC_DEFUN([AX_LIB_MYSQL],
     )
 
     MYSQL_CFLAGS=""
-    MYSQL_LDFLAGS=""
+    MYSQL_LIBS=""
     MYSQL_VERSION=""
 
     dnl
@@ -86,7 +86,7 @@ AC_DEFUN([AX_LIB_MYSQL],
 
 dnl         MYSQL_CFLAGS="`$MYSQL_CONFIG --cflags`"
             MYSQL_CFLAGS="`$MYSQL_CONFIG --include`"
-            MYSQL_LDFLAGS="`$MYSQL_CONFIG --libs`"
+            MYSQL_LIBS="`$MYSQL_CONFIG --libs`"
 
             MYSQL_VERSION=`$MYSQL_CONFIG --version`
 
@@ -97,7 +97,7 @@ dnl not be installed
     	    EMBCFLAGS=$CFLAGS
 	    EMBLDFLAGS=$LDFLAGS
 	    CFLAGS="$MYSQL_CFLAGS $EMBCFLAGS"
-	    LDFLAGS="$MYSQL_LDFLAGS $EMBLDFLAGS"
+        LIBS="${MYSQL_LIBS} ${LIBS}"
 
             AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
                                               #include "mysql.h"]],
@@ -115,7 +115,7 @@ dnl not be installed
                 AC_MSG_RESULT([yes])
             else
 	        MYSQL_CFLAGS=""
-	        MYSQL_LDFLAGS=""
+	        MYSQL_LIBS=""
                 found_mysql="no"
                 AC_MSG_RESULT([no])
             fi
@@ -172,5 +172,5 @@ dnl not be installed
 
     AC_SUBST([MYSQL_VERSION])
     AC_SUBST([MYSQL_CFLAGS])
-    AC_SUBST([MYSQL_LDFLAGS])
+    AC_SUBST([MYSQL_LIBS])
 ])
diff --git a/m4/pngdriver.m4 b/m4/pngdriver.m4
index 5934321..de970ae 100644
--- a/m4/pngdriver.m4
+++ b/m4/pngdriver.m4
@@ -44,8 +44,8 @@ ALT_HOME=/usr
 #
 # Locate png/gd/zlib, if wanted
 #
-if test -d "${ALT_HOME}"
-then
+#if test -d "${ALT_HOME}"
+#then
 
 #
 # Keep a copy if it fails
@@ -69,20 +69,26 @@ then
           fi
 
 
+PKG_CHECK_MODULES([PNG], [libpng])
+PKG_CHECK_MODULES([ZLIB], [zlib])
+AC_CHECK_PROG(GD_LIBS, [gdlib-config])
+GD_LIBS="`gdlib-config --libs` -lgd"
+GD_CFLAGS="`gdlib-config --cflags`"
+
 
 
 
 #
 # Check for zlib in ALT_HOME
 #
-        AC_CHECK_LIB(z, inflateEnd, CHECK=1, CHECK=0, -L${ALT_HOME}/lib -lz)
+        AC_CHECK_LIB(z, inflateEnd, CHECK=1, CHECK=0, ${ZLIB_LIBS})
 #
 
 #
 # Check for png
 #
 	if test $CHECK = "1" ; then
-	  AC_CHECK_LIB(png, png_destroy_read_struct, CHECK=1, CHECK=0 , -L${ALT_HOME}/lib -lz)
+	  AC_CHECK_LIB(png, png_destroy_read_struct, CHECK=1, CHECK=0 , ${ZLIB_LIBS})
 	fi
 	
 
@@ -92,7 +98,7 @@ then
 # Check for gd
 #
 	if test $CHECK = "1"; then
-	  AC_CHECK_LIB(gd, gdImageCreateFromPng, CHECK=1, CHECK=0 , -L${ALT_HOME}/lib -lgd -lpng -lz -lm)
+	  AC_CHECK_LIB(gd, gdImageCreateFromPng, CHECK=1, CHECK=0 , ${PNG_LIBS} ${ZLIB_LIBS} ${GD_LIBS})
           if test $CHECK = "0"; then
 		echo need to upgrade gd for png driver for plplot
 	  fi
@@ -101,7 +107,7 @@ then
 # If everything found okay then proceed to include png driver in config.
 #
 	if test $CHECK = "1" ; then
-	  LIBS="$LIBS -lgd -lpng -lz -lm"
+	  LIBS="$LIBS ${PNG_LIBS} ${ZLIB_LIBS} ${GD_LIBS}"
 
 	  if test $ICCHECK = "1" ; then
 		  LIBS="$LIBS -liconv"
@@ -114,10 +120,11 @@ then
 	  AC_DEFINE(PLD_png)
 	  AM_CONDITIONAL(AMPNG, true)
 	  echo PNG libraries found
-	    if test $ALT_HOME = "/usr" ; then
+#	    if test $ALT_HOME = "/usr" ; then
 		  LDFLAGS="$ALT_LDFLAGS"
 		  CPPFLAGS="$ALT_CPPFLAGS"
-	    fi
+        CFLAGS="${CFLAGS} ${PNG_CFLAGS} ${ZLIB_CFLAGS} ${GD_CFLAGS}"
+#	    fi
 	else
 #
 # If not okay then reset FLAGS.
@@ -134,10 +141,4 @@ then
 #       echo CPPFLAGS = $CPPFLAGS
 
 
-else
-        if test $withval != "no"; then
-		echo "Directory $ALT_HOME does not exist"
-		exit 0
-        fi
-fi
 ])
diff --git a/m4/postgresql.m4 b/m4/postgresql.m4
index e6a804d..ef4d49c 100644
--- a/m4/postgresql.m4
+++ b/m4/postgresql.m4
@@ -26,7 +26,7 @@
 #   This macro calls:
 #
 #     AC_SUBST(POSTGRESQL_CFLAGS)
-#     AC_SUBST(POSTGRESQL_LDFLAGS)
+#     AC_SUBST(POSTGRESQL_LIBS)
 #     AC_SUBST(POSTGRESQL_VERSION)
 #
 #   And sets:
@@ -65,7 +65,7 @@ AC_DEFUN([AX_LIB_POSTGRESQL],
     )
 
     POSTGRESQL_CFLAGS=""
-    POSTGRESQL_LDFLAGS=""
+    POSTGRESQL_LIBS=""
     POSTGRESQL_POSTGRESQL=""
 
     dnl
@@ -82,7 +82,7 @@ AC_DEFUN([AX_LIB_POSTGRESQL],
             AC_MSG_CHECKING([for PostgreSQL libraries])
 
             POSTGRESQL_CFLAGS="-I`$PG_CONFIG --includedir`"
-            POSTGRESQL_LDFLAGS="-L`$PG_CONFIG --libdir` -lpq"
+            POSTGRESQL_LIBS="-L`$PG_CONFIG --libdir` `$PG_CONFIG --libs`"
 
             POSTGRESQL_VERSION=`$PG_CONFIG --version | sed -e 's#PostgreSQL ##'`
 
@@ -144,5 +144,5 @@ AC_DEFUN([AX_LIB_POSTGRESQL],
 
     AC_SUBST([POSTGRESQL_VERSION])
     AC_SUBST([POSTGRESQL_CFLAGS])
-    AC_SUBST([POSTGRESQL_LDFLAGS])
+    AC_SUBST([POSTGRESQL_LIBS])
 ])
diff --git a/src/Makefile.am b/src/Makefile.am
index e6ba236..fae5c0f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -17,9 +17,9 @@ INCLUDES = -I../../../nucleus -I../../../ajax/pcre \
 	-I../../../ajax/ensembl -I../../../ajax/ajaxdb \
 	-I../../../ajax/acd -I../../../plplot
 else
-INCLUDES = -I${embprefix}/include -I${embprefix}/include/eplplot \
-	$(NLINCLUDES) \
-	-I${embprefix}/include/epcre
+INCLUDES = -I${embprefix}/include $(PLPLOT_CFLAGS) \
+	$(NLINCLUDES) -I${embprefix}/include/emboss \
+	-I${embprefix}/include/pcre
 endif
 
 if ISSHARED
@@ -57,6 +57,6 @@ LDADD = ../../../nucleus/libnucleus.la ../../../ajax/acd/libacd.la \
 	../../../plplot/libeplplot.la \
 	$(XLIB)
 else
-LDADD = -L${embprefix}/lib -lnucleus -lacd -lajaxdb -lensembl -lajaxg \
-        -lajax -lepcre $(NLADD) -leplplot $(XLIB)
+LDADD = -lnucleus -lacd -lajaxdb -lensembl -lajaxg \
+        -lajax -lpcre $(NLADD) $(PLPLOT_LIBS) $(XLIB)
 endif
