diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfc3f9b..8fef4af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,8 +52,8 @@ set_target_properties (utf8proc PROPERTIES
 if (UTF8PROC_INSTALL)
   install(TARGETS utf8proc
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
   install(
     FILES
