--- src/util/test/Makefile.in.ori	2012-12-21 14:26:16.000000000 +0100
+++ src/util/test/Makefile.in	2012-12-21 14:26:26.000000000 +0100
@@ -4,10 +4,10 @@
 #################################
 
 APP_PROJ = test_utf8 test_rangemap test_resize_iter \
-           test_logrotate test_regexp test_compress test_checksum \
+           test_logrotate test_regexp test_checksum \
            test_staticmap test_strsearch test_table test_tar \
            test_transmissionrw test_align \
-           test_cache_mt test_id_mux test_compress_mt test_thread_pool \
+           test_cache_mt test_id_mux test_thread_pool \
            test_uttp test_line_reader test_queue_mt test_thread_pool_old \
            test_scheduler test_value_convert example_value_convert \
            formatguess_unit_test test_floating_point_comparison \
