Index: Makefile.test_compress.app
===================================================================
--- src/util/test/Makefile.test_compress.app  (revision 356584)
+++ src/util/test/Makefile.test_compress.app  (working copy)
@@ -3,7 +3,7 @@
 
 APP = test_compress
 SRC = test_compress
-LIB = xutil xcompress $(CMPRS_LIB) xncbi
+LIB = xcompress xutil $(CMPRS_LIB) xncbi
 LIBS = $(CMPRS_LIBS) $(ORIG_LIBS)
 CPPFLAGS = $(ORIG_CPPFLAGS) $(CMPRS_INCLUDE)
 
--- src/util/test/Makefile.test_compress_mt.app.ori	2012-12-21 16:12:11.000000000 +0100
+++ src/util/test/Makefile.test_compress_mt.app	2012-12-21 16:12:41.000000000 +0100
@@ -3,7 +3,7 @@
 
 APP = test_compress_mt
 SRC = test_compress_mt
-LIB = xutil xcompress $(CMPRS_LIB) test_mt xncbi
+LIB = xcompress xutil $(CMPRS_LIB) test_mt xncbi
 LIBS = $(CMPRS_LIBS) $(ORIG_LIBS)
 CPPFLAGS = $(ORIG_CPPFLAGS) $(CMPRS_INCLUDE)
 
