Index: c++.18.0.0/src/app/compart/Makefile.compart.app
===================================================================
--- c++.18.0.0/src/app/compart/Makefile.compart.app	(revision 523253)
+++ c++.18.0.0/src/app/compart/Makefile.compart.app	(working copy)
@@ -5,8 +5,8 @@
 APP = compart
 SRC = compart
 
-LIB =  xalgoalignsplign xalgoalignutil xalgoalignnw xqueryparse \
-       $(BLAST_LIBS:%=%$(STATIC)) \
+LIB =  xalgoalignsplign xalgoalignutil xalgoalignnw xalgoseq taxon1 \
+       $(BLAST_LIBS:%=%$(STATIC)) xqueryparse xregexp $(PCRE_LIB) \
        $(OBJMGR_LIBS:%=%$(STATIC))
 
 LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
Index: c++.18.0.0/src/app/compart/Makefile.compartp.app
===================================================================
--- c++.18.0.0/src/app/compart/Makefile.compartp.app	(revision 523253)
+++ c++.18.0.0/src/app/compart/Makefile.compartp.app	(working copy)
@@ -5,7 +5,8 @@
 APP = compartp
 SRC = compartp
 
-LIB = prosplign  xalgoalignutil $(BLAST_LIBS)  xqueryparse $(OBJMGR_LIBS)
+LIB = prosplign xalgoalignutil xalgoseq taxon1 $(BLAST_LIBS) xqueryparse \
+      xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 
 LIBS = $(CMPRS_LIBS) $(PCRE_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
