--- ncbi_cxx--12_0_0/src/sample/app/cgi/Makefile.cgi_session_sample.app.ori	2014-06-24 21:50:43.000000000 +0200
+++ ncbi_cxx--12_0_0/src/sample/app/cgi/Makefile.cgi_session_sample.app	2014-06-24 21:50:48.000000000 +0200
@@ -11,7 +11,7 @@
 # the lines reading "### BEGIN/END COPIED SETTINGS" in any way.
 
 ### BEGIN COPIED SETTINGS
-LIB = xgridcgi xcgi xhtml xconnserv xconnect xutil xncbi
+LIB = xgridcgi xcgi xhtml xconnserv xthrserv xconnect xutil xncbi
 
 LIBS = $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
