Index: c++.18.0.0/src/app/vecscreen/Makefile.vecscreen.app
===================================================================
--- c++.18.0.0/src/app/vecscreen/Makefile.vecscreen.app	(revision 523253)
+++ c++.18.0.0/src/app/vecscreen/Makefile.vecscreen.app	(working copy)
@@ -4,7 +4,7 @@
 
 APP = vecscreen
 SRC = vecscreen_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = $(LIB_:%=%$(STATIC))
 # FIXME: do we need blast_app_util
 #LIB = blast_app_util $(LIB_:%=%$(STATIC))
