Index: c++.18.0.0/src/app/annotwriter/Makefile.annotwriter.app
===================================================================
--- c++.18.0.0/src/app/annotwriter/Makefile.annotwriter.app	(revision 523253)
+++ c++.18.0.0/src/app/annotwriter/Makefile.annotwriter.app	(working copy)
@@ -8,8 +8,8 @@
 
 APP = annotwriter
 SRC = annotwriter
-LIB = xobjwrite variation_utils $(OBJREAD_LIBS) xalnmgr xobjutil \
-      gbseq entrez2cli entrez2 tables $(OBJMGR_LIBS)
+LIB = xobjwrite $(XFORMAT_LIBS) variation_utils $(OBJREAD_LIBS) xalnmgr \
+      xobjutil entrez2cli entrez2 tables xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 
 LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
