Index: c++.18.0.0/src/app/igblast/Makefile.igblastn.app
===================================================================
--- c++.18.0.0/src/app/igblast/Makefile.igblastn.app	(revision 523253)
+++ c++.18.0.0/src/app/igblast/Makefile.igblastn.app	(working copy)
@@ -2,7 +2,8 @@
 
 APP = igblastn
 SRC = igblastn_app
-LIB_ = xalgoalignutil $(BLAST_INPUT_LIBS) xqueryparse $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = xalgoalignutil $(BLAST_INPUT_LIBS) xqueryparse $(BLAST_LIBS) \
+	xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util igblast $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/igblast/Makefile.igblastp.app
===================================================================
--- c++.18.0.0/src/app/igblast/Makefile.igblastp.app	(revision 523253)
+++ c++.18.0.0/src/app/igblast/Makefile.igblastp.app	(working copy)
@@ -2,7 +2,8 @@
 
 APP = igblastp
 SRC = igblastp_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = xalgoalignutil $(BLAST_INPUT_LIBS) xqueryparse $(BLAST_LIBS) \
+	xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util igblast $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
