Index: c++.18.0.0/src/app/blast/Makefile.blast_formatter.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.blast_formatter.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.blast_formatter.app	(working copy)
@@ -2,7 +2,7 @@
 
 APP = blast_formatter
 SRC = blast_formatter
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.blastp.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.blastp.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.blastp.app	(working copy)
@@ -4,7 +4,7 @@
 
 APP = blastp
 SRC = blastp_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.blastx.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.blastx.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.blastx.app	(working copy)
@@ -4,7 +4,7 @@
 
 APP = blastx
 SRC = blastx_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.deltablast.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.deltablast.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.deltablast.app	(working copy)
@@ -1,7 +1,7 @@
 
 APP = deltablast
 SRC = deltablast_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.psiblast.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.psiblast.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.psiblast.app	(working copy)
@@ -2,7 +2,7 @@
 
 APP = psiblast
 SRC = psiblast_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.rpsblast.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.rpsblast.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.rpsblast.app	(working copy)
@@ -2,7 +2,7 @@
 
 APP = rpsblast
 SRC = rpsblast_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.rpstblastn.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.rpstblastn.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.rpstblastn.app	(working copy)
@@ -2,7 +2,7 @@
 
 APP = rpstblastn
 SRC = rpstblastn_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.seedtop.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.seedtop.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.seedtop.app	(working copy)
@@ -2,7 +2,7 @@
 
 APP = seedtop
 SRC = seedtop_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.tblastn.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.tblastn.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.tblastn.app	(working copy)
@@ -4,7 +4,7 @@
 
 APP = tblastn
 SRC = tblastn_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
Index: c++.18.0.0/src/app/blast/Makefile.tblastx.app
===================================================================
--- c++.18.0.0/src/app/blast/Makefile.tblastx.app	(revision 523253)
+++ c++.18.0.0/src/app/blast/Makefile.tblastx.app	(working copy)
@@ -4,7 +4,7 @@
 
 APP = tblastx
 SRC = tblastx_app
-LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = blast_app_util $(LIB_:%=%$(STATIC))
 
 # De-universalize Mac builds to work around a PPC toolchain limitation
