Index: c++.18.0.0/src/sample/app/blast/Makefile.blast_sample.app
===================================================================
--- c++.18.0.0/src/sample/app/blast/Makefile.blast_sample.app	(revision 523253)
+++ c++.18.0.0/src/sample/app/blast/Makefile.blast_sample.app	(working copy)
@@ -12,7 +12,7 @@
 # the lines reading "### BEGIN/END COPIED SETTINGS" in any way.
 
 ### BEGIN COPIED SETTINGS
-LIB_ = xobjsimple $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = xobjsimple $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = $(LIB_:%=%$(STATIC))
 LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
Index: c++.18.0.0/src/sample/app/blast/Makefile.vsrun_sample.app
===================================================================
--- c++.18.0.0/src/sample/app/blast/Makefile.vsrun_sample.app	(revision 523253)
+++ c++.18.0.0/src/sample/app/blast/Makefile.vsrun_sample.app	(working copy)
@@ -12,7 +12,7 @@
 # the lines reading "### BEGIN/END COPIED SETTINGS" in any way.
 
 ### BEGIN COPIED SETTINGS
-LIB_ = xobjsimple $(BLAST_INPUT_LIBS) $(BLAST_LIBS) $(OBJMGR_LIBS)
+LIB_ = xobjsimple $(BLAST_INPUT_LIBS) $(BLAST_LIBS) xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 LIB = $(LIB_:%=%$(STATIC))
 LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
