Index: c++.18.0.0/src/app/hfilter/Makefile.hfilter.app
===================================================================
--- c++.18.0.0/src/app/hfilter/Makefile.hfilter.app	(revision 523253)
+++ c++.18.0.0/src/app/hfilter/Makefile.hfilter.app	(working copy)
@@ -5,7 +5,8 @@
 APP = hfilter
 SRC = hitfilter_app
 
-LIB = xalgoalignutil $(BLAST_LIBS) xqueryparse $(OBJMGR_LIBS)
+LIB = xalgoalignutil xalgoseq taxon1 $(BLAST_LIBS) \
+	xqueryparse xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 
 LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
