--- src/app/agpconvert/Makefile.agpconvert.app.ori	2013-09-13 23:56:22.660032261 +0200
+++ src/app/agpconvert/Makefile.agpconvert.app	2013-09-13 23:57:31.340032441 +0200
@@ -6,13 +6,12 @@
 APP = agpconvert
 SRC = agpconvert
 
-LIB  = xalgoseq xobjedit $(OBJREAD_LIBS) taxon1 xalnmgr xobjutil submit \
-    ncbi_xdbapi_ftds $(FTDS_LIB) tables xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
-LIBS = $(FTDS_LIBS) $(CMPRS_LIBS) $(PCRE_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
+LIB  = $(OBJREAD_LIBS) taxon1 xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
+LIBS = $(CMPRS_LIBS) $(PCRE_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
 
 CXXFLAGS = $(FAST_CXXFLAGS)
 LDFLAGS  = $(FAST_LDFLAGS)
 
-REQUIRES = objects algo
+REQUIRES = objects
 
 WATCHERS = xiangcha
