--- ncbi_cxx--18_0_0/src/app/hfilter/Makefile.hfilter.app.ori	2017-03-23 19:47:10.204018194 +0100
+++ ncbi_cxx--18_0_0/src/app/hfilter/Makefile.hfilter.app	2017-03-23 19:51:51.601108063 +0100
@@ -5,9 +5,10 @@
 APP = hfilter
 SRC = hitfilter_app
 
-LIB = xalgoalignutil $(BLAST_LIBS) xqueryparse $(OBJMGR_LIBS)
+LIB = xalgoalignutil xalgoseq taxon1 $(BLAST_LIBS) \
+	xqueryparse xregexp $(PCRE_LIB) $(OBJMGR_LIBS)
 
-LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(ORIG_LIBS)
+LIBS = $(CMPRS_LIBS) $(NETWORK_LIBS) $(DL_LIBS) $(PCRE_LIBS) $(ORIG_LIBS)
 
 CXXFLAGS = $(FAST_CXXFLAGS)
 LDFLAGS  = $(FAST_LDFLAGS)
