diff --git a/Makefile.am b/Makefile.am
index 1778219..59c4e5a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,9 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = mmdb.pc
+
 lib_LTLIBRARIES = mmdb/libmmdb.la
 
 pkginclude_HEADERS = \
diff --git a/configure.ac b/configure.ac
index 7e75afb..751b2e5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,4 +9,10 @@ AM_INIT_AUTOMAKE([1.11 foreign subdir-objects silent-rules -Wall])
 LT_INIT([disable-shared])
 AC_PROG_CXX
 AC_SEARCH_LIBS([cos], [m], [], [AC_MSG_ERROR([need math library])])
-AC_OUTPUT(Makefile)
+
+AC_CONFIG_FILES([
+	Makefile 
+	mmdb.pc
+	])
+
+AC_OUTPUT
\ No newline at end of file
diff --git a/mmdb.pc.in b/mmdb.pc.in
new file mode 100644
index 0000000..6156025
--- /dev/null
+++ b/mmdb.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mmdb
+Description: Macromolecular coordinate library
+Version: @VERSION@
+Requires:
+Conflicts:
+Libs: -L${libdir} -lmmdb
+Cflags: -I${includedir}
