--- libscigraphica-2.1.1/configure.in	2005-06-25 02:38:20.000000000 +0100
+++ libscigraphica-2.1.1.new/configure.in	2006-06-08 15:41:52.000000000 +0100
@@ -324,9 +324,9 @@
                   PY_CFLAGS="-I$PY_PREFIX/include/python$PY_VERSION"
 		  PY_LIBS="python$PY_VERSION"
 		  PY_LIB_LOC="$PY_EXEC_PREFIX/lib/python$PY_VERSION/"
-                  array_loc=`find $python_numeric_path -name arrayobject.h | sed s/arrayobject\.h//`
+                  array_loc=`find $python_numeric_path -name arrayobject.h | sed s/arrayobject\.h// | head -n 1`
                   if test x$array_loc = x; then 
-                     array_loc=`find $PY_PREFIX/ -name arrayobject.h | sed s/arrayobject\.h//`
+                     array_loc=`find $PY_PREFIX/ -name arrayobject.h | sed s/arrayobject\.h// | head -n 1`
                   fi
                   if test -n $array_loc -a -d $array_loc; then
                      echo "$array_loc"
