Index: dhcp-3.0.2/Makefile
===================================================================
--- dhcp-3.0.2.orig/Makefile
+++ dhcp-3.0.2/Makefile
@@ -33,7 +33,7 @@ all:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make all); \
+	   (cd work.$$sysname; $(MAKE) all); \
 	fi
 
 install:
@@ -41,7 +41,7 @@ install:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make install); \
+	   (cd work.$$sysname; $(MAKE) install); \
 	fi
 
 depend:
@@ -49,7 +49,7 @@ depend:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make depend); \
+	   (cd work.$$sysname; $(MAKE) depend); \
 	fi
 
 clean:
@@ -57,7 +57,7 @@ clean:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make clean); \
+	   (cd work.$$sysname; $(MAKE) clean); \
 	fi
 
 realclean:
@@ -65,7 +65,7 @@ realclean:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make realclean); \
+	   (cd work.$$sysname; $(MAKE) realclean); \
 	fi
 
 distclean:
@@ -73,7 +73,7 @@ distclean:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make distclean); \
+	   (cd work.$$sysname; $(MAKE) distclean); \
 	fi
 
 links:
@@ -81,6 +81,6 @@ links:
 	 if [ ! -d work.$$sysname ]; then \
 	   echo No build directory for $$sysname - please run ./configure.; \
 	else \
-	   (cd work.$$sysname; make links); \
+	   (cd work.$$sysname; $(MAKE) links); \
 	fi
 
