diff -ur common.orig/bpf.c common/bpf.c
--- common.orig/bpf.c	2006-09-20 12:52:47 +0100
+++ common/bpf.c	2006-09-20 12:53:22 +0100
@@ -354,10 +354,6 @@
 	int result;
 	int fudge;
 
-	if (!strcmp (interface -> name, "fallback"))
-		return send_fallback (interface, packet, raw,
-				      len, from, to, hto);
-
 	/* Assemble the headers... */
 	assemble_hw_header (interface, (unsigned char *)hw, &hbufp, hto);
 	assemble_udp_ip_header (interface,
@@ -531,17 +527,5 @@
 
 void maybe_setup_fallback ()
 {
-	isc_result_t status;
-	struct interface_info *fbi = (struct interface_info *)0;
-	if (setup_fallback (&fbi, MDL)) {
-		if_register_fallback (fbi);
-		status = omapi_register_io_object ((omapi_object_t *)fbi,
-						   if_readsocket, 0,
-						   fallback_discard, 0, 0);
-		if (status != ISC_R_SUCCESS)
-			log_fatal ("Can't register I/O handle for %s: %s",
-				   fbi -> name, isc_result_totext (status));
-		interface_dereference (&fbi, MDL);
-	}
 }
 #endif
diff -ur includes.orig/osdep.h includes/osdep.h
--- includes.orig/osdep.h	2006-09-20 12:54:52 +0100
+++ includes/osdep.h	2006-09-20 12:56:40 +0100
@@ -190,7 +190,7 @@
    Currently, all low-level packet interfaces use BSD sockets as a
    fallback. */
 
-#if defined (USE_BPF_SEND) || defined (USE_NIT_SEND) || \
+#if defined (USE_NIT_SEND) || \
     defined (USE_DLPI_SEND) || defined (USE_UPF_SEND) || \
     defined (USE_LPF_SEND) || \
     (defined (USE_SOCKET_SEND) && defined (HAVE_SO_BINDTODEVICE))
