Fix pc file
--- a/fortrancl.pc.in
+++ b/fortrancl.pc.in
@@ -9,25 +9,3 @@
 URL: @PACKAGE_URL@
 Libs: -L${libdir} -lfortrancl
 Cflags: -I${includedir}/fortrancl
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-
-Name: @PACKAGE_NAME@
-Description: OpenCL interface for Fortran 90
-Version: @PACKAGE_VERSION@
-URL: @PACKAGE_URL@
-Libs: -L${libdir} -lfortrancl
-Cflags: -I${includedir}/fortrancl
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-
-Name: @PACKAGE_NAME@
-Description: OpenCL interface for Fortran 90
-Version: @PACKAGE_VERSION@
-URL: @PACKAGE_URL@
-Libs: -L${libdir} -lfortrancl
-Cflags: -I${includedir}/fortrancl
