diff -Naurp libanomaly-0.1-orig/include/anomaly.h libanomaly-0.1-ptch/include/anomaly.h
--- libanomaly-0.1-orig/include/anomaly.h	2003-10-07 23:29:46.000000000 +0000
+++ libanomaly-0.1-ptch/include/anomaly.h	2009-12-12 21:17:38.000000000 +0000
@@ -31,10 +31,17 @@
 #include <set>
 #include <map>
 #include <string>
-#if defined(__GNUC__) && __GNUC__ == 3
+#ifdef __GNUC__
+#if __GNUC__ < 3
+#include <hash_map>
+#elif __GNUC__ < 4
 #include <ext/hash_map>
 using namespace __gnu_cxx;
 #else
+#include <ext/hash_map>
+using namespace __gnu_cxx;
+#endif
+#else
 #include <hash_map>
 #endif
 
diff -Naurp libanomaly-0.1-orig/include/libAnomaly/item.h libanomaly-0.1-ptch/include/libAnomaly/item.h
--- libanomaly-0.1-orig/include/libAnomaly/item.h	2003-03-25 05:44:14.000000000 +0000
+++ libanomaly-0.1-ptch/include/libAnomaly/item.h	2009-12-12 21:18:12.000000000 +0000
@@ -136,7 +136,7 @@ namespace std {
 
 
 
-#if defined(__GNUC__) && __GNUC__ == 3
+#if defined(__GNUC__) && __GNUC__ > 2
 namespace __gnu_cxx {
 #else
 namespace std {
@@ -307,7 +307,7 @@ namespace std {
    * This structure is used as a function object to encapsulate the
    * less-than operation for the STL hashtable
    */
-  template<> struct std::greater<NumericalItem*>
+  template<> struct greater<NumericalItem*>
     {
     /** The function operator invokes a comparison between two items 
      * @param left: first operand in less-than operation
