--- a/background.c
+++ b/background.c
@@ -28,6 +28,9 @@
 #include <ctype.h>
 #define MIN(a, b) (a > b ? b : a)
 
+extern int *seed;
+extern int nuc_code[255];
+
 /*
  
  Collects length and nucleotide composition of the input reads. Simultaneously counts how many sequences are covered by x,x+1 ... percent library sequences
--- a/detect.h
+++ b/detect.h
@@ -70,11 +70,11 @@ struct stats_dat{
 	int *p_counts;
 };
 
-int nuc_code[255]; 
+extern int nuc_code[255]; 
 
-int *seed;
+extern int *seed;
 
-int linewrap;
+extern int linewrap;
 
 void init_nuc_code();
 
--- a/input.c
+++ b/input.c
@@ -28,6 +28,10 @@
 #include <ctype.h>
 
 static int numseq = 0;
+extern int linewrap;
+extern int *seed;
+extern int nuc_code[255];
+
 
 
 int compare(const void* a, const void* b)
--- a/interface.c
+++ b/interface.c
@@ -33,6 +33,8 @@
 #define OPT_FASTA 5
 #define OPT_MODEL 6
 
+extern int linewrap;
+
 struct parameters* interface(struct parameters* param,int argc, char **argv)
 {
 	int c;
--- a/main.c
+++ b/main.c
@@ -27,6 +27,10 @@
 #include "detect.h"
 #include "sys/time.h"
 
+int linewrap;
+int *seed;
+int nuc_code[255];
+
 
 int main (int argc, char * argv[]) {
 	struct parameters* param = 0;
--- a/test_sequence.c
+++ b/test_sequence.c
@@ -26,6 +26,8 @@
 
 #include "detect.h"
 
+extern int *seed;
+
 /* 
  Checks if a sequence is covered by library sequences. Return the fraction of residues covered by library sequences.  
 */
