diff -ru InsightToolkit-4.4.0/Modules/Compatibility/V3Compatibility/include/itkImageToVectorImageFilter.hxx InsightToolkit-4.4.0.new/Modules/Compatibility/V3Compatibility/include/itkImageToVectorImageFilter.hxx
--- InsightToolkit-4.4.0/Modules/Compatibility/V3Compatibility/include/itkImageToVectorImageFilter.hxx	2013-05-29 22:23:15.000000000 +0200
+++ InsightToolkit-4.4.0.new/Modules/Compatibility/V3Compatibility/include/itkImageToVectorImageFilter.hxx	2013-06-09 22:04:35.157129143 +0200
@@ -61,7 +61,7 @@
 
   for ( unsigned int i = 0; i < numberOfInputs; i++ )
     {
-    InputImageType *input = this->Superclass::GetInput(i);
+    const InputImageType *input = this->Superclass::GetInput(i);
     if ( !input )
       {
       itkExceptionMacro(<< "Input " << i << " not set!");
