diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Base.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Base.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Base.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Base.cxx	2008-08-22 19:57:13.000000000 +0200
@@ -74,7 +74,7 @@
 ostream & GraphBase::Base::AnyValue( const CORBA::Any & anAny ) {
   switch (anAny.type()->kind()) {
     case CORBA::tk_string: {
-      char * t;
+      const char * t;
       anAny >>= t;
       *_fdebug << " " << t << " (tk_string)" << endl ;
       break;
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_DataPort.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_DataPort.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_DataPort.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_DataPort.cxx	2008-08-22 21:55:22.000000000 +0200
@@ -151,7 +151,7 @@
 #endif
   switch (aDataValue->type()->kind()) { // Input Value
   case CORBA::tk_string: { // Input string Value
-    char * t;
+    const char * t;
     *aDataValue >>= t;
 #if ValueTrace
     cdebug << t << " (string)" ;
@@ -242,7 +242,7 @@
         char t[40] ;
         sprintf( t , "%ld" , ll ) ;
         *theValue <<=  t ;
-        char *tt ;
+        const char *tt ;
         *theValue >>= tt ;
 #if ValueTrace
         cdebug << " --> Value( " << t << ") (string) kind " << theValue->type()->kind() ;
@@ -311,7 +311,7 @@
         char t[40] ;
         sprintf( t , "%lf" , d ) ;
         *theValue <<=  t ;
-        char *tt ;
+        const char *tt ;
         *theValue >>= tt ;
 #if ValueTrace
         cdebug << " --> Value( " << t << ") (string) kind " << theValue->type()->kind() ;
@@ -473,7 +473,7 @@
 //JR 30.03.2005    switch ( theValue->type()->kind() ) {
     switch ( theValue.type()->kind() ) {
     case CORBA::tk_string: {
-      char * t;
+      const char * t;
 //JR 30.03.2005      *theValue >>= t;
       theValue >>= t;
 //      cdebug << "GraphBase::DataPort::Value() : " << t << " (string) " << endl ;
@@ -583,7 +583,7 @@
     const CORBA::Any * theValue = *_Value ;
     switch (theValue->type()->kind()) {
     case CORBA::tk_string:
-      char * t;
+      const char * t;
       *theValue >>= t;
       f << t << " (string)" ;
       break;
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Graph.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Graph.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Graph.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_Graph.cxx	2008-08-22 22:29:22.000000000 +0200
@@ -1598,7 +1598,7 @@
   bool isUnknown = false;
   switch (aValue.type()->kind()) {
     case CORBA::tk_string:
-      char * t;
+      const char * t;
       aValue >>= t;
       cdebug << t << " (string) " ;
       if ( !strcmp( t, "Unknown CORBA::Any Type" ) ) 
@@ -1688,7 +1688,7 @@
             << ToServiceParameterName << "' , Any " ;
   switch (aValue.type()->kind()) {
     case CORBA::tk_string:
-      char * t;
+      const char * t;
       aValue >>= t;
       cdebug << t << " (string) " ;
       break;
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_XmlHandler.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_XmlHandler.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_XmlHandler.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphBase/DataFlowBase_XmlHandler.cxx	2008-08-22 22:00:15.000000000 +0200
@@ -718,7 +718,7 @@
             case CORBA::tk_string: {
 //Mandrake 10.1/Salome 3              aLink.aLinkValue <<= (char *) NULL ;
               aLink.aLinkValue <<= (char *) "" ;
-              char * t;
+              const char * t;
               aLink.aLinkValue >>= t;
 #if TRACE
               MESSAGE( t << " (string)" );
@@ -778,7 +778,7 @@
               switch ( aLink.aLinkValue.type()->kind() ) {
               case CORBA::tk_string: {
                 aLink.aLinkValue <<= fieldvalue[depth].c_str() ;
-                char * t;
+                const char * t;
                 aLink.aLinkValue >>= t;
 #if TRACE
                 MESSAGE( t << " (string)" );
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphEditor/DataFlowEditor_OutNode.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphEditor/DataFlowEditor_OutNode.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphEditor/DataFlowEditor_OutNode.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphEditor/DataFlowEditor_OutNode.cxx	2008-08-22 22:44:27.000000000 +0200
@@ -1200,7 +1200,7 @@
     valuetype.appendChild( aField ) ;
     switch (aLink.aLinkValue.type()->kind()) {
       case CORBA::tk_string: {
-        char* retstr ;
+        const char* retstr ;
         aLink.aLinkValue >>= retstr;
 //        f << Tabs << "	<value>" << retstr << "</value>" << endl ;
         QDomElement value = Graph.createElement( "value" ) ;
@@ -1345,7 +1345,7 @@
       << ".Input( " ;
     switch (aLink.aLinkValue.type()->kind()) {
       case CORBA::tk_string: {
-        char* retstr ;
+        const char* retstr ;
         aLink.aLinkValue >>= retstr;
         f << "'" << retstr << "'" ;
         break ;
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_InNodeThreads.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_InNodeThreads.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_InNodeThreads.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_InNodeThreads.cxx	2008-08-22 22:59:18.000000000 +0200
@@ -3277,7 +3277,7 @@
       const char * Type = anInPort->GetServicesParameter().Parametertype ;
       switch ( D.Value.type()->kind() ) { // { string , long , double , objref }
       case CORBA::tk_string:
-        char * t;
+        const char * t;
         D.Value >>= t;
 #if InParametersSetTrace
         cdebug << t << " (string)" ;
@@ -3675,7 +3675,7 @@
 #if InParametersSetTrace
     switch (D.Value.type()->kind()) { // { string , long , double , objref }
     case CORBA::tk_string:
-      char * t;
+      const char * t;
       D.Value >>= t;
       cdebug << ThreadNo() << " " << t << "(string)" << endl ;
       break;
@@ -3764,7 +3764,7 @@
         ServicesAnyData D = OutParametersList[i] ;
         switch (D.Value.type()->kind()) { // { string , long , double , objref }
         case CORBA::tk_string: {
-          char * t;
+          const char * t;
           D.Value >>= t;
 #if OutParametersSetTrace
           cdebug << ThreadNo() << " " << t << "(string)" << endl ;
@@ -3999,7 +3999,7 @@
 #if OutParametersSetTrace
         switch ( anOutPort->Value().type()->kind() ) {
         case CORBA::tk_string:
-          char * t;
+          const char * t;
           (anOutPort->Value()) >>= t;
           cdebug << ThreadNo() << " Out" << i << " : " << t << "(string)" << endl ;
           break;
diff -abur src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_PyDynInvoke.cxx src3.2.6/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_PyDynInvoke.cxx
--- src3.2.6_orig/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_PyDynInvoke.cxx	2008-08-22 19:55:15.000000000 +0200
+++ src3.2.6/SUPERV_SRC_3.2.6/src/GraphExecutor/DataFlowExecutor_PyDynInvoke.cxx	2008-08-22 22:48:17.000000000 +0200
@@ -320,7 +320,7 @@
       sname = InParametersList[i].Name.c_str() ;
       switch ( data.type()->kind() ) {
       case CORBA::tk_string : {
-        char * t ;
+        const char * t ;
         data >>= t ;
         PyObject * ArgValue = Py_BuildValue( "s" , t ) ;
         PyTuple_SetItem( ArgsList , i , ArgValue ) ;
