--- CMakeLists.txt.orig	2010-07-02 23:31:56.663962398 +1200
+++ CMakeLists.txt	2010-07-09 21:49:29.056895751 +1200
@@ -46,6 +46,7 @@
 
 option(ENABLE_ZLIB "Enable compression using ZLIB" ON)
 option(ENABLE_PNG "Enable PNG support" ON)
+option(ENABLE_DOC "Build the PDF documentation" OFF)
 
 set(GL2PS_MAJOR_VERSION 1)
 set(GL2PS_MINOR_VERSION 3)
@@ -113,19 +114,18 @@
     set_target_properties(shared PROPERTIES COMPILE_FLAGS "-DGL2PSDLL -DGL2PSDLL_EXPORTS")
   endif(MSVC)
 
-  install(TARGETS lib shared DESTINATION lib)
+  install(TARGETS lib shared DESTINATION lib${LIB_SUFFIX})
 endif(OPENGL_FOUND)
 
 if(WIN32)
   set(GL2PS_DOC .)
 else(WIN32)
-  set(GL2PS_DOC share/doc/gl2ps)
+  set(GL2PS_DOC share/doc/gl2ps-${GL2PS_VERSION})
 endif(WIN32)
 
 install(FILES gl2ps.h DESTINATION include)
 install(FILES ${CMAKE_SOURCE_DIR}/README.txt DESTINATION ${GL2PS_DOC})
-install(FILES ${CMAKE_SOURCE_DIR}/COPYING.LGPL DESTINATION ${GL2PS_DOC})
-install(FILES ${CMAKE_SOURCE_DIR}/COPYING.GL2PS DESTINATION ${GL2PS_DOC})
+install(FILES ${CMAKE_SOURCE_DIR}/TODO.txt DESTINATION ${GL2PS_DOC})
 install(FILES ${CMAKE_SOURCE_DIR}/gl2psTest.c DESTINATION ${GL2PS_DOC})
 install(FILES ${CMAKE_SOURCE_DIR}/gl2psTestSimple.c DESTINATION ${GL2PS_DOC})
 
@@ -136,6 +136,7 @@
   target_link_libraries(gl2psTestSimple lib ${EXTERNAL_LIBRARIES})
 endif(GLUT_FOUND)
 
+if(ENABLE_DOC)
 find_package(LATEX)
 if(PDFLATEX_COMPILER)
   add_custom_command(OUTPUT gl2ps.pdf DEPENDS gl2ps.tex 
@@ -154,6 +155,7 @@
     add_custom_target(html DEPENDS gl2ps.html)
   endif(TTH)
 endif(PDFLATEX_COMPILER)
+endif(ENABLE_DOC)
 
 set(CPACK_PACKAGE_VENDOR "Christophe Geuzaine")
 set(CPACK_PACKAGE_VERSION_MAJOR ${GL2PS_MAJOR_VERSION})
