diff -aur src3.2.6.old/VISU_SRC_3.2.6/src/PIPELINE/VISU_PipeLine.cxx src3.2.6/VISU_SRC_3.2.6/src/PIPELINE/VISU_PipeLine.cxx
--- src3.2.6.old/VISU_SRC_3.2.6/src/PIPELINE/VISU_PipeLine.cxx	2008-05-25 19:32:55.000000000 +0200
+++ src3.2.6/VISU_SRC_3.2.6/src/PIPELINE/VISU_PipeLine.cxx	2008-05-26 04:24:56.000000000 +0200
@@ -32,6 +32,7 @@
 #include "SALOME_ExtractGeometry.h"
 
 #include <float.h>
+#include <climits>
 
 #include <vtkObjectFactory.h>
 #include <vtkDataSetMapper.h>
diff -aur src3.2.6.old/VISU_SRC_3.2.6/src/PIPELINE/VISU_StreamLinesPL.cxx src3.2.6/VISU_SRC_3.2.6/src/PIPELINE/VISU_StreamLinesPL.cxx
--- src3.2.6.old/VISU_SRC_3.2.6/src/PIPELINE/VISU_StreamLinesPL.cxx	2008-05-25 19:32:55.000000000 +0200
+++ src3.2.6/VISU_SRC_3.2.6/src/PIPELINE/VISU_StreamLinesPL.cxx	2008-05-26 04:35:35.000000000 +0200
@@ -30,7 +30,7 @@
 #include "VISU_UsedPointsFilter.hxx"
 #include "VTKViewer_GeometryFilter.h"
 
-#include <algo.h>
+#include <algorithm>
 
 #include <vtkCell.h>
 #include <vtkPointSet.h>
diff -aur src3.2.6.old/VISU_SRC_3.2.6/src/VVTK/VVTK_PrimitiveBox.cxx src3.2.6/VISU_SRC_3.2.6/src/VVTK/VVTK_PrimitiveBox.cxx
--- src3.2.6.old/VISU_SRC_3.2.6/src/VVTK/VVTK_PrimitiveBox.cxx	2008-05-25 19:32:55.000000000 +0200
+++ src3.2.6/VISU_SRC_3.2.6/src/VVTK/VVTK_PrimitiveBox.cxx	2008-05-26 04:39:50.000000000 +0200
@@ -43,7 +43,7 @@
 #include <qpushbutton.h>
 #include <qfiledialog.h>
 
-#include <iostream.h>
+#include <iostream>
 
 using namespace std;
 
