diff -Nur numarray-1.5.2/Include/numarray/arraybase.h numarray-1.5.2.new/Include/numarray/arraybase.h
--- numarray-1.5.2/Include/numarray/arraybase.h	2006-01-23 11:45:03.000000000 +0000
+++ numarray-1.5.2.new/Include/numarray/arraybase.h	2006-08-30 19:54:21.000000000 +0100
@@ -1,6 +1,8 @@
 #if !defined(_arraybase_h)
 #define _arraybase_h 1
 
+#include <Python.h>
+
 #if defined(SIZEOF_VOID_P)
 #if SIZEOF_VOID_P == 8
 #define LP64 1
@@ -17,6 +19,8 @@
 #define HAS_UINT64 1
 #endif
 
+#include "numconfig.h"
+
 #define REFCOUNT(obj) (((PyObject *)(obj))->ob_refcnt)
 #define MAX_ELSIZE 16
 
diff -Nur numarray-1.5.2/Include/numarray/cfunc.h numarray-1.5.2.new/Include/numarray/cfunc.h
--- numarray-1.5.2/Include/numarray/cfunc.h	2005-10-27 22:06:35.000000000 +0100
+++ numarray-1.5.2.new/Include/numarray/cfunc.h	2006-08-30 19:54:21.000000000 +0100
@@ -1,6 +1,8 @@
 #if !defined(__cfunc__)
 #define __cfunc__ 1
 
+#include <Python.h>
+
 typedef PyObject *(*CFUNCasPyValue)(void *);
 typedef int (*UFUNC)(long, long, long, void **, long*);
 /* typedef void (*CFUNC_2ARG)(long, void *, void *); */
diff -Nur numarray-1.5.2/Include/numarray/nummacro.h numarray-1.5.2.new/Include/numarray/nummacro.h
--- numarray-1.5.2/Include/numarray/nummacro.h	2005-11-28 01:07:02.000000000 +0000
+++ numarray-1.5.2.new/Include/numarray/nummacro.h	2006-08-30 19:54:21.000000000 +0100
@@ -1,6 +1,8 @@
 #if !defined(_ndarraymacro)
 #define _ndarraymacro
 
+#include <Python.h>
+
 /* The structs defined here are private implementation details of numarray
 which are subject to change w/o notice.
 */
diff -Nur numarray-1.5.2/Include/numarray/tc.h numarray-1.5.2.new/Include/numarray/tc.h
--- numarray-1.5.2/Include/numarray/tc.h	2003-05-09 18:11:56.000000000 +0100
+++ numarray-1.5.2.new/Include/numarray/tc.h	2006-08-30 19:54:21.000000000 +0100
@@ -1,5 +1,7 @@
 #if defined(MEASURE_TIMING)
 
+#include <Python.h>
+
 /* ----------------------------------------------------------------------- */
 
 typedef struct 
