diff -U 3 -dHrN Normaliz2.5.orig/source/Makefile Normaliz2.5/source/Makefile
--- Normaliz2.5.orig/source/Makefile	2010-08-04 15:04:02.000000000 +0200
+++ Normaliz2.5/source/Makefile	2010-10-20 22:47:23.000000000 +0200
@@ -1,9 +1,7 @@
 ##
 ## Makefile for normaliz
 ##
-CXX = g++
 CXXFLAGS += -Wall -Wno-sign-compare
-CXXFLAGS += -O3 -funroll-loops -pipe
 
 ## use OpenMP?
 ifeq ($(OPENMP),no)
@@ -12,8 +10,8 @@
  CXXFLAGS += -fopenmp
 endif
 
-N64FLAGS = -Dnorm64 -static
-NBIGFLAGS = -Dnormbig -static
+N64FLAGS = -Dnorm64
+NBIGFLAGS = -Dnormbig
 GMPFLAGS = -lgmpxx -lgmp
 
 SOURCES = full_cone.cpp integer.cpp cone_dual_mode.cpp lineare_transformation.cpp list_operations.cpp matrix.cpp mode.cpp Normaliz.cpp output.cpp simplex.cpp sublattice_representation.cpp vector_operations.cpp
@@ -29,13 +27,13 @@
 	@mkdir -p obj64
 	$(CXX) $(CXXFLAGS) $(N64FLAGS) -c $< -o $@
 norm64: Normaliz.cpp $(N64OBJ)
-	$(CXX) $(CXXFLAGS) $(N64FLAGS) Normaliz.cpp $(N64OBJ) -o norm64
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(N64FLAGS) Normaliz.cpp $(N64OBJ) -o norm64
 
 objBig/%.o: %.cpp $(HEADERS)
 	@mkdir -p objBig
 	$(CXX) $(CXXFLAGS) $(NBIGFLAGS) -c $< -o $@
 normbig: Normaliz.cpp $(NBIGOBJ)
-	$(CXX) $(CXXFLAGS) $(NBIGFLAGS) Normaliz.cpp $(NBIGOBJ) $(GMPFLAGS) -o normbig
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(NBIGFLAGS) Normaliz.cpp $(NBIGOBJ) $(GMPFLAGS) -o normbig
 
 clean:
 	-rm -rf obj64 objBig
