--- GUI_SRC_3.2.6/doc/salome/gui/Makefile.in.org	2007-04-24 18:41:04.000000000 +0200
+++ GUI_SRC_3.2.6/doc/salome/gui/Makefile.in	2008-01-03 12:51:11.000000000 +0100
@@ -28,7 +28,7 @@
 @COMMENCE@
 
 usr_docs:
-	cp -fr $(srcdir)/GUI ./
+#	cp -fr $(srcdir)/GUI ./
 	-find $(PWD) -name CVS -prune -exec rm -rf {} \;
 
 docs: usr_docs
--- ./GUI_SRC_3.2.6/adm_local/Makefile.in.org     2007-04-24 18:41:04.000000000 +0200
+++ ./GUI_SRC_3.2.6/adm_local/Makefile.in 2007-09-26 23:23:59.000000000 +0200
@@ -26,12 +26,12 @@
 all: resources

 install:
-	cp -rf @top_srcdir@/adm_local @prefix@
+#	cp -rf @top_srcdir@/adm_local @prefix@

 bin:

 resources :
-	cp -rf @top_srcdir@/adm_local $(top_builddir)
+#	cp -rf @top_srcdir@/adm_local $(top_builddir)

 inc:

--- ./GUI_SRC_3.2.6/build_configure.org   2007-04-24 18:41:04.000000000 +0200
+++ ./GUI_SRC_3.2.6/build_configure       2007-05-26 12:29:53.000000000 +0200
@@ -109,7 +109,7 @@

 # make a link allowing AC_OUTPUT to find the salome_adm/.../*.in  files
 echo "" >> configure.in_tmp1
-echo 'ln -fs ${KERNEL_ROOT_DIR}/salome_adm ${ROOT_SRCDIR}/.' >> configure.in_tmp1
+echo 'cp -prf ${KERNEL_ROOT_DIR}/salome_adm ${ROOT_SRCDIR}/.' >> configure.in_tmp1

 echo  "" >> configure.in_tmp1
 echo "AC_OUTPUT([ \\" >> configure.in_tmp1
diff -aburN src3.2.6_orig/GUI_SRC_3.2.6/adm_local/unix/config_files/check_sip.m4 src3.2.6/GUI_SRC_3.2.6/adm_local/unix/config_files/check_sip.m4
--- ./GUI_SRC_3.2.6/adm_local/unix/config_files/check_sip.m4.org	2007-11-25 19:13:29.000000000 +0100
+++ ./GUI_SRC_3.2.6/adm_local/unix/config_files/check_sip.m4	2007-11-25 19:15:45.000000000 +0100
@@ -50,6 +50,10 @@
            SIP_VERS=v4_new ;;
          4.5*)
            SIP_VERS=v4_new ;;
+	 4.6*)
+	   SIP_VERS=v4_new ;;
+	 4.7*)
+	   SIP_VERS=v4_new ;;
          *)
            AC_MSG_RESULT(sip version $version not supported)
            SIP_VERS=unsupported ;;
diff -aburN src3.2.6_orig/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/Makefile.in src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/Makefile.in
--- ./GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/Makefile.in.org	2007-11-25 19:13:29.000000000 +0100
+++ ./GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/Makefile.in	2007-11-25 19:14:50.000000000 +0100
@@ -41,6 +41,12 @@
 ifeq ($(QT_VERS),Qt_3_3_7)
 QT_VERS = Qt_3_3_0
 endif
+ifeq ($(QT_VERS),Qt_3_3_8)
+QT_VERS = Qt_3_3_0
+endif
+ifeq ($(QT_VERS),Qt_3_3_8b)
+QT_VERS = Qt_3_3_6
+endif
 
 # Sip flags
 SIP_FLAGS = -t WS_X11 -t $(QT_VERS) $(PYQT_SIPFLAGS) -s ".cc" -c . -I $(PYQT_SIPS)
diff -aburN src3.2.6_orig/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/Makefile.in src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/Makefile.in
--- ./GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/Makefile.in.org	2007-11-25 19:13:29.000000000 +0100
+++ ./GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/Makefile.in	2007-11-25 19:14:21.000000000 +0100
@@ -44,6 +44,14 @@
 QT_VERS = Qt_3_3_0
 endif
 
+ifeq ($(QT_VERS),Qt_3_3_8)
+QT_VERS = Qt_3_3_0
+endif
+
+ifeq ($(QT_VERS),Qt_3_3_8b)
+QT_VERS = Qt_3_3_6
+endif
+
 # Sip flags
 SIP_FLAGS = -t WS_X11 -t $(QT_VERS) $(PYQT_SIPFLAGS) -s ".cc" -c . -I $(PYQT_SIPS)
 
