diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/GLViewer/GLViewer_Tools.cxx src3.2.6/GUI_SRC_3.2.6/src/GLViewer/GLViewer_Tools.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/GLViewer/GLViewer_Tools.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/GLViewer/GLViewer_Tools.cxx	2008-05-26 00:04:12.000000000 +0200
@@ -27,7 +27,9 @@
 
 #include <GL/gl.h>
 
-#include <iostream.h>
+#include <iostream>
+
+using std::cout;
 
 /*!
   Constructor
diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/ResExporter/ResourceExporter.cxx src3.2.6/GUI_SRC_3.2.6/src/ResExporter/ResourceExporter.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/ResExporter/ResourceExporter.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/ResExporter/ResourceExporter.cxx	2008-05-26 00:38:10.000000000 +0200
@@ -47,6 +47,7 @@
 #include <qstringlist.h>
 #include <qapplication.h>
 #include <iostream>
+#include <cstdlib>
 
 /*!
   \return the SALOME version number
diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/SalomeApp/SalomeApp_Study.cxx src3.2.6/GUI_SRC_3.2.6/src/SalomeApp/SalomeApp_Study.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/SalomeApp/SalomeApp_Study.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/SalomeApp/SalomeApp_Study.cxx	2008-05-26 00:28:19.000000000 +0200
@@ -36,7 +36,7 @@
 #include <qdict.h>
 
 #include "utilities.h"
-#include <iostream.h>
+#include <iostream>
 #include <string>
 #include <vector>
 
diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/SUIT/SUIT_ResourceMgr.cxx src3.2.6/GUI_SRC_3.2.6/src/SUIT/SUIT_ResourceMgr.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/SUIT/SUIT_ResourceMgr.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/SUIT/SUIT_ResourceMgr.cxx	2008-05-25 23:41:44.000000000 +0200
@@ -22,6 +22,7 @@
 #include <qdir.h>
 #include <qapplication.h>
 #include <qregexp.h>
+#include <cstdlib>
 
 /*!
     Constructor
diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/SVTK/SVTK_InteractorStyle.cxx src3.2.6/GUI_SRC_3.2.6/src/SVTK/SVTK_InteractorStyle.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/SVTK/SVTK_InteractorStyle.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/SVTK/SVTK_InteractorStyle.cxx	2008-05-26 00:19:17.000000000 +0200
@@ -67,7 +67,7 @@
 //VRV: porting on Qt 3.0.5
 #include <algorithm>
 
-#include <iostream.h>
+#include <iostream>
 
 using namespace std;
 
diff -aur src3.2.6.old/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ConvexTool.cxx src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ConvexTool.cxx
--- src3.2.6.old/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ConvexTool.cxx	2008-05-25 19:32:46.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ConvexTool.cxx	2008-05-26 00:12:09.000000000 +0200
@@ -31,6 +31,7 @@
 
 #include <set>
 #include <map>
+#include <algorithm>
 
 #include <vtkUnstructuredGrid.h>
 #include <vtkGeometryFilter.h>
