--- ./src3.2.6/GUI_SRC_3.2.6/adm_local/unix/config_files/check_vtk.m4.org	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/adm_local/unix/config_files/check_vtk.m4	2007-05-26 12:32:50.000000000 +0200
@@ -76,7 +76,7 @@
 if test -z $VTKHOME
 then 
    AC_MSG_WARN(undefined VTKHOME variable which specify where vtk was compiled)
-   if test -f /usr/include/vtk/vtkPlane.h ; then
+   if test -f /usr/include/vtk-5.0/vtkPlane.h ; then
       AC_MSG_RESULT(trying /usr)
       VTKHOME="/usr"
    fi
@@ -84,9 +84,9 @@
 
 if test ! -z $VTKHOME
 then
-   LOCAL_INCLUDES="-I$VTKHOME/include/vtk $LOCAL_INCLUDES"
-   LOCAL_LIBS="-L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk -L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk/python $LOCAL_LIBS"
-   TRY_LINK_LIBS="-L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk -L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk/python $TRY_LINK_LIBS"
+   LOCAL_INCLUDES="-I$VTKHOME/include/vtk-5.0 $LOCAL_INCLUDES"
+   LOCAL_LIBS="-L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk-5.0 -L/usr/lib/python2.4/site-packages/vtk $LOCAL_LIBS"
+   TRY_LINK_LIBS="-L$VTKHOME/lib${LIB_LOCATION_SUFFIX}/vtk-5.0 -L/usr/lib/python2.4/site-packages/vtk/ $TRY_LINK_LIBS"
 fi
 
 dnl vtk headers
@@ -142,4 +142,4 @@
 # Save cache
 AC_CACHE_SAVE
 
-])dnl
\ No newline at end of file
+])dnl
Seulement dans ./src3.2.6/GUI_SRC_3.2.6: autom4te.cache
Seulement dans ./src3.2.6/GUI_SRC_3.2.6: configure
Seulement dans ./src3.2.6/GUI_SRC_3.2.6: configure.in
Seulement dans ./src3.2.6/GUI_SRC_3.2.6: salome_adm
diff -abur /home/francois/Desktop/salome/src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/SalomePyQt.sip ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/SalomePyQt.sip
--- ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/SalomePyQt.sip	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt/SalomePyQt.sip	2007-11-26 00:18:40.000000000 +0100
@@ -13,7 +13,7 @@
 
 class SALOME_Selection : QObject
 {
-%HeaderCode
+%TypeHeaderCode
 #include <SalomePyQt.h>
 %End
 
@@ -75,7 +75,7 @@
 
 class SalomePyQt
 {
-%HeaderCode
+%TypeHeaderCode
 #include <SalomePyQt.h>
 %End
 
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SalomePyQt: SalomePyQt.sip~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx	2007-11-25 23:34:31.000000000 +0100
@@ -46,8 +46,8 @@
 #include "sipSalomePyQtGUIDeclSalomePyQtGUI.h"
 #endif
 
-#include <sipqtQWidget.h>
-#include <sipqtQPopupMenu.h>
+//#include <sipqtQWidget.h>
+//#include <sipqtQPopupMenu.h>
 
 #include <CORBA.h>
 
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI: SALOME_PYQT_Module.cxx~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_AppendFilter.cxx ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_AppendFilter.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_AppendFilter.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_AppendFilter.cxx	2007-11-25 21:39:54.000000000 +0100
@@ -119,8 +119,8 @@
   vtkIdType aPntStartId = 0;
   vtkIdType aCellStartId = 0;
 
-  for(vtkIdType aDataSetId = 0; aDataSetId < this->NumberOfInputs; ++aDataSetId){
-    vtkDataSet* aDataSet = (vtkDataSet *)(this->Inputs[aDataSetId]);
+  for(vtkIdType aDataSetId = 0; aDataSetId < this->InputList->GetNumberOfItems(); ++aDataSetId){
+    vtkDataSet* aDataSet = (vtkDataSet *)(this->GetInput(aDataSetId));
     // Do mapping of the nodes
     if(!GetSharedPointsDataSet()){
       vtkIdType aNbPnts = aDataSet->GetNumberOfPoints();
@@ -265,11 +265,11 @@
 
   numCells = 0;
 
-  vtkDataSetAttributes::FieldList cellList(this->NumberOfInputs);
+  vtkDataSetAttributes::FieldList cellList(this->InputList->GetNumberOfItems());
   int firstCD=1;
 
-  for (idx = 0; idx < this->NumberOfInputs; ++idx) {
-    ds = (vtkDataSet *)(this->Inputs[idx]);
+  for (idx = 0; idx < this->InputList->GetNumberOfItems(); ++idx) {
+    ds = (vtkDataSet *)(this->GetInput(idx));
     if (ds != NULL)  {
       if ( ds->GetNumberOfPoints() <= 0 && ds->GetNumberOfCells() <= 0 )  {
         continue; //no input, just skip
@@ -308,8 +308,8 @@
   output->GetPointData()->PassData(GetSharedPointsDataSet()->GetPointData());
 
   // 2.cells
-  for (idx = 0; idx < this->NumberOfInputs; ++idx) {
-    ds = (vtkDataSet *)(this->Inputs[idx]);
+  for (idx = 0; idx < this->InputList->GetNumberOfItems(); ++idx) {
+    ds = (vtkDataSet *)(this->GetInput(idx));
     if (ds != NULL) {
       numCells = ds->GetNumberOfCells(); 
       cd = ds->GetCellData();
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer: VTKViewer_AppendFilter.cxx~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ExtractUnstructuredGrid.cxx ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ExtractUnstructuredGrid.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ExtractUnstructuredGrid.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ExtractUnstructuredGrid.cxx	2007-11-25 20:53:53.000000000 +0100
@@ -240,7 +240,7 @@
 	for(vtkIdType i = 0, *pts, npts; aConnectivity->GetNextCell(npts,pts); i++){
 	  aCellLocationsArray->SetValue(i,aConnectivity->GetTraversalLocation(npts));
 	}
-	anOutput->SetCells(aCellTypesArray,aCellLocationsArray,aConnectivity);
+	anOutput->SetCells(aCellTypesArray,(vtkIdTypeArray*)aCellLocationsArray,aConnectivity);
 	anOutput->SetPoints(anInput->GetPoints());
 	aCellLocationsArray->Delete();
       }
@@ -329,7 +329,7 @@
       for(vtkIdType i = 0, *pts, npts; aConnectivity->GetNextCell(npts,pts); i++){
 	aCellLocationsArray->SetValue(i,aConnectivity->GetTraversalLocation(npts));
       }
-      anOutput->SetCells(aCellTypesArray,aCellLocationsArray,aConnectivity);
+      anOutput->SetCells(aCellTypesArray,(vtkIdTypeArray*)aCellLocationsArray,aConnectivity);
       anOutput->SetPoints(anInput->GetPoints());
       aCellLocationsArray->Delete();
     }
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer: VTKViewer_ExtractUnstructuredGrid.cxx~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_GeometryFilter.cxx ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_GeometryFilter.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_GeometryFilter.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_GeometryFilter.cxx	2007-11-25 20:55:38.000000000 +0100
@@ -80,7 +80,7 @@
 VTKViewer_GeometryFilter
 ::Execute()
 {
-  vtkDataSet *input= this->GetInput();
+  vtkDataSet *input= (vtkDataSet*) this->GetInput();
   vtkIdType numCells=input->GetNumberOfCells();
 
   if (numCells == 0)
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer: VTKViewer_GeometryFilter.cxx~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ShrinkFilter.cxx ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ShrinkFilter.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ShrinkFilter.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_ShrinkFilter.cxx	2007-11-25 21:40:41.000000000 +0100
@@ -58,7 +58,7 @@
   vtkFloatingPointType center[3], *p, pt[3];
   vtkPointData *pd, *outPD;;
   vtkIdList *ptIds, *newPtIds;
-  vtkDataSet *input= this->GetInput();
+  vtkDataSet *input= (vtkDataSet*) this->GetInput();
   vtkUnstructuredGrid *output = this->GetOutput();
   vtkIdType tenth;
   vtkFloatingPointType decimal;
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer: VTKViewer_ShrinkFilter.cxx~
diff -abur ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_TransformFilter.cxx ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_TransformFilter.cxx
--- ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_TransformFilter.cxx	2007-04-24 18:41:04.000000000 +0200
+++ ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer/VTKViewer_TransformFilter.cxx	2007-11-25 21:41:40.000000000 +0100
@@ -43,8 +43,8 @@
   vtkPoints *inPts;
   vtkPoints *newPts;
   int numPts, numCells;
-  vtkPointSet *input = this->GetInput();
-  vtkPointSet *output = this->GetOutput();
+  vtkPointSet *input = (vtkPointSet*) this->GetInput();
+  vtkPointSet *output = (vtkPointSet*) this->GetOutput();
   vtkPointData *pd=input->GetPointData(), *outPD=output->GetPointData();
   vtkCellData *cd=input->GetCellData(), *outCD=output->GetCellData();
   output->CopyStructure( input );
Seulement dans ./src3.2.6/GUI_SRC_3.2.6/src/VTKViewer: VTKViewer_TransformFilter.cxx~
