--- src3.2.6.orig/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx	2008-09-04 10:20:58.000000000 +0200
+++ src3.2.6/GUI_SRC_3.2.6/src/SALOME_PYQT/SALOME_PYQT_GUI/SALOME_PYQT_Module.cxx	2008-09-04 10:21:43.000000000 +0200
@@ -764,7 +764,7 @@
       if ( PyDict_Check( res1 ) ) {
         PyObject* key;
         PyObject* value;
-        int pos = 0;
+        Py_ssize_t pos = 0;
         while ( PyDict_Next( res1, &pos, &key, &value ) ) {
 	  // parse the return value
 	  // it should be a map: {integer:integer}
