--- tests/unit/security/certificate_authority.cpp.orig	2014-02-15 22:29:43.663465723 -0800
+++ tests/unit/security/certificate_authority.cpp	2014-02-15 22:30:02.649579686 -0800
@@ -3,7 +3,6 @@
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-#include <boost/make_shared.hpp>
 #include <hpx/hpx_fwd.hpp>
 #include <hpx/hpx_init.hpp>
 #include <hpx/components/security/certificate_store.hpp>
@@ -16,6 +15,7 @@
 #include <hpx/util/lightweight_test.hpp>
 #include <hpx/util/security/root_certificate_authority.hpp>
 #include <hpx/util/security/subordinate_certificate_authority.hpp>
+#include <boost/make_shared.hpp>
 
 #include <iostream>
 #include <vector>
