diff -Nur casacore-1.2.20110516.orig/casa/Arrays/IPosition.h casacore-1.2.20110516/casa/Arrays/IPosition.h
--- casacore-1.2.20110516.orig/casa/Arrays/IPosition.h	2011-05-17 06:03:06.000000000 +0100
+++ casacore-1.2.20110516/casa/Arrays/IPosition.h	2011-05-17 06:04:15.000000000 +0100
@@ -33,6 +33,7 @@
 #include <casa/iosfwd.h>
 #include <casa/BasicSL/String.h>
 #include <vector>
+#include <cstddef>
 #include <unistd.h>
 
 namespace casa { //# NAMESPACE CASA - BEGIN
diff -Nur casacore-1.2.20110516.orig/casa/Containers/Block.h casacore-1.2.20110516/casa/Containers/Block.h
--- casacore-1.2.20110516.orig/casa/Containers/Block.h	2011-05-17 06:03:05.000000000 +0100
+++ casacore-1.2.20110516/casa/Containers/Block.h	2011-05-17 06:05:55.000000000 +0100
@@ -28,6 +28,7 @@
 #ifndef CASA_BLOCK_H
 #define CASA_BLOCK_H
 
+#include <cstddef>
 #include <casa/aips.h>
 ///#include <casa/Containers/BlockIO.h>
 #include <casa/Utilities/Copy.h>
