diff -Naur scalasca-1.1.orig/mf/common.defs scalasca-1.1/mf/common.defs
--- scalasca-1.1.orig/mf/common.defs	2009-04-21 20:43:55.000000000 +0200
+++ scalasca-1.1/mf/common.defs	2009-05-17 11:48:49.000000000 +0200
@@ -1,10 +1,11 @@
 MKDIR = $(TOPDIR)/utils/build/mkinstalldirs
 INSTALL = $(TOPDIR)/utils/build/install-sh
 
-BINDIR = ${PREFIX}/bin$(DSUF)
-LIBDIR = ${PREFIX}/lib$(DSUF)
-INCDIR = ${PREFIX}/include$(DSUF)
-DOCDIR = ${PREFIX}/doc
+BINDIR = ${DESTDIR}${PREFIX}/bin$(DSUF)
+LIBDIR = ${DESTDIR}${PREFIX}/lib$(DSUF)
+INCDIR = ${DESTDIR}${PREFIX}/include$(DSUF)
+DOCDIR = ${DESTDIR}${PREFIX}/share/doc/${PF}
+EXPDIR = ${DESTDIR}${PREFIX}/share/doc/${PF}/examples
 
 OMPTRGT = opari
 
diff -Naur scalasca-1.1.orig/utils/build/RealMakefile scalasca-1.1/utils/build/RealMakefile
--- scalasca-1.1.orig/utils/build/RealMakefile	2009-04-21 20:43:55.000000000 +0200
+++ scalasca-1.1/utils/build/RealMakefile	2009-05-17 11:48:49.000000000 +0200
@@ -224,20 +224,20 @@
 
 install-examples:
 	@echo "++++++++++ EXAMPLES +++++++++++++++++++++++++++"
-	$(MKDIR) -m 755 ${PREFIX}/example
+	$(MKDIR) -m 755 ${EXPDIR}
 	@list='example/*'; for p in $$list; do \
-		echo "$(INSTALL) -c -m 644 $$p ${PREFIX}/example" ; \
-		eval "$(INSTALL) -c -m 644 $$p ${PREFIX}/example" ; \
+		echo "$(INSTALL) -c -m 644 $$p ${EXPDIR}" ; \
+		eval "$(INSTALL) -c -m 644 $$p ${EXPDIR}" ; \
 	done
-	@rm -f ${PREFIX}/example/EXAMPLE.tgz
+	@rm -f ${EXPDIR}/EXAMPLE.tgz
 	@if test -f ../Makefile.defs.be; then \
-		$(INSTALL) -c -m 644 ../Makefile.defs.be ${PREFIX}/example/Makefile.defs; \
+		$(INSTALL) -c -m 644 ../Makefile.defs.be ${EXPDIR}/Makefile.defs; \
 	else \
-		sed -e 's/DEFAULTPREC/PREC/' ../Makefile.defs > ${PREFIX}/example/Makefile.defs; \
-		chmod 644 ${PREFIX}/example/Makefile.defs; \
+		sed -e 's/DEFAULTPREC/PREC/' ../Makefile.defs > ${EXPDIR}/Makefile.defs; \
+		chmod 644 ${EXPDIR}/Makefile.defs; \
 	fi
 	gunzip -dc example/EXAMPLE.tgz | \
-		( cd ${PREFIX}/example; tar xpvf - )
+		( cd ${EXPDIR}; tar xpvf - )
 
 install-config:
 	$(MKDIR) -m 755 ${DOCDIR}
diff -Naur scalasca-1.1.orig/utils/scalasca/Makefile scalasca-1.1/utils/scalasca/Makefile
--- scalasca-1.1.orig/utils/scalasca/Makefile	2009-04-21 20:43:55.000000000 +0200
+++ scalasca-1.1/utils/scalasca/Makefile	2009-05-17 11:57:22.000000000 +0200
@@ -39,10 +39,11 @@
 	$(ECXX) $(CXXFLAGS) $(LDFLAGS) $(SCAN_INCS) scan.cpp -o $@ $(SCAN_LIBS)
 
 square: square.in
-	sed -e "s!@DOCDIR@!$(DOCDIR)/html"! square.in > square
+	sed -e "s!@DOCDIR@!$(DOCDIR)/html!" square.in > square
 
 scalasca: scalasca.in
-	sed -e "s!@CONFIG@!$(CFGKEY): $(CONFIG)!" scalasca.in > scalasca
+	sed -e "s!@CONFIG@!$(CFGKEY): $(CONFIG)!" \
+	    -e "s!@DOCDIR@!$(DOCDIR)!" scalasca.in > scalasca
 
 
 #------------------------------------------------------------------------------
diff -Naur scalasca-1.1.orig/utils/scalasca/scalasca.in scalasca-1.1/utils/scalasca/scalasca.in
--- scalasca-1.1.orig/utils/scalasca/scalasca.in	2009-04-21 20:43:55.000000000 +0200
+++ scalasca-1.1/utils/scalasca/scalasca.in	2009-05-17 12:37:46.000000000 +0200
@@ -43,8 +43,8 @@
 
 # report location and try to present quickref guide
 quickref () {
-  REFLOC=doc/manuals/quickref.pdf
-  REFPDF=`dirname $BINDIR`/$REFLOC
+  REFLOC=manuals/quickref.pdf
+  REFPDF=@DOCDIR@/$REFLOC
   echo "Scalasca $VERSION - quick reference guide:"
   if [ ! -f "$REFPDF" ]
   then
