--- a/cmake/tribits/package_arch/TribitsWriteClientExportFiles.cmake 2012-11-12 14:58:00.000000000 +0100
+++ b/cmake/tribits/package_arch/TribitsWriteClientExportFiles.cmake 2012-11-12 14:42:47.000000000 +0100
@@ -478,8 +478,8 @@
   # directories using the installed config file. This is to deal with
   # installers that allow relocation of the install tree at *install*
   # time.
-  SET(FULL_LIBRARY_DIRS_SET "\${CMAKE_CURRENT_LIST_DIR}/../../../${${PROJECT_NAME}_INSTALL_LIB_DIR}")
-  SET(FULL_INCLUDE_DIRS_SET "\${CMAKE_CURRENT_LIST_DIR}/../../../${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}")
+  SET(FULL_LIBRARY_DIRS_SET "${${PROJECT_NAME}_INSTALL_LIB_DIR}")
+  SET(FULL_INCLUDE_DIRS_SET "${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}")
 
   # Custom code in configuration file.
   SET(PACKAGE_CONFIG_CODE "")
@@ -548,7 +548,7 @@
   IF (${PROJECT_NAME}_ENABLE_INSTALL_CMAKE_CONFIG_FILES)
     INSTALL(
       FILES ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${PACKAGE_NAME}Config_install.cmake
-      DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PACKAGE_NAME}"
+      DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PACKAGE_NAME}"
       RENAME ${PACKAGE_NAME}Config.cmake
       )
 
@@ -766,8 +766,8 @@
   # directories using the installed config file. This is to deal with
   # installers that allow relocation of the install tree at *install*
   # time.
-  SET(${PROJECT_NAME}_CONFIG_INCLUDE_DIRS "\${CMAKE_CURRENT_LIST_DIR}/../../../${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}")
-  SET(${PROJECT_NAME}_CONFIG_LIBRARY_DIRS "\${CMAKE_CURRENT_LIST_DIR}/../../../${${PROJECT_NAME}_INSTALL_LIB_DIR}")
+  SET(${PROJECT_NAME}_CONFIG_INCLUDE_DIRS "${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}")
+  SET(${PROJECT_NAME}_CONFIG_LIBRARY_DIRS "${${PROJECT_NAME}_INSTALL_LIB_DIR}")
 
   # Write the specification of the rpath if necessary. This is only needed if we're building shared libraries.
   IF(BUILD_SHARED_LIBS)
@@ -782,7 +782,7 @@
   IF(HAS_INSTALL_TARGETS)
     INSTALL(
       EXPORT ${PROJECT_NAME}
-      DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+      DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
       FILE ${PROJECT_NAME}Targets.cmake
       )
     # Import the targets in applications.
@@ -809,7 +809,7 @@
 
     INSTALL(
       FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config_install.cmake
-      DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+      DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
       RENAME ${PROJECT_NAME}Config.cmake
       )
   ENDIF()
@@ -853,7 +853,7 @@
 
   INSTALL(
     FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
-    DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+    DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
   )
 
 ENDFUNCTION()
