diff --git a/packages/fei/test_utils/snl_fei_tester.cpp b/packages/fei/test_utils/snl_fei_tester.cpp
index 0ebfa9d..17278ee 100644
--- a/packages/fei/test_utils/snl_fei_tester.cpp
+++ b/packages/fei/test_utils/snl_fei_tester.cpp
@@ -523,7 +523,7 @@ int snl_fei_tester::save_block_node_soln(DataReader& data, fei::Vector* vec,
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
@@ -591,7 +591,7 @@ int snl_fei_tester::save_block_elem_soln(DataReader& data, fei::Vector* vec,
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
@@ -668,7 +668,7 @@ int snl_fei_tester::save_multiplier_soln(DataReader& data, fei::Vector* vec,
   FEI_OFSTREAM outfile(str.c_str());
 
   if (!outfile || outfile.bad()) {
-    fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
+    fei::console_out() << "ERROR opening solution output file " << fileName.str() << FEI_ENDL;
     return(-1);
   }
 
diff --git a/packages/ifpack/src/Ifpack_Hypre.cpp b/packages/ifpack/src/Ifpack_Hypre.cpp
index ea6ba35..26a1281 100644
--- a/packages/ifpack/src/Ifpack_Hypre.cpp
+++ b/packages/ifpack/src/Ifpack_Hypre.cpp
@@ -51,6 +51,7 @@
 
 using Teuchos::RCP;
 using Teuchos::rcp;
+using std::endl;
 
 Ifpack_Hypre::Ifpack_Hypre(Epetra_RowMatrix* A):
   A_(rcp(A,false)),
