--- a/cmake/tribits/package_arch/TribitsGlobalMacros.cmake   2013-04-03 11:00:54.342539245 +0200
+++ b/cmake/tribits/package_arch/TribitsGlobalMacros.cmake   2013-04-03 11:04:30.283837064 +0200
@@ -209,26 +209,6 @@
     ADVANCED_SET(${PROJECT_NAME}_LINK_SEARCH_START_STATIC OFF CACHE BOOL
       "If on, then the properter LINK_SEARCH_START_STATIC will be added to all executables." )
   ENDIF()
-  
-  ADVANCED_SET(${PROJECT_NAME}_INSTALL_INCLUDE_DIR "include"
-    CACHE PATH
-    "Location where the headers will be installed.  If given as a relative path, it will be relative to ${CMAKE_INSTALL_PREFIX}.  If given as an absolute path, it will used as such.  Default is 'include'"
-    )
-  
-  ADVANCED_SET(${PROJECT_NAME}_INSTALL_LIB_DIR "lib"
-    CACHE PATH
-    "Location where the libraries will be installed.  If given as a relative path, it will be relative to ${CMAKE_INSTALL_PREFIX}.  If given as an absolute path, it will used as such.  Default is 'lib'"
-    )
-  
-  ADVANCED_SET(${PROJECT_NAME}_INSTALL_RUNTIME_DIR "bin"
-    CACHE PATH
-    "Location where the runtime DLLs and designated programs will be installed.  If given as a relative path, it will be relative to ${CMAKE_INSTALL_PREFIX}.  If given as an absolute path, it will used as such.  Default is 'bin'"
-    )
-  
-  ADVANCED_SET(${PROJECT_NAME}_INSTALL_EXAMPLE_DIR "example"
-    CACHE PATH
-    "Location where assorted examples will be installed.  If given as a relative path, it will be relative to ${CMAKE_INSTALL_PREFIX}.  If given as an absolute path, it will used as such.  Default is 'example'"
-    )
 
   IF ("${${PROJECT_NAME}_INSTALL_LIBRARIES_AND_HEADERS_DEFAULT}" STREQUAL "")
     # Assume the TriBITS project wants to install headers and libraries by default
--- a/cmake/tribits/package_arch/TribitsPackageWritePackageConfig.cmake	2012-11-12 14:58:00.000000000 +0100
+++ b/cmake/tribits/package_arch/TribitsPackageWritePackageConfig.cmake	2012-11-12 14:42:47.000000000 +0100
@@ -367,7 +367,7 @@
 
   INSTALL(
     FILES ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${PACKAGE_NAME}Config_install.cmake
-    DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PACKAGE_NAME}"
+    DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PACKAGE_NAME}"
     RENAME ${PACKAGE_NAME}Config.cmake
   )
 
@@ -563,7 +563,7 @@
   IF(HAS_INSTALL_TARGETS)
     INSTALL(
       EXPORT ${PROJECT_NAME}
-      DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+      DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
       FILE ${PROJECT_NAME}Targets.cmake
       )
     # Import the targets in applications.
@@ -589,7 +589,7 @@
 
   INSTALL(
     FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config_install.cmake
-    DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+    DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
     RENAME ${PROJECT_NAME}Config.cmake
   )
   
@@ -632,7 +632,7 @@
 
   INSTALL(
     FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
-    DESTINATION "${${PROJECT_NAME}_INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}"
+    DESTINATION "${${PROJECT_NAME}_INSTALL_CONFIG_DIR}/${PROJECT_NAME}"
   )
 
 ENDFUNCTION()
--- a/packages/fei/base/CMakeLists.txt	2012-11-12 14:05:18.000000000 +0100
+++ b/packages/fei/base/CMakeLists.txt	2012-11-12 14:05:18.000000000 +0100
@@ -54,5 +54,5 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-   ${CMAKE_INSTALL_PREFIX}/include)
+   ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
 
