diff -Naur trilinos-12.10.1-Source.orig/commonTools/gtest/CMakeLists.txt trilinos-12.10.1-Source/commonTools/gtest/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/commonTools/gtest/CMakeLists.txt	2016-12-05 13:46:36.981584433 -0700
+++ trilinos-12.10.1-Source/commonTools/gtest/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -31,7 +31,7 @@
   )
 
 #install gtest header into the correct subdir.
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${Trilinos_INSTALL_INCLUDE_DIR}/gtest/)
+INSTALL(FILES ${HEADERS} DESTINATION ${Trilinos_INSTALL_INCLUDE_DIR}/gtest/)
 
 
 
diff -Naur trilinos-12.10.1-Source.orig/packages/kokkos/containers/src/CMakeLists.txt trilinos-12.10.1-Source/packages/kokkos/containers/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/kokkos/containers/src/CMakeLists.txt	2016-12-05 13:46:37.071535392 -0700
+++ trilinos-12.10.1-Source/packages/kokkos/containers/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -15,7 +15,7 @@
 FILE(GLOB HEADERS_IMPL impl/*.hpp)
 FILE(GLOB SOURCES impl/*.cpp)
 
-SET(TRILINOS_INCDIR ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+SET(TRILINOS_INCDIR ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
 
 INSTALL(FILES ${HEADERS_IMPL} DESTINATION ${TRILINOS_INCDIR}/impl/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/kokkos/core/src/CMakeLists.txt trilinos-12.10.1-Source/packages/kokkos/core/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/kokkos/core/src/CMakeLists.txt	2016-12-05 13:46:37.061540841 -0700
+++ trilinos-12.10.1-Source/packages/kokkos/core/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -40,7 +40,7 @@
 
 #-----------------------------------------------------------------------------
 
-SET(TRILINOS_INCDIR ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+SET(TRILINOS_INCDIR ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
 
 #-----------------------------------------------------------------------------
 
diff -Naur trilinos-12.10.1-Source.orig/packages/kokkos/core/src/impl/CMakeLists.txt trilinos-12.10.1-Source/packages/kokkos/core/src/impl/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/kokkos/core/src/impl/CMakeLists.txt	2016-12-05 13:46:37.061540841 -0700
+++ trilinos-12.10.1-Source/packages/kokkos/core/src/impl/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -12,7 +12,7 @@
     DEPLIBS 
     )
 
-SET(TRILINOS_INCDIR ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+SET(TRILINOS_INCDIR ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
 
 INSTALL(FILES ${HEADERS} DESTINATION ${TRILINOS_INCDIR}/impl/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/belos/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/belos/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/belos/src/CMakeLists.txt	2016-12-05 13:46:37.111513596 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/belos/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -39,4 +39,4 @@
   )
 ENDIF()
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/epetra/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/epetra/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/epetra/src/CMakeLists.txt	2016-12-05 13:46:37.111513596 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/epetra/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -42,4 +42,4 @@
   )
 ENDIF()
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/sacado/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/sacado/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/sacado/src/CMakeLists.txt	2016-12-05 13:46:37.101519046 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/sacado/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -37,4 +37,4 @@
   )
 ENDIF()
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/teuchos/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/teuchos/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/teuchos/src/CMakeLists.txt	2016-12-05 13:46:37.101519046 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/teuchos/src/CMakeLists.txt	2016-12-05 14:37:03.361602958 -0700
@@ -35,4 +35,4 @@
 )
 
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/thyra/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/thyra/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/thyra/src/CMakeLists.txt	2016-12-05 13:46:37.111513596 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/thyra/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -49,4 +49,4 @@
   )
 ENDIF()
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/rol/adapters/tpetra/src/CMakeLists.txt trilinos-12.10.1-Source/packages/rol/adapters/tpetra/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/rol/adapters/tpetra/src/CMakeLists.txt	2016-12-05 13:46:37.111513596 -0700
+++ trilinos-12.10.1-Source/packages/rol/adapters/tpetra/src/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -46,4 +46,4 @@
   )
 ENDIF()
 
-INSTALL(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+INSTALL(FILES ${HEADERS} DESTINATION ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_adapt/stk_adapt/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_adapt/stk_adapt/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_adapt/stk_adapt/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_adapt/stk_adapt/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -46,7 +46,7 @@
 )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_adapt/)
+${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_adapt/)
 
 ADD_SUBDIRECTORY(main)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_adapt/stk_adapt/sierra_element/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_adapt/stk_adapt/sierra_element/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_adapt/stk_adapt/sierra_element/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_adapt/stk_adapt/sierra_element/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -34,5 +34,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_adapt/sierra_element/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_adapt/sierra_element/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_algsup/stk_algsup/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_algsup/stk_algsup/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_algsup/stk_algsup/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_algsup/stk_algsup/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -21,5 +21,5 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_algsup/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_algsup/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_expreval/stk_expreval/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_expreval/stk_expreval/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_expreval/stk_expreval/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_expreval/stk_expreval/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -46,4 +46,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_expreval)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_expreval)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_io/stk_io/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_io/stk_io/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_io/stk_io/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_io/stk_io/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -49,7 +49,7 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/)
+${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/)
 
 # AGS: I moved this to the bottom so that the link order was correct
 # with  -lstk_io_uti before -lstk_io.  I think the DEPLIBS added in
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_io/stk_io/util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_io/stk_io/util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_io/stk_io/util/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_io/stk_io/util/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -45,4 +45,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/util/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/base/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/base/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/base/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/base/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -54,7 +54,7 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/base/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/base/)
 
 INSTALL(FILES ${HEADERS_IMPL} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/fem/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/fem/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/fem/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/fem/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -48,5 +48,5 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fem/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fem/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/fixtures/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/fixtures/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_mesh/stk_mesh/fixtures/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_mesh/stk_mesh/fixtures/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -54,4 +54,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -68,5 +68,5 @@
 )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/)
+${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/fixtures/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/fixtures/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/fixtures/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/fixtures/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -35,5 +35,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/fixtures/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/fixtures/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/function/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/function/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/function/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/function/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -37,5 +37,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/function/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/function/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/function/internal/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/function/internal/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/function/internal/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/function/internal/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -36,5 +36,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/function/internal/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/function/internal/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/math/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/math/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/math/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/math/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -35,5 +35,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/math/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/math/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/mesh/gen/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/mesh/gen/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/mesh/gen/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/mesh/gen/CMakeLists.txt	2016-12-05 13:53:32.265171949 -0700
@@ -35,5 +35,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/mesh/gen/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/mesh/gen/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/norm/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/norm/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/norm/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/norm/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -35,5 +35,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/norm/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/norm/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_percept/stk_percept/util/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_percept/stk_percept/util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -35,5 +35,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_percept/util/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_rebalance/stk_rebalance/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_rebalance/stk_rebalance/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_rebalance/stk_rebalance/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_rebalance/stk_rebalance/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -46,4 +46,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_rebalance/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_rebalance/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_rebalance/stk_rebalance_utils/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_rebalance/stk_rebalance_utils/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_rebalance/stk_rebalance_utils/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_rebalance/stk_rebalance_utils/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -18,4 +18,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-  ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_rebalance_utils)
+  ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_rebalance_utils)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search/stk_search/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search/stk_search/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search/stk_search/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search/stk_search/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -62,4 +62,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search/stk_search/diag/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search/stk_search/diag/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search/stk_search/diag/CMakeLists.txt	2016-12-05 13:46:37.381366472 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search/stk_search/diag/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -34,5 +34,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search/diag/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search/diag/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search_util/stk_search_util/stk_mesh/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search_util/stk_search_util/stk_mesh/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_search_util/stk_search_util/stk_mesh/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_search_util/stk_search_util/stk_mesh/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -54,4 +54,4 @@
   )
 
 #INSTALL(FILES ${HEADERS} DESTINATION
-#    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
+#    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -16,6 +16,6 @@
 TRIBITS_CONFIGURE_FILE(${PACKAGE_NAME}_config.h)
 
 INSTALL(FILES stk_config.h DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/STKClassic_config.h DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/diag/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/diag/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/diag/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/diag/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -82,5 +82,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/diag/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/diag/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/environment/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/environment/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/environment/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/environment/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -80,4 +80,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-        ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/environment/)
+        ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/environment/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/parallel/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/parallel/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/parallel/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/parallel/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -61,4 +61,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/parallel/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/parallel/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/unit_test_support/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/unit_test_support/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/unit_test_support/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/unit_test_support/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -50,4 +50,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/unit_test_support/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/unit_test_support/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/use_cases/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/use_cases/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/use_cases/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/use_cases/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -49,4 +49,4 @@
   )
 
 #INSTALL(FILES ${HEADERS} DESTINATION
-#    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/use_cases/)
+#    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/use_cases/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_classic/stk_util/stk_util/util/CMakeLists.txt	2016-12-05 13:46:37.391361023 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_classic/stk_util/stk_util/util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -46,4 +46,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/util/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_expreval/stk_expreval/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_expreval/stk_expreval/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_expreval/stk_expreval/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_expreval/stk_expreval/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -77,4 +77,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_expreval)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_expreval)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_io/stk_io/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_io/stk_io/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_io/stk_io/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_io/stk_io/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -78,7 +78,7 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/)
+${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/)
 
 # AGS: I moved this to the bottom so that the link order was correct
 # with  -lstk_io_uti before -lstk_io.  I think the DEPLIBS added in
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_io/stk_io/util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_io/stk_io/util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_io/stk_io/util/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_io/stk_io/util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -76,4 +76,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_io/util/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_mesh/stk_mesh/base/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_mesh/stk_mesh/base/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_mesh/stk_mesh/base/CMakeLists.txt	2016-12-05 13:46:37.421344676 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_mesh/stk_mesh/base/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -84,14 +84,13 @@
 TRIBITS_CONFIGURE_FILE(${PACKAGE_NAME}_config.h)
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/base/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/base/)
 
 INSTALL(FILES ${HEADERS_IMPL} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/)
 
 INSTALL(FILES ${HEADERS_ELEMGRAPH} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/elementGraph)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/baseImpl/elementGraph)
 
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE_NAME}_config.h DESTINATION
-        ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
-
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_mesh/stk_mesh/fixtures/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_mesh/stk_mesh/fixtures/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_mesh/stk_mesh/fixtures/CMakeLists.txt	2016-12-05 13:46:37.421344676 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_mesh/stk_mesh/fixtures/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -71,4 +71,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_search/stk_search/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_search/stk_search/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_search/stk_search/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_search/stk_search/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -69,4 +69,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_search_util/stk_search_util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_search_util/stk_search_util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_search_util/stk_search_util/CMakeLists.txt	2016-12-05 13:46:37.431339227 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_search_util/stk_search_util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -68,5 +68,5 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search_util)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_search_util)
     
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_topology/stk_topology/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_topology/stk_topology/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_topology/stk_topology/CMakeLists.txt	2016-12-05 13:46:37.401355574 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_topology/stk_topology/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -71,7 +71,7 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_topology/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_topology/)
 
 INSTALL(FILES ${DETAIL_HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_topology/topology_detail)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_topology/topology_detail)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_transfer/stk_transfer/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_transfer/stk_transfer/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_transfer/stk_transfer/CMakeLists.txt	2016-12-05 13:46:37.431339227 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_transfer/stk_transfer/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -5,4 +5,4 @@
 FILE(GLOB HEADERS *.hpp)
 
 INSTALL(FILES ${HEADERS} DESTINATION
-${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_transfer/)
+${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_transfer/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/CMakeLists.txt	2016-12-05 13:56:03.782564712 -0700
@@ -18,6 +18,6 @@
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 
 INSTALL(FILES stk_config.h stk_kokkos_macros.h DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/STK_Trilinos_config.h DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/diag/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/diag/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/diag/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/diag/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -75,5 +75,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/diag/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/diag/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/environment/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/environment/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/environment/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/environment/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -73,4 +73,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-        ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/environment/)
+        ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/environment/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/parallel/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/parallel/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/parallel/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/parallel/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -92,4 +92,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/parallel/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/parallel/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/registry/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/registry/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/registry/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/registry/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -70,5 +70,5 @@
 
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/registry/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/registry/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/unit_test_support/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/unit_test_support/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/unit_test_support/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/unit_test_support/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -78,4 +78,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/unit_test_support/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/unit_test_support/)
diff -Naur trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/util/CMakeLists.txt trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/util/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/stk/stk_util/stk_util/util/CMakeLists.txt	2016-12-05 13:46:37.411350125 -0700
+++ trilinos-12.10.1-Source/packages/stk/stk_util/stk_util/util/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -75,4 +75,4 @@
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/util/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_util/util/)
diff -Naur trilinos-12.10.1-Source.orig/packages/teuchos/kokkoscompat/src/CMakeLists.txt trilinos-12.10.1-Source/packages/teuchos/kokkoscompat/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/teuchos/kokkoscompat/src/CMakeLists.txt	2016-12-05 13:46:37.371371921 -0700
+++ trilinos-12.10.1-Source/packages/teuchos/kokkoscompat/src/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -27,7 +27,7 @@
 # FIXME (mfh 19 Dec 2014) We should rely on TriBITS to handle
 # installation.
 
-SET(TRILINOS_INCDIR ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
+SET(TRILINOS_INCDIR ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR})
 
 INSTALL(FILES ${HEADERS} DESTINATION ${TRILINOS_INCDIR}/)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/trios/libraries/support/CMakeLists.txt trilinos-12.10.1-Source/packages/trios/libraries/support/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/trios/libraries/support/CMakeLists.txt	2016-12-05 13:46:37.531284737 -0700
+++ trilinos-12.10.1-Source/packages/trios/libraries/support/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -168,6 +168,6 @@
 
 ########### install targets ###############
 
-# INSTALL(FILES ${MODULES} DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+# INSTALL(FILES ${MODULES} DESTINATION include)
 
 TRIBITS_SUBPACKAGE_POSTPROCESS()
diff -Naur trilinos-12.10.1-Source.orig/packages/zoltan/src/CMakeLists.txt trilinos-12.10.1-Source/packages/zoltan/src/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/zoltan/src/CMakeLists.txt	2016-12-05 13:46:37.231448208 -0700
+++ trilinos-12.10.1-Source/packages/zoltan/src/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -678,5 +678,5 @@
   IMPORTEDLIBS ${IMPORTEDLIBS}
   )
 
-INSTALL(FILES ${MODULES} DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+INSTALL(FILES ${MODULES} DESTINATION include)
 
diff -Naur trilinos-12.10.1-Source.orig/packages/zoltan/src/fdriver/CMakeLists.txt trilinos-12.10.1-Source/packages/zoltan/src/fdriver/CMakeLists.txt
--- trilinos-12.10.1-Source.orig/packages/zoltan/src/fdriver/CMakeLists.txt	2016-12-05 13:46:37.231448208 -0700
+++ trilinos-12.10.1-Source/packages/zoltan/src/fdriver/CMakeLists.txt	2016-12-05 13:53:32.275166500 -0700
@@ -33,6 +33,6 @@
 
 #  What is a good location for executables zdrive and zCPPdrive?
 #  When should they be moved there?  During install?  Or build?
-#INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/zfdrive.exe DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
+#INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/zfdrive.exe DESTINATION bin)
 
 ENDIF()
