diff --git a/packages/stk/stk_unit_tests/stk_mesh_fixtures/CMakeLists.txt b/packages/stk/stk_unit_tests/stk_mesh_fixtures/CMakeLists.txt
index 8f9fa8f..e57d25f 100644
--- a/packages/stk/stk_unit_tests/stk_mesh_fixtures/CMakeLists.txt
+++ b/packages/stk/stk_unit_tests/stk_mesh_fixtures/CMakeLists.txt
@@ -71,4 +71,4 @@ TRIBITS_ADD_LIBRARY(
   )
 
 INSTALL(FILES ${HEADERS} DESTINATION
-    ${CMAKE_INSTALL_PREFIX}/${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
+    ${${PROJECT_NAME}_INSTALL_INCLUDE_DIR}/stk_mesh/fixtures/)
