diff --git a/TNear.h b/TNear.h
index d848b61..a5d255e 100644
--- a/TNear.h
+++ b/TNear.h
@@ -735,7 +735,7 @@ void BelongsToPoints( const T& t1, const T& t2, ContainerType& group1, Container
 {
     group1.clear();
     group2.clear();
-    CNearTree<T>::iterator it;
+    typename CNearTree<T>::iterator it;
 
     for ( it=this->begin( ); it!=this->end( ); ++it )
     {
@@ -756,11 +756,11 @@ void SeparateByRadius( const DistanceType radius, const T& tProbe, ContainerType
 {
     inside.clear();
     outside.clear();
-    CNearTree<T>::iterator it;
+    typename CNearTree<T>::iterator it;
 
     for ( it=this->begin( ); it!=this->end( ); ++it )
     {
-        if( DistanceBetween( (*it), probe ) > radius )
+        if( DistanceBetween( (*it), tProbe ) > radius )
         {
             inside.insert( inside.end( ), (*it) );
         }
