--- CNearTreeTest.cpp	2009-06-08 04:54:29.000000000 +0200
+++ CNearTreeTest.cpp.new	2009-07-25 12:36:33.000000000 +0200
@@ -47,6 +47,7 @@
 #include <cmath>
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 
 #ifndef USE_LOCAL_HEADERS
 #include <TNear.h>
--- TNear.h	2009-06-08 04:54:29.000000000 +0200
+++ TNear.h.new	2009-07-25 12:39:09.000000000 +0200
@@ -272,6 +272,7 @@
 #define TRIANG(a,b,c) (  (((b)+(c))-(a) >= 0))
 #endif
 
+#include <algorithm>
 
 //=======================================================================
 // CNearTree is the root class for the neartree. The actual data of the
