diff -U 3 -dHrN polymake-2.11.orig/Makefile polymake-2.11/Makefile
--- polymake-2.11.orig/Makefile	2011-12-23 15:51:41.762207359 +0100
+++ polymake-2.11/Makefile	2011-12-23 15:52:13.889973019 +0100
@@ -134,11 +134,11 @@
 	@case "$$(uname -s)" in \
 	 *linux*|*Linux*) \
 	   for d in linux32 linux64; do \
-	     ${PERL} ${INSTALL_PL} -m ${DirMask} -U external/jreality/jni/$$d ${InstallArch}/lib/jreality/jni/$$d; \
+		true; \
 	   done; \
 	   ;; \
          Darwin*|darwin*) \
-	   ${PERL} ${INSTALL_PL} -m ${DirMask} -U external/jreality/jni/macosx ${InstallArch}/lib/jreality/jni/macosx; \
+		true; \
 	   ;; \
 	 esac
 	[ -d ${InstallBin} ] || $(call _CreateDir,${InstallBin})
@@ -166,7 +166,6 @@
 	@+$(call _MakeApps, install-src)
 	$(if ${.svn},${CallPolymake} --no-config --script filter_rules -m ${InstallTop} ${Apps})
 	[ ! -d java_build ] || $(call _InstallSubdir,java_build,-X de)
-	[ ! -d java_build ] || $(call _InstallSubdirRenamed,external/jreality/lib,java_build/jreality/lib)
 	$(call _InstallSubdir,povray)
 	$(call _InstallSubdir,scripts)
 	$(call _InstallSubdir,xml)
