--- ../vmd-1.8.6-orig/configure	2007-04-14 17:08:43.000000000 -0500
+++ vmd-1.8.6/configure	2007-04-14 17:30:28.000000000 -0500
@@ -13,10 +13,10 @@
 $install_name = "vmd";
 
 # Directory where VMD startup script is installed, should be in users' paths.
-$install_bin_dir="/usr/local/bin";
+$install_bin_dir="gentoo-bindir";
 
 # Directory where VMD files and executables are installed
-$install_library_dir="/usr/local/lib/$install_name";
+$install_library_dir="gentoo-libdir/$install_name";
 
 
 # optionally override hard-coded defaults above with environment variables
@@ -123,29 +123,6 @@
 # Configure script code begins here
 ############################################################################
 
-## Test for existence of a properly compiled plugin tree
-if (!(-e "plugins")) {
-  print "\n";
-  print "VMD plugin directory not found.\n";
-  print "Please compile the VMD plugins and do 'make distrib' to copy them\n";
-  print "into a 'plugins' directory linked or located in this directory.\n";
-  print "Be sure your PLUGINDIR environment variable is set before running\n";
-  print "'make distrib' in the plugin build area.\n";
-  print "\n";
-  die "Halting configuration until VMD plugins are built.\n";
-} else {
-  if (!(-e "plugins/include/vmdplugin.h")) {
-    print "\n";
-    print "VMD plugin include files or directory not found.\n";
-    print "Please compile the VMD plugins and do 'make distrib' to copy them\n";
-    print "into a 'plugins' directory linked or located in this directory.\n";
-    print "Be sure your PLUGINDIR environment variable is set before running\n";
-    print "'make distrib' in the plugin build area.\n";
-    print "\n";
-    die "Halting configuration until VMD plugins are built.\n";
-  }
-} 
-
 #################### Parse command line options   ###########
 # list of allowed architectures
 @archlist=('IRIX6', 'IRIX6_64', 'FREEBSD', 'HPUX11', 'AIX4', 'AIX5', 'AIX5_64', 'LINUX', 'LINUXALPHA', 'LINUXAMD64', 'LINUXIA64', 'LINUXPPC', 'MACOSX', 'MACOSXX86', 'TRU64', 'SOLARIS2', 'SOLARIS2_64', 'SOLARISX86', 'SOLARISX86_64', 'WIN32', 'WIN64');
@@ -423,8 +400,8 @@
 
 ################ Plugin options
 #
-$plugin_dir	= "../plugins";
-$molfile_dir = "$plugin_dir/$config_arch/molfile";
+$plugin_dir    = "gentoo-plugindir";
+$molfile_dir = "$plugin_dir/compile/lib_LINUX/molfile";
 $plugin_include = "-I$plugin_dir/include -I$molfile_dir";
 $plugin_library = "-L$molfile_dir";
 $plugin_libs    = "-lmolfile_plugin";
@@ -529,8 +506,8 @@
 ################ FLTK GUI
 $fltk_defines     = "-DVMDGUI -DVMDFLTK";
 $fltk_dir         = "$vmd_library_dir/fltk";
-$fltk_include     = "-I$fltk_dir/include";
-$fltk_library     = "-L$fltk_dir/$config_arch";
+$fltk_include     = "-Igentoo-fltk-include";
+$fltk_library     = "-Lgentoo-fltk-libs";
 $fltk_libs        = "-lfltk -lX11";
 #@fltk_cc          = ('forms_ui.c');
 @fltk_cu          = ();
@@ -886,8 +863,8 @@
 #######################
 $netcdf_defines     = "";
 $netcdf_dir         = "$vmd_library_dir/netcdf";
-$netcdf_include     = "-I$netcdf_dir/include";
-$netcdf_library     = "-L$netcdf_dir/lib_$config_arch";
+$netcdf_include     = "-Igentoo-netcdf-include";
+$netcdf_library     = "-Lgentoo-netcdf-libs";
 $netcdf_libs        = "-lnetcdf";
 @netcdf_cc          = ();
 @netcdf_cu          = ();
@@ -954,9 +931,9 @@
 ###################
 $python_defines     = "-DVMDPYTHON";
 $python_dir         = "$vmd_library_dir/python";
-$python_include     = "-I$python_dir/lib_$config_arch/include/python2.2";
-$python_library     = "-L$python_dir/lib_$config_arch/lib/python2.2/config";
-$python_libs        = "-lpython2.2 -lpthread";
+$python_include     = "-Igentoo-python-include";
+$python_library     = "-Lgentoo-python-lib";
+$python_libs        = "-lgentoo-python-link -lpthread";
 @python_h           = ('PythonTextInterp.h',
                        'VMDTkinterMenu.h',
 		       'py_commands.h',
@@ -1555,8 +1532,8 @@
     $opengl_libs        = "-lGL -lGLU"; 
     # $opengl_libs      = "-L/usr/X11R6/lib -lGL -lGLU -lXext -lX11";
 
-    $arch_cc          = "gcc";
-    $arch_ccpp        = "g++";
+    $arch_cc          = "gentoo-gcc";
+    $arch_ccpp        = "gentoo-g++";
     # test hack to statically link libstd++
     # $arch_ccpp        = "gcc";
     $arch_depend_flag = "-MM";
@@ -1565,8 +1542,8 @@
     $arch_shcppopts   = "-fPIC";
     $arch_shldopts    = "";
 
-    $arch_opt_flag    = "-m32 -fno-for-scope -Wno-deprecated -Wall -O3";
-    $arch_copts       = "-m32 -Wall -O3";
+    $arch_opt_flag    = "gentoo-cflags";
+    $arch_copts       = "gentoo-cflags";
     if ($config_swig) {
       $swig_cflags     .= " -fno-strict-aliasing";
     }
