--- pymol-1.2_rc1/setup.py.old  2009-07-17 00:05:10.339203020 -0500
+++ pymol-1.2_rc1/setup.py      2009-07-17 00:09:03.375274561 -0500
@@ -138,6 +138,7 @@
                 ("_PYMOL_INLINE",None),
                 ("_PYMOL_FREETYPE",None),
                 ("_PYMOL_LIBPNG",None),
+                ("_PYMOL_OPENGL_SHADERS",None),
 # Numeric Python support                    
 #                ("_PYMOL_NUMPY",None),
 # VMD plugin support   
