diff -up freefem++-2.23/src/Graphics/getprog-unix.hpp.gcc43 freefem++-2.23/src/Graphics/getprog-unix.hpp
--- freefem++-2.23/src/Graphics/getprog-unix.hpp.gcc43	2007-12-11 22:37:39.000000000 +0100
+++ freefem++-2.23/src/Graphics/getprog-unix.hpp	2008-02-20 21:50:16.000000000 +0100
@@ -1,3 +1,4 @@
+#include <cstring>
 
 extern long verbosity;
 
diff -up freefem++-2.23/src/Graphics/sansrgraph.cpp.gcc43 freefem++-2.23/src/Graphics/sansrgraph.cpp
diff -up freefem++-2.23/src/Algo/LineSearch.hpp.gcc43 freefem++-2.23/src/Algo/LineSearch.hpp
--- freefem++-2.23/src/Algo/LineSearch.hpp.gcc43	2006-04-03 19:23:33.000000000 +0200
+++ freefem++-2.23/src/Algo/LineSearch.hpp	2008-02-21 17:39:08.000000000 +0100
@@ -30,7 +30,7 @@ public:
 	typedef M Mat;
 	typedef P Param;
 	typedef V Vect;
-	typedef  NRJ<Param,Vect,VMat,Real> NRJ;
+	typedef  NRJ<Param,Vect,VMat,Real> lsNRJ;
 
 private:
   //ne sert que si on calcule le gradient de mani?re num?rique
@@ -57,7 +57,7 @@ protected:
   // c'est du meme type que le gradient
   Vect* step;
   //pointer to the nrj
-  NRJ *	nrj;
+  lsNRJ *	nrj;
   
   void appendSearchNumber();
   
@@ -65,7 +65,7 @@ public:
   
   //a constructor with pointer to the nrj and
   //to the step if numerical gradient is computed
-  LineSearch(NRJ* f,Vect* interval = NULL);
+  LineSearch(lsNRJ* f,Vect* interval = NULL);
   virtual ~LineSearch();
 
   // point de d?part et direction de descente + des param?tres
@@ -92,7 +92,7 @@ public:
 };
 
 template <class P, class V,class M,class VM,class R>
-LineSearch<P,V,M,VM,R>::LineSearch(NRJ* p, Vect* interval )
+LineSearch<P,V,M,VM,R>::LineSearch(lsNRJ* p, Vect* interval )
 {
    iterMax 	= 	MAX_IT_LS;
    iterNum 	= 	0;
diff -up freefem++-2.23/src/Algo/CubicLS.hpp.gcc43 freefem++-2.23/src/Algo/CubicLS.hpp
--- freefem++-2.23/src/Algo/CubicLS.hpp.gcc43	2006-04-03 19:23:32.000000000 +0200
+++ freefem++-2.23/src/Algo/CubicLS.hpp	2008-02-21 17:39:08.000000000 +0100
@@ -31,12 +31,12 @@ class CubicLineSearch : public LS
   typedef typename LS::Vect Vect;
   typedef typename LS::VMat VMat;
   typedef LS  LineSearch;
-  typedef NRJ<Param,Vect,VMat,Real> NRJ;
+  typedef NRJ<Param,Vect,VMat,Real> clsNRJ;
 public:
   //a constructor with the default delta
-  CubicLineSearch(NRJ* f, int iter);
+  CubicLineSearch(clsNRJ* f, int iter);
   //a constructor with the specified delta
-  CubicLineSearch(NRJ* f, int iter, Vect* delta);
+  CubicLineSearch(clsNRJ* f, int iter, Vect* delta);
    /*    The parameter $delta$ is not used by the line search 
          itself. Rather it is used in the numerical computation
          of the derivatives using centered differences. For
@@ -65,13 +65,13 @@ public:
 };
 
 template <class LS>
-CubicLineSearch<LS>::CubicLineSearch(NRJ* f, int it) 
+CubicLineSearch<LS>::CubicLineSearch(clsNRJ* f, int it) 
 : LS(f)
 {	this->iterMax	=	it;
 }
 
 template <class LS>
-CubicLineSearch<LS>::CubicLineSearch(NRJ* f, int it, Vect* interval)
+CubicLineSearch<LS>::CubicLineSearch(clsNRJ* f, int it, Vect* interval)
 : LS(f,interval)
 {	this->iterMax	=	it;
 }
diff -up freefem++-2.23/src/ide/testhighlight.cpp.gcc43 freefem++-2.23/src/ide/testhighlight.cpp
--- freefem++-2.23/src/ide/testhighlight.cpp.gcc43	2007-04-20 17:18:59.000000000 +0200
+++ freefem++-2.23/src/ide/testhighlight.cpp	2008-02-21 17:39:08.000000000 +0100
@@ -33,6 +33,7 @@ using namespace std;
 #include "hl_lexyacc.hpp"
 #include <iostream>
 #include <fstream>
+#include <cstdlib>
 
 extern int yydebug;
 
diff -up freefem++-2.23/src/ide/highlight.cpp.gcc43 freefem++-2.23/src/ide/highlight.cpp
--- freefem++-2.23/src/ide/highlight.cpp.gcc43	2006-09-29 22:30:15.000000000 +0200
+++ freefem++-2.23/src/ide/highlight.cpp	2008-02-21 17:39:08.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include <iostream>
 #include <cassert>
+#include <cstring>
 using namespace std;
 #include "highlight.hpp"
 #include "hl_lexyacc.hpp"
diff -up freefem++-2.23/src/ide/commands.cpp.gcc43 freefem++-2.23/src/ide/commands.cpp
--- freefem++-2.23/src/ide/commands.cpp.gcc43	2006-09-25 17:36:43.000000000 +0200
+++ freefem++-2.23/src/ide/commands.cpp	2008-02-21 17:39:08.000000000 +0100
@@ -32,6 +32,7 @@
 // together, or through sockets if they are separated.
 
 #include <FL/Fl.H>
+#include <cstdlib>
 #include "commands.hpp"
 #include "editor.hpp"
 #include "draw.hpp"
diff -up freefem++-2.23/src/ide/socket.cpp.gcc43 freefem++-2.23/src/ide/socket.cpp
--- freefem++-2.23/src/ide/socket.cpp.gcc43	2006-09-29 22:30:15.000000000 +0200
+++ freefem++-2.23/src/ide/socket.cpp	2008-02-21 17:39:08.000000000 +0100
@@ -33,6 +33,7 @@
 
 #include <iostream>
 #include <cassert>
+#include <cstring>
 #include "socket.hpp"
 using namespace std;
 
diff -up freefem++-2.23/src/ide/spawn.cpp.gcc43 freefem++-2.23/src/ide/spawn.cpp
--- freefem++-2.23/src/ide/spawn.cpp.gcc43	2006-09-29 22:30:15.000000000 +0200
+++ freefem++-2.23/src/ide/spawn.cpp	2008-02-21 17:39:08.000000000 +0100
@@ -35,6 +35,7 @@
 #define SPAWN_HPP
 
 #include <iostream>
+#include <cstdlib>
 using namespace std;
 #include <cassert>
 #include <FL/Fl.H>
diff -up freefem++-2.23/src/fflib/Serialize.hpp.gcc43 freefem++-2.23/src/fflib/Serialize.hpp
--- freefem++-2.23/src/fflib/Serialize.hpp.gcc43	2008-01-13 11:17:49.000000000 +0100
+++ freefem++-2.23/src/fflib/Serialize.hpp	2008-02-20 20:57:33.000000000 +0100
@@ -28,6 +28,8 @@
 #ifndef SERIALEZE_HPP_
 #define SERIALEZE_HPP_
 
+#include <cstring>
+
 struct MPIrank;
  class Serialize {  
    // we store a refcounter in the pointer p a adresse p-sizeof(long)
diff -up freefem++-2.23/src/femlib/fem.hpp.gcc43 freefem++-2.23/src/femlib/fem.hpp
--- freefem++-2.23/src/femlib/fem.hpp.gcc43	2008-01-17 18:00:42.000000000 +0100
+++ freefem++-2.23/src/femlib/fem.hpp	2008-02-21 17:39:57.000000000 +0100
@@ -5,6 +5,7 @@
 #include "Serialize.hpp"
 // some usefull function 
 
+#include <algorithm>
 
 template<class K> class KN_;
 
@@ -132,7 +133,6 @@ inline R2 Maxc(const R3 & A,const R3& B,
 // def de numerotation dans un triangles direct sens (trigo)
 // the edge is oposite of the vertex
 ////  [3] is a edge
-#include <algorithm>
 //#include <Functional>
 struct SortedTriplet {
     static const int  empty = -1;
diff -up freefem++-2.23/src/femlib/MatriceCreuse_tpl.hpp.gcc43 freefem++-2.23/src/femlib/MatriceCreuse_tpl.hpp
--- freefem++-2.23/src/femlib/MatriceCreuse_tpl.hpp.gcc43	2007-12-12 18:12:37.000000000 +0100
+++ freefem++-2.23/src/femlib/MatriceCreuse_tpl.hpp	2008-02-20 19:37:28.000000000 +0100
@@ -4,6 +4,7 @@
 #include <set>
 #include <list>
 #include <map>
+#include <limits>
 #endif
 
 #ifndef __MWERKS__
