diff -urN freefem++-2.24-2.old/configure.ac freefem++-2.24-2.new/configure.ac
--- freefem++-2.24-2.old/configure.ac	2008-03-27 13:22:17.000000000 +0100
+++ freefem++-2.24-2.new/configure.ac	2008-10-26 16:46:21.000000000 +0100
@@ -331,10 +331,7 @@
 
 if test "$ff_mingw" = no && test "$with_x" != "no";
 then
-	if test -d /usr/X11R6
-	then
-		ff_x11=yes
-	fi
+	ff_x11=yes
 fi
 
 AC_MSG_RESULT($ff_x11)
