diff -urN freefem++-3.2.old/configure.ac freefem++-3.2.new/configure.ac
--- freefem++-3.2.old/configure.ac	2009-05-13 16:11:44.000000000 +0200
+++ freefem++-3.2.new/configure.ac	2009-05-24 17:36:34.000000000 +0200
@@ -403,7 +403,12 @@
 ff_glx_ok=no
 ff_agl_ok=no
 ff_libs="$LIBS"
+AC_ARG_WITH(flib,[  --with-flib= the fortran library ])
+AC_ARG_ENABLE([opengl],AS_HELP_STRING([--enable-opengl],
+                                      [Enable/disable support for OpenGL]))
+
 # echo "  .... LIBS = $LIBS .... " 
+if test x$enable_opengl != xno; then
 # GL libraries seem to depend on some dlopen capabilities
 if test "$enable_static" != yes -a "$ff_glut_ok" != yes
 then
@@ -476,6 +481,7 @@
     
 
 fi
+fi
 
 if test "$ff_glut_ok" = yes
 then
