diff -urN freefem++-3.2.old/regtests.sh freefem++-3.2.new/regtests.sh
--- freefem++-3.2.old/regtests.sh	2008-11-25 17:21:06.000000000 +0100
+++ freefem++-3.2.new/regtests.sh	2009-05-24 16:51:49.000000000 +0200
@@ -76,7 +76,7 @@
     if [ $mpich -ne 0 ] ; then
     dotest "mpirun -np $nprocs -machinefile machinefile ${PROGLOC}/mpi/FreeFem++-mpi${EXEEXT}" mpi $script
     else
-    lamboot 
+	[[ -f "$(which lamboot 2>/dev/null)" ]] && lamboot	
      dotest "mpirun -np $nprocs ${PROGLOC}/mpi/FreeFem++-mpi${EXEEXT}" mpi $script
     fi
 fi
