diff -urN freefem++-3.2.old/download/fftw/Makefile freefem++-3.2.new/download/fftw/Makefile
--- freefem++-3.2.old/download/fftw/Makefile	2009-04-30 09:52:57.000000000 +0200
+++ freefem++-3.2.new/download/fftw/Makefile	2009-05-24 16:43:03.000000000 +0200
@@ -40,6 +40,8 @@
 $(PKGDIR)/$(PACKAGE):
 	cd $(PKGDIR); $(WGET)  $(SERVER)/$(PACKAGE)
 
+clean: clean-local
+
 clean-local:
 	-rm -rf fftw-* FAIT 
 
