diff -Naurp cudatoolkit_orig/include/cuda.h cudatoolkit/include/cuda.h
--- cudatoolkit_orig/include/cuda.h	2009-04-13 10:39:12.000000000 +0200
+++ cudatoolkit/include/cuda.h	2009-05-09 17:28:25.000000000 +0200
@@ -86,7 +86,7 @@ typedef enum CUctx_flags_enum {
     CU_CTX_SCHED_MASK  = 0x3,
     CU_CTX_BLOCKING_SYNC = 4,   ///< Use blocking synchronization
     CU_CTX_MAP_HOST = 8,        ///< Support mapped pinned allocations
-    CU_CTX_FLAGS_MASK  = 0xf,
+    CU_CTX_FLAGS_MASK  = 0xf
 } CUctx_flags;
 
 /**
@@ -94,7 +94,7 @@ typedef enum CUctx_flags_enum {
  */
 typedef enum CUevent_flags_enum {
     CU_EVENT_DEFAULT       = 0, ///< Default event flag
-    CU_EVENT_BLOCKING_SYNC = 1, ///< Event uses blocking synchronization
+    CU_EVENT_BLOCKING_SYNC = 1  ///< Event uses blocking synchronization
 } CUevent_flags;
 
 /**
@@ -117,7 +117,7 @@ typedef enum CUarray_format_enum {
 typedef enum CUaddress_mode_enum {
     CU_TR_ADDRESS_MODE_WRAP = 0,    ///< Wrapping address mode
     CU_TR_ADDRESS_MODE_CLAMP = 1,   ///< Clamp to edge address mode
-    CU_TR_ADDRESS_MODE_MIRROR = 2,  ///< Mirror address mode
+    CU_TR_ADDRESS_MODE_MIRROR = 2   ///< Mirror address mode
 } CUaddress_mode;
 
 /**
diff -Naurp cudatoolkit_orig/include/driver_types.h cudatoolkit/include/driver_types.h
--- cudatoolkit_orig/include/driver_types.h	2009-04-13 10:39:06.000000000 +0200
+++ cudatoolkit/include/driver_types.h	2009-05-09 17:31:17.000000000 +0200
@@ -140,7 +140,7 @@ enum cudaChannelFormatKind
   cudaChannelFormatKindSigned           =   0,      ///< Signed channel format
   cudaChannelFormatKindUnsigned         =   1,      ///< Unsigned channel format
   cudaChannelFormatKindFloat            =   2,      ///< Float channel format
-  cudaChannelFormatKindNone             =   3,      ///< No channel format
+  cudaChannelFormatKindNone             =   3       ///< No channel format
 };
 
 /**
