diff -ruN labplot.orig/CMakeLists.txt labplot/CMakeLists.txt
--- labplot.orig/CMakeLists.txt	2012-01-21 15:09:43.792562578 +0100
+++ labplot/CMakeLists.txt	2012-01-21 15:13:07.421573963 +0100
@@ -105,7 +105,7 @@
 
 ### OPJ #####################################
 FIND_LIBRARY(OPJ_LIBRARY
-  NAMES origin
+  NAMES origin origin2
 )
 IF(OPJ_LIBRARY)
 MESSAGE ("-- Found liborigin library: ${OPJ_LIBRARY}")
