diff -Naur med-2.3.1.orig/configure med-2.3.1/configure
--- med-2.3.1.orig/configure	2006-05-18 10:47:22.000000000 +0100
+++ med-2.3.1/configure	2008-04-22 01:18:14.000000000 +0100
@@ -1049,6 +1049,7 @@
 Optional Packages:
   --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
   --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
+  --with-med_int=<C type> Use <C type> for med_int
   --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
   --with-pic              try to use only PIC/non-PIC objects [default=use
                           both]
@@ -4635,6 +4636,21 @@
 
 
 
+## Guess where as med_int type is provided by --with-med_int option
+## or auto-detection must be used
+
+
+# Check whether --with-med_int was given.
+if test "${with_med_int+set}" = set; then
+  withval=$with_med_int;
+else
+  withval=no
+fi
+
+
+if test "x$withval" = "xno"
+then
+
   ## Guess the fortran int size
 
 
@@ -5718,6 +5734,11 @@
    { (exit 1); exit 1; }; }
   fi
 
+else
+  LONG_OR_INT="$withval"
+  { echo "$as_me:$LINENO: Using type $withval for med_int" >&5
+echo "$as_me: Using type $withval for med_int" >&6;}
+fi
 
 
 ## Check if user want to disable explicit error messages
