--- CalculiX/ccx_2.1/src/Makefile.orig	2009-08-28 09:53:59.000000000 +0200
+++ CalculiX/ccx_2.1/src/Makefile	2009-08-28 09:59:58.000000000 +0200
@@ -1,5 +1,5 @@
 
-CFLAGS = -Wall -O3  -I ../../../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE
+CFLAGS = -Wall -O3 $(SPOOLESINC) -DARCH="Linux" -DMATRIXSTORAGE $(ARPACK) $(USE_MT)
 FFLAGS = -Wall -O3 -fopenmp
 
 CC=cc
@@ -18,12 +18,7 @@
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../../../SPOOLES.2.2
-
-LIBS = \
-       $(DIR)/spooles.a \
-	../../../ARPACK/libarpack_INTEL.a \
-       -lm -lc
+LIBS = $(SPOOLESLIB) $(ARPACKLIB) $(BLAS) -lm -lc
 
 ccx_2.2: $(OCCXMAIN) ccx_2.2.a  $(LIBS)
 	./date.pl; $(CC) $(CFLAGS) -c ccx_2.2.c; $(FC) -Wall -O3 -o $@ $(OCCXMAIN) ccx_2.2.a  -lpthread  $(LIBS)
