diff -uhr CalculiX/ccx_1.8/src/Makefile CalculiX.mine/ccx_1.8/src/Makefile
--- CalculiX/ccx_1.8/src/Makefile	2008-07-31 15:24:46.000000000 -0400
+++ CalculiX.mine/ccx_1.8/src/Makefile	2008-08-06 08:28:02.000000000 -0400
@@ -1,6 +1,5 @@
-
-CFLAGS = -Wall -O  -I ../../../SPOOLES.2.2 -DARCH="Linux" -DSPOOLES -DARPACK -DMATRIXSTORAGE
-FFLAGS = -Wall -O 
+CFLAGS =  -Wall -O3 $(SPOOLESINC) -DARCH="Linux" -DMATRIXSTORAGE $(ARPACK) $(USE_MT)
+FFLAGS =  -Wall -O3
 
 CC=cc
 FC=gfortran
@@ -462,16 +461,10 @@
 OCCXC = $(SCCXC:.c=.o)
 OCCXMAIN = $(SCCXMAIN:.c=.o)
 
-DIR=../../../SPOOLES.2.2
-
-LIBS = \
-       $(DIR)/spooles.a \
-	../../../ARPACK/libarpack_INTEL.a \
-       -lm
+LIBS = $(SPOOLESLIB) $(ARPACKLIB) $(BLAS) -lm 
 
 ccx_1.8: $(OCCXMAIN) ccx_1.8.a  $(LIBS)
-	./date.pl; $(CC) $(CFLAGS) -c ccx_1.8.c; gfortran -Wall -O -o $@ $(OCCXMAIN) ccx_1.8.a  $(LIBS)
+	./date.pl; $(CC) $(CFLAGS) -c ccx_1.8.c; $(FC) -Wall -O3 -o $@ $(OCCXMAIN) ccx_1.8.a  $(LIBS)
 
 ccx_1.8.a: $(OCCXF) $(OCCXC)
 	ar vr $@ $?
-                                                                               
