diff -Nur scilab-5.1.1.orig/modules/helptools/macros/add_module_help_chapter.sci scilab-5.1.1/modules/helptools/macros/add_module_help_chapter.sci
--- scilab-5.1.1.orig/modules/helptools/macros/add_module_help_chapter.sci	2009-04-14 09:30:56.000000000 +0200
+++ scilab-5.1.1/modules/helptools/macros/add_module_help_chapter.sci	2009-04-14 18:07:23.000000000 +0200
@@ -19,10 +19,8 @@
 		path_addchapter = SCI+"/modules/"+modulename+"/help/"+getdefaultlanguage()+"/addchapter.sce";
 	end
 	
-	//try
+	if ( fileinfo(path_addchapter) <> [] ) then
 		exec(path_addchapter);
-	//catch
-	//    warning(gettext("Incorrect module path ")+path_addchapter);
-	//end
+	end
 	
 endfunction