commit 409faec413e7491450f51059af02b0469cbb01eb
Author: Sylvestre Ledru <sylvestre.ledru@scilab.org>
Date:   Fri Feb 19 23:42:38 2010 +0100

    full support of sparc64

diff --git a/scilab/bin/scilab b/scilab/bin/scilab
index c4d6548..7ec4bbc 100755
--- a/scilab/bin/scilab
+++ b/scilab/bin/scilab
@@ -371,9 +371,12 @@ fi
 					"ppc")
 						proc="ppc"
 						;;
-					"alpha")  
+					"alpha")
 						proc="alpha"
 						;;
+					"sparc64")
+						proc="sparc"
+						;;
 					*)
 						echo "Could not find the Java configuration for the model <${MODEL}>. Please contact us."
 						;;
