diff -Nur -x '*.orig' -x '*~' scilab-5.2.0/modules/fileio/sci_gateway/c/sci_copyfile.c scilab-5.2.0.new/modules/fileio/sci_gateway/c/sci_copyfile.c
--- scilab-5.2.0/modules/fileio/sci_gateway/c/sci_copyfile.c	2009-12-16 08:24:45.000000000 +0100
+++ scilab-5.2.0.new/modules/fileio/sci_gateway/c/sci_copyfile.c	2010-01-18 13:22:03.360717656 +0100
@@ -24,6 +24,7 @@
 #include "isdir.h"
 #include "splitpath.h"
 #include "freeArrayOfString.h"
+#include "charEncoding.h"
 /*--------------------------------------------------------------------------*/
 static wchar_t* getFilenameWithExtension(wchar_t* wcFullFilename);
 static void returnCopyFileResultOnStack(int ierr, char *fname);
diff -Nur -x '*.orig' -x '*~' scilab-5.2.0/modules/fileio/sci_gateway/c/sci_movefile.c scilab-5.2.0.new/modules/fileio/sci_gateway/c/sci_movefile.c
--- scilab-5.2.0/modules/fileio/sci_gateway/c/sci_movefile.c	2009-12-16 08:24:45.000000000 +0100
+++ scilab-5.2.0.new/modules/fileio/sci_gateway/c/sci_movefile.c	2010-01-18 13:21:09.540737969 +0100
@@ -24,6 +24,7 @@
 #include "isdir.h"
 #include "splitpath.h"
 #include "freeArrayOfString.h"
+#include "charEncoding.h"
 /*--------------------------------------------------------------------------*/
 static wchar_t* getFilenameWithExtensionForMove(wchar_t* wcFullFilename);
 static void returnMoveFileResultOnStack(int ierr, char *fname);
