--- bin/scilab.orig	2009-02-13 10:12:45.000000000 +0100
+++ bin/scilab	2009-02-13 10:14:06.000000000 +0100
@@ -439,7 +439,7 @@
 #export PVM_ARCH
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
+#LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -453,6 +453,7 @@
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:$LD_LIBRARY_PATH
 
 if test $SCIVERBOSE -ne 0; then
 	echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
--- bin/scilab-cli.orig	2009-02-13 10:12:45.000000000 +0100
+++ bin/scilab-cli	2009-02-13 10:14:06.000000000 +0100
@@ -439,7 +439,7 @@
 #export PVM_ARCH
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
+#LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -453,6 +453,7 @@
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:$LD_LIBRARY_PATH
 
 if test $SCIVERBOSE -ne 0; then
 	echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
--- bin/scilab-adv-cli.orig	2009-02-13 10:12:45.000000000 +0100
+++ bin/scilab-adv-cli	2009-02-13 10:14:06.000000000 +0100
@@ -439,7 +439,7 @@
 #export PVM_ARCH
 
 SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/lib64/scilab/
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
+#LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -453,6 +453,7 @@
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:$LD_LIBRARY_PATH
 
 if test $SCIVERBOSE -ne 0; then
 	echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
