commit aad7b644ccc97638598c2831d550edb51bdb13f1
Author: Sylvestre Ledru <sylvestre.ledru@scilab.org>
Date:   Mon Apr 27 15:16:20 2009 +0200

    Missing addchapter.sce

diff --git a/scilab/modules/call_scilab/help/en_US/addchapter.sce b/scilab/modules/call_scilab/help/en_US/addchapter.sce
new file mode 100644
index 0000000..29d4a54
--- /dev/null
+++ b/scilab/modules/call_scilab/help/en_US/addchapter.sce
@@ -0,0 +1,13 @@
+// Scilab ( http://www.scilab.org/ ) - This file is part of Scilab
+// Copyright (C) 2007-2008 - INRIA - Allan CORNET
+// Copyright (C) 2009 - DIGITEO
+// 
+// This file must be used under the terms of the CeCILL.
+// This source file is licensed as described in the file COPYING, which
+// you should have received as part of this distribution.  The terms
+// are also available at    
+// http://www.cecill.info/licences/Licence_CeCILL_V2-en.txt
+
+path = SCI+"/modules/call_scilab/help/en_US";
+add_help_chapter("call_scilab API",path,%T);
+clear path add_help_chapter;
