commit 9586bf474819f11734927d5138056c0c837b9bdb
Author: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date:   Fri Feb 19 23:44:51 2010 +0100

    Support of Renesas SH(sh4)

diff --git a/scilab/bin/scilab b/scilab/bin/scilab
index 7ec4bbc..f453f9a 100755
--- a/scilab/bin/scilab
+++ b/scilab/bin/scilab
@@ -377,6 +377,9 @@ fi
 					"sparc64")
 						proc="sparc"
 						;;
+					sh*)
+						proc="sh"
+						;;
 					*)
 						echo "Could not find the Java configuration for the model <${MODEL}>. Please contact us."
 						;;
diff --git a/scilab/m4/java.m4 b/scilab/m4/java.m4
index 469019b..918fbf7 100644
--- a/scilab/m4/java.m4
+++ b/scilab/m4/java.m4
@@ -446,6 +446,9 @@ AC_DEFUN([AC_JAVA_JNI_LIBS], [
 		  s390x) # s390 arch can also returns s390x
 		  machine=s390
 		  ;;
+	    sh*)
+	  	  machine=sh
+		  ;;
     esac
 
 	AC_MSG_LOG([Looking for JNI libs with $machine as machine hardware name])
