--- configure.ac.orig	2010-02-11 20:54:10.611709937 +0100
+++ configure.ac	2010-02-11 21:03:58.184516409 +0100
@@ -747,6 +747,48 @@
 				JROSETTA_ENGINE=$PACKAGE_JAR_FILE
 				AC_SUBST(JROSETTA_ENGINE)
 
+				# MathML rendering solution
+				# Used in both graphic & help 
+				AC_JAVA_CHECK_PACKAGE([jeuclid-core],[net.sourceforge.jeuclid.LayoutContext],[MathML rendering solution (at least version 3.1.X)])
+				JEUCLID_CORE=$PACKAGE_JAR_FILE
+				AC_SUBST(JEUCLID_CORE)
+
+				# xml.apache.org SVG Library (under mandriva for example)
+				AC_JAVA_CHECK_PACKAGE([batik-all],[org.apache.batik.parser.Parser],[Apache SVG Library])
+				BATIK=$PACKAGE_JAR_FILE
+
+				if test -z "$BATIK"; then
+					# Other other distribs
+					AC_JAVA_CHECK_PACKAGE([batik],[org.apache.batik.parser.Parser],[Apache SVG Library],"yes")
+					BATIK=$PACKAGE_JAR_FILE
+				fi
+				AC_SUBST(BATIK)
+
+				# XML to PDF/other Translator
+				AC_JAVA_CHECK_PACKAGE([fop],[org.apache.fop.pdf.PDFInfo],[XML to PDF Translator (fop)])
+				FOP=$PACKAGE_JAR_FILE
+				AC_SUBST(FOP)
+
+				# Commons I/O library
+				AC_JAVA_CHECK_PACKAGE([commons-io],[org.apache.commons.io.output.CountingOutputStream],[Commons I/O library])
+				COMMONS_IO=$PACKAGE_JAR_FILE
+				AC_SUBST(COMMONS_IO)
+
+				# XML graphics common
+				AC_JAVA_CHECK_PACKAGE([xmlgraphics-commons],[org.apache.xmlgraphics.util.Service],[Commons graphics library])
+				XMLGRAPHICS_COMMONS=$PACKAGE_JAR_FILE
+				AC_SUBST(XMLGRAPHICS_COMMONS)
+
+				# XML API EXT (conversion of a SVG => PNG)
+				AC_JAVA_CHECK_PACKAGE([xml-apis-ext],[org.w3c.dom.svg.SVGDocument],[XML Commons external code])
+				XML_APIS_EXT=$PACKAGE_JAR_FILE
+				AC_SUBST(XML_APIS_EXT)
+
+				# Avalon Framework (PDF)
+				AC_JAVA_CHECK_PACKAGE([avalon-framework],[org.apache.avalon.framework.configuration.ConfigurationException],[Common framework for Java server application])
+				AVALON_FRAMEWORK=$PACKAGE_JAR_FILE
+				AC_SUBST(AVALON_FRAMEWORK)
+
 				if test $HDF5_ENABLE = yes;  then
 				# HDF5 java lib
 				  AC_JAVA_CHECK_PACKAGE([jhdf5],[ncsa.hdf.hdf5lib.HDF5Constants],[HDF5 Java library])
@@ -1763,13 +1805,7 @@
 if test $HELP_ENABLE = yes; then
    echo "Documentation building configuration:"
    echo "  Docbook XSL path ....... = $DOCBOOK_ROOT"
-   echo "  Batik (SVG) ............ = $BATIK"
    echo "  Saxon XSLT ............. = $SAXON"
-   echo "  FOP (XML => PDF) ....... = $FOP"
-   echo "  Jeuclid (MathML) ....... = $JEUCLID_CORE"
-   echo "  Commons I/O ............ = $COMMONS_IO"
-   echo "  XML graphics commons ... = $XMLGRAPHICS_COMMONS"
-   echo "  XML commons external ... = $XML_APIS_EXT"
 else
    echo "No documentation generated"
 fi
@@ -1801,6 +1837,12 @@
 echo "  JOGL LIBS (JNI) ..... = $JOGL_LIBS"
 echo "  Gluegen ............. = $GLUEGEN_RT"
 echo "  Gluegen LIBS (JNI) .. = $GLUEGEN_RT_LIBS"
+echo "  Batik (SVG) ......... = $BATIK"
+echo "  FOP (XML => PDF) .... = $FOP"
+echo "  Jeuclid (MathML) .... = $JEUCLID_CORE"
+echo "  Commons I/O ......... = $COMMONS_IO"
+echo "  XML graphics commons. = $XMLGRAPHICS_COMMONS"
+echo "  XML commons external. = $XML_APIS_EXT"
 echo "  Jhall  .............. = $JHALL"
 echo "  Jrosetta (API) ...... = $JROSETTA_API"
 echo "  Jrosetta (Engine) ... = $JROSETTA_ENGINE"
--- m4/docbook.m4.orig	2010-02-11 20:39:12.014267405 +0100
+++ m4/docbook.m4	2010-02-11 20:53:47.786517217 +0100
@@ -32,53 +32,11 @@
 	if test -z "$DOCBOOK_ROOT"; then
 		AC_MSG_ERROR([Could not find the Docbook root directory. If you have installed it on your system and we haven't been able to find it. Please report a bug])
 	fi
-	# xml.apache.org SVG Library (under mandriva for example)
-	AC_JAVA_CHECK_PACKAGE([batik-all],[org.apache.batik.parser.Parser],[Apache SVG Library])
-	BATIK=$PACKAGE_JAR_FILE
-
-	if test -z "$BATIK"; then
-		# Other other distribs
-		AC_JAVA_CHECK_PACKAGE([batik],[org.apache.batik.parser.Parser],[Apache SVG Library],"yes")
-		BATIK=$PACKAGE_JAR_FILE
-	fi
-	AC_SUBST(BATIK)
-
 
 	# Saxon XSLT Processor
 	AC_JAVA_CHECK_PACKAGE([saxon],[com.icl.saxon.Loader],[Saxon XSLT Processor])
 	SAXON=$PACKAGE_JAR_FILE
 	AC_SUBST(SAXON)
-
-	# XML to PDF/other Translator
-	AC_JAVA_CHECK_PACKAGE([fop],[org.apache.fop.pdf.PDFInfo],[XML to PDF Translator (fop)])
-	FOP=$PACKAGE_JAR_FILE
-	AC_SUBST(FOP)
-
-	# MathML rendering solution
-	AC_JAVA_CHECK_PACKAGE([jeuclid-core],[net.sourceforge.jeuclid.LayoutContext],[MathML rendering solution (at least version 3.1.X)])
-	JEUCLID_CORE=$PACKAGE_JAR_FILE
-	AC_SUBST(JEUCLID_CORE)
-
-	# Commons I/O library
-	AC_JAVA_CHECK_PACKAGE([commons-io],[org.apache.commons.io.output.CountingOutputStream],[Commons I/O library])
-	COMMONS_IO=$PACKAGE_JAR_FILE
-	AC_SUBST(COMMONS_IO)
-
-	# XML graphics common
-	AC_JAVA_CHECK_PACKAGE([xmlgraphics-commons],[org.apache.xmlgraphics.util.Service],[Commons graphics library])
-	XMLGRAPHICS_COMMONS=$PACKAGE_JAR_FILE
-	AC_SUBST(XMLGRAPHICS_COMMONS)
-
-	# XML API EXT (conversion of a SVG => PNG)
-	AC_JAVA_CHECK_PACKAGE([xml-apis-ext],[org.w3c.dom.svg.SVGDocument],[XML Commons external code])
-	XML_APIS_EXT=$PACKAGE_JAR_FILE
-	AC_SUBST(XML_APIS_EXT)
-
-
-	# Avalon Framework (PDF)
-	AC_JAVA_CHECK_PACKAGE([avalon-framework],[org.apache.avalon.framework.configuration.ConfigurationException],[Common framework for Java server application])
-	AVALON_FRAMEWORK=$PACKAGE_JAR_FILE
-	AC_SUBST(AVALON_FRAMEWORK)
 	
 
 AC_SUBST(DOCBOOK_ROOT)
