diff -uNr scilab-5.5.2-orig/configure.ac scilab-5.5.2/configure.ac
--- a/configure.ac	2016-06-05 12:30:06.582735449 -0400
+++ b/configure.ac	2016-06-05 12:31:29.706708157 -0400
@@ -1024,7 +1024,7 @@
                     AC_SUBST(FREEHEP_IO)
 
                     # Freehep Util
-                    AC_JAVA_CHECK_PACKAGE([freehep-util],[org.freehep.util.StringUtilities],[Freehep Util])
+                    AC_JAVA_CHECK_PACKAGE([freehep-graphicsbase],[org.freehep.graphicsbase.util.UserProperties],[Freehep Util])
                     FREEHEP_UTIL=$PACKAGE_JAR_FILE
                     AC_SUBST(FREEHEP_UTIL)
                 fi
