diff -Naurp scilab-5.3.3-r1-orig/m4/java-thirdparty.m4 scilab-5.3.3-r1/m4/java-thirdparty.m4
--- scilab-5.3.3-r1-orig/m4/java-thirdparty.m4	2013-01-07 14:55:58.516374556 +0100
+++ scilab-5.3.3-r1/m4/java-thirdparty.m4	2013-01-07 16:19:21.256370075 +0100
@@ -38,9 +38,16 @@ AC_DEFUN([AC_JAVA_CHECK_VERSION_PACKAGE]
    AC_JAVA_TRY_COMPILE($2, [String minVersion="$4";
                             $6
                             System.out.println($5);
-                                  if (minVersion.compareTo($5) > 0) {
-                                   System.exit(-1);
-                                } ]
+                            String[] minV = minVersion.split("\\.");
+                            String[] curV = $5.split("\\.");
+                            for (int i=0; i<Math.max(minV.length,curV.length); i++) {
+                                int mE = i<minV.length ? Integer.parseInt(minV[i]) : 0;
+                                int cE = i<curV.length ? Integer.parseInt(curV[i]) : 0;
+                                if (mE < cE)
+                                    break;
+                                if (mE > cE)
+                                    System.exit(-1);
+                            } ]
                             , "yes", echo "yes" , AC_MSG_ERROR([Wrong version of $1. Expected at least $4. Found $STDOUT]))
    else
    AC_JAVA_TRY_COMPILE($2, [String minVersion="$4";
