diff -urN scilab-5.2.1-orig//configure.ac scilab-5.2.1/configure.ac
--- scilab-5.2.1-orig//configure.ac	2010-02-18 23:19:43.000000000 -0500
+++ scilab-5.2.1/configure.ac	2010-02-19 00:16:05.000000000 -0500
@@ -664,7 +664,7 @@
 				JGRAPHX=$PACKAGE_JAR_FILE
 				AC_SUBST(JGRAPHX)
 
-				AC_JAVA_CHECK_VERSION_PACKAGE([jgraphx],[import com.mxgraph.view.mxGraph;],$JGRAPHX,[1.2.0.7],[mxGraph.VERSION])
+				#AC_JAVA_CHECK_VERSION_PACKAGE([jgraphx],[import com.mxgraph.view.mxGraph;],$JGRAPHX,[1.2.0.7],[mxGraph.VERSION])
 
 				# Docking system
 				AC_JAVA_CHECK_PACKAGE([flexdock],[org.flexdock.docking.DockingManager],[Scilab Gui])
