--- m4/java.m4.orig	2014-02-11 23:43:03.030430258 -0500
+++ m4/java.m4	2014-02-11 23:43:54.008428766 -0500
@@ -839,7 +839,7 @@
       for jar in "$jardir/$1.jar" "$jardir/lib$1.jar" "$jardir/lib$1-java.jar" "$jardir/$1*.jar"; do
 
 # TODO check the behaviour when spaces
-        jars_resolved=`ls $jar 2>/dev/null`
+        jars_resolved=$(realpath $(ls $jar 2>/dev/null) 2>/dev/null)
         for jar_resolved in $jars_resolved; do # If several jars matches
           if test -e "$jar_resolved"; then
             export ac_java_classpath="$jar_resolved:$ac_java_classpath"
