Index: scilab-5.2.0/configure.ac
===================================================================
--- scilab-5.2.0.orig/configure.ac	2010-02-12 17:27:55.000000000 +0100
+++ scilab-5.2.0/configure.ac	2010-02-12 17:28:48.000000000 +0100
@@ -747,6 +747,33 @@
 				JROSETTA_ENGINE=$PACKAGE_JAR_FILE
 				AC_SUBST(JROSETTA_ENGINE)
 
+	# xml.apache.org SVG Library (under mandriva for example)
+	AC_JAVA_CHECK_PACKAGE([batik-all],[org.apache.batik.parser.Parser],[Apache SVG Library])
+	BATIK=$PACKAGE_JAR_FILE
+
+	if test -z "$BATIK"; then
+		# Other other distribs
+		AC_JAVA_CHECK_PACKAGE([batik],[org.apache.batik.parser.Parser],[Apache SVG Library],"yes")
+		BATIK=$PACKAGE_JAR_FILE
+	fi
+	AC_SUBST(BATIK)
+
+	# XML to PDF/other Translator
+	AC_JAVA_CHECK_PACKAGE([fop],[org.apache.fop.pdf.PDFInfo],[XML to PDF Translator (fop)])
+	FOP=$PACKAGE_JAR_FILE
+	AC_SUBST(FOP)
+
+	# MathML rendering solution
+	AC_JAVA_CHECK_PACKAGE([jeuclid-core],[net.sourceforge.jeuclid.LayoutContext],[MathML rendering solution (at least version 3.1.X)])
+	JEUCLID_CORE=$PACKAGE_JAR_FILE
+	AC_SUBST(JEUCLID_CORE)
+
+	# Avalon Framework (PDF)
+	AC_JAVA_CHECK_PACKAGE([avalon-framework],[org.apache.avalon.framework.configuration.ConfigurationException],[Common framework for Java server application])
+	AVALON_FRAMEWORK=$PACKAGE_JAR_FILE
+	AC_SUBST(AVALON_FRAMEWORK)
+
+
 				if test $HDF5_ENABLE = yes;  then
 				# HDF5 java lib
 				  AC_JAVA_CHECK_PACKAGE([jhdf5],[ncsa.hdf.hdf5lib.HDF5Constants],[HDF5 Java library])
Index: scilab-5.2.0/m4/docbook.m4
===================================================================
--- scilab-5.2.0.orig/m4/docbook.m4	2010-02-12 17:27:48.000000000 +0100
+++ scilab-5.2.0/m4/docbook.m4	2010-02-12 17:28:58.000000000 +0100
@@ -32,16 +32,6 @@
 	if test -z "$DOCBOOK_ROOT"; then
 		AC_MSG_ERROR([Could not find the Docbook root directory. If you have installed it on your system and we haven't been able to find it. Please report a bug])
 	fi
-	# xml.apache.org SVG Library (under mandriva for example)
-	AC_JAVA_CHECK_PACKAGE([batik-all],[org.apache.batik.parser.Parser],[Apache SVG Library])
-	BATIK=$PACKAGE_JAR_FILE
-
-	if test -z "$BATIK"; then
-		# Other other distribs
-		AC_JAVA_CHECK_PACKAGE([batik],[org.apache.batik.parser.Parser],[Apache SVG Library],"yes")
-		BATIK=$PACKAGE_JAR_FILE
-	fi
-	AC_SUBST(BATIK)
 
 
 	# Saxon XSLT Processor
@@ -49,16 +39,6 @@
 	SAXON=$PACKAGE_JAR_FILE
 	AC_SUBST(SAXON)
 
-	# XML to PDF/other Translator
-	AC_JAVA_CHECK_PACKAGE([fop],[org.apache.fop.pdf.PDFInfo],[XML to PDF Translator (fop)])
-	FOP=$PACKAGE_JAR_FILE
-	AC_SUBST(FOP)
-
-	# MathML rendering solution
-	AC_JAVA_CHECK_PACKAGE([jeuclid-core],[net.sourceforge.jeuclid.LayoutContext],[MathML rendering solution (at least version 3.1.X)])
-	JEUCLID_CORE=$PACKAGE_JAR_FILE
-	AC_SUBST(JEUCLID_CORE)
-
 	# Commons I/O library
 	AC_JAVA_CHECK_PACKAGE([commons-io],[org.apache.commons.io.output.CountingOutputStream],[Commons I/O library])
 	COMMONS_IO=$PACKAGE_JAR_FILE
@@ -74,11 +54,6 @@
 	XML_APIS_EXT=$PACKAGE_JAR_FILE
 	AC_SUBST(XML_APIS_EXT)
 
-
-	# Avalon Framework (PDF)
-	AC_JAVA_CHECK_PACKAGE([avalon-framework],[org.apache.avalon.framework.configuration.ConfigurationException],[Common framework for Java server application])
-	AVALON_FRAMEWORK=$PACKAGE_JAR_FILE
-	AC_SUBST(AVALON_FRAMEWORK)
 	
 
 AC_SUBST(DOCBOOK_ROOT)
