diff -urN scilab-5.5.0-orig/bin/scilab scilab-5.5.0/bin/scilab
--- scilab-5.5.0-orig/bin/scilab	2014-07-12 15:24:17.105686656 -0400
+++ scilab-5.5.0/bin/scilab	2014-07-12 15:25:29.307725409 -0400
@@ -647,17 +647,6 @@
 GTK_MODULES=$(echo $GTK_MODULES | sed s/gnomebreakpad//g)
 export GTK_MODULES
 
-# libEGL may be partially supported by the closed-source NVIDIA driver.
-# Until they release an EGL driver, we force the egl_glx driver.
-# http://bugzilla.scilab.org/show_bug.cgi?id=12940
-if test -x /usr/bin/glxinfo; then
-    NVIDIA_GLX_VENDOR=$(/usr/bin/glxinfo |grep "NVIDIA Corporation")
-    if test ! -z "$NVIDIA_GLX_VENDOR"; then
-        EGL_DRIVER=egl_glx
-        export EGL_DRIVER
-    fi
-fi
-
 if test $SCIVERBOSE -ne 0; then
     echo "SCI : $SCI"
     echo "SCIBIN : $SCIBIN"
