--- bin/scilab.orig	2013-10-28 01:08:06.623143085 -0400
+++ bin/scilab	2013-10-30 22:57:00.781289235 -0400
@@ -650,11 +650,13 @@
 # libEGL may be partially supported by the closed-source NVIDIA driver.
 # Until they release an EGL driver, we force the egl_glx driver.
 # http://bugzilla.scilab.org/show_bug.cgi?id=12940
-if test -x /usr/bin/glxinfo; then
-    NVIDIA_GLX_VENDOR=$(/usr/bin/glxinfo |grep "NVIDIA Corporation")
-    if test ! -z "$NVIDIA_GLX_VENDOR"; then
-        EGL_DRIVER=egl_glx
-        export EGL_DRIVER
+if test $SCILAB_MODE != "nw"; then
+    if test -x /usr/bin/glxinfo; then
+    	NVIDIA_GLX_VENDOR=$(/usr/bin/glxinfo |grep "NVIDIA Corporation")
+    	if test ! -z "$NVIDIA_GLX_VENDOR"; then
+            EGL_DRIVER=egl_glx
+            export EGL_DRIVER
+    	fi
     fi
 fi
 
