--- configure.ac.orig	2010-02-04 20:07:14.000000000 +0000
+++ configure.ac	2010-02-04 20:07:53.000000000 +0000
@@ -894,10 +894,6 @@
 
 AC_CHECK_FUNCS([pow] [sqrt] [finite] [floor])
 
-AC_CHECK_FUNC([exp10],,[
-				AC_DEFINE([log_10_],[2.3025850929940456840179914546844],[Provide a macro to do exp10])
-				AC_DEFINE([exp10(x)],[exp( (log_10_) * (x) )],[Provide a macro to do exp10])
-				])
 
 ### If isinf exists or not (used to not be the case under Solaris)
 ### See bug #4164
