--- a/util/pfam_runner.pl
+++ b/util/pfam_runner.pl
@@ -24,7 +24,7 @@
 my $workdir;
 my $verbose;
 my ($reuse,$pfam_out);
-my $CPU = 2;
+my $CPU = 1;
 
 my $usage =  <<_EOH_;
 
@@ -59,7 +59,7 @@
 # -h                                     print this option menu and quit
 # -v                                     verbose
 #
-# --CPU <int>                            number of threads to use; (default: 2)
+# --CPU <int>                            number of threads to use; (default: 1)
 #
 # --MPI                                  use MPI (via ffindex_apply_mpi)
 #
