--- TransDecoder-2.0.1/TransDecoder.LongOrfs.ori	2015-11-19 21:05:53.340219051 +0100
+++ TransDecoder-2.0.1/TransDecoder.LongOrfs	2015-11-19 21:20:44.870221380 +0100
@@ -64,7 +64,7 @@
 use TransDecoder::Fasta_reader;
 use TransDecoder::Longest_orf;
 
-my $UTIL_DIR = "$FindBin::RealBin/util";
+my $UTIL_DIR = "/usr/share/TransDecoder/util/";
 $ENV{PATH} = "$UTIL_DIR/bin:$ENV{PATH}";
 
 
--- TransDecoder-2.0.1/TransDecoder.Predict.ori	2015-11-19 21:06:04.280219080 +0100
+++ TransDecoder-2.0.1/TransDecoder.Predict	2015-11-19 21:21:22.560221479 +0100
@@ -52,7 +52,7 @@
 use TransDecoder::Fasta_reader;
 use TransDecoder::Longest_orf;
 
-my $UTIL_DIR = "$FindBin::RealBin/util";
+my $UTIL_DIR = "/usr/share/TransDecoder/util/";
 $ENV{PATH} = "$UTIL_DIR/bin:$ENV{PATH}";
 
 
