--- ./src5.1.3.orig/MED_SRC_5.1.3/src/ParaMEDMEM/MxN_Mapping.hxx	2010-03-07 19:42:49.442822826 +0100
+++ ./src5.1.3/MED_SRC_5.1.3/src/ParaMEDMEM/MxN_Mapping.hxx	2010-03-07 19:43:38.710252330 +0100
@@ -39,8 +39,8 @@
     void addElementFromSource(int distant_proc, int distant_elem);
     void prepareSendRecv();
     void sendRecv(MEDCouplingFieldDouble& field);
-    void sendRecv(double* field, MEDCouplingFieldDouble& field) const ;
-    void reverseSendRecv(double* field, MEDCouplingFieldDouble& field) const ;
+    void sendRecv(double* sendfield, MEDCouplingFieldDouble& field) const ;
+    void reverseSendRecv(double* recvfield, MEDCouplingFieldDouble& field) const ;
  
     //
     const std::vector<std::pair<int,int> >& getSendingIds() const { return _sending_ids; }
