--- src/MedClient/test/environ/Makefile.in_orig	2008-01-04 17:11:06.000000000 +0100
+++ src/MedClient/test/environ/Makefile.in	2008-01-04 17:12:08.000000000 +0100
@@ -29,6 +29,7 @@
 top_builddir=../../../..
 srcdir=@srcdir@
 VPATH=.:@srcdir@
+prefix=@prefix@
 
 BIN=	runEnvironTests 
 
@@ -53,11 +54,11 @@
 	chmod +x csh/*
 
 install:
-	mkdir -p @prefix@/Tests/environ
-	cp -rf ${BIN} csh @prefix@/Tests/environ
-	cd @prefix@/Tests/environ ; chmod u+x ${BIN}
-	cp -f ${top_builddir}/bin/runTestMedCorba @prefix@/bin
-	chmod +x @prefix@/bin/runTestMedCorba
+	mkdir -p ${prefix}/Tests/environ
+	cp -rf ${BIN} csh ${prefix}/Tests/environ
+	cd ${prefix}/Tests/environ ; chmod u+x ${BIN}
+	cp -f ${top_builddir}/bin/runTestMedCorba ${prefix}/bin
+	chmod +x ${prefix}/bin/runTestMedCorba
 
 ./%: @srcdir@/%.in
 	cd $(top_builddir) ; \
