--- ./src5.1.4.orig/MED_SRC_5.1.4/src/INTERPOLATION/create_mesh_interpolation.c	2010-03-13 13:14:50.721209673 +0100
+++ ./src5.1.4/MED_SRC_5.1.4/src/INTERPOLATION/create_mesh_interpolation.c	2010-03-13 13:15:35.420479255 +0100
@@ -21,7 +21,7 @@
 //
 #include <med.h>
 #include <string.h>
-
+#include <stdlib.h>
 
 /*****************************************************************************************************/
 
