--- src5.1.3.orig/MED_SRC_5.1.3/adm_local/unix/config_files/check_qt.m4	2010-03-06 21:00:58.281572535 +0100
+++ src5.1.3/MED_SRC_5.1.3/adm_local/unix/config_files/check_qt.m4	2010-03-06 21:03:34.110325466 +0100
@@ -333,7 +333,7 @@
 
   LIBS_old=$LIBS
   if test "x$QTDIR" = "x/usr" ; then
-    QT_LIB_DIR=""
+    QT_LIB_DIR="-L/usr/lib${LIB_LOCATION_SUFFIX}/qt4"
   elif test -d ${QTDIR}/lib; then
     QT_LIB_DIR="-L$QTDIR/lib"
   elif test -d ${QTDIR}/lib64; then
