--- src3.2.6/MED_SRC_3.2.6/src/MEDMEM/MEDMEM_MEDMEMchampLire.cxx	2007-11-26 23:27:29.000000000 +0100
+++ src3.2.6/MED_SRC_3.2.6/src/MEDMEM/MEDMEM_MEDMEMchampLire.cxx	2007-11-26 23:28:23.000000000 +0100
@@ -222,7 +222,7 @@
 	case MED_FLOAT64 :
 	  if ( _MEDdatasetNumLire(datagroup3,MED_NOM_CO,MED_FLOAT64,
 				  interlace,ncomp,numco,
-				  psize,pflmod,pfltab,ngauss,val)< 0)
+				  psize,pflmod,(med_size*)pfltab,ngauss,val)< 0)
 	    goto ERROR;
 	  break;
 
@@ -230,12 +230,12 @@
 #if defined(F77INT64)
 	  if ( _MEDdatasetNumLire(datagroup3,MED_NOM_CO,MED_INT64,
 				  interlace,ncomp,numco,
-				  psize,pflmod,pfltab,ngauss,val)< 0)
+				  psize,pflmod,(med_size*)pfltab,ngauss,val)< 0)
 	    goto ERROR;
 #else
 	  if ( _MEDdatasetNumLire(datagroup3,MED_NOM_CO,MED_INT32,
 				  interlace,ncomp,numco,
-				  psize,pflmod,pfltab,ngauss,val)< 0)
+				  psize,pflmod,(med_size*)pfltab,ngauss,val)< 0)
 	    goto ERROR;
 #endif
 	  break;
