--- ./src5.1.3.orig/MED_SRC_5.1.3/src/ParaMEDMEM/Test/ParaMEDMEMTest.cxx	2010-03-07 23:18:07.064072621 +0100
+++ ./src5.1.3/MED_SRC_5.1.3/src/ParaMEDMEM/Test/ParaMEDMEMTest.cxx	2010-03-07 23:19:44.920271906 +0100
@@ -23,6 +23,7 @@
 #include <cmath>
 #include <list>
 #include <stdexcept>
+#include <cstdlib>
 
 //================================================================================
 /*!
--- ./src5.1.3.orig/MED_SRC_5.1.3/src/ParaMEDMEM/Test/test_AllToAllvTimeDEC.cxx	2010-03-07 23:18:07.102331566 +0100
+++ ./src5.1.3/MED_SRC_5.1.3/src/ParaMEDMEM/Test/test_AllToAllvTimeDEC.cxx	2010-03-07 23:24:44.300321009 +0100
@@ -21,7 +21,7 @@
 #include <map>
 #include <iostream>
 #include <mpi.h>
-#include <time.h>
+#include <ctime>
 
 #include "MPIAccessDECTest.hxx"
 #include <cppunit/TestAssert.h>
--- ./src5.1.3.orig/MED_SRC_5.1.3/src/ParaMEDMEM/Test/test_AllToAllvTimeDoubleDEC.cxx	2010-03-07 23:18:07.084009650 +0100
+++ ./src5.1.3/MED_SRC_5.1.3/src/ParaMEDMEM/Test/test_AllToAllvTimeDoubleDEC.cxx	2010-03-07 23:25:22.930322599 +0100
@@ -22,7 +22,7 @@
 #include <map>
 #include <iostream>
 #include <mpi.h>
-#include <time.h>
+#include <ctime>
 
 #include "MPIAccessDECTest.hxx"
 #include <cppunit/TestAssert.h>
--- ./src5.1.3.orig/MED_SRC_5.1.3/src/INTERPOLATION/create_mesh_interpolation.c	2010-03-13 13:14:50.721209673 +0100
+++ ./src5.1.3/MED_SRC_5.1.3/src/INTERPOLATION/create_mesh_interpolation.c	2010-03-13 13:15:35.420479255 +0100
@@ -21,7 +21,7 @@
 //
 #include <med.h>
 #include <string.h>
-
+#include <stdlib.h>
 
 /*****************************************************************************************************/
 
