--- src3.2.6_orig/MED_SRC_3.2.6/src/MEDWrapper/Base/MED_Wrapper.cxx	2008-09-04 21:10:27.000000000 +0200
+++ src3.2.6/MED_SRC_3.2.6/src/MEDWrapper/Base/MED_Wrapper.cxx	2008-09-04 21:14:06.000000000 +0200
@@ -23,7 +23,7 @@
   ::TLockProxy(TWrapper* theWrapper):
     myWrapper(theWrapper)
   {
-    boost::detail::thread::lock_ops<TWrapper::TMutex>::lock(myWrapper->myMutex);
+    myWrapper->myMutex.lock();
     INITMSG(MYDEBUG,"TLockProxy() - this -"<<this<<"; myWrapper = "<<myWrapper<<endl);
   }
   
@@ -31,7 +31,7 @@
   ::~TLockProxy()
   {
     INITMSG(MYDEBUG,"~TLockProxy() - this -"<<this<<"; myWrapper = "<<myWrapper<<endl);
-    boost::detail::thread::lock_ops<TWrapper::TMutex>::unlock(myWrapper->myMutex);
+    myWrapper->myMutex.unlock();
   }
   
   TWrapper*
