--- src5.1.3.orig/MED_SRC_5.1.3/src/MEDSPLITTER/MEDSPLITTER_SCOTCHGraph.cxx	2008-12-10 10:14:43.000000000 +0100
+++ src5.1.3/MED_SRC_5.1.3/src/MEDSPLITTER/MEDSPLITTER_SCOTCHGraph.cxx	2009-09-27 23:26:59.010112791 +0200
@@ -19,7 +19,7 @@
 #include <cstdio>
 extern "C" {
 #define restrict
-#include "bin/scotch.h"
+#include "scotch.h"
 }
 #include "MEDSPLITTER_Graph.hxx"
 #include "MEDSPLITTER_SCOTCHGraph.hxx"
