diff -Naur why-2.19-orig/Makefile.in why-2.19-ptch/Makefile.in
--- why-2.19-orig/Makefile.in	2009-06-23 12:09:43.000000000 +0000
+++ why-2.19-ptch/Makefile.in	2009-09-04 14:43:22.000000000 +0000
@@ -777,9 +777,9 @@
 	cp -f $(VO7) $(LIBDIR)/why/coq7
 install-coq-v8: 
 	if test -w $(COQLIB) ; then \
-	 mkdir -p $(COQLIB)/user-contrib ; \
-	 cp -f $(V8FILES) $(COQLIB)/user-contrib ; \
-	 cp -f $(VO8) $(COQLIB)/user-contrib ; \
+	 mkdir -p $(DESTDIR)/$(COQLIB)/user-contrib ; \
+	 cp -f $(V8FILES) $(DESTDIR)/$(COQLIB)/user-contrib ; \
+	 cp -f $(VO8) $(DESTDIR)/$(COQLIB)/user-contrib ; \
 	else \
 	echo "Cannot copy to Coq standard library. Add $(LIBDIR)/why/coq to Coq include path." ;\
 	mkdir -p $(LIBDIR)/why/coq ;\
@@ -788,18 +788,18 @@
 
 install-pvs-no:
 install-pvs-yes: $(PVSFILES)
-	mkdir -p $(PVSLIB)/why
-	cp $(PVSFILES) $(PVSFILES:.pvs=.prf) $(PVSLIB)/why
-	cp lib/pvs/top.pvs lib/pvs/pvscontext.el $(PVSLIB)/why
+	mkdir -p $(DESTDIR)/$(PVSLIB)/why
+	cp $(PVSFILES) $(PVSFILES:.pvs=.prf) $(DESTDIR)/$(PVSLIB)/why
+	cp lib/pvs/top.pvs lib/pvs/pvscontext.el $(DESTDIR)/$(PVSLIB)/why
 	@echo "======  Compiling PVS theories, this may take some time ======"
-	(cd $(PVSLIB)/why ; @PVSC@ -batch -l pvscontext.el -q -v 2 > top.out)
+	(cd $(DESTDIR)/$(PVSLIB)/why ; @PVSC@ -batch -l pvscontext.el -q -v 2 > top.out)
 	@echo "======  Done compiling PVS theories ======"
 
 install-mizar-no:
 install-mizar-yes:
-	mkdir -p @MIZARLIB@/mml/dict
-	cp lib/mizar/why.miz @MIZARLIB@/mml
-	cp lib/mizar/dict/why.voc @MIZARLIB@/mml/dict
+	mkdir -p $(DESTDIR)/@MIZARLIB@/mml/dict
+	cp lib/mizar/why.miz $(DESTDIR)/@MIZARLIB@/mml
+	cp lib/mizar/dict/why.voc $(DESTDIR)/@MIZARLIB@/mml/dict
 
 local-install: $(BINARY) $(WHYCONFIG) $(CADUCEUS) $(JESSIE) bin/gwhy.$(OCAMLBEST) byte bin/gwhy.byte 
 	cp $(BINARY) $$HOME/bin/why
@@ -817,15 +817,15 @@
 # 	cp -f $(WHY2HTML) $$HOME/bin/$$OSTYPE/why2html
 # 	cp -f $(DP) $$HOME/bin/$$OSTYPE/dp
 # 	cp -f $(RVMERGE) $$HOME/bin/$$OSTYPE/rv_merge
-# #	mkdir -p $(COQLIB)/contrib7/why
-# #	cp -f $(VO7) $(VFILES) $(COQLIB)/contrib7/why
-# #	mkdir -p $(COQLIB)/contrib/why
-# #	cp -f $(VO8) $(VFILES) $(COQLIB)/contrib/why
-# 	mkdir -p $(PVSLIB)/why
-# 	cp $(PVSFILES) $(PVSLIB)/why
-# 	mkdir -p $(MIZFILES)/mml/dict
-# 	cp lib/mizar/why.miz $(MIZFILES)/mml
-# 	cp lib/mizar/dict/why.voc $(MIZFILES)/mml/dict
+# #	mkdir -p $(DESTDIR)/$(COQLIB)/contrib7/why
+# #	cp -f $(VO7) $(VFILES) $(DESTDIR)/$(COQLIB)/contrib7/why
+# #	mkdir -p $(DESTDIR)/$(COQLIB)/contrib/why
+# #	cp -f $(VO8) $(VFILES) $(DESTDIR)/$(COQLIB)/contrib/why
+# 	mkdir -p $(DESTDIR)/$(PVSLIB)/why
+# 	cp $(PVSFILES) $(DESTDIR)/$(PVSLIB)/why
+# 	mkdir -p $(DESTDIR)/$(MIZFILES)/mml/dict
+# 	cp lib/mizar/why.miz $(DESTDIR)/$(MIZFILES)/mml
+# 	cp lib/mizar/dict/why.voc $(DESTDIR)/$(MIZFILES)/mml/dict
 # 	mkdir -p $$HOME/man/man1
 # 	cp -f doc/*.1 $$HOME/man/man1
 
@@ -1119,7 +1119,6 @@
 	rm -f $(GENERATED)
 	make -C atp clean
 	make -C doc clean
-	make -C examples-v7 clean
 	make -C examples clean
 
 dist-clean:: clean
