diff -Naurp why-2.28-orig/jc/jc_annot_inference.ml why-2.28/jc/jc_annot_inference.ml
--- why-2.28/jc/jc_annot_inference.ml	2010-12-17 20:06:44.000000000 +0000
+++ why-2.28/jc/jc_annot_inference.ml	2011-02-14 11:34:21.000000000 +0000
@@ -150,8 +150,8 @@ let rec destruct_pointer t =
     | JCTvar _ | JCTderef _ | JCTapp _ | JCTold _ | JCTat _ | JCTif _ 
     | JCTrange _ | JCTmatch _ | JCTaddress _ | JCTbase_block _
     | JCTconst _ | JCTbinary _ | JCTunary _ | JCToffset _ | JCTinstanceof _ 
-    | JCTreal_cast _ | JCTrange_cast _ | JCTbitwise_cast _ | JCTcast _ ->
-	None
+    | JCTreal_cast _ | JCTrange_cast _ | JCTbitwise_cast _ | JCTcast _
+    | JCTlet (_, _, _) -> None
 
 let equality_operator_of_type ty = `Beq, operator_of_type ty
 
@@ -610,8 +610,8 @@ let rec atp_of_term t =
 	Atp.Var (Vwp.variable_for_term t)
     | JCTshift _ | JCTold _ | JCTat _ | JCTmatch _ | JCTinstanceof _ 
     | JCTcast _ | JCTrange_cast _ | JCTbitwise_cast _ | JCTreal_cast _ 
-    | JCTaddress _ | JCTif _ | JCTrange _ | JCTunary _ | JCTbase_block _ ->
-	err ()
+    | JCTaddress _ | JCTif _ | JCTrange _ | JCTunary _ | JCTbase_block _
+    | JCTlet (_, _, _) -> err ()
 
 let rec term_of_atp tm =
   try
@@ -1198,8 +1198,8 @@ let linearize t =
       | JCTunary _ | JCTshift _ | JCTinstanceof _ | JCTmatch _ 
       | JCTold _ | JCTat _ | JCTcast _ | JCTbitwise_cast _ 
       | JCTrange_cast _ | JCTreal_cast _ | JCTaddress _ | JCTbase_block _
-      | JCTrange _ | JCTif _ -> 
-	  err ()
+      | JCTrange _ | JCTif _
+      | JCTlet (_, _, _) -> err ()
     with Failure "linearize" -> 
       (TermMap.add t (Int 1) TermMap.empty, Int 0)
   in aux t
