--- Singular/Makefile.in	2010-06-24 15:10:14.000000000 +0200
+++ Singular/Makefile.in.new	2010-07-14 21:30:08.000000000 +0200
@@ -332,7 +332,7 @@
 
 libsingular: mod2.h Makefile version.h $(WIN_LIBS) scanner.cc  ${OBJS} \
          iparith.o libsingular-tesths.o libparse.cc claptmpl.o mpsr_Tok.o $(DL_LIBS) 
-	$(LIBSINGULAR_LD) ${SLDFLAGS} ${LIBSINGULAR_FLAGS} -o libsingular.${SO_SUFFIX} \
+	$(LIBSINGULAR_LD) ${SLDFLAGS} ${LIBSINGULAR_FLAGS} -Wl,-soname,libsingular.so.3 -o libsingular.${SO_SUFFIX} \
 	libsingular-tesths.o iparith.o mpsr_Tok.o claptmpl.o \
 	${OBJS} -lkernel -L../kernel -L../factory -L../libfac -L${libdir} ${LIBSINGULAR_LIBS}
 
