diff -Naurp bigdft-1.7-dev.12.orig/S_GPU/src/Makefile.in bigdft-1.7-dev.12/S_GPU/src/Makefile.in
--- bigdft-1.7-dev.12.orig/S_GPU/src/Makefile.in	2012-07-16 08:24:00.000000000 +0000
+++ bigdft-1.7-dev.12/S_GPU/src/Makefile.in	2013-02-19 15:59:06.142835878 +0000
@@ -88,7 +88,7 @@ am__objects_1 = libs_gpu_la-s_gpu_C.lo \
 	libs_gpu_la-s_gpu_F.lo
 am_libs_gpu_la_OBJECTS = $(am__objects_1)
 libs_gpu_la_OBJECTS = $(am_libs_gpu_la_OBJECTS)
-libs_gpu_la_LINK = $(LIBTOOL) --tag=CXX $(libs_gpu_la_LIBTOOLFLAGS) \
+libs_gpu_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
 	$(CXXFLAGS) $(libs_gpu_la_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
