diff -Naurp bigdft-1.7-dev.12.orig/src/Makefile.am bigdft-1.7-dev.12/src/Makefile.am
--- bigdft-1.7-dev.12.orig/src/Makefile.am	2012-07-16 07:46:11.000000000 +0000
+++ bigdft-1.7-dev.12/src/Makefile.am	2013-01-09 14:09:58.198905489 +0000
@@ -45,6 +45,9 @@ EXTRA_DIST = \
 # -------------
 if BUILD_LIB_BIGDFT
 lib_bigdft = libbigdft.a libdft_common.a
+else
+lib_bigdft =
+endif
 PS_SOURCES = PSolver/Build_Kernel.f90 \
 	PSolver/xcenergy.f90 \
 	PSolver/3Dgradient.f90 \
@@ -53,11 +56,6 @@ PS_SOURCES = PSolver/Build_Kernel.f90 \
 	PSolver/wofz.f90 \
 	lib/fft/fft3d.f90
 PS_MODULES = PSolver/Poisson_Solver.f90
-else
-lib_bigdft =
-PS_SOURCES =
-PS_MODULES =
-endif
 if BUILD_BINARIES
 binaries_sources = bigdft memguess splsad NEB frequencies MDanalysis $(minima_hopping_sources) bart abscalc BigDFT2Wannier WaCo
 local_binaries = test_forces
