diff -Naur orig/bigdft-1.6-tuto.1.orig/src/CUDA/Makefile.am patched/bigdft-1.6-tuto.1/src/CUDA/Makefile.am
--- bigdft-1.6-tuto.1.orig/src/CUDA/Makefile.am	2011-09-27 12:43:06.000000000 +0000
+++ bigdft-1.6-tuto.1/src/CUDA/Makefile.am	2012-07-04 12:08:27.000000000 +0000
@@ -43,7 +43,8 @@
 # Not used anymore, to be removed
 #INC_CUDA = -I@LIB_CUTIL_PATH@
 EXTRA_PROGRAMS = conv_check
-noinst_LIBRARIES = libGPU.a libbasef.a
+lib_LIBRARIES = libGPU.a
+noinst_LIBRARIES = libbasef.a
 
 libGPU_a_SOURCES = $(SOURCES_CU_CONV1) $(SOURCES_SHARED)  $(SOURCES_CONV)
 AM_CPPFLAGS = $(INC_CUDA) -I. -I$(top_builddir) -I@CUDA_PATH@/include/ @LIBSGPU_INCLUDE@ -DCUBLAS_USE_THUNKING
diff -Naur orig/bigdft-1.6-tuto.1.orig/src/OpenCL/Makefile.am patched/bigdft-1.6-tuto.1/src/OpenCL/Makefile.am
--- bigdft-1.6-tuto.1.orig/src/OpenCL/Makefile.am	2011-09-27 12:43:06.000000000 +0000
+++ bigdft-1.6-tuto.1/src/OpenCL/Makefile.am	2012-07-04 12:09:02.000000000 +0000
@@ -26,7 +26,8 @@
 
 EXTRA_PROGRAMS = conv_check conv_check_fft bench_conv bench_conv_quick bench_zgemm stream_check
 
-noinst_LIBRARIES = libbasef.a libbasec.a libOCL.a libStream.a
+lib_LIBRARIES = libOCL.a
+noinst_LIBRARIES = libbasef.a libbasec.a libStream.a
 
 libbasef_a_SOURCES = $(mpi_source)
 libbasef_a_LIBADD = ../modules/defs.o ../modules/base.o \
