--- configure.ac.orig	2011-11-23 14:02:15.000000000 +0000
+++ configure.ac	2011-11-23 14:09:19.000000000 +0000
@@ -551,14 +551,17 @@
 ac_build_libxc="no"
 AC_ARG_ENABLE(libxc, AS_HELP_STRING([--disable-libxc], [Do not link with libXC library (default = yes).]), ac_use_libxc=$enableval, ac_use_libxc="yes")
 AC_ARG_ENABLE(internal-libxc, AS_HELP_STRING([--disable-internal-libxc], [Do not build and link with internal libXC library (default = yes).]), ac_internal_libxc=$enableval, ac_internal_libxc="yes")
-AC_ARG_WITH(libxc-path, AS_HELP_STRING([--with-libxc-path], [Give the path for an external libXC library, lib and module files (default = None).]),
+AC_ARG_WITH(libxc-path, AS_HELP_STRING([--with-libxc-path], [Give the path for an external libXC library, lib files (default = None).]),
             ac_libxc_dir=$withval, ac_libxc_dir=)
+AC_ARG_WITH(libxc-include, AS_HELP_STRING([--with-libxc-include], [Give the path for an external libXC library module files (default = None).]),
+            ac_libxc_dir_include=$withval, ac_libxc_dir_include=)
 if test x"$ac_use_libxc" = x"yes" ; then
   ac_build_libxc=$ac_internal_libxc
   if test "$ac_internal_libxc" = "yes" ; then
     ac_build_libxc="yes"
     ac_use_libxc="yes"
     ac_libxc_dir="$top_builddir"/libxc-1.1.0/src
+    ac_libxc_dir_include="$ac_libxc_dir"/include
     AC_SUBST(LIBXC_DIR, "\$(top_builddir)"/libxc-1.1.0/src)
     AC_SUBST(LIBXC_INCLUDE, "-I\$(top_builddir)"/libxc-1.1.0/src)
     LDFLAGS="$LDFLAGS -L\$(top_builddir)/libxc-1.1.0/src/.libs"
@@ -566,6 +569,9 @@
   else
     if test -n "$ac_libxc_dir" ; then
       LDFLAGS="$LDFLAGS -L$ac_libxc_dir/lib"
+      if test x"$ac_libxc_dir_include" = x ; then
+        ac_libxc_dir_include="$ac_libxc_dir"/include
+      fi
       
       dnl Test the given implementation of libXC.
       ac_libxc_libname="xc"
@@ -574,13 +580,13 @@
         ac_libxc_libname="dft_xc"
         AC_CHECK_LIB($ac_libxc_libname, xc_f90_lda_vxc, withlibxc=yes, withlibxc=no)
       fi
-      AC_CHECK_FILE($ac_libxc_dir/include/libxc_funcs_m.$capitalize_module_ext, funcs="yes", funcs="no")
-      AC_CHECK_FILE($ac_libxc_dir/include/xc_f90_lib_m.$capitalize_module_ext, lib="yes", lib="no")
-      AC_CHECK_FILE($ac_libxc_dir/include/xc_f90_types_m.$capitalize_module_ext, types="yes", types="no")
+      AC_CHECK_FILE($ac_libxc_dir_include/libxc_funcs_m.$capitalize_module_ext, funcs="yes", funcs="no")
+      AC_CHECK_FILE($ac_libxc_dir_include/xc_f90_lib_m.$capitalize_module_ext, lib="yes", lib="no")
+      AC_CHECK_FILE($ac_libxc_dir_include/xc_f90_types_m.$capitalize_module_ext, types="yes", types="no")
       if test "$withlibxc" = "yes" -a "$funcs" = "yes" -a "$lib" = "yes" -a "$types" = "yes"; then
         ac_use_libxc="yes"
-        AC_SUBST(LIBXC_DIR, $ac_libxc_dir/include)
-        AC_SUBST(LIBXC_INCLUDE, "-I"$ac_libxc_dir/include)
+        AC_SUBST(LIBXC_DIR, $ac_libxc_dir_include)
+        AC_SUBST(LIBXC_INCLUDE, "-I"$ac_libxc_dir_include)
         LIBXC_LIBS="-l"$ac_libxc_libname
       fi
     else
