diff --git a/Packages/cdms2/Src/bindex.c b/Packages/cdms2/Src/bindex.c
index 9a91a3b..d8e6e4d 100644
--- a/Packages/cdms2/Src/bindex.c
+++ b/Packages/cdms2/Src/bindex.c
@@ -2,6 +2,7 @@
    with data on non-rectilinear grids.
  */
 #include <math.h>
+#include <stdio.h>
 /* NBINI, NBINJ should match definition in bindex.bindexHorizontalGrid */
 #define NBINI 720			     /* number of bins in the i (longitude) direction */
 #define NBINJ 360			     /* number of bins in the j (latitude) direction */
diff --git a/libcdms/src/cdunif/cdunifnc.c b/libcdms/src/cdunif/cdunifnc.c
index 75f2b2f..8af93e1 100644
--- a/libcdms/src/cdunif/cdunifnc.c
+++ b/libcdms/src/cdunif/cdunifnc.c
@@ -28,6 +28,7 @@
 
 #ifdef netcdf
 #include <stdlib.h>
+#include <string.h>
 #include "netcdf.h"
 #include "cdunifint.h"
 
diff --git a/libcdms/src/cdunif/gaddes.c b/libcdms/src/cdunif/gaddes.c
index aac8152..0dcd634 100644
--- a/libcdms/src/cdunif/gaddes.c
+++ b/libcdms/src/cdunif/gaddes.c
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <math.h>
 #include <ctype.h>
+#include <string.h>
 #ifndef __APPLE__
 #include <malloc.h>
 #endif
diff --git a/libcdms/src/cdunif/gaio.c b/libcdms/src/cdunif/gaio.c
index c64ae83..21b61f8 100644
--- a/libcdms/src/cdunif/gaio.c
+++ b/libcdms/src/cdunif/gaio.c
@@ -4,6 +4,7 @@
 
 #include <math.h>
 #include <stdio.h>
+#include <string.h>
 #ifndef __APPLE__
 #include <malloc.h>
 #endif
diff --git a/libcdms/src/db_util/cdRegTrfm.c b/libcdms/src/db_util/cdRegTrfm.c
index cd88441..97d0185 100644
--- a/libcdms/src/db_util/cdRegTrfm.c
+++ b/libcdms/src/db_util/cdRegTrfm.c
@@ -32,6 +32,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 #include <cdmsint.h>
 #include <isdb.h>
 					     /* copy CDMS geom to NEONS ngeom */
