diff --git a/setup_util.py b/setup_util.py
index 1eea295..e17342a 100644
--- a/setup_util.py
+++ b/setup_util.py
@@ -116,7 +116,7 @@ def check_ifnetcdf4(netcdf4_incdir):
 
 netcdf_incdir, netcdf_libdir = DepFinder('NetCDF', 'NETCDF_HOME',
                                          includefile='netcdf.h',
-                                         libfile='libnetcdf.a').find()
+                                         libfile='libnetcdf.so').find()
 # If using NetCDF4 find the HDF5 libraries
 if check_ifnetcdf4(netcdf_incdir):
     print 'NetCDF4 detected.  Including HDF libraries'
