Index: SRC/CMakeLists.txt
===================================================================
--- SRC/CMakeLists.txt	(revision 975)
+++ SRC/CMakeLists.txt	(working copy)
@@ -373,7 +373,7 @@
    zla_lin_berr.f zlarscl2.f zlascl2.f zla_wwaddw.f)
 
 
-if( USEXBLAS)
+if( USE_XBLAS)
   set(ALLXOBJ ${SXLASRC} ${DXLASRC} ${CXLASRC} ${ZXLASRC} ${ALLXAUX})
 endif()
 
@@ -392,5 +392,5 @@
   set(ALLOBJ  ${ZLASRC} ${ALLAUX} ${DZLAUX} ${ZCLASRC})
 endif()
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
-target_link_libraries(lapack ${BLAS_LIBRARIES})
+target_link_libraries(lapack ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 lapack_install_library(lapack)
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 975)
+++ CMakeLists.txt	(working copy)
@@ -67,6 +67,7 @@
 
 option(USE_OPTIMIZED_BLAS "Whether or not to use an optimized BLAS library instead of included netlib BLAS" OFF)
 
+
 # Check the usage of the user provided BLAS libraries
 if(BLAS_LIBRAIRIES)
   include(CheckFortranFunctionExists)
@@ -103,6 +104,11 @@
     "${CMAKE_SHARED_LINKER_FLAGS} ${BLAS_LINKER_FLAGS}" 
     CACHE STRING "Linker flags for shared libs" FORCE)
 endif( NOT BLAS_FOUND )
+
+option(USE_XBLAS "Build extended precision (needs XBLAS)" OFF)
+if (USE_XBLAS)
+  find_library(XBLAS_LIBRARY NAMES xblas)
+endif(USE_XBLAS)
    
 add_subdirectory(SRC)
 if(BUILD_TESTING)
