--- a/branding/cernbox-theme.h	2021-10-21 13:10:22.832842670 +0200
+++ b/branding/cernbox-theme.h	2021-10-21 13:14:57.394479817 +0200
@@ -62,7 +62,7 @@
 
 
     QString about() const  {
-        QString devString = tr("<p>Version %1. For more information visit <a href=\"%2\">https://%3</a></p>").arg(Utility::escape(QStringLiteral(MIRALL_VERSION_STRING)), Utility::escape(QStringLiteral("https://" MIRALL_STRINGIFY(APPLICATION_DOMAIN))), Utility::escape(QStringLiteral(MIRALL_STRINGIFY(APPLICATION_DOMAIN))));
+        QString devString = tr("<p>Version %1. For more information visit <a href=\"%2\">https://%3</a></p>").arg(Utility::escape(QStringLiteral(MIRALL_VERSION_STRING)), Utility::escape(QStringLiteral("https://" APPLICATION_DOMAIN)), Utility::escape(QStringLiteral(APPLICATION_DOMAIN)));
 
 
         return devString;
