diff -Nur scalapack-2.0.1.orig/SRC/pblas.h scalapack-2.0.1/SRC/pblas.h
--- scalapack-2.0.1.orig/SRC/pblas.h	2007-04-03 01:06:06.000000000 +0100
+++ scalapack-2.0.1/SRC/pblas.h	2012-02-23 05:13:01.959529911 +0000
@@ -543,6 +543,7 @@
 #define pscasum_          PSCASUM
 #define pcamax_           PCAMAX
 #define pcrot_            PCROT
+#define crot_             CROT
 
 #define pzswap_           PZSWAP
 #define pzscal_           PZSCAL
@@ -555,6 +556,7 @@
 #define pdzasum_          PDZASUM
 #define pzamax_           PZAMAX
 #define pzrot_            PZROT
+#define zrot_             ZROT
                                                     /* Level-2 PBLAS */
 #define pcgemv_           PCGEMV
 #define pcgeru_           PCGERU
diff -Nur scalapack-2.0.1.orig/SRC/pdsyevr.f scalapack-2.0.1/SRC/pdsyevr.f
--- scalapack-2.0.1.orig/SRC/pdsyevr.f	2011-11-11 03:10:38.000000000 +0000
+++ scalapack-2.0.1/SRC/pdsyevr.f	2012-02-23 05:13:02.149534905 +0000
@@ -675,7 +675,7 @@
 *
 *     Each processor computes the work assignments for all processors
 *
-      CALL CMPIM2( IIL, IIU, NPROCS,
+      CALL DMPIM2( IIL, IIU, NPROCS,
      $             IWORK(INDILU), IWORK(INDILU+NPROCS) )
 *
 *     Find local work assignment
@@ -863,7 +863,7 @@
 *
 *           Find collaborators of MYPROC            
             IF( (NPROCS.GT.1).AND.(MYIL.GT.0) ) THEN
-               CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+               CALL DMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
             ELSE
@@ -949,7 +949,7 @@
             IINDWLC = INDWORK + INDWLC - 1
             IF(.NOT.FINISH) THEN
                IF((NEEDIL.LT.DOL).OR.(NEEDIU.GT.DOU)) THEN
-                  CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
+                  CALL DMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
      $                 IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
                ELSE
@@ -1172,7 +1172,7 @@
 *     
 *
 ***********************************************************************
-      SUBROUTINE CMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
+      SUBROUTINE DMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
 
       IMPLICIT NONE
 
@@ -1234,7 +1234,7 @@
       END
 
 
-      SUBROUTINE CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+      SUBROUTINE DMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   PMYILS, PMYIUS,
      $                   COLBRT, FRSTCL, LASTCL )
 
diff -Nur scalapack-2.0.1.orig/SRC/pssyevr.f scalapack-2.0.1/SRC/pssyevr.f
--- scalapack-2.0.1.orig/SRC/pssyevr.f	2011-11-11 03:10:38.000000000 +0000
+++ scalapack-2.0.1/SRC/pssyevr.f	2012-02-23 05:13:02.349540161 +0000
@@ -675,7 +675,7 @@
 *
 *     Each processor computes the work assignments for all processors
 *
-      CALL CMPIM2( IIL, IIU, NPROCS,
+      CALL SMPIM2( IIL, IIU, NPROCS,
      $             IWORK(INDILU), IWORK(INDILU+NPROCS) )
 *
 *     Find local work assignment
@@ -863,7 +863,7 @@
 *
 *           Find collaborators of MYPROC            
             IF( (NPROCS.GT.1).AND.(MYIL.GT.0) ) THEN
-               CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+               CALL SMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
             ELSE
@@ -949,7 +949,7 @@
             IINDWLC = INDWORK + INDWLC - 1
             IF(.NOT.FINISH) THEN
                IF((NEEDIL.LT.DOL).OR.(NEEDIU.GT.DOU)) THEN
-                  CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
+                  CALL SMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
      $                 IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
                ELSE
@@ -1172,7 +1172,7 @@
 *     
 *
 ***********************************************************************
-      SUBROUTINE CMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
+      SUBROUTINE SMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
 
       IMPLICIT NONE
 
@@ -1234,7 +1234,7 @@
       END
 
 
-      SUBROUTINE CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+      SUBROUTINE SMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   PMYILS, PMYIUS,
      $                   COLBRT, FRSTCL, LASTCL )
 
diff -Nur scalapack-2.0.1.orig/SRC/pzheevr.f scalapack-2.0.1/SRC/pzheevr.f
--- scalapack-2.0.1.orig/SRC/pzheevr.f	2011-11-11 03:10:38.000000000 +0000
+++ scalapack-2.0.1/SRC/pzheevr.f	2012-02-23 05:13:01.899528334 +0000
@@ -727,7 +727,7 @@
 *
 *     Each processor computes the work assignments for all processors
 *
-      CALL CMPIM2( IIL, IIU, NPROCS,
+      CALL ZMPIM2( IIL, IIU, NPROCS,
      $             IWORK(INDILU), IWORK(INDILU+NPROCS) )
 *
 *     Find local work assignment
@@ -914,7 +914,7 @@
 
 *           Find collaborators of MYPROC            
             IF( (NPROCS.GT.1).AND.(MYIL.GT.0) ) THEN
-               CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+               CALL ZMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
             ELSE
@@ -999,7 +999,7 @@
             IINDWLC = INDRWORK + INDWLC - 1
             IF(.NOT.FINISH) THEN
                IF((NEEDIL.LT.DOL).OR.(NEEDIU.GT.DOU)) THEN
-                  CALL CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
+                  CALL ZMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU,
      $                 IWORK(INDILU), IWORK(INDILU+NPROCS),
      $                   COLBRT, FRSTCL, LASTCL )
                ELSE
@@ -1223,7 +1223,7 @@
 *
 ***********************************************************************
 
-      SUBROUTINE CMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
+      SUBROUTINE ZMPIM2( IL, IU, NPROCS, PMYILS, PMYIUS )
 
       IMPLICIT NONE
 
@@ -1288,7 +1288,7 @@
 *
 *
 ***********************************************************************
-      SUBROUTINE CMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
+      SUBROUTINE ZMPCOL( MYPROC, NPROCS, IIL, NEEDIL, NEEDIU, 
      $                   PMYILS, PMYIUS,
      $                   COLBRT, FRSTCL, LASTCL )
 
