diff -Naur orsa-0.7.0/src/liborsa/orsa_body.h orsa-0.7.0.new/src/liborsa/orsa_body.h
--- orsa-0.7.0/src/liborsa/orsa_body.h	2004-12-30 03:58:03.000000000 +0000
+++ orsa-0.7.0.new/src/liborsa/orsa_body.h	2006-06-01 16:27:02.000000000 +0100
@@ -183,7 +183,7 @@
     inline void SetVelocity(const double x, const double y, const double z) { Vector v(x,y,z); SetVelocity(v); }
     
     // b position - this position
-    inline Vector Body::distanceVector(const Body & b) const { return b.position()-position(); }
+    inline Vector distanceVector(const Body & b) const { return b.position()-position(); }
     inline double distance(const Body & b) const { return distanceVector(b).Length(); }
     
     // alias
diff -Naur orsa-0.7.0/src/libxorsa/xorsa_analysis.h orsa-0.7.0.new/src/libxorsa/xorsa_analysis.h
--- orsa-0.7.0/src/libxorsa/xorsa_analysis.h	2004-06-25 01:58:30.000000000 +0100
+++ orsa-0.7.0.new/src/libxorsa/xorsa_analysis.h	2006-06-01 16:27:02.000000000 +0100
@@ -129,7 +129,7 @@
   XOrsaPeaksListItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int XOrsaPeaksListItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
diff -Naur orsa-0.7.0/src/libxorsa/xorsa_import_astorb_objects.h orsa-0.7.0.new/src/libxorsa/xorsa_import_astorb_objects.h
--- orsa-0.7.0/src/libxorsa/xorsa_import_astorb_objects.h	2005-01-05 03:04:17.000000000 +0000
+++ orsa-0.7.0.new/src/libxorsa/xorsa_import_astorb_objects.h	2006-06-01 16:27:02.000000000 +0100
@@ -600,7 +600,7 @@
   inline XOrsaAstorbObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null) : QListViewItem(parent, label1, label2, label3, label4, label5, label6, label7, label8) { };
   
  public:
-  inline int XOrsaAstorbObjectItem::compare(QListViewItem *i, int col, bool ascending) const {
+  inline int compare(QListViewItem *i, int col, bool ascending) const {
 
     using std::atof;
         
diff -Naur orsa-0.7.0/src/libxorsa/xorsa_object_selector.cc orsa-0.7.0.new/src/libxorsa/xorsa_object_selector.cc
--- orsa-0.7.0/src/libxorsa/xorsa_object_selector.cc	2004-06-25 04:12:57.000000000 +0100
+++ orsa-0.7.0.new/src/libxorsa/xorsa_object_selector.cc	2006-06-01 16:27:02.000000000 +0100
@@ -40,7 +40,7 @@
   XOrsaObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int XOrsaObjectItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
diff -Naur orsa-0.7.0/src/orsa/xorsa.h orsa-0.7.0.new/src/orsa/xorsa.h
--- orsa-0.7.0/src/orsa/xorsa.h	2004-07-13 03:21:19.000000000 +0100
+++ orsa-0.7.0.new/src/orsa/xorsa.h	2006-06-01 16:27:02.000000000 +0100
@@ -62,7 +62,7 @@
   ObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int ObjectItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
