--- src5.1.3.orig/YACS_SRC_5.1.3/adm/unix/make_begin.am	2010-03-10 15:45:32.281101526 +0100
+++ src5.1.3/YACS_SRC_5.1.3/adm/unix/make_begin.am	2010-03-10 15:46:58.440251953 +0100
@@ -22,7 +22,7 @@
 
 # Standard directory for installation
 salomeincludedir   = $(includedir)/salome
-libdir             = $(prefix)/lib/salome
+libdir             = $(prefix)/lib@LIB_LOCATION_SUFFIX@/salome
 bindir             = $(prefix)/bin/salome
 salomescriptdir    = $(bindir)
 salomepythondir    = $(pythondir)/salome
--- src5.1.3.orig/YACS_SRC_5.1.3/src/genericgui/Makefile.am	2010-03-10 16:26:05.424066867 +0100
+++ src5.1.3/YACS_SRC_5.1.3/src/genericgui/Makefile.am	2010-03-10 16:26:41.450302694 +0100
@@ -16,7 +16,7 @@
 #
 #  See http://www.salome-platform.org/ or email : webmaster.salome@opencascade.com
 #
-libdir             = $(prefix)/lib/salome
+libdir             = $(prefix)/lib@LIB_LOCATION_SUFFIX@/salome
 bindir             = $(prefix)/bin/salome
 
 lib_LTLIBRARIES = libGenericGui.la
