 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 5401c07..ffd8157 100644
--- a/Makefile
+++ b/Makefile
@@ -6,8 +6,8 @@
 # E-Mail:   anip@redpoll.pharmacy.ualberta.ca
 #
 #CC=gcc -Xcpluscomm -cckr
-CC=/usr/bin/gcc
-CFLAGS=-O
+#CC=/usr/bin/gcc
+#CFLAGS=-O
 SYSLIB=
 INCLUDES=-I/usr/include
 LIBS=$(SYSLIB) -lm 
@@ -25,7 +25,7 @@ OBJS=Main.o Nn.o Ssbond.o Residue.o Hetatm.o Vector.o Torison.o Cs.o Phipsi.o \
 all: shiftx
 
 shiftx:	$(OBJS)
-	$(CC) $(OBJS) $(CFLAGS) $(LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS) $(LIBS) -o $@
 
 Main.o:		Main.c main.h states.h nn.h ssbond.h residue.h hetatm.h cs.h \
                 phipsi.h torison.h hydrogens.h rings.h es.h proline.h hbond.h \
