diff -pur octave-forge-2005.06.13/FIXES/randmtzig.c octave-forge-2005.06.13.patched/FIXES/randmtzig.c
--- octave-forge-2005.06.13/FIXES/randmtzig.c	2005-05-28 18:08:43.000000000 +0200
+++ octave-forge-2005.06.13.patched/FIXES/randmtzig.c	2005-07-21 12:50:02.000000000 +0200
@@ -153,6 +153,7 @@
    instead.
 */ 
 #include <sys/types.h>
+#include <stdint.h>
 
    
 /* XXX FIXME XXX may want to suppress X86 if sizeof(long)>4 */
