diff -urN wannier90-1.1.old/src/disentangle.F90 wannier90-1.1.new/src/disentangle.F90
--- wannier90-1.1.old/src/disentangle.F90	2009-03-07 15:44:11.000000000 +0100
+++ wannier90-1.1.new/src/disentangle.F90	2009-03-07 15:48:05.000000000 +0100
@@ -1784,9 +1784,9 @@
       ! [BIG ITERATION LOOP (iter)]
 
       if (.not.dis_converged) then
-         write(stdout,'(/5x,a)') '<<< Warning: Maximum number of disentanglement &
-              &iterations reached >>>'
-         write(stdout,'(10x,a)')  '<<< Disentanglement convergence criteria not satisfied >>>'
+         write(stdout,'(/5x,a)') &
+          '<<< Warning: Maximum number of disentanglement iterations reached >>>'
+         write(stdout,'(10x,a)') '<<< Disentanglement convergence criteria not satisfied >>>'
       endif
 
       if (icompflag.eq.1) then
@@ -2551,8 +2551,8 @@
       ! [BIG ITERATION LOOP (iter)]
 
       if (.not.dis_converged) then
-         write(stdout,'(/5x,a)') '<<< Warning: Maximum number of disentanglement &
-              &iterations reached >>>'
+         write(stdout,'(/5x,a)') &
+           '<<< Warning: Maximum number of disentanglement iterations reached >>>'
          write(stdout,'(10x,a)')  '<<< Disentanglement convergence criteria not satisfied >>>'
       endif
 
diff -urN wannier90-1.1.old/src/overlap.F90 wannier90-1.1.new/src/overlap.F90
--- wannier90-1.1.old/src/overlap.F90	2009-03-07 15:44:11.000000000 +0100
+++ wannier90-1.1.new/src/overlap.F90	2009-03-07 15:48:36.000000000 +0100
@@ -141,8 +141,8 @@
                    nn_found=.true.
                    nn=inn
                 else
-                   call io_error('Error reading '//trim(seedname)//'.mmn.&
-                        & More than one matching nearest neighbour found')
+                   call io_error('Error reading '//trim(seedname)// &
+                        '.mmn. More than one matching nearest neighbour found')
                 endif
              endif
           end do
diff -urN wannier90-1.1.old/src/parameters.F90 wannier90-1.1.new/src/parameters.F90
--- wannier90-1.1.old/src/parameters.F90	2009-03-07 15:44:11.000000000 +0100
+++ wannier90-1.1.new/src/parameters.F90	2009-03-07 15:50:46.000000000 +0100
@@ -1054,8 +1054,9 @@
        write(stdout,'(1x,a)') '+----------------------------------------------------------------------------+'
        do nsp=1,num_species
           do nat=1,atoms_species_num(nsp)
-             write(stdout,'(1x,a1,1x,a2,1x,i3,3F10.5,3x,a1,1x,3F10.5,4x,a1)') '|',atoms_symbol(nsp),nat,atoms_pos_frac(:,nat,nsp),&
-                  '|',atoms_pos_cart(:,nat,nsp)*lenconfac,'|'
+             write(stdout,'(1x,a1,1x,a2,1x,i3,3F10.5,3x,a1,1x,3F10.5,4x,a1)') &
+&                 '|',atoms_symbol(nsp),nat,atoms_pos_frac(:,nat,nsp),&
+&                 '|',atoms_pos_cart(:,nat,nsp)*lenconfac,'|'
           end do
        end do
        write(stdout,'(1x,a)') '*----------------------------------------------------------------------------*'
@@ -1073,8 +1074,8 @@
        write(stdout,'(1x,a)') '|     Frac. Coord.   l mr  r        z-axis               x-axis          Z/a |'
        write(stdout,'(1x,a)') '+----------------------------------------------------------------------------+'
        do nsp=1,num_proj
-          write(stdout,'(1x,a1,3(1x,f5.2),1x,i2,1x,i2,1x,i2,3(1x,f6.3),3(1x,f6.3),&
-               & 2x,f4.1,1x,a1)')  '|',proj_site(1,nsp),proj_site(2,nsp),&
+          write(stdout,'(1x,a1,3(1x,f5.2),1x,i2,1x,i2,1x,i2,3(1x,f6.3),3(1x,f6.3),2x,f4.1,1x,a1)')&
+&              '|',proj_site(1,nsp),proj_site(2,nsp),&
                proj_site(3,nsp),proj_l(nsp), proj_m(nsp),proj_radial(nsp),&
                proj_z(1,nsp),proj_z(2,nsp),proj_z(3,nsp),proj_x(1,nsp),&
                proj_x(2,nsp),proj_x(3,nsp),proj_zona(nsp),'|'
@@ -2738,8 +2739,8 @@
            dummy=utility_strip(in_data(line))
            dummy=adjustl(dummy)
            pos1=index(dummy,':')
-           if(pos1==0) call io_error('param_read_projection: malformed projection &
-                &definition: '//trim(dummy))
+           if(pos1==0) &
+              call io_error('param_read_projection: malformed projection definition: '//trim(dummy))
            sites=0
            ctemp=dummy(:pos1-1)
            ! Read the atomic site
@@ -2980,11 +2981,11 @@
               endif
            end if
            if(sites==-1) then
-              if(counter+sum(ang_states) > num_proj) call io_error('param_get_projection: &
-                   &too many projections defined')
+              if(counter+sum(ang_states) > num_proj) &
+                 call io_error('param_get_projection: too many projections defined')
            else
-              if(counter+sites*sum(ang_states) > num_proj) call io_error('param_get_projection:&
-                   & too many projections defined')
+              if(counter+sites*sum(ang_states) > num_proj) &
+                 call io_error('param_get_projection: too many projections defined')
            end if
            !
            if(sites==-1) then
@@ -3025,8 +3026,8 @@
 
         ! check there are enough projections and add random projections if required
         if (.not. lpartrandom) then
-              if (counter.ne.num_proj) call io_error('param_get_projections:&
-                   & Fewer projections defined than the number of Wannier functions requested')
+              if (counter.ne.num_proj) call io_error(&
+               'param_get_projections: Fewer projections defined than the number of Wannier functions requested')
         else
            call random_seed()
            do loop=counter+1,num_proj
