diff --git a/other_builds/Makefile.linux_fedora_19_64 b/other_builds/Makefile.linux_fedora_19_64
index ebaa9d5..1471081 100644
--- a/other_builds/Makefile.linux_fedora_19_64
+++ b/other_builds/Makefile.linux_fedora_19_64
@@ -11,8 +11,6 @@ LGIFTI    = -lexpat
 
 # ------------------------------
 # python from C
-IPYTHON  = -DSELENIUM_READY -I/usr/include/python2.7
-LDPYTHON = -lpython2.7
 
 # ----------------------------------------------------------------------
 # X configuration
