--- a/other_builds/Makefile.linux_fedora_19_64      2020-10-09 07:07:58.348874414 -0400
+++ b/other_builds/Makefile.linux_fedora_19_64	2020-10-10 01:15:22.337755934 -0400
@@ -11,8 +11,6 @@

 # ------------------------------
 # python from C
-IPYTHON  = -DSELENIUM_READY -I/usr/include/python2.7
-LDPYTHON = -lpython2.7

 # ----------------------------------------------------------------------
 # X configuration
