--- klayout-0.18.1/build.sh	2009-08-02 15:17:12.000000000 -0400
+++ klayout-0.18.1/build.sh.new	2009-09-18 16:15:47.000000000 -0400
@@ -237,34 +237,3 @@
 echo "Building plugins: $PLUGINS"
 echo ""
 
-# -- Running build  --
-
-echo "Running build .."
-echo "$MAKE_PRG"
-cd $BUILD
-$MAKE_PRG $MAKE_OPT all || exit 1
-echo ""
-
-cd $CURR_DIR
-
-# -- Installing binaries  --
-
-echo "Installing binaries .."
-cd $BUILD
-$MAKE_PRG install
-cd $CURR_DIR
-echo "Build done."
-
-if [ $WIN_CYGWIN = 0 ]; then
-  echo ""
-  echo "Final binary is ready in $BIN/$EXEC_NAME."
-  echo "To run it, you may need to add the following to your LD_LIBRARY_PATH:"
-  echo "$QTLIB"
-  
-  LD_LIBRARY_PATH=$QTLIB:$LD_LIBRARY_PATH
-  $BIN/$EXEC_NAME -v
-  $BIN/$EXEC_NAME -h
-fi
-
-exit 0
-
