Index: opticks-0.0.1_p20191110/cmake/Modules/OpticksBuildOptions.cmake
===================================================================
--- opticks-0.0.1_p20191110.orig/cmake/Modules/OpticksBuildOptions.cmake
+++ opticks-0.0.1_p20191110/cmake/Modules/OpticksBuildOptions.cmake
@@ -26,7 +26,7 @@ RPATH setup docs
 
 #]=]
 
-
+find_package(okconf REQUIRED CONFIG)
 #message(STATUS "OpticksBuildOptions.cmake Configuring ${name}")
 message(STATUS "Configuring ${name}")
 
@@ -92,31 +92,6 @@ set(BUILD_SHARED_LIBS ON)
 # to check the RPATH of a library or executable use chrpath on it, eg: chrpath $(which OKTest) 
 #
 
-
-if(UNIX AND NOT APPLE)
-    if(CMAKE_INSTALL_PREFIX STREQUAL ${OPTICKS_PREFIX})
-       set(CMAKE_INSTALL_RPATH "$ORIGIN/../lib64:$ORIGIN/../externals/lib:$ORIGIN/../externals/lib64:$ORIGIN/../externals/OptiX/lib64")
-    else()
-       message(STATUS " Below two strings differ : forced to use absolute RPATH ")
-       message(STATUS " CMAKE_INSTALL_PREFIX : ${CMAKE_INSTALL_PREFIX} ")
-       message(STATUS " OPTICKS_PREFIX       : ${OPTICKS_PREFIX} ")
-       set(ABSOLUTE_INSTALL_RPATH
-                     ${OPTICKS_PREFIX}/lib64  
-                     ${OPTICKS_PREFIX}/externals/lib  
-                     ${OPTICKS_PREFIX}/externals/lib64  
-                     ${OPTICKS_PREFIX}/externals/OptiX/lib64  
-          ) 
-       set(CMAKE_INSTALL_RPATH  "${ABSOLUTE_INSTALL_RPATH}")
-    endif()
-
-elseif(APPLE)
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-endif()
-
-
-
-
-
 include(OpticksCXXFlags)   
 
 
