Index: implicitmesher-0_p20191111/CMakeLists.txt
===================================================================
--- implicitmesher-0_p20191111.orig/CMakeLists.txt
+++ implicitmesher-0_p20191111/CMakeLists.txt
@@ -3,19 +3,12 @@ set(name ImplicitMesher)
 project(${name} VERSION 0.1.0)
 
 include(CTest)
-include(OpticksBuildOptions)
-
-include(GNUInstallDirs)
-set(CMAKE_INSTALL_INCLUDEDIR "externals/include/${name}")
-set(CMAKE_INSTALL_LIBDIR     "externals/lib")
-set(CMAKE_INSTALL_BINDIR     "lib")
-set(CMAKE_INSTALL_RPATH      "${CMAKE_INSTALL_FULL_LIBDIR}")
 
 find_package(BCM REQUIRED)
 include(BCMDeploy)
 #include(EchoTarget)
 
-find_package(GLM  REQUIRED)
+find_package(glm REQUIRED)
 
 set(SOURCES
 
@@ -45,7 +38,7 @@ set(HEADERS
 )
 
 add_library(${name} SHARED ${SOURCES})
-target_link_libraries(${name} PUBLIC Opticks::GLM)
+target_link_libraries(${name} PUBLIC glm)
 
 # needed for tests that use non-installed headers
 target_include_directories(${name} PUBLIC
