diff --git a/CMakeLists.txt b/CMakeLists.txt
index d1a0cf3..f9013cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,6 @@
 cmake_minimum_required (VERSION 3.5)
 set(name ImplicitMesher)
 project(${name} VERSION 0.1.0)
-include(OpticksBuildOptions)
 
 #[=[
 Hmm OpticksBuildOptions sets CMAKE_INSTALL_INCLUDEDIR to "include/${name}"
@@ -14,7 +13,7 @@ set(CMAKE_INSTALL_LIBDIR     "externals/lib")
 set(CMAKE_INSTALL_BINDIR     "lib")
 
 
-find_package(GLM  REQUIRED)
+find_package(glm  REQUIRED)
 
 set(SOURCES
 
@@ -44,7 +43,7 @@ set(HEADERS
 )
 
 add_library(${name} SHARED ${SOURCES})
-target_link_libraries(${name} PUBLIC Opticks::GLM)
+target_link_libraries(${name} PUBLIC glm)
 
 if(UNIX AND NOT APPLE)
 message(STATUS "adding stdc++ m for UNIX AND NOT APPLE")
@@ -74,7 +73,6 @@ if(PC_VERBOSE)
 endif()
 
 
-bcm_deploy(TARGETS ${name} NAMESPACE Opticks:: SKIP_HEADER_INSTALL) 
 install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 add_subdirectory(tests)
