--- tune.h	2008-10-01 19:15:41.000000000 +0200
+++ tune.h.new	2009-03-22 11:59:42.000000000 +0100
@@ -281,10 +281,10 @@
  *
  * If you want no standard assertion, define STD_ASSERT to 0.
  */
-/*
+
 #define STD_ASSERT	"cpu(i386)", "machine(i386)", "system(unix)", \
 			"system(freebsd)"
-*/
+
 
 /* ====================================================================== */
 /*
@@ -293,10 +293,10 @@
  * Each string must be either "name" or "name=token-list". If you want
  * no predefined macro, define STD_MACROS to 0.
  */
-/*
+
 #define STD_MACROS	"__FreeBSD=4", "__unix", "__i386", \
 			"__FreeBSD__=4", "__unix__", "__i386__"
-*/
+
 
 /* ====================================================================== */
 /*
