--- configure.in.orig	2009-11-22 11:15:42.000000000 +1300
+++ configure.in	2009-11-22 11:59:16.000000000 +1300
@@ -1583,8 +1583,6 @@
     ;;
 esac
 
-AM_CONDITIONAL(BUILD_YASM, test "$want_yasm" = "yes")
-
 CFLAGS_or_unset=${CFLAGS-'(unset)'}
 CPPFLAGS_or_unset=${CPPFLAGS-'(unset)'}
 
@@ -1982,7 +1980,7 @@
 GMP_CHECK_LIBM_FOR_BUILD
 
 
-MPIR_AS="/yasm/yasm"
+MPIR_AS="yasm"
 
 # How to assemble, used with CFLAGS etc, see mpn/Makeasm.am.
 # Using the compiler is a lot easier than figuring out how to invoke the
@@ -3354,9 +3352,6 @@
 # FIXME: Upcoming version of autoconf/automake don't like broken lines.
 #        Right now automake isn't accepting the new AC_CONFIG_FILES scheme.
 
-#     allways configure all subdirectorys , even if there are not going to be built , 
-AC_CONFIG_SUBDIRS([yasm])
-
 if test $enable_fat = yes; then
 YASM_MAC_INC=yasm_mac.inc.fat
 else
--- Makefile.am.orig	2009-11-22 11:16:02.000000000 +1300
+++ Makefile.am	2009-11-22 12:00:01.000000000 +1300
@@ -91,10 +91,6 @@
 
 SUBDIRS = tests
 
-if BUILD_YASM
-SUBDIRS += yasm
-endif
-
 SUBDIRS += mpn mpz mpq mpf printf scanf cxx demos tune doc bench
 
 EXTRA_DIST = configfsf.guess configfsf.sub .gdbinit INSTALL.autoconf
--- mpn/Makeasm.am.orig	2009-11-22 11:16:41.000000000 +1300
+++ mpn/Makeasm.am	2009-11-22 12:01:28.000000000 +1300
@@ -111,4 +111,4 @@
 #  .as assembler, assembled with Yasm
 
 .as.lo:
-	$(LIBTOOL) --mode=compile --tag=CC $(top_srcdir)/strip_fPIC.sh $(top_builddir)$(MPIR_AS) -I $(top_builddir) $(OBJECT_FORMAT) $(GSYM_FLAG) -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
+	$(LIBTOOL) --mode=compile --tag=CC $(top_srcdir)/strip_fPIC.sh $(MPIR_AS) -I $(top_builddir) $(OBJECT_FORMAT) $(GSYM_FLAG) -o $@ `test -f '$<' || echo '$(srcdir)/'`$<
