--- src/configure_at	2009-03-22 20:55:51.394457545 +0100
+++ src/configure_at.new	2009-03-22 20:50:01.155363408 +0100
@@ -163,7 +163,7 @@
 blas=install
 f2c=install
 ucpp=install
-cpp="\$(BINDIR)/ucpp -l"
+cpp="/usr/bin/ucpp -l"
 
 #-----------------------------------
 # skip building of sleap?
@@ -501,7 +501,7 @@
         echo ""
         echo "Obtaining the C++ compiler version:"
         echo "      $cplusplus -v"
-        version=`$cplusplus -v 2>&1 | grep version | cut -d' ' -f3`
+        version=`$cplusplus -v 2>&1 | grep " version " | cut -d' ' -f3`
         echo "The version is $version"
         majorversion=`echo $version | cut -d'.' -f1`
         minorversion=`echo $version | cut -d'.' -f2`
@@ -698,7 +698,7 @@
 LDFLAGS=$ldflags
 
 LEX=   $lex
-YACC=  \$(BINDIR)/yacc
+YACC=  /usr/bin/byacc
 AR=    ar rv
 M4=    $m4
 RANLIB=$ranlib
