--- src/Makefile_at	2008-07-11 00:09:01.000000000 +0200
+++ src/Makefile_at.new	2009-03-25 15:35:51.000000000 +0100
@@ -5,29 +5,29 @@
 install:
 	-mkdir $(BINDIR) $(LIBDIR) $(INCDIR)
 #	utility routines and libraries:
-	(cd ucpp-1.3 && make $(UCPP) )
-	(cd byacc && make install )
-	(cd carpack && make install );
-	(cd clapack && make $(LAPACK) )
-	(cd cblas && make $(BLAS) )
-	(cd f2c &&  make $(F2C)  )
-	(cd c9x-complex && make $(C9XCOMPLEX) )
+#	(cd ucpp-1.3 && make $(UCPP) )
+#	(cd byacc && make install )
+#	(cd carpack && make install );
+#	(cd clapack && make $(LAPACK) )
+#	(cd cblas && make $(BLAS) )
+#	(cd f2c &&  make $(F2C)  )
+#	(cd c9x-complex && make $(C9XCOMPLEX) )
 
 #	nab, leap, and sleap:
-	(cd cifparse && make install )
+#	(cd cifparse && make install )
 	(cd nab && make install )
 	(cd nss && make install )
 
 #	antechamber:
 	(cd antechamber && make install )
-	(cd mopac6/src && make $(G77) )
+#	(cd mopac6/src && make $(G77) )
 	(cd resp && make $(G77) )
 
 #   ptraj:
 	(cd ptraj && make install )
 
 #   miscellaneous:
-	(cd reduce && make install )
+#	(cd reduce && make install )
 
 #   leap and gleap:
 	(cd leap && make install )
