--- amber11/AmberTools/src/configure.orig	2011-03-07 00:05:24.000000000 +0300
+++ amber11/AmberTools/src/configure	2011-03-07 00:07:59.000000000 +0300
@@ -288,11 +288,11 @@
 mpi_flag=
 lex=flex
 flibs_mkl=
-lapack=install
-blas=install
+lapack=skip
+blas=skip
 f2c=skip
-ucpp=install
-cpp="\$(BINDIR)/ucpp -l"
+ucpp=skip
+cpp="\$(EPREFIX)/usr/bin/ucpp -l"
 
 #-----------------------------------
 # skip building of sleap?
@@ -1451,7 +1451,7 @@
 LDFLAGS=$ldflags \$(AMBERBUILDFLAGS)
 
 LEX=   $lex
-YACC=  \$(BINDIR)/yacc
+YACC=  \$(EPREFIX)/usr/bin/yacc
 AR=    ar rv
 M4=    $m4
 RANLIB=$ranlib
