--- amber11/AmberTools/src/configure.orig	2011-03-07 00:28:56.000000000 +0300
+++ amber11/AmberTools/src/configure	2011-03-07 00:30:55.000000000 +0300
@@ -1241,24 +1241,24 @@
   #--------------------------------------------------------------------------
   #  Configure fftw-3.2.2:
   #--------------------------------------------------------------------------
-  if [ "$mdgx" = 'yes' ]; then
-      mdgxflag="--prefix=$AMBERHOME --disable-fortran" 
-      echo
-      echo "Configuring fftw-3.2.2; (may be time-consuming)"
-      echo
-      cd fftw-3.2.2 && \
-      env CC="$cc" CFLAGS="$cflags" ./configure $mdgxflag \
-              > ../fftw3_config.log 2>&1
-      ncerror=$?
-      if [ $ncerror -gt 0 ]; then
-          echo "    Error: fftw configure returned $ncerror"
-          echo "    fftw configure failed!  Check the fftw3_config.log file."
-          exit 1
-      else
-          echo "    fftw-3.2.2 configure succeeded."
-      fi
-      cd ..
-  fi
+  #if [ "$mdgx" = 'yes' ]; then
+  #    mdgxflag="--prefix=$AMBERHOME --disable-fortran" 
+  #    echo
+  #    echo "Configuring fftw-3.2.2; (may be time-consuming)"
+  #    echo
+  #    cd fftw-3.2.2 && \
+  #    env CC="$cc" CFLAGS="$cflags" ./configure $mdgxflag \
+  #            > ../fftw3_config.log 2>&1
+  #    ncerror=$?
+  #    if [ $ncerror -gt 0 ]; then
+  #        echo "    Error: fftw configure returned $ncerror"
+  #        echo "    fftw configure failed!  Check the fftw3_config.log file."
+  #        exit 1
+  #    else
+  #        echo "    fftw-3.2.2 configure succeeded."
+  #    fi
+  #    cd ..
+  #fi
 
   #--------------------------------------------------------------------------
   #  Set up MPI 3D-RISM in NAB and SANDER
@@ -1307,23 +1307,23 @@
           echo
           echo "Configuring fftw-2.1.5; (may be time-consuming)"
           echo
-          cd fftw-2.1.5
-          if [ "$mpi" = 'yes' ]; then
-              ./configure $rismflag --enable-mpi CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
-                  > ../fftw2_config.log 2>&1
-          else
-              ./configure $rismflag CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
-                  > ../fftw2_config.log 2>&1
-          fi
-          ncerror=$?
-          if [ $ncerror -gt 0 ]; then
-              echo "    Error: fftw configure returned $ncerror"
-              echo "    fftw configure failed!  Check the fftw2_config.log file."
-              exit 1
-          else
-              echo "    fftw-2.1.5 configure succeeded."
-          fi
-          cd ..
+ #         cd fftw-2.1.5
+ #         if [ "$mpi" = 'yes' ]; then
+ #             ./configure $rismflag --enable-mpi CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
+ #                 > ../fftw2_config.log 2>&1
+ #         else
+ #             ./configure $rismflag CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
+ #                 > ../fftw2_config.log 2>&1
+ #         fi
+ #         ncerror=$?
+ #         if [ $ncerror -gt 0 ]; then
+ #             echo "    Error: fftw configure returned $ncerror"
+ #             echo "    fftw configure failed!  Check the fftw2_config.log file."
+ #             exit 1
+ #         else
+ #             echo "    fftw-2.1.5 configure succeeded."
+ #         fi
+ #         cd ..
           flibs_fftw2="\$(LIBDIR)/libdrfftw.a \$(LIBDIR)/libdfftw.a"
           if [ "$mpi" = 'yes' ]; then
               flibs_fftw2="\$(LIBDIR)/libdrfftw_mpi.a \$(LIBDIR)/libdfftw_mpi.a $flibs_fftw2"
