diff -urN amber11.orig/AmberTools/src/configure amber11/AmberTools/src/configure
--- amber11.orig/AmberTools/src/configure	2010-04-23 18:17:17.000000000 +0400
+++ amber11/AmberTools/src/configure	2011-03-07 03:39:14.000000000 +0300
@@ -288,11 +288,11 @@
 mpi_flag=
 lex=flex
 flibs_mkl=
-lapack=install
-blas=install
+lapack=skip
+blas=skip
 f2c=skip
-ucpp=install
-cpp="\$(BINDIR)/ucpp -l"
+ucpp=skip
+cpp="\$(EPREFIX)/usr/bin/ucpp -l"
 
 #-----------------------------------
 # skip building of sleap?
@@ -1241,24 +1241,24 @@
   #--------------------------------------------------------------------------
   #  Configure fftw-3.2.2:
   #--------------------------------------------------------------------------
-  if [ "$mdgx" = 'yes' ]; then
-      mdgxflag="--prefix=$AMBERHOME --disable-fortran" 
-      echo
-      echo "Configuring fftw-3.2.2; (may be time-consuming)"
-      echo
-      cd fftw-3.2.2 && \
-      env CC="$cc" CFLAGS="$cflags" ./configure $mdgxflag \
-              > ../fftw3_config.log 2>&1
-      ncerror=$?
-      if [ $ncerror -gt 0 ]; then
-          echo "    Error: fftw configure returned $ncerror"
-          echo "    fftw configure failed!  Check the fftw3_config.log file."
-          exit 1
-      else
-          echo "    fftw-3.2.2 configure succeeded."
-      fi
-      cd ..
-  fi
+  #if [ "$mdgx" = 'yes' ]; then
+  #    mdgxflag="--prefix=$AMBERHOME --disable-fortran" 
+  #    echo
+  #    echo "Configuring fftw-3.2.2; (may be time-consuming)"
+  #    echo
+  #    cd fftw-3.2.2 && \
+  #    env CC="$cc" CFLAGS="$cflags" ./configure $mdgxflag \
+  #            > ../fftw3_config.log 2>&1
+  #    ncerror=$?
+  #    if [ $ncerror -gt 0 ]; then
+  #        echo "    Error: fftw configure returned $ncerror"
+  #        echo "    fftw configure failed!  Check the fftw3_config.log file."
+  #        exit 1
+  #    else
+  #        echo "    fftw-3.2.2 configure succeeded."
+  #    fi
+  #    cd ..
+  #fi
 
   #--------------------------------------------------------------------------
   #  Set up MPI 3D-RISM in NAB and SANDER
@@ -1307,23 +1307,23 @@
           echo
           echo "Configuring fftw-2.1.5; (may be time-consuming)"
           echo
-          cd fftw-2.1.5
-          if [ "$mpi" = 'yes' ]; then
-              ./configure $rismflag --enable-mpi CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
-                  > ../fftw2_config.log 2>&1
-          else
-              ./configure $rismflag CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
-                  > ../fftw2_config.log 2>&1
-          fi
-          ncerror=$?
-          if [ $ncerror -gt 0 ]; then
-              echo "    Error: fftw configure returned $ncerror"
-              echo "    fftw configure failed!  Check the fftw2_config.log file."
-              exit 1
-          else
-              echo "    fftw-2.1.5 configure succeeded."
-          fi
-          cd ..
+ #         cd fftw-2.1.5
+ #         if [ "$mpi" = 'yes' ]; then
+ #             ./configure $rismflag --enable-mpi CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
+ #                 > ../fftw2_config.log 2>&1
+ #         else
+ #             ./configure $rismflag CC="$cc" CFLAGS="$cflags" F77="$fc" FFLAGS="$fflags" \
+ #                 > ../fftw2_config.log 2>&1
+ #         fi
+ #         ncerror=$?
+ #         if [ $ncerror -gt 0 ]; then
+ #             echo "    Error: fftw configure returned $ncerror"
+ #             echo "    fftw configure failed!  Check the fftw2_config.log file."
+ #             exit 1
+ #         else
+ #             echo "    fftw-2.1.5 configure succeeded."
+ #         fi
+ #         cd ..
           flibs_fftw2="\$(LIBDIR)/libdrfftw.a \$(LIBDIR)/libdfftw.a"
           if [ "$mpi" = 'yes' ]; then
               flibs_fftw2="\$(LIBDIR)/libdrfftw_mpi.a \$(LIBDIR)/libdfftw_mpi.a $flibs_fftw2"
diff -urN amber11.orig/AmberTools/src/Makefile amber11/AmberTools/src/Makefile
--- amber11.orig/AmberTools/src/Makefile	2010-04-14 17:36:31.000000000 +0400
+++ amber11/AmberTools/src/Makefile	2011-03-07 03:39:38.000000000 +0300
@@ -8,15 +8,15 @@
 
 install: serial
 
-serial: configured_serial $(NETCDFLIB)
+serial: configured_serial
 	@echo "Starting installation of ${AMBERTOOLS} serial at `date`".
 #	utility routines and libraries:
-	(cd ucpp-1.3 && $(MAKE) $(UCPP) )
+#	(cd ucpp-1.3 && $(MAKE) $(UCPP) )
 	(cd byacc && $(MAKE) install )
-	(cd arpack && $(MAKE) install );
-	(cd lapack && $(MAKE) $(LAPACK) )
-	(cd blas && $(MAKE) $(BLAS) )
-	(cd c9x-complex && $(MAKE) $(C9XCOMPLEX) )
+#	(cd arpack && $(MAKE) install );
+#	(cd lapack && $(MAKE) $(LAPACK) )
+#	(cd blas && $(MAKE) $(BLAS) )
+#	(cd c9x-complex && $(MAKE) $(C9XCOMPLEX) )
 	(cd etc && $(MAKE) install )
 	(cd chamber && $(MAKE) install )
 	(cd pbsa && $(MAKE) install )
@@ -30,14 +30,14 @@
 	(cd ptraj && $(MAKE) install )
 
 #   miscellaneous:
-	(cd reduce && $(MAKE) install )
+#	(cd reduce && $(MAKE) install )
 
 #   leap and gleap:
 	(cd leap && $(MAKE) install )
 	(cd gleap && $(MAKE) install )
 
 #	nab:
-	(cd cifparse && $(MAKE) install )
+#	(cd cifparse && $(MAKE) install )
 	(cd nab && $(MAKE) install )
 	(cd nss && $(MAKE) install )
 
@@ -50,15 +50,15 @@
 
 nabonly: $(NETCDFLIB) 
 #	utility routines and libraries:
-	(cd ucpp-1.3 && $(MAKE) $(UCPP) )
+#	(cd ucpp-1.3 && $(MAKE) $(UCPP) )
 	(cd byacc && $(MAKE) install )
-	(cd arpack && $(MAKE) install );
-	(cd lapack && $(MAKE) $(LAPACK) )
-	(cd blas && $(MAKE) $(BLAS) )
-	(cd c9x-complex && $(MAKE) $(C9XCOMPLEX) )
+#	(cd arpack && $(MAKE) install );
+#	(cd lapack && $(MAKE) $(LAPACK) )
+#	(cd blas && $(MAKE) $(BLAS) )
+#	(cd c9x-complex && $(MAKE) $(C9XCOMPLEX) )
 
 #	nab:
-	(cd cifparse && $(MAKE) install )
+#	(cd cifparse && $(MAKE) install )
 	(cd nab && $(MAKE) install )
 	(cd nss && $(MAKE) install )
 
diff -urN amber11.orig/AmberTools/src/pbsa/Makefile amber11/AmberTools/src/pbsa/Makefile
--- amber11.orig/AmberTools/src/pbsa/Makefile	2010-04-14 17:26:17.000000000 +0400
+++ amber11/AmberTools/src/pbsa/Makefile	2011-03-07 03:39:15.000000000 +0300
@@ -47,10 +47,10 @@
 	$(LIBDIR)/libpbsa.a $(LIBDIR)/libFpbsa.a
 
 #---------------------------------------------------------------------------
-pbsa$(SFX): $(OBJ) syslib netlib c9x-complex
+pbsa$(SFX): $(OBJ) syslib
 	$(FC) -o pbsa$(SFX) $(OBJ) ../lib/nxtsec.o $(FLIBSF) $(LDFLAGS)
 
-simplepbsa$(SFX): simplepbsa.o gopt.o libpbsa.a sfflib syslib netlib c9x-complex
+simplepbsa$(SFX): simplepbsa.o gopt.o libpbsa.a sfflib syslib
 	$(FC) -o simplepbsa$(SFX) simplepbsa.o gopt.o libpbsa.a \
 	$(LIBDIR)/libsff.a ../lib/nxtsec.o $(FLIBSF) $(LDFLAGS)
 	/bin/mv simplepbsa$(SFX) $(BINDIR)
@@ -76,15 +76,15 @@
 sfflib:
 	cd ../sff && $(MAKE) install
 
-netlib: 
-	cd ../lapack && $(MAKE) $(LAPACK)
-	cd ../blas && $(MAKE) $(BLAS)
-	cd ../arpack && $(MAKE) install
-
-c9x-complex:
-	@if test $(C9XCOMPLEX) != "skip"; then \
-	   cd ../c9x-complex && $(MAKE) libmc.a; \
-	 fi
+#netlib: 
+#	cd ../lapack && $(MAKE) $(LAPACK)
+#	cd ../blas && $(MAKE) $(BLAS)
+#	cd ../arpack && $(MAKE) install
+
+#c9x-complex:
+#	@if test $(C9XCOMPLEX) != "skip"; then \
+#	   cd ../c9x-complex && $(MAKE) libmc.a; \
+#	 fi
 
 %.LIBPBSA.o: %.f
 	$(FPP) -DLIBPBSA $< > _$<
diff -urN amber11.orig/AmberTools/src/ptraj/Makefile amber11/AmberTools/src/ptraj/Makefile
--- amber11.orig/AmberTools/src/ptraj/Makefile	2010-03-31 07:05:16.000000000 +0400
+++ amber11/AmberTools/src/ptraj/Makefile	2011-03-07 03:39:16.000000000 +0300
@@ -60,20 +60,20 @@
 rdparm$(SFX): libs $(NETCDFLIB) $(OBJECTS)
 	$(CC) $(LDFLAGS) -o rdparm$(SFX) $(OBJECTS) $(LIBS) $(NETCDFLIB) $(LM)
 
-ptraj$(SFX): libs netlib $(NETCDFLIB) $(OBJECTS)
+ptraj$(SFX): libs $(NETCDFLIB) $(OBJECTS)
 	$(CC) $(LDFLAGS) -o ptraj$(SFX) $(OBJECTS) $(LIBS) $(NETCDFLIB) $(LM)
 
-ptraj.MPI$(SFX): libs netlib $(PNETCDFLIB) $(OBJECTS)
+ptraj.MPI$(SFX): libs $(PNETCDFLIB) $(OBJECTS)
 	$(CC) $(LDFLAGS) -o ptraj.MPI$(SFX) $(OBJECTS) $(LIBS) $(PNETCDFLIB) $(LM)
 #
 
 libs: 
 	cd pdb && $(MAKE) 
-	cd ../arpack && $(MAKE)
+#	cd ../arpack && $(MAKE)
 
-netlib:
-	cd ../lapack; $(MAKE) $(LAPACK)
-	cd ../blas; $(MAKE) $(BLAS)
+#netlib:
+#	cd ../lapack; $(MAKE) $(LAPACK)
+#	cd ../blas; $(MAKE) $(BLAS)
 
 clean:
 	cd pdb; $(MAKE) clean
diff -urN amber11.orig/AmberTools/src/ptraj/netcdf_ptraj.h amber11/AmberTools/src/ptraj/netcdf_ptraj.h
--- amber11.orig/AmberTools/src/ptraj/netcdf_ptraj.h	2010-03-31 07:05:16.000000000 +0400
+++ amber11/AmberTools/src/ptraj/netcdf_ptraj.h	2011-03-07 03:39:17.000000000 +0300
@@ -4,7 +4,7 @@
 #    include "../pnetcdf/include/pnetcdf.h"
 #    define nc_strerror ncmpi_strerror
 #  else
-#    include "../netcdf/include/netcdf.h"
+#    include <netcdf.h>
 #  endif 
 #endif
 
diff -urN amber11.orig/AmberTools/src/ptraj/ptraj.h amber11/AmberTools/src/ptraj/ptraj.h
--- amber11.orig/AmberTools/src/ptraj/ptraj.h	2010-03-31 07:05:16.000000000 +0400
+++ amber11/AmberTools/src/ptraj/ptraj.h	2011-03-07 03:39:17.000000000 +0300
@@ -94,7 +94,7 @@
 #define nc_strerror ncmpi_strerror
 #include "../pnetcdf/include/pnetcdf.h"
 #else
-#include "../netcdf/include/netcdf.h"
+#include <netcdf.h>
 #endif
 #endif
 
diff -urN amber11.orig/AmberTools/src/sqm/Makefile amber11/AmberTools/src/sqm/Makefile
--- amber11.orig/AmberTools/src/sqm/Makefile	2010-04-14 16:13:28.000000000 +0400
+++ amber11/AmberTools/src/sqm/Makefile	2011-03-07 03:39:17.000000000 +0300
@@ -77,7 +77,7 @@
 install: sqm$(SFX)
 	mv sqm$(SFX) $(BINDIR)
 
-sqm$(SFX): $(SQMOBJ) $(QMOBJ) netlib sys
+sqm$(SFX): $(SQMOBJ) $(QMOBJ) sys
 	$(FC) -o sqm$(SFX) $(SQMOBJ) $(QMOBJ) $(FLIBSF) ../lib/sys.a $(LDFLAGS)
 
 #-----------LIBS
@@ -93,10 +93,10 @@
 sys:
 	cd ../lib; $(MAKE) sys.a
 
-netlib:
-	cd ../lapack; $(MAKE) $(LAPACK)
-	cd ../blas; $(MAKE) $(BLAS)
-	cd ../arpack && $(MAKE) install
+#netlib:
+#	cd ../lapack; $(MAKE) $(LAPACK)
+#	cd ../blas; $(MAKE) $(BLAS)
+#	cd ../arpack && $(MAKE) install
 
 clean:
 	/bin/rm -f *.o _*.f *.mod *.d sqm$(SFX)
