diff -urN amber11.orig/AmberTools/src/nab/database.c amber11/AmberTools/src/nab/database.c
--- amber11.orig/AmberTools/src/nab/database.c	2011-03-10 20:12:15.000000000 +0200
+++ amber11/AmberTools/src/nab/database.c	2012-03-03 23:59:25.225368331 +0200
@@ -671,7 +671,7 @@
 
 static  Bool	zbDBReadLine( db, sLine )
 DATABASE        db;
-char*           sLine;
+char            sLine[MAXDATALINELEN];
 {
 
 
@@ -706,7 +706,7 @@
 
 static  Bool	zbDBReadDataLine( db, sLine )
 DATABASE        db;
-char*           sLine;
+char            sLine[MAXDATALINELEN];
 {
 
 
@@ -1135,7 +1135,7 @@
 char*           PBuffer;
 int             iBufferInc;
 {
-String          sLine;
+String          sLine[MAXDATALINELEN];
 
 
 
