 Makefile_linux_template |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile_linux_template b/Makefile_linux_template
index 3535cab..5ed4719 100755
--- a/Makefile_linux_template
+++ b/Makefile_linux_template
@@ -1,7 +1,8 @@
 #
-OBJ=	PROGRAMNAME.o ../gklib/linux_kleylib  ../ccp4libs_latest_m32_linux/libccp4f.a  ../ccp4libs_latest_m32_linux/libccp4c.a
+OBJ=	PROGRAMNAME.o
+LIBS= -L../gklib -lkley -lccp4f -lccp4c
 EXEC =	PROGRAMNAME
-INCL=	Makefile_linux
+INCL=	Makefile
 SUBS = 
 STATIC = 
 ##SUBS## SUBS = PROGRAMNAME_subs.o
@@ -14,7 +15,8 @@ opt= -O -u -ffixed-line-length-132 -I..
 link= -O
 
 $(EXEC):	$(OBJ) $(SUBS)
-	$(fc) $(link) -o $(EXEC) $(SUBS) $(OBJ) $(STATIC) ; strip $(EXEC)
+	$(fc) $(link) -o $(EXEC) $(SUBS) $(OBJ) $(STATIC) $(LIBS)
+	cp -fv $(EXEC) ../bin/
 
 PROGRAMNAME.o:	PROGRAMNAME.fpp $(INCL)
 	$(fc) $(opt) -c PROGRAMNAME.fpp
