From b8a606460fddc050d0df7aa0e8dfedd3dedb90a8 Mon Sep 17 00:00:00 2001
From: Christoph Junghans <junghans@lanl.gov>
Date: Mon, 21 Oct 2013 19:02:17 -0600
Subject: [PATCH] fixed cmake python checks

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e17b01..6e92faa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,8 +51,8 @@ link_directories(${Boost_LIBRARY_DIRS})
 
 
 # for python
-find_package(PythonInterp REQUIRED VERSION 2)
-find_package(PythonLibs REQUIRED VERSION 2)
+find_package(PythonInterp 2 REQUIRED)
+find_package(PythonLibs 2 REQUIRED)
 include_directories(${PYTHON_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIR} ${PYTHON_INCLUDE_PATH})
 link_directories(${PYTHON_LIBRARY_DIRS})
 
-- 
1.8.1.5

