--- makefile.orig	2009-02-22 02:56:49.000000000 +0200
+++ makefile	2009-02-22 03:07:49.000000000 +0200
@@ -21,7 +21,7 @@
 
 # === Linker to create Main program =====
 ifndef LINKER
-  LINKER = gcc -s
+  LINKER = gcc
 endif
 
 # === C Compiler ===============
@@ -31,7 +31,7 @@
 
 # === C Compiler Options =======
 ifndef C_OPTIONS
-  C_OPTIONS = $(P_INCL) -ansi -O3 -c
+  C_OPTIONS = $(P_INCL) -ansi ${CFLAGS} -c
 #  C_OPTIONS = $(P_INCL) -ansi -O1 -fsigned-char -ffunction-sections -fexpensive-optimizations -fstack-check -fexceptions -Wall -pedantic -Wbad-function-cast -Wreturn-type -Wformat -Wuninitialized -Wcast-align -Wshadow -Wunused -Wunused-value -Wunused-variable -Wunused-function -Wunused-parameter -Wunused-label -Wcomment -Wcast-qual -Wconversion -Wimplicit-function-dec -Wimplicit-int -Wmissing-braces -Wmissing-declarations -Wmissing-prototypes -Wredundant-decls -Wsign-compare -Wfloat-equal -Wstrict-prototypes -Wwrite-strings -Wundef -Waggregate-return -Wchar-subscripts -Wformat-nonliteral -Wnested-externs -Wsequence-point -Wpointer-arith -m32 -mfancy-math-387 -mieee-fp -mno-soft-float -c
 endif
 
