--- CMake/ParaViewCommon.cmake.orig	2012-08-29 13:38:19.392381797 +1200
+++ CMake/ParaViewCommon.cmake	2012-08-29 13:38:47.271407061 +1200
@@ -497,9 +497,9 @@
 
 #########################################################################
 # Configure mpi4py
-IF(PARAVIEW_ENABLE_PYTHON AND PARAVIEW_USE_MPI)
-  ADD_SUBDIRECTORY(Utilities/mpi4py)
-ENDIF(PARAVIEW_ENABLE_PYTHON AND PARAVIEW_USE_MPI)
+#IF(PARAVIEW_ENABLE_PYTHON AND PARAVIEW_USE_MPI)
+#  ADD_SUBDIRECTORY(Utilities/mpi4py)
+#ENDIF(PARAVIEW_ENABLE_PYTHON AND PARAVIEW_USE_MPI)
 
 
 
