--- a/VTK/Utilities/vtkmetaio/metaUtils.cxx
+++ b/VTK/Utilities/vtkmetaio/metaUtils.cxx
@@ -29,6 +29,7 @@
 #include <arpa/inet.h>
 #endif
 
+#include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
 #include <string>
--- a/VTK/Wrapping/Python/PyVTKObject.cxx
+++ b/VTK/Wrapping/Python/PyVTKObject.cxx
@@ -33,6 +33,7 @@
 #include "vtkObjectBase.h"
 #include "vtkDataArray.h"
 
+#include <vtksys/cstddef>
 #include <vtksys/ios/sstream>
 
 //--------------------------------------------------------------------
