diff -Naur ParaView-2.6.0/CMake/ParaViewCommon.cmake ParaView-2.6.0-new/CMake/ParaViewCommon.cmake
--- ParaView-2.6.0/CMake/ParaViewCommon.cmake	2006-10-24 11:12:48.000000000 -0400
+++ ParaView-2.6.0-new/CMake/ParaViewCommon.cmake	2007-02-19 09:42:19.000000000 -0500
@@ -293,7 +293,11 @@
   SET(VTKHDF5_INSTALL_NO_RUNTIME ${PV_INSTALL_NO_RUNTIME})
   SET(VTKHDF5_INSTALL_LIB_DIR ${PV_INSTALL_LIB_DIR})
   SET(PARAVIEW_HDF5_LIBRARIES vtkhdf5)
-  SET(HDF5_ZLIB_HEADER "vtk_zlib.h")
+  IF(VTK_USE_SYSTEM_ZLIB)
+    SET(HDF5_ZLIB_HEADER "zlib.h")
+  ELSE(VTK_USE_SYSTEM_ZLIB)
+    SET(HDF5_ZLIB_HEADER "vtk_zlib.h")
+  ENDIF(VTK_USE_SYSTEM_ZLIB)
   SET(HDF5_INCLUDE_DIR 
     ${ParaView_SOURCE_DIR}/Utilities/hdf5
     ${ParaView_BINARY_DIR}/Utilities/hdf5)
