--- Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt.orig	2010-06-11 22:54:37.423952532 +1200
+++ Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt	2010-06-11 22:55:07.546951156 +1200
@@ -6,5 +6,5 @@
 target_link_libraries(${exe} PointSprite_Rendering vtkIO)
 
 install(TARGETS ${exe} 
-    RUNTIME DESTINATION ${PROJECT_BINARY_DIR}
+    RUNTIME DESTINATION ${PV_INSTALL_BIN_DIR}
 )
