--- Utilities/CMakeLists.txt.orig	2012-08-31 13:55:56.616827748 +1200
+++ Utilities/CMakeLists.txt	2012-08-31 13:56:43.487830247 +1200
@@ -47,7 +47,7 @@
 SET( verdict_INSTALL_LIB_DIR ${VTK_INSTALL_LIB_DIR_CM24})
 ADD_SUBDIRECTORY(verdict)
 
-ADD_SUBDIRECTORY(vtknetcdf)
+#ADD_SUBDIRECTORY(vtknetcdf)
 IF(VTK_USE_METAIO)
   SET(METAIO_FOR_VTK 1)
   ADD_SUBDIRECTORY(vtkmetaio)
--- Utilities/vtk_netcdf.h.orig	2012-08-31 13:57:25.704832693 +1200
+++ Utilities/vtk_netcdf.h	2012-08-31 13:57:59.823834877 +1200
@@ -16,6 +16,6 @@
 #define __vtk_netcdf_h
 
 /* Use the netcdf library configured for VTK.  */
-#include <vtknetcdf/include/netcdf.h>
+#include <netcdf.h>
 
 #endif
--- Utilities/vtk_netcdfcpp.h.orig	2012-08-31 13:58:48.800838210 +1200
+++ Utilities/vtk_netcdfcpp.h	2012-08-31 13:59:31.223841343 +1200
@@ -16,6 +16,6 @@
 #define __vtk_netcdfcpp_h
 
 /* Use the netcdf library configured for VTK.  */
-#include <vtknetcdf/cxx/netcdfcpp.h>
+#include <netcdfcpp.h>
 
 #endif
--- Utilities/vtkexodus2/CMakeLists.txt.orig	2012-08-31 14:01:26.536856845 +1200
+++ Utilities/vtkexodus2/CMakeLists.txt	2012-08-31 14:02:11.503872711 +1200
@@ -241,7 +241,7 @@
 
 
 VTK_ADD_LIBRARY(vtkexoIIc ${cbind_SRCS})
-TARGET_LINK_LIBRARIES(vtkexoIIc vtkNetCDF)
+TARGET_LINK_LIBRARIES(vtkexoIIc netcdf)
 
 # Apply user-defined properties to the library target.
 IF(VTK_LIBRARY_PROPERTIES)
--- Utilities/vtkexodus2/test/CMakeLists.txt.orig	2012-08-31 14:02:39.696882673 +1200
+++ Utilities/vtkexodus2/test/CMakeLists.txt	2012-08-31 14:02:59.743889790 +1200
@@ -21,7 +21,7 @@
   )
 
 ADD_EXECUTABLE( ExoIICTests ${ExoIICTests} )
-TARGET_LINK_LIBRARIES( ExoIICTests vtkCommon vtkexoIIc vtkNetCDF )
+TARGET_LINK_LIBRARIES( ExoIICTests vtkCommon vtkexoIIc netcdf )
 
 SET( TestsToRun ${ExoIICTests} )
 REMOVE( TestsToRun ExoIICTests.cxx )
--- IO/CMakeLists.txt.orig	2012-08-31 14:05:00.657933170 +1200
+++ IO/CMakeLists.txt	2012-08-31 14:06:18.024961299 +1200
@@ -18,7 +18,7 @@
 SET(KIT_PYTHON_LIBS vtkFilteringPythonD)
 SET(KIT_JAVA_LIBS vtkFilteringJava)
 SET(KIT_INTERFACE_LIBRARIES vtkFiltering)
-SET(KIT_LIBS vtkDICOMParser vtkNetCDF vtkNetCDF_cxx LSDyna
+SET(KIT_LIBS vtkDICOMParser netcdf netcdf_c++ LSDyna
   ${_VTK_METAIO_LIB} sqlite3
   ${VTK_PNG_LIBRARIES} ${VTK_ZLIB_LIBRARIES} ${VTK_JPEG_LIBRARIES}
   ${VTK_TIFF_LIBRARIES} ${VTK_EXPAT_LIBRARIES} ${VTK_OGGTHEORA_LIBRARIES}
--- vtkIncludeDirectories.cmake.orig	2012-08-31 15:36:06.599955661 +1200
+++ vtkIncludeDirectories.cmake	2012-08-31 15:37:35.759045094 +1200
@@ -193,7 +193,7 @@
 VTK_THIRD_PARTY_INCLUDE2(TIFF)
 VTK_THIRD_PARTY_INCLUDE2(EXPAT)
 VTK_THIRD_PARTY_INCLUDE2(LIBXML2)
-VTK_THIRD_PARTY_INCLUDE(NETCDF vtknetcdf/include)
+#VTK_THIRD_PARTY_INCLUDE(NETCDF vtknetcdf/include)
 VTK_THIRD_PARTY_INCLUDE2(Exodus2)
 IF(VTK_USE_GL2PS)
   VTK_THIRD_PARTY_INCLUDE2(GL2PS)
