diff -Naur ParaView3/Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt ParaView3.new/Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt
--- ParaView3/Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt	2009-07-20 15:37:07.000000000 -0400
+++ ParaView3.new/Plugins/PointSprite/Examples/Rendering/Cxx/CMakeLists.txt	2009-07-21 11:42:56.000000000 -0400
@@ -5,6 +5,6 @@
 
 target_link_libraries(${exe} CSCS_PointSprite_Rendering vtkIO)
 
-install(TARGETS ${exe} 
-    RUNTIME DESTINATION ${PROJECT_BINARY_DIR}
-)
+#install(TARGETS ${exe} 
+#    RUNTIME DESTINATION ${PROJECT_BINARY_DIR}
+#)
