--- Utilities/Xdmf2/libsrc/CMakeLists.txt.orig	2010-06-05 23:02:46.495999925 +1200
+++ Utilities/Xdmf2/libsrc/CMakeLists.txt	2010-06-07 21:57:25.261616104 +1200
@@ -503,8 +503,8 @@
       RENAME Xdmf.py) # always named the same regardless of build with or without mpi
 
   # Compile Byte Code
-  INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} ${Xdmf_SOURCE_DIR}/CMake/CompileXdmfByteCode.py 
-    ${CMAKE_INSTALL_PREFIX}/${XDMF_WRAP_PYTHON_INSTALL_PY_DIR})")
+#  INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} ${Xdmf_SOURCE_DIR}/CMake/CompileXdmfByteCode.py
+#    ${XDMF_WRAP_PYTHON_INSTALL_DIR})")
   
 ENDIF(XDMF_WRAP_PYTHON)
 
