diff -Naur ParaView3/Utilities/Xdmf2/libsrc/XdmfH5Driver.cxx ParaView3.new/Utilities/Xdmf2/libsrc/XdmfH5Driver.cxx
--- ParaView3/Utilities/Xdmf2/libsrc/XdmfH5Driver.cxx	2009-07-20 15:39:10.000000000 -0400
+++ ParaView3.new/Utilities/Xdmf2/libsrc/XdmfH5Driver.cxx	2009-07-22 16:05:22.000000000 -0400
@@ -132,9 +132,15 @@
 static herr_t H5FD_dsm_flush(H5FD_t *_file);
 #endif
 static int H5FD_dsm_cmp(const H5FD_t *_f1, const H5FD_t *_f2);
+#ifdef H5_USE_16_API
+static haddr_t H5FD_dsm_get_eoa(const H5FD_t *_file, H5FD_mem_t);
+static herr_t H5FD_dsm_set_eoa(H5FD_t *_file, H5FD_mem_t, haddr_t addr);
+static haddr_t H5FD_dsm_get_eof(const H5FD_t *_file);
+#else
 static haddr_t H5FD_dsm_get_eoa(H5FD_t *_file);
 static herr_t H5FD_dsm_set_eoa(H5FD_t *_file, haddr_t addr);
 static haddr_t H5FD_dsm_get_eof(H5FD_t *_file);
+#endif
 static herr_t H5FD_dsm_read(H5FD_t *_file, H5FD_mem_t type, hid_t fapl_id, haddr_t addr,
            DSM_HSIZE_T size, void *buf);
 static herr_t H5FD_dsm_write(H5FD_t *_file, H5FD_mem_t type, hid_t fapl_id, haddr_t addr,
@@ -159,6 +165,7 @@
     H5FD_dsm_close,             /*close         */
     H5FD_dsm_cmp,               /*cmp           */
     NULL,                       /*query         */
+    NULL,                       /*type map      */
     NULL,                       /*alloc         */
     NULL,                       /*free          */
     H5FD_dsm_get_eoa,           /*get_eoa       */
@@ -168,6 +175,7 @@
     H5FD_dsm_read,              /*read          */
     H5FD_dsm_write,             /*write         */
     NULL,                       /*flush         */
+    NULL,                       /*truncate      */
     NULL,                       /*lock          */
     NULL,                       /*unlock        */
     H5FD_FLMAP_SINGLE           /*fl_map        */
@@ -594,7 +602,11 @@
  *-------------------------------------------------------------------------
  */
 static haddr_t
+#ifdef H5_USE_16_API
+H5FD_dsm_get_eoa(const H5FD_t *_file, H5FD_mem_t)
+#else
 H5FD_dsm_get_eoa(H5FD_t *_file)
+#endif
 {
     H5FD_dsm_t  *file = (H5FD_dsm_t*)_file;
 
@@ -622,7 +634,11 @@
  *-------------------------------------------------------------------------
  */
 static herr_t
+#ifdef H5_USE_16_API
+H5FD_dsm_set_eoa(H5FD_t *_file, H5FD_mem_t, haddr_t addr)
+#else
 H5FD_dsm_set_eoa(H5FD_t *_file, haddr_t addr)
+#endif
 {
     H5FD_dsm_t  *file = (H5FD_dsm_t*)_file;
 
@@ -663,7 +679,11 @@
  *-------------------------------------------------------------------------
  */
 static haddr_t
+#ifdef H5_USE_16_API
+H5FD_dsm_get_eof(const H5FD_t *_file)
+#else
 H5FD_dsm_get_eof(H5FD_t *_file)
+#endif
 {
     H5FD_dsm_t  *file = (H5FD_dsm_t*)_file;
 
