--- Applications/AdaptiveParaView/CMakeLists.txt.orig	2010-06-04 22:40:43.973547693 +1200
+++ Applications/AdaptiveParaView/CMakeLists.txt	2010-06-04 22:41:37.889549494 +1200
@@ -45,6 +45,8 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/CustomResources/CustomFilters.xml
     ${CMAKE_CURRENT_SOURCE_DIR}/CustomResources/CustomSources.xml
   SOURCES ${Application_SOURCE_FILES}
+  INSTALL_BIN_DIR "${PV_INSTALL_BIN_DIR}"
+  INSTALL_LIB_DIR "${PV_INSTALL_LIB_DIR}"
 )
 
 ADD_EXECUTABLE(ppRawStridedReader2 ppRawStridedReader2.cxx)
--- Applications/ParaView/CMakeLists.txt.orig	2010-06-04 22:42:36.500548837 +1200
+++ Applications/ParaView/CMakeLists.txt	2010-06-04 22:43:21.355548188 +1200
@@ -63,6 +63,8 @@
       ${PARAVIEW_EXTRA_GUI_RESOURCES} # Add any compiled in GUI configuration xmls
     COMPRESSED_HELP_FILE "${ParaView_BINARY_DIR}/Documentation/paraview.qch"
     SOURCES ${ParaView_SOURCE_FILES}
+    INSTALL_BIN_DIR "${PV_INSTALL_BIN_DIR}"
+    INSTALL_LIB_DIR "${PV_INSTALL_LIB_DIR}"
 )
 
 # Configure the ".plugins" configuration xml for making paraview aware of the
--- Applications/StreamingParaView/CMakeLists.txt.orig	2010-06-04 22:43:46.287547403 +1200
+++ Applications/StreamingParaView/CMakeLists.txt	2010-06-04 22:44:44.047547909 +1200
@@ -45,6 +45,8 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/CustomResources/CustomFilters.xml
     ${CMAKE_CURRENT_SOURCE_DIR}/CustomResources/CustomSources.xml
   SOURCES ${Application_SOURCE_FILES}
+  INSTALL_BIN_DIR "${PV_INSTALL_BIN_DIR}"
+  INSTALL_LIB_DIR "${PV_INSTALL_LIB_DIR}"
 )
 
 IF(FALSE)#BUILD_TESTING)
