--- ParaViewCore/ClientServerCore/vtkPythonProgrammableFilter.cxx.orig	2012-08-28 15:42:34.410099857 +1200
+++ ParaViewCore/ClientServerCore/vtkPythonProgrammableFilter.cxx	2012-08-28 15:45:25.731279418 +1200
@@ -80,6 +80,16 @@
 }
 
 //----------------------------------------------------------------------------
+void vtkPythonProgrammableFilter::DeleteGlobalPythonInterpretor()
+{
+  if(GlobalPipelineInterpretor)
+    {
+    GlobalPipelineInterpretor->Delete();
+    GlobalPipelineInterpretor = 0;
+    }
+}
+
+//----------------------------------------------------------------------------
 vtkPythonProgrammableFilter::vtkPythonProgrammableFilter() :
   Implementation(new vtkPythonProgrammableFilterImplementation())
 {
--- ParaViewCore/ClientServerCore/vtkPythonProgrammableFilter.h.orig	2012-08-28 15:42:45.892355278 +1200
+++ ParaViewCore/ClientServerCore/vtkPythonProgrammableFilter.h	2012-08-28 15:45:54.883213682 +1200
@@ -102,6 +102,7 @@
 //BTX
   static vtkPVPythonInterpretor* GetGlobalPipelineInterpretor();
 //ETX
+static void DeleteGlobalPythonInterpretor();
 protected:
   vtkPythonProgrammableFilter();
   ~vtkPythonProgrammableFilter();
--- Utilities/VTKPythonWrapping/paraview/servermanager.py.orig	2012-08-28 15:47:14.611830975 +1200
+++ Utilities/VTKPythonWrapping/paraview/servermanager.py	2012-08-28 15:48:11.995520340 +1200
@@ -40,7 +40,7 @@
 #     PURPOSE.  See the above copyright notice for more information.
 #
 #==============================================================================
-import paraview, re, os, os.path, new, sys, vtk
+import paraview, re, os, os.path, new, sys, atexit, vtk
 
 if not paraview.compatibility.minor:
     paraview.compatibility.major = 3
@@ -3027,3 +3027,5 @@
 
 def GetNumberOfConnections():
    return len(MultiServerConnections)
+
+atexit.register(vtkPythonProgrammableFilter.DeleteGlobalPythonInterpretor)
