diff -ru ParaView3/Applications/Client/CMakeLists.txt ParaView3.new/Applications/Client/CMakeLists.txt
--- ParaView3/Applications/Client/CMakeLists.txt	2009-07-20 15:36:01.000000000 -0400
+++ ParaView3.new/Applications/Client/CMakeLists.txt	2009-07-22 18:35:55.000000000 -0400
@@ -499,17 +499,17 @@
 
 # Install system libraries on Linux (will not work on any other Unix
 # yet)
-FOREACH(fmpglib avcodec avformat avutil)
-  IF(FFMPEG_${fmpglib}_LIBRARY)
-    GET_FILENAME_COMPONENT(FFMPEG_LIB_DIR_tmp ${FFMPEG_${fmpglib}_LIBRARY} PATH)
-    GET_FILENAME_COMPONENT(FFMPEG_LIB_NAME_tmp ${FFMPEG_${fmpglib}_LIBRARY} NAME)
-    FILE(GLOB FFMPEG_LIB_LIST RELATIVE ${FFMPEG_LIB_DIR_tmp} "${FFMPEG_${fmpglib}_LIBRARY}*")
-    INSTALL(CODE "
-MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${FFMPEG_LIB_NAME_tmp}\")
-EXECUTE_PROCESS (WORKING_DIRECTORY ${FFMPEG_LIB_DIR_tmp}
-                 COMMAND tar c ${FFMPEG_LIB_LIST}
-                 COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
-        " COMPONENT Runtime)
+#FOREACH(fmpglib avcodec avformat avutil)
+#  IF(FFMPEG_${fmpglib}_LIBRARY)
+#    GET_FILENAME_COMPONENT(FFMPEG_LIB_DIR_tmp ${FFMPEG_${fmpglib}_LIBRARY} PATH)
+#    GET_FILENAME_COMPONENT(FFMPEG_LIB_NAME_tmp ${FFMPEG_${fmpglib}_LIBRARY} NAME)
+#    FILE(GLOB FFMPEG_LIB_LIST RELATIVE ${FFMPEG_LIB_DIR_tmp} "${FFMPEG_${fmpglib}_LIBRARY}*")
+#    INSTALL(CODE "
+#MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${FFMPEG_LIB_NAME_tmp}\")
+#EXECUTE_PROCESS (WORKING_DIRECTORY ${FFMPEG_LIB_DIR_tmp}
+#                 COMMAND tar c ${FFMPEG_LIB_LIST}
+#                 COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
+#        " COMPONENT Runtime)
     #INSTALL(FILES ${FFMPEG_${fmpglib}_LIBRARY} DESTINATION ${PV_INSTALL_LIB_DIR})
     
   ENDIF(FFMPEG_${fmpglib}_LIBRARY)
@@ -527,26 +527,26 @@
   SET (QTLIBLIST ${QTLIBLIST} QTOPENGL)
 ENDIF (VTK_USE_QVTK_QTOPENGL)
 
-IF(NOT Q_WS_MAC)
-  FOREACH(qtlib ${QTLIBLIST})
-    IF (NOT WIN32)
-      #INSTALL(FILES ${QT_${qtlib}_LIBRARY_RELEASE} DESTINATION ${PV_INSTALL_LIB_DIR})
-      GET_FILENAME_COMPONENT(QT_LIB_DIR_tmp ${QT_${qtlib}_LIBRARY_RELEASE} PATH)
-      GET_FILENAME_COMPONENT(QT_LIB_NAME_tmp ${QT_${qtlib}_LIBRARY_RELEASE} NAME)
-      FILE(GLOB QT_LIB_LIST RELATIVE ${QT_LIB_DIR_tmp} "${QT_${qtlib}_LIBRARY_RELEASE}*")
-      INSTALL(CODE "
-MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${QT_LIB_NAME_tmp}\")
-EXECUTE_PROCESS (WORKING_DIRECTORY ${QT_LIB_DIR_tmp}
-                 COMMAND tar c ${QT_LIB_LIST}
-                 COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
-        " COMPONENT Runtime)
-    ELSE (NOT WIN32)
-      GET_FILENAME_COMPONENT(QT_DLL_PATH_tmp ${QT_QMAKE_EXECUTABLE} PATH)
-      INSTALL(FILES ${QT_DLL_PATH_tmp}/${qtlib}4.dll DESTINATION ${PV_INSTALL_BIN_DIR} COMPONENT Runtime)
-    ENDIF (NOT WIN32)
-    
-  ENDFOREACH(qtlib)
-ENDIF(NOT Q_WS_MAC)
+#IF(NOT Q_WS_MAC)
+#  FOREACH(qtlib ${QTLIBLIST})
+#    IF (NOT WIN32)
+#      #INSTALL(FILES ${QT_${qtlib}_LIBRARY_RELEASE} DESTINATION ${PV_INSTALL_LIB_DIR})
+#      GET_FILENAME_COMPONENT(QT_LIB_DIR_tmp ${QT_${qtlib}_LIBRARY_RELEASE} PATH)
+#      GET_FILENAME_COMPONENT(QT_LIB_NAME_tmp ${QT_${qtlib}_LIBRARY_RELEASE} NAME)
+#      FILE(GLOB QT_LIB_LIST RELATIVE ${QT_LIB_DIR_tmp} "${QT_${qtlib}_LIBRARY_RELEASE}*")
+#      INSTALL(CODE "
+#MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${QT_LIB_NAME_tmp}\")
+#EXECUTE_PROCESS (WORKING_DIRECTORY ${QT_LIB_DIR_tmp}
+#                 COMMAND tar c ${QT_LIB_LIST}
+#                 COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
+#        " COMPONENT Runtime)
+#    ELSE (NOT WIN32)
+#      GET_FILENAME_COMPONENT(QT_DLL_PATH_tmp ${QT_QMAKE_EXECUTABLE} PATH)
+#      INSTALL(FILES ${QT_DLL_PATH_tmp}/${qtlib}4.dll DESTINATION ${PV_INSTALL_BIN_DIR} COMPONENT Runtime)
+#    ENDIF (NOT WIN32)
+#    
+#  ENDFOREACH(qtlib)
+#ENDIF(NOT Q_WS_MAC)
 
 # Hard-coded install rules for anything left to install. These should
 # probably go to a local cmake file. Change to match the build system's
diff -ru ParaView3/Applications/StreamingParaView/CMakeLists.txt ParaView3.new/Applications/StreamingParaView/CMakeLists.txt
--- ParaView3/Applications/StreamingParaView/CMakeLists.txt	2009-07-20 15:36:15.000000000 -0400
+++ ParaView3.new/Applications/StreamingParaView/CMakeLists.txt	2009-07-22 18:36:49.000000000 -0400
@@ -198,26 +198,26 @@
   SET (QTLIBLIST ${QTLIBLIST} QTASSISTANT)
 ENDIF (WIN32)
 
-IF(NOT Q_WS_MAC)
-  FOREACH(qtlib ${QTLIBLIST})
-    IF (NOT WIN32)
+#IF(NOT Q_WS_MAC)
+#  FOREACH(qtlib ${QTLIBLIST})
+#    IF (NOT WIN32)
       #INSTALL(FILES ${QT_${qtlib}_LIBRARY_RELEASE} DESTINATION ${PV_INSTALL_LIB_DIR})
-      GET_FILENAME_COMPONENT(QT_LIB_DIR_tmp ${QT_${qtlib}_LIBRARY_RELEASE} PATH)
-      GET_FILENAME_COMPONENT(QT_LIB_NAME_tmp ${QT_${qtlib}_LIBRARY_RELEASE} NAME)
-      FILE(GLOB QT_LIB_LIST RELATIVE ${QT_LIB_DIR_tmp} "${QT_${qtlib}_LIBRARY_RELEASE}*")
-      INSTALL(CODE "
-MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${QT_LIB_NAME_tmp}\")
-EXECUTE_PROCESS (WORKING_DIRECTORY ${QT_LIB_DIR_tmp}
-                 COMMAND tar c ${QT_LIB_LIST}
-                 COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
-        " COMPONENT Runtime)
-    ELSE (NOT WIN32)
-      GET_FILENAME_COMPONENT(QT_DLL_PATH_tmp ${QT_QMAKE_EXECUTABLE} PATH)
-      INSTALL(FILES ${QT_DLL_PATH_tmp}/${qtlib}4.dll DESTINATION ${PV_INSTALL_BIN_DIR} COMPONENT Runtime)
-    ENDIF (NOT WIN32)
+      #      GET_FILENAME_COMPONENT(QT_LIB_DIR_tmp ${QT_${qtlib}_LIBRARY_RELEASE} PATH)
+      #GET_FILENAME_COMPONENT(QT_LIB_NAME_tmp ${QT_${qtlib}_LIBRARY_RELEASE} NAME)
+      #      FILE(GLOB QT_LIB_LIST RELATIVE ${QT_LIB_DIR_tmp} "${QT_${qtlib}_LIBRARY_RELEASE}*")
+      #      INSTALL(CODE "
+      #MESSAGE(STATUS \"Installing \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR}/${QT_LIB_NAME_tmp}\")
+      #EXECUTE_PROCESS (WORKING_DIRECTORY ${QT_LIB_DIR_tmp}
+      #           COMMAND tar c ${QT_LIB_LIST}
+      #           COMMAND tar -xC \${CMAKE_INSTALL_PREFIX}/${PV_INSTALL_LIB_DIR})
+      #  " COMPONENT Runtime)
+      #    ELSE (NOT WIN32)
+      #GET_FILENAME_COMPONENT(QT_DLL_PATH_tmp ${QT_QMAKE_EXECUTABLE} PATH)
+      #INSTALL(FILES ${QT_DLL_PATH_tmp}/${qtlib}4.dll DESTINATION ${PV_INSTALL_BIN_DIR} COMPONENT Runtime)
+      # ENDIF (NOT WIN32)
     
-  ENDFOREACH(qtlib)
-ENDIF(NOT Q_WS_MAC)
+      #  ENDFOREACH(qtlib)
+      #ENDIF(NOT Q_WS_MAC)
 
 # Hard-coded install rules for anything left to install. These should
 # probably go to a local cmake file. Change to match the build system's
