--- CommandLineExecutables/CMakeLists.txt.orig	2012-08-28 14:29:41.472454000 +1200
+++ CommandLineExecutables/CMakeLists.txt	2012-08-28 14:30:07.959473756 +1200
@@ -69,15 +69,15 @@
 
 #------------------------------------------------------------------------
 # Install ffmpeg, if used.
-IF (VTK_USE_FFMPEG_ENCODER)
-  FOREACH(fmpglib avcodec avformat avutil)
-    IF (FFMPEG_${fmpglib}_LIBRARY)
-      pv_install_library(${FFMPEG_${fmpglib}_LIBRARY}
-        ${PV_INSTALL_LIB_DIR}
-        BrandedRuntime)
-    ENDIF(FFMPEG_${fmpglib}_LIBRARY)
-  ENDFOREACH(fmpglib)
-ENDIF (VTK_USE_FFMPEG_ENCODER)
+#IF (VTK_USE_FFMPEG_ENCODER)
+#  FOREACH(fmpglib avcodec avformat avutil)
+#    IF (FFMPEG_${fmpglib}_LIBRARY)
+#      pv_install_library(${FFMPEG_${fmpglib}_LIBRARY}
+#        ${PV_INSTALL_LIB_DIR}
+#        BrandedRuntime)
+#    ENDIF(FFMPEG_${fmpglib}_LIBRARY)
+#  ENDFOREACH(fmpglib)
+#ENDIF (VTK_USE_FFMPEG_ENCODER)
 
 IF (NOT PV_INSTALL_NO_RUNTIME)
   INSTALL(TARGETS smTestDriver
