--- EXAMPLE/Makefile.orig	2008-01-31 17:16:16.008471873 +0000
+++ EXAMPLE/Makefile	2008-01-31 17:20:16.581889615 +0000
@@ -19,4 +19,9 @@
-include ../make.inc
+SUPERLULIB   = -lsuperlu
+BLASLIB      = $(BLAS_LIBS)
+FORTRAN	     = $(F77)
+LOADER       = $(LD)
+LOADOPTS     = $(LDFLAGS)
+CDEFS        = -DAdd_
 
 #######################################################################
 #  This makefile creates the example programs for the linear equation
