--- make.inc	2008-08-02 08:06:33.000000000 +0200
+++ make.inc.new	2009-08-28 23:01:27.000000000 +0200
@@ -29,7 +29,7 @@
 # which may be slower than ESSL
 #
 BLASDEF	     	= -DUSE_VENDOR_BLAS
-BLASLIB      	= -lessl
+BLASLIB      	= 'pkg-config --libs blas'
 
 TMGLIB       	= libtmglib.a
 LIBS		= $(SUPERLULIB) $(BLASLIB)
@@ -42,12 +42,9 @@
 ARCHFLAGS    = cr
 RANLIB       = ranlib
 
-CC           = xlc
-CFLAGS       = -O3 -qarch=pwr3 -qalias=allptrs
-FORTRAN	     = xlf
-FFLAGS       = -O3 -qarch=pwr3
-LOADER       = xlc
-LOADOPTS     = -bmaxdata:0x80000000
+FORTRAN	     = $(F77)
+LOADER       = $(CC)
+LOADOPTS     = $(LDFLAGS)
 #
 #  C preprocessor defs for compilation for the Fortran interface
 #  (-DNoChange, -DAdd_, -DAdd__, or -DUpCase)
