diff -Naur octave-3.0.1/scripts/statistics/distributions/chi2rnd.m octave-3.0.1.new/scripts/statistics/distributions/chi2rnd.m
--- octave-3.0.1/scripts/statistics/distributions/chi2rnd.m	2008-04-21 11:58:40.000000000 -0400
+++ octave-3.0.1.new/scripts/statistics/distributions/chi2rnd.m	2008-06-02 06:42:34.000000000 -0400
@@ -68,7 +68,7 @@
      if (find (!(n > 0) | !(n < Inf)))
        rnd = NaN * ones (sz);
      else
-       rnd = 2 * randg(n/2, sz)
+       rnd = 2 * randg(n/2, sz);
      endif
   else
     [retval, n, dummy] = common_size (n, ones (sz));
@@ -84,7 +84,7 @@
 
     k = find ((n > 0) & (n < Inf));
     if (any (k))
-      rnd(k) = 2 * randg(n(k)/2, size(k))
+      rnd(k) = 2 * randg(n(k)/2, size(k));
     endif
   endif
 
