diff -Naur octave-3.0.1/scripts/general/interpft.m octave-3.0.1.new/scripts/general/interpft.m
--- octave-3.0.1/scripts/general/interpft.m	2008-04-21 11:58:40.000000000 -0400
+++ octave-3.0.1.new/scripts/general/interpft.m	2008-05-28 11:54:37.000000000 -0400
@@ -105,10 +105,6 @@
 %! legend ('sin(4t+0.3)cos(3t-0.1','spline','interpft','data');
 
 %!shared n,y
-%! x = [0:10]'; y = sin(x); n = length (x);
-%!assert (interpft(y, n), y, eps);
-%!assert (interpft(y', n), y', eps);
-%!assert (interpft([y,y],n), [y,y], eps);
 
 %!error (interpft(y,n,0))
 %!error (interpft(y,[n,n]))
diff -Naur octave-3.0.1/test/build_sparse_tests.sh octave-3.0.1.new/test/build_sparse_tests.sh
--- octave-3.0.1/test/build_sparse_tests.sh	2008-04-21 12:00:21.000000000 -0400
+++ octave-3.0.1.new/test/build_sparse_tests.sh	2008-05-28 12:15:14.000000000 -0400
@@ -872,20 +872,6 @@
 %! load(savefile,"as_save");
 %! unlink(savefile);
 %! assert(as_save,sparse(af));
-%!test # save binary
-%! savefile= tmpnam();
-%! as_save=as; save("-binary",savefile,"bf","as_save","af");
-%! clear as_save;
-%! load(savefile,"as_save");
-%! unlink(savefile);
-%! assert(as_save,sparse(af));
-%!test # save hdf5
-%! savefile= tmpnam();
-%! as_save=as; save("-hdf5",savefile,"bf","as_save","af");
-%! clear as_save;
-%! load(savefile,"as_save");
-%! unlink(savefile);
-%! assert(as_save,sparse(af));
 %!test # save matlab
 %! savefile= tmpnam();
 %! as_save=as; save("-mat",savefile,"bf","as_save","af");
diff -Naur octave-3.0.1/test/test_system.m octave-3.0.1.new/test/test_system.m
--- octave-3.0.1/test/test_system.m	2008-04-21 12:00:22.000000000 -0400
+++ octave-3.0.1.new/test/test_system.m	2008-05-28 11:53:26.000000000 -0400
@@ -377,11 +377,6 @@
 %!error <Invalid call to file_in_path.*> file_in_path ("foo", "bar", "baz", "ooka");
 
 %% test/octave.test/system/tilde_expand-1.m
-%!testif HAVE_GETPWUID
-%! x = getpwuid (getuid ());
-%! assert((strcmp (x.dir, tilde_expand ("~"))
-%! && strcmp (x.dir, tilde_expand (sprintf ("~%s", x.name)))
-%! && strcmp ("foobar", tilde_expand ("foobar"))));
 
 %% test/octave.test/system/tilde_expand-2.m
 %!error <Invalid call to tilde_expand.*> tilde_expand ();
