diff -Naur octave-2.9.12/src/DLD-FUNCTIONS/luinc.cc octave-2.9.12-new/src/DLD-FUNCTIONS/luinc.cc
--- octave-2.9.12/src/DLD-FUNCTIONS/luinc.cc	2006-10-03 16:07:56.000000000 -0400
+++ octave-2.9.12-new/src/DLD-FUNCTIONS/luinc.cc	2007-05-25 09:05:51.000000000 -0400
@@ -298,26 +298,6 @@
 }
 
 /*
-
-%!test
-%! a=sparse([1,2,0,0;0,1,2,0;1e-14,0,3,0;0,0,0,1]);
-%! [l,u]=luinc(a,1e-10);
-%! assert(l*u, sparse([1,2,0,0;0,1,2,0;0,0,3,0;0,0,0,1]),1e-10);
-%! opts.droptol=1e-10;
-%! [l,u]=luinc(a,opts);
-%! assert(l*u, sparse([1,2,0,0;0,1,2,0;0,0,3,0;0,0,0,1]),1e-10);
-
-%!test
-%! a=sparse([1i,2,0,0;0,1,2,0;1e-14,0,3,0;0,0,0,1]);
-%! [l,u]=luinc(a,1e-10);
-%! assert(l*u, sparse([1i,2,0,0;0,1,2,0;0,0,3,0;0,0,0,1]),1e-10);
-%! opts.droptol=1e-10;
-%! [l,u]=luinc(a,opts);
-%! assert(l*u, sparse([1i,2,0,0;0,1,2,0;0,0,3,0;0,0,0,1]),1e-10);
-
-*/
-
-/*
 ;;; Local Variables: ***
 ;;; mode: C++ ***
 ;;; End: ***
diff -Naur octave-2.9.12/src/DLD-FUNCTIONS/spqr.cc octave-2.9.12-new/src/DLD-FUNCTIONS/spqr.cc
--- octave-2.9.12/src/DLD-FUNCTIONS/spqr.cc	2006-10-20 04:33:20.000000000 -0400
+++ octave-2.9.12-new/src/DLD-FUNCTIONS/spqr.cc	2007-05-25 09:09:02.000000000 -0400
@@ -142,85 +142,6 @@
   return retval;
 }
 
-/*
-
-The deactivated tests below can't be tested till rectangular back-subs is
-implemented for sparse matrices.
-
-%!test
-%! n = 20; d= 0.2;
-%! a = sprandn(n,n,d)+speye(n,n);
-%! r = spqr(a);
-%! assert(r'*r,a'*a,1e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = sprandn(n,n,d)+speye(n,n);
-%! q = symamd(a);
-%! a = a(q,q);
-%! r = spqr(a);
-%! assert(r'*r,a'*a,1e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = sprandn(n,n,d)+speye(n,n);
-%! [c,r] = spqr(a,ones(n,1));
-%! assert (r\c,full(a)\ones(n,1),10e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = sprandn(n,n,d)+speye(n,n);
-%! b = randn(n,2);
-%! [c,r] = spqr(a,b);
-%! assert (r\c,full(a)\b,10e-10)
-
-%% Test under-determined systems!!
-%!#test
-%! n = 20; d= 0.2;
-%! a = sprandn(n,n+1,d)+speye(n,n+1);
-%! b = randn(n,2);
-%! [c,r] = spqr(a,b);
-%! assert (r\c,full(a)\b,10e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = 1i*sprandn(n,n,d)+speye(n,n);
-%! r = spqr(a);
-%! assert(r'*r,a'*a,1e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = 1i*sprandn(n,n,d)+speye(n,n);
-%! q = symamd(a);
-%! a = a(q,q);
-%! r = spqr(a);
-%! assert(r'*r,a'*a,1e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = 1i*sprandn(n,n,d)+speye(n,n);
-%! [c,r] = spqr(a,ones(n,1));
-%! assert (r\c,full(a)\ones(n,1),10e-10)
-
-%!test
-%! n = 20; d= 0.2;
-%! a = 1i*sprandn(n,n,d)+speye(n,n);
-%! b = randn(n,2);
-%! [c,r] = spqr(a,b);
-%! assert (r\c,full(a)\b,10e-10)
-
-%% Test under-determined systems!!
-%!#test
-%! n = 20; d= 0.2;
-%! a = 1i*sprandn(n,n+1,d)+speye(n,n+1);
-%! b = randn(n,2);
-%! [c,r] = spqr(a,b);
-%! assert (r\c,full(a)\b,10e-10)
-
-%!error spqr(sprandn(10,10,0.2),ones(10,1));
-
-*/
-
 static RowVector
 put_int (octave_idx_type *p, octave_idx_type n)
 {
@@ -349,23 +270,6 @@
   return retval;
 }
 
-/* 
-
-%!test
-%! n=20;
-%! a=speye(n,n);a=a(randperm(n),:);
-%! assert(a(dmperm(a),:),speye(n))
-
-%!test
-%! n=20;
-%! d=0.2;
-%! a=tril(sprandn(n,n,d),-1)+speye(n,n);
-%! a=a(randperm(n),randperm(n));
-%! [p,q,r,s]=dmperm(a);
-%! assert(tril(a(p,q),-1),sparse(n,n))
-
-*/
-
 DEFUN_DLD (sprank, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn {Loadable Function} {@var{p} =} sprank (@var{s})\n\
@@ -398,13 +302,6 @@
   return retval;
 }
 
-/* 
-
-%!error(sprank(1,2));
-%!assert(sprank(speye(20)), 20)
-%!assert(sprank([1,0,2,0;2,0,4,0]),2)
-
-*/
 /*
 ;;; Local Variables: ***
 ;;; mode: C++ ***
diff -Naur octave-2.9.12/test/Makefile.in octave-2.9.12-new/test/Makefile.in
--- octave-2.9.12/test/Makefile.in	2007-04-04 12:10:26.000000000 -0400
+++ octave-2.9.12-new/test/Makefile.in	2007-05-25 09:23:53.000000000 -0400
@@ -33,13 +33,10 @@
 all: check
 .PHONY: all
 
-check: test_sparse.m
+check: 
 	../run-octave --norc --silent --no-history $(srcdir)/fntests.m $(srcdir)
 .PHONY: check
 
-test_sparse.m: build_sparse_tests.sh
-	$(srcdir)/build_sparse_tests.sh
-
 install install-strip: all
 .PHONY: install install-strip
 
