diff -Naur octave-3.0.0/scripts/pkg/pkg.m octave-3.0.0.new/scripts/pkg/pkg.m
--- octave-3.0.0/scripts/pkg/pkg.m	2007-12-20 11:55:26.000000000 -0500
+++ octave-3.0.0.new/scripts/pkg/pkg.m	2008-03-16 06:32:07.000000000 -0400
@@ -514,7 +514,7 @@
     ## Warn about non existent files 
     for i = 1:length (files)
       if (isempty (glob(files{i}))) 
-	warning ("file %s does not exist", files{i});
+	error ("file %s does not exist", files{i});
       endif
     endfor
 
