diff --git a/Makefile.am b/Makefile.am
index c522001..4513bc8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -19,7 +19,7 @@ EXTRA_DIST = autogen.sh gsl-config.in gsl.pc.in configure.ac THANKS BUGS gsl.spe
 
 lib_LTLIBRARIES = libgsl.la
 libgsl_la_SOURCES = version.c
-libgsl_la_LIBADD = $(GSL_LIBADD) $(SUBLIBS)
+libgsl_la_LIBADD = $(GSL_LIBADD) $(SUBLIBS) @CBLAS_LINK_LIBS@
 libgsl_la_LDFLAGS = $(GSL_LDFLAGS) -version-info $(GSL_LT_VERSION)
 noinst_HEADERS = templates_on.h templates_off.h build.h
 
@@ -29,10 +29,10 @@ m4data_DATA = gsl.m4
 bin_PROGRAMS = gsl-randist gsl-histogram
 
 gsl_randist_SOURCES = gsl-randist.c
-gsl_randist_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_randist_LDADD = libgsl.la
 
 gsl_histogram_SOURCES = gsl-histogram.c
-gsl_histogram_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_histogram_LDADD = libgsl.la
 
 check_SCRIPTS = test_gsl_histogram.sh pkgconfig.test
 TESTS = test_gsl_histogram.sh pkgconfig.test
@@ -51,6 +51,8 @@ edit = $(SED) \
 	-e 's|@GSL_CFLAGS[@]|$(GSL_CFLAGS)|g' \
 	-e 's|@GSL_LIBM[@]|$(GSL_LIBM)|g' \
 	-e 's|@GSL_LIBS[@]|$(GSL_LIBS)|g' \
+	-e 's|@CBLAS_CFLAGS[@]|$(CBLAS_CFLAGS)|g' \
+	-e 's|@CBLAS_LIBS[@]|$(CBLAS_LIBS)|g' \
 	-e 's|@LIBS[@]|$(LIBS)|g' \
 	-e 's|@VERSION[@]|$(VERSION)|g'
 
diff --git a/configure.ac b/configure.ac
index a26fc1e..564d426 100644
--- a/configure.ac
+++ b/configure.ac
@@ -208,6 +208,16 @@ if test "x$LIBS" = "x" ; then
   AC_CHECK_LIB(m, cos)
 fi
 
+sinclude(ax_cblas.m4)
+AX_CBLAS
+if test "x$CBLAS_LIBS" != "x"; then
+   CBLAS_LINK_LIBS="$CBLAS_LIBS"
+else
+   CBLAS_LINK_LIBS="\$(top_builddir)/cblas/libgslcblas.la"
+   CBLAS_LIBS="-lgslcblas"
+fi
+AC_SUBST(CBLAS_LINK_LIBS)
+
 dnl Remember to put a definition in acconfig.h for each of these
 AC_CHECK_DECLS(feenableexcept,,,[#define _GNU_SOURCE 1
 #include <fenv.h>]) 
diff --git a/doc/examples/Makefile.am b/doc/examples/Makefile.am
index f0dcc02..b2f0c0e 100644
--- a/doc/examples/Makefile.am
+++ b/doc/examples/Makefile.am
@@ -8,5 +8,5 @@ examples_txt = blas.txt block.txt bspline.txt cblas.txt cdf.txt cheb.txt combina
 
 dist_noinst_DATA = $(examples_src) $(examples_txt)
 
-LDADD = ../../libgsl.la ../../cblas/libgslcblas.la
+LDADD = ../../libgsl.la @CBLAS_LINK_LIBS@
 AM_DEFAULT_SOURCE_EXT = .c
diff --git a/gsl-config.in b/gsl-config.in
old mode 100755
new mode 100644
index 3f3fa61..c9c4262
--- a/gsl-config.in
+++ b/gsl-config.in
@@ -58,11 +58,11 @@ while test $# -gt 0; do
 	;;
 
     --cflags)
-       	echo @GSL_CFLAGS@ 
+       	echo @GSL_CFLAGS@ @CBLAS_CFLAGS@ 
        	;;
 
     --libs)
-        : ${GSL_CBLAS_LIB=-lgslcblas}
+        : ${GSL_CBLAS_LIB=@CBLAS_LIBS@}
 	echo @GSL_LIBS@ $GSL_CBLAS_LIB @GSL_LIBM@
        	;;
 
diff --git a/gsl.pc.in b/gsl.pc.in
index 5e9ef21..5a7a0f3 100644
--- a/gsl.pc.in
+++ b/gsl.pc.in
@@ -2,7 +2,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-GSL_CBLAS_LIB=-lgslcblas
+GSL_CBLAS_LIB=@CBLAS_LIBS@
 
 Name: GSL
 Description: GNU Scientific Library
 
