diff -Nur gsl-1.15.orig/blas/Makefile.am gsl-1.15/blas/Makefile.am
--- gsl-1.15.orig/blas/Makefile.am	2010-12-26 17:57:08.000000000 +0000
+++ gsl-1.15/blas/Makefile.am	2012-02-23 17:41:52.000000000 +0000
@@ -5,7 +5,7 @@
 INCLUDES = -I$(top_srcdir)
 
 libgslblas_la_SOURCES = blas.c 
-
+libgslblas_la_LIBADD = $(top_builddir)/cblas/libgslcblas.la
 #check_PROGRAMS = test
 #TESTS = test
 #test_LDADD = libgslblas.la ../ieee-utils/libgslieeeutils.la ../err/libgslerr.la ../test/libgsltest.la ../sys/libgslsys.la
diff -Nur gsl-1.15.orig/gsl-config.in gsl-1.15/gsl-config.in
--- gsl-1.15.orig/gsl-config.in	2010-12-26 17:57:08.000000000 +0000
+++ gsl-1.15/gsl-config.in	2012-02-23 17:47:30.000000000 +0000
@@ -58,12 +58,11 @@
 	;;
 
     --cflags)
-       	echo @GSL_CFLAGS@ 
+       	echo @GSL_CFLAGS@ $(pkg-config --cflags cblas)
        	;;
 
     --libs)
-        : ${GSL_CBLAS_LIB=-lgslcblas}
-	echo @GSL_LIBS@ $GSL_CBLAS_LIB @GSL_LIBM@
+	echo $(pkg-config --libs gsl) @GSL_LIBM@
        	;;
 
     --libs-without-cblas)
diff -Nur gsl-1.15.orig/gsl.pc.in gsl-1.15/gsl.pc.in
--- gsl-1.15.orig/gsl.pc.in	2011-01-31 16:59:17.000000000 +0000
+++ gsl-1.15/gsl.pc.in	2012-02-23 17:40:12.000000000 +0000
@@ -2,10 +2,11 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-GSL_CBLAS_LIB=-lgslcblas
 
 Name: GSL
 Description: GNU Scientific Library
 Version: @VERSION@
-Libs: @GSL_LIBS@ ${GSL_CBLAS_LIB} @GSL_LIBM@ @LIBS@
+Libs: @GSL_LIBS@ 
+Private: @GSL_LIBM@ @LIBS@
 Cflags: @GSL_CFLAGS@
+Requires: cblas
diff -Nur gsl-1.15.orig/Makefile.am gsl-1.15/Makefile.am
--- gsl-1.15.orig/Makefile.am	2011-04-14 16:13:48.000000000 +0100
+++ gsl-1.15/Makefile.am	2012-02-23 17:53:10.000000000 +0000
@@ -25,7 +25,6 @@
 
 MINGW32_HOST = @MINGW32_HOST@
 if MINGW32_HOST
-libgsl_la_LIBADD += cblas/libgslcblas.la
 libgsl_la_LDFLAGS += -no-undefined
 endif
 
@@ -35,13 +34,13 @@
 bin_PROGRAMS = gsl-randist gsl-histogram
 
 gsl_randist_SOURCES = gsl-randist.c
-gsl_randist_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_randist_LDADD = libgsl.la
 
 gsl_histogram_SOURCES = gsl-histogram.c
-gsl_histogram_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_histogram_LDADD = libgsl.la
 
 check_SCRIPTS = test_gsl_histogram.sh pkgconfig.test
-TESTS = test_gsl_histogram.sh pkgconfig.test
+TESTS = test_gsl_histogram.sh
 
 #bin_PROGRAMS = main dummy
 #dummy_SOURCES = version.c
