--- python/setup.py.orig	2009-10-09 21:53:17.000000000 +0100
+++ python/setup.py	2009-10-09 21:53:54.000000000 +0100
@@ -5,7 +5,8 @@
 module = Extension('lwpr',
                     include_dirs = ['../include', 
                        os.path.join(get_python_lib(),'numpy','core','include')],
-                    libraries = ['lwpr'],    
+                    libraries = ['lwpr'],
+                   library_dirs=['../src/.libs'],
                     sources = ['lwprmodule.c'])
 
 setup (name = 'LWPR Module',
