diff -ur wcstools-3.6.5.orig/edhead.c wcstools-3.6.5/edhead.c
--- wcstools-3.6.5.orig/edhead.c	2006-06-22 01:45:55.000000000 +0200
+++ wcstools-3.6.5/edhead.c	2006-08-30 23:31:26.000000000 +0200
@@ -140,7 +140,7 @@
     char headline[160];
     char newname[128];
     char space;
-    char *temphead = "/tmp/edheadXXXXXX";
+    char temphead[] = "/tmp/edheadXXXXXX";
     FILE *fd;
     char *ext, *fname, *imext, *imext1;
     char *editcom;
