+++ wcstools-3.6.6/getdate.c	2006-10-10 12:07:22.000000000 +0200
--- wcstools-3.6.6.orig/getdate.c	2006-10-10 12:06:11.000000000 +0200
@@ -413,7 +413,7 @@
     char *fitsdate, *newfdate, *stdate;
     char temp[64];
     char fyear[16];
-    char ts0[8];
+    char ts0[9];
     char *tchar;
     int its, its1;
     time_t lts;
diff -ur wcstools-3.6.6.orig/libwcs/binread.c wcstools-3.6.6/libwcs/binread.c
--- wcstools-3.6.6.orig/libwcs/binread.c	2006-10-10 12:10:47.000000000 +0200
+++ wcstools-3.6.6/libwcs/binread.c	2006-10-10 12:11:22.000000000 +0200
@@ -1358,7 +1358,7 @@
 	for (i = 0; i < nmag; i++) {
 	    moveb (sc->catline, (char *) st->mag, 2, sc->entmag[0]+(i*2), i*2);
 	    if (sc->byteswapped)
-		binswap2 (&st->mag[i], 2);
+		binswap2 ((char*)&st->mag[i], 2);
 	    st->xmag[i] = 0.01 * (double) st->mag[i];
 	    }
 	}
diff -ur wcstools-3.6.6.orig/libwcs/catutil.c wcstools-3.6.6/libwcs/catutil.c
--- wcstools-3.6.6.orig/libwcs/catutil.c	2006-10-10 12:10:47.000000000 +0200
+++ wcstools-3.6.6/libwcs/catutil.c	2006-10-10 12:12:08.000000000 +0200
@@ -722,7 +722,7 @@
 	    lname = 0;
 	else
 	    lname = strlen (refcatname);
-	catname = (char *)calloc (lname + 16, 1);
+	catname = (char *)calloc (lname + 37, 1);
 	if (lname > 0)
 	    sprintf (catname, "%s sources", refcatname);
 	else
