diff -Nur wcstools-3.6.6/configure.ac wcstools-3.6.6.new/configure.ac
--- wcstools-3.6.6/configure.ac	1970-01-01 01:00:00.000000000 +0100
+++ wcstools-3.6.6.new/configure.ac	2006-12-29 15:55:28.000000000 +0000
@@ -0,0 +1,11 @@
+AC_PREREQ([2.57])
+AC_INIT(wcstools, 3.6.5, [http://tdc-www.harvard.edu/wcstools/])
+AM_INIT_AUTOMAKE([dist-bzip2])
+AM_CONFIG_HEADER(config.h)
+
+AC_PROG_CC
+AC_PROG_LIBTOOL
+AC_C_CONST
+AC_CHECK_LIB(m, sin)
+AC_CONFIG_FILES([libwcs/Makefile Makefile])
+AC_OUTPUT
diff -Nur wcstools-3.6.6/libwcs/Makefile.am wcstools-3.6.6.new/libwcs/Makefile.am
--- wcstools-3.6.6/libwcs/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ wcstools-3.6.6.new/libwcs/Makefile.am	2006-12-29 15:56:04.000000000 +0000
@@ -0,0 +1,14 @@
+lib_LTLIBRARIES = libwcs.la
+wcsdir = $(pkgincludedir)/wcs
+wcs_HEADERS = \
+	wcscat.h wcslib.h wcs.h lwcs.h fitshead.h fitsfile.h imio.h
+libwcs_la_SOURCES = \
+	imsetwcs.c imgetwcs.c matchstar.c findstar.c daoread.c wcscon.c \
+	fitswcs.c wcsinit.c wcs.c ty2read.c webread.c tmcread.c \
+	gscread.c gsc2read.c ujcread.c uacread.c ubcread.c ucacread.c \
+	sdssread.c tabread.c binread.c ctgread.c actread.c catutil.c \
+	imrotate.c fitsfile.c imhfile.c \
+	hget.c hput.c imio.c dateutil.c imutil.c \
+	worldpos.c tnxpos.c dsspos.c platepos.c \
+	sortstar.c platefit.c iget.c fileutil.c polfit.c \
+	wcslib.c lin.c cel.c proj.c sph.c wcstrig.c distort.c
diff -Nur wcstools-3.6.6/Makefile.am wcstools-3.6.6.new/Makefile.am
--- wcstools-3.6.6/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ wcstools-3.6.6.new/Makefile.am	2006-12-29 15:55:40.000000000 +0000
@@ -0,0 +1,149 @@
+AUTOMAKE_OPTIONS = foreign no-dependencies
+SUBDIRS = libwcs
+INCLUDES = -I$(top_srcdir)/libwcs
+
+bin_PROGRAMS = \
+	cphead delwcs delhead edhead fixpix gethead i2f imcat imhead immatch \
+	imrot imsize imstar imwcs scat sethead addpix getpix setpix sky2xy \
+	keyhead skycoor subpix xy2sky wcshead conpix gettab newfits getfits \
+	imstack imextract sumpix remap getcol getdate imfill imsmooth imresize \
+	fileroot filename filext char2sp sp2char crlf isnum isrange simpos
+
+WCSLIBS = -L$(top_srcdir)/libwcs -lwcs
+
+addpix_SOURCES = addpix.c
+addpix_LDADD = $(WCSLIBS)
+
+char2sp_SOURCES = char2sp.c
+char2sp_LDADD = $(WCSLIBS)
+
+conpix_SOURCES = conpix.c
+conpix_LDADD = $(WCSLIBS)
+
+cphead_SOURCES = cphead.c
+cphead_LDADD = $(WCSLIBS)
+
+crlf_SOURCES = crlf.c
+
+delwcs_SOURCES = delwcs.c
+delwcs_LDADD = $(WCSLIBS)
+
+delhead_SOURCES = delhead.c
+delhead_LDADD = $(WCSLIBS)
+
+edhead_SOURCES = edhead.c
+edhead_LDADD = $(WCSLIBS)
+
+filename_SOURCES = filename.c
+
+fileroot_SOURCES = fileroot.c
+
+filext_SOURCES = filext.c
+
+fixpix_SOURCES = fixpix.c
+fixpix_LDADD = $(WCSLIBS)
+
+getcol_SOURCES = getcol.c
+getcol_LDADD = $(WCSLIBS)
+
+getdate_SOURCES = getdate.c
+getdate_LDADD = $(WCSLIBS)
+
+gethead_SOURCES = gethead.c
+gethead_LDADD = $(WCSLIBS)
+
+getfits_SOURCES = getfits.c
+getfits_LDADD = $(WCSLIBS)
+
+getpix_SOURCES = getpix.c
+getpix_LDADD = $(WCSLIBS)
+
+gettab_SOURCES = gettab.c
+gettab_LDADD = $(WCSLIBS)
+
+i2f_SOURCES = i2f.c
+i2f_LDADD = $(WCSLIBS)
+
+imcat_SOURCES = imcat.c
+imcat_LDADD = $(WCSLIBS)
+
+imfill_SOURCES = imfill.c
+imfill_LDADD = $(WCSLIBS)
+
+imresize_SOURCES = imresize.c
+imresize_LDADD = $(WCSLIBS)
+
+imsmooth_SOURCES = imsmooth.c
+imsmooth_LDADD = $(WCSLIBS)
+
+imhead_SOURCES = imhead.c
+imhead_LDADD = $(WCSLIBS)
+
+imrot_SOURCES = imrot.c
+imrot_LDADD = $(WCSLIBS)
+
+imsize_SOURCES = imsize.c
+imsize_LDADD = $(WCSLIBS)
+
+imstack_SOURCES = imstack.c
+imstack_LDADD = $(WCSLIBS)
+
+imextract_SOURCES = imextract.c
+imextract_LDADD = $(WCSLIBS)
+
+imstar_SOURCES = imstar.c
+imstar_LDADD = $(WCSLIBS)
+
+isnum_SOURCES = isnum.c
+isnum_LDADD = $(WCSLIBS)
+
+isrange_SOURCES = isrange.c
+
+imwcs_SOURCES = imwcs.c
+imwcs_LDADD = $(WCSLIBS)
+
+immatch_SOURCES = immatch.c
+immatch_LDADD = $(WCSLIBS)
+
+keyhead_SOURCES = keyhead.c
+keyhead_LDADD = $(WCSLIBS)
+
+newfits_SOURCES = newfits.c
+newfits_LDADD = $(WCSLIBS)
+
+remap_SOURCES = remap.c
+remap_LDADD = $(WCSLIBS)
+
+scat_SOURCES = scat.c
+scat_LDADD = $(WCSLIBS)
+
+sethead_SOURCES = sethead.c
+sethead_LDADD = $(WCSLIBS)
+
+setpix_SOURCES = setpix.c
+setpix_LDADD = $(WCSLIBS)
+
+sky2xy_SOURCES = sky2xy.c
+sky2xy_LDADD = $(WCSLIBS)
+
+skycoor_SOURCES = skycoor.c
+skycoor_LDADD = $(WCSLIBS)
+
+sp2char_SOURCES = sp2char.c
+sp2char_LDADD = $(WCSLIBS)
+
+subpix_SOURCES = subpix.c
+subpix_LDADD = $(WCSLIBS)
+
+sumpix_SOURCES = sumpix.c
+sumpix_LDADD = $(WCSLIBS)
+
+wcshead_SOURCES = wcshead.c
+wcshead_LDADD = $(WCSLIBS)
+
+xy2sky_SOURCES = xy2sky.c
+xy2sky_LDADD = $(WCSLIBS)
+
+simpos_SOURCES = simpos.c
+simpos_LDADD = $(WCSLIBS)
+
