--- libs/libCore/IO.cpp.orig	2006-08-29 02:50:09.000000000 +0200
+++ libs/libCore/IO.cpp	2006-08-29 02:52:10.000000000 +0200
@@ -1785,5 +1785,5 @@
       char header[116];
       time_t t = time(NULL);
-      mat_snprintf(header, 116, 
+      snprintf(header, 116, 
 		   "MATLAB 5.0 MAT-file, Platform: %s, Created on: %s by %s", 
 		   MATIO_PLATFORM, ctime(&t), 
@@ -2190,5 +2190,5 @@
       mat_t *mat = Mat_Open(fname, MAT_ACC_RDONLY);
       if (!mat) {
-	mat = Mat_Open(fname, MAT_ACC_RDONLY | MAT_V4);
+	mat = Mat_Open(fname, MAT_ACC_RDONLY | MAT_FT_MAT4);
       }
       if (mat) {
