diff -Naur pvs-4.2-orig/Makefile.in pvs-4.2-ptch/Makefile.in
--- pvs-4.2-orig/Makefile.in	2008-07-20 08:58:09.000000000 +0000
+++ pvs-4.2-ptch/Makefile.in	2009-09-08 18:38:08.000000000 +0000
@@ -131,6 +131,9 @@
                             : '[^X]*\(X*Emacs [0-9][0-9]\)')
 
 
+ifeq ($(emacsversion),Emacs 23)
+EMACSSUBDIR = emacs23
+else
 ifeq ($(emacsversion),Emacs 22)
 EMACSSUBDIR = emacs22
 else
@@ -156,6 +159,7 @@
 endif
 endif
 endif
+endif
 
 endif # end of buildcmds check
 
