diff -Naur pvs-4.2-orig/BDD/ix86_64-Linux/Makefile pvs-4.2-ptch/BDD/ix86_64-Linux/Makefile
--- pvs-4.2-orig/BDD/ix86_64-Linux/Makefile	2009-07-30 06:55:58.000000000 +0000
+++ pvs-4.2-ptch/BDD/ix86_64-Linux/Makefile	2009-07-30 06:58:50.000000000 +0000
@@ -5,7 +5,7 @@
 LD = gcc
 LDFLAGS = -shared -L./
 CC = gcc
-CFLAGS = -D_POSIX_SOURCE -DSYSV $(INCLUDES) -DLINUX -DLINUX_REDHAT5 -DSIGNALS_LINUX 
+CFLAGS = -D_POSIX_SOURCE -DSYSV $(INCLUDES) -DLINUX -DLINUX_REDHAT5 -DSIGNALS_LINUX -fPIC
 XCFLAGS = -O
 SHELL = /bin/sh
 VPATH = ..:../bdd/utils:../bdd/src:../mu/src
@@ -17,7 +17,7 @@
 
 .SUFFIXES:
 .SUFFIXES: .c .o
-.c.o : ; $(CC) $(XCFLAGS) ${CFLAGS} -c $< -o $@
+.c.o : ; $(CC) $(XCFLAGS) $(CFLAGS) -c $< -o $@
 
 all : mu.so
 
diff -Naur pvs-4.2-orig/src/WS1S/ix86_64-Linux/Makefile pvs-4.2-ptch/src/WS1S/ix86_64-Linux/Makefile
--- pvs-4.2-orig/src/WS1S/ix86_64-Linux/Makefile	2009-07-30 06:55:58.000000000 +0000
+++ pvs-4.2-ptch/src/WS1S/ix86_64-Linux/Makefile	2009-07-30 06:57:19.000000000 +0000
@@ -9,7 +9,7 @@
 LD = gcc
 LDFLAGS = -shared -L./
 CC = gcc
-CFLAGS += -D_POSIX_SOURCE -DSYSV $(INCLUDES)
+CFLAGS += -D_POSIX_SOURCE -DSYSV $(INCLUDES) -fPIC
 XCFLAGS = -O
 SHELL = /bin/sh
 VPATH = ..:../mona/BDD:../mona/DFA:../mona/Mem
@@ -26,12 +26,12 @@
 
 .SUFFIXES:
 .SUFFIXES: .c .o
-.c.o : ; $(CC) $(XCFLAGS) ${CFLAGS} -c $< -o $@
+.c.o : ; $(CC) $(XCFLAGS) $(CFLAGS) -c $< -o $@
 
 all : ws1s.so
 
 ws1s_extended_interface.o : ../ws1s_extended_interface.c
-	$(CC) $(XCFLAGS) ${CFLAGS} -c $< -o $@
+	$(CC) $(XCFLAGS) $(CFLAGS) -c $< -o $@
 
 ws1s.so : ${obj}
 	$(LD) ../ws1s-ld-table $(LDFLAGS) -o ws1s.so ${obj}
