diff -dur pvs-sbcl-4.2.ORIG/src/groundeval/pvs2clean.lisp pvs-sbcl-4.2/src/groundeval/pvs2clean.lisp
--- pvs-sbcl-4.2.ORIG/src/groundeval/pvs2clean.lisp	2007-07-02 14:07:40.000000000 -0600
+++ pvs-sbcl-4.2/src/groundeval/pvs2clean.lisp	2008-08-22 22:12:18.000000000 -0600
@@ -177,7 +177,7 @@
 				      (updateable-free-formal-vars operator)
 				      livevars))))
 	  (if (clean-updateable? (type operator))
-	      (format nil "(pvsSelect ~a ~a)"
+	      (format nil "(pvsSelect ~a ~a ~a)"
 		clean-op clean-arg
 		(mk-clean-funcall clean-op
 				  (list clean-arg))))))))
@@ -375,6 +375,7 @@
 				     livevars))))))
     (if else-part
 	(format nil "~a ~% _ -> ~a"
+	  selections-clean
 	  (pvs2clean* (expression else-part) bindings livevars))
 	selections-clean)))
 
diff -durN pvs-sbcl-4.2.ORIG/src/groundeval/eval-utils.lisp pvs-sbcl-4.2/src/groundeval/eval-utils.lisp
--- pvs-sbcl-4.2.ORIG/src/groundeval/eval-utils.lisp	2007-07-02 14:07:40.000000000 -0600
+++ pvs-sbcl-4.2/src/groundeval/eval-utils.lisp	2008-08-24 14:58:20.000000000 -0600
@@ -68,14 +68,14 @@
 (defmethod print-object ((obj eval-info) stream)
   (if *debugging-print-object*
       (call-next-method)
-      (format stream "~@<#<eval-info ~2I~_~:0Iinternal: ~W~:@_external: ~W>~:>"
+      (format stream "~@<#<eval-info ~2I~_~0:Iinternal: ~W~:@_external: ~W>~:>"
 	(internal obj) (external obj))))
 
 (defmethod print-object ((obj eval-defn-info) stream)
   (if *debugging-print-object*
       (call-next-method)
       (format stream
-	  "~@<#<eval-defn-info ~2I~_~:0Iunary:       ~W~:@_multiary:    ~W~:@_destructive: ~W>~:>"
+	  "~@<#<eval-defn-info ~2I~_~0:Iunary:       ~W~:@_multiary:    ~W~:@_destructive: ~W>~:>"
 	(unary obj) (multiary obj) (destructive obj))))
 
 (defmethod print-object ((obj eval-defn) stream)
diff -durN pvs-sbcl-4.2.ORIG/src/ground-prover/process.lisp pvs-sbcl-4.2/src/ground-prover/process.lisp
--- pvs-sbcl-4.2.ORIG/src/ground-prover/process.lisp	2007-07-02 14:07:40.000000000 -0600
+++ pvs-sbcl-4.2/src/ground-prover/process.lisp	2008-08-24 15:13:06.000000000 -0600
@@ -729,7 +729,7 @@
 ;	(set     (setsolve atf))
 	(tuple   (tupsolve atf))
 	(array   (arraysolve atf))
-	(t       (error "No solver for type " (prtype (arg1 atf))))
+	(t       (error "No solver for type ~a" (prtype (arg1 atf))))
 	)))))
 
 ; ------------------------------------------------------------------
@@ -808,7 +808,7 @@
 ;	(set     (setnsolve atf))
 	(tuple   (tupnsolve atf))
 	(array   (arraynsolve atf))
-	(t       (error "No nsolve for type "
+	(t       (error "No nsolve for type ~a"
 			(or (prtype (arg1 atf)) (prtype (arg2 atf))) ))
 	)))))
 
diff -dur pvs-sbcl-4.2.ORIG/src/prover/eproofcheck.lisp pvs-sbcl-4.2/src/prover/eproofcheck.lisp
--- pvs-sbcl-4.2.ORIG/src/prover/eproofcheck.lisp	2008-01-14 04:12:56.000000000 -0700
+++ pvs-sbcl-4.2/src/prover/eproofcheck.lisp	2008-08-22 22:07:06.000000000 -0600
@@ -3446,7 +3446,7 @@
 			(values 'X nil nil))
 		       ((and (or (digit-char-p (char strlbl 0))
 				 (char= (char strlbl 0) #\-))
-			     (every #'digit-char-p (subseq strlbl) 1))
+			     (every #'digit-char-p (subseq strlbl 1)))
 			(error-format-if
 			 "~%Label cannot be an integer"))
 		       (t
diff -durN pvs-sbcl-4.2.ORIG/src/tex-support.lisp pvs-sbcl-4.2/src/tex-support.lisp
--- pvs-sbcl-4.2.ORIG/src/tex-support.lisp	2008-03-11 16:53:37.000000000 -0600
+++ pvs-sbcl-4.2/src/tex-support.lisp	2008-08-24 15:18:16.000000000 -0600
@@ -532,7 +532,7 @@
 	     (when (and *report-mode*
 			(loop for sf in  neg-s-forms
 			      thereis (memq sf par-sforms)))
-	       (format stream "~% & $\\vdots$ \\\\" *prover-indent*)))
+	       (format stream "~%~VT & $\\vdots$ \\\\" *prover-indent*)))
 	    (t (format stream "\\strut\\\\")))
       (format stream "\\hline~%")
       (cond (pos-s-forms
diff -durN pvs-sbcl-4.2.ORIG/src/WS1S/lisp/ws1s-strategy.lisp pvs-sbcl-4.2/src/WS1S/lisp/ws1s-strategy.lisp
--- pvs-sbcl-4.2.ORIG/src/WS1S/lisp/ws1s-strategy.lisp	2007-07-02 14:07:39.000000000 -0600
+++ pvs-sbcl-4.2/src/WS1S/lisp/ws1s-strategy.lisp	2008-08-24 15:20:16.000000000 -0600
@@ -193,7 +193,7 @@
     (loop* 0 (emptyset-operator))))
 
 (defun ws1s-automaton-output (p num fvars offsets)
-  (format t "~2%Free vars:~2%" fvars)
+  (format t "~2%Free vars:~{ ~s~}~2%" fvars)
   (dfa-print p num fvars offsets)
   (format t "~%"))
