diff -Naur wxMaxima-0.6.5/data/Makefile.am wxMaxima-0.6.5.new/data/Makefile.am
--- wxMaxima-0.6.5/data/Makefile.am	2006-01-30 22:52:16.000000000 +0000
+++ wxMaxima-0.6.5.new/data/Makefile.am	2006-05-11 16:12:58.000000000 +0100
@@ -1,4 +1,4 @@
 wxmaximadatadir = ${datadir}/wxMaxima
-wxmaximadata_DATA = docs.zip intro.zip tips.txt wxmathml.lisp
+wxmaximadata_DATA = tips.txt wxmathml.lisp
 
-EXTRA_DIST = docs.zip intro.zip tips.txt wxmathml.lisp Info.plist.in PkgInfo
+EXTRA_DIST = tips.txt wxmathml.lisp
diff -Naur wxMaxima-0.6.5/Makefile.am wxMaxima-0.6.5.new/Makefile.am
--- wxMaxima-0.6.5/Makefile.am	2006-03-25 04:09:42.000000000 +0000
+++ wxMaxima-0.6.5.new/Makefile.am	2006-05-11 16:11:45.000000000 +0100
@@ -12,7 +12,7 @@
              art/maximaicon.xpm art/maximaicon.ico                      \
              art/wxmac.icns
 
-docdir = ${datadir}/wxMaxima
+docdir = ${datadir}/doc/#PF#
 doc_DATA = README COPYING
 
 wxMaxima.app: all
diff -Naur wxMaxima-0.6.5/src/wxMaxima.cpp wxMaxima-0.6.5.new/src/wxMaxima.cpp
--- wxMaxima-0.6.5/src/wxMaxima.cpp	2006-04-07 23:49:41.000000000 +0100
+++ wxMaxima-0.6.5.new/src/wxMaxima.cpp	2006-05-11 16:14:27.000000000 +0100
@@ -2143,7 +2143,7 @@
       filename = wxT(MACPREFIX);
 #else
       filename = wxT(PREFIX);
-      filename += wxT("/share/wxMaxima/");
+      filename += wxT("/share/doc/#PF#/");
       if (!wxFileExists(filename + wxT("docs.zip")))
       {
         filename = wxT(PREFIX);
