--- src/Makefile	2009-06-04 14:25:15.000000000 +0200
+++ src/Makefile.new	2009-06-04 14:27:35.000000000 +0200
@@ -1175,7 +1175,7 @@
 
 rasmol: $(OBJS) $(DEPLIBS)
 	$(RM) $@
-	$(CCLINK) -o $@ $(LDOPTIONS) $(OBJS) $(LOCAL_LIBRARIES) $(LDLIBS)  $(EXTRA_LOAD_FLAGS)
+	$(CCLINK) $(LDOPTIONS) $(LDFLAGS) -o $@ $(OBJS) $(LOCAL_LIBRARIES) $(LDLIBS)  $(EXTRA_LOAD_FLAGS)
 
 install:: rasmol
 	@if [ -d $(DESTDIR)$(BINDIR) ]; then \
