diff --git a/libcfs/libcfs/util/parser.c b/libcfs/libcfs/util/parser.c
index f395fa3..3fc0373 100644
--- a/libcfs/libcfs/util/parser.c
+++ b/libcfs/libcfs/util/parser.c
@@ -308,7 +308,7 @@ int init_input()
                 rl_deprep_term_function = (rl_voidfunc_t *)noop_fn;
         }
 
-        rl_attempted_completion_function = (CPPFunction *)command_completion;
+        rl_attempted_completion_function = (rl_completion_func_t *)command_completion;
         rl_completion_entry_function = (void *)command_generator;
 #endif
         return interactive;
