diff -urN lustre-2.8.0.orig/config/lustre-build.m4 lustre-2.8.0/config/lustre-build.m4
--- lustre-2.8.0.orig/config/lustre-build.m4	2016-06-12 23:18:39.992712332 +0300
+++ lustre-2.8.0/config/lustre-build.m4	2016-06-12 23:21:39.928433319 +0300
@@ -266,7 +266,7 @@
 #
 AC_DEFUN([LB_CONFIG_HEADERS], [
 AC_CONFIG_HEADERS([config.h])
-CPPFLAGS="-include $PWD/undef.h -include $PWD/config.h $CPPFLAGS"
+CPPFLAGS="-include $PWD/config.h $CPPFLAGS"
 EXTRA_KCFLAGS="-include $PWD/undef.h -include $PWD/config.h $EXTRA_KCFLAGS"
 AC_SUBST(EXTRA_KCFLAGS)
 ]) # LB_CONFIG_HEADERS
