diff -Naurp frama-c-Boron-20100401/share/Makefile.dynamic frama-c-Boron-20100401b/share/Makefile.dynamic
--- frama-c-Boron-20100401/share/Makefile.dynamic	2010-04-13 11:52:53.000000000 +0000
+++ frama-c-Boron-20100401/share/Makefile.dynamic	2010-07-04 19:48:52.000000000 +0000
@@ -113,7 +113,7 @@ endif #FRAMAC_MAKE
 
 PLUGIN_LIB_DIR	?= $(PLUGIN_DIR)
 PLUGIN_GUI_LIB_DIR ?= $(PLUGIN_DIR)/gui
-PLUGIN_INSTALL_DIR ?=$(FRAMAC_PLUGINDIR)
+PLUGIN_INSTALL_DIR ?=$(DESTDIR)$(FRAMAC_PLUGINDIR)
 
 PLUGIN_FLAGS:=$(FLAGS) $(DEBUG) $(FRAMAC_INCLUDES) -I $(FRAMAC_LIBDIR)
 PLUGIN_BFLAGS:=$(PLUGIN_FLAGS) $(PLUGIN_BFLAGS)
@@ -182,6 +182,7 @@ endif
 ifneq ($(PLUGIN_ENABLE),no)
 install::
 	$(PRINT_CP) $(PLUGIN_INSTALL_DIR)
+	$(MKDIR) $(PLUGIN_INSTALL_DIR)
 	$(CP) $(TARGETS) $(PLUGIN_INSTALL_DIR)
 	$(PRINT_CP) $(BINDIR)
 	if [ -f frama-c-$(PLUGIN_NAME).byte$(EXE) ]; then \
@@ -191,9 +192,9 @@ install::
 	  $(CP) frama-c-$(PLUGIN_NAME).$(OCAMLBEST)$(EXE) \
 		$(BINDIR)/frama-c-$(PLUGIN_NAME)$(EXE); \
 	fi
-	$(PRINT_UPDATE) $(FRAMAC_SHARE)/known_plugins.ac
-	echo "ENABLE_`echo $(PLUGIN_NAME) | tr "a-z" "A-Z"`=$(PLUGIN_ENABLE)" \
-	  >> $(FRAMAC_SHARE)/known_plugins.ac
+	#$(PRINT_UPDATE) $(FRAMAC_SHARE)/known_plugins.ac
+	#echo "ENABLE_`echo $(PLUGIN_NAME) | tr "a-z" "A-Z"`=$(PLUGIN_ENABLE)" \
+	#  >> $(FRAMAC_SHARE)/known_plugins.ac
 ifeq ($(HAS_GUI),yes)
 	$(PRINT_CP) $(PLUGIN_INSTALL_DIR)/gui
 	$(CP) $(TARGETS_GUI) $(PLUGIN_INSTALL_DIR)/gui
