diff -Naur frama-c-Lithium-20081201-orig/src/from/register.ml frama-c-Lithium-20081201-patch/src/from/register.ml
--- frama-c-Lithium-20081201-orig/src/from/register.ml	2008-12-16 10:29:48.000000000 +0100
+++ frama-c-Lithium-20081201-patch/src/from/register.ml	2009-02-16 14:43:55.000000000 +0100
@@ -572,7 +572,7 @@
     Db.Value.is_reachable (Values_To_Use.get_state (Kstmt stmt))
 
   (* Remove all local variables and formals from table *)
-  let externalize return fundec state =
+  let externalize return fundec state strip =
     let deps_return = (match return.skind with
                          | Return (Some (Lval v),_) ->
                              let deps, looking_for =
@@ -596,9 +596,10 @@
                              Lmap_bitwise.From_Model.LOffset.empty
                          | _ -> assert false)
     in
-    let deps_table = Lmap_bitwise.From_Model.filter_base
+    let deps_table = if strip then Lmap_bitwise.From_Model.filter_base
       (fun v -> not (Base.is_formal_or_local v fundec))
       state.deps_table
+      else state.deps_table
     in
     { deps_return = deps_return;
       Function_Froms.deps_table = deps_table }
@@ -617,7 +618,7 @@
     else Dataflow.GUnreachable
 end
 
-let compute_using_cfg kf = match kf.fundec with
+let compute_using_cfg kf ksid = match kf.fundec with
   | Declaration _ -> assert false
   | Definition (f,_) ->
       try
@@ -656,6 +657,10 @@
               Format.printf "[from] stack with %s\n@\n" (get_name g))
               call_stack;*)
 
+            let sid,strip = match ksid with
+                              None -> ret_id.sid,true
+                            | Some id -> id,false
+            in
             let last_from =
               try
 		if Db.Value.is_reachable (Values_To_Use.get_state (Kstmt ret_id))
@@ -663,7 +668,8 @@
                   Computer.externalize
                   ret_id
                   f
-                  (Computer.StmtStartData.find ret_id.sid)
+                  (Computer.StmtStartData.find sid)
+                  strip
 		else
 		  raise Not_found
               with Not_found -> begin
@@ -806,7 +812,7 @@
 
   let result = match kf.fundec with
     | Definition _ ->
-        compute_using_cfg kf
+        compute_using_cfg kf None
     | Declaration _ ->
         compute_using_prototype kf
   in
@@ -846,7 +852,8 @@
 let () =
   force_compute := From2.compute;
   Db.From.compute := (fun kf -> ignore (Functionwise_From_to_use.memo kf));
-  Db.From.get := Functionwise_From_to_use.memo
+  Db.From.get := Functionwise_From_to_use.memo;
+  Db.From.compute_using_cfg := From2.compute_using_cfg
 
 let () = Db.From.pretty :=
   (fun fmt v ->
diff -Naur frama-c-Lithium-20081201-orig/src/kernel/db.ml frama-c-Lithium-20081201-patch/src/kernel/db.ml
--- frama-c-Lithium-20081201-orig/src/kernel/db.ml	2009-02-14 00:10:32.000000000 +0100
+++ frama-c-Lithium-20081201-patch/src/kernel/db.ml	2009-02-16 14:43:55.000000000 +0100
@@ -437,6 +437,7 @@
   let is_computed = mk_fun "From.is_computed"
   let pretty = mk_fun "From.pretty"
   let get = mk_fun "From.get"
+  let compute_using_cfg = mk_fun "From.compute_using_cfg"
   let self = ref Project.Computation.dummy
 
   let display = mk_fun "From.display"
diff -Naur frama-c-Lithium-20081201-orig/src/kernel/db.mli frama-c-Lithium-20081201-patch/src/kernel/db.mli
--- frama-c-Lithium-20081201-orig/src/kernel/db.mli	2009-02-14 00:10:32.000000000 +0100
+++ frama-c-Lithium-20081201-patch/src/kernel/db.mli	2009-02-16 14:43:55.000000000 +0100
@@ -299,6 +299,7 @@
 	@return true iff the analysis has been performed *)
  
   val get : (kernel_function -> Function_Froms.t) p_ref
+  val compute_using_cfg : (kernel_function -> int option -> Function_Froms.t) p_ref
   val access : (Locations.Zone.t -> from_model -> Locations.Zone.t) p_ref
   val find_deps_no_transitivity : (kinstr -> exp -> Locations.Zone.t) p_ref
   val self: Project.Computation.t p_ref
