diff -Naur frama-c-Beryllium-20090601-beta1-orig/Makefile.in frama-c-Beryllium-20090601-beta1-ptch/Makefile.in
--- frama-c-Beryllium-20090601-beta1-orig/Makefile.in	2009-06-23 11:24:39.000000000 +0000
+++ frama-c-Beryllium-20090601-beta1-ptch/Makefile.in	2009-09-04 15:32:40.000000000 +0000
@@ -358,10 +358,12 @@
 GRAPH_GUICMX= $(GRAPH_GUICMO:.cmo=.cmx)
 GRAPH_GUIO= $(GRAPH_GUICMO:.cmo=.o)
 HAS_VIEWGRAPH=yes
+ifneq ("$(OCAMLGRAPH_LOCAL)","")
 BYTE_GUI_LIBS+= $(GRAPH_GUICMO)
 OPT_GUI_LIBS+= $(GRAPH_GUICMX)
+endif
 endif
 endif
 
 endif # whether ocamlgraph is local
 
