From: Mehdi Dogguy <mehdi@debian.org>
Date: Wed, 9 Sep 2009 17:25:18 +0200
Subject: [PATCH] Add dGraphView.cmo when linking

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 65d9f77..3a70533 100644
--- a/Makefile
+++ b/Makefile
@@ -357,7 +357,7 @@ OPT_LIBS+= graph.cmxa
 ifeq ($(HAS_GNOMECANVAS),yes)
 ifeq ($(ENABLE_SYNTACTIC_CALLGRAPH),yes)
 GRAPH_GUICMO_BASE= xDotDraw.cmo dGraphModel.cmo xDot.cmo dGraphViewItem.cmo \
-	concreteModel.cmo
+	dGraphView.cmo concreteModel.cmo
 GRAPH_GUICMO_BASE+= viewGraph.cmo viewGraph_select.cmo viewGraph_utils.cmo
 GRAPH_GUICMO=$(GRAPH_GUICMO_BASE:%=$(OCAMLGRAPH_HOME)/%)
 GRAPH_GUICMX= $(GRAPH_GUICMO:.cmo=.cmx)
-- 
