diff -Naur frama-c-Beryllium-20090601-beta1-orig/src/ai/base.ml frama-c-Beryllium-20090601-beta1-ptch/src/ai/base.ml
--- frama-c-Beryllium-20090601-beta1-orig/src/ai/base.ml	2009-05-15 09:10:25.000000000 +0000
+++ frama-c-Beryllium-20090601-beta1-ptch/src/ai/base.ml	2009-07-12 17:10:54.000000000 +0000
@@ -271,6 +271,11 @@
   | Var (t,_) | Initialized_Var (t,_) -> t
   | _ -> assert false
 
+let get_varinfo t =
+  match t with
+  | Var (t,_) | Initialized_Var (t,_) -> t
+  | _ -> assert false
+
 let create_varinfo varinfo =
   assert (not varinfo.vlogic);
   Var (varinfo,validity_from_type varinfo)
diff -Naur frama-c-Beryllium-20090601-beta1-orig/src/ai/base.mli frama-c-Beryllium-20090601-beta1-ptch/src/ai/base.mli
--- frama-c-Beryllium-20090601-beta1-orig/src/ai/base.mli	2009-05-05 14:29:35.000000000 +0000
+++ frama-c-Beryllium-20090601-beta1-ptch/src/ai/base.mli	2009-07-12 17:09:54.000000000 +0000
@@ -61,6 +61,7 @@
 val is_block_local: t -> Cil_types.block -> bool
 val is_hidden_variable : t -> bool
 val validity_from_type : Cil_types.varinfo -> validity
+val get_varinfo : t -> Cil_types.varinfo
 val create_varinfo : Cil_types.varinfo -> t
 val create_logic :  Cil_types.varinfo -> validity -> t
 val create_initialized :  Cil_types.varinfo -> validity -> t
