--- gmsh-2.2.5.orig/Mesh/Partition.cpp  2008-11-26 18:07:27.000000000 +0100
+++ gmsh-2.2.5/Mesh/Partition.cpp       2008-11-26 18:08:04.000000000 +0100
@@ -146,8 +146,8 @@
   int ier = 0;

   switch(options.partitioner){
-  case 1:  // Chacho
 #ifdef HAVE_CHACO
+  case 1:  // Chacho
     {
       Msg::Info("Launching Chaco graph partitioner");
       // Some setup (similar to that of Chaco/input/input.c)
@@ -196,8 +196,8 @@
     }
     break;
 #endif
-  case 2:  // Metis
 #ifdef HAVE_METIS
+  case 2:  // Metis
     {
       Msg::Info("Launching METIS graph partitioner");
       // "C" numbering for Metis
