--- maxima.el~	2004-07-20 19:18:05.000000000 +0200
+++ maxima.el	2004-08-03 18:17:47.559653376 +0200
@@ -50,7 +50,7 @@
 ;; Some variables that may have to be set in order to use the maxima help:
 ;;  maxima-info-dir
 ;;         This should be the directory where the maxima info files are kept.
-;;         By default, it is "/usr/local/info/"
+;;         By default, it is "/usr/share/info/"
 ;;  maxima-info-index-file
 ;;         This should be the name of the maxima info file that contains
 ;;         the index, by default, it is "maxima.info-15"
@@ -213,7 +213,7 @@
 (defvar maxima-newline-style nil
   "For compatability.")

-(defcustom maxima-info-dir "/usr/local/info/"
+(defcustom maxima-info-dir "/usr/share/info/"
   "*The directory where the maxima info files are kept."
   :group 'maxima
   :type '(directory))
