diff -U2 --recursive maxima-5.11.0-orig/src/Makefile.am maxima-5.11.0/src/Makefile.am
--- maxima-5.11.0-orig/src/Makefile.am	2006-11-25 13:00:04.000000000 +0100
+++ maxima-5.11.0/src/Makefile.am	2006-12-26 23:53:46.000000000 +0100
@@ -152,5 +152,5 @@
 
 if SBCL
-EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --eval
+EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --disable-debugger --eval
 all-local: binary-sbcl/maxima.core
 install-exec-local: install-sbcl
diff -U2 --recursive maxima-5.11.0-orig/src/Makefile.in maxima-5.11.0/src/Makefile.in
--- maxima-5.11.0-orig/src/Makefile.in	2006-12-21 17:49:36.000000000 +0100
+++ maxima-5.11.0/src/Makefile.in	2006-12-26 23:54:21.000000000 +0100
@@ -202,5 +202,5 @@
 @SCL_TRUE@EXECUTESCL = $(SCL_NAME) -noinit -batch
 @ACL_TRUE@EXECUTEACL = $(ACL_NAME) -batch
-@SBCL_TRUE@EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --eval
+@SBCL_TRUE@EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --disable-debugger --eval
 @GCL_TRUE@EXECUTEGCL = $(GCL_NAME) -batch -eval
 @OPENMCL_TRUE@EXECUTEOPENMCL = $(OPENMCL_NAME) -e
