diff -U2 --recursive maxima-5.11.0-orig/interfaces/xmaxima/Tkmaxima/Menu.tcl maxima-5.11.0/interfaces/xmaxima/Tkmaxima/Menu.tcl
--- maxima-5.11.0-orig/interfaces/xmaxima/Tkmaxima/Menu.tcl	2006-12-09 20:52:11.000000000 +0100
+++ maxima-5.11.0/interfaces/xmaxima/Tkmaxima/Menu.tcl	2006-12-26 22:59:33.000000000 +0100
@@ -246,7 +246,7 @@
     } else {
 	
-	set selectedbrowser mozilla
+	set selectedbrowser firefox
 
-	foreach b { firefox mozilla konqueror epiphany galeon amaya netscape } {
+	foreach b { firefox konqueror epiphany galeon mozilla opera } {
 	    if { ! [catch {exec which $b} ] } {
 		set selectedbrowser $b
