diff -U2 --recursive maxima-5.11.0-orig/interfaces/emacs/misc/bookmode.el maxima-5.11.0/interfaces/emacs/misc/bookmode.el
--- maxima-5.11.0-orig/interfaces/emacs/misc/bookmode.el	2000-05-08 08:09:44.000000000 +0200
+++ maxima-5.11.0/interfaces/emacs/misc/bookmode.el	2006-12-26 23:29:42.000000000 +0100
@@ -681,5 +681,5 @@
 		  (and tem
 		       (list (concat "View "(nth 1 tem))
-			     'call-process  "ghostview" nil nil nil
+			     'call-process  "gv" nil nil nil
 			     (expand-file-name(nth 1 tem))
 			     ))
@@ -706,5 +706,5 @@
 			  (cond ((looking-at "%PS") "")
 				(t "gzip -dc | "))
-			  "ghostview -")
+			  "gv -")
 			 
 	)))
diff -U2 --recursive maxima-5.11.0-orig/interfaces/xmaxima/Tkmaxima/Constants.tcl maxima-5.11.0/interfaces/xmaxima/Tkmaxima/Constants.tcl
--- maxima-5.11.0-orig/interfaces/xmaxima/Tkmaxima/Constants.tcl	2006-10-02 00:42:48.000000000 +0200
+++ maxima-5.11.0/interfaces/xmaxima/Tkmaxima/Constants.tcl	2006-12-26 23:31:47.000000000 +0100
@@ -144,5 +144,5 @@
     image/png  netmath
     image/jpeg netmath
-    application/postscript "ghostview -safer %s"
+    application/postscript "gv -safer %s"
     application/pdf "acroread %s"
     application/x-dvi "xdvi %s"
diff -U2 --recursive maxima-5.11.0-orig/src/plot.lisp maxima-5.11.0/src/plot.lisp
--- maxima-5.11.0-orig/src/plot.lisp	2006-10-10 23:13:17.000000000 +0200
+++ maxima-5.11.0/src/plot.lisp	2006-12-26 23:33:25.000000000 +0100
@@ -76,5 +76,5 @@
                         ))
 
-(defvar $viewps_command  "(ghostview \"~a\")")
+(defvar $viewps_command  "(gv \"~a\")")
 
 ;; $plot_realpart option is false by default but *plot-realpart* is true because coerce-float-fun
