diff -r -U4 maxima-5.10.0.orig/interfaces/emacs/emaxima/maxima.el maxima-5.10.0/interfaces/emacs/emaxima/maxima.el
--- maxima-5.10.0.orig/interfaces/emacs/emaxima/maxima.el	2005-06-22 05:48:42.000000000 +0200
+++ maxima-5.10.0/interfaces/emacs/emaxima/maxima.el	2006-09-21 20:04:48.000000000 +0200
@@ -2704,9 +2704,9 @@
               (process-mark inferior-maxima-process))))
 
 (defun inferior-maxima-wait-for-output ()
   "Wait for output from the Maxima process."
-  (while (and 
+  (when (and 
           inferior-maxima-waiting-for-output
           (inferior-maxima-running))
     (accept-process-output inferior-maxima-process))
   (if maxima-running-xemacs
@@ -2759,10 +2759,10 @@
         (add-to-list 'comint-output-filter-functions
                      'inferior-maxima-output-filter)
 ;        (add-to-list 'comint-output-filter-functions
 ;                     'inferior-maxima-replace-tabs-by-spaces)
-        (add-to-list 'comint-output-filter-functions
-                     'inferior-maxima-remove-double-input-prompt)
+;        (add-to-list 'comint-output-filter-functions
+;                     'inferior-maxima-remove-double-input-prompt)
 	(if maxima-fix-double-prompt
             (add-to-list 'comint-output-filter-functions
                          'inferior-maxima-remove-double-prompt))
         (inferior-maxima-wait-for-output)
diff -r -U4 maxima-5.10.0.orig/interfaces/emacs/misc/bookmode.el maxima-5.10.0/interfaces/emacs/misc/bookmode.el
--- maxima-5.10.0.orig/interfaces/emacs/misc/bookmode.el	2000-05-08 08:09:44.000000000 +0200
+++ maxima-5.10.0/interfaces/emacs/misc/bookmode.el	2006-09-21 20:18:03.000000000 +0200
@@ -679,9 +679,9 @@
 			    'book-set-postscript-value beg end (nth 1 tem)))
 		
 		  (and tem
 		       (list (concat "View "(nth 1 tem))
-			     'call-process  "ghostview" nil nil nil
+			     'call-process  "gv" nil nil nil
 			     (expand-file-name(nth 1 tem))
 			     ))
 		  (list "Cancel")
 		  )))
@@ -704,9 +704,9 @@
 			 ;; construct command to pass to the shell.
 			 (concat
 			  (cond ((looking-at "%PS") "")
 				(t "gzip -dc | "))
-			  "ghostview -")
+			  "gv -")
 			 
 	)))
 	 
 (defun book-set-postscript-value ( beg end &optional file)
diff -r -U4 maxima-5.10.0.orig/interfaces/xmaxima/Tkmaxima/Constants.tcl maxima-5.10.0/interfaces/xmaxima/Tkmaxima/Constants.tcl
--- maxima-5.10.0.orig/interfaces/xmaxima/Tkmaxima/Constants.tcl	2006-08-03 15:21:57.000000000 +0200
+++ maxima-5.10.0/interfaces/xmaxima/Tkmaxima/Constants.tcl	2006-09-21 20:20:32.000000000 +0200
@@ -142,9 +142,9 @@
     text/plain netmath
     image/gif  netmath
     image/png  netmath
     image/jpeg netmath
-    application/postscript "ghostview -safer %s"
+    application/postscript "gv -safer %s"
     application/pdf "acroread %s"
     application/x-dvi "xdvi %s"
 }
 set maxima_priv(imagecounter) 0
diff -r -U4 maxima-5.10.0.orig/interfaces/xmaxima/Tkmaxima/Menu.tcl maxima-5.10.0/interfaces/xmaxima/Tkmaxima/Menu.tcl
--- maxima-5.10.0.orig/interfaces/xmaxima/Tkmaxima/Menu.tcl	2006-09-11 17:38:01.000000000 +0200
+++ maxima-5.10.0/interfaces/xmaxima/Tkmaxima/Menu.tcl	2006-09-21 20:08:45.000000000 +0200
@@ -214,11 +214,11 @@
 	    lappend browse cmd.exe /c start
 	}
     } else {
 	
-	set selectedbrowser mozilla
+	set selectedbrowser firefox
 
-	foreach b { mozilla konqueror epiphany firefox netscape } {
+	foreach b { firefox konqueror epiphany mozilla opera netscape } {
 	    if { ! [catch {exec which $b} ] } {
 		set selectedbrowser $b
 		break } }
 
diff -r -U4 maxima-5.10.0.orig/src/Makefile.am maxima-5.10.0/src/Makefile.am
--- maxima-5.10.0.orig/src/Makefile.am	2006-09-05 01:52:42.000000000 +0200
+++ maxima-5.10.0/src/Makefile.am	2006-09-21 20:11:16.000000000 +0200
@@ -150,9 +150,9 @@
 endif
 
 
 if SBCL
-EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --eval
+EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --disable-debugger --eval
 all-local: binary-sbcl/maxima.core
 install-exec-local: install-sbcl
 uninstall: uninstall-sbcl
 clean: clean-sbcl
diff -r -U4 maxima-5.10.0.orig/src/Makefile.in maxima-5.10.0/src/Makefile.in
--- maxima-5.10.0.orig/src/Makefile.in	2006-09-21 06:00:57.000000000 +0200
+++ maxima-5.10.0/src/Makefile.in	2006-09-21 20:14:59.000000000 +0200
@@ -203,9 +203,9 @@
 @CLISP_TRUE@EXECUTECLISP = $(CLISP_NAME) -norc -q -x
 @CMUCL_TRUE@EXECUTECMUCL = $(CMUCL_NAME) -noinit -batch
 @SCL_TRUE@EXECUTESCL = $(SCL_NAME) -noinit -batch
 @ACL_TRUE@EXECUTEACL = $(ACL_NAME) -batch
-@SBCL_TRUE@EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --eval
+@SBCL_TRUE@EXECUTESBCL = $(SBCL_NAME) --noinform --noprint --disable-debugger --eval
 @GCL_TRUE@EXECUTEGCL = $(GCL_NAME) -batch -eval
 @OPENMCL_TRUE@EXECUTEOPENMCL = $(OPENMCL_NAME) -e
 genericdir = $(instsrcdir)
 real_lisp_sources = acall.lisp \
diff -r -U4 maxima-5.10.0.orig/src/plot.lisp maxima-5.10.0/src/plot.lisp
--- maxima-5.10.0.orig/src/plot.lisp	2006-08-26 13:13:40.000000000 +0200
+++ maxima-5.10.0/src/plot.lisp	2006-09-21 20:19:02.000000000 +0200
@@ -74,9 +74,9 @@
                         ((mlist) $logy nil)
                         ((mlist) $plot_realpart nil)
                         ))
 
-(defvar $viewps_command  "(ghostview \"~a\")")
+(defvar $viewps_command  "(gv \"~a\")")
 
 ;; $plot_realpart option is false by default but *plot-realpart* is true because coerce-float-fun
 ;; is used outside of plot package too.
 (defvar *plot-realpart* t)
