diff -Naur maxima-5.9.3/src/numerical/f2cl-lib.lisp maxima-5.9.3.new/src/numerical/f2cl-lib.lisp
--- maxima-5.9.3/src/numerical/f2cl-lib.lisp	2006-01-31 10:22:26.000000000 -0500
+++ maxima-5.9.3.new/src/numerical/f2cl-lib.lisp	2006-04-07 12:03:43.000000000 -0400
@@ -1413,7 +1413,7 @@
 ;;; These changes come from maxima.
 ;;;
 ;;; Revision 1.61  2005/03/28 20:38:18  rtoy
-;;; Make strings with an element-type of character instead of base-char,
+;;; Make strings with an element-type of character instead of character,
 ;;; in case the Lisp implementation has unicode support.
 ;;;
 ;;; Revision 1.60  2004/09/01 14:17:57  rtoy
diff -Naur maxima-5.9.3/src/server.lisp maxima-5.9.3.new/src/server.lisp
--- maxima-5.9.3/src/server.lisp	2005-11-07 12:37:12.000000000 -0500
+++ maxima-5.9.3.new/src/server.lisp	2006-04-07 12:03:43.000000000 -0400
@@ -52,7 +52,7 @@
 	      :element-type (if bin '(unsigned-byte 8) 'character)))
     #+gcl (si::socket port :host host)
     #+lispworks (comm:open-tcp-stream host port :direction :io :element-type
-                                      (if bin 'unsigned-byte 'base-char))
+                                      (if bin 'unsigned-byte 'character))
     #-(or allegro clisp cmu scl sbcl gcl lispworks)
     (error 'not-implemented :proc (list 'open-socket host port bin))))
 
