--- src/polybori-0.6/SConstruct.orig	2009-10-28 10:47:24.000000000 +0000
+++ src/polybori-0.6/SConstruct	2010-01-23 09:25:21.106548437 +0000
@@ -1232,6 +1232,11 @@
     env.AlwaysBuild(ipboribin)   
     env.Alias('install', ipboribin)
 
+    # we dump the flags for reuse by Sage
+    if not path.exists(InstPath()):
+        Execute(Mkdir(InstPath()))
+    opts.Save(InstPath("flags.conf"), env)
+
 env.Alias('prepare-devel', devellibs + readabledevellibs)
 env.Alias('prepare-install', [pyroot, DocPath()])
 
