--- a/stp/ocaml/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/stp/ocaml/Makefile	2013-03-14 20:22:49.993194747 -0600
@@ -14,3 +14,6 @@
 		Makefile
 
 include $(OCAMLMAKEFILE)
+
+install: rawinstall
+
--- a/stp/ocaml/OCamlMakefile	2009-09-24 15:45:57.000000000 -0600
+++ b/stp/ocaml/OCamlMakefile	2013-03-14 20:20:23.012069386 -0600
@@ -1126,10 +1126,10 @@
 .PHONY: rawinstall
 rawinstall:	all
 	$(QUIET)printf "\nInstalling library to: $(OCAML_LIB_INSTALL)\n"
-	-install -d $(OCAML_LIB_INSTALL)
+	-install -d $(DESTDIR)$(OCAML_LIB_INSTALL)
 	for i in $(LIBINSTALL_FILES); do \
 	  if [ -f $$i ]; then \
-	    install -c -m 0644 $$i $(OCAML_LIB_INSTALL); \
+	    install -c -m 0644 $$i $(DESTDIR)$(OCAML_LIB_INSTALL); \
 	  fi; \
 	done
 	$(QUIET)printf "\nInstallation successful.\n"

--- a/ocaml/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/ocaml/Makefile	2013-03-14 20:22:49.993194747 -0600
@@ -14,3 +14,6 @@
 
 
 include $(OCAMLMAKEFILE)
+
+install: rawinstall
+
--- a/ocaml/OCamlMakefile	2009-09-24 15:45:57.000000000 -0600
+++ b/ocaml/OCamlMakefile	2013-03-14 20:20:23.012069386 -0600
@@ -1126,10 +1126,10 @@
 .PHONY: rawinstall
 rawinstall:	all
 	$(QUIET)printf "\nInstalling library to: $(OCAML_LIB_INSTALL)\n"
-	-install -d $(OCAML_LIB_INSTALL)
+	-install -d $(DESTDIR)$(OCAML_LIB_INSTALL)
 	for i in $(LIBINSTALL_FILES); do \
 	  if [ -f $$i ]; then \
-	    install -c -m 0644 $$i $(OCAML_LIB_INSTALL); \
+	    install -c -m 0644 $$i $(DESTDIR)$(OCAML_LIB_INSTALL); \
 	  fi; \
 	done
 	$(QUIET)printf "\nInstallation successful.\n"
--- a/trace/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/trace/Makefile	2013-03-14 20:22:49.993194747 -0600
@@ -14,3 +14,6 @@
 
 
 include $(OCAMLMAKEFILE)
+
+install: rawinstall
+
--- a/trace_utils/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/trace_utils/Makefile	2013-03-16 14:31:46.375690732 -0600
@@ -88,6 +88,8 @@
 #    * subprjs
 
 
+install:
+	
 
 %:
 	make -f $(OCAMLMAKEFILE) subprojs SUBTARGET=$@
--- a/utils/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/utils/Makefile	2013-03-16 14:31:46.375690732 -0600
@@ -88,6 +88,8 @@
 #    * subprjs
 
 
+install:
+	
 
 %:
 	make -f $(OCAMLMAKEFILE) subprojs SUBTARGET=$@
--- a/doc/Makefile	2009-09-24 15:45:57.000000000 -0600
+++ b/doc/Makefile	2013-03-16 14:41:59.182926821 -0600
@@ -17,3 +17,6 @@
 
 clean:
 	(cd howto && $(MAKE) clean)
+
+install:
+	
