From: Mehdi Dogguy <mehdi@debian.org>
Date: Sun, 2 Oct 2011 11:53:22 +0200
Subject: [PATCH] ocamlopt is not really required

---
 configure |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure b/configure
index 5bd80d3..980aedd 100755
--- a/configure
+++ b/configure
@@ -186,7 +186,7 @@ checkcc()
 
 searchbinreq $ocaml
 searchbinreq $ocamlc
-searchbinreq $ocamlopt
+searchbin    $ocamlopt
 searchbinreq $ocamldep
 searchbinreq $ocamlmklib
 searchbinreq $ocamldoc
-- 
