--- project.mak.orig	2011-11-14 18:18:05.702705850 -0500
+++ project.mak	2011-11-14 18:18:58.142704311 -0500
@@ -23,12 +23,14 @@
 
 CMIOBJ = $(MLISRC:%.mli=%.cmi)
 TOINSTALL = zarith.a zarith.cma zarith.cmxa zarith.cmxs libzarith.a $(MLISRC) $(CMIOBJ)
-
+TOINSTALLbyte = zarith.cma libzarith.a $(MLISRC) $(CMIOBJ)
 
 # build targets
 ###############
 
-all: $(TOINSTALL) test
+all: $(TOINSTALL)
+
+all-byte: $(TOINSTALLbyte)
 
 zarith.cma: $(MLSRC:%.ml=%.cmo)
 	$(OCAMLMKLIB) -o zarith $+ $(LIBS)
@@ -80,8 +82,9 @@
 
 ifeq ($(INSTMETH),findlib)
 install:
-	$(OCAMLFIND) install -destdir $(INSTALLDIR) zarith META $(TOINSTALL) dllzarith.so
-
+	$(OCAMLFIND) install zarith META $(TOINSTALL) dllzarith.so
+install-byte:
+	$(OCAMLFIND) install zarith META $(TOINSTALLbyte) dllzarith.so
 uninstall:
 	$(OCAMLFIND) remove -destdir $(INSTALLDIR) zarith
 endif
