--- src/Makefile.am.orig	2009-01-22 09:41:46.000000000 +0000
+++ src/Makefile.am	2009-01-22 11:12:09.000000000 +0000
@@ -17,14 +17,16 @@
 lib_LTLIBRARIES     = libmatio.la
 libmatio_la_SOURCES = snprintf.c endian.c io.c $(ZLIB_SRC) read_data.c \
                       mat5.c mat4.c mat.c
+libmatio_la_LIBADD  = $(ZLIB_LIBS)
 
 if ENABLE_FORTRAN
     AM_FCFLAGS             = -I. -I$(top_builddir)/src/fortran
     AM_LDFLAGS            +=  $(FCLDFLAGS)
-    nodist_include_HEADERS =  $(top_builddir)/src/matio.mod
+    nodist_include_HEADERS = $(top_builddir)/src/matio.mod
     noinst_HEADERS        += fortran/create.f90 fortran/read_data.f90 \
                              fortran/write_data.f90 fortran/write.f90
     libmatio_la_SOURCES   += fortran/matio_internal.c fortran/matio.f90
+$(top_builddir)/src/matio.mod: libmatio.la
 endif
 
 dosubst = sed -e 's,[@]PACKAGE[@],$(PACKAGE),g' \
@@ -32,7 +34,7 @@
               -e 's,\/,\\,g'
 
 doc:
-	
+
 
 print:
 	@list='$(libmatio_a_SOURCES)'; \
