diff -r -U2 matio.orig/Makefile.am matio/Makefile.am
--- matio.orig/Makefile.am	2006-01-11 15:34:34.000000000 +0100
+++ matio/Makefile.am	2006-06-16 21:16:52.000000000 +0200
@@ -8,7 +8,5 @@
 ACLOCAL_AMFLAGS = -Iconfig
 
-
 EXTRA_DIST = matio.pc.in
-data_DATA  = share/test_file.mat share/test_file_v6.mat
 
 pkgconfigdir = $(libdir)/pkgconfig
@@ -21,5 +19,5 @@
 endif
 
-SUBDIRS = src test $(DOCS)
+SUBDIRS = src $(DOCS)
 
 dosubst = sed -e 's,[@]PACKAGE[@],$(PACKAGE),g' \
@@ -27,7 +25,4 @@
               -e 's,\/,\\,g'
 
-doc:
-	
-
 print:
 	subdirs='$(SUBDIRS)'; \
diff -r -U2 matio.orig/doxygen/Makefile.am matio/doxygen/Makefile.am
--- matio.orig/doxygen/Makefile.am	2006-01-11 15:34:29.000000000 +0100
+++ matio/doxygen/Makefile.am	2006-06-16 21:07:05.000000000 +0200
@@ -8,5 +8,5 @@
 VPATH = @srcdir@
 
-docdir = $(prefix)/doc
+docdir = $(prefix)/share/doc/matio-1.0
 doc_DATA = $(top_builddir)/doxygen/latex/libmatio.pdf
 
diff -r -U2 matio.orig/src/Makefile.am matio/src/Makefile.am
--- matio.orig/src/Makefile.am	2006-03-11 03:36:40.000000000 +0100
+++ matio/src/Makefile.am	2006-06-16 23:36:48.000000000 +0200
@@ -29,7 +29,4 @@
               -e 's,\/,\\,g'
 
-doc:
-	
-
 print:
 	@list='$(libmatio_a_SOURCES)'; \
diff -r -U2 matio.orig/test/Makefile.am matio/test/Makefile.am
--- matio.orig/test/Makefile.am	2006-02-09 20:48:24.000000000 +0100
+++ matio/test/Makefile.am	2006-06-16 21:36:44.000000000 +0200
@@ -7,5 +7,5 @@
 TEST_LFLAGS = $(all_libraries) -L$(top_builddir)/src
 
-bin_PROGRAMS = test_mat test_snprintf
+noinst_PROGRAMS = test_mat test_snprintf
 
 test_mat_SOURCES = test_mat.c
@@ -18,5 +18,5 @@
 
 if ENABLE_FORTRAN
-bin_PROGRAMS += test_matf
+noinst_PROGRAMS += test_matf
 
 test_matf_SOURCES = test_matf.f90
