diff -ur -x Makefile -x '*.o' -x '*.lo' -x '*.Plo' -x .libs -x '*.Tpo' -x '*.loT' -x '*.Po' -x '*.la' -x 'config.*' -x autom4te.cache -x libtool -x '*.in' coot-0.5-pre-1.orig/atom-utils/Makefile.am coot-0.5-pre-1/atom-utils/Makefile.am
--- coot-0.5-pre-1.orig/atom-utils/Makefile.am	2008-03-23 22:30:36.000000000 -0700
+++ coot-0.5-pre-1/atom-utils/Makefile.am	2008-03-23 23:09:42.000000000 -0700
@@ -9,6 +9,7 @@
 	-I$(top_srcdir)/coords 		\
 	-I$(top_srcdir)/geometry 	\
 	-I$(top_srcdir)/coot-utils	\
+	-I$(top_srcdir)/ccp4mg-utils	\
 	-I$(top_srcdir)/angles		\
 	-I$(top_srcdir)/build		\
 	-I$(top_srcdir)/mini-mol	\
diff -ur -x Makefile -x '*.o' -x '*.lo' -x '*.Plo' -x .libs -x '*.Tpo' -x '*.loT' -x '*.Po' -x '*.la' -x 'config.*' -x autom4te.cache -x libtool -x '*.in' coot-0.5-pre-1.orig/geometry/Makefile.am coot-0.5-pre-1/geometry/Makefile.am
--- coot-0.5-pre-1.orig/geometry/Makefile.am	2008-03-23 22:30:36.000000000 -0700
+++ coot-0.5-pre-1/geometry/Makefile.am	2008-03-23 23:21:29.000000000 -0700
@@ -47,3 +47,5 @@
 test_read_components_SOURCES = test-read-components.cc test-read-components.hh
 
 test_geometry_LDADD = ./libcoot-geometry.la
+
+test_read_components_LDADD = ./libcoot-geometry.la
diff -ur -x Makefile -x '*.o' -x '*.lo' -x '*.Plo' -x .libs -x '*.Tpo' -x '*.loT' -x '*.Po' -x '*.la' -x 'config.*' -x autom4te.cache -x libtool -x '*.in' coot-0.5-pre-1.orig/sequence-view/Makefile.am coot-0.5-pre-1/sequence-view/Makefile.am
--- coot-0.5-pre-1.orig/sequence-view/Makefile.am	2008-03-23 22:30:37.000000000 -0700
+++ coot-0.5-pre-1/sequence-view/Makefile.am	2008-03-23 23:27:15.000000000 -0700
@@ -35,6 +35,7 @@
 INCLUDES = \
 	-I $(top_srcdir)/coords		\
 	-I $(top_srcdir)/coot-utils	\
+	-I $(top_srcdir)/ccp4mg-utils	\
 	-I $(top_srcdir)/src		\
 	-I $(top_srcdir)/geometry	\
 	-I $(top_srcdir)/mini-mol	\
