--- ./x-windows/Makefile.in.orig	2008-08-01 14:21:05.000000000 -0700
+++ ./x-windows/Makefile.in	2008-08-01 14:21:37.000000000 -0700
@@ -35,7 +35,7 @@
 #
 # all
 #
-all : xdlview libjwc_c libjwc_f rotgen_ hklview xjiffy ipmosflm
+all : xdlview libjwc_c libjwc_f rotgen_ hklview xjiffy
 #
 # xdl_view
 #
@@ -161,7 +161,6 @@
 	$(INSTALL_PROGRAM) $(srcdir)/hklview $(bindir)
 	cd $(srcdir)/XCCPJIFFY ; $(MAKE) install
 	cd $(rotgen_dir); $(MAKE) install
-	$(INSTALL_PROGRAM) $(mosflm_dir)/bin/ipmosflm $(bindir)
 #
 # clean
 #
