--- ccp4-6.0.99e.orig/include/ccp4.setup-bash	2008-07-31 10:37:22.000000000 -0700
+++ ccp4-6.0.99e/include/ccp4.setup-bash	2008-07-31 12:47:14.000000000 -0700
@@ -181,7 +181,7 @@
 
 #Python path for the subprocess fixed module
 
-if ($ccp4_first_in_path) then
+if [ $ccp4_first_in_path -eq 1 ]; then
     setenv PYTHONPATH ${CCP4}/share/python:${PYTHONPATH}
 else
     setenv PYTHONPATH ${PYTHONPATH}:${CCP4}/share/python
