--- ccp4-6.0.99e.orig/lib/clipper/clipper/mmdbold/clipper_mmdb.cpp	2008-07-31 13:23:13.000000000 -0700
+++ ccp4-6.0.99e/lib/clipper/clipper/mmdbold/clipper_mmdb.cpp	2008-07-31 13:47:35.000000000 -0700
@@ -43,7 +43,7 @@
 
 #include <set>
 #include <algorithm>
-
+#include <string.h>
 
 namespace clipper {
 
--- ccp4-6.0.99e.orig/lib/clipper/clipper/mmdbold/clipper_mmdb_wrapper.cpp.orig	2008-07-31 13:49:53.000000000 -0700
+++ ccp4-6.0.99e/lib/clipper/clipper/mmdbold/clipper_mmdb_wrapper.cpp	2008-07-31 13:50:25.000000000 -0700
@@ -40,7 +40,7 @@
 //L  MA 02111-1307 USA
 
 #include "clipper_mmdb_wrapper.h"
-
+#include <string.h>
 
 namespace clipper {
 
