--- ccp4-5.99.5.orig/x-windows/Makefile.in	2005-12-05 01:30:59.000000000 -0800
+++ ccp4-5.99.5.orig/x-windows/Makefile.in	2005-12-05 01:31:50.000000000 -0800
@@ -42,8 +42,7 @@
 xdlview_dir = $(srcdir)/xdl_view/src
 xdlview :
 	cd $(xdlview_dir) ; \
-	$(MAKE) $(MFLAGS) ; \
-	$(MAKE) install 
+	$(MAKE) $(MFLAGS)
 
 #
 # rotgen and libraries
@@ -56,15 +55,13 @@
 #
 libjwc_c :
 	cd $(libjwc_c_dir) ; \
-	$(MAKE) $(MFLAGS) ; \
-	$(MAKE) install
+	$(MAKE) $(MFLAGS)
 #
 # libjwc_f
 #
 libjwc_f :
 	cd $(libjwc_f_dir) ; \
-	$(MAKE) $(MFLAGS) ; \
-	$(MAKE) install
+	$(MAKE) $(MFLAGS)
 #
 # rotgen
 #
@@ -167,6 +164,9 @@
 	$(INSTALL_PROGRAM) $(rasmol_dir)/rasmol $(bindir)
 	cd $(rotgen_dir); $(MAKE) install
 	$(INSTALL_PROGRAM) $(mosflm_dir)/bin/ipmosflm $(bindir)
+	cd $(xdlview_dir); $(MAKE) install
+	cd $(libjwc_c_dir); $(MAKE) install
+	cd $(libjwc_f_dir); $(MAKE) install
 #
 # clean
 #
