diff -urN coot-0.1.0-pre-1.orig/aclocal.m4 coot-0.1.0-pre-1/aclocal.m4
--- coot-0.1.0-pre-1.orig/aclocal.m4	2006-01-01 05:37:20.000000000 -0800
+++ coot-0.1.0-pre-1/aclocal.m4	2006-01-08 10:01:37.000000000 -0800
@@ -5826,15 +5826,15 @@
  # HACK! FIXME
  # added lz, we should have proper autoconf check for this.
  #
- CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lmccp4 $MMDB_LIBS -lrfftw -lfftw -lz -lm"
- CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lccp4c $MMDB_LIBS -lrfftw -lfftw -lz -lm"
+ CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lmccp4 $MMDB_LIBS -lsrfftw -lsfftw -lz -lm"
+ CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lccp4c $MMDB_LIBS -lsrfftw -lsfftw -lz -lm"
 # -L$clipper_prefix/boost/lib -lclipper-cctbx -L$clipper_prefix/cctbx/lib -lsgtbx -luctbx 
 else
  # the compiler looks in the "standard" places for clipper.  In real life,
  # it would be quite unlikely that clipper would be installed in /usr/include, 
  # /usr/lib etc. so this code will not usually find the right dependencies.
  CLIPPER_CXXFLAGS=""
- CLIPPER_LDOPTS="-lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lmccp4 $MMDB_LIBS -lrfftw -lfftw -lz -lm"
+ CLIPPER_LDOPTS="-lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lmccp4 $MMDB_LIBS -lsrfftw -lsfftw -lz -lm"
 fi
 
 AC_MSG_CHECKING([for Clipper])
