diff -urN coot-0.1.0-pre-1.orig/setup/Makefile.am coot-0.1.0-pre-1/setup/Makefile.am
--- coot-0.1.0-pre-1.orig/setup/Makefile.am	2005-12-30 11:48:18.000000000 -0800
+++ coot-0.1.0-pre-1/setup/Makefile.am	2006-01-08 10:00:49.000000000 -0800
@@ -1,8 +1,8 @@
 # Install in both bin and setup
-setupdir = $(prefix)/setup
+#setupdir = $(prefix)/setup
 
-# bin_SCRIPTS = coot.csh coot.sh
-setup_SCRIPTS = coot.csh coot.sh
+bin_SCRIPTS = coot.csh coot.sh
+# setup_SCRIPTS = coot.csh coot.sh
 
 
 
