--- ccp4-5.99.5.orig/src/Makefile.in	2006-01-05 22:52:56.000000000 -0800
+++ ccp4-5.99.5/src/Makefile.in	2006-01-05 22:55:58.000000000 -0800
@@ -57,7 +57,7 @@
 
 PDB_EXTRACT_TARGETS = pdb_extract extract pdb_extract_sf
 
-all : $(CTARGETS) $(CXXTARGETS) $(FTARGETS) $(OTHERS) $(PDB_EXTRACT_SUITE) ccp4mapwish phaser
+all : $(CTARGETS) $(CXXTARGETS) $(FTARGETS) $(OTHERS) ccp4mapwish phaser
 
 .PHONY : all install instsome clean realclean distclean
 
@@ -67,9 +67,6 @@
 	-for i in $(CTARGETS) $(CXXTARGETS) $(FTARGETS) $(OTHERS) $(JAVATARGETS); do \
 	  test -s $$i  &&  $(INSTALL_PROGRAM) `pwd`/$$i $(bindir)/$$i || true ;\
 	done
-	for i in $(PDB_EXTRACT_TARGETS); do \
-	  test -s harvest_app_/pdb_extract/bin/$$i  &&  $(INSTALL_PROGRAM) harvest_app_/pdb_extract/bin/$$i $(bindir)/$$i || true ;\
-	done
 	if test -f ccp4mapwish_/Makefile; then \
 	   cd ccp4mapwish_; $(MAKE) -i $(MFLAGS) $@; \
 	else true; fi
