--- ccp4-5.99.5.orig/lib/cctbx/cctbx_sources/libtbx/libtbx/config.py	2006-01-06 21:48:09.000000000 -0800
+++ ccp4-5.99.5/lib/cctbx/cctbx_sources/libtbx/libtbx/config.py	2006-01-06 21:46:24.000000000 -0800
@@ -756,8 +756,9 @@
     else:
       action = self.write_bin_sh_dispatcher
       ext = ""
-      try: os.chmod(source_file, 0755)
-      except OSError: pass
+      if source_file != self.python_exe:
+        try: os.chmod(source_file, 0755)
+        except OSError: pass
     target_file_ext = target_file + ext
     remove_or_rename(target_file_ext)
     try: action(source_file, target_file_ext)
