--- coot-utils/coot-coord-utils.cc.orig	2007-11-14 09:38:31.000000000 -0800
+++ coot-utils/coot-coord-utils.cc	2007-11-14 09:44:50.000000000 -0800
@@ -395,28 +395,30 @@
    return SelHnd;
 }
 
-std::ostream& coot::operator<< (std::ostream& s, const coot::atom_spec_t &spec) {
+namespace coot {
+   std::ostream& operator<< (std::ostream& s, const atom_spec_t &spec) {
 
-   s << "[spec: ";
-   s << "\"";
-   s << spec.chain;
-   s << "\" ";
-   s << spec.resno;
-   s << " ";
-   s << "\"";
-   s << spec.insertion_code;
-   s << "\"";
-   s << " ";
-   s << "\"";
-   s  << spec.atom_name;
-   s << "\"";
-   s << " ";
-   s << "\"";
-   s << spec.alt_conf;
-   s << "\"]";
+      s << "[spec: ";
+      s << "\"";
+      s << spec.chain;
+      s << "\" ";
+      s << spec.resno;
+      s << " ";
+      s << "\"";
+      s << spec.insertion_code;
+      s << "\"";
+      s << " ";
+      s << "\"";
+      s  << spec.atom_name;
+      s << "\"";
+      s << " ";
+      s << "\"";
+      s << spec.alt_conf;
+      s << "\"]";
 
-   return s;
+      return s;
 
+   }
 }
 
 
