--- ccp4-5.99.5.orig/src/phaser/conf/configure-options	2006-01-06 11:27:28.000000000 -0800
+++ ccp4-5.99.5/src/phaser/conf/configure-options	2006-01-06 11:26:29.000000000 -0800
@@ -33,9 +33,9 @@
   set font_dirs = ''
   set strip_command = 'strip'
   set wxpython_env = ''
-  if ( "$PHASER_MTYPE" == "intel-linux" ) then
-    setenv LD_ASSUME_KERNEL 2.4.1
-  endif
+#  if ( "$PHASER_MTYPE" == "intel-linux" ) then
+#    setenv LD_ASSUME_KERNEL 2.4.1
+#  endif
 else if ( "$PHASER_MTYPE" == "alpha-tru64" ) then
   set python_configure = '--with-cxx=cxx'
   set libpng_arch = 'dec'
