diff -aur src3.2.6.old/SMESH_SRC_3.2.6/src/SMESHGUI/SMESHGUI_AddQuadraticElementDlg.cxx src3.2.6/SMESH_SRC_3.2.6/src/SMESHGUI/SMESHGUI_AddQuadraticElementDlg.cxx
--- src3.2.6.old/SMESH_SRC_3.2.6/src/SMESHGUI/SMESHGUI_AddQuadraticElementDlg.cxx	2008-07-25 09:16:09.000000000 +0200
+++ src3.2.6/SMESH_SRC_3.2.6/src/SMESHGUI/SMESHGUI_AddQuadraticElementDlg.cxx	2008-10-16 01:33:46.000000000 +0200
@@ -86,7 +86,7 @@
 
 namespace SMESH {
 
-  void ReverseConnectivity( vector<int> & ids, int type )
+  void ReverseConnectivity( vector<vtkIdType> & ids, int type )
   {
     // for reverse connectivity of other types keeping the first id, see
     // void SMESH_VisualObjDef::buildElemPrs() in SMESH_Object.cxx:900
@@ -134,7 +134,7 @@
       reverse( ids.begin(), ids.end() );
     }
     else {
-      vector<int> aRevIds( ids.size() );
+      vector<vtkIdType> aRevIds( ids.size() );
       for ( int i = 0; i < ids.size(); i++)
         aRevIds[ i ] = ids[ conn[ i ]];
       ids = aRevIds;
@@ -597,7 +597,7 @@
   if (IsValid() && !mySMESHGUI->isActiveStudyLocked()) {
     myBusy = true;
     
-    vector<int> anIds;
+    vector<vtkIdType> anIds;
 
     switch (myType) {
     case QUAD_EDGE:
