diff -aur src3.2.6.old/SMESH_SRC_3.2.6/src/DriverUNV/UNV_Utilities.hxx src3.2.6/SMESH_SRC_3.2.6/src/DriverUNV/UNV_Utilities.hxx
--- src3.2.6.old/SMESH_SRC_3.2.6/src/DriverUNV/UNV_Utilities.hxx	2008-05-25 19:32:52.000000000 +0200
+++ src3.2.6/SMESH_SRC_3.2.6/src/DriverUNV/UNV_Utilities.hxx	2008-05-26 03:23:15.000000000 +0200
@@ -26,6 +26,7 @@
 #include <string>
 #include <stdexcept>
 #include <cassert>
+#include <cstdlib>
 
 namespace UNV{
   using namespace std;
diff -aur src3.2.6.old/SMESH_SRC_3.2.6/src/MEFISTO2/aptrte.h src3.2.6/SMESH_SRC_3.2.6/src/MEFISTO2/aptrte.h
--- src3.2.6.old/SMESH_SRC_3.2.6/src/MEFISTO2/aptrte.h	2008-05-25 19:32:52.000000000 +0200
+++ src3.2.6/SMESH_SRC_3.2.6/src/MEFISTO2/aptrte.h	2008-05-26 03:38:58.000000000 +0200
@@ -31,8 +31,8 @@
 #include <limits.h>   // limites min max int long real ...
 #include <unistd.h>   // gethostname, ...
 #include <stdio.h>
-#include <iostream.h> // pour cout cin ...
-#include <iomanip.h>  // pour le format des io setw, stx, setfill, ...
+#include <iostream> // pour cout cin ...
+#include <iomanip>  // pour le format des io setw, stx, setfill, ...
 #include <string.h>   // pour les fonctions sur les chaines de caracteres
 #include <ctype.h>
 #include <stdlib.h>
diff -aur src3.2.6.old/SMESH_SRC_3.2.6/src/StdMeshers/StdMeshers_Prism_3D.hxx src3.2.6/SMESH_SRC_3.2.6/src/StdMeshers/StdMeshers_Prism_3D.hxx
--- src3.2.6.old/SMESH_SRC_3.2.6/src/StdMeshers/StdMeshers_Prism_3D.hxx	2008-05-25 19:32:52.000000000 +0200
+++ src3.2.6/SMESH_SRC_3.2.6/src/StdMeshers/StdMeshers_Prism_3D.hxx	2008-05-26 03:47:51.000000000 +0200
@@ -37,8 +37,8 @@
 #include "SMESH_MesherHelper.hxx"
 #include "SMESH_Comment.hxx"
 
-#include <vector.h>
-#include <map.h>
+#include <vector>
+#include <map>
 
 #include <Adaptor3d_Curve.hxx>
 #include <Adaptor3d_Surface.hxx>
