diff -Naur ATLAS/include/contrib/camm_dpa.h ATLAS-new/include/contrib/camm_dpa.h
--- ATLAS/include/contrib/camm_dpa.h	2006-09-09 15:33:15.000000000 -0400
+++ ATLAS-new/include/contrib/camm_dpa.h	2006-09-26 10:18:24.000000000 -0400
@@ -1619,7 +1619,7 @@
 #if defined(SCPLX) || (defined(DCPLX) && defined(ATL_SSE2))
 	 ,"m" (w)
 #endif
-	 :"ax","bx","cx","dx","si","di");
+	 :"ax","cx","dx","si","di");
 
 
 }
