diff -urN kseg-0.402.orig/Makefile kseg-0.402/Makefile
--- kseg-0.402.orig/Makefile	2004-11-24 18:07:27.000000000 +0100
+++ kseg-0.402/Makefile	2005-08-04 15:33:03.000000000 +0200
@@ -1,7 +1,7 @@
 # Makefile for KSEG
 CC = g++
 QTINCLUDE = -I$(QTDIR)/include
-CCFLAGS = -c -Wall -O2 -fomit-frame-pointer -ffast-math
+CCFLAGS = -c $(CXXFLAGS)
 LIBS = -lm -L$(QTDIR)/lib -lqt -lz
 MOC = $(QTDIR)/bin/moc
 LUPDATE = $(QTDIR)/bin/lupdate
diff -urN kseg-0.402.orig/formula/Makefile kseg-0.402/formula/Makefile
--- kseg-0.402.orig/formula/Makefile	2004-11-24 18:07:28.000000000 +0100
+++ kseg-0.402/formula/Makefile	2005-08-04 15:33:16.000000000 +0200
@@ -1,7 +1,7 @@
 # Makefile for formula
 CC = g++
 QTINCLUDE = -I$(QTDIR)/include
-CCFLAGS = -c -Wall -O2 -fomit-frame-pointer -ffast-math
+CCFLAGS = -c $(CXXFLAGS)
 LIBS = 
 MOC = $(QTDIR)/bin/moc
 
diff -urN kseg-0.402.orig/main.cpp kseg-0.402/main.cpp
--- kseg-0.402.orig/main.cpp	2004-11-24 18:07:27.000000000 +0100
+++ kseg-0.402/main.cpp	2005-08-04 15:32:48.000000000 +0200
@@ -37,7 +37,7 @@
 
   //set some default properties
   KSegProperties::setPropertyDefault("Language", "");
-  KSegProperties::setPropertyDefault("LanguageDir", ".");
+  KSegProperties::setPropertyDefault("LanguageDir", "KSEG_HELP_DIR");
   KSegProperties::setPropertyDefault("SelectType", "BorderSelect");
   KSegProperties::setPropertyDefault("QuickPlayDirectory", ".");
   KSegProperties::setPropertyDefault("RecentListSize", "4");
