--- Makefile.PL.orig	2009-02-09 15:32:59.000000000 +0200
+++ Makefile.PL	2009-02-09 15:49:44.000000000 +0200
@@ -7,16 +7,15 @@
 @ENV{"LD_LIBRARY_PATH"});
 
 $CC = 'g++';
-$srcdir = '../../src';
 
 # check if we're compiling in the source directory
 # check first for the static, then change to dynamic if available
-$ldfrom = "\$(OBJECT) $srcdir/.libs/libopenbabel.a -lz"
-    if (-r "$srcdir/.libs/libopenbabel.a") and (-s _) and (-B _);
-$ldfrom = "\$(OBJECT) -L$srcdir/.libs -lopenbabel -lz"
-    if (-r "$srcdir/.libs/libopenbabel.dylib") and (-s _) and (-B _);
-$ldfrom = "\$(OBJECT) -L$srcdir/.libs -lopenbabel -lz"
-    if (-r "$srcdir/.libs/libopenbabel.so") and (-s _) and (-B _);
+$ldfrom = "\$(OBJECT) /usr/lib/libopenbabel.a -lz"
+    if (-r "/usr/lib/libopenbabel.a") and (-s _) and (-B _);
+$ldfrom = "\$(OBJECT) -L/usr/lib -lopenbabel -lz"
+    if (-r "/usr/lib/libopenbabel.dylib") and (-s _) and (-B _);
+$ldfrom = "\$(OBJECT) -L/usr/lib -lopenbabel -lz"
+    if (-r "/usr/lib/libopenbabel.so") and (-s _) and (-B _);
 
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
