--- setup.py.orig	2009-02-23 13:59:28.000000000 +0200
+++ setup.py	2009-02-28 15:31:30.000000000 +0200
@@ -58,7 +58,7 @@
   platforms = ["Unix", "Windows", "hopefully other OSes able to run Python"],
   long_description = "BKchem is a chemical drawing program written in Python",
   
-  packages=[ 'bkchem', 'bkchem/plugins', 'bkchem/oasa', 'bkchem/oasa/oasa', 'bkchem/oasa/oasa/graph', 'bkchem/plugins/piddle'],
+  packages=[ 'bkchem', 'bkchem/plugins', 'bkchem/plugins/piddle'],
 
   data_files=[ ('share/bkchem/templates', glob.glob( 'templates/*.cdml')+glob.glob('templates/*.xml')),
                ('share/bkchem/images', ['images/logo.ppm','images/icon.ico']+glob.glob('images/bkchem*.png')),
