diff --git a/Src/superlu3module.c b/Src/superlu3module.c
index 810191b..a74edfb 100644
--- a/Src/superlu3module.c
+++ b/Src/superlu3module.c
@@ -267,7 +267,7 @@ static PyObject *newSuperLUObject(int n, CSRMatObject *matrix, PyObject *issym,
   //printf("Factorizing with drop_tol = %g\n", drop_tol);
   /* Perform factorization (perm_c and perm_r are swapped because our matrix
    * is stored in compressed-row format and not in compressed-column format! */
-  dgstrf(&self->options, &AC, drop_tol, relax, panel_size,
+  dgstrf(&self->options, &AC, relax, panel_size,
          etree, NULL, 0, self->perm_c, self->perm_r,
          &self->L, &self->U, &self->stat, &info);
 
