--- setup.py.orig	2010-05-11 19:32:35.000000000 +0100
+++ setup.py	2010-05-11 19:32:49.000000000 +0100
@@ -8,7 +8,7 @@
 
 # default settings
 library_dirs_list= []
-libraries_list = ['lapack', 'blas', 'g2c']
+libraries_list = ['lapack', 'blas']
 superlu_defs = [('USE_VENDOR_BLAS',1)]
 if sys.platform == 'win32':
     superlu_defs += [('NO_TIMER', 1)]
@@ -18,40 +18,40 @@
 compily=[]
 
 # Specify whether to link against user's SuperLU library
-use_users_superlu = False #True
+use_users_superlu = True
 umfpack_defs = [('DINT', 1), ('NBLAS', 1)] # No BLAS. Ok if using your system's BLAS.
 #umfpack_defs = [('DINT', 1), ('CBLAS', 1)] # with atlas c-blas (http://math-atlas.sf.net)
 
 if use_users_superlu:
     # Specify location of include files
-    superlu_include_dirs = ['/Users/dpo/local/linalg/SuperLU/SuperLU_3.1/SRC']
+    superlu_include_dirs = ['/usr/include/superlu']
     # AND specify ONE of the following:
     # 1) Location of source files (overrides linking with existing library)
-    superlu_src_dir = '/Users/dpo/local/linalg/SuperLU/SuperLU_3.1/SRC'
+    superlu_src_dir = ''
     # 2) Location of precompiled library
     superlu_lib_dir = ['']
-    superlu_libraries = ['']
+    superlu_libraries = ['superlu']
 
 # Specify whether to link against user's UMFPACK library
-use_users_umfpack = False #True
+use_users_umfpack = True
 if use_users_umfpack:
     # Specify location of include files
-    umfpack_include_dirs = ['/Users/dpo/local/linalg/UMFPACK/UMFPACK/Include']
+    umfpack_include_dirs = ['']
     # AND specify ONE of the following:
     # 1) Location of source files (overrides linking with existing library)
-    umfpack_src_dir = '/Users/dpo/local/linalg/UMFPACK/UMFPACK/Source'
+    umfpack_src_dir = ''
     # 2) Location of precompiled library
     umfpack_lib_dir = ['']
-    umfpack_libraries = ['']
+    umfpack_libraries = ['umfpack']
 
     # Do the same for AMD
-    amd_include_dirs = ['/Users/dpo/local/linalg/UMFPACK/AMD/Include']
+    amd_include_dirs = ['']
     # AND specify ONE of the following:
     # 1) Location of source files (overrides linking with existing library)
-    amd_src_dir = '/Users/dpo/local/linalg/UMFPACK/AMD/Source'
+    amd_src_dir = ''
     # 2) Location of precompiled library
     amd_lib_dir = ['']
-    amd_libraries = ['']
+    amd_libraries = ['amd']
     
 else:
     umfpack_libraries = []
