--- Src/superlu3module.c.orig	2010-05-11 19:22:02.000000000 +0100
+++ Src/superlu3module.c	2010-05-11 19:17:02.000000000 +0100
@@ -267,7 +267,7 @@
   //printf("Factorizing with drop_tol = %g\n", drop_tol);
   /* Perform factorization (perm_c and perm_r are swapped because our matrix
    * is stored in compressed-row format and not in compressed-column format! */
-  dgstrf(&self->options, &AC, drop_tol, relax, panel_size,
+  dgstrf(&self->options, &AC, relax, panel_size,
          etree, NULL, 0, self->perm_c, self->perm_r,
          &self->L, &self->U, &self->stat, &info);
 
