diff -ur old/INCHI-1-API/INCHI/gcc/inchi-1/makefile new/INCHI-1-API/INCHI/gcc/inchi-1/makefile
--- old/INCHI-1-API/INCHI/gcc/inchi-1/makefile	2010-03-01 12:42:14.000000000 +0200
+++ new/INCHI-1-API/INCHI/gcc/inchi-1/makefile	2010-11-13 17:45:45.000000000 +0200
@@ -13,7 +13,7 @@
 endif
 
 ifndef LINKER
-  LINKER = g++ -s
+  LINKER = g++
 endif
 
 ifndef INCHI_EXECUTABLE
@@ -37,7 +37,7 @@
 
 P_INCL = -I$(P_MAIN) -I$(P_LIBR) -I$(P_CMLP)
 
-C_COMPILER_OPTIONS = $(P_INCL) -DADD_CMLPP=1 -ansi -O3 -c
+C_COMPILER_OPTIONS = $(P_INCL) -DADD_CMLPP=1 -ansi ${CFLAGS} -c
 #C_COMPILER_OPTIONS = -c $(P_INCL) -DADD_CMLPP=1 -ansi -O3 -fsigned-char -ffunction-sections -fexpensive-optimizations -fstack-check -fexceptions -Wall -pedantic -Wbad-function-cast -Wreturn-type -Wformat -Wuninitialized -Wcast-align -Wshadow -Wunused -Wunused-value -Wunused-variable -Wunused-function -Wunused-parameter -Wunused-label -Wcomment -Wcast-qual -Wconversion -Wimplicit-int -Wmissing-braces -Wmissing-declarations -Wmissing-prototypes -Wredundant-decls -Wsign-compare -Wfloat-equal -Wstrict-prototypes -Wwrite-strings -Wundef -Waggregate-return -Wchar-subscripts -Wformat-nonliteral -Wnested-externs -Wsequence-point -Wpointer-arith -mfancy-math-387 -mieee-fp -mno-soft-float
 
 ifdef windir
@@ -45,12 +45,12 @@
   CPP_COMPILER_OPTIONS = $(P_INCL) -D_LIB -DADD_CMLPP=1 -O3 -frtti -c
   #CPP_COMPILER_OPTIONS = -c $(P_INCL) -D_LIB -DADD_CMLPP=1 -O3 -frtti -Wall -pedantic -Wreturn-type -Wformat -Wuninitialized -ffunction-sections -fexpensive-optimizations -fstack-check -fexceptions -Wcast-align -Wshadow -Wunused -Wunused-value -Wunused-variable -Wunused-function -fsigned-char -Wcast-qual -Wconversion -Wmissing-braces -Wredundant-decls -Wsign-compare -Wfloat-equal -Wwrite-strings -mfancy-math-387 -mieee-fp -mno-soft-float
 else
-  CPP_COMPILER_OPTIONS = $(P_INCL) -D_LIB -DADD_CMLPP=1 -ansi -O3 -frtti -c
+  CPP_COMPILER_OPTIONS = $(P_INCL) -D_LIB -DADD_CMLPP=1 -ansi ${CXXFLAGS} -c
   #CPP_COMPILER_OPTIONS = -c $(P_INCL) -D_LIB -DADD_CMLPP=1 -ansi -O3 -frtti -Wall -pedantic -Wreturn-type -Wformat -Wuninitialized -ffunction-sections -fexpensive-optimizations -fstack-check -fexceptions -Wcast-align -Wshadow -Wunused -Wunused-value -Wunused-variable -Wunused-function -fsigned-char -Wcast-qual -Wconversion -Wmissing-braces -Wredundant-decls -Wsign-compare -Wfloat-equal -Wwrite-strings -mfancy-math-387 -mieee-fp -mno-soft-float
 endif
 
 
-LINKER_OPTIONS =
+LINKER_OPTIONS = ${LDFLAGS}
 #LINKER_OPTIONS = -Wall -Wunused -Wunused-function
 
 
diff -ur old/INCHI-1-API/INCHI_API/gcc_so_makefile/makefile new/INCHI-1-API/INCHI_API/gcc_so_makefile/makefile
--- old/INCHI-1-API/INCHI_API/gcc_so_makefile/makefile	2010-02-02 02:24:54.000000000 +0200
+++ new/INCHI-1-API/INCHI_API/gcc_so_makefile/makefile	2010-11-13 17:47:34.000000000 +0200
@@ -55,7 +55,7 @@
 
 # === Linker to create (Shared) InChI library ====
 ifndef SHARED_LINK
-  SHARED_LINK = gcc -shared
+  SHARED_LINK = gcc -shared ${LDFLAGS}
 endif
 
 # === Linker to create Main program =====
@@ -65,7 +65,7 @@
      LINKER_CWD_PATH = -Wl,-R,""
   endif
   endif
-  LINKER = gcc -s $(LINKER_CWD_PATH)
+  LINKER = gcc $(LINKER_CWD_PATH)
 endif
 
 ifndef P_LIBR
@@ -84,7 +84,7 @@
 
 # === C Compiler Options =======
 ifndef C_OPTIONS
-  C_OPTIONS = -ansi -O3 -c
+  C_OPTIONS = -ansi ${CFLAGS} -c
   ifndef windir
   ifdef ISLINUX
     ifndef C_SO_OPTIONS
