--- a/lib/fold.c
+++ b/lib/fold.c
@@ -64,9 +64,9 @@ PRIVATE void backtrack(const char *sequence, int s);
 PRIVATE int fill_arrays(const char *sequence);
 /*@unused@*/
 INLINE PRIVATE  int oldLoopEnergy(int i, int j, int p, int q, int type, int type_2);
-INLINE int  LoopEnergy(int n1, int n2, int type, int type_2,
+int  LoopEnergy(int n1, int n2, int type, int type_2,
 			 int si1, int sj1, int sp1, int sq1);
-INLINE int  HairpinE(int size, int type, int si1, int sj1, const char *string);
+int  HairpinE(int size, int type, int si1, int sj1, const char *string);
 
 #define MAXSECTORS      500     /* dimension for a backtrack array */
 #define LOCALITY        0.      /* locality parameter for base-pairs */
