--- gui/Makefile.in.orig	2010-08-20 22:59:47.208483158 +1200
+++ gui/Makefile.in	2010-08-20 23:00:51.888482426 +1200
@@ -19,7 +19,7 @@
 main.o: $(OBJS) Makefile
 
 xcadabra: $(OBJS)
-	@CXX@ -o xcadabra $+ `pkg-config modglue --libs` `pkg-config --libs gtkmm-2.4` -lpcrecpp
+	@CXX@ @CFLAGS@ @LDFLAGS@ -o xcadabra $+ `pkg-config modglue --libs` `pkg-config --libs gtkmm-2.4` -lpcrecpp
 
 xcadabra_static: $(OBJS)
 	@CXX@ -o xcadabra -static $+  -L@prefix@/lib `pkg-config modglue --libs` \
