--- src/Makefile.in.orig	2010-03-28 23:01:45.367865648 +1300
+++ src/Makefile.in	2010-03-28 23:02:11.518865659 +1300
@@ -146,9 +146,9 @@
 # Installation and cleanup.
 
 install:
-ifeq ($(strip $(MACTEST)),)
-	strip cadabra
-endif
+#ifeq ($(strip $(MACTEST)),)
+#	strip cadabra
+#endif
 #	strip -S cadabra
 #endif
 	@INSTALL@ -m 0755 -d ${DESTDIR}@prefix@/bin
--- gui/Makefile.in.orig	2010-03-28 23:01:34.485865848 +1300
+++ gui/Makefile.in	2010-03-28 23:02:30.490865700 +1300
@@ -30,9 +30,9 @@
 	@CXX@ -o test_texit `pkg-config modglue --libs` `pkg-config --libs gtkmm-2.4` $+ 
 
 install:
-ifeq ($(strip $(MACTEST)),)
-	strip xcadabra
-endif
+#ifeq ($(strip $(MACTEST)),)
+#	strip xcadabra
+#endif
 	install -d ${DESTDIR}@prefix@/bin
 	install -m 0755 xcadabra ${DESTDIR}@prefix@/bin
 	install -d ${DESTDIR}@prefix@/share/texmf/tex/latex/cadabra
