--- src/Makefile.in	2009-03-10 23:59:12.000000000 +1300
+++ src/Makefile.in.old	2009-04-04 22:36:27.000000000 +1300
@@ -37,7 +37,7 @@
 modules: $(MOBJS)
 
 cadabra: $(OBJS) $(MOBJS)
-	@CXX@ -o cadabra ${LDFLAGS} `pkg-config modglue --libs` $+ -lgmpxx -lpcrecpp -lgmp
+	@CXX@ -o cadabra $+ ${LDFLAGS} `pkg-config modglue --libs` -lgmpxx -lpcrecpp -lgmp
 
 #`pkg-config glib-2.0 --libs` 
 # Static linking now works!
