diff -aur OpenCASCADE6.3.0.old/ros/src/WOKUnix/WOKUnix_FDescr.cxx OpenCASCADE6.3.0/ros/src/WOKUnix/WOKUnix_FDescr.cxx
--- OpenCASCADE6.3.0.old/ros/src/WOKUnix/WOKUnix_FDescr.cxx	2009-02-15 14:00:22.000000000 +0100
+++ OpenCASCADE6.3.0/ros/src/WOKUnix/WOKUnix_FDescr.cxx	2009-02-15 14:08:50.000000000 +0100
@@ -202,7 +202,7 @@
   if(mknod(apath.ToCString(), 0700 |  S_IFIFO, 0)) 
     { perror(apath.ToCString());}
 
-  myFileChannel = open(apath.ToCString(),  O_RDONLY | O_NDELAY | O_CREAT);
+  myFileChannel = open(apath.ToCString(),  O_RDONLY | O_NDELAY );
   SetUnBuffered();
 
   // write end of pipe is unbuffered also 
