 cctbx_sources/libtbx/SConscript | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/cctbx_sources/libtbx/SConscript b/cctbx_sources/libtbx/SConscript
index c273536..a70dae8 100644
--- a/cctbx_sources/libtbx/SConscript
+++ b/cctbx_sources/libtbx/SConscript
@@ -120,7 +120,7 @@ def patch_scons_env_for_ad_hoc_debug(env, O123_replacement=["-O0", "-g"]):
     for f in env[kw]:
       if   (f in ["-O1", "-O2", "-O3"]):
         dbg_ccflags.extend(O123_replacement)
-      elif (f not in ["-ffast-math", "-funroll-loops"]):
+      elif (f not in []):
         dbg_ccflags.append(f)
     env.Replace(**{kw: dbg_ccflags})
 env_etc.patch_scons_env_for_ad_hoc_debug = patch_scons_env_for_ad_hoc_debug
@@ -453,9 +453,9 @@ libtbx.scons: Warning: compiling with /MDd:
       warn_options = gcc_common_warn_options() + ["-Werror"]
     env_etc.ccflags_base.extend(warn_options)
     if (libtbx.env.build_options.optimization):
-      opts = ["-DNDEBUG", "-O3", "-ffast-math"]
+      opts = ["-DNDEBUG"]
       if (env_etc.gcc_version >= 40100 and is_64bit_architecture):
-        opts.insert(2, "-funroll-loops")
+        opts.insert(2, )
     else:
       opts = ["-O0", "-fno-inline"]
     if (libtbx.env.build_options.mode == "profile"):
@@ -565,9 +565,7 @@ else:
       warn_options = gcc_common_warn_options() + ["-Werror"]
     env_etc.ccflags_base.extend(warn_options)
     if (libtbx.env.build_options.optimization):
-      opts = ["-DNDEBUG", "-O3", "-ffast-math"]
-      if (env_etc.gcc_version >= 40100 and is_64bit_architecture):
-        opts.insert(2, "-funroll-loops")
+      opts = ["-DNDEBUG"]
     else:
       opts = ["-O0", "-fno-inline"]
     if (libtbx.env.build_options.mode == "profile"):
@@ -633,10 +631,7 @@ int main() {
     env_etc.cxxflags_base = []
     setup_clang_warning(env_etc, libtbx.env.build_options.warning_level)
     if (libtbx.env.build_options.optimization):
-      opts = [
-        "-DNDEBUG",
-        "-O3",
-        "-ffast-math"]
+      opts = ["-DNDEBUG"]
     else:
       opts = ["-O0", "-fno-inline"]
     if (libtbx.env.build_options.debug_symbols):
@@ -782,7 +777,7 @@ int main() {
       env_etc.cxxflags_base.append("-fcoalesce-templates")
     env_etc.cxxflags_base.append("-DBOOST_ALL_NO_LIB")
     if (libtbx.env.build_options.optimization):
-      opts = ["-DNDEBUG", "-O3", "-ffast-math"]
+      opts = ["-DNDEBUG"]
       if (cxx == "c++" and env_etc.gcc_version
             and env_etc.gcc_version == 40000):
         opts[1] = "-O2" # Apple's optimizer is broken
