diff --git a/Makefile b/Makefile
index c5f6fcf..25e328f 100644
--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,9 @@ C_OBJS = *.o
 #INCLUDES = -IC:/Tcl/include -I../CBLAS/src
 #LIBRARIES = -lm *.dll
 # LINUX:
-INCLUDES = -I/usr/include/tcl8.5
-LIBRARIES = -lm -lgslcblas -llapack -ltcl8.5
+INCLUDES = -I@GENTOO_PORTAGE_EPREFIX@/usr/include/
+BLASLIBS = `$(PKG_CONFIG) --libs cblas lapack`
+LIBRARIES = -lm ${BLASLIBS} -ltcl
 
 FLAGS = -c -O3
 CC = gcc
@@ -16,9 +17,9 @@ RM = rm
 TAR = tar
 
 simpson: $(C_OBJS)
-	$(CC) $(LIBRARIES) $(C_OBJS) -o simpson 
+	$(CC) $(LDFLAGS) -o simpson $(C_OBJS) $(LIBRARIES)
 $(C_OBJS):
-	$(CC) $(FLAGS) $(INCLUDES) *.c
+	$(CC) $(CFLAGS) $(INCLUDES) -c *.c
 clean:
 	$(RM) -f *.o simpson
 dist:
