 Makefile |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 7206ba5..832e76a 100644
--- a/Makefile
+++ b/Makefile
@@ -11,9 +11,9 @@ BINDIR = /usr/bin
 #LIBRARIES = -lm *.dll
 
 # Linux
-#INCLUDES = -I/usr/include/tcl8.4
-#LIBRARIES = -lm -llapack -lblas -ltcl8.4
-#BINDIR = /usr/local/bin
+INCLUDES = -I@GENTOO_PORTAGE_EPREFIX@/usr/include/
+BLASLIBS = `$(PKG_CONFIG) --libs cblas lapack`
+LIBRARIES = -lm ${BLASLIBS} -ltcl
 
 # Linux MPI 
 #INCLUDES = -I/usr/include/tcl8.4 -I/com/intel/mkl/10.1.1.019/include/ 
@@ -38,9 +38,9 @@ TAR = tar
 MKDIR = mkdir
 
 simpson: $(OBJ)
-	$(CC) $(LIBRARIES) $(OBJ) -o simpson
+	$(CC) $(LDFLAGS) -o $@ $(OBJ) $(LIBRARIES)
 .c.o:
-	$(CC) $(FLAGS) $(EXTRA_FLAGS) $(INCLUDES) -DVERSION=\"$(VERSION)\" $<
+	$(CC) $(CFLAGS) $(EXTRA_FLAGS) $(INCLUDES) -DVERSION=\"$(VERSION)\" -c $<
 clean:
 	$(RM) -f *.o simpson
 dist:
