--- configure.in.orig	2006-06-07 22:14:12.000000000 +0200
+++ configure.in	2006-06-07 22:41:41.000000000 +0200
@@ -151,2 +151,2 @@
-AC_PATH_PROG(HTML, lynx)
-AC_PATH_PROGS(XHTML, mozilla netscape mosaic)
+AC_PATH_PROG(HTML, links elinks lynx)
+AC_PATH_PROGS(XHTML, firefox mozilla konqueror epiphany opera seamonkey galeon skipstone kazehakase netscape)
@@ -154 +154 @@
-AC_PATH_PROG(XTERM, xterm)
+AC_PATH_PROG(XTERM, xterm konsole gnome-terminal rxvt eterm aterm)
