 Makefile.in | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 8887b2e..61d185d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -112,7 +112,7 @@ SHELL = /bin/sh
 
 # Subdirectories to make recursively.
 BINSUBDIRS = doc etc examples src test timing tools
-SUBDIRS = blas lapack superlu arpack fpe $(BINSUBDIRS)
+SUBDIRS = fpe $(BINSUBDIRS)
 
 # Files in the source distribution from this directory.
 DISTFILES = COPYING INSTALL LICENSE Makefile.in NEWS PROBLEMS README \
@@ -149,22 +149,12 @@ configure: configure.in
 
 # Compile the source code.
 
-source: @BLAS@ @LAPACK@
-	cd superlu; $(MAKE) RC0=$(RC0)
-	cd arpack; $(MAKE)
-	cd src; $(MAKE) RC0=$(RC0)
+source: 
+	$(MAKE) -C src RC0=$(RC0)
 .PHONY: source
 
 # Compile the BLAS and LAPACK libraries, if required.
 
-xblas:
-	cd blas; $(MAKE)
-.PHONY: xblas
-
-xlapack:
-	cd lapack; $(MAKE)
-.PHONY: xlapack
-
 check: source
 times: source
 
