--- configure.ac.orig	2010-07-12 20:13:09.000000000 +0200
+++ configure.ac	2010-07-12 20:13:20.000000000 +0200
@@ -193,11 +193,6 @@
 else M2_CPPFLAGS="$M2_CPPFLAGS -DNDEBUG"
 fi
 
-if test "$OPTIMIZE" = yes
-then CFLAGS="$CFLAGS -O3"
-     CXXFLAGS="$CFLAGS -O3"
-fi
-
 AC_PROG_CC() 			# set CFLAGS before this
 AC_SUBST(GCC)			# gets set to yes or no by AC_PROG_CC
 AC_PROG_CXX()			# set CXXFLAGS before this
