Index: pytorch-1.4.0_p0/setup.py
===================================================================
--- pytorch-1.4.0_p0.orig/setup.py
+++ pytorch-1.4.0_p0/setup.py
@@ -288,9 +288,7 @@ def build_deps():
             sys.exit(1)
 
     check_file(os.path.join(third_party_path, "gloo", "CMakeLists.txt"))
-    check_file(os.path.join(third_party_path, "pybind11", "CMakeLists.txt"))
     check_file(os.path.join(third_party_path, 'cpuinfo', 'CMakeLists.txt'))
-    check_file(os.path.join(third_party_path, 'tbb', 'Makefile'))
     check_file(os.path.join(third_party_path, 'onnx', 'CMakeLists.txt'))
     check_file(os.path.join(third_party_path, 'foxi', 'CMakeLists.txt'))
     check_file(os.path.join(third_party_path, 'QNNPACK', 'CMakeLists.txt'))
@@ -341,9 +339,6 @@ def build_deps():
         if not same:
             shutil.copyfile(orig_file, sym_file)
 
-    dir_util.copy_tree('third_party/pybind11/include/pybind11/',
-                       'torch/include/pybind11')
-
 ################################################################################
 # Building dependent libraries
 ################################################################################
