Index: pytorch-1.4.0/CMakeLists.txt
===================================================================
--- pytorch-1.4.0.orig/CMakeLists.txt
+++ pytorch-1.4.0/CMakeLists.txt
@@ -173,7 +173,7 @@ option(USE_REDIS "Use Redis" OFF)
 option(USE_ROCKSDB "Use RocksDB" OFF)
 option(USE_SNPE "Use Qualcomm's SNPE library" OFF)
 option(USE_SYSTEM_EIGEN_INSTALL
-    "Use system Eigen instead of the one under third_party" OFF)
+    "Use system Eigen instead of the one under third_party" ON)
 option(USE_TENSORRT "Using Nvidia TensorRT library" OFF)
 option(USE_ZMQ "Use ZMQ" OFF)
 option(USE_ZSTD "Use ZSTD" OFF)
