--- ./COMPONENT_SRC_3.2.6/build_configure.org	2007-04-24 18:40:50.000000000 +0200
+++ ./COMPONENT_SRC_3.2.6/build_configure	2008-02-21 09:57:25.000000000 +0100
@@ -144,7 +144,7 @@
 
 # make a link allowing AC_OUTPUT to find the salome_adm/.../*.in  files
 echo "" >> configure.in_tmp1
-echo 'ln -fs ${KERNEL_ROOT_DIR}/salome_adm ${ROOT_SRCDIR}/.' >> configure.in_tmp1
+echo 'cp -prf ${KERNEL_ROOT_DIR}/salome_adm ${ROOT_SRCDIR}/.' >> configure.in_tmp1
 
 echo  "" >> configure.in_tmp1
 echo "AC_OUTPUT([ \\" >> configure.in_tmp1
--- ./COMPONENT_SRC_3.2.6/adm_local/Makefile.in.org	2007-04-24 18:40:50.000000000 +0200
+++ ./COMPONENT_SRC_3.2.6/adm_local/Makefile.in	2008-02-21 09:58:54.000000000 +0100
@@ -26,12 +26,12 @@
 all: resources
 
 install:
-	cp -rf @top_srcdir@/adm_local @prefix@
+#	cp -rf @top_srcdir@/adm_local @prefix@
 
 bin:
 
 resources :
-	cp -rf @top_srcdir@/adm_local $(top_builddir)
+#	cp -rf @top_srcdir@/adm_local $(top_builddir)
 
 inc:
 
