--- src5.1.3.orig/COMPONENT_SRC_5.1.3/src/AddComponent/Makefile.in	2010-03-07 22:08:31.371569651 +0100
+++ src5.1.3/COMPONENT_SRC_5.1.3/src/AddComponent/Makefile.in	2010-03-07 22:09:05.060321118 +0100
@@ -571,7 +571,7 @@
 	done
 AddComponent_CheckOfUndefined$(EXEEXT): $(AddComponent_CheckOfUndefined_OBJECTS) $(AddComponent_CheckOfUndefined_DEPENDENCIES) 
 	@rm -f AddComponent_CheckOfUndefined$(EXEEXT)
-	$(AddComponent_CheckOfUndefined_LINK) $(AddComponent_CheckOfUndefined_OBJECTS) $(AddComponent_CheckOfUndefined_LDADD) $(LIBS)
+	$(AddComponent_CheckOfUndefined_LINK) $(AddComponent_CheckOfUndefined_OBJECTS) $(AddComponent_CheckOfUndefined_LDADD) $(LIBS) $(MPI_LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
