--- Makefile.orig	2008-09-19 15:46:36.000000000 +0200
+++ Makefile	2008-09-19 15:47:50.000000000 +0200
@@ -288,13 +288,13 @@
 #
 # Program to use to decompress a data file
 #
-DECOMPRESS = /usr/bin/bunzip2
+DECOMPRESS = /bin/bunzip2
 
 
 #
 # Program to use to compress a data file
 #
-COMPRESS = /usr/bin/bzip2
+COMPRESS = /bin/bzip2
 
 #
 # Program to use to generate a signature

## justin 090919 I am not able to fix this.

--- Makefile.orig	2008-09-19 16:35:43.000000000 +0200
+++ Makefile	2008-09-19 16:36:25.000000000 +0200
@@ -335,7 +335,7 @@
 F90FLAGS = -O2 -march=prescott -pipe -fomit-frame-pointer -fno-range-check
 F90LDFLAGS = 
 M4FLAGS = -Dfcb_bytes_in_rec=131072
-TIME = time
+#TIME = time
 
 ifneq ($(NOFORTRAN),)
 F90C =

