--- Makefile.orig	2008-09-16 17:03:36.602973240 +0200
+++ Makefile	2008-09-16 17:04:49.054975881 +0200
@@ -358,7 +358,7 @@
 DATADIRI  = $(ROOT)/../CBFlib_$(VERSION)_Data_Files_Input
 DATADIRO  = $(ROOT)/../CBFlib_$(VERSION)_Data_Files_Output
 DATADIRS  = $(ROOT)/../CBFlib_$(VERSION)_Data_Files_Output_Sigs_Only
-INSTALLDIR  = $(HOME)
+INSTALLDIR  = $(DESTDIR)/usr
 
 #
 # URLs from which to retrieve the data directories
