--- rstudio-0.96.331/src/cpp/desktop/resources/freedesktop/rstudio.desktop.in.old	2012-10-11 16:11:09.247971451 +0200
+++ rstudio-0.96.331/src/cpp/desktop/resources/freedesktop/rstudio.desktop.in	2012-10-11 16:12:30.967971366 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Exec=${CMAKE_INSTALL_PREFIX}/${RSTUDIO_INSTALL_BIN}/rstudio %F
-Icon=rstudio.png
+Icon=rstudio
 Type=Application
 Terminal=false
 Name=RStudio
-Categories=Development
-MimeType=text/x-r-source;text/x-r;text-x-R;text/x-r-doc;text/x-r-sweave;text/x-r-markdown;text/x-r-html;text/x-r-html;application/x-r-data;application/x-r-project;text/x-r-history;text/x-r-profile;text/x-tex;text/x-markdown;text/html;text/css;text/javascript;
+Categories=Development;
+MimeType=text/x-r-source;text/x-r;text/x-R;text/x-r-doc;text/x-r-sweave;text/x-r-markdown;text/x-r-html;application/x-r-data;application/x-r-project;text/x-r-history;text/x-r-profile;text/x-tex;text/x-markdown;text/html;text/css;text/javascript;
 
 
