 burrow/spectrum/spectrum.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/burrow/spectrum/spectrum.c b/burrow/spectrum/spectrum.c
index 5e099d8..38040bb 100644
--- a/burrow/spectrum/spectrum.c
+++ b/burrow/spectrum/spectrum.c
@@ -35,6 +35,7 @@
 #include "spectrum_priv.h"
 #include "skiplist.h"
 #include "debug.h"
+#include <stdlib.h>
 
 /* for debugging purposes; make value available */
 static const gdouble datum_unknown_value_substitute=DATUM_UNKNOWN_VALUE_SUBSTITUTE;
