From 605e69e70ce7b4c505be91696612e98649ec383f Mon Sep 17 00:00:00 2001
From: Alexey Shvetsov <alexxy@gentoo.org>
Date: Thu, 15 Nov 2012 08:29:04 +0400
Subject: [PATCH 1/4] Fix detection of nouveau in case its builtin

Signed-off-by: Alexey Shvetsov <alexxy@gentoo.org>
---
 common/autogen.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/autogen.sh b/common/autogen.sh
index 469d3d8..1dca70f 100644
--- a/common/autogen.sh
+++ b/common/autogen.sh
@@ -4,6 +4,8 @@ if [ ! $(lsmod | grep nvidia | wc -l) -eq 0 ] ; then
 	driver='nvi'
 elif [ ! $(lsmod | grep nouveau | wc -l) -eq 0 ] ; then
 	driver='nouveau'
+elif [ ! $(zgrep NOUVEAU /proc/config.gz | grep y | wc -l) -eq 0 ] ; then
+	driver='nouveau'
 elif [ ! $(lsmod | grep pscnv | wc -l) -eq 0 ] ; then
 	driver='pscnv'
 else
-- 
1.7.12.3

