--- ccp4-6.1.1/lib/cctbx/cctbx_sources/libtbx/libtbx/config.py.orig	2009-05-27 16:55:30.712547511 -0500
+++ ccp4-6.1.1/lib/cctbx/cctbx_sources/libtbx/libtbx/config.py	2009-05-27 16:55:44.446546721 -0500
@@ -802,8 +802,6 @@
     else:
       action = self.write_bin_sh_dispatcher
       ext = ""
-      try: os.chmod(source_file, 0755)
-      except OSError: pass
     target_file_ext = target_file + ext
     remove_or_rename(target_file_ext)
     try: action(source_file, target_file_ext)
