--- src/frontend/control.c.old	2004-12-29 13:53:16.012519944 +0000
+++ src/frontend/control.c	2004-12-29 13:53:17.683265952 +0000
@@ -818,6 +818,17 @@
     return (0); /* va: which value? */
 }
 
+/* va: This totally frees the control structures */
+void cp_free_control(void) 
+{
+    int i;
+    
+    for (i=stackp; i>=0; i--) ctl_free(control[i]);
+    
+    control[0] = cend[0] = NULL;
+    stackp = 0;
+}
+
 /* This blows away the control structures... */
 void
 cp_resetcontrol(void)
@@ -878,16 +889,3 @@
             cend[stackp] = cend[stackp]->co_parent;
     return;
 }
-
-
-/* va: This totally frees the control structures */
-void cp_free_control(void) 
-{
-    int i;
-    
-    for (i=stackp; i>=0; i--) ctl_free(control[i]);
-    
-    control[0] = cend[0] = NULL;
-    stackp = 0;
-}
-
