diff -Nadurp ng-spice-rework-17/src/frontend/numparam/general.h ng-spice-rework-17-r1/src/frontend/numparam/general.h
--- ng-spice-rework-17/src/frontend/numparam/general.h	2005-04-24 23:18:02.000000000 +0200
+++ ng-spice-rework-17-r1/src/frontend/numparam/general.h	2006-05-06 23:07:39.000000000 +0200
@@ -136,7 +136,7 @@ Type(Pchar, char AT)
 #define Str(n,a)      char a[n+3]={0,0,(char)n}  /* n<255 ! */
 #define Sini(s)       sini(s,sizeof(s)-4)
 
-Cconst(Maxstr,2004) /* was 255, string maxlen, may be up to 32000 or so */
+Cconst(Maxstr,32004) /* was 255, string maxlen, may be up to 32000 or so */
 
 typedef char string[258];
 
diff -Nadurp ng-spice-rework-17/src/frontend/numparam/numparam.h ng-spice-rework-17-r1/src/frontend/numparam/numparam.h
--- ng-spice-rework-17/src/frontend/numparam/numparam.h	2005-05-29 03:18:20.000000000 +0200
+++ ng-spice-rework-17-r1/src/frontend/numparam/numparam.h	2006-05-06 23:06:37.000000000 +0200
@@ -28,8 +28,8 @@ Cconst(Llen,1000)
 typedef char str20 [24];
 typedef char str80 [84];
 
-Cconst(Maxline, 1000) /* size of initial unexpanded circuit code */
-Cconst(Maxckt,  5000)  /* size of expanded circuit code */
+Cconst(Maxline, 10000) /* size of initial unexpanded circuit code */
+Cconst(Maxckt,  30000)  /* size of expanded circuit code */
 
 
 typedef Pchar auxtable; /* dummy */
