diff -Nadurp ng-spice-rework-17/src/main.c ng-spice-rework-17-r1/src/main.c
--- ng-spice-rework-17/src/main.c	2005-05-31 18:47:48.000000000 +0200
+++ ng-spice-rework-17-r1/src/main.c	2006-05-09 10:24:47.000000000 +0200
@@ -516,6 +516,7 @@ show_help(void)
 	   "  -i, --interactive         run in interactive mode\n"
 	   "  -n, --no-spiceinit        don't load the local or user's config file\n"
 	   "  -o, --output=FILE         set the outputfile\n"
+	   "  -p, --pipe                run in I/O pipe mode\n"
 	   "  -q, --completion          activate command completion\n"
 	   "  -r, --rawfile=FILE        set the rawfile output\n"             
 	   "  -s, --server              run spice as a server process\n"
@@ -738,6 +739,7 @@ main(int argc, char **argv)
 	    {"interactive", 0, 0, 'i'},
 	    {"no-spiceinit", 0, 0, 'n'},
 	    {"output", 0, 0, 'o'},
+	    {"pipe", 0, 0, 'p'},
 	    {"completion", 0, 0, 'q'},
 	    {"rawfile", 1, 0, 'r'},
 	    {"server", 0, 0, 's'},
@@ -745,7 +747,7 @@ main(int argc, char **argv)
 	    {0, 0, 0, 0}
 	};
 
-	c = getopt_long (argc, argv, "hvbc:ihno:qr:st:",
+	c = getopt_long (argc, argv, "hvbc:ihno:pqr:st:",
 			 long_options, &option_index);
 	if (c == -1)
 	    break;
@@ -779,6 +781,11 @@ main(int argc, char **argv)
 		iflag = TRUE;
 		break;
 
+	    case 'p':		/* Pipe mode */
+		iflag = TRUE;
+		istty = TRUE;
+		break;
+
 	    case 'n':		/* Don't read initialisation file */
 		readinit = FALSE;
 		break;
