diff -Nadurp ng-spice-rework-17/src/frontend/com_let.c ng-spice-rework-17-r1/src/frontend/com_let.c
--- ng-spice-rework-17/src/frontend/com_let.c	2003-12-25 18:30:50.000000000 +0100
+++ ng-spice-rework-17-r1/src/frontend/com_let.c	2006-05-06 23:14:10.000000000 +0200
@@ -156,7 +156,7 @@ com_let(wordlist *wl)
 	n->v_flags = (t->v_flags | VF_PERMANENT);
 	n->v_length = t->v_length;
 
-	if (!t->v_numdims) {
+	if ((!t->v_numdims)<=1) {
 	    n->v_numdims = 1;
 	    n->v_dims[0] = n->v_length;
 	} else {
