--- hisat2-2.2.1/Makefile.ori	2020-08-07 13:12:25.393396171 +0200
+++ hisat2-2.2.1/Makefile	2020-08-07 13:15:56.845080223 +0200
@@ -152,30 +152,30 @@
 VERSION = $(shell cat VERSION)
 
 # Convert BITS=?? to a -m flag
-BITS=32
+BITS?=32
 ifeq (x86_64,$(shell uname -m))
-BITS=64
+BITS?=64
 endif
 # msys will always be 32 bit so look at the cpu arch instead.
 ifneq (,$(findstring AMD64,$(PROCESSOR_ARCHITEW6432)))
 	ifeq (1,$(MINGW))
-		BITS=64
+		BITS?=64
 	endif
 endif
 BITS_FLAG =
 
 ifeq (32,$(BITS))
-	BITS_FLAG = -m32
+	BITS_FLAG ?= -m32
 endif
 
 ifeq (64,$(BITS))
-	BITS_FLAG = -m64
+	BITS_FLAG ?= -m64
 endif
-SSE_FLAG=-msse2
+SSE_FLAG?=-msse2
 
-DEBUG_FLAGS    = -O0 -g3 $(BITS_FLAG) $(SSE_FLAG)
+DEBUG_FLAGS    = -g3 -O0 $(CXXFLAGS) $(BITS_FLAG) $(SSE_FLAG)
 DEBUG_DEFS     = -DCOMPILER_OPTIONS="\"$(DEBUG_FLAGS) $(EXTRA_FLAGS)\""
-RELEASE_FLAGS  = -O3 $(BITS_FLAG) $(SSE_FLAG) -funroll-loops -g3
+RELEASE_FLAGS  = $(CXXFLAGS) $(BITS_FLAG) $(SSE_FLAG) -funroll-loops
 RELEASE_DEFS   = -DCOMPILER_OPTIONS="\"$(RELEASE_FLAGS) $(EXTRA_FLAGS)\""
 NOASSERT_FLAGS = -DNDEBUG
 FILE_FLAGS     = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
