 source/DRAWxtl55/Makefile | 27 ++++++++-------------------
 1 file changed, 8 insertions(+), 19 deletions(-)

diff --git a/source/DRAWxtl55/Makefile b/source/DRAWxtl55/Makefile
index 24b2bbc..a6e03b4 100644
--- a/source/DRAWxtl55/Makefile
+++ b/source/DRAWxtl55/Makefile
@@ -42,21 +42,20 @@ ifeq ($(OS),Linux)
   endif
 
 # compiler name:
-  CXX             = g++
+  CXX             ?= g++
 
 # flags for C++ compiler:
-  OPTIM           = -g -Wall -Wunused  -fno-exceptions
 # work around a bug in freeglut 2.4 - only enable this if you have to use a version 
 # of fltk before 1.1.8 AND exactly version 2.4 of freeglut
-#  GLUT		  = -DFREEGLUT24
-  CXXFLAGS        = $(OPTIM)  -I/usr/X11R6/include $(GLUT)
+#  GLUT		  = -DFREEGLUT26
+ FLTKFLAGS        = -I/usr/X11R6/include $(GLUT) `fltk-config --cxxflags`
 
 # libraries to link with:
-  GLDLIBS  =  -L/usr/X11R6/lib  -lGLU -lGL -lXinerama -lXft -lpthread -lm  -lXext -lX11  -lXpm \
+  GLDLIBS  =  -L/usr/X11R6/lib  -lGLU -lGL -lXft -lpthread -lm  -lXext -lX11  -lXpm \
 		-lfltk \
 # -lglut # only needed with old versions of fltk before 1.1.8
 #       -lXi -lgdkglext-x11-1.0a  # Fedora Core 2 needs these libraries
-  LINKFLTKGL  = $(libdir)libfltk.a $(libdir)libfltk_gl.a
+  LINKFLTKGL  = `fltk-config --ldflags` -lfltk_gl
 
 # The extension to use for executables...
   EXEEXT          =
@@ -155,16 +154,6 @@ endif
 #
 ##################################################################################
 
-ifeq ($(OPTIM),error)
-  OPTIM=-O
-  CXX=cc
-  CXXFLAGS=$(OPTIM)
-  GLDLIBS=-lfltk_gl -lfltk -lglut -lGLU -lGL
-endif
-
-# Be quiet when building...
-.SILENT:
-
 # Build commands and filename extensions...
 .SUFFIXES:	.rc .cxx .h .o $(EXEEXT)
 
@@ -174,7 +163,7 @@ endif
 
 .cxx.o:
 	echo Compiling $< ...
-	$(CXX) -I$(includedir) $(CXXFLAGS) -c $<
+	$(CXX) $(CXXFLAGS) $(FLTKFLAGS) -c $<
 
 CPPFILES =\
 	CrystalView.cxx \
@@ -216,7 +205,7 @@ clean:
 .deps:  $(CPPFILES)
 	echo "Building dependencies..."
 	echo >.deps
-	for i in $(CPPFILES); do gcc -MM $(PREINC) $$i >>.deps; done
+	for i in $(CPPFILES); do $(CXX) -MM $(PREINC) $$i >>.deps; done
 
 -include .deps
 
@@ -225,7 +214,7 @@ clean:
 $(TARGET) : .deps $(OBJFILES) 
 	echo Linking $@...
 	mkdir -p ../../exe
-	$(CXX) $(CXXFLAGS) -o $@ $(OBJFILES) \
+	$(CXX) $(LDFLAGS) $(CXXFLAGS) $(FLTKFLAGS) -o $@ $(OBJFILES) \
 		$(LINKFLTKGL) $(GLDLIBS) 
 	$(POSTLINK) $@
 
