diff -Naurp dx-4.4.4-orig/bin/dx.in dx-4.4.4/bin/dx.in
--- dx-4.4.4-orig/bin/dx.in	2008-01-21 17:04:51.000000000 +0100
+++ dx-4.4.4/bin/dx.in	2008-01-21 17:05:14.000000000 +0100
@@ -60,8 +60,8 @@ exit 1
 #
 longhelp() {
 
-if test -f "$dxroot/man/catl/dx.l" ; then
-    more "$dxroot/man/catl/dx.l"
+if test -f "$dxroot"/man/catl/dx.l* ; then
+    man "$dxroot"/man/catl/dx.l*
 else
     echo "cannot find $dxroot/man/catl/dx.l"
     echo "set the DXROOT environment variable to the root of the"
