diff -Naur gle-4.0.11.orig/Makefile.gcc gle-4.0.11/Makefile.gcc
--- gle-4.0.11.orig/Makefile.gcc	2005-12-17 03:08:30.000000000 +0000
+++ gle-4.0.11/Makefile.gcc	2006-04-14 14:34:00.000000000 +0100
@@ -12,26 +12,26 @@
 	-mkdir include
 	-mkdir lib
 	-mkdir bin
-	make -C src/gle/bitmap -f Makefile.gcc
-	make -C src/gle/surface -f Makefile.gcc
-	make -C src/gle/letzfitz -f Makefile.gcc
-	make -C src/gle/tokens -f Makefile.gcc
-	make -C src/gle -f Makefile.gcc
-	make -C src/fbuild -f Makefile.gcc
-	make -C src/makefmt -f Makefile.gcc
-	make -C src/fonts -f Makefile.gcc
-	make -C src/TeX -f Makefile.gcc
+	$(MAKE) -C src/gle/bitmap $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/gle/surface $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/gle/letzfitz $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/gle/tokens $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/gle $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/fbuild $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/makefmt $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/fonts $(MAKECMDGOALS) -f Makefile.gcc
+	$(MAKE) -C src/TeX $(MAKECMDGOALS) -f Makefile.gcc
 
 clean:
-	make -C src/gle/bitmap -f Makefile.gcc clean
-	make -C src/gle/surface -f Makefile.gcc clean
-	make -C src/gle/letzfitz -f Makefile.gcc clean
-	make -C src/gle/tokens -f Makefile.gcc clean
-	make -C src/gle -f Makefile.gcc clean
-	make -C src/fbuild -f Makefile.gcc clean
-	make -C src/makefmt -f Makefile.gcc clean
-	make -C src/fonts -f Makefile.gcc clean
-	make -C src/TeX -f Makefile.gcc clean
+	$(MAKE) -C src/gle/bitmap $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/gle/surface $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/gle/letzfitz $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/gle/tokens $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/gle $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/fbuild $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/makefmt $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/fonts $(MAKECMDGOALS) -f Makefile.gcc clean
+	$(MAKE) -C src/TeX $(MAKECMDGOALS) -f Makefile.gcc clean
 	-$(RMDIR) include
 	-$(RMDIR) lib
 	-$(RMDIR) bin
@@ -68,7 +68,7 @@
 	cp -r src/fbuild src/fonts src/gle src/makefmt src/TeX $(ZIP_SRC_DIR)/src/
 	mkdir -p $(ZIP_SRC_DIR)/src/samples/
 	cp -r src/samples/Makefile.* src/samples/sample* $(ZIP_SRC_DIR)/src/samples/
-	make -C $(ZIP_SRC_DIR) -f Makefile.gcc clean
+	$(MAKE) -C $(ZIP_SRC_DIR) $(MAKECMDGOALS) -f Makefile.gcc clean
 	-(cd $(ZIP_SRC_DIR); find . -name "CVS" -exec rm -rf {} \;)
 	-(cd $(ZIP_SRC_DIR); find . -name ".svn" -exec rm -rf {} \;)
 	zip -r $(ZIP_SRC_NAME) $(ZIP_SRC_DIR)
diff -Naur gle-4.0.11.orig/config.gcc gle-4.0.11/config.gcc
--- gle-4.0.11.orig/config.gcc	2005-12-17 03:08:30.000000000 +0000
+++ gle-4.0.11/config.gcc	2006-04-14 14:29:29.000000000 +0100
@@ -65,5 +65,5 @@
 	LIBS			:= $(LIBS) -lX11
 endif
 
-CXXFLAGS	=-g $(VERSION_INFO) $(EXTRA_DEFS) $(EXTRA_INCLUDE_PATHS) -DINSTALL_DIR="\"$(INSTALL_DIR)\""
+CXXFLAGS	=$(VERSION_INFO) $(EXTRA_DEFS) $(EXTRA_INCLUDE_PATHS) -DINSTALL_DIR="\"$(INSTALL_DIR)\" $(EXTRA_CXXFLAGS)"
 LINKFLAGS	=$(EXTRA_LIB_PATHS) $(LIBS)
