--- libAfterImage-1.20/ungif.h.orig	2011-01-15 08:52:22.000000000 +0300
+++ libAfterImage-1.20/ungif.h	2013-10-19 10:48:28.241094055 +0400
@@ -7,6 +7,12 @@
 extern "C" {
 #endif
 
+#if (GIFLIB_MAJOR>4) || ((GIFLIB_MAJOR==4) && (GIFLIB_MINOR>=2)) 
+static inline void PrintGifError(void) {
+    fprintf(stderr, "%s\n", GifErrorString());
+}
+#endif
+
 #ifdef __GNUC__
 #define ASIM_PrintGifError() do{ fprintf( stderr, "%s():%d:<%s> ",__FUNCTION__, __LINE__, path?path:"null" ); PrintGifError(); }while(0)
 #else
