diff -Nur libAfterImage-1.07/apps/Makefile.in libAfterImage-1.07.new/apps/Makefile.in
--- libAfterImage-1.07/apps/Makefile.in	2005-02-28 14:52:47.000000000 +0000
+++ libAfterImage-1.07.new/apps/Makefile.in	2006-10-15 15:26:09.000000000 +0100
@@ -40,7 +40,7 @@
 AFTER_DIR	= @with_afterdir@
 
 # need this for dependancies :
-LIBAFTERIMAGE_PATH = ../../libAfterImage
+LIBAFTERIMAGE_PATH = ../
 LIB_STATIC      = $(LIBAFTERIMAGE_PATH)/libAfterImage.a
 LIB_SHARED      = $(LIBAFTERIMAGE_PATH)/libAfterImage.so
 
diff -Nur libAfterImage-1.07/Makefile.in libAfterImage-1.07.new/Makefile.in
--- libAfterImage-1.07/Makefile.in	2005-06-03 21:44:06.000000000 +0100
+++ libAfterImage-1.07.new/Makefile.in	2006-10-15 15:25:24.000000000 +0100
@@ -78,7 +78,7 @@
 MV		= @MV@
 RM		= @RM@
 RMF		= @RM@ -f
-MKDIR		= @MKDIR@
+MKDIR		= @MKDIR@ -p
 FIND		= @FIND@
 XARGS		= @XARGS@
 LDCONFIG	= @LDCONFIG@
@@ -135,21 +135,6 @@
 		@(if [ -d $(LIBDIR) ] && [ -w $(LIBDIR) ]; then \
 		    echo "$(INSTALL_LIB) $(LIB_STATIC) $(LIBDIR)"; \
 		    $(INSTALL_LIB) $(LIB_STATIC) $(LIBDIR); \
-		    if [ `uname` = "Linux" ]; then \
-			if test $(LIBDIR) = "/lib" || test $(LIBDIR) = "/usr/lib"; then \
-		    	    echo "" > /dev/null; \
-			elif grep -q $(LIBDIR) /etc/ld.so.conf > /dev/null 2>&1; then \
-		    	    echo "" > /dev/null; \
-			else \
-		    	    echo "Unable to find $(LIBDIR) in ld.so.conf. In order to use "; \
-		    	    echo "$(LIB_STATIC), you may need to add it or set LD_LIBRARY_PATH."; \
-			fi; \
-			if test -w /etc; then \
-			    echo "$(LDCONFIG)"; \
-			    $(LDCONFIG); \
-			fi; \
-		    fi ;\
-		fi \
 		)
 
 install.script:
@@ -161,18 +146,6 @@
 		$(RM) -f $(LIBDIR)/$(LIB_SHARED).$(LIBVERMAJOR) $(LIBDIR)/$(LIB_SHARED); \
 		$(LN_S) -f $(LIB_SHARED).$(LIBVER) $(LIBDIR)/$(LIB_SHARED).$(LIBVERMAJOR); \
 		$(LN_S) -f $(LIB_SHARED).$(LIBVERMAJOR) $(LIBDIR)/$(LIB_SHARED); \
-		if test `uname` = "Linux"; then \
-		   if test $(LIBDIR) = "/lib" || test $(LIBDIR) = "/usr/lib"; then \
-		     echo "" > /dev/null; \
-		   elif grep -q $(LIBDIR) /etc/ld.so.conf > /dev/null 2>&1; then \
-		     echo "" > /dev/null; \
-		   else \
-		     echo "Unable to find $(LIBDIR) in ld.so.conf. In order to use "; \
-		     echo "$(LIB_SHARED), you may need to add it or set LD_LIBRARY_PATH."; \
-		   fi; \
-		   echo "$(LDCONFIG)"; \
-		   $(LDCONFIG); \
-		 fi \
 		)
 
 install.cyg:
@@ -221,7 +194,7 @@
 		)
 
 install.apps: install.lib
-		cd apps; make install; cd ..
+		cd apps; $(MAKE) install; cd ..
 
 uninstall.bin:
 
@@ -349,7 +322,6 @@
 	  			$(INSTALL_DATA) $$page_name.man $(AFTER_MAN_DIR)/$$page_name.1x; \
 			done; cd ../..; \
 		fi
-			
 
 install.doc:
 		@if [ -d doc/html ] ; then \
@@ -364,7 +336,7 @@
 			cd ../..; \
 		fi
 
-install:	install.lib install.man install.doc install.apps
+install:	install.lib install.man install.apps
 
 
 uninstall:	uninstall.lib
