--- src/config/makefile.h	2011-12-04 12:00:52.000000000 +0100
+++ src/config/makefile.h	2011-12-04 12:01:56.000000000 +0100
@@ -2133,7 +2133,7 @@
 endif
 ifdef USE_PYTHON64
-  CORE_LIBS += $(PYTHONHOME)/lib64/python$(PYTHONVERSION)/config/libpython$(PYTHONVERSION).$(PYTHONLIBTYPE)
+  CORE_LIBS += $(PYTHONHOME)/lib64/libpython$(PYTHONVERSION).$(PYTHONLIBTYPE)
 else
-  CORE_LIBS += $(PYTHONHOME)/lib/python$(PYTHONVERSION)/config/libpython$(PYTHONVERSION).$(PYTHONLIBTYPE)
+  CORE_LIBS += $(PYTHONHOME)/lib/libpython$(PYTHONVERSION).$(PYTHONLIBTYPE)
 endif
 endif
