--- config/amos.m4.ori	2010-12-05 17:33:13.000000000 +0100
+++ config/amos.m4	2010-12-05 17:46:26.000000000 +0100
@@ -219,7 +219,7 @@
        ); then
 
       if test x"$amos_qt_dir" = x; then
-        amos_qt_dir="$amos_qt_bin_dir"
+        amos_qt_dir=
       fi
       if test x"$amos_qt_include_dir" = x; then
         amos_qt_include_dir="$amos_qt_dir/include"
@@ -279,16 +279,6 @@
     QT_LIBS="$amos_qt_LIBS"
     # If amos_qt_dir is defined, utilities are expected to be in the
     # bin subdirectory
-    if test x"$amos_qt_dir" != x; then
-        if test -x "$amos_qt_dir/bin/uic"; then
-          QT_UIC="$amos_qt_dir/bin/uic"
-        else
-          # Old versions of Qt don't have uic
-          QT_UIC=
-        fi
-      QT_MOC="$amos_qt_dir/bin/moc"
-    else
-      # Or maybe we are told where to look for the utilities
       if test x"$amos_qt_bin_dir" != x; then
         if test -x "$amos_qt_bin_dir/uic"; then
           QT_UIC="$amos_qt_bin_dir/uic"
@@ -297,11 +287,19 @@
           QT_UIC=
         fi
         QT_MOC="$amos_qt_bin_dir/moc"
-      else
+      if test x"$QT_MOC" = "x/moc"; then
       # Last possibility is that they are in $PATH
         QT_UIC="`which uic`"
         QT_MOC="`which moc`"
       fi
+    elif test x"$amos_qt_dir" != x; then
+        if test -x "$amos_qt_dir/bin/uic"; then
+          QT_UIC="$amos_qt_dir/bin/uic"
+        else
+          # Old versions of Qt don't have uic
+          QT_UIC=
+        fi
+      QT_MOC="$amos_qt_dir/bin/moc"
     fi
     # All variables are defined, report the result
     AC_MSG_RESULT([$have_qt])
