diff --git a/setup.py b/setup.py
index c8681c2..41c21a5 100644
--- a/setup.py
+++ b/setup.py
@@ -20,7 +18,7 @@ MAINTAINER_EMAIL = 'mwaskom@stanford.edu'
 URL = 'http://pysurfer.github.com'
 LICENSE = 'BSD (3-clause)'
 DOWNLOAD_URL = 'https://github.com/nipy/PySurfer'
-VERSION = version
+VERSION = "0.6"
 
 import setuptools  # we are using a setuptools namespace
 from numpy.distutils.core import setup
