From b83750ab917b12500f9755dc4c81f99eb55bb67e Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@tu-dortmund.de>
Date: Sat, 6 Jun 2015 10:35:32 +0200
Subject: [PATCH] Install cmake configuration to lib/cmake/clBLAS

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a23cd37..5c378dc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -300,7 +300,7 @@ endif( )
 if(WIN32)
   set(destdir CMake)
 else()
-  set(destdir share/clBLAS)
+  set(destdir lib${SUFFIX_LIB}/cmake/clBLAS)
 endif()
 string(REGEX REPLACE "[^/]+" ".." reldir "${destdir}")
 configure_file(
-- 
2.3.6

