--- a/Makefile	2005-08-25 22:18:31.000000000 +0200
+++ b/Makefile	2009-03-21 15:11:55.000000000 +0100
@@ -4,8 +4,8 @@
 # You'll probably have to tweak this for your platform.
 
 
-CC = cc
-CFLAGS = -c -g -ansi -I/usr/local/include
+CC ?= gcc
+CFLAGS += -c -ansi -I/usr/include
 LIBS = -L/usr/X11R6/lib -lglut -lGLU -lGL -lX11 -lXmu -lXext -lm
 
 
@@ -19,13 +19,13 @@
 	$(CC) $(CFLAGS) tr.c
 
 trdemo1: trdemo1.o tr.o
-	$(CC) trdemo1.o tr.o $(LIBS) -o $@
+	$(CC) $(LDFLAGS) trdemo1.o tr.o $(LIBS) -o $@
 
 trdemo1.o: trdemo1.c tr.h
 	$(CC) $(CFLAGS) trdemo1.c
 
 trdemo2: trdemo2.o tr.o
-	$(CC) trdemo2.o tr.o $(LIBS) -o $@
+	$(CC) $(LDFLAGS) trdemo2.o tr.o $(LIBS) -o $@
 
 trdemo2.o: trdemo2.c tr.h
 	$(CC) $(CFLAGS) trdemo2.c
