Index: setup.py
===================================================================
--- setup.py	(revision 3908)
+++ setup.py	(working copy)
@@ -130,8 +130,8 @@
               "/usr/include/freetype2",
 #                 "/users/warren/ext/include",
 # VMD plugin support
-#              "contrib/uiuc/plugins/include",
-#              "contrib/uiuc/plugins/molfile_plugin/src",
+              "contrib/uiuc/plugins/include",
+              "contrib/uiuc/plugins/molfile_plugin/src",
               "modules/cealign/src",
 	      "modules/cealign/src/tnt", ]
     libs=["GL","GLU","glut","png","z","freetype",
@@ -151,7 +151,7 @@
 # Numeric Python support                    
 #                ("_PYMOL_NUMPY",None),
 # VMD plugin support           
-#               ("_PYMOL_VMD_PLUGINS",None)
+               ("_PYMOL_VMD_PLUGINS",None)
                 ]
     ext_comp_args=["-ffast-math","-funroll-loops","-O3"]
     ext_link_args=[]
@@ -299,7 +299,7 @@
     "layer5/main.c"
 # VMD plugin support
     # switch the 0 to 1 to activate the additional source code
-    ] + 0 * [
+    ] + 1 * [
 # (incomplete support -- only TRJ, TRR, XTC, DCD so far...)
     'contrib/uiuc/plugins/molfile_plugin/src/PlugIOManagerInit.c',
     'contrib/uiuc/plugins/molfile_plugin/src/avsplugin.cpp',
