 setup.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index b0f4dad..4ec5145 100644
--- a/setup.py
+++ b/setup.py
@@ -132,8 +132,8 @@ else: # linux or other unix
               "/usr/include/freetype2",
 #                 "/users/warren/ext/include",
 # VMD plugin support
-#              "contrib/uiuc/plugins/include",
-#              "contrib/uiuc/plugins/molfile_plugin/src",
+              "contrib/uiuc/plugins/include",
+              "contrib/uiuc/plugins/molfile_plugin/src",
               "modules/cealign/src",
 	      "modules/cealign/src/tnt", ]
     libs=["GL","GLU","glut","png","z","freetype", "GLEW",
@@ -153,7 +153,7 @@ else: # linux or other unix
 # Numeric Python support                    
 #                ("_PYMOL_NUMPY",None),
 # VMD plugin support           
-#               ("_PYMOL_VMD_PLUGINS",None)
+               ("_PYMOL_VMD_PLUGINS",None),
 		("NO_MMLIBS",None),
                 ]
     ext_comp_args=["-ffast-math","-funroll-loops","-O3"]
@@ -307,7 +307,7 @@ setup ( # Distribution meta-data
     "layer5/main.c"
 # VMD plugin support
     # switch the 0 to 1 to activate the additional source code
-    ] + 0 * [
+    ] + 1 * [
 # (incomplete support -- only TRJ, TRR, XTC, DCD so far...)
     'contrib/uiuc/plugins/molfile_plugin/src/PlugIOManagerInit.c',
     'contrib/uiuc/plugins/molfile_plugin/src/avsplugin.cpp',
