 setup.py |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index b0f4dad..94d292d 100644
--- a/setup.py
+++ b/setup.py
@@ -180,10 +180,7 @@ setup ( # Distribution meta-data
                 'pymol/opengl/gl',
                 'pymol/opengl/glu',
                 'pymol/opengl/glut',
-                'pymol/wizard',
-		'web',
-		'web/examples',
-		'web/javascript', ],
+                'pymol/wizard', ],
 	ext_modules = [
     Extension("pymol._cmd", [
     "modules/cealign/src/ccealignmodule.cpp",
