--- pymol-0.99rc8/modules/pymol/commanding.py	2007-04-18 02:11:48.000000000 -0400
+++ commanding.py	2007-04-24 02:17:53.000000000 -0400
@@ -129,11 +129,11 @@
                 unlock(0)
             r = DEFAULT_SUCCESS
             if show_splash==1: # generic / open-source
-                png_path = cmd.exp_path("$PYMOL_PATH/data/pymol/splash.png")
+                png_path = cmd.exp_path("$PYMOL_DATA/pymol/splash.png")
             elif show_splash==2: # evaluation builds
-                png_path = cmd.exp_path("$PYMOL_PATH/data/pymol/epymol.png")
+                png_path = cmd.exp_path("$PYMOL_DATA/pymol/epymol.png")
             else: # incentive builds
-                png_path = cmd.exp_path("$PYMOL_PATH/data/pymol/ipymol.png")
+                png_path = cmd.exp_path("$PYMOL_DATA/pymol/ipymol.png")
             if os.path.exists(png_path):
                 cmd.do("_ cmd.load_png('%s',0,quiet=1)"%png_path)
         else:
--- pymol-0.99rc8/modules/pymol/fitting.py	2007-04-18 02:11:48.000000000 -0400
+++ fitting.py	2007-04-24 02:18:07.000000000 -0400
@@ -61,7 +61,7 @@
         r = DEFAULT_ERROR
         source = selector.process(source)
         target = selector.process(target)
-        mfile = cmd.exp_path("$PYMOL_PATH/data/pymol/matrices/"+matrix)
+        mfile = cmd.exp_path("$PYMOL_DATA/pymol/matrices/"+matrix)
         if object==None: object=''
         try:
             lock()
--- pymol-0.99rc8/modules/pymol/importing.py	2007-04-18 02:11:47.000000000 -0400
+++ importing.py	2007-04-24 02:18:46.000000000 -0400
@@ -177,7 +177,7 @@
         r = DEFAULT_ERROR
         
         
-        tables = { 'cmyk' : "$PYMOL_PATH/data/pymol/cmyk.png",
+        tables = { 'cmyk' : "$PYMOL_DATA/pymol/cmyk.png",
                       'pymol' : 'pymol',
                       'rgb' : 'rgb' }
 
--- pymol-0.99rc8/modules/pymol/wizard/mutagenesis.py	2007-04-18 02:11:50.000000000 -0400
+++ mutagenesis.py	2007-04-24 04:08:44.000000000 -0400
@@ -35,8 +35,8 @@
 
         self.dep = default_dep
 
-        self.ind_library = io.pkl.fromFile(os.environ['PYMOL_PATH']+
-                                           "/data/chempy/sidechains/sc_bb_ind.pkl")
+        self.ind_library = io.pkl.fromFile(os.environ['PYMOL_DATA']+
+                                           "/chempy/sidechains/sc_bb_ind.pkl")
         self.load_library()
         self.status = 0 # 0 no selection, 1 mutagenizing
         self.bump_check = 1
@@ -108,8 +108,8 @@
     def load_library(self):
         if self.dep == 'dep':
             if not hasattr(self,'dep_library'):
-                self.dep_library = io.pkl.fromFile(os.environ['PYMOL_PATH']+
-                                           "/data/chempy/sidechains/sc_bb_dep.pkl")
+                self.dep_library = io.pkl.fromFile(os.environ['PYMOL_DATA']+
+                                           "/chempy/sidechains/sc_bb_dep.pkl")
             
     def set_mode(self,mode):
         if mode in self.modes:

