 modules/pymol/invocation.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/modules/pymol/invocation.py b/modules/pymol/invocation.py
index de47849..c5f4487 100644
--- a/modules/pymol/invocation.py
+++ b/modules/pymol/invocation.py
@@ -397,7 +397,5 @@ if __name__=='pymol.invocation':
         if loaded_something and (options.after_load_script!=""):
             options.deferred.append(options.after_load_script)
         options.deferred.extend(final_actions)
-        if options.show_splash and not options.no_gui and not restricted:
-            options.deferred.insert(0,"_do__ cmd.splash(1)")
         if options.full_screen:
             options.deferred.append("_do__ full_screen on")
