 setup.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 366d552..ce0d271 100755
--- a/setup.py
+++ b/setup.py
@@ -161,8 +161,8 @@ else: # linux or other unix
                  "layer5",
                  "/usr/include/freetype2",
               # VMD plugin support
-              #              "contrib/uiuc/plugins/include",
-              #              "contrib/uiuc/plugins/molfile_plugin/src",
+                            "contrib/uiuc/plugins/include",
+                            "contrib/uiuc/plugins/molfile_plugin/src",
                  "modules/cealign/src",
                  "modules/cealign/src/tnt",
                  "generated/include",
@@ -190,7 +190,7 @@ else: # linux or other unix
                    # Numeric Python support                    
                    #                ("_PYMOL_NUMPY",None),
                    # VMD plugin support           
-                   #               ("_PYMOL_VMD_PLUGINS",None)
+                                  ("_PYMOL_VMD_PLUGINS",None),
                    ("_PYMOL_CGO_DRAWARRAYS",None),
                    ("_PYMOL_CGO_DRAWBUFFERS",None),
                    ("_CGO_DRAWARRAYS",None),
@@ -354,7 +354,7 @@ distribution = setup ( # Distribution meta-data
                 "layer5/main.c"
                 # VMD plugin support
                 # switch the 0 to 1 to activate the additional source code
-                ] + 0 * [
+                ] + 1 * [
                 # (incomplete support -- only TRJ, TRR, XTC, DCD so far...)
                 'contrib/uiuc/plugins/molfile_plugin/src/PlugIOManagerInit.c',
                 'contrib/uiuc/plugins/molfile_plugin/src/avsplugin.cpp',
