--- setup.py.old	2009-10-03 01:01:22.444881137 -0500
+++ setup.py	2009-10-03 01:01:41.580752440 -0500
@@ -139,7 +139,7 @@
                 ("_PYMOL_FREETYPE",None),
                 ("_PYMOL_LIBPNG",None),
 # OpenGL shaders
-#               ("_PYMOL_OPENGL_SHADERS",None),
+               ("_PYMOL_OPENGL_SHADERS",None),
 # Numeric Python support                    
 #                ("_PYMOL_NUMPY",None),
 # VMD plugin support           
