diff -uarN pymol-9999.orig/setup.py pymol-9999/setup.py
--- pymol-9999.orig/setup.py	2008-06-18 11:47:58.000000000 +0200
+++ pymol-9999/setup.py	2008-06-18 11:48:52.000000000 +0200
@@ -143,6 +143,7 @@
     def_macros=[("_PYMOL_MODULE",None),
                     ("_PYMOL_INLINE",None),
                     ("_PYMOL_FREETYPE",None),
+                    ("_PYMOL_OPENGL_SHADERS",None),
 # Numeric Python support                    
 #                        ("_PYMOL_NUMPY",None),
 # VMD plugin support                    
