diff -urN pymol-9999.orig/modules/pymol/commanding.py pymol-9999/modules/pymol/commanding.py
--- pymol-9999.orig/modules/pymol/commanding.py	2008-01-11 13:23:32.000000000 +0100
+++ pymol-9999/modules/pymol/commanding.py	2008-01-11 13:22:57.000000000 +0100
@@ -198,11 +198,11 @@
                 _self.unlock(0,_self)
             r = DEFAULT_SUCCESS
             if show_splash==1: # generic / open-source
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/splash.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/splash.png")
             elif show_splash==2: # evaluation builds
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/epymol.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/epymol.png")
             else: # incentive builds
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/ipymol.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/ipymol.png")
             if os.path.exists(png_path):
                 _self.do("_ cmd.load_png('%s',0,quiet=1)"%png_path)
         else:
#diff -urN pymol-9999.orig/modules/pymol/fitting.py pymol-9999/modules/pymol/fitting.py
#--- pymol-9999.orig/modules/pymol/fitting.py	2008-01-11 13:23:32.000000000 +0100
#+++ pymol-9999/modules/pymol/fitting.py	2008-01-11 13:23:00.000000000 +0100
#@@ -68,7 +68,7 @@
#         if string.lower(matrix)=='none':
#             matrix=''
#         if len(matrix):
#-            mfile = cmd.exp_path("$PYMOL_PATH/data/pymol/matrices/"+matrix)
#+            mfile = cmd.exp_path("$PYMOL_DATA/pymol/matrices/"+matrix)
#         else:
#             mfile = ''        
#         # delete existing alignment object (if asked to reset it)
#@@ -135,7 +135,7 @@
#         if string.lower(matrix)=='none':
#             matrix=''
#         if len(matrix):
#-            mfile = cmd.exp_path("$PYMOL_PATH/data/pymol/matrices/"+matrix)
#+            mfile = cmd.exp_path("$PYMOL_DATA/pymol/matrices/"+matrix)
#         else:
#             mfile = ''
#         if object==None: object=''
diff -urN pymol-9999.orig/modules/pymol/importing.py pymol-9999/modules/pymol/importing.py
--- pymol-9999.orig/modules/pymol/importing.py	2008-01-11 13:23:32.000000000 +0100
+++ pymol-9999/modules/pymol/importing.py	2008-01-11 13:23:02.000000000 +0100
@@ -171,7 +171,7 @@
         r = DEFAULT_ERROR
         
         
-        tables = { 'cmyk' : "$PYMOL_PATH/data/pymol/cmyk.png",
+        tables = { 'cmyk' : "$PYMOL_DATA/pymol/cmyk.png",
                       'pymol' : 'pymol',
                       'rgb' : 'rgb' }
 
