 setup.py |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 2b6fa35..9c0556d 100755
--- a/setup.py
+++ b/setup.py
@@ -197,10 +197,7 @@ else: # linux or other unix
                    ("_PYMOL_GL_CALLLISTS",None),
                    ("OPENGL_ES_2",None),
                    ]
-    ext_comp_args = [ "-ffast-math",
-                      "-funroll-loops",
-                      "-O3",
-                      "-g" ]
+    ext_comp_args = []
     ext_link_args = []
 
 
