 setup.py |   28 ----------------------------
 1 files changed, 0 insertions(+), 28 deletions(-)

diff --git a/setup.py b/setup.py
index 18866a2..705b0c5 100755
--- a/setup.py
+++ b/setup.py
@@ -487,31 +487,3 @@ distribution = setup ( # Distribution meta-data
                   define_macros = def_macros
                   )
         ])
-
-# make available for setup2.py
-try:
-    site_packages = distribution.command_obj['install'].install_libbase
-except KeyError:
-    print """
- Error: Please run, 'setup.py install' not 'setup build' or other variant.
-
-"""
-    sys.exit(2)
-
-f = open('setup3.py', 'w')
-print >> f, 'site_packages =', repr(site_packages)
-f.close()
-
-print '''
- After running:
-
-     python setup.py install
-     
- Please run, to complete the installation:
-
-     python setup2.py install
-
- To uninstall PyMOL, run:
-
-     python setup2.py uninstall
-'''
