 pymol/setup.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/pymol/setup.py b/pymol/setup.py
index b8d7070..378bbf6 100644
--- a/pymol/setup.py
+++ b/pymol/setup.py
@@ -185,12 +185,12 @@ elif sys.platform=='darwin':
                   "/sw/include/freetype2/freetype",
                   "/sw/include/freetype2",
                   "/sw/include",
-                  "/usr/X11/include",
+                  "@GENTOO_PORTAGE_EPREFIX@/usr/X11/include",
                   ]
         ext_link_args=[
                        "-L/sw/lib", "-lpng",
-                       "/usr/X11/lib/libGL.dylib",
-                       "/usr/X11/lib/libGLU.dylib",
+                       "@GENTOO_PORTAGE_EPREFIX@/usr/X11/lib/libGL.dylib",
+                       "@GENTOO_PORTAGE_EPREFIX@/usr/X11/lib/libGLU.dylib",
                        "-lfreeglut",
                        "-lglew",
                        "-L/sw/lib/freetype219/lib", "-lfreetype"
@@ -220,7 +220,7 @@ elif sys.platform=='darwin':
 else: # linux or other unix
 
     inc_dirs += [
-                 "/usr/include/freetype2",
+                 "@GENTOO_PORTAGE_EPREFIX@/usr/include/freetype2",
                  ]
     libs = [ "GL",
              "GLU",
@@ -235,7 +235,7 @@ else: # linux or other unix
                    "GLU",
                    "glut",
                    "GLEW"]
-    lib_dirs = [ "/usr/X11R6/lib64", ]
+    lib_dirs = [ "@GENTOO_PORTAGE_EPREFIX@/usr/X11R6/lib64", ]
     def_macros += [
                    ("_PYMOL_INLINE",None),
                    ("_PYMOL_FREETYPE",None),
