Index: setup.py
===================================================================
--- setup.py	(revision 3963)
+++ setup.py	(working copy)
@@ -101,7 +101,7 @@
         #
         # REMEMEBER to use "./ext/bin/python ..."
         #
-        EXT = os.getcwd()+"/ext"
+        EXT = "@GENTOO_PORTAGE_EPREFIX@/usr"
         inc_dirs=["ov/src",
                   "layer0","layer1","layer2",
                   "layer3","layer4","layer5", 
@@ -121,7 +121,7 @@
                     ]
         ext_comp_args=[]
         ext_link_args=[
-	               "-L/usr/X11R6/lib64", "-lGL", "-lXxf86vm",
+	               "-lGL", "-lXxf86vm",
                    "-L"+EXT+"/lib", "-lpng", "-lglut", "-lfreetype"
                    ]
 #============================================================================
@@ -129,7 +129,7 @@
     inc_dirs=["ov/src",
               "layer0","layer1","layer2",
               "layer3","layer4","layer5",
-              "/usr/include/freetype2",
+              "@GENTOO_PORTAGE_EPREFIX@/usr/include/freetype2",
 #                 "/users/warren/ext/include",
 # VMD plugin support
 #              "contrib/uiuc/plugins/include",
@@ -141,7 +141,7 @@
           ]	
     pyogl_libs = ["GL","GLU","glut", "GLEW"]
     lib_dirs=[
-        "/usr/X11R6/lib64",
+        "@GENTOO_PORTAGE_EPREFIX@/usr/usr/X11R6/lib",
 #        "/users/warren/pymol/ext/lib"
         ]
     def_macros=[("_PYMOL_MODULE",None),
