 layer0/ShaderMgr.c |   16 ++++++++--------
 layer1/Setting.c   |    2 +-
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/layer0/ShaderMgr.c b/layer0/ShaderMgr.c
index 929167f..ecb4130 100644
--- a/layer0/ShaderMgr.c
+++ b/layer0/ShaderMgr.c
@@ -386,7 +386,7 @@ int CShaderMgr_ShadersPresent(CShaderMgr * I)
 char * CShaderMgr_ReadShaderFromDisk(PyMOLGlobals * G, const char * fileName) {
   FILE* f;
   long size;
-  char* buffer = NULL, *p, *pymol_path, *shader_path, *fullFile;
+  char* buffer = NULL, *p, *pymol_data, *shader_path, *fullFile;
 
   PRINTFB(G, FB_ShaderMgr, FB_Debugging)
     "CShaderMgr_ReadShaderFromDisk: fileName='%s'\n", fileName
@@ -398,16 +398,16 @@ char * CShaderMgr_ReadShaderFromDisk(PyMOLGlobals * G, const char * fileName) {
     return NULL;
   }
   
-  pymol_path = getenv("PYMOL_PATH");
-  if (!pymol_path){
+  pymol_data = getenv("PYMOL_DATA");
+  if (!pymol_data){
     PRINTFB(G, FB_ShaderMgr, FB_Warnings)
-      " PyMOLShader_NewFromFile-Warning: PYMOL_PATH not set, cannot read shader config files (%s) from disk\n", fileName ENDFB(G);
+      " PyMOLShader_NewFromFile-Warning: PYMOL_DATA not set, cannot read shader config files (%s) from disk\n", fileName ENDFB(G);
     return NULL;
   }
   /* make this a setting */
-  shader_path = "/data/shaders/";
-  fullFile = malloc( sizeof(char) * (strlen(pymol_path)+strlen(shader_path)+strlen(fileName)+1));
-  fullFile = strcpy(fullFile, pymol_path);
+  shader_path = "/shaders/";
+  fullFile = malloc( sizeof(char) * (strlen(pymol_data)+strlen(shader_path)+strlen(fileName)+1));
+  fullFile = strcpy(fullFile, pymol_data);
   fullFile = strcat(fullFile, shader_path);
   fullFile = strcat(fullFile, fileName);
 
@@ -416,7 +416,7 @@ char * CShaderMgr_ReadShaderFromDisk(PyMOLGlobals * G, const char * fileName) {
 
   if (!f) {
     PRINTFB(G, FB_ShaderMgr, FB_Errors)
-      " PyMOLShader_NewFromFile-Error: Unable to open file '%s' PYMOL_PATH='%s'\n", fullFile, pymol_path ENDFB(G);
+      " PyMOLShader_NewFromFile-Error: Unable to open file '%s' PYMOL_DATA='%s'\n", fullFile, pymol_data ENDFB(G);
     return NULL;
   } else {
     PRINTFB(G, FB_ShaderMgr, FB_Blather)
diff --git a/layer1/Setting.c b/layer1/Setting.c
index 3aa0b8d..f9b5789 100644
--- a/layer1/Setting.c
+++ b/layer1/Setting.c
@@ -3956,7 +3956,7 @@ void SettingInitGlobal(PyMOLGlobals * G, int alloc, int reset_gui, int use_defau
     set_b(I, cSetting_line_use_shader, 1);
     set_b(I, cSetting_sphere_use_shader, 1);
     set_b(I, cSetting_use_shaders, 0);  /* disable by default until optimized shaders present; doesn't effect vol */
-    set_s(I, cSetting_shader_path, "data/shaders");
+    set_s(I, cSetting_shader_path, "shaders");
     set_i(I, cSetting_volume_bit_depth, 8);
     set_color(I, cSetting_volume_color, "-1");
     set_f(I, cSetting_volume_layers, 256);
