diff -arNu -uarN pymol/modules/pymol/commanding.py pymol.new/modules/pymol/commanding.py
--- pymol/modules/pymol/commanding.py	2009-03-31 12:43:14.585834066 +0200
+++ pymol.new/modules/pymol/commanding.py	2009-03-31 12:44:26.436354330 +0200
@@ -219,11 +219,11 @@
                 _self.unlock(0,_self)
             r = DEFAULT_SUCCESS
             if show_splash==1: # generic / open-source
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/splash.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/splash.png")
             elif show_splash==2: # evaluation builds
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/epymol.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/epymol.png")
             else: # incentive builds
-                png_path = _self.exp_path("$PYMOL_PATH/data/pymol/ipymol.png")
+                png_path = _self.exp_path("$PYMOL_DATA/pymol/ipymol.png")
             if os.path.exists(png_path):
                 _self.do("_ cmd.load_png('%s',0,quiet=1)"%png_path)
         else:
diff -arNu -uarN pymol/modules/pymol/importing.py pymol.new/modules/pymol/importing.py
--- pymol/modules/pymol/importing.py	2009-03-31 12:43:14.546250540 +0200
+++ pymol.new/modules/pymol/importing.py	2009-03-31 12:44:48.876657949 +0200
@@ -191,7 +191,7 @@
     '''
         r = DEFAULT_ERROR
         
-        tables = { 'cmyk' : "$PYMOL_PATH/data/pymol/cmyk.png",
+        tables = { 'cmyk' : "$PYMOL_DATA/pymol/cmyk.png",
                    'pymol' : 'pymol',
                    'rgb' : 'rgb' }
         
