 setup.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index ce0d271..2b6fa35 100755
--- a/setup.py
+++ b/setup.py
@@ -104,7 +104,7 @@ elif sys.platform=='darwin':
         inc_dirs=["ov/src",
                   "layer0","layer1","layer2",
                   "layer3","layer4","layer5", 
-                  "/usr/X11R6/include",
+                  "@GENTOO_PORTAGE_EPREFIX@/usr/include",
                   EXT+"/include",
                   EXT+"/include/GL",
                   EXT+"/include/freetype2",
@@ -120,7 +120,7 @@ elif sys.platform=='darwin':
                     ]
         ext_comp_args=[]
         ext_link_args=[
-            "-L/usr/X11R6/lib64", "-lGL", "-lXxf86vm",
+            "-L@GENTOO_PORTAGE_EPREFIX@/usr/lib64", "-lGL", "-lXxf86vm",
             "-L"+EXT+"/lib", "-lpng", "-lglut", "-lfreetype"
             ]
 #============================================================================
@@ -159,7 +159,7 @@ else: # linux or other unix
                  "layer3",
                  "layer4",
                  "layer5",
-                 "/usr/include/freetype2",
+                 "@GENTOO_PORTAGE_EPREFIX@/usr/include/freetype2",
               # VMD plugin support
                             "contrib/uiuc/plugins/include",
                             "contrib/uiuc/plugins/molfile_plugin/src",
@@ -180,7 +180,7 @@ else: # linux or other unix
                    "GLU",
                    "glut",
                    "GLEW"]
-    lib_dirs = [ "/usr/X11R6/lib64", ]
+    lib_dirs = [ "@GENTOO_PORTAGE_EPREFIX@/usr/lib64", ]
     def_macros = [ ("_PYMOL_MODULE",None),
                    ("_PYMOL_INLINE",None),
                    ("_PYMOL_FREETYPE",None),
