Index: setup.py
===================================================================
--- setup.py	(revision 3983)
+++ setup.py	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@GENTOO_PORTAGE_EPREFIX@/usr/bin/env python
 #
 # This script only applies if you are performing a Python Distutils-based
 # installation of PyMOL.
@@ -35,7 +35,7 @@
 	      "modules/cealign/src/tnt" ]
     libs=["glut32","opengl32","glu32","png"]
     pyogl_libs = ["glut32","opengl32","glu32"]
-    lib_dirs=["/usr/lib/w32api"]
+    lib_dirs=["@GENTOO_PORTAGE_EPREFIX@/usr/lib/w32api"]
     def_macros=[("_PYMOL_MODULE",None),
                 ("CYGWIN",None),
                 ("_PYMOL_LIBPNG",None)]
@@ -87,7 +87,7 @@
                   "/sw/include/freetype2/freetype",
                   "/sw/include/freetype2",
                   "/sw/include",
-                  "/usr/X11/include",
+                  "@GENTOO_PORTAGE_EPREFIX@/usr/X11/include",
 		  "modules/cealign/src",
 		  "modules/cealign/src/tnt",
 		  #"contrib/uiuc/plugins/include/",
@@ -114,8 +114,8 @@
         ext_comp_args=[]
         ext_link_args=[
                        "-L/sw/lib", "-lpng",
-                       "/usr/X11/lib/libGL.dylib",
-                       "/usr/X11/lib/libGLU.dylib",
+                       "@GENTOO_PORTAGE_EPREFIX@/usr/X11/lib/libGL.dylib",
+                       "@GENTOO_PORTAGE_EPREFIX@/usr/X11/lib/libGLU.dylib",
                        "-lfreeglut",
                        "-lglew",
                        "-L/sw/lib/freetype219/lib", "-lfreetype"
@@ -221,7 +221,7 @@
                  "layer3",
                  "layer4",
                  "layer5",
-                 "/usr/include/freetype2",
+                 "@GENTOO_PORTAGE_EPREFIX@/usr/include/freetype2",
               # VMD plugin support
               #              "contrib/uiuc/plugins/include",
               #              "contrib/uiuc/plugins/molfile_plugin/src",
@@ -242,7 +242,7 @@
                    "GLU",
                    "glut",
                    "GLEW"]
-    lib_dirs = [ "/usr/X11R6/lib64", ]
+    lib_dirs = [ "@GENTOO_PORTAGE_EPREFIX@/usr/X11R6/lib64", ]
     def_macros = [ ("_PYMOL_MODULE",None),
                    ("_PYMOL_INLINE",None),
                    ("_PYMOL_FREETYPE",None),
