 pymol/setup.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/pymol/setup.py b/pymol/setup.py
index b8d7070..7fc11fb 100644
--- a/pymol/setup.py
+++ b/pymol/setup.py
@@ -252,10 +252,7 @@ else: # linux or other unix
                    ("_PYMOL_GL_CALLLISTS",None),
                    ("OPENGL_ES_2",None),
                    ]
-    ext_comp_args = [ "-ffast-math",
-                      "-funroll-loops",
-                      "-O3",
-                      "-g" ]
+    ext_comp_args = []
 
 def get_pymol_version():
     return re.findall(r'_PyMOL_VERSION "(.*)"', open('layer0/Version.h').read())[0]
