diff --git a/doc/Makefile.in b/doc/Makefile.in
index b1a61d5..c5e8546 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -113,7 +113,7 @@ help:
 clean:
 	rm -rf $(BUILDDIR)/*
 
-html:
+html: man
 	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
@@ -316,4 +316,4 @@ distdir: $(DISTFILES)
 	    || cp -p $$d/$$file "$(distdir)/$$file" \
 	    || exit 1; \
 	  fi; \
-	done
\ No newline at end of file
+	done
