--- configure.ac.ori	2018-04-22 12:19:37.318794996 +0200
+++ configure.ac	2018-04-22 12:20:43.310663598 +0200
@@ -172,8 +172,8 @@
 
 
 AC_CONFIG_HEADERS([config.h])
-AC_CONFIG_FILES([Makefile doc/Makefile doc/source/conf.py lib/kat.pc lib/Makefile src/Makefile tests/Makefile tests/compat.sh deps/seqan-library-2.0.0/Makefile scripts/setup.py scripts/kat/__init__.py scripts/Makefile])
-AC_CONFIG_SUBDIRS([deps/jellyfish-2.2.0])
+AC_CONFIG_FILES([Makefile doc/Makefile doc/source/conf.py lib/kat.pc lib/Makefile src/Makefile tests/Makefile tests/compat.sh scripts/setup.py scripts/kat/__init__.py scripts/Makefile])
+AC_CONFIG_SUBDIRS([])
 AC_OUTPUT
 
 pystr="detected ${PYTHON_VERSION}.  Plotting and downstream analysis enabled."
--- Makefile.am.ori	2018-04-22 12:23:58.756197763 +0200
+++ Makefile.am	2018-04-22 12:24:21.176832618 +0200
@@ -19,7 +19,7 @@
 dist_noinst_SCRIPTS = autogen.sh antigen.sh build_boost.sh
 
 # SRC DIRS
-make_dirs=deps/jellyfish-2.2.0 deps/seqan-library-2.0.0 lib src tests
+make_dirs=lib src tests
 
 if MAKE_PYKAT
     make_dirs += scripts
--- tests/Makefile.am.ori	2018-04-22 12:31:02.738203085 +0200
+++ tests/Makefile.am	2018-04-22 12:34:15.043648336 +0200
@@ -29,8 +29,7 @@
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/include \
-	-I$(top_srcdir)/deps/seqan-library-2.0.0/include \
-	-I$(top_srcdir)/deps/jellyfish-2.2.0/include \
+	-I$(top_srcdir)/usr/include/seqan-2.2 \
 	-I$(top_srcdir)/deps/boost/build/include \
 	-DDATADIR=\"$(srcdir)/data\" \
 	-DKAT_SITE_PKGS='"$(libdir)/python$(PYTHON_VERSION)/site-packages"' \
@@ -60,7 +59,6 @@
 	-static \
 	-L. \
 	-L../lib \
-	-L../deps/jellyfish-2.2.0 \
 	-L../deps/boost/build/lib \
 	@AM_LDFLAGS@
 
@@ -69,11 +67,11 @@
 	-lgtest \
 	-lkat \
 	-lkat_jellyfish \
-	../deps/boost/build/lib/libboost_timer.a \
-    ../deps/boost/build/lib/libboost_chrono.a \
-    ../deps/boost/build/lib/libboost_filesystem.a \
-    ../deps/boost/build/lib/libboost_program_options.a \
-    ../deps/boost/build/lib/libboost_system.a \
+	-lboost_timer \
+	-lboost_chrono \
+	-lboost_filesystem \
+	-lboost_program_options \
+	-lboost_system \
 	@AM_LIBS@
 
 include gtest.mk
--- src/Makefile.am.ori	2018-04-22 12:34:41.874408065 +0200
+++ src/Makefile.am	2018-04-22 12:36:33.567570728 +0200
@@ -12,9 +12,7 @@
 	@AM_CXXFLAGS@
 
 kat_CPPFLAGS = \
-	-I$(top_srcdir)/deps/seqan-library-2.0.0/include \
-	-I$(top_srcdir)/deps/jellyfish-2.2.0/include \
-	-I$(top_srcdir)/deps/boost/build/include \
+	-I/usr/include/seqan-2.2 \
 	-I$(top_srcdir)/lib/include \
 	-DKAT_SITE_PKGS='"$(libdir)/python$(PYTHON_VERSION)/site-packages"' \
 	@AM_CPPFLAGS@
@@ -22,18 +20,16 @@
 kat_LDFLAGS = \
 	-static \
 	-L../lib \
-	-L../deps/jellyfish-2.2.0 \
-	-L../deps/boost/build/lib \
 	@AM_LDFLAGS@
 
 kat_LDADD = \
 	-lkat \
 	-lkat_jellyfish \
-	../deps/boost/build/lib/libboost_timer.a \
-	../deps/boost/build/lib/libboost_chrono.a \
-	../deps/boost/build/lib/libboost_filesystem.a \
-	../deps/boost/build/lib/libboost_program_options.a \
-	../deps/boost/build/lib/libboost_system.a \
+	-lboost_timer \
+	-lboost_chrono \
+	-lboost_filesystem \
+	-lboost_program_options \
+	-lboost_system \
 	@AM_LIBS@
 
 noinst_HEADERS = \
--- lib/Makefile.am.ori	2018-04-22 12:38:29.680858551 +0200
+++ lib/Makefile.am	2018-04-22 12:38:58.401671799 +0200
@@ -6,9 +6,7 @@
 lib_LTLIBRARIES = libkat.la
 
 libkat_la_LDFLAGS = \
-	-version-info 2:4:0 \
-	-L../deps/jellyfish-2.2.0 \
-	-L../deps/boost/build/lib
+	-version-info 2:4:0
 
 libkat_la_SOURCES = \
 	src/matrix_metadata_extractor.cc \
@@ -30,8 +28,6 @@
 			    $(KI)/comp_counters.hpp
 
 libkat_la_CPPFLAGS = \
-	-I$(top_srcdir)/deps/jellyfish-2.2.0/include \
-	-I$(top_srcdir)/deps/boost/build/include \
 	-I$(top_srcdir)/lib/include \
 	-DKAT_SITE_PKGS='"$(libdir)/python$(PYTHON_VERSION)/site-packages"' \
 	@AM_CPPFLAGS@
