--- lib/include/kat/jellyfish_helper.hpp.ori	2018-04-22 12:47:42.296506217 +0200
+++ lib/include/kat/jellyfish_helper.hpp	2018-04-22 12:49:22.209335310 +0200
@@ -41,18 +41,18 @@
 using boost::lexical_cast;
 using boost::timer::auto_cpu_timer;
 
-#include <jellyfish/err.hpp>
-#include <jellyfish/file_header.hpp>
-#include <jellyfish/hash_counter.hpp>
-#include <jellyfish/mapped_file.hpp>
-#include <jellyfish/mer_dna.hpp>
-#include <jellyfish/jellyfish.hpp>
-#include <jellyfish/large_hash_array.hpp>
-#include <jellyfish/large_hash_iterator.hpp>
-#include <jellyfish/mer_iterator.hpp>
-#include <jellyfish/mer_overlap_sequence_parser.hpp>
-#include <jellyfish/storage.hpp>
-#include <jellyfish/stream_manager.hpp>
+#include <jellyfish2/err.hpp>
+#include <jellyfish2/file_header.hpp>
+#include <jellyfish2/hash_counter.hpp>
+#include <jellyfish2/mapped_file.hpp>
+#include <jellyfish2/mer_dna.hpp>
+#include <jellyfish2/jellyfish.hpp>
+#include <jellyfish2/large_hash_array.hpp>
+#include <jellyfish2/large_hash_iterator.hpp>
+#include <jellyfish2/mer_iterator.hpp>
+#include <jellyfish2/mer_overlap_sequence_parser.hpp>
+#include <jellyfish2/storage.hpp>
+#include <jellyfish2/stream_manager.hpp>
 using jellyfish::mer_dna;
 using jellyfish::file_header;
 using jellyfish::mapped_file;
--- lib/src/jellyfish_helper.cc.ori	2018-04-22 12:47:50.166729065 +0200
+++ lib/src/jellyfish_helper.cc	2018-04-22 12:49:28.379510022 +0200
@@ -35,15 +35,15 @@
 using boost::lexical_cast;
 using boost::timer::auto_cpu_timer;
 
-#include <jellyfish/binary_dumper.hpp>
-#include <jellyfish/err.hpp>
-#include <jellyfish/file_header.hpp>
-#include <jellyfish/mapped_file.hpp>
-#include <jellyfish/mer_dna.hpp>
-#include <jellyfish/jellyfish.hpp>
-#include <jellyfish/large_hash_array.hpp>
-#include <jellyfish/large_hash_iterator.hpp>
-#include <jellyfish/storage.hpp>
+#include <jellyfish2/binary_dumper.hpp>
+#include <jellyfish2/err.hpp>
+#include <jellyfish2/file_header.hpp>
+#include <jellyfish2/mapped_file.hpp>
+#include <jellyfish2/mer_dna.hpp>
+#include <jellyfish2/jellyfish.hpp>
+#include <jellyfish2/large_hash_array.hpp>
+#include <jellyfish2/large_hash_iterator.hpp>
+#include <jellyfish2/storage.hpp>
 using jellyfish::large_hash::reprobe_limit_t;
 using jellyfish::Offsets;
 using jellyfish::quadratic_reprobes;
--- src/sect.cc.ori	2018-04-22 12:47:57.906948236 +0200
+++ src/sect.cc	2018-04-22 12:49:33.479654437 +0200
@@ -53,7 +53,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/jellyfish_helper.hpp>
 #include <kat/matrix_metadata_extractor.hpp>
--- src/histogram.cc.ori	2018-04-22 12:48:06.427189490 +0200
+++ src/histogram.cc	2018-04-22 12:49:37.719774498 +0200
@@ -44,7 +44,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/matrix_metadata_extractor.hpp>
 #include <kat/jellyfish_helper.hpp>
--- src/sect.hpp.ori	2018-04-22 12:48:13.167380343 +0200
+++ src/sect.hpp	2018-04-22 12:49:41.549882951 +0200
@@ -45,7 +45,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/matrix_metadata_extractor.hpp>
 #include <kat/jellyfish_helper.hpp>
--- src/gcp.cc.ori	2018-04-22 12:48:20.457586772 +0200
+++ src/gcp.cc	2018-04-22 12:49:45.469993952 +0200
@@ -44,7 +44,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/jellyfish_helper.hpp>
 #include <kat/sparse_matrix.hpp>
--- src/comp.cc.ori	2018-04-22 12:48:28.447813020 +0200
+++ src/comp.cc	2018-04-22 12:49:50.720142612 +0200
@@ -48,7 +48,7 @@
 namespace bfs = boost::filesystem;
 using bfs::path;
 
-#include <jellyfish/large_hash_iterator.hpp>
+#include <jellyfish2/large_hash_iterator.hpp>
 
 #include <kat/matrix_metadata_extractor.hpp>
 #include <kat/sparse_matrix.hpp>
--- src/cold.cc.ori	2018-04-22 12:27:18.151843779 +0200
+++ src/cold.cc	2018-04-22 12:27:35.482334502 +0200
@@ -53,7 +53,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/jellyfish_helper.hpp>
 #include <kat/matrix_metadata_extractor.hpp>
--- src/comp.hpp.ori	2018-04-22 12:27:53.022831172 +0200
+++ src/comp.hpp	2018-04-22 12:28:10.173316799 +0200
@@ -33,7 +33,7 @@
 namespace bfs = boost::filesystem;
 using bfs::path;
 
-#include <jellyfish/large_hash_iterator.hpp>
+#include <jellyfish2/large_hash_iterator.hpp>
 
 #include <kat/matrix_metadata_extractor.hpp>
 #include <kat/sparse_matrix.hpp>
--- src/cold.hpp.ori	2018-04-22 12:29:23.725399473 +0200
+++ src/cold.hpp	2018-04-22 12:29:33.885687168 +0200
@@ -45,7 +45,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/matrix_metadata_extractor.hpp>
 #include <kat/jellyfish_helper.hpp>
--- src/gcp.hpp.ori	2018-04-22 12:30:06.576612834 +0200
+++ src/gcp.hpp	2018-04-22 12:30:13.056796324 +0200
@@ -36,7 +36,7 @@
 using bfs::path;
 using boost::lexical_cast;
 
-#include <jellyfish/mer_dna.hpp>
+#include <jellyfish2/mer_dna.hpp>
 
 #include <kat/pyhelper.hpp>
 #include <kat/jellyfish_helper.hpp>
