diff --git a/configure.ac b/configure.ac
index b68c514..00c8a2e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -159,11 +159,11 @@ if [[ "${make_pykat}" == "yes" ]]; then
     AM_CPPFLAGS="${ZLIB_CPPFLAGS} ${PYTHON_CPPFLAGS}"
     AM_LIBS="${PTHREAD_CFLAGS} ${ZLIB_LIB} ${RT_LIB} ${PYTHON_EXTRA_LIBS} ${PYTHON_LIBS}"
     #AM_LDFLAGS="${ZLIB_LDFLAGS} ${PYTHON_EXTRA_LDFLAGS} ${PYTHON_RPATH} -static-libstdc++"
-    AM_LDFLAGS="${ZLIB_LDFLAGS} -static-libstdc++ ${PYTHON_RPATH}"
+    AM_LDFLAGS="${ZLIB_LDFLAGS} ${PYTHON_RPATH}"
 else
     AM_CPPFLAGS="${ZLIB_CPPFLAGS}"
     AM_LIBS="${PTHREAD_CFLAGS} ${ZLIB_LIB} ${RT_LIB}"
-    AM_LDFLAGS="${ZLIB_LDFLAGS} -static-libstdc++"
+    AM_LDFLAGS="${ZLIB_LDFLAGS}"
 fi
 
 AC_SUBST([AM_CPPFLAGS])
diff --git a/deps/jellyfish-2.2.0/Makefile.am b/deps/jellyfish-2.2.0/Makefile.am
index 8f6fffd..d6e0a3d 100644
--- a/deps/jellyfish-2.2.0/Makefile.am
+++ b/deps/jellyfish-2.2.0/Makefile.am
@@ -7,7 +7,7 @@ man1_MANS = doc/kat_jellyfish.man
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = kat_jellyfish.pc
 
-AM_LDFLAGS = $(ZLIB_LDFLAGS) $(ZLIB_LIB) -lpthread -static-libstdc++ # $(VALGRIND_LIBS)
+AM_LDFLAGS = $(ZLIB_LDFLAGS) $(ZLIB_LIB) -lpthread # $(VALGRIND_LIBS)
 AM_CPPFLAGS = $(ZLIB_CPPFLAGS) -I$(top_srcdir) -I$(top_srcdir)/include # $(VALGRIND_CFLAGS)
 AM_CXXFLAGS = $(ALL_CXXFLAGS) -Wall -Wnon-virtual-dtor -Wno-deprecated-declarations
 
diff --git a/src/Makefile.am b/src/Makefile.am
index a2ae071..71def5c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,7 +19,6 @@ kat_CPPFLAGS = \
 	@AM_CPPFLAGS@
 
 kat_LDFLAGS = \
-	-static \
 	@AM_LDFLAGS@
 
 kat_LDADD = \
