diff --git a/ez-viz.py b/ez-viz.py
index 1d8e847..8d50e6b 100755
--- a/ez-viz.py
+++ b/ez-viz.py
@@ -363,27 +363,27 @@ class PGUI:
     global splash
     
     # Images used in Presets tab
-    defaultImage = PhotoImage( file="modules\pmg_tk\startup\default.gif")
-    chains = PhotoImage(file="modules\pmg_tk\startup\chains.GIF")
-    heteroImg = PhotoImage(file="modules\pmg_tk\startup\heteroAtom.GIF")
-    ballStickImg = PhotoImage(file="C:\Program Files\DeLano Scientific\PyMOL\modules\pmg_tk\startup\showBallStick.GIF")
-    surfaceImg = PhotoImage(file="modules\pmg_tk\startup\surface.GIF")
-    polarityImg = PhotoImage(file="modules\pmg_tk\startup\polarity.GIF")
-    puttyImg = PhotoImage(file="modules\pmg_tk\startup\putty.GIF")
-    aromaticsImg = PhotoImage(file="modules\pmg_tk\startup\showAromatics.GIF")
-    dnaImg = PhotoImage(file="modules\pmg_tk\startup\dna.GIF")
+    defaultImage = PhotoImage( file="GENTOOPYMOL/pmg_tk/startup/ez-viz/default.gif")
+    chains = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/chains.gif")
+    heteroImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/heteroAtom.gif")
+    ballStickImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/showBallStick.gif")
+    surfaceImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/surface.gif")
+    polarityImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/polarity.gif")
+    puttyImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/putty.gif")
+    aromaticsImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/showAromatics.gif")
+    dnaImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/dna.gif")
     
     # Movie tab images
-    ligand = PhotoImage(file="modules\pmg_tk\startup\zoom.GIF")
-    build = PhotoImage(file="modules\pmg_tk\startup\prot.GIF")
-    chainsMovieImg = PhotoImage(file="modules\pmg_tk\startup\chainsMovie.GIF")
-    rotateImg = PhotoImage(file="modules\pmg_tk\startup\Rotate.GIF")
-    playButton = PhotoImage(file="modules\pmg_tk\startup\play.GIF")
-    stopButton = PhotoImage(file="modules\pmg_tk\startup\stop.GIF")
-    rewindButton = PhotoImage(file="modules\pmg_tk\startup\Rewind.GIF")
+    ligand = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/zoom.gif")
+    build = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/prot.gif")
+    chainsMovieImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/chainsMovie.gif")
+    rotateImg = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/Rotate.gif")
+    playButton = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/play.gif")
+    stopButton = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/stop.gif")
+    rewindButton = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/Rewind.gif")
     
     # Welcome tab image
-    splash = PhotoImage(file="modules\pmg_tk\startup\splash.GIF")
+    splash = PhotoImage(file="GENTOOPYMOL/pmg_tk/startup/ez-viz/splash.gif")
     #--------------------------------------#
     #					   #
     #         PRESETS METHODS              #
@@ -1035,7 +1035,7 @@ class PGUI:
 	if tkMessageBox.askyesno('Ray Trace Image', "Do you want to ray trace you image before saving it? Depending on your image "+
 					"this make take a few minutes if you select yes."):
 		cmd.ray()    	
-    	file = tkFileDialog.asksaveasfilename(defaultextension=".png", initialdir="C:/Program Files/DeLano Scientific/PyMOL/")
+    	file = tkFileDialog.asksaveasfilename(defaultextension=".png", initialdir="./")
     	if len(file)>0:
     		cmd.save(file)
     	
@@ -1045,7 +1045,7 @@ class PGUI:
     	import tkFileDialog
     	import tkMessageBox 
     	
-    	file = tkFileDialog.asksaveasfilename(defaultextension=".pse", initialdir='C:/Program Files/Delano Scientific/PyMOL/')
+    	file = tkFileDialog.asksaveasfilename(defaultextension=".pse", initialdir='./')
     	print file
         if len(file)>0:
     		cmd.save(file)      
@@ -1574,7 +1574,7 @@ class PGUI:
     #------Open Help File---------#
     def help(self):
     	import webbrowser
-    	webbrowser.open('modules\pmg_tk\startup\Help\EZ-VizWebMain.html')
+    	webbrowser.open('GENTOOPYMOL/pmg_tk/startup/ez-vizHelp\EZ-VizWebMain.html')
     			
     #---------------------------------------------------------------#
     #								    #
@@ -2188,7 +2188,7 @@ class PGUI:
         import tkFileDialog
     
         if result == 'Open':
-            file=tkFileDialog.askopenfilename(initialdir='C:/Documents and Settings/Administrator/My Documents/')
+            file=tkFileDialog.askopenfilename(initialdir='./')
             if file:
             	cmd.load(file)
             	self.pdbCode = file
