diff -Naur nfft-3.1.3.orig/Makefile.am nfft-3.1.3/Makefile.am
--- nfft-3.1.3.orig/Makefile.am	2009-12-23 11:57:40.000000000 +0100
+++ nfft-3.1.3/Makefile.am	2010-09-12 20:46:27.000000000 +0200
@@ -31,8 +27,8 @@
 EXTRA_DIST = bootstrap.sh VERSION doxygen.c
 
 install-data-hook:
-	mkdir -p $(pkgdatadir)
-	cp -R doc $(pkgdatadir)
+	mkdir -p $(DESTDIR)$(pkgdatadir)
+	cp -R doc $(DESTDIR)$(pkgdatadir)
 
 maintainer-clean-local:
 	rm -f doc/api/html/*.html
