--- perl/est_prepro.pm.ori	2011-02-23 10:56:06.000000000 +0100
+++ perl/est_prepro.pm	2011-02-23 19:53:13.000000000 +0100
@@ -499,15 +499,17 @@
 					$sth->bind_param(5,$unique_seq_control);
 					# we get the clone names from the data file
 					# if not in the data file, we guess it following user specifications in config file
-					unless (${%$clone}{$real_name}) {
-						${%$clone}{$real_name}=get_clone_from_est($config,$real_name);
-					}
-					$sth->bind_param(6,${%$clone}{$real_name});
-					$$seq_info{clone}=${%$clone}{$real_name};
-					if (!${%$clone}{$real_name}){
-						print localtime().": Unable to get clone name for EST $real_name\n";
-						return undef;
-					}
+					#unless (${%$clone}{$real_name}) {
+					#	${%$clone}{$real_name}=get_clone_from_est($config,$real_name);
+					#}
+					#$sth->bind_param(6,${%$clone}{$real_name});
+					#$$seq_info{clone}=${%$clone}{$real_name};
+					#if (!${%$clone}{$real_name}){
+					#	print localtime().": Unable to get clone name for EST $real_name\n";
+					#	return undef;
+					#}
+					$sth->bind_param(6,'blah');
+					$$seq_info{clone}='blah';
 				}
 				if (not $qual and not ($unique_seq_control eq 'raw_est' or $unique_seq_control eq 'clean_est')) {
 					$sth->bind_param(5,$unique_seq_control)
