--- perl/pipe_db.pm	2011-02-23 18:38:47.000000000 +0100
+++ pipe_db.pm	2011-02-23 19:41:06.000000000 +0100
@@ -1468,10 +1468,11 @@
 	my $tables=shift;
 	my $ok=1;
 
+	my %t = %$tables; # Can't use string ("38/64") as a HASH ref while "strict refs" in use at /opt/est2uni/pipe_db.pm line 1474.
 	my $table;
 	foreach $table(keys %$tables) {
 		if ($ok) {
-			$ok=create_table($db,$table,${%{$tables}}{$table});
+			$ok=create_table($db,$table,$t{$table});
 			if ($ok) {
 				print localtime().": Table $table created\n";
 			} else {
