--- est2uni/configuration.php.example   2011-02-19 02:05:44.000000000 +0100
+++ est2uni/configuration.php   2011-12-12 16:07:15.000000000 +0100
@@ -16,7 +16,7 @@
        $config['web_master_email']="admin@estpipe.org";
 
        # base directory for web pages at this Apache installation
-       $config['apache_dir']="/var/www";
+       $config['apache_dir']="/var/www/localhost/htdocs/est2uni";
 
        # directory name for this group of EST projects
        $config['group']= "";

