--- est_annot_psnp.pm.ori	2011-02-22 20:12:25.000000000 +0100
+++ est_annot_psnp.pm	2011-02-22 20:12:46.000000000 +0100
@@ -773,7 +773,7 @@
 	}
 
 	my $dbh=open_db($db) or return 0;
-	my $st_psnp="INSERT INTO $table_psnp (psnp_id,unigene,location,is_snp,is_indel,score,enzymes) VALUES (?,?,?,?,?,?,'');";
+	my $st_psnp="INSERT INTO $table_psnp (psnp_id,unigene,location,is_snp,is_indel,score,enzymes) VALUES (?,?,?,?,?,?,NULL);";
 	my $sth_psnp=$dbh->prepare($st_psnp);
 	my $st_seq="INSERT INTO $table_seq (psnp_id,nucleotide,est,accession) VALUES (?,?,?,?);";
 	my $sth_seq=$dbh->prepare($st_seq);
