--- est_annot_go.pm.ori	2011-02-22 15:33:33.000000000 +0100
+++ est_annot_go.pm	2011-02-23 00:12:48.000000000 +0100
@@ -213,13 +213,13 @@
 sub annotate_go_from_blast($$){
 	my $config=shift;
 	my $db_info=shift;
-	print localtime().": Annotating the sequences with GO terms looking at the BLAST results.\n";
+	print localtime().": Annotating the sequences with GO terms from ".read_config($config,'go_assoc_table','')." looking at the BLAST results in ".read_config($config,'go_db_path','').".\n";
 
 	#the go-blast_db association is stored in a table on a database
 	my $go_db_path=read_config($config,'go_db_path','');
 	my $assoc_table=read_config($config,'go_assoc_table','');
 	my $go_dbh=open_db($go_db_path) or (
-		print localtime().": There was a problem opening the database $go_db_path\n" and
+		print localtime().": There was a problem opening the database $go_db_path: $DBI::errstr\n" and
 		return 0
 	);
 	
@@ -227,13 +227,13 @@
 	my $result_table=read_config($config,'go_table','');
 	my $est_db_path=read_config($config,'db_path','');
 	my $est_dbh=open_db($est_db_path) or (
-		print localtime().": There was a problem opening the database $est_db_path\n" and
+		print localtime().": There was a problem opening the database $est_db_path: $DBI::errstr\n" and
 		return 0
 	);
 	#we use ignore because we're interested just in the first annotation for the combined seq_go_slim, it could be more but they're not important
 	my $est_st="INSERT IGNORE INTO $result_table (db,db_symbol,qualifier,go_id,ori_evidence,evidence,aspect,taxon_id,e_value,is_slim,seq_id) VALUES (?,?,?,?,?,?,?,?,?,?,?);";
 	my $est_sth=prepare_st($est_dbh,$est_st) or (
-		print localtime().": There was a problem executing the statement $est_st\n" and
+		print localtime().": There was a problem executing the statement $est_st: $DBI::errstr\n" and
 		return 0
 	);
 	
@@ -255,7 +255,7 @@
 		my $real_table=$assoc_table.$go_db;
 		my $go_st="SELECT db,db_symbol,qualifier,go_id,evidence,aspect,taxon_id,is_slim FROM $real_table WHERE db_symbol=? ;";
 		my $go_sth=prepare_st($go_dbh,$go_st) or (
-			print localtime().": There was a problem executing the statement $go_st\n" and
+			print localtime().": There was a problem executing the statement $go_st: $DBI::errstr\n" and
 			return 0
 		);
 		unless ($go_db){
@@ -264,7 +264,7 @@
 		}
 		#we go through the blast files using the bioperl blast parser
 		my $in = new Bio::SearchIO(-format => 'blast',-file   => $blast_file) or(
-			print localtime().": There was a problem creating a bioperl handler for the BLAST file $blast_file\n" and
+			print localtime().": There was a problem creating a bioperl handler for the BLAST file $blast_file: $!\n" and
 			return 0
 		);
 
@@ -277,9 +277,12 @@
 				my $significative_hit=0;
 				my $query=$result->query_name;
 				my $hit=$result->next_hit;
-				unless ($hit){$done=1;last};	#if there isn't more hits we're finnished
+				unless ($hit){$done=1;last};	#if there isn't more hits we're finished
 				my $acc=$hit->name;
-				#the arabidopsis project puts an .number in the hit name that should be removed
+				if (! $acc ) {
+					print localtime().": No \$acc value parsed from \$hit->name\n";
+				}
+				#the arabidopsis project puts a .number in the hit name that should be removed
 				if (${$$db_info{$go_db}}{'orthologues_arabi_hack'}){ 
 				$acc =~ s/\.\d$//;
 				}
@@ -299,7 +302,7 @@
 					#is there go terms associated with this acc?
 					$go_sth->bind_param(1,$acc);
 					$go_sth->execute or (
-						print localtime().": There was a problem executing the statement $go_st with accession $acc\n" and
+						print localtime().": There was a problem executing the statement $go_st with accession $acc: $DBI::errstr\n" and
 						return 0
 					);
 					my $there_is_go=0;
@@ -319,7 +322,7 @@
 							$est_sth->bind_param(10,$$go[7]);	#is this a slim term?
 							$est_sth->bind_param(11,$query);	#annotated sequence
 							$est_sth->execute or (
-								print localtime().": There was a problem executing the statement $est_st\n" and
+								print localtime().": There was a problem executing the statement $est_st: $DBI::errstr\n" and
 								return 0
 							);
 							$done=1;		#we do not need any more hits
@@ -333,11 +336,12 @@
 			$i_uni++;
 		}
 		close_st ($go_sth);
+		print localtime().": $i_uni BLAST unigenes processed from $blast_file, $i_uni_with_go with GO annotations, and $i_go_assoc GO terms associated\n";
 	}
 	close_st ($est_sth);
 	close_db ($go_dbh);
 	close_db ($est_dbh);
-	print localtime().": $i_uni BLAST unigenes processed, $i_uni_with_go with GO annotations, and $i_go_assoc GO terms associated\n";
+	print localtime().": $i_uni BLAST unigenes processed from $blast_file_base and "."@go_dbs\n".", $i_uni_with_go with GO annotations, and $i_go_assoc GO terms associated\n";
 	return 1;
 }
 
@@ -367,7 +371,7 @@
 		my $assoc_file=${$$db_info{$db}}{'local_go_assoc_file'}; 
 		my $cmd="$bin $slim_file $go_def_file $assoc_file -o $temp_dir$base_out_file".$sep."$db";
 		not (system $cmd) or (
-			print localtime().": There was a problem executing: $cmd\n" and
+			print localtime().": There was a problem executing $cmd: $!\n" and
 			return 0
 		);
 		$i++;
@@ -447,7 +451,8 @@
 	my $table=read_config($config,'go_assoc_table','');
 	#now it's time to fill the table with the data
 	my $real_table=$table.$go_db;
-	my $st="INSERT INTO $real_table (db,db_symbol,qualifier,go_id,evidence,aspect,taxon_id,is_slim) VALUES (?,?,?,?,?,?,?,?);";
+	# this is a wastefull insert inserting qualifier, taxon_id, is_slim equal to NULL
+	my $st="INSERT DELAYED INTO $real_table (db,db_symbol,qualifier,go_id,evidence,aspect,taxon_id,is_slim) VALUES (?,?,?,?,?,?,?,?);";
 	my $dbh=open_db($db_path) or (
 		print localtime().": There was a problem opening the database $db_path\n" and
 		return 0
@@ -491,7 +496,7 @@
 				$sth->bind_param(5,$items[6]);	#evidence
 				$sth->bind_param(6,$items[8]);	#aspect
 				$sth->bind_param(7,$items[12]);	#taxon_id
-				$sth->bind_param(8,$is_slim);
+				$sth->bind_param(8,$is_slim);	#is_slim
 			}else{
 				print localtime().": Gene-GO association file has no defined format in $config go_assoc_file_format\n";
 				print localtime().": Or the format is not supported. Supported formats are goa and tair\n";
@@ -542,7 +547,7 @@
 	my $db=read_config($config,'db_path','');
 	
 	open IN,"<$file" or (
-		print localtime().": Imposible to open file: $file\n" and
+		print localtime().": Error opening file $file: $!\n" and
 		return 0
 	);
 	
