--- est_annot_reciprocal_blast.pm.ori	2011-02-22 01:19:43.000000000 +0100
+++ est_annot_reciprocal_blast.pm	2011-02-22 01:39:37.000000000 +0100
@@ -106,8 +106,9 @@
 	my @dbs=split (',',read_config($config,'reci_dbs',''));
 	foreach my $db (@dbs){
 		if (!$$db_info{$db}){
-			print localtime().": Database $db is not defined in db table on the databases\n";
-			return 0;
+			print localtime().": Database $db included in \$reci_dbs is not described in table db, skipping.\n";
+			#return 0;
+			next;
 		}
 		my $in_file=$unigenes_seq_file;
 		my $outf=$in_file.$separator.$db.$separator."blast";
