--- perl/est_clustering.pm.ori	2011-02-17 22:41:29.000000000 +0100
+++ perl/est_clustering.pm	2011-02-17 22:42:54.000000000 +0100
@@ -279,8 +279,11 @@
 	$command="cat ";
 	for ($i=1; $i<=$num_micros; $i++)	# a loop for every directory created by tgicl
 	{
+		if (-e $temp_dir.$asm_dir.$i."/".$tgicl_ace_file)
+		{
 		$command=$command.$temp_dir.$asm_dir.$i."/".$tgicl_ace_file." ";
 	}
+	}
 	$command.=" >".$output_ace_file;
 	$tgicl_success=not(system($command));
 	unless ($tgicl_success)
@@ -295,8 +298,11 @@
 	$command="cat ";
 	for ($i=1; $i<=$num_micros; $i++)	# a loop for every directory created by tgicl
 	{
+		if (-e $temp_dir.$asm_dir.$i."/".$tgicl_fasta_file)
+		{
 		$command=$command.$temp_dir.$asm_dir.$i."/".$tgicl_fasta_file." ";
 	}
+	}
 	$command.=" >".$output_contig_fasta_file;
 	$tgicl_success=not(system($command));
 	unless ($tgicl_success)
