--- cmake/gsInstall.cmake	2015-08-08 12:39:45.000000000 +0200
+++ cmake/gsInstall.cmake	2015-10-15 19:08:34.735776106 +0200
@@ -1,6 +1,15 @@
-## #################################################################
+######################################################################
+## gsIntall.cmake ---
+## This file is part of the G+Smo library.
+## 
+## Author: Angelos Mantzaflaris 
+## Author: Harald Weiner
+## Copyright (C) 2012-2015 - RICAM-Linz.
+######################################################################
+
+######################################################################
 ## Installation
-## #################################################################
+######################################################################
 
 message ("  CMAKE_INSTALL_PREFIX    ${CMAKE_INSTALL_PREFIX}")
 
@@ -128,3 +137,25 @@
    message ("Configure with -DGISMO_BUILD_LIB=ON to compile the library")
 endif(GISMO_BUILD_LIB)
 
+# Install docs (if available)
+set(DOC_SRC_DIR "${PROJECT_BINARY_DIR}/doc/html/")
+#message("DOC_SRC_DIR='${DOC_SRC_DIR}'")
+
+set(TMP_VERSION "${gismo_VERSION}")
+string(REGEX REPLACE "[a-zA-Z]+" "" TMP_VERSION ${TMP_VERSION})
+#message("TMP_VERSION='${TMP_VERSION}'")
+set(DOC_INSTALL_DIR share/doc/gismo-${TMP_VERSION} CACHE PATH 
+                       "Installation directory for documentation")
+#message("DOC_INSTALL_DIR='${DOC_INSTALL_DIR}'")
+
+install(DIRECTORY "${DOC_SRC_DIR}"
+               DESTINATION "${DOC_INSTALL_DIR}/"
+               USE_SOURCE_PERMISSIONS
+               OPTIONAL 
+               FILES_MATCHING
+                       PATTERN "*.css"
+                       PATTERN "*.html"
+                       PATTERN "*.js"
+                       PATTERN "*.jpg"
+                       PATTERN "*.png"
+)
