diff --git a/Examples/PrintHeader.cxx b/Examples/PrintHeader.cxx
index d1e7586..048d934 100644
--- a/Examples/PrintHeader.cxx
+++ b/Examples/PrintHeader.cxx
@@ -281,7 +281,7 @@ PrintHeader(int argc, char * argv[])
       // Get the metadata as a generic object
       string                                                   key = itMeta->first, v_string;
       itk::SpatialOrientation::ValidCoordinateOrientationFlags v_oflags =
-        itk::SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_INVALID;
+        itk::SpatialOrientation::ITK_COORDINATE_ORIENTATION_INVALID;
 
       if (itk::ExposeMetaData<string>(mdd, key, v_string))
       {
