--- ccp4-6.0.99e.orig/x-windows/Makefile.in	2008-07-31 10:36:21.000000000 -0700
+++ ccp4-6.0.99e/x-windows/Makefile.in	2008-07-31 12:25:36.000000000 -0700
@@ -35,7 +35,7 @@
 #
 # all
 #
-all : xdlview libjwc_c libjwc_f rotgen_ hklview xjiffy rasmol2 ipmosflm
+all : xdlview libjwc_c libjwc_f rotgen_ hklview xjiffy ipmosflm
 #
 # xdl_view
 #
@@ -157,9 +157,9 @@
 #
 install_obj = hklview rotgen ipmosflm
 install: all
+	mkdir -p $(bindir)
 	$(INSTALL_PROGRAM) $(srcdir)/hklview $(bindir)
 	cd $(srcdir)/XCCPJIFFY ; $(MAKE) install
-	$(INSTALL_PROGRAM) $(rasmol_dir)/rasmol $(bindir)
 	cd $(rotgen_dir); $(MAKE) install
 	$(INSTALL_PROGRAM) $(mosflm_dir)/bin/ipmosflm $(bindir)
 #
