From e5a6dd4bdc4db828ea1de2a48e96cbdb80465e86 Mon Sep 17 00:00:00 2001
From: "John T. Wodder II" <git@varonathe.org>
Date: Tue, 25 Jan 2022 18:53:53 -0500
Subject: [PATCH] Mark a test with missing `skipif_no_network` marker

---
 dandi/tests/test_metadata.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dandi/tests/test_metadata.py b/dandi/tests/test_metadata.py
index 5896e6ef..a0a6bce1 100644
--- a/dandi/tests/test_metadata.py
+++ b/dandi/tests/test_metadata.py
@@ -351,6 +351,7 @@ def test_parseobourl(url, value):
     assert parse_purlobourl(url) == value
 
 
+@mark.skipif_no_network
 def test_species():
     m = {"species": "http://purl.obolibrary.org/obo/NCBITaxon_28584"}
     assert extract_species(m).json_dict() == {
