--- src/conv/step/Makefile.in.orig	2010-02-02 19:37:34.000000000 +0000
+++ src/conv/step/Makefile.in	2010-02-02 19:38:15.000000000 +0000
@@ -894,6 +894,11 @@
 	VolumeSiUnit.h \
 	VolumeUnit.h 
 
+libstep2g_nil_la_LIBADD = \
+	${BU} \
+	${WDB} \
+	${RT} \
+	${STEP_LIBS}
 
 # FIXME: these declarations are required for proper distchecking if
 # the files do not exist before configure is run.
@@ -903,10 +908,6 @@
 
 step_g_SOURCES = 
 step_g_LDADD = \
-	${BU} \
-	${WDB} \
-	${RT} \
-	${STEP_LIBS} \
 	libstep2g_nil.la
 
 
