diff -r -U2 scidavis-0.1.3.orig/fitPlugins/fitRational0/fitRational0.pro scidavis-0.1.3/fitPlugins/fitRational0/fitRational0.pro
--- scidavis-0.1.3.orig/fitPlugins/fitRational0/fitRational0.pro	2008-02-26 01:32:02.000000000 +0600
+++ scidavis-0.1.3/fitPlugins/fitRational0/fitRational0.pro	2008-05-19 07:38:55.000000000 +0700
@@ -26,5 +26,6 @@
 # This is used as default on unix systems such as
 # Linux, Mac OS X and *BSD.
-unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 # where to install the plugins
diff -r -U2 scidavis-0.1.3.orig/fitPlugins/fitRational1/fitRational1.pro scidavis-0.1.3/fitPlugins/fitRational1/fitRational1.pro
--- scidavis-0.1.3.orig/fitPlugins/fitRational1/fitRational1.pro	2008-02-26 01:32:02.000000000 +0600
+++ scidavis-0.1.3/fitPlugins/fitRational1/fitRational1.pro	2008-05-19 07:39:24.000000000 +0700
@@ -26,5 +26,6 @@
 # This is used as default on unix systems such as
 # Linux, Mac OS X and *BSD.
-unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:INCLUDEPATH += /usr/include/gsl
+unix:LIBS += $$system(pkg-config --libs gsl)
 
 # where to install the plugins
