--- gs/time_.h~	Thu Apr  1 07:31:53 1999
+++ gs/time_.h	Wed Mar 14 17:11:12 2001
@@ -43,9 +43,7 @@
  */
 #ifdef HAVE_SYS_TIME_H
 #  include <sys/time.h>
-#  if defined(Plan9) || defined(M_UNIX) || defined(_IBMR2) || defined(_SEQUENT_)	/* Plan 9, SCO, AIX and Sequent's DYNIX/ptx need both time.h and sys/time.h! */
-#    include <time.h>
-#  endif
+#  include <time.h>
 #else
 #  include <time.h>
 #  ifndef __DECC
