--- DPS/clients/makepsres/makepsres.c.old	2004-04-22 19:48:50.568331512 +0100
+++ DPS/clients/makepsres/makepsres.c	2004-04-22 19:50:34.076595864 +0100
@@ -124,8 +124,6 @@
 
 char *program;
 
-extern char *malloc(), *realloc();
-
 #if !defined(__NetBSD__) && !defined(__FreeBSD__)
 # if !defined(__GLIBC__)
 extern char *sys_errlist[];
--- DPS/clients/pswrap/psw.c.old	2004-04-22 19:54:13.609221832 +0100
+++ DPS/clients/pswrap/psw.c	2004-04-22 19:54:39.149339144 +0100
@@ -1906,7 +1906,6 @@
 
 char *psw_malloc(s) int s; {
     char *temp;
-    extern char *malloc();
     if ((temp = malloc((unsigned) s)) == NULL)
         AllocFailure();
     return(temp);
@@ -1914,7 +1913,6 @@
 
 char *psw_calloc(n,s) int n,s; {
     char *temp;
-    extern char *calloc();
     if ((temp = calloc((unsigned) n, (unsigned) s)) == NULL)
         AllocFailure();
     return(temp);
