diff -urN dgs-0.5.10.ORIG/configure.in dgs-0.5.10/configure.in
--- dgs-0.5.10.ORIG/configure.in	2000-05-08 12:38:17.000000000 +0900
+++ dgs-0.5.10/configure.in	2004-04-16 02:14:40.029905480 +0900
@@ -163,9 +163,13 @@
 # TCP wrappers check from gnome-core
 #--------------------------------------------------------------------
 
+AC_ARG_ENABLE(tcpd,
+   [  --disable-tcpd           Disable tcpd support.],,
+   enable_tcpd=yes)
 AC_CHECK_HEADERS(tcpd.h)
 AC_CHECK_HEADERS(syslog.h)
 AC_MSG_CHECKING("whether to use TCP wrappers")
+if test "x$enable_tcpd" = "xyes"; then
 LIBWRAP_PATH=""
 for I in $LDFLAGS $LIBS -L/usr/lib -L/usr/local/lib; do
         case "$I" in
@@ -246,6 +250,10 @@
         AC_DEFINE(HAVE_HOSTS_ACCESS)
 fi
 AC_SUBST(LIBWRAP_LIBS)
+else
+	AC_MSG_RESULT(no)
+	enable_tcpd=no
+fi
 
 #--------------------------------------------------------------------
 # For now we assume that gcc will be used to compile ghostscript.
