diff --git a/python/Makefile.am b/python/Makefile.am
index 4a8b5d5..b976893 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -19,13 +19,14 @@ if WITH_PYTHON
 AM_CPPFLAGS = \
 	-I$(top_builddir)/include \
 	-I$(top_srcdir)/include	\
-	-I$(PYTHON_INCLUDES)
+	-I$(PYTHON_INCLUDES) \
+	-shared
 
 python_LTLIBRARIES = libxml2mod.la
 
 libxml2mod_la_SOURCES = libxml.c libxml_wrap.h libxml2-py.h libxml2-py.c types.c
 libxml2mod_la_LDFLAGS = $(CYGWIN_EXTRA_LDFLAGS) $(WIN32_EXTRA_LDFLAGS) -module -avoid-version \
-        $(top_builddir)/libxml2.la $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) $(PYTHON_LIBS)
+        $(top_builddir)/libxml2.la $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) $(PYTHON_LIBS) -shared
 
 BUILT_SOURCES = libxml2-export.c libxml2-py.h libxml2-py.c
 
