--- a/src/CMakeLists.txt	2024-02-16 21:36:58.808186061 +0100
+++ b/src/CMakeLists.txt	2024-02-16 21:36:55.288131610 +0100
@@ -8,7 +8,7 @@
   Message("Boost found" ${Boost_LIBRARY_DIR})
   INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
   #MESSAGE("Boost was found" ${Boost_INCLUDE_DIR})
-  TARGET_LINK_LIBRARIES (cipi boost_thread-mt)
+  TARGET_LINK_LIBRARIES (cipi boost_thread)
 ENDIF (Boost_FOUND)
 
 INSTALL(PROGRAMS cipi DESTINATION bin)
