--- Makefile.orig	2013-03-03 09:41:33.534481539 -0800
+++ Makefile	2013-03-03 10:00:21.008239835 -0800
@@ -26,7 +26,7 @@
 libopenlibm.a: $(OBJS)  
 	$(AR) -rcs libopenlibm.a $(OBJS)
 libopenlibm.$(SHLIB_EXT): $(OBJS)
-	$(FC) -shared $(OBJS) $(LDFLAGS) -o libopenlibm.$(SHLIB_EXT)
+	$(FC) -shared $(OBJS) $(LDFLAGS) -Wl,--soname=libopenlibm.$(SHLIB_EXT) -o libopenlibm.$(SHLIB_EXT)
 
 distclean:
 	rm -f $(OBJS) *.a *.$(SHLIB_EXT)
