
--- util/check_lin_tif.c.or   2007-10-17 12:53:19.000000000 +0300
+++ util/check_lin_tif.c   2007-10-17 13:02:29.000000000 +0300
@@ -251,7 +251,8 @@
         if (!force_bad &&
             ((XmVERSION == 2 && XmREVISION == 1) ||
              (XmVERSION == 2 && XmREVISION == 2 && XmUPDATE_LEVEL == 3 &&
-                 strcmp("@(#)Motif Version 2.2.3", vs) == 0))) {
+                 strcmp("@(#)Motif Version 2.2.3", vs) == 0) ||
+             (XmVERSION == 2 && XmREVISION >= 3))) {
             exit(0);
         }
  
