--- src5.1.3.orig/KERNEL_SRC_5.1.3/salome_adm/unix/config_files/check_mpi.m4	2008-12-10 10:14:34.000000000 +0100
+++ src5.1.3/KERNEL_SRC_5.1.3/salome_adm/unix/config_files/check_mpi.m4	2009-09-29 10:14:56.821299268 +0200
@@ -76,7 +76,7 @@
 
   if test "$WITHMPI" = "yes";then
     mpi_ok=yes
-    MPI_LIBS="$MPI_LIBS -lmpi -lmpio -lmpiCC"
+    MPI_LIBS="$MPI_LIBS -lmpi -lmpi_cxx"
   else
     mpi_ok=no
   fi
--- src5.1.3.orig/KERNEL_SRC_5.1.3/src/Communication/SALOME_Comm_i.hxx	2008-12-10 10:14:33.000000000 +0100
+++ src5.1.3/KERNEL_SRC_5.1.3/src/Communication/SALOME_Comm_i.hxx	2009-09-29 11:10:13.227293648 +0200
@@ -25,6 +25,7 @@
 #include <SALOME_Communication.hxx>
 
 #ifdef HAVE_MPI2
+#define OMPI_IGNORE_CXX_SEEK 1
 #include "mpi.h"
 #endif
 #include <string>
