--- salome_adm/unix/config_files/check_openpbs.m4_orig	2008-01-18 23:50:02.000000000 +0100
+++ salome_adm/unix/config_files/check_openpbs.m4	2008-01-18 23:50:57.000000000 +0100
@@ -32,15 +32,15 @@
  
    if test "x${OPENPBS}" = "x/usr"
    then
-     OPENPBS_LIBDIR=""
+     OPENPBS_LIBDIR="-L/usr/lib/pbs/lib"
    else
      OPENPBS_LIBDIR="-L${OPENPBS}/lib"
    fi
-   OPENPBS_LIBS="-lpbs"
+   OPENPBS_LIBS="-ltorque"
    saved_LDFLAGS=${LDFLAGS}
    saved_LIBS=${LIBS}
    LDFLAGS="${LDFLAGS} ${OPENPBS_LIBDIR}"
-   AC_CHECK_LIB([pbs], [pbs_connect], [openpbs_lib_found=yes], [openpbs_lib_found=no], [])
+   AC_CHECK_LIB([torque], [pbs_connect], [openpbs_lib_found=yes], [openpbs_lib_found=no], [])
    test x${openpbs_lib_found} = xno && AC_MSG_WARN([OpenPBS library not found])
    LIBS="${LIBS} ${OPENPBS_LIBS}"
    LDFLAGS=${saved_LDFLAGS}
