--- src3.2.6_orig/KERNEL_SRC_3.2.6/src/GenericObj/SALOME_GenericObj_i.cc	2008-08-22 14:53:38.000000000 +0200
+++ src3.2.6/KERNEL_SRC_3.2.6/src/GenericObj/SALOME_GenericObj_i.cc	2008-08-22 15:20:23.000000000 +0200
@@ -42,7 +42,7 @@
 	    "; CORBA::is_nil(thePOA) = "<<CORBA::is_nil(thePOA));
   if(CORBA::is_nil(thePOA))
 #ifndef WNT
-    myPOA = PortableServer::RefCountServantBase::_default_POA();
+    myPOA = PortableServer::ServantBase::_default_POA();
 #else
     myPOA = RefCountServantBase::_default_POA();
 #endif
--- src3.2.6_orig/KERNEL_SRC_3.2.6/src/Notification/NOTIFICATION_Consumer.cxx	2008-08-22 14:53:38.000000000 +0200
+++ src3.2.6/KERNEL_SRC_3.2.6/src/Notification/NOTIFICATION_Consumer.cxx	2008-08-22 15:22:54.000000000 +0200
@@ -93,14 +93,14 @@
         };
 
         if (status) {
-            char*        Asender;
-            char*        Agraph;
-            char*        Atype;
+            const char*        Asender;
+            const char*        Agraph;
+            const char*        Atype;
             CORBA::ULong Acounter;
             CORBA::ULong Astamp;
-            char*        Adate;
-            char*        Anode;
-            char*        Amessage;
+            const char*        Adate;
+            const char*        Anode;
+            const char*        Amessage;
             event->filterable_data[0].value >>= Asender;
             event->filterable_data[1].value >>= Agraph;
             event->filterable_data[2].value >>= Atype;
