--- src5.1.3.old/KERNEL_SRC_5.1.3/salome_adm/unix/config_files/ac_linker_options.m4	2009-07-10 19:08:57.000000000 +0200
+++ src5.1.3/KERNEL_SRC_5.1.3/salome_adm/unix/config_files/ac_linker_options.m4	2009-07-26 19:47:05.000000000 +0200
@@ -26,6 +26,10 @@
 
   AC_CHECKING(for LIB_LOCATION_SUFFIX)
   LIB_LOCATION_SUFFIX=""
+  case "$build_cpu" in
+  x86_64*)  LIB_LOCATION_SUFFIX="64" ;;
+  *)        LIB_LOCATION_SUFFIX="" ;;
+  esac
   AC_SUBST(LIB_LOCATION_SUFFIX)
   AC_MSG_RESULT(LIB_LOCATION_SUFFIX is $LIB_LOCATION_SUFFIX)
 
