--- lib/libxview/textsw/es_file.c	1997-04-28 14:45:37.000000000 +0200
+++ lib/libxview/textsw/es_file.c.new	2009-02-03 09:51:34.000000000 +0100
@@ -114,7 +114,7 @@
 
 extern int      errno, sys_nerr;
 extern char    *sys_errlist[];
-extern long     lseek();
+extern off_t     lseek();
 
 static void update_read_buf();  /* update the read buf if overlaps write buf */
 static Es_status es_file_commit();
--- lib/libxview/sel/sel_agent.c	1997-06-19 22:45:57.000000000 +0200
+++ lib/libxview/sel/sel_agent.c.new	2009-02-03 09:53:34.000000000 +0100
@@ -309,7 +309,7 @@
     struct stat     stat_buf;
     int             count, size;
     char           *destp;
-    extern long     lseek();
+    extern off_t     lseek();
 
     if (fstat(fd, &stat_buf) != 0) {
 	perror(XV_MSG("Agent couldn't reply about a file"));
