--- lib/libxview/textsw/es_file.c.old	2006-02-18 23:26:14.000000000 -0500
+++ lib/libxview/textsw/es_file.c	2006-02-18 23:26:37.000000000 -0500
@@ -119,9 +119,7 @@
 extern int      errno, sys_nerr;
 extern char    *sys_errlist[];
 #endif
-#ifndef __CYGWIN__
-extern long     lseek();
-#endif
+extern off_t     lseek();
 
 static void update_read_buf();  /* update the read buf if overlaps write buf */
 static Es_status es_file_commit();
--- lib/libxview/sel/sel_agent.c.old	2006-02-18 23:17:33.000000000 -0500
+++ lib/libxview/sel/sel_agent.c	2006-02-18 23:18:23.000000000 -0500
@@ -314,9 +314,7 @@
     struct stat     stat_buf;
     int             count, size;
     char           *destp;
-#ifndef __CYGWIN__
-    extern long     lseek();
-#endif
+    extern off_t     lseek();
     if (fstat(fd, &stat_buf) != 0) {
 	perror(XV_MSG("Agent couldn't reply about a file"));
 	return SELN_FAILED;
